{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Extracting Unstructured Data from PDFs"
      ],
      "metadata": {
        "id": "Fsga60c2ddSY"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 1. Introduction\n",
        "\n",
        "A PDF document is human-readable and may *look* structured but does not actually have a logical internal structure containing paragraphs, sentences or even words. It is only aware of the location of individual characters.  Programmatically extracting text in a meaningful way is therefore challenging.\n",
        "\n",
        "Each page consists of a set of layout objects and instructions for their positioning that collectively describe its appearance. Page layout analysis will be performed to determine which objects exist, and reconstruct and visualise the document structure before extracting text.\n",
        "\n",
        "There are basically three types of PDF:\n",
        "\n",
        "*   Real or digitally created PDFs, which are the type used for this project as specified in the [arXiv submission guidelines](https://info.arxiv.org/help/submit_pdf.html#pdf-must-be-machine-readable).\n",
        "*   Scanned or \"image-only\" PDFs without an underlying text layer.\n",
        "*   Searchable or OCRed PDFs with an image layer and a searchable text layer.\n",
        "\n"
      ],
      "metadata": {
        "id": "QVvJ11Qedm7T"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 2. Install/import libraries"
      ],
      "metadata": {
        "id": "ez32vBLWdqNQ"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "-WUK7USv7RUh"
      },
      "outputs": [],
      "source": [
        "!pip install pdfminer.six pdfplumber unidecode"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "9PFGZ44V-PYg"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "import matplotlib.pyplot as plt\n",
        "import requests\n",
        "import re\n",
        "import pickle\n",
        "import time\n",
        "import pdfplumber\n",
        "import pdfminer\n",
        "import unicodedata\n",
        "\n",
        "from matplotlib import patches\n",
        "%matplotlib inline\n",
        "\n",
        "from io import BytesIO\n",
        "from io import StringIO\n",
        "\n",
        "from pdfminer.pdfparser import PDFParser\n",
        "from pdfminer.pdfinterp import PDFResourceManager, PDFPageInterpreter\n",
        "from pdfminer.layout import *\n",
        "from pdfminer.pdfpage import PDFPage\n",
        "from pdfminer.converter import PDFPageAggregator\n",
        "from pdfminer.high_level import extract_pages\n",
        "\n",
        "from collections  import Counter\n",
        "from typing import Iterable, Any\n",
        "from itertools import chain\n",
        "from unidecode import unidecode\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "YWbQgrWmFJLa"
      },
      "source": [
        "## 3. Import data\n",
        "\n",
        "Import DataFrame with metadata for arXiv PDF with missing full text."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "nM24zxvV-5Ca"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-15_pmc_arxiv_full_text_merged_new.pickle', 'rb') as f:\n",
        "    pmc_arxiv_full_text_merged_new = pickle.load(f)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 229
        },
        "id": "xVIcxVvLDY6W",
        "outputId": "c5a0b6e9-7617-48cf-f7bf-8c3892d386c6"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         article_id   published     revised  \\\n",
              "11566  2007.09186v3  2020-07-17  2020-10-07   \n",
              "\n",
              "                                                   title journal  \\\n",
              "11566  AWS CORD-19 Search: A Neural Search Engine for...    None   \n",
              "\n",
              "                                                 authors   doi  \\\n",
              "11566  Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...  None   \n",
              "\n",
              "                                        pdf_url  text  \n",
              "11566  http://export.arxiv.org/pdf/2007.09186v3  None  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1b48d5c6-9622-444a-949f-6fb2207caaaf\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>11566</th>\n",
              "      <td>2007.09186v3</td>\n",
              "      <td>2020-07-17</td>\n",
              "      <td>2020-10-07</td>\n",
              "      <td>AWS CORD-19 Search: A Neural Search Engine for...</td>\n",
              "      <td>None</td>\n",
              "      <td>Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2007.09186v3</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1b48d5c6-9622-444a-949f-6fb2207caaaf')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-1b48d5c6-9622-444a-949f-6fb2207caaaf button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-1b48d5c6-9622-444a-949f-6fb2207caaaf');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 4
        }
      ],
      "source": [
        "pmc_arxiv_missing_text = pmc_arxiv_full_text_merged_new.loc[pmc_arxiv_full_text_merged_new['text'].isna()]\n",
        "pmc_arxiv_missing_text"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_missing_text.pdf_url"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "K_yru3d-hViW",
        "outputId": "12fc3867-2051-4ead-9920-39ee0f9b2861"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "11566    http://export.arxiv.org/pdf/2007.09186v3\n",
              "Name: pdf_url, dtype: object"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WMTYN1Lpgjiq"
      },
      "source": [
        "## 4. Page layout analysis\n",
        "\n",
        "We will use the page layout function in the [pdfminer](https://github.com/pdfminer/pdfminer.six) library to output an ordered hierarchy of layout objects for each page in the document in order to visualise and understand its structure for the purpose of text extraction.  \n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 4.1 Extract page layouts\n",
        "\n",
        "Layout analysis function to extract `LTPage` objects from PDF. An `LTPage` object represents an entire page and consists of child objects.\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "M68eXeAzlaWO"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "pmAdSuiOKu2X"
      },
      "outputs": [],
      "source": [
        "def extract_page_layouts(pdf_memory_file):\n",
        "    \"\"\"Extract LTPage objects from PDF\"\"\"\n",
        "\n",
        "    ext_layouts = []\n",
        "\n",
        "    # Create a PDF resource manager object that stores shared resources such as fonts or images\n",
        "    rsrcmgr = PDFResourceManager()\n",
        "    # Set parameters for analysis\n",
        "    laparams = LAParams()\n",
        "    # Create PDF page aggregator object\n",
        "    device = PDFPageAggregator(rsrcmgr, laparams=laparams)\n",
        "    # Create PDF interpreter object to process the page contents\n",
        "    interpreter = PDFPageInterpreter(rsrcmgr, device)\n",
        "\n",
        "    for page in PDFPage.get_pages(pdf_memory_file):\n",
        "        interpreter.process_page(page)\n",
        "        # Get the LTPage object for the page\n",
        "        layout = device.get_result()\n",
        "        # Append LTPage object to list\n",
        "        ext_layouts.append(layout)\n",
        "\n",
        "    return ext_layouts"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JAS2Dj1NEoF7"
      },
      "source": [
        "### 4.2 Output page layouts\n",
        "\n",
        "The layout objects and structural information are represented as a single self-contained sequence of bytes. Write the binary data to a memory buffer, extract page layouts and return objects as a list."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "wuirvF5hHF6A"
      },
      "outputs": [],
      "source": [
        "def output_page_layouts(pdf_url: str):\n",
        "    \"\"\"\n",
        "    Write PDF binary data to BytesIO buffer\n",
        "    and return list of LTPage objects\n",
        "    \"\"\"\n",
        "    out_layouts = []\n",
        "\n",
        "    if isinstance(pdf_url, str):\n",
        "        url = pdf_url\n",
        "        response = requests.get(url, allow_redirects=True)\n",
        "        if response.status_code != 200:\n",
        "          raise Exception(response.text)\n",
        "        pdf_memory_file = BytesIO()\n",
        "        pdf_memory_file.write(response.content)\n",
        "        layout = extract_page_layouts(pdf_memory_file)\n",
        "        out_layouts.append(layout)\n",
        "        time.sleep(3)\n",
        "    else:\n",
        "      raise TypeError(\"Need to supply a url\")\n",
        "\n",
        "    return out_layouts"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "An `LTPage` object is returned for each page in the PDF as a list."
      ],
      "metadata": {
        "id": "eOLE_6KUnFwo"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# LTPage represents entire page\n",
        "\n",
        "page_layouts = output_page_layouts(pmc_arxiv_missing_text.pdf_url[11566])\n",
        "page_layouts"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KQwaEHJHhAIY",
        "outputId": "b5e67e63-c05b-4602-ce95-570acd11aade"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[<LTPage(1) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(2) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(3) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(4) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(5) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(6) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(7) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(8) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              "  <LTPage(9) 0.000,0.000,595.276,841.890 rotate=0>]]"
            ]
          },
          "metadata": {},
          "execution_count": 7
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Here we just have a single PDF."
      ],
      "metadata": {
        "id": "fdozvVQ5Ypsv"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dnktfTkLLxn6",
        "outputId": "5f858988-0e67-4675-bf63-2f74aaa03edf"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ],
      "source": [
        "len(page_layouts)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Iterate over `LTPage` objects to get article page count."
      ],
      "metadata": {
        "id": "IOO5LWBXY7Op"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mQywiaJJL0FO",
        "outputId": "f9cada7e-0f12-4355-f4ef-55a651718b18"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "9\n"
          ]
        }
      ],
      "source": [
        "for num_pages in page_layouts:   # number of pages in each article\n",
        "  print(len(num_pages))"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cvqRRO6V_FCl"
      },
      "source": [
        "### 4.3 Get list of child objects\n",
        "\n",
        "Function to return list of child objects for each `LTPage` object.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def get_page_layouts(article):\n",
        "\n",
        "    page_layout_objs = [[obj for obj in set(type(o) for o in page)] for page in article]\n",
        "\n",
        "    return page_layout_objs"
      ],
      "metadata": {
        "id": "NvVsjp0XjNL-"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Call function on list of `LTPage` objects."
      ],
      "metadata": {
        "id": "gwdOAb0zaGbK"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "all_pages = list(map(get_page_layouts, page_layouts))\n",
        "all_pages"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "I8pyDeHXjSR9",
        "outputId": "033b7bd2-b732-4e43-f74d-fa406212b848"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[[pdfminer.layout.LTTextBoxHorizontal,\n",
              "   pdfminer.layout.LTLine,\n",
              "   pdfminer.layout.LTTextLineHorizontal],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal, pdfminer.layout.LTLine],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal,\n",
              "   pdfminer.layout.LTLine,\n",
              "   pdfminer.layout.LTFigure],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal,\n",
              "   pdfminer.layout.LTLine,\n",
              "   pdfminer.layout.LTFigure],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal, pdfminer.layout.LTLine],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal, pdfminer.layout.LTLine],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal, pdfminer.layout.LTLine],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal, pdfminer.layout.LTLine],\n",
              "  [pdfminer.layout.LTTextBoxHorizontal]]]"
            ]
          },
          "metadata": {},
          "execution_count": 49
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "P_KPHe4bUbcF"
      },
      "source": [
        "### 4.4 Get list of unique child objects\n",
        "\n",
        "Flatten nested list of child objects of `LTPage` objects to return list of unique child objects."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def get_unique_objs(article):\n",
        "\n",
        "    unique_LT_objs = list(set(obj for page in article for obj in page))\n",
        "\n",
        "    return unique_LT_objs"
      ],
      "metadata": {
        "id": "8W325AS1nSpE"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "unique_objects = list(map(get_unique_objs, all_pages))\n",
        "unique_objects"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aXw37dWnn1vx",
        "outputId": "61a3fec1-f061-4a2c-df5d-03ea7d674343"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[pdfminer.layout.LTTextBoxHorizontal,\n",
              "  pdfminer.layout.LTLine,\n",
              "  pdfminer.layout.LTTextLineHorizontal,\n",
              "  pdfminer.layout.LTFigure]]"
            ]
          },
          "metadata": {},
          "execution_count": 61
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The child objects here are `LLTextBox` objects which represent groups of text chunks and can contain a list of `LTTextLine` objects, `LTLine` objects representing a single straight line used for separating text or figures, and `LTFigure` objects representing an area used by PDF Form objects to present figures or pictures by embedding yet another PDF document within a page."
      ],
      "metadata": {
        "id": "CTAllI-jdduE"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dvLvbFBBmvTD"
      },
      "source": [
        "### 4.5 Get count of unique child objects"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def get_num_unique_objs(article):\n",
        "\n",
        "    num_unique_LT_objs = Counter([obj for page in article for obj in page]).most_common()\n",
        "\n",
        "    return num_unique_LT_objs"
      ],
      "metadata": {
        "id": "3gB0pBIlqU6D"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "The most common object is `LTTextBoxHorizontal` which occurs on each page, then eight pages with `LTLine` which represents a single straight line and can be used for separating text or figures, two pages with `LTFigure`, and one page with `LTTextLineHorizontal`."
      ],
      "metadata": {
        "id": "2bYPSurglW9T"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_objs = list(map(get_num_unique_objs, all_pages))\n",
        "num_unique_objs"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fYCy53O7qWjz",
        "outputId": "cce16f25-7efb-45f7-df30-0f34add21cd9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[(pdfminer.layout.LTTextBoxHorizontal, 9),\n",
              "  (pdfminer.layout.LTLine, 8),\n",
              "  (pdfminer.layout.LTFigure, 2),\n",
              "  (pdfminer.layout.LTTextLineHorizontal, 1)]]"
            ]
          },
          "metadata": {},
          "execution_count": 63
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "GX7iy976tDDk"
      },
      "source": [
        "### 4.6 Extract all layout objects in the hierarchy\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "LT_ELEMENTS = [pdfminer.layout.LTTextBox,\n",
        "            pdfminer.layout.LTTextBoxHorizontal,\n",
        "            pdfminer.layout.LTTextLine,\n",
        "            pdfminer.layout.LTTextLineHorizontal,\n",
        "            pdfminer.layout.LTLine,\n",
        "            pdfminer.layout.LTFigure]\n",
        "\n",
        "\n",
        "def extract_LT_elems(article):\n",
        "    for page in article:\n",
        "      for elements in page:\n",
        "        for element in elements:\n",
        "          if any(isinstance(element, i) for i in LT_ELEMENTS):\n",
        "            return [[elem for elem in elems] for elems in page]"
      ],
      "metadata": {
        "id": "jIHrc62stNLy"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Layout objects containing bounding box coordinates and contents for all pages.\n"
      ],
      "metadata": {
        "id": "q_lzKEL8wX63"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "LT_elem_text = extract_LT_elems(page_layouts)\n",
        "LT_elem_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Y8ilRDdNtQq4",
        "outputId": "6d296b9d-d1ae-4e58-fd29-0b1da09333dd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[<LTTextBoxHorizontal(0) 103.371,746.071,493.659,776.357 'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 77.083,639.527,523.454,734.427 'Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\\nBen Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\\nRamesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\\nAmazon Web Services AI\\nparmib, liuall, arumae, nimpourd, suyogd, snivelyb,\\nmonamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\\nzhiheng, bxiang, tahak @amazon.com\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 158.891,605.891,203.376,617.847 'Abstract\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 88.261,322.925,274.910,595.902 'Coronavirus disease (COVID-19) has been de-\\nclared as a pandemic by WHO with thousands\\nof cases being reported each day. Numerous\\nscientiﬁc articles are being published on the\\ndisease raising the need for a service which\\ncan organize, and query them in a reliable fash-\\nion. To support this cause we present AWS\\nCORD-19 Search (ACS), a public, COVID-19\\nspeciﬁc, neural search engine that is powered\\nby several machine learning systems to sup-\\nport natural language based searches. ACS\\nwith capabilities such as document ranking,\\npassage ranking, question answering and topic\\nclassiﬁcation provides a scalable solution to\\nCOVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high\\npriority scientiﬁc questions. We present a\\nquantitative evaluation and qualitative analysis\\nof the system against other leading COVID-\\n19 search platforms. ACS is top performing\\nacross these systems yielding quality results\\nwhich we detail with relevant examples in this\\nwork.\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 72.000,300.286,77.978,312.242 '1\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 89.933,300.286,154.814,312.242 'Introduction\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 71.182,76.568,292.082,290.715 'With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 307.276,389.247,527.359,616.943 'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 306.883,101.737,527.356,385.118 'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 319.928,77.606,425.632,87.419 '1https://cord19.aws/\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 16.340,526.960,36.340,566.960 '0\\n2\\n0\\n2\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 16.340,478.080,36.340,521.960 't\\nc\\nO\\n7\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 16.340,461.420,36.340,468.080 ']\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 16.340,448.080,36.340,461.420 'R\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 16.340,413.100,36.340,448.080 'I\\n.\\ns\\nc\\n[\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 16.340,237.000,36.340,403.100 '3\\nv\\n6\\n8\\n1\\n9\\n0\\n.\\n7\\n0\\n0\\n2\\n:\\nv\\ni\\nX\\nr\\na\\n'>,\n",
              "  <LTLine 307.276,89.599,367.052,89.599>,\n",
              "  <LTTextLineHorizontal 16.340,403.100,36.340,408.100 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,408.100,36.340,413.100 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,468.080,36.340,473.080 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,473.080,36.340,478.080 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,488.080,36.340,493.080 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,521.960,36.340,526.960 ' \\n'>],\n",
              " [<LTTextBoxHorizontal(0) 82.729,766.636,101.138,774.003 'Query\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 181.035,766.636,201.485,774.003 'Article\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 279.865,766.636,308.101,774.003 'Response\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 81.749,742.699,166.567,758.658 '“Are IL-6 inhibitors key to\\nCOVID-19?”\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 180.744,725.510,266.356,758.710 'SARS-CoV-2 and COVID-\\n19:\\nis interleukin-6 (IL-6)\\nthe ’culprit lesion’ of ARDS\\nonset?\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 278.886,716.915,448.769,758.658 '“...monoclonal antibody against IL-6, is being tested in a\\nclinical trial against COVID-19 (Sarilimumab COVID-\\n19). Another drug that showed potential inhibition of IL-\\n6 related JAK/STAT pathway is glatiramer acetate which\\nshowed potential to downregulate both IL-17 and IL-6”\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 462.055,766.636,486.423,774.003 'Topic(s)\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 462.280,751.292,517.470,758.658 'Clinical Treatment\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 81.749,682.539,167.790,707.093 '“When is the salivary vi-\\nral load highest for COVID-\\n19?”\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 181.297,673.944,266.358,707.093 'Elective, Non-urgent Pro-\\ncedures\\nAesthetic\\nand\\nSurgery in the Wake of\\nSARS–COVID-19 [...]\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 278.886,673.944,448.770,707.144 '“Patients with COVID-19 have demonstrated high viral\\nloads in the upper respiratory tract soon after their infec-\\ntion, with the highest load assumed to be the day before\\nsymptoms appear.”\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 462.280,699.726,517.470,707.093 'Clinical Treatment\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 81.749,648.161,167.790,664.122 '“Is convalescent plasma ther-\\napy a precursor to vaccine?”\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 181.297,648.161,236.745,664.122 'COVID-19\\nplasma transfusion\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 227.566,656.755,265.135,664.122 'convalescent\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 278.886,639.566,448.766,664.122 '“...a passive immunotherapy, has been used as a possi-\\nble therapeutic option when no proven speciﬁc vaccine\\nor drug is available for emerging infections. ”\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 462.052,639.566,519.098,664.122 'Clinical Treatment,\\nImmunology, Lab\\nTrials\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 72.000,592.787,525.547,626.661 'Figure 1: Sample natural language queries and responses using AWS CORD-19 Search. The response ﬁeld is\\ntaken directly from the top result of this service. Also provided are the article titles where the answer is taken from\\nas well as selected topics for the response.\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 72.000,368.424,292.083,569.273 'viral load. We provide a further advantage from\\nan organizational perspective using topics with do-\\nmain speciﬁc relations. The user can select clinical-\\ntreatments to reﬁne search articles to understand\\nthat “...the highest load [is] assumed to be the day\\nbefore symptoms appear.” Similarly if commis-\\nsioned to understand convalescent plasma therapy\\nand selecting the three topics as shown in the last\\nexample the system identiﬁes the most relevant ar-\\nticle and answers the question in a highlight. There\\nare a few search systems have been developed to\\nsupport COVID-19 search leveraging CORD-19\\ncorpus. We evaluate the ACS against two other\\nsearch engines, and ACS is proved to be a top-\\nperforming system.\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 71.727,282.940,292.075,361.595 'In the following sections we will discuss the\\nvarious individual Amazon Web Services (AWS)\\nproducts that allow ACS to provide this function-\\nality, highlighting the value to scientists who can\\nquickly query, validate their research, and advance\\ntheir investigations.\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 72.000,249.561,178.342,261.517 '2 System Overview\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 71.607,76.568,292.172,236.518 'AWS CORD-19 Search (ACS) relies on a deep\\nsemantic search model to return a ranked list of\\nrelevant documents. This system performs docu-\\nment ranking, passage ranking, question answer-\\ning and FAQ matching. Additionally, it leverages\\nknowledge graphs and topic modeling to enrich the\\nbiomedical data and boost searching performance.\\nThis provides a scalable solution to COVID-19 re-\\nsearchers and policy makers. In this section we\\npresent the overall architecture of the system and a\\ncloser look at several individual components. Fig-\\nure 2 provides an overview of this architecture.\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 306.883,418.693,527.452,569.098 '2.1 Amazon Kendra\\nAmazon Kendra2 is a semantic search and question\\nanswering service provided by AWS for enterprise\\ncustomers. Kendra allows its customers to power\\nnatural language based searching across their own\\ndata. As response to the worldwide COVID-19\\npandemic, Kendra has also been tooled to support\\nCOVID-19 related searching and question answer-\\ning using the document corpus from CORD-19.\\nThe end-to-end Kendra system consists of several\\ncomponents.\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 307.276,323.849,527.356,416.053 'To complement the extracted answers, Kendra\\nuses a deep learning based semantic search model\\nto return a ranked list of relevant documents. Ama-\\nzon Kendra’s ability to understand natural language\\nquestions is at the core of its search engine return-\\ning the most relevant passage and related docu-\\nments\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 320.294,220.508,527.359,312.789 '• Document ranking (DR) - Like any tradi-\\ntional search engine, Kendra returns a ranked\\nlist of relevant documents based on the user’s\\nquery to fulﬁll their information needs. A\\ndeep semantic search model is used to under-\\nstand natural language questions in addition\\nto the keyword search.\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 320.294,144.329,527.360,209.511 '• Passage ranking (PR) & Question Answer-\\ning (QA) - Kendra ranks the passages and\\ntries to extract the answer from the top rele-\\nvant passages with a deep reading comprehen-\\nsion model.\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 320.294,95.249,527.351,133.332 '• FAQ Matching (FAQM) - If there exists Fre-\\nquently Answered Questions and their corre-\\nsponding answers, Kendra will automatically\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 319.928,77.606,484.810,87.419 '2https://aws.amazon.com/kendra/\\n'>,\n",
              "  <LTLine 75.362,778.544,525.543,778.544>,\n",
              "  <LTLine 75.362,762.998,525.543,762.998>,\n",
              "  <LTLine 75.362,635.726,525.543,635.726>,\n",
              "  <LTLine 307.276,89.599,367.052,89.599>],\n",
              " [<LTTextBoxHorizontal(0) 237.044,540.631,360.501,550.594 'Figure 2: System Architecture.\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 93.818,492.407,290.280,516.866 'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 72.000,325.745,292.082,472.146 'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 72.000,297.540,197.858,308.449 '2.2 Comprehend Medical\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 71.575,141.116,292.075,289.172 'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 307.276,506.033,470.563,516.942 '2.3 COVID-19 Knowledge Graph\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 307.276,148.089,527.359,497.728 'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 84.653,118.369,217.255,128.182 '3https://docs.aws.amazon.\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 72.000,98.443,245.150,117.372 'com/comprehend/latest/dg/\\nhow-document-classification.html\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 84.653,87.569,271.054,97.382 '4https://aws.amazon.com/comprehend/\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 72.000,77.606,115.039,86.572 'medical/\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 307.276,122.881,394.156,133.790 '2.4 Topic Models\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 306.937,76.568,525.549,114.576 'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n'>,\n",
              "  <LTFigure(Im1) 72.000,559.122,526.860,779.082 matrix=[454.86,0.00,0.00,219.96, (72.00,559.12)]>,\n",
              "  <LTLine 72.000,130.362,131.776,130.362>],\n",
              " [<LTTextBoxHorizontal(0) 186.115,447.945,411.429,457.908 'Figure 3: Visualization of COVID-19 Knowledge graph.\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 71.607,304.876,292.075,424.179 'for topic modeling is Latent Dirichlet Allocation\\n(LDA) (Blei et al., 2003), a generative model which\\ngroups documents together by observed content,\\noften giving each document a mixture of topics it\\nbelongs to. An extension of this work termed Z-\\nlabel LDA (Andrzejewski and Zhu, 2009) utilizes\\npriors to allow the model to force certain topics\\nwhich the users have manually curated, or wish to\\nsee clustered together.\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 71.607,105.493,292.083,295.072 '2.4.1 Generating Topics\\nFor the purposes of this work we experimented\\nwith 5, 10, and 20 topic models 5. The outputs\\nof each clustering size were manually inspected\\nand topic labels were provided by us when inspect-\\ning the top ten terms for each cluster. The ﬁnal\\ngranularity of the topic models was chosen by man-\\nually deleting and merging topics from the 20 topic\\nmodel. In general we were able to clearly extract\\ngroups which centered around important topics in-\\ncluding virology, proteomics, epidemiology, and\\ncellular biology to name a few. However when\\nfaced with 20 topics the less populated ones tended\\nto be noisy, and captured peripheral information\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 307.003,182.933,527.359,424.179 'present in the input, such as language (e.g. Spanish\\nand French) or provide redundancies with existing\\ntopics (e.g. two topics for Inﬂuenza). As a con-\\ntrol we ran a publicly available implementation of\\nZ-label LDA 6 with no priors which yields topics\\nclose to those extracted using Comprehend Topic\\nModeling. Although similar we observed better\\ndeﬁnition in certain groups (such as pulmonary\\ndiseases, and policy/industry), and decided to use\\nthis as the curation entry-point. Our goal was to\\nlimit these topics to ten, and compile them in ad-\\nvance as much as possible. With the help of medi-\\ncal professionals we eliminated and combined top-\\nics to form the following: Vaccines/immunology,\\nGenomics, Public health Policies, Epidemiology,\\nClinical Treatment, Virology, Inﬂuenza, Healthcare\\nIndustry, Pulmonary Infections, and Lab Trials (hu-\\nman).\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 307.276,105.493,527.349,173.130 '2.4.2 Multi-Label Classiﬁcation\\nHaving to manually feed a topic model and re-train\\non the entire corpus once new data becomes avail-\\nable is largely inefﬁcient. We therefore used the\\ntopic model labels to train a multi-label classiﬁer\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 84.653,86.950,290.268,97.382 '5These models were trained using CORD-19 data available\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 72.000,76.987,148.708,85.954 'as of April 6th, 2020.\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 319.928,85.850,511.709,97.382 '6http://pages.cs.wisc.edu/˜andrzeje/\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 307.276,77.606,414.873,86.572 'research/zl_lda.html\\n'>,\n",
              "  <LTFigure(Im2) 97.924,466.435,499.624,779.085 matrix=[401.70,0.00,0.00,312.65, (97.92,466.44)]>,\n",
              "  <LTLine 72.000,99.562,131.776,99.562>,\n",
              "  <LTLine 307.276,99.562,367.052,99.562>],\n",
              " [<LTTextBoxHorizontal(0) 114.189,763.315,177.505,774.224 'Search Engine\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 194.622,763.315,216.189,774.224 'P@1\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 236.578,763.315,258.145,774.224 'P@5\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 275.807,763.315,302.829,774.224 'P@10\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 317.764,763.315,429.298,774.224 'P@20 R@10 R@20\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 442.138,763.315,484.302,774.224 'ndcg@20\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 260.421,745.841,337.123,756.750 'Keyword Queries\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 127.525,713.265,164.169,737.723 'ACS\\nCovidex\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 113.245,699.716,220.405,737.723 '0.5250\\n0.3421\\nCOVID-19 RE 0.5750\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 232.359,699.792,262.361,737.799 '0.5650\\n0.2316\\n0.5650\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 274.314,699.716,304.318,737.799 '0.5325\\n0.2079\\n0.4775\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 316.270,699.716,346.274,737.799 '0.4775\\n0.1658\\n0.4412\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 358.220,699.716,388.228,737.799 '0.0260\\n0.0109\\n0.0236\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 400.176,699.716,430.185,737.799 '0.0459\\n0.0173\\n0.0429\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 448.216,699.716,478.220,737.799 '0.4380\\n0.1633\\n0.4022\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 236.361,682.241,361.183,693.150 'Natural Language Questions\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 127.525,649.665,164.169,674.123 'ACS\\nCovidex\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 113.245,636.116,220.405,674.199 '0.8750\\n0.4750\\nCOVID-19 RE 0.6000\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 232.360,636.116,262.361,674.199 '0.7000\\n0.4800\\n0.5300\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 274.317,636.116,304.318,674.199 '0.6400\\n0.4225\\n0.4925\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 316.273,636.116,346.274,674.199 '0.5550\\n0.3625\\n0.4600\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 358.220,636.116,388.230,674.199 '0.0345\\n0.0204\\n0.0267\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 400.175,636.116,430.176,674.199 '0.0582\\n0.0356\\n0.0474\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 448.219,636.116,478.220,674.199 '0.5357\\n0.3229\\n0.4133\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 192.835,613.165,404.401,623.128 'Table 1: Evaluation results on TREC-COVID dataset\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 71.640,497.195,290.269,589.399 '(Read et al., 2011). To evaluate the performance\\nof this model we calculate the average F1 across\\ntest samples by calculating the set overlap between\\nour gold standard (topic model) and system labels\\n(multi-label classiﬁcation). This held-out test set\\ncontains 20% of the CORD-19 data available at the\\ntime.\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 72.000,429.449,290.273,494.555 'Using this metric our trained model achieved an\\naverage F1 of 91.92, with on average 2.37 labels\\nper document. Fewer than 1% of the documents\\nin the test set received no label, using 0.5 as the\\nconﬁdence threshold.\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 72.000,406.223,145.620,418.179 '3 Evaluation\\n'>,\n",
              "  <LTTextBoxHorizontal(28) 71.182,196.446,292.177,397.044 'ACS supports document ranking (DR), passage\\nranking (PR) and question answering (QA) on the\\ntop suggested results. In this section we evaluate\\nthe overall performance of ACS with publicly avail-\\nable datasets. The TREC-COVID track (Voorhees\\net al., 2020) built DR test collections that contain\\ndocument relevance ground truth and are used to\\nassess DR only. For PR and QA, we manually anno-\\ntate questions collected from CovidQA (Tang et al.,\\n2020), the ﬁrst open sourced Covid-19 QA dataset.\\nThe ACS results are compared against Covidex7\\nand Google Covid-19 Research Explorer8 (COVID-\\n19 RE), which are also COVID-19 public search\\nengines that facilitate DR and answer highlighting\\nin passage.\\n'>,\n",
              "  <LTTextBoxHorizontal(29) 72.000,115.151,292.075,193.806 'The TREC-COVID challenge track contains 40\\ntopic sets along with their document relevance\\njudgement. The topic sets are written by its or-\\nganizers with biomedical training, and motivated\\nby search submitted to the National Library of\\nMedicine and social media. Each topic consists\\n'>,\n",
              "  <LTTextBoxHorizontal(30) 84.653,87.569,272.049,108.256 '7https://covidex.ai/\\n8https://covid19-research-explorer.\\n'>,\n",
              "  <LTTextBoxHorizontal(31) 72.000,77.606,136.558,86.572 'appspot.com/\\n'>,\n",
              "  <LTTextBoxHorizontal(32) 307.276,361.703,527.359,589.399 'of three ﬁelds with different levels of granularity, a\\nkeyword-based query (KQ), a more precise natural\\nlanguage question (NQ), and a longer descriptive\\nnarrative. The DR assessments are performed fol-\\nlowing the TREC pooling mechanism. The par-\\nticipants submitted ranked lists of documents for\\neach topic set, based on which a depth of 10 to 20\\ndocuments are pooled and combined as a collection\\nof (q, D) pairs. The pairs are then assessed by an-\\nnotators with in-domain expertise. There are three\\nrounds of judgement available that corresponding\\nto three different versions of CORD-19 corpus. To\\nensure sufﬁcient coverage of annotation, we aggre-\\ngate all rounds results into 33,064 (q, D) relevance\\njudgements. Since the document id may change\\nacross versions, we map ids from each round to the\\nMay 19 release of CORD-19 corpus.\\n'>,\n",
              "  <LTTextBoxHorizontal(33) 306.883,253.109,527.355,358.862 'To collect DR results from the three systems, we\\ncrawled the top 50 articles by querying the engines\\nwith KQ and its NQ variation from the topic sets on\\nJune 15, 2020. The crawled data, which are char-\\nacterized by the article title and link, are mapped\\nto May 19 CORD-19 corpus. Note that articles\\nthat cannot be found in the corpus are removed to\\nensure fair comparison.\\n'>,\n",
              "  <LTTextBoxHorizontal(34) 306.883,103.867,527.356,250.268 'We use the standard DR metrics in our evalua-\\ntion, namely, the precision and recall at k (P@k,\\nR@k) and normalized discounted cumulative gain\\nin the top k documents (NDCG@k). Note that we\\nevaluate with k up to 20 since TREC-COVID has\\na pooling depth with 20 at most. Table 1 presents\\nthe DR performance of the engines over KQ and\\nNQ, respectively. ACS performs consistently better\\nthan the other engines on NQ and mostly on KQ,\\nand all engines perform better on NQ comparing\\nwith KQ.\\n'>,\n",
              "  <LTTextBoxHorizontal(35) 307.276,76.568,525.549,101.026 'In addition, we are able to evaluate how robust\\neach system is against query variation from KQ\\n'>,\n",
              "  <LTLine 107.268,778.645,490.276,778.645>,\n",
              "  <LTLine 107.268,761.170,490.276,761.170>,\n",
              "  <LTLine 107.268,741.980,490.276,741.980>,\n",
              "  <LTLine 107.268,697.571,490.276,697.571>,\n",
              "  <LTLine 107.268,678.381,490.276,678.381>,\n",
              "  <LTLine 107.268,632.092,490.276,632.092>,\n",
              "  <LTLine 72.000,110.353,131.776,110.353>],\n",
              " [<LTTextBoxHorizontal(0) 173.032,756.541,236.348,767.450 'Search Engine\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 251.554,749.766,296.543,774.224 'Top3\\nEM F1\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 313.646,749.766,358.635,774.224 'Top10\\nEM F1\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 375.738,749.766,420.727,774.224 'Top30\\nEM F1\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 173.032,703.642,269.281,741.725 '11.7\\nACS\\nCovidex\\n0.90\\nCOVID-19 RE 10.0\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 281.237,703.642,300.328,741.725 '35.6\\n18.9\\n31.8\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 312.282,703.642,331.373,741.725 '16.0\\n2.10\\n13.5\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 343.329,703.642,362.420,741.725 '42.2\\n24.2\\n36.9\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 374.374,703.642,393.465,741.725 '26.0\\n3.60\\n18.2\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 405.421,703.642,424.512,741.725 '50.4\\n27.6\\n43.8\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 154.912,680.691,442.323,690.654 'Table 2: Top results variation from KQ to NQ on TREC-COVID dataset\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 140.305,644.760,203.621,655.669 'Search Engine\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 221.679,637.986,243.246,648.895 'P@1\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 140.305,591.861,247.463,629.944 '0.4074\\nACS\\n0.4074\\nCovidex\\nCOVID-19 RE 0.4074\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 263.635,637.986,285.203,662.444 'PR\\nP@2\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 259.418,591.861,289.419,629.944 '0.5370\\n0.4444\\n0.3704\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 305.592,637.986,327.159,648.895 'P@3\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 347.545,637.986,369.112,648.895 'P@1\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 301.374,591.861,331.376,629.944 '0.4938\\n0.4074\\n0.3580\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 343.328,591.861,373.328,629.944 '0.3333\\n0.1481\\n0.2963\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 389.501,637.986,411.069,662.444 'QA\\nP@2\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 385.284,591.861,415.284,629.944 '0.2593\\n0.1852\\n0.2407\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 431.458,637.986,453.025,648.895 'P@3\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 427.239,591.937,457.241,629.868 '0.2099\\n0.2099\\n0.2593\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 109.892,568.910,487.345,578.873 'Table 3: Passage ranking (PR) and question answering (QA) performance on CovidQA dataset\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 203.824,478.151,238.804,492.777 'q , KQk\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 71.607,317.448,292.083,545.144 'to NQ. Ideally, the results shall remain unchanged\\nwith query variation that requests the same infor-\\nmation. We deﬁne exact match (EM) and F1 score\\namong top k results to evaluate the robustness. Let\\nQ be the topic sets, and N Qk\\nq denote the\\ntop k searching results of natural language ques-\\ntion and keyword query corresponding to the same\\ntopic q ∈ Q, respectively. As shown in Eq.(1), we\\ntake the top k results KQk\\nq as ground truth, and\\ncompute the average exact match and F1 score of\\nthe top k results N Qk\\nq , and then average over all\\nqueries. The article title string is used as com-\\nparison key, and EM and F1 are standard that the\\nmaximum is taken over all the ground truth arti-\\ncles. Table 2 demonstrates that ACS has the best\\ncapability to provide consistent results with query\\nvariation.\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 109.069,288.073,190.977,298.982 'EM (N Q, KQ, k)\\n'>,\n",
              "  <LTTextBoxHorizontal(28) 100.584,259.761,109.069,270.670 '=\\n'>,\n",
              "  <LTTextBoxHorizontal(29) 110.264,252.278,124.950,278.050 '1\\n|Q|\\n'>,\n",
              "  <LTTextBoxHorizontal(30) 147.349,252.278,153.088,278.050 '1\\nk\\n'>,\n",
              "  <LTTextBoxHorizontal(31) 128.270,265.696,144.027,276.605 '(cid:88)\\n'>,\n",
              "  <LTTextBoxHorizontal(32) 127.963,247.066,144.336,255.036 'q∈Q\\n'>,\n",
              "  <LTTextBoxHorizontal(33) 156.386,265.696,172.143,281.937 'k\\n(cid:88)\\n'>,\n",
              "  <LTTextBoxHorizontal(34) 157.413,247.244,171.117,255.214 'i=1\\n'>,\n",
              "  <LTTextBoxHorizontal(35) 173.962,259.761,215.850,272.805 'em(N Qk\\n'>,\n",
              "  <LTTextBoxHorizontal(36) 211.441,257.634,261.684,272.805 'q (i), KQk\\nq )\\n'>,\n",
              "  <LTTextBoxHorizontal(37) 277.548,258.972,290.268,269.881 '(1)\\n'>,\n",
              "  <LTTextBoxHorizontal(38) 71.607,76.568,292.075,222.969 'Next, we evaluate the performance of PR and QA\\nwith CovidQA dataset. CovidQA contains 27 ques-\\ntions and their answers from 124 question-article\\npairs, which are selected by in-domain volunteers\\nas the most promising COVID-19 literature. Since\\nthe annotation does not possess sufﬁcient answer\\ncoverage over the entire corpus, we leverage our\\ninternal annotation resources to make PR and QA\\njudgement. More explicitly, we crawled the top 3\\nresults characterized by the article, displayed pas-\\nsage and the highlighted text snippet on June 15,\\n'>,\n",
              "  <LTTextBoxHorizontal(39) 307.276,425.841,527.359,545.144 '2020 from ACS, Covidex and COVID-19 RE, re-\\nspectively. After that, the annotators assess PR\\nand QA in terms of whether the displayed passage\\ncontains relevant information and whether the high-\\nlighted text snippet answers the given question. To\\navoid bias, the crawled results are combined and\\nshufﬂed randomly, therefore, the annotators do not\\nhave access to the source engine and the rank posi-\\ntion information during the judgement.\\n'>,\n",
              "  <LTTextBoxHorizontal(40) 306.457,136.037,527.486,404.380 'Table 3 presents precision of top PR and QA\\nresults. We use P@k instead of EM and F1 to eval-\\nuate QA since ACS highlights the answer while\\nCovidex and COVID-19 RE highlight the sentence\\nthat contains the answer. Instead of extracting the\\nanswer, it is more reasonable to judge whether the\\nhighlighted text answers the question and compute\\nprecision. ACS achieves better accuracy on both\\nPR and QA on most metrics. Note that ACS high-\\nlights answer snippet for at most three passages\\n, while Covidex and COVID-19 RE highlight all\\ndisplayed passages. This explains why ACS P@3\\nunderperforms COVID-19 RE. Explicitly with an\\nexample, Figure 4 displays the topmost results of\\nquerying “What is the incubation period of virus”\\nfrom the three systems. ACS highlights the answer\\nin the passage. In contrast, Covidex presents an\\narticle published at 2004 with information of virus\\nwhich is irrelevant with coronavirus, and COVID-\\n19 RE does not answer the question at all.\\n'>,\n",
              "  <LTTextBoxHorizontal(41) 307.276,76.568,525.549,114.576 'As evident from the above results, ACS is one\\nof the top-performing systems that provides high\\nquality informative results over CORD-19 search.\\n'>,\n",
              "  <LTLine 167.054,778.645,430.489,778.645>,\n",
              "  <LTLine 244.213,761.607,244.213,775.156>,\n",
              "  <LTLine 306.305,761.607,306.305,775.156>,\n",
              "  <LTLine 368.397,761.607,368.397,775.156>,\n",
              "  <LTLine 244.213,748.057,244.213,761.606>,\n",
              "  <LTLine 306.305,748.057,306.305,761.606>,\n",
              "  <LTLine 368.397,748.057,368.397,761.606>,\n",
              "  <LTLine 167.054,745.906,430.489,745.906>,\n",
              "  <LTLine 244.213,729.031,244.213,742.580>,\n",
              "  <LTLine 306.305,729.031,306.305,742.580>,\n",
              "  <LTLine 368.397,729.031,368.397,742.580>,\n",
              "  <LTLine 244.213,715.482,244.213,729.031>,\n",
              "  <LTLine 306.305,715.482,306.305,729.031>,\n",
              "  <LTLine 368.397,715.482,368.397,729.031>,\n",
              "  <LTLine 244.213,701.933,244.213,715.482>,\n",
              "  <LTLine 306.305,701.933,306.305,715.482>,\n",
              "  <LTLine 368.397,701.933,368.397,715.482>,\n",
              "  <LTLine 167.054,699.618,430.489,699.618>,\n",
              "  <LTLine 134.327,666.864,463.216,666.864>,\n",
              "  <LTLine 211.485,649.826,211.485,663.375>,\n",
              "  <LTLine 337.351,649.826,337.351,663.375>,\n",
              "  <LTLine 211.485,636.277,211.485,649.826>,\n",
              "  <LTLine 337.351,636.277,337.351,649.826>,\n",
              "  <LTLine 134.327,634.125,463.216,634.125>,\n",
              "  <LTLine 211.485,617.250,211.485,630.799>,\n",
              "  <LTLine 337.351,617.250,337.351,630.799>,\n",
              "  <LTLine 211.485,603.701,211.485,617.250>,\n",
              "  <LTLine 337.351,603.701,337.351,617.250>,\n",
              "  <LTLine 211.485,590.152,211.485,603.701>,\n",
              "  <LTLine 337.351,590.152,337.351,603.701>,\n",
              "  <LTLine 134.327,587.837,463.216,587.837>,\n",
              "  <LTLine 110.264,264.604,124.949,264.604>,\n",
              "  <LTLine 147.349,264.604,153.372,264.604>],\n",
              " [<LTTextBoxHorizontal(0) 142.598,764.340,159.220,773.066 'ACS\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 286.114,764.340,315.423,773.066 'Covidex\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 424.543,764.340,476.695,773.066 'COVID-19 RE\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 226.245,748.431,375.285,757.157 'What is the incubation period of the virus?\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 234.561,730.257,342.408,738.982 'Deadly viral syndrome mimics\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 84.707,699.717,218.564,738.982 'Importance of Social Distancing:\\nModeling\\n2019-\\nof\\nSusceptible-Infected-\\nnCoV using\\nQuarantined-Recovered-t model\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 148.734,720.077,171.507,728.802 'spread\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 128.351,720.077,139.014,728.802 'the\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 83.546,618.280,218.563,688.144 '“Studies on the nature of the virus\\nhave suggested different incubation\\nperiods of the virus, and reports have\\nsuggested a median incubation pe-\\nriod of 5-6 days and a very high symp-\\ntom probability period of 14 days [5]\\n”\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 233.401,577.561,368.418,688.144 '“...The incubation period is typically\\n3 to 14 days with the symptoms of\\nan acute nonspeciﬁc, ﬂulike illness\\ndeveloping suddenly...The incuba-\\ntion period is 7 to 10 days before\\nthe onset of symptoms [37]. This\\nincubation period provides the po-\\ntential for worldwide exposure be-\\ncause a person harboring the virus\\ncan expose the world at large via air\\ntravel...”\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 384.417,709.897,518.274,738.982 'Prediction of the virus incubation pe-\\nriod for COVID-19 and future out-\\nbreaks\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 383.257,618.280,518.274,688.083 '“. . . while minimizing the negative\\nconsequences of the quarantine. 70 71\\nThe length of the incubation period\\nvaries both across and within virus\\nfamilies 4 . To our knowledge, 72\\ngenomic features (if any) that corre-\\nlate with the incubation. . . ”\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 73.917,553.886,523.629,563.849 'Figure 4: Top-1 result of article title and displayed passage by querying ’What is the incubation period of virus?’\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 72.000,504.668,133.115,516.624 '4 Analysis\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 72.000,294.788,292.083,495.386 'In this section we look into a number of sample\\nqueries to shed light on how different components\\nof ACS help in improving search results. We begin\\nby observing how small semantic differences in the\\nquery alter the results. The ﬁrst sample in Figure 5\\nis speciﬁc to medications. While the top result does\\nnot include the term medication the system high-\\nlights ribavirin and corticosteroids. The CORD-19\\nsystem understands that these terms represent med-\\nications with the help of CM NERe engine. In the\\nsecond example we change medications to mea-\\nsures and observe the top result discussing border\\ncontrol, and quarantine. This clearly demonstrates\\nthat Amazon Kendra has a deep comprehension of\\ntoken and query meanings.\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 71.487,145.747,292.173,292.148 'Finally, we take a look at the effects of topic mod-\\neling when grouping and ﬁltering results. The last\\ntwo examples in Figure 5 showcase the difference\\nthis makes in the top result. Without specifying any\\ntopic the resulting article discusses high level pol-\\nicy, speciﬁcally quarantine measures in Singapore.\\nWhen we ﬁlter by clinical treatment the top result\\ninstead focuses on infections which is covered in\\nthe clinical setting. Furthermore the extracted text\\nreturned to the user still focuses on lessons learned\\nstaying true to the query.\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 72.000,122.417,264.515,134.373 '5 Limitations and Future Directions\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 71.607,76.568,292.083,114.576 'AWS CORD-19 is an initial step towards help-\\ning medical researchers ﬁnd relevant content in\\na timely and meaningful way. In order to improve\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 307.276,491.262,525.548,515.720 'the robustness, we see following areas as direction\\nfor future research.\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 306.883,175.521,527.359,484.589 'Feedback Loop - Since ACS is a search engine\\nthe motivation would be to evaluate it as such; us-\\ning well-established methodologies based on test\\ncollections—comprising topics (information needs)\\nand human annotations. Since no designated evalu-\\nation data exist, our initial focus is to capture dif-\\nferent interactions and feedback. Currently, ACS\\nlacks the feedback loop and federated learning ap-\\nproaches where the system would continuously\\nlearn and improve the search. However, the system\\ncaptures feedback from the researchers in the form\\nof implicit and explicit reactions. Implicit feed-\\nback evaluation consists of topics of interests, their\\nclicks as well as the ranking of the results which\\nwere selected by medical researchers. Explicit feed-\\nback evaluation is captured by providing up-down\\nrating associated with each search results. In the\\nfuture results can be personalized based on this\\nfeedback. Now that we have a system in place, our\\nefforts have shifted to broader engagement with po-\\ntential stakeholders to solicit additional guidance,\\nwhile trying to balance between the features and\\nranking.\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 306.883,76.568,525.727,168.848 'Q&A Curation - Curation and normalization of\\nquestions have potential a use-case of presenting\\ntrending questions asked by the medical research\\ncommunity at a particular point. However, curation\\nwould involve capturing the questions asked as well\\nas identifying similar questions that can be later\\nnormalized. Currently, there is no mechanism to\\n'>,\n",
              "  <LTLine 75.981,778.444,525.546,778.444>,\n",
              "  <LTLine 75.981,762.535,525.546,762.535>,\n",
              "  <LTLine 75.981,744.121,525.546,744.121>,\n",
              "  <LTLine 75.981,573.013,525.546,573.013>],\n",
              " [<LTTextBoxHorizontal(0) 83.088,766.219,102.115,773.833 'Query\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 213.583,766.219,234.719,773.833 'Article\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 344.618,766.219,373.802,773.833 'Response\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 82.075,741.477,199.890,758.149 '“What medications were most beneﬁ-\\ncial in the SARS outbreak?”\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 213.853,732.593,330.658,757.974 'Development of chemical inhibitors\\nof the SARS coronavirus: Viral heli-\\ncase as a potential target\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 82.075,679.296,198.627,695.968 '“What measures were most beneﬁcial\\nin the SARS outbreak?”\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 213.853,670.413,330.658,695.793 'Impact of quarantine on the 2003\\nSARS outbreak: A retrospective mod-\\neling study\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 82.075,634.882,198.629,651.378 '“What did we learn from the SARS\\noutbreak?” (no topic)\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 213.853,634.882,329.394,651.378 'Use of quarantine in the control of\\nSARS in Singapore\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 82.075,590.467,198.629,606.964 '“What did we learn from the SARS\\noutbreak?” (clinical-treatment)\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 213.853,590.467,330.658,606.964 'Characteristics of COVID-19 infec-\\ntion in Beijing\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 343.606,705.944,519.196,757.974 '“...spread of SARS, a number of broad-spectrum antiviral\\nmedications were empirically administered to the SARS\\npatients during the SARS outbreak in 2003. These medi-\\ncations include ribavirin, HIV protease inhibitors, cor-\\nticosteroids, and alpha-interferon (IFN-a). In a retro-\\nspective review of treatment...”\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 343.606,661.530,517.931,695.793 '“During the 2003 Severe Acute Respiratory Syndrome\\n(SARS) outbreak, traditional intervention measures\\nsuch as quarantine and border control were found to\\nbe useful in containing the outbreak...”\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 343.606,617.116,517.926,651.378 '“The main lesson to learn from the SARS outbreak is the\\ncapability of an emerging infection to cause a pandemic\\nin a short span of time and the paradigm shift needed to\\nrespond to such a disease.”\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 343.606,581.584,519.190,606.964 '“We compared the epidemic features between COVID-19\\nand 2003 SARS for learn lessons and control the out-\\nbreak.”\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 72.000,546.613,525.547,568.532 'Figure 5: Sample queries demonstrating the semantic understanding of ACS, the use-fullness of Comprehend\\nMedical NERe, and the utility of topic modeling for ﬁltering.\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 72.000,511.939,270.317,522.848 'curate the questions asked by the researchers.\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 307.276,511.796,400.802,523.752 'Acknowledgments\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 72.000,424.903,292.075,503.634 'Summarization - Currently, ACS outputs the\\nrelevant passage based on the query. It would be\\nbeneﬁcial to get the overall summary of the pa-\\nper. A potential future direction would be to gen-\\nerates summaries (Raffel et al., 2019) from paper\\nabstracts and full body.\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 306.763,409.683,526.910,501.888 'We acknowledge the broader collaboration with\\nAI2 team and White house as well as the broader\\nAWS CORD-19 team including Kendra Science\\nteam, Tyler Stepke, Kingston Bosco, Victor Wang,\\nVaibhav Chaddha, Miguel Calvo, Ninad Kulkani,\\nKevin Longofer, Ray Chang, Adrian Bordone,\\nTony Nguyen, and Kyle Johnson.\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 72.000,381.109,147.067,393.065 '6 Conclusion\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 71.607,76.568,292.084,358.461 'This paper describes our efforts in building AWS\\nCORD-19 Search with its capabilities consisting\\nof topic based, knowledge graph, and natural lan-\\nguage search queries. This is further enhanced\\nwith reading comprehension and FAQ matching\\nas well as document ranking providing a scalable\\nsolution to COVID-19 researchers and policy mak-\\ners in their search and discovery for answers to\\nhigh priority scientiﬁc questions. Our solution is\\npowered by Amazon Kendra, Comprehend Medi-\\ncal and Neptune which incorporate the latest neural\\narchitectures to provide information access capa-\\nbilities to the CORD-19 challenge. By comparing\\nwith other public search engines that support sim-\\nilar functionalities over COVID-19 search, ACS\\nis demonstrated to be powerful on its document\\nranking and question answering components. We\\nhope that our solution can prove useful in the ﬁght\\nagainst this global pandemic, and that the capabili-\\nties we have developed can be applied to analyzing\\nthe scientiﬁc literature more broadly.\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 307.276,371.126,362.820,383.082 'References\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 307.276,297.584,527.200,362.342 'David Andrzejewski and Xiaojin Zhu. 2009. Latent\\ndirichlet allocation with topic-in-set knowledge. In\\nProceedings of the NAACL HLT 2009 Workshop\\non Semi-Supervised Learning for Natural Language\\nProcessing, pages 43–48. Association for Computa-\\ntional Linguistics.\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 307.276,253.340,527.200,285.221 'Iz Beltagy, Kyle Lo, and Arman Cohan. 2019. Scib-\\nert: Pretrained language model for scientiﬁc text. In\\nEMNLP.\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 307.276,176.219,527.290,240.977 'P. Bhatia, B. Celikkaya, M. Khalilia, and S. Senthivel.\\n2019. Comprehend medical: A named entity recog-\\nIn\\nnition and relationship extraction web service.\\n2019 18th IEEE International Conference On Ma-\\nchine Learning And Applications (ICMLA), pages\\n1844–1851.\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 307.276,132.204,527.200,163.856 'Parminder Bhatia, Busra Celikkaya, and Mohammed\\nJoint entity extraction and asser-\\narXiv preprint\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 318.185,121.016,428.371,152.897 'Khalilia. 2018.\\ntion detection for clinical\\narXiv:1812.05270.\\n'>,\n",
              "  <LTTextBoxHorizontal(28) 434.110,131.975,451.395,141.938 'text.\\n'>,\n",
              "  <LTTextBoxHorizontal(29) 307.276,76.772,527.290,108.653 'David M Blei, Andrew Y Ng, and Michael I Jordan.\\n2003. Latent dirichlet allocation. Journal of ma-\\nchine Learning research, 3(Jan):993–1022.\\n'>,\n",
              "  <LTLine 75.473,778.526,525.542,778.526>,\n",
              "  <LTLine 75.473,762.458,525.542,762.458>,\n",
              "  <LTLine 75.473,577.615,525.542,577.615>],\n",
              " [<LTTextBoxHorizontal(0) 72.000,722.135,292.014,775.934 'Jeffrey Dalton, Laura Dietz, and James Allan. 2014.\\nEntity query feature expansion using knowledge\\nbase links. In Proceedings of the 37th international\\nACM SIGIR conference on Research & development\\nin information retrieval, pages 365–374.\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 307.276,722.135,527.200,775.934 'Da Zheng, Xiang Song, Chao Ma, Zeyuan Tan, Zi-\\nhao Ye, Jin Dong, Hao Xiong, Zheng Zhang, and\\nGeorge Karypis. 2020. Dgl-ke: Training knowl-\\narXiv preprint\\nedge graph embeddings at scale.\\narXiv:2004.08532.\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 72.000,567.628,292.014,709.097 'Wei-jie Guan, Zheng-yi Ni, Yu Hu, Wen-hua Liang,\\nChun-quan Ou, Jian-xing He, Lei Liu, Hong Shan,\\nChun-liang Lei, David S.C. Hui, Bin Du, Lan-juan\\nLi, Guang Zeng, Kwok-Yung Yuen, Ru-chong Chen,\\nChun-li Tang, Tao Wang, Ping-yan Chen, Jie Xiang,\\nShi-yue Li, Jin-lin Wang, Zi-jing Liang, Yi-xiang\\nPeng, Li Wei, Yong Liu, Ya-hua Hu, Peng Peng,\\nJian-ming Wang, Ji-yang Liu, Zhong Chen, Gang\\nLi, Zhi-jian Zheng, Shao-qin Qiu, Jie Luo, Chang-\\njiang Ye, Shao-yong Zhu, and Nan-shan Zhong.\\n2020. Clinical characteristics of coronavirus disease\\n2019 in china. New England Journal of Medicine,\\n382(18):1708–1720.\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 72.000,500.791,291.924,554.590 'Colin Raffel, Noam Shazeer, Adam Roberts, Katherine\\nLee, Sharan Narang, Michael Matena, Yanqi Zhou,\\nWei Li, and Peter J Liu. 2019. Exploring the limits\\nof transfer learning with a uniﬁed text-to-text trans-\\nformer. arXiv preprint arXiv:1910.10683.\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 72.000,455.873,290.271,487.753 'Jesse Read, Bernhard Pfahringer, Geoff Holmes, and\\nEibe Frank. 2011. Classiﬁer chains for multi-label\\nclassiﬁcation. Machine learning, 85(3):333.\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 72.000,399.995,291.516,442.835 'Maya Rotmensch, Yoni Halpern, Abdulhakim Tlimat,\\nSteven Horng, and David Sontag. 2017. Learning\\na health knowledge graph from electronic medical\\nrecords. Scientiﬁc reports, 7(1):1–11.\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 72.000,355.077,290.271,386.957 'Gaurav Singh and Parminder Bhatia. 2019. Relation\\nextraction using explicit context conditioning. arXiv\\npreprint arXiv:1902.09271.\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 72.000,288.240,291.924,342.039 'Raphael Tang, Rodrigo Nogueira, Edwin Zhang, Nikhil\\nGupta, Phuong Cam, Kyunghyun Cho, and Jimmy\\nLin. 2020. Rapidly Bootstrapping a Question An-\\narXiv preprint\\nswering Dataset for COVID-19.\\narXiv:2004.11339.\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 72.000,210.445,292.014,275.203 'Ellen Voorhees, Tasmeer Alam, Steven Bedrick, Dina\\nDemner-Fushman, William R Hersh, Kyle Lo, Kirk\\nRoberts, Ian Soborof, and Lucy Lu Wang. 2020.\\nTREC-COVID: Constructing a Pandemic Informa-\\narXiv preprint\\ntion Retrieval Test Collection.\\narXiv:2005.04474.\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 72.000,143.609,291.516,197.407 'Lucy Lu Wang, Kyle Lo, Yoganand Chandrasekhar,\\nRussell Reas, Jiangjiang Yang, Darrin Eide, Kathryn\\nFunk, Rodney Kinney, Ziyang Liu, William Merrill,\\net al. 2020. Cord-19: The covid-19 open research\\ndataset. arXiv preprint arXiv:2004.10706.\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 72.000,76.772,292.014,130.571 'Quan Wang, Zhendong Mao, Bin Wang, and Li Guo.\\n2017. Knowledge graph embedding: A survey of\\nIEEE Transactions\\napproaches and applications.\\non Knowledge and Data Engineering, 29(12):2724–\\n2743.\\n'>]]"
            ]
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(LT_elem_text)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YTvjAtnKuubI",
        "outputId": "b509b1b9-d9b4-4268-adae-fcb29973aff0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "9"
            ]
          },
          "metadata": {},
          "execution_count": 75
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Layout objects for page 3 of PDF."
      ],
      "metadata": {
        "id": "mFAHO8WqwfgG"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yUka33J8Idu2",
        "outputId": "f353d35f-b112-4b1c-c7ed-55fc1acd82c6"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTTextBoxHorizontal(0) 237.044,540.631,360.501,550.594 'Figure 2: System Architecture.\\n'>,\n",
              " <LTTextBoxHorizontal(1) 93.818,492.407,290.280,516.866 'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n'>,\n",
              " <LTTextBoxHorizontal(2) 72.000,325.745,292.082,472.146 'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n'>,\n",
              " <LTTextBoxHorizontal(3) 72.000,297.540,197.858,308.449 '2.2 Comprehend Medical\\n'>,\n",
              " <LTTextBoxHorizontal(4) 71.575,141.116,292.075,289.172 'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n'>,\n",
              " <LTTextBoxHorizontal(5) 307.276,506.033,470.563,516.942 '2.3 COVID-19 Knowledge Graph\\n'>,\n",
              " <LTTextBoxHorizontal(6) 307.276,148.089,527.359,497.728 'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n'>,\n",
              " <LTTextBoxHorizontal(7) 84.653,118.369,217.255,128.182 '3https://docs.aws.amazon.\\n'>,\n",
              " <LTTextBoxHorizontal(8) 72.000,98.443,245.150,117.372 'com/comprehend/latest/dg/\\nhow-document-classification.html\\n'>,\n",
              " <LTTextBoxHorizontal(9) 84.653,87.569,271.054,97.382 '4https://aws.amazon.com/comprehend/\\n'>,\n",
              " <LTTextBoxHorizontal(10) 72.000,77.606,115.039,86.572 'medical/\\n'>,\n",
              " <LTTextBoxHorizontal(11) 307.276,122.881,394.156,133.790 '2.4 Topic Models\\n'>,\n",
              " <LTTextBoxHorizontal(12) 306.937,76.568,525.549,114.576 'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n'>,\n",
              " <LTFigure(Im1) 72.000,559.122,526.860,779.082 matrix=[454.86,0.00,0.00,219.96, (72.00,559.12)]>,\n",
              " <LTLine 72.000,130.362,131.776,130.362>]"
            ]
          },
          "metadata": {},
          "execution_count": 76
        }
      ],
      "source": [
        "LT_elem_text[2]"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "G4eTti49_J8Q"
      },
      "source": [
        "### 4.7 Alternative method\n",
        "\n",
        "An alternative method to show the location and text of a layout object and all its descendants is demonstrated below by one of the maintainers of pdfminer.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "wnbyXjWo_J8b"
      },
      "outputs": [],
      "source": [
        "# https://stackoverflow.com/questions/22898145/how-to-extract-text-and-text-coordinates-from-a-pdf-file\n",
        "\n",
        "def show_ltitem_hierarchy(o: Any, depth=0):\n",
        "    \"\"\"Show location and text of LTItem and all its descendants\"\"\"\n",
        "    if depth == 0:\n",
        "        print('element                        x0  y0  x1  y1   text')\n",
        "        print('------------------------------ --- --- --- ---- -----')\n",
        "\n",
        "    print(\n",
        "        f'{get_indented_name(o, depth):<30.30s} '\n",
        "        f'{get_optional_bbox(o)} '\n",
        "        f'{get_optional_text(o)}'\n",
        "    )\n",
        "\n",
        "    if isinstance(o, Iterable):\n",
        "        for i in o:\n",
        "            show_ltitem_hierarchy(i, depth=depth + 1)\n",
        "\n",
        "\n",
        "def get_indented_name(o: Any, depth: int) -> str:\n",
        "    \"\"\"Indented name of LTItem\"\"\"\n",
        "    return '  ' * depth + o.__class__.__name__\n",
        "\n",
        "\n",
        "def get_optional_bbox(o: Any) -> str:\n",
        "    \"\"\"Bounding box of LTItem if available, otherwise empty string\"\"\"\n",
        "    if hasattr(o, 'bbox'):\n",
        "        return ''.join(f'{i:<4.0f}' for i in o.bbox)\n",
        "    return ''\n",
        "\n",
        "\n",
        "def get_optional_text(o: Any) -> str:\n",
        "    \"\"\"Text of LTItem if available, otherwise empty string\"\"\"\n",
        "    if hasattr(o, 'get_text'):\n",
        "        return o.get_text().strip()\n",
        "    return ''"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We will try this for the arXiv PDF we are analysing in order to extract text."
      ],
      "metadata": {
        "id": "rHNvGPZ-Bk7P"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0513f9e9-f506-4177-b520-b958d72c4a6c",
        "id": "pOnwbdoQ_J8b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "element                        x0  y0  x1  y1   text\n",
            "------------------------------ --- --- --- ---- -----\n",
            "generator                       \n",
            "  LTPage                       0   0   595 842  \n",
            "    LTTextBoxHorizontal        103 746 494 776  AWS CORD-19 Search: A Neural Search Engine for COVID-19\n",
            "Literature\n",
            "      LTTextLineHorizontal     103 762 494 776  AWS CORD-19 Search: A Neural Search Engine for COVID-19\n",
            "        LTChar                 103 762 114 776  A\n",
            "        LTChar                 112 762 126 776  W\n",
            "        LTChar                 126 762 134 776  S\n",
            "        LTAnno                  \n",
            "        LTChar                 138 762 148 776  C\n",
            "        LTChar                 148 762 159 776  O\n",
            "        LTChar                 159 762 170 776  R\n",
            "        LTChar                 170 762 180 776  D\n",
            "        LTChar                 180 762 185 776  -\n",
            "        LTChar                 185 762 192 776  1\n",
            "        LTChar                 192 762 199 776  9\n",
            "        LTAnno                  \n",
            "        LTChar                 203 762 211 776  S\n",
            "        LTChar                 211 762 217 776  e\n",
            "        LTChar                 217 762 224 776  a\n",
            "        LTChar                 224 762 231 776  r\n",
            "        LTChar                 230 762 237 776  c\n",
            "        LTChar                 237 762 245 776  h\n",
            "        LTChar                 245 762 249 776  :\n",
            "        LTAnno                  \n",
            "        LTChar                 254 762 264 776  A\n",
            "        LTAnno                  \n",
            "        LTChar                 268 762 278 776  N\n",
            "        LTChar                 278 762 285 776  e\n",
            "        LTChar                 285 762 293 776  u\n",
            "        LTChar                 293 762 299 776  r\n",
            "        LTChar                 299 762 306 776  a\n",
            "        LTChar                 306 762 310 776  l\n",
            "        LTAnno                  \n",
            "        LTChar                 314 762 322 776  S\n",
            "        LTChar                 322 762 328 776  e\n",
            "        LTChar                 328 762 335 776  a\n",
            "        LTChar                 335 762 342 776  r\n",
            "        LTChar                 341 762 348 776  c\n",
            "        LTChar                 348 762 356 776  h\n",
            "        LTAnno                  \n",
            "        LTChar                 359 762 369 776  E\n",
            "        LTChar                 369 762 377 776  n\n",
            "        LTChar                 377 762 384 776  g\n",
            "        LTChar                 384 762 388 776  i\n",
            "        LTChar                 388 762 396 776  n\n",
            "        LTChar                 396 762 402 776  e\n",
            "        LTAnno                  \n",
            "        LTChar                 406 762 411 776  f\n",
            "        LTChar                 410 762 417 776  o\n",
            "        LTChar                 417 762 424 776  r\n",
            "        LTAnno                  \n",
            "        LTChar                 427 762 438 776  C\n",
            "        LTChar                 438 762 449 776  O\n",
            "        LTChar                 448 762 459 776  V\n",
            "        LTChar                 459 762 464 776  I\n",
            "        LTChar                 464 762 475 776  D\n",
            "        LTChar                 475 762 479 776  -\n",
            "        LTChar                 479 762 486 776  1\n",
            "        LTChar                 486 762 494 776  9\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     267 746 331 760  Literature\n",
            "        LTChar                 267 746 277 760  L\n",
            "        LTChar                 277 746 281 760  i\n",
            "        LTChar                 281 746 285 760  t\n",
            "        LTChar                 285 746 292 760  e\n",
            "        LTChar                 292 746 298 760  r\n",
            "        LTChar                 298 746 305 760  a\n",
            "        LTChar                 305 746 310 760  t\n",
            "        LTChar                 310 746 318 760  u\n",
            "        LTChar                 318 746 324 760  r\n",
            "        LTChar                 324 746 331 760  e\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        77  640 523 734  Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\n",
            "Ben Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\n",
            "Ramesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\n",
            "Amazon Web Services AI\n",
            "parmib, liuall, arumae, nimpourd, suyogd, snivelyb,\n",
            "monamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\n",
            "zhiheng, bxiang, tahak @amazon.com\n",
            "      LTTextLineHorizontal     77  722 523 734  Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\n",
            "        LTChar                 77  722 84  734  P\n",
            "        LTChar                 84  722 90  734  a\n",
            "        LTChar                 90  722 96  734  r\n",
            "        LTChar                 96  722 106 734  m\n",
            "        LTChar                 106 722 109 734  i\n",
            "        LTChar                 109 722 115 734  n\n",
            "        LTChar                 115 722 122 734  d\n",
            "        LTChar                 122 722 127 734  e\n",
            "        LTChar                 127 722 133 734  r\n",
            "        LTAnno                  \n",
            "        LTChar                 136 722 144 734  B\n",
            "        LTChar                 144 722 150 734  h\n",
            "        LTChar                 150 722 156 734  a\n",
            "        LTChar                 156 722 160 734  t\n",
            "        LTChar                 160 722 164 734  i\n",
            "        LTChar                 164 722 170 734  a\n",
            "        LTChar                 170 722 173 734  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 176 722 184 734  L\n",
            "        LTChar                 184 722 190 734  a\n",
            "        LTChar                 190 722 196 734  n\n",
            "        LTAnno                  \n",
            "        LTChar                 199 722 207 734  L\n",
            "        LTChar                 207 722 210 734  i\n",
            "        LTChar                 210 722 217 734  u\n",
            "        LTChar                 217 722 220 734  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 223 722 232 734  K\n",
            "        LTChar                 232 722 238 734  r\n",
            "        LTChar                 238 722 241 734  i\n",
            "        LTChar                 241 722 246 734  s\n",
            "        LTChar                 246 722 250 734  t\n",
            "        LTChar                 250 722 254 734  j\n",
            "        LTChar                 254 722 260 734  a\n",
            "        LTChar                 260 722 266 734  n\n",
            "        LTAnno                  \n",
            "        LTChar                 269 722 278 734  A\n",
            "        LTChar                 278 722 283 734  r\n",
            "        LTChar                 283 722 290 734  u\n",
            "        LTChar                 290 722 300 734  m\n",
            "        LTChar                 300 722 306 734  a\n",
            "        LTChar                 306 722 311 734  e\n",
            "        LTChar                 311 722 314 734  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 317 722 326 734  N\n",
            "        LTChar                 326 722 329 734  i\n",
            "        LTChar                 329 722 339 734  m\n",
            "        LTChar                 339 722 345 734  a\n",
            "        LTAnno                  \n",
            "        LTChar                 348 722 355 734  P\n",
            "        LTChar                 355 722 361 734  o\n",
            "        LTChar                 361 722 368 734  u\n",
            "        LTChar                 368 722 373 734  r\n",
            "        LTChar                 373 722 380 734  d\n",
            "        LTChar                 380 722 386 734  a\n",
            "        LTChar                 386 722 396 734  m\n",
            "        LTChar                 396 722 402 734  g\n",
            "        LTChar                 402 722 408 734  h\n",
            "        LTChar                 408 722 414 734  a\n",
            "        LTChar                 414 722 421 734  n\n",
            "        LTChar                 421 722 424 734  i\n",
            "        LTChar                 424 722 427 734  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 430 722 437 734  S\n",
            "        LTChar                 437 722 443 734  u\n",
            "        LTChar                 443 722 449 734  y\n",
            "        LTChar                 449 722 455 734  o\n",
            "        LTChar                 455 722 461 734  g\n",
            "        LTAnno                  \n",
            "        LTChar                 464 722 473 734  D\n",
            "        LTChar                 473 722 478 734  e\n",
            "        LTChar                 478 722 483 734  s\n",
            "        LTChar                 483 722 489 734  h\n",
            "        LTChar                 489 722 496 734  p\n",
            "        LTChar                 496 722 502 734  a\n",
            "        LTChar                 502 722 509 734  n\n",
            "        LTChar                 509 722 515 734  d\n",
            "        LTChar                 515 722 520 734  e\n",
            "        LTChar                 520 722 523 734  ,\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     80  709 520 720  Ben Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\n",
            "        LTChar                 80  709 88  720  B\n",
            "        LTChar                 88  709 94  720  e\n",
            "        LTChar                 94  709 100 720  n\n",
            "        LTAnno                  \n",
            "        LTChar                 103 709 110 720  S\n",
            "        LTChar                 110 709 117 720  n\n",
            "        LTChar                 117 709 120 720  i\n",
            "        LTChar                 120 709 126 720  v\n",
            "        LTChar                 126 709 131 720  e\n",
            "        LTChar                 131 709 134 720  l\n",
            "        LTChar                 134 709 140 720  y\n",
            "        LTChar                 140 709 143 720  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 146 709 157 720  M\n",
            "        LTChar                 157 709 163 720  o\n",
            "        LTChar                 163 709 169 720  n\n",
            "        LTChar                 169 709 175 720  a\n",
            "        LTAnno                  \n",
            "        LTChar                 178 709 190 720  M\n",
            "        LTChar                 190 709 196 720  o\n",
            "        LTChar                 196 709 202 720  n\n",
            "        LTChar                 202 709 208 720  a\n",
            "        LTChar                 208 709 211 720  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 214 709 223 720  C\n",
            "        LTChar                 223 709 229 720  o\n",
            "        LTChar                 229 709 232 720  l\n",
            "        LTChar                 232 709 239 720  b\n",
            "        LTChar                 239 709 245 720  y\n",
            "        LTAnno                  \n",
            "        LTChar                 248 709 260 720  W\n",
            "        LTChar                 260 709 263 720  i\n",
            "        LTChar                 263 709 268 720  s\n",
            "        LTChar                 268 709 273 720  e\n",
            "        LTChar                 273 709 276 720  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 279 709 288 720  G\n",
            "        LTChar                 288 709 293 720  e\n",
            "        LTChar                 293 709 299 720  o\n",
            "        LTChar                 299 709 305 720  r\n",
            "        LTChar                 305 709 311 720  g\n",
            "        LTChar                 311 709 316 720  e\n",
            "        LTAnno                  \n",
            "        LTChar                 319 709 326 720  P\n",
            "        LTChar                 326 709 331 720  r\n",
            "        LTChar                 331 709 335 720  i\n",
            "        LTChar                 335 709 340 720  c\n",
            "        LTChar                 340 709 345 720  e\n",
            "        LTChar                 345 709 348 720  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 351 709 358 720  S\n",
            "        LTChar                 358 709 365 720  h\n",
            "        LTChar                 364 709 370 720  y\n",
            "        LTChar                 370 709 376 720  a\n",
            "        LTChar                 376 709 386 720  m\n",
            "        LTAnno                  \n",
            "        LTChar                 389 709 398 720  R\n",
            "        LTChar                 398 709 404 720  a\n",
            "        LTChar                 404 709 414 720  m\n",
            "        LTChar                 414 709 420 720  a\n",
            "        LTChar                 420 709 425 720  s\n",
            "        LTChar                 425 709 433 720  w\n",
            "        LTChar                 433 709 439 720  a\n",
            "        LTChar                 439 709 449 720  m\n",
            "        LTChar                 449 709 455 720  y\n",
            "        LTChar                 454 709 457 720  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 460 709 469 720  X\n",
            "        LTChar                 469 709 472 720  i\n",
            "        LTChar                 472 709 478 720  a\n",
            "        LTChar                 478 709 484 720  o\n",
            "        LTChar                 484 709 488 720  f\n",
            "        LTChar                 488 709 494 720  e\n",
            "        LTChar                 494 709 497 720  i\n",
            "        LTAnno                  \n",
            "        LTChar                 500 709 511 720  M\n",
            "        LTChar                 511 709 517 720  a\n",
            "        LTChar                 517 709 520 720  ,\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     135 695 465 707  Ramesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\n",
            "        LTChar                 135 695 144 707  R\n",
            "        LTChar                 144 695 150 707  a\n",
            "        LTChar                 150 695 160 707  m\n",
            "        LTChar                 160 695 165 707  e\n",
            "        LTChar                 165 695 170 707  s\n",
            "        LTChar                 170 695 177 707  h\n",
            "        LTAnno                  \n",
            "        LTChar                 180 695 188 707  N\n",
            "        LTChar                 188 695 194 707  a\n",
            "        LTChar                 194 695 198 707  l\n",
            "        LTChar                 198 695 201 707  l\n",
            "        LTChar                 201 695 207 707  a\n",
            "        LTChar                 207 695 213 707  p\n",
            "        LTChar                 213 695 219 707  a\n",
            "        LTChar                 219 695 223 707  t\n",
            "        LTChar                 223 695 227 707  i\n",
            "        LTChar                 227 695 230 707  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 233 695 241 707  Z\n",
            "        LTChar                 241 695 247 707  h\n",
            "        LTChar                 247 695 251 707  i\n",
            "        LTChar                 251 695 257 707  h\n",
            "        LTChar                 257 695 263 707  e\n",
            "        LTChar                 263 695 269 707  n\n",
            "        LTChar                 269 695 275 707  g\n",
            "        LTAnno                  \n",
            "        LTChar                 278 695 288 707  H\n",
            "        LTChar                 288 695 294 707  u\n",
            "        LTChar                 294 695 300 707  a\n",
            "        LTChar                 300 695 307 707  n\n",
            "        LTChar                 307 695 313 707  g\n",
            "        LTChar                 313 695 316 707  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 319 695 327 707  B\n",
            "        LTChar                 327 695 330 707  i\n",
            "        LTChar                 330 695 337 707  n\n",
            "        LTChar                 337 695 343 707  g\n",
            "        LTAnno                  \n",
            "        LTChar                 346 695 354 707  X\n",
            "        LTChar                 354 695 358 707  i\n",
            "        LTChar                 358 695 364 707  a\n",
            "        LTChar                 364 695 370 707  n\n",
            "        LTChar                 370 695 376 707  g\n",
            "        LTChar                 376 695 379 707  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 382 695 390 707  T\n",
            "        LTChar                 389 695 395 707  a\n",
            "        LTChar                 395 695 402 707  h\n",
            "        LTChar                 402 695 408 707  a\n",
            "        LTAnno                  \n",
            "        LTChar                 411 695 420 707  K\n",
            "        LTChar                 420 695 426 707  a\n",
            "        LTChar                 426 695 431 707  s\n",
            "        LTChar                 431 695 435 707  s\n",
            "        LTChar                 435 695 439 707  -\n",
            "        LTChar                 439 695 449 707  H\n",
            "        LTChar                 449 695 455 707  o\n",
            "        LTChar                 455 695 461 707  u\n",
            "        LTChar                 461 695 465 707  t\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     238 681 362 692  Amazon Web Services AI\n",
            "        LTChar                 238 681 247 692  A\n",
            "        LTChar                 247 681 256 692  m\n",
            "        LTChar                 256 681 261 692  a\n",
            "        LTChar                 261 681 267 692  z\n",
            "        LTChar                 267 681 273 692  o\n",
            "        LTChar                 273 681 279 692  n\n",
            "        LTAnno                  \n",
            "        LTChar                 282 681 293 692  W\n",
            "        LTChar                 292 681 297 692  e\n",
            "        LTChar                 297 681 303 692  b\n",
            "        LTAnno                  \n",
            "        LTChar                 306 681 313 692  S\n",
            "        LTChar                 313 681 318 692  e\n",
            "        LTChar                 318 681 322 692  r\n",
            "        LTChar                 322 681 328 692  v\n",
            "        LTChar                 328 681 331 692  i\n",
            "        LTChar                 331 681 337 692  c\n",
            "        LTChar                 337 681 342 692  e\n",
            "        LTChar                 342 681 347 692  s\n",
            "        LTAnno                  \n",
            "        LTChar                 350 681 358 692  A\n",
            "        LTChar                 358 681 362 692  I\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     117 667 483 679  parmib, liuall, arumae, nimpourd, suyogd, snivelyb,\n",
            "        LTChar                 117 667 125 679  p\n",
            "        LTChar                 125 667 132 679  a\n",
            "        LTChar                 132 667 139 679  r\n",
            "        LTChar                 139 667 146 679  m\n",
            "        LTChar                 146 667 153 679  i\n",
            "        LTChar                 153 667 160 679  b\n",
            "        LTChar                 160 667 168 679  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 175 667 182 679  l\n",
            "        LTChar                 182 667 189 679  i\n",
            "        LTChar                 189 667 196 679  u\n",
            "        LTChar                 196 667 203 679  a\n",
            "        LTChar                 203 667 211 679  l\n",
            "        LTChar                 211 667 218 679  l\n",
            "        LTChar                 218 667 225 679  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 232 667 239 679  a\n",
            "        LTChar                 239 667 246 679  r\n",
            "        LTChar                 246 667 254 679  u\n",
            "        LTChar                 254 667 261 679  m\n",
            "        LTChar                 261 667 268 679  a\n",
            "        LTChar                 268 667 275 679  e\n",
            "        LTChar                 275 667 282 679  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 290 667 297 679  n\n",
            "        LTChar                 297 667 304 679  i\n",
            "        LTChar                 304 667 311 679  m\n",
            "        LTChar                 311 667 318 679  p\n",
            "        LTChar                 318 667 325 679  o\n",
            "        LTChar                 325 667 333 679  u\n",
            "        LTChar                 333 667 340 679  r\n",
            "        LTChar                 340 667 347 679  d\n",
            "        LTChar                 347 667 354 679  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 361 667 368 679  s\n",
            "        LTChar                 368 667 376 679  u\n",
            "        LTChar                 376 667 383 679  y\n",
            "        LTChar                 383 667 390 679  o\n",
            "        LTChar                 390 667 397 679  g\n",
            "        LTChar                 397 667 404 679  d\n",
            "        LTChar                 404 667 411 679  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 419 667 426 679  s\n",
            "        LTChar                 426 667 433 679  n\n",
            "        LTChar                 433 667 440 679  i\n",
            "        LTChar                 440 667 447 679  v\n",
            "        LTChar                 447 667 454 679  e\n",
            "        LTChar                 454 667 462 679  l\n",
            "        LTChar                 462 667 469 679  y\n",
            "        LTChar                 469 667 476 679  b\n",
            "        LTChar                 476 667 483 679  ,\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     114 653 487 665  monamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\n",
            "        LTChar                 114 653 121 665  m\n",
            "        LTChar                 121 653 128 665  o\n",
            "        LTChar                 128 653 135 665  n\n",
            "        LTChar                 135 653 142 665  a\n",
            "        LTChar                 142 653 150 665  m\n",
            "        LTChar                 150 653 157 665  o\n",
            "        LTChar                 157 653 164 665  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 171 653 178 665  c\n",
            "        LTChar                 178 653 185 665  o\n",
            "        LTChar                 185 653 193 665  l\n",
            "        LTChar                 193 653 200 665  b\n",
            "        LTChar                 200 653 207 665  y\n",
            "        LTChar                 207 653 214 665  w\n",
            "        LTChar                 214 653 221 665  i\n",
            "        LTChar                 221 653 229 665  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 236 653 243 665  g\n",
            "        LTChar                 243 653 250 665  w\n",
            "        LTChar                 250 653 257 665  p\n",
            "        LTChar                 257 653 264 665  r\n",
            "        LTChar                 264 653 272 665  i\n",
            "        LTChar                 272 653 279 665  c\n",
            "        LTChar                 279 653 286 665  e\n",
            "        LTChar                 286 653 293 665  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 300 653 307 665  s\n",
            "        LTChar                 307 653 315 665  h\n",
            "        LTChar                 315 653 322 665  y\n",
            "        LTChar                 322 653 329 665  a\n",
            "        LTChar                 329 653 336 665  r\n",
            "        LTChar                 336 653 343 665  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 350 653 358 665  x\n",
            "        LTChar                 358 653 365 665  i\n",
            "        LTChar                 365 653 372 665  a\n",
            "        LTChar                 372 653 379 665  o\n",
            "        LTChar                 379 653 386 665  f\n",
            "        LTChar                 386 653 394 665  e\n",
            "        LTChar                 394 653 401 665  i\n",
            "        LTChar                 401 653 408 665  m\n",
            "        LTChar                 408 653 415 665  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 422 653 429 665  r\n",
            "        LTChar                 429 653 437 665  n\n",
            "        LTChar                 437 653 444 665  a\n",
            "        LTChar                 444 653 451 665  l\n",
            "        LTChar                 451 653 458 665  l\n",
            "        LTChar                 458 653 465 665  a\n",
            "        LTChar                 465 653 472 665  p\n",
            "        LTChar                 472 653 480 665  a\n",
            "        LTChar                 480 653 487 665  ,\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     178 640 422 651  zhiheng, bxiang, tahak @amazon.com\n",
            "        LTChar                 178 640 185 651  z\n",
            "        LTChar                 185 640 193 651  h\n",
            "        LTChar                 193 640 200 651  i\n",
            "        LTChar                 200 640 207 651  h\n",
            "        LTChar                 207 640 214 651  e\n",
            "        LTChar                 214 640 221 651  n\n",
            "        LTChar                 221 640 229 651  g\n",
            "        LTChar                 229 640 236 651  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 243 640 250 651  b\n",
            "        LTChar                 250 640 257 651  x\n",
            "        LTChar                 257 640 264 651  i\n",
            "        LTChar                 264 640 272 651  a\n",
            "        LTChar                 272 640 279 651  n\n",
            "        LTChar                 279 640 286 651  g\n",
            "        LTChar                 286 640 293 651  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 300 640 307 651  t\n",
            "        LTChar                 307 640 315 651  a\n",
            "        LTChar                 315 640 322 651  h\n",
            "        LTChar                 322 640 329 651  a\n",
            "        LTChar                 329 640 336 651  k\n",
            "        LTAnno                  \n",
            "        LTChar                 343 640 350 651  @\n",
            "        LTChar                 350 640 358 651  a\n",
            "        LTChar                 358 640 365 651  m\n",
            "        LTChar                 365 640 372 651  a\n",
            "        LTChar                 372 640 379 651  z\n",
            "        LTChar                 379 640 386 651  o\n",
            "        LTChar                 386 640 394 651  n\n",
            "        LTChar                 394 640 401 651  .\n",
            "        LTChar                 401 640 408 651  c\n",
            "        LTChar                 408 640 415 651  o\n",
            "        LTChar                 415 640 422 651  m\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        159 606 203 618  Abstract\n",
            "      LTTextLineHorizontal     159 606 203 618  Abstract\n",
            "        LTChar                 159 606 168 618  A\n",
            "        LTChar                 168 606 174 618  b\n",
            "        LTChar                 174 606 179 618  s\n",
            "        LTChar                 179 606 183 618  t\n",
            "        LTChar                 183 606 188 618  r\n",
            "        LTChar                 188 606 194 618  a\n",
            "        LTChar                 194 606 199 618  c\n",
            "        LTChar                 199 606 203 618  t\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        88  323 275 596  Coronavirus disease (COVID-19) has been de-\n",
            "clared as a pandemic by WHO with thousands\n",
            "of cases being reported each day. Numerous\n",
            "scientiﬁc articles are being published on the\n",
            "disease raising the need for a service which\n",
            "can organize, and query them in a reliable fash-\n",
            "ion. To support this cause we present AWS\n",
            "CORD-19 Search (ACS), a public, COVID-19\n",
            "speciﬁc, neural search engine that is powered\n",
            "by several machine learning systems to sup-\n",
            "port natural language based searches. ACS\n",
            "with capabilities such as document ranking,\n",
            "passage ranking, question answering and topic\n",
            "classiﬁcation provides a scalable solution to\n",
            "COVID-19 researchers and policy makers in\n",
            "their search and discovery for answers to high\n",
            "priority scientiﬁc questions. We present a\n",
            "quantitative evaluation and qualitative analysis\n",
            "of the system against other leading COVID-\n",
            "19 search platforms. ACS is top performing\n",
            "across these systems yielding quality results\n",
            "which we detail with relevant examples in this\n",
            "work.\n",
            "      LTTextLineHorizontal     89  586 275 596  Coronavirus disease (COVID-19) has been de-\n",
            "        LTChar                 89  586 96  596  C\n",
            "        LTChar                 96  586 101 596  o\n",
            "        LTChar                 101 586 104 596  r\n",
            "        LTChar                 104 586 109 596  o\n",
            "        LTChar                 109 586 114 596  n\n",
            "        LTChar                 114 586 118 596  a\n",
            "        LTChar                 118 586 123 596  v\n",
            "        LTChar                 123 586 126 596  i\n",
            "        LTChar                 126 586 129 596  r\n",
            "        LTChar                 129 586 134 596  u\n",
            "        LTChar                 134 586 138 596  s\n",
            "        LTAnno                  \n",
            "        LTChar                 141 586 146 596  d\n",
            "        LTChar                 146 586 148 596  i\n",
            "        LTChar                 148 586 152 596  s\n",
            "        LTChar                 152 586 157 596  e\n",
            "        LTChar                 157 586 161 596  a\n",
            "        LTChar                 161 586 165 596  s\n",
            "        LTChar                 165 586 169 596  e\n",
            "        LTAnno                  \n",
            "        LTChar                 172 586 175 596  (\n",
            "        LTChar                 175 586 182 596  C\n",
            "        LTChar                 182 586 189 596  O\n",
            "        LTChar                 188 586 196 596  V\n",
            "        LTChar                 196 586 199 596  I\n",
            "        LTChar                 199 586 206 596  D\n",
            "        LTChar                 206 586 209 596  -\n",
            "        LTChar                 209 586 214 596  1\n",
            "        LTChar                 214 586 219 596  9\n",
            "        LTChar                 219 586 223 596  )\n",
            "        LTAnno                  \n",
            "        LTChar                 225 586 230 596  h\n",
            "        LTChar                 230 586 235 596  a\n",
            "        LTChar                 235 586 238 596  s\n",
            "        LTAnno                  \n",
            "        LTChar                 241 586 246 596  b\n",
            "        LTChar                 246 586 250 596  e\n",
            "        LTChar                 250 586 255 596  e\n",
            "        LTChar                 255 586 260 596  n\n",
            "        LTAnno                  \n",
            "        LTChar                 262 586 267 596  d\n",
            "        LTChar                 267 586 272 596  e\n",
            "        LTChar                 272 586 275 596  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  574 273 584  clared as a pandemic by WHO with thousands\n",
            "        LTChar                 89  574 93  584  c\n",
            "        LTChar                 93  574 96  584  l\n",
            "        LTChar                 96  574 101 584  a\n",
            "        LTChar                 101 574 104 584  r\n",
            "        LTChar                 104 574 108 584  e\n",
            "        LTChar                 108 574 113 584  d\n",
            "        LTAnno                  \n",
            "        LTChar                 116 574 120 584  a\n",
            "        LTChar                 120 574 124 584  s\n",
            "        LTAnno                  \n",
            "        LTChar                 127 574 131 584  a\n",
            "        LTAnno                  \n",
            "        LTChar                 133 574 138 584  p\n",
            "        LTChar                 138 574 143 584  a\n",
            "        LTChar                 143 574 148 584  n\n",
            "        LTChar                 148 574 153 584  d\n",
            "        LTChar                 153 574 157 584  e\n",
            "        LTChar                 157 574 165 584  m\n",
            "        LTChar                 165 574 168 584  i\n",
            "        LTChar                 168 574 172 584  c\n",
            "        LTAnno                  \n",
            "        LTChar                 175 574 180 584  b\n",
            "        LTChar                 180 574 185 584  y\n",
            "        LTAnno                  \n",
            "        LTChar                 187 574 196 584  W\n",
            "        LTChar                 196 574 204 584  H\n",
            "        LTChar                 204 574 211 584  O\n",
            "        LTAnno                  \n",
            "        LTChar                 213 574 220 584  w\n",
            "        LTChar                 220 574 223 584  i\n",
            "        LTChar                 223 574 226 584  t\n",
            "        LTChar                 226 574 231 584  h\n",
            "        LTAnno                  \n",
            "        LTChar                 233 574 236 584  t\n",
            "        LTChar                 236 574 241 584  h\n",
            "        LTChar                 241 574 246 584  o\n",
            "        LTChar                 246 574 251 584  u\n",
            "        LTChar                 251 574 255 584  s\n",
            "        LTChar                 255 574 259 584  a\n",
            "        LTChar                 259 574 264 584  n\n",
            "        LTChar                 264 574 269 584  d\n",
            "        LTChar                 269 574 273 584  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  562 273 572  of cases being reported each day. Numerous\n",
            "        LTChar                 89  562 94  572  o\n",
            "        LTChar                 94  562 97  572  f\n",
            "        LTAnno                  \n",
            "        LTChar                 101 562 105 572  c\n",
            "        LTChar                 105 562 110 572  a\n",
            "        LTChar                 110 562 114 572  s\n",
            "        LTChar                 114 562 118 572  e\n",
            "        LTChar                 118 562 122 572  s\n",
            "        LTAnno                  \n",
            "        LTChar                 125 562 130 572  b\n",
            "        LTChar                 130 562 135 572  e\n",
            "        LTChar                 135 562 138 572  i\n",
            "        LTChar                 138 562 142 572  n\n",
            "        LTChar                 142 562 147 572  g\n",
            "        LTAnno                  \n",
            "        LTChar                 151 562 154 572  r\n",
            "        LTChar                 154 562 159 572  e\n",
            "        LTChar                 159 562 164 572  p\n",
            "        LTChar                 164 562 169 572  o\n",
            "        LTChar                 169 562 172 572  r\n",
            "        LTChar                 172 562 175 572  t\n",
            "        LTChar                 175 562 179 572  e\n",
            "        LTChar                 179 562 184 572  d\n",
            "        LTAnno                  \n",
            "        LTChar                 188 562 192 572  e\n",
            "        LTChar                 192 562 197 572  a\n",
            "        LTChar                 197 562 201 572  c\n",
            "        LTChar                 201 562 206 572  h\n",
            "        LTAnno                  \n",
            "        LTChar                 209 562 214 572  d\n",
            "        LTChar                 214 562 219 572  a\n",
            "        LTChar                 219 562 224 572  y\n",
            "        LTChar                 223 562 226 572  .\n",
            "        LTAnno                  \n",
            "        LTChar                 232 562 239 572  N\n",
            "        LTChar                 239 562 244 572  u\n",
            "        LTChar                 244 562 252 572  m\n",
            "        LTChar                 252 562 256 572  e\n",
            "        LTChar                 256 562 259 572  r\n",
            "        LTChar                 259 562 264 572  o\n",
            "        LTChar                 264 562 269 572  u\n",
            "        LTChar                 269 562 273 572  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  550 273 560  scientiﬁc articles are being published on the\n",
            "        LTChar                 89  550 93  560  s\n",
            "        LTChar                 93  550 97  560  c\n",
            "        LTChar                 97  550 100 560  i\n",
            "        LTChar                 100 550 104 560  e\n",
            "        LTChar                 104 550 109 560  n\n",
            "        LTChar                 109 550 112 560  t\n",
            "        LTChar                 112 550 115 560  i\n",
            "        LTChar                 115 550 121 560  ﬁ\n",
            "        LTChar                 121 550 125 560  c\n",
            "        LTAnno                  \n",
            "        LTChar                 129 550 133 560  a\n",
            "        LTChar                 133 550 137 560  r\n",
            "        LTChar                 137 550 140 560  t\n",
            "        LTChar                 140 550 142 560  i\n",
            "        LTChar                 142 550 147 560  c\n",
            "        LTChar                 147 550 150 560  l\n",
            "        LTChar                 150 550 154 560  e\n",
            "        LTChar                 154 550 158 560  s\n",
            "        LTAnno                  \n",
            "        LTChar                 162 550 166 560  a\n",
            "        LTChar                 166 550 170 560  r\n",
            "        LTChar                 170 550 174 560  e\n",
            "        LTAnno                  \n",
            "        LTChar                 178 550 183 560  b\n",
            "        LTChar                 183 550 187 560  e\n",
            "        LTChar                 187 550 190 560  i\n",
            "        LTChar                 190 550 195 560  n\n",
            "        LTChar                 195 550 200 560  g\n",
            "        LTAnno                  \n",
            "        LTChar                 204 550 209 560  p\n",
            "        LTChar                 209 550 214 560  u\n",
            "        LTChar                 214 550 219 560  b\n",
            "        LTChar                 219 550 222 560  l\n",
            "        LTChar                 222 550 225 560  i\n",
            "        LTChar                 225 550 229 560  s\n",
            "        LTChar                 229 550 234 560  h\n",
            "        LTChar                 234 550 238 560  e\n",
            "        LTChar                 238 550 243 560  d\n",
            "        LTAnno                  \n",
            "        LTChar                 247 550 252 560  o\n",
            "        LTChar                 252 550 257 560  n\n",
            "        LTAnno                  \n",
            "        LTChar                 261 550 264 560  t\n",
            "        LTChar                 264 550 269 560  h\n",
            "        LTChar                 269 550 273 560  e\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  538 273 548  disease raising the need for a service which\n",
            "        LTChar                 89  538 94  548  d\n",
            "        LTChar                 94  538 97  548  i\n",
            "        LTChar                 97  538 101 548  s\n",
            "        LTChar                 101 538 105 548  e\n",
            "        LTChar                 105 538 109 548  a\n",
            "        LTChar                 109 538 113 548  s\n",
            "        LTChar                 113 538 118 548  e\n",
            "        LTAnno                  \n",
            "        LTChar                 122 538 125 548  r\n",
            "        LTChar                 125 538 130 548  a\n",
            "        LTChar                 130 538 132 548  i\n",
            "        LTChar                 132 538 136 548  s\n",
            "        LTChar                 136 538 139 548  i\n",
            "        LTChar                 139 538 144 548  n\n",
            "        LTChar                 144 538 149 548  g\n",
            "        LTAnno                  \n",
            "        LTChar                 153 538 156 548  t\n",
            "        LTChar                 156 538 161 548  h\n",
            "        LTChar                 161 538 165 548  e\n",
            "        LTAnno                  \n",
            "        LTChar                 169 538 174 548  n\n",
            "        LTChar                 174 538 179 548  e\n",
            "        LTChar                 179 538 183 548  e\n",
            "        LTChar                 183 538 188 548  d\n",
            "        LTAnno                  \n",
            "        LTChar                 192 538 196 548  f\n",
            "        LTChar                 196 538 201 548  o\n",
            "        LTChar                 201 538 204 548  r\n",
            "        LTAnno                  \n",
            "        LTChar                 208 538 212 548  a\n",
            "        LTAnno                  \n",
            "        LTChar                 217 538 220 548  s\n",
            "        LTChar                 220 538 225 548  e\n",
            "        LTChar                 225 538 228 548  r\n",
            "        LTChar                 228 538 233 548  v\n",
            "        LTChar                 233 538 236 548  i\n",
            "        LTChar                 236 538 240 548  c\n",
            "        LTChar                 240 538 245 548  e\n",
            "        LTAnno                  \n",
            "        LTChar                 249 538 256 548  w\n",
            "        LTChar                 256 538 261 548  h\n",
            "        LTChar                 261 538 264 548  i\n",
            "        LTChar                 264 538 268 548  c\n",
            "        LTChar                 268 538 273 548  h\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  526 275 536  can organize, and query them in a reliable fash-\n",
            "        LTChar                 89  526 93  536  c\n",
            "        LTChar                 93  526 98  536  a\n",
            "        LTChar                 98  526 103 536  n\n",
            "        LTAnno                  \n",
            "        LTChar                 105 526 110 536  o\n",
            "        LTChar                 110 526 113 536  r\n",
            "        LTChar                 113 526 118 536  g\n",
            "        LTChar                 118 526 122 536  a\n",
            "        LTChar                 122 526 127 536  n\n",
            "        LTChar                 127 526 130 536  i\n",
            "        LTChar                 130 526 135 536  z\n",
            "        LTChar                 135 526 139 536  e\n",
            "        LTChar                 139 526 141 536  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 144 526 148 536  a\n",
            "        LTChar                 148 526 153 536  n\n",
            "        LTChar                 153 526 158 536  d\n",
            "        LTAnno                  \n",
            "        LTChar                 160 526 165 536  q\n",
            "        LTChar                 165 526 170 536  u\n",
            "        LTChar                 170 526 174 536  e\n",
            "        LTChar                 174 526 178 536  r\n",
            "        LTChar                 178 526 183 536  y\n",
            "        LTAnno                  \n",
            "        LTChar                 185 526 188 536  t\n",
            "        LTChar                 188 526 193 536  h\n",
            "        LTChar                 193 526 197 536  e\n",
            "        LTChar                 197 526 205 536  m\n",
            "        LTAnno                  \n",
            "        LTChar                 207 526 210 536  i\n",
            "        LTChar                 210 526 215 536  n\n",
            "        LTAnno                  \n",
            "        LTChar                 217 526 221 536  a\n",
            "        LTAnno                  \n",
            "        LTChar                 223 526 226 536  r\n",
            "        LTChar                 226 526 231 536  e\n",
            "        LTChar                 231 526 234 536  l\n",
            "        LTChar                 234 526 236 536  i\n",
            "        LTChar                 236 526 241 536  a\n",
            "        LTChar                 241 526 246 536  b\n",
            "        LTChar                 246 526 249 536  l\n",
            "        LTChar                 249 526 253 536  e\n",
            "        LTAnno                  \n",
            "        LTChar                 255 526 258 536  f\n",
            "        LTChar                 258 526 263 536  a\n",
            "        LTChar                 263 526 267 536  s\n",
            "        LTChar                 267 526 272 536  h\n",
            "        LTChar                 272 526 275 536  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  514 273 524  ion. To support this cause we present AWS\n",
            "        LTChar                 89  514 92  524  i\n",
            "        LTChar                 92  514 97  524  o\n",
            "        LTChar                 97  514 102 524  n\n",
            "        LTChar                 102 514 104 524  .\n",
            "        LTAnno                  \n",
            "        LTChar                 112 514 118 524  T\n",
            "        LTChar                 117 514 122 524  o\n",
            "        LTAnno                  \n",
            "        LTChar                 126 514 130 524  s\n",
            "        LTChar                 130 514 135 524  u\n",
            "        LTChar                 135 514 140 524  p\n",
            "        LTChar                 140 514 145 524  p\n",
            "        LTChar                 145 514 150 524  o\n",
            "        LTChar                 150 514 153 524  r\n",
            "        LTChar                 153 514 156 524  t\n",
            "        LTAnno                  \n",
            "        LTChar                 160 514 162 524  t\n",
            "        LTChar                 162 514 167 524  h\n",
            "        LTChar                 167 514 170 524  i\n",
            "        LTChar                 170 514 174 524  s\n",
            "        LTAnno                  \n",
            "        LTChar                 178 514 182 524  c\n",
            "        LTChar                 182 514 187 524  a\n",
            "        LTChar                 187 514 192 524  u\n",
            "        LTChar                 192 514 196 524  s\n",
            "        LTChar                 196 514 200 524  e\n",
            "        LTAnno                  \n",
            "        LTChar                 204 514 211 524  w\n",
            "        LTChar                 211 514 215 524  e\n",
            "        LTAnno                  \n",
            "        LTChar                 219 514 224 524  p\n",
            "        LTChar                 224 514 228 524  r\n",
            "        LTChar                 228 514 232 524  e\n",
            "        LTChar                 232 514 236 524  s\n",
            "        LTChar                 236 514 240 524  e\n",
            "        LTChar                 240 514 245 524  n\n",
            "        LTChar                 245 514 248 524  t\n",
            "        LTAnno                  \n",
            "        LTChar                 252 514 259 524  A\n",
            "        LTChar                 258 514 268 524  W\n",
            "        LTChar                 268 514 273 524  S\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  502 273 512  CORD-19 Search (ACS), a public, COVID-19\n",
            "        LTChar                 89  502 96  512  C\n",
            "        LTChar                 96  502 103 512  O\n",
            "        LTChar                 103 502 109 512  R\n",
            "        LTChar                 109 502 117 512  D\n",
            "        LTChar                 117 502 120 512  -\n",
            "        LTChar                 120 502 125 512  1\n",
            "        LTChar                 125 502 130 512  9\n",
            "        LTAnno                  \n",
            "        LTChar                 132 502 138 512  S\n",
            "        LTChar                 138 502 142 512  e\n",
            "        LTChar                 142 502 147 512  a\n",
            "        LTChar                 147 502 150 512  r\n",
            "        LTChar                 150 502 154 512  c\n",
            "        LTChar                 154 502 159 512  h\n",
            "        LTAnno                  \n",
            "        LTChar                 162 502 165 512  (\n",
            "        LTChar                 165 502 172 512  A\n",
            "        LTChar                 172 502 179 512  C\n",
            "        LTChar                 179 502 184 512  S\n",
            "        LTChar                 184 502 187 512  )\n",
            "        LTChar                 187 502 190 512  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 192 502 197 512  a\n",
            "        LTAnno                  \n",
            "        LTChar                 199 502 204 512  p\n",
            "        LTChar                 204 502 209 512  u\n",
            "        LTChar                 209 502 214 512  b\n",
            "        LTChar                 214 502 217 512  l\n",
            "        LTChar                 217 502 220 512  i\n",
            "        LTChar                 220 502 224 512  c\n",
            "        LTChar                 224 502 227 512  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 229 502 236 512  C\n",
            "        LTChar                 236 502 243 512  O\n",
            "        LTChar                 242 502 249 512  V\n",
            "        LTChar                 249 502 253 512  I\n",
            "        LTChar                 253 502 260 512  D\n",
            "        LTChar                 260 502 263 512  -\n",
            "        LTChar                 263 502 268 512  1\n",
            "        LTChar                 268 502 273 512  9\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  490 273 500  speciﬁc, neural search engine that is powered\n",
            "        LTChar                 89  490 93  500  s\n",
            "        LTChar                 93  490 98  500  p\n",
            "        LTChar                 98  490 102 500  e\n",
            "        LTChar                 102 490 107 500  c\n",
            "        LTChar                 107 490 109 500  i\n",
            "        LTChar                 109 490 115 500  ﬁ\n",
            "        LTChar                 115 490 119 500  c\n",
            "        LTChar                 119 490 122 500  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 125 490 130 500  n\n",
            "        LTChar                 130 490 135 500  e\n",
            "        LTChar                 135 490 140 500  u\n",
            "        LTChar                 140 490 143 500  r\n",
            "        LTChar                 143 490 147 500  a\n",
            "        LTChar                 147 490 150 500  l\n",
            "        LTAnno                  \n",
            "        LTChar                 153 490 157 500  s\n",
            "        LTChar                 157 490 161 500  e\n",
            "        LTChar                 161 490 166 500  a\n",
            "        LTChar                 166 490 169 500  r\n",
            "        LTChar                 169 490 174 500  c\n",
            "        LTChar                 174 490 179 500  h\n",
            "        LTAnno                  \n",
            "        LTChar                 182 490 186 500  e\n",
            "        LTChar                 186 490 191 500  n\n",
            "        LTChar                 191 490 196 500  g\n",
            "        LTChar                 196 490 199 500  i\n",
            "        LTChar                 199 490 204 500  n\n",
            "        LTChar                 204 490 208 500  e\n",
            "        LTAnno                  \n",
            "        LTChar                 211 490 214 500  t\n",
            "        LTChar                 214 490 219 500  h\n",
            "        LTChar                 219 490 224 500  a\n",
            "        LTChar                 224 490 226 500  t\n",
            "        LTAnno                  \n",
            "        LTChar                 229 490 232 500  i\n",
            "        LTChar                 232 490 236 500  s\n",
            "        LTAnno                  \n",
            "        LTChar                 239 490 244 500  p\n",
            "        LTChar                 244 490 249 500  o\n",
            "        LTChar                 249 490 256 500  w\n",
            "        LTChar                 256 490 261 500  e\n",
            "        LTChar                 261 490 264 500  r\n",
            "        LTChar                 264 490 268 500  e\n",
            "        LTChar                 268 490 273 500  d\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  478 275 488  by several machine learning systems to sup-\n",
            "        LTChar                 89  478 94  488  b\n",
            "        LTChar                 94  478 99  488  y\n",
            "        LTAnno                  \n",
            "        LTChar                 103 478 107 488  s\n",
            "        LTChar                 107 478 111 488  e\n",
            "        LTChar                 111 478 116 488  v\n",
            "        LTChar                 116 478 120 488  e\n",
            "        LTChar                 120 478 124 488  r\n",
            "        LTChar                 124 478 128 488  a\n",
            "        LTChar                 128 478 131 488  l\n",
            "        LTAnno                  \n",
            "        LTChar                 135 478 143 488  m\n",
            "        LTChar                 143 478 147 488  a\n",
            "        LTChar                 147 478 152 488  c\n",
            "        LTChar                 152 478 157 488  h\n",
            "        LTChar                 157 478 160 488  i\n",
            "        LTChar                 160 478 165 488  n\n",
            "        LTChar                 165 478 169 488  e\n",
            "        LTAnno                  \n",
            "        LTChar                 173 478 176 488  l\n",
            "        LTChar                 176 478 180 488  e\n",
            "        LTChar                 180 478 185 488  a\n",
            "        LTChar                 185 478 188 488  r\n",
            "        LTChar                 188 478 193 488  n\n",
            "        LTChar                 193 478 196 488  i\n",
            "        LTChar                 196 478 201 488  n\n",
            "        LTChar                 201 478 206 488  g\n",
            "        LTAnno                  \n",
            "        LTChar                 210 478 214 488  s\n",
            "        LTChar                 214 478 219 488  y\n",
            "        LTChar                 219 478 223 488  s\n",
            "        LTChar                 223 478 226 488  t\n",
            "        LTChar                 226 478 230 488  e\n",
            "        LTChar                 230 478 238 488  m\n",
            "        LTChar                 238 478 242 488  s\n",
            "        LTAnno                  \n",
            "        LTChar                 246 478 249 488  t\n",
            "        LTChar                 249 478 254 488  o\n",
            "        LTAnno                  \n",
            "        LTChar                 258 478 262 488  s\n",
            "        LTChar                 262 478 267 488  u\n",
            "        LTChar                 267 478 272 488  p\n",
            "        LTChar                 272 478 275 488  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  466 273 476  port natural language based searches. ACS\n",
            "        LTChar                 89  466 94  476  p\n",
            "        LTChar                 94  466 99  476  o\n",
            "        LTChar                 99  466 102 476  r\n",
            "        LTChar                 102 466 105 476  t\n",
            "        LTAnno                  \n",
            "        LTChar                 109 466 114 476  n\n",
            "        LTChar                 114 466 119 476  a\n",
            "        LTChar                 119 466 122 476  t\n",
            "        LTChar                 122 466 127 476  u\n",
            "        LTChar                 127 466 130 476  r\n",
            "        LTChar                 130 466 134 476  a\n",
            "        LTChar                 134 466 137 476  l\n",
            "        LTAnno                  \n",
            "        LTChar                 142 466 144 476  l\n",
            "        LTChar                 144 466 149 476  a\n",
            "        LTChar                 149 466 154 476  n\n",
            "        LTChar                 154 466 159 476  g\n",
            "        LTChar                 159 466 164 476  u\n",
            "        LTChar                 164 466 168 476  a\n",
            "        LTChar                 168 466 173 476  g\n",
            "        LTChar                 173 466 178 476  e\n",
            "        LTAnno                  \n",
            "        LTChar                 182 466 187 476  b\n",
            "        LTChar                 187 466 191 476  a\n",
            "        LTChar                 191 466 195 476  s\n",
            "        LTChar                 195 466 200 476  e\n",
            "        LTChar                 200 466 205 476  d\n",
            "        LTAnno                  \n",
            "        LTChar                 209 466 213 476  s\n",
            "        LTChar                 213 466 217 476  e\n",
            "        LTChar                 217 466 222 476  a\n",
            "        LTChar                 222 466 225 476  r\n",
            "        LTChar                 225 466 230 476  c\n",
            "        LTChar                 230 466 235 476  h\n",
            "        LTChar                 235 466 239 476  e\n",
            "        LTChar                 239 466 243 476  s\n",
            "        LTChar                 243 466 245 476  .\n",
            "        LTAnno                  \n",
            "        LTChar                 254 466 261 476  A\n",
            "        LTChar                 261 466 268 476  C\n",
            "        LTChar                 268 466 273 476  S\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  454 275 464  with capabilities such as document ranking,\n",
            "        LTChar                 89  454 96  464  w\n",
            "        LTChar                 96  454 99  464  i\n",
            "        LTChar                 99  454 101 464  t\n",
            "        LTChar                 101 454 106 464  h\n",
            "        LTAnno                  \n",
            "        LTChar                 111 454 116 464  c\n",
            "        LTChar                 116 454 120 464  a\n",
            "        LTChar                 120 454 125 464  p\n",
            "        LTChar                 125 454 129 464  a\n",
            "        LTChar                 129 454 134 464  b\n",
            "        LTChar                 134 454 137 464  i\n",
            "        LTChar                 137 454 140 464  l\n",
            "        LTChar                 140 454 143 464  i\n",
            "        LTChar                 143 454 145 464  t\n",
            "        LTChar                 145 454 148 464  i\n",
            "        LTChar                 148 454 153 464  e\n",
            "        LTChar                 153 454 157 464  s\n",
            "        LTAnno                  \n",
            "        LTChar                 161 454 165 464  s\n",
            "        LTChar                 165 454 170 464  u\n",
            "        LTChar                 170 454 175 464  c\n",
            "        LTChar                 175 454 180 464  h\n",
            "        LTAnno                  \n",
            "        LTChar                 184 454 189 464  a\n",
            "        LTChar                 189 454 193 464  s\n",
            "        LTAnno                  \n",
            "        LTChar                 197 454 202 464  d\n",
            "        LTChar                 202 454 207 464  o\n",
            "        LTChar                 207 454 212 464  c\n",
            "        LTChar                 212 454 217 464  u\n",
            "        LTChar                 217 454 225 464  m\n",
            "        LTChar                 225 454 229 464  e\n",
            "        LTChar                 229 454 234 464  n\n",
            "        LTChar                 234 454 237 464  t\n",
            "        LTAnno                  \n",
            "        LTChar                 242 454 245 464  r\n",
            "        LTChar                 245 454 249 464  a\n",
            "        LTChar                 249 454 254 464  n\n",
            "        LTChar                 254 454 259 464  k\n",
            "        LTChar                 259 454 262 464  i\n",
            "        LTChar                 262 454 267 464  n\n",
            "        LTChar                 267 454 272 464  g\n",
            "        LTChar                 272 454 275 464  ,\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  442 273 452  passage ranking, question answering and topic\n",
            "        LTChar                 89  442 94  452  p\n",
            "        LTChar                 94  442 98  452  a\n",
            "        LTChar                 98  442 102 452  s\n",
            "        LTChar                 102 442 106 452  s\n",
            "        LTChar                 106 442 111 452  a\n",
            "        LTChar                 111 442 116 452  g\n",
            "        LTChar                 116 442 120 452  e\n",
            "        LTAnno                  \n",
            "        LTChar                 122 442 126 452  r\n",
            "        LTChar                 126 442 130 452  a\n",
            "        LTChar                 130 442 135 452  n\n",
            "        LTChar                 135 442 140 452  k\n",
            "        LTChar                 140 442 143 452  i\n",
            "        LTChar                 143 442 148 452  n\n",
            "        LTChar                 148 442 153 452  g\n",
            "        LTChar                 153 442 155 452  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 157 442 162 452  q\n",
            "        LTChar                 162 442 167 452  u\n",
            "        LTChar                 167 442 172 452  e\n",
            "        LTChar                 172 442 176 452  s\n",
            "        LTChar                 176 442 179 452  t\n",
            "        LTChar                 179 442 181 452  i\n",
            "        LTChar                 181 442 186 452  o\n",
            "        LTChar                 186 442 191 452  n\n",
            "        LTAnno                  \n",
            "        LTChar                 193 442 198 452  a\n",
            "        LTChar                 198 442 203 452  n\n",
            "        LTChar                 203 442 207 452  s\n",
            "        LTChar                 207 442 214 452  w\n",
            "        LTChar                 214 442 218 452  e\n",
            "        LTChar                 218 442 222 452  r\n",
            "        LTChar                 222 442 224 452  i\n",
            "        LTChar                 224 442 229 452  n\n",
            "        LTChar                 229 442 234 452  g\n",
            "        LTAnno                  \n",
            "        LTChar                 237 442 241 452  a\n",
            "        LTChar                 241 442 246 452  n\n",
            "        LTChar                 246 442 251 452  d\n",
            "        LTAnno                  \n",
            "        LTChar                 253 442 256 452  t\n",
            "        LTChar                 256 442 261 452  o\n",
            "        LTChar                 261 442 266 452  p\n",
            "        LTChar                 266 442 269 452  i\n",
            "        LTChar                 269 442 273 452  c\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  431 273 440  classiﬁcation provides a scalable solution to\n",
            "        LTChar                 89  431 93  440  c\n",
            "        LTChar                 93  431 96  440  l\n",
            "        LTChar                 96  431 101 440  a\n",
            "        LTChar                 101 431 104 440  s\n",
            "        LTChar                 104 431 108 440  s\n",
            "        LTChar                 108 431 111 440  i\n",
            "        LTChar                 111 431 117 440  ﬁ\n",
            "        LTChar                 117 431 121 440  c\n",
            "        LTChar                 121 431 126 440  a\n",
            "        LTChar                 126 431 128 440  t\n",
            "        LTChar                 128 431 131 440  i\n",
            "        LTChar                 131 431 136 440  o\n",
            "        LTChar                 136 431 141 440  n\n",
            "        LTAnno                  \n",
            "        LTChar                 145 431 150 440  p\n",
            "        LTChar                 150 431 154 440  r\n",
            "        LTChar                 154 431 159 440  o\n",
            "        LTChar                 158 431 163 440  v\n",
            "        LTChar                 163 431 166 440  i\n",
            "        LTChar                 166 431 171 440  d\n",
            "        LTChar                 171 431 176 440  e\n",
            "        LTChar                 176 431 180 440  s\n",
            "        LTAnno                  \n",
            "        LTChar                 184 431 188 440  a\n",
            "        LTAnno                  \n",
            "        LTChar                 193 431 197 440  s\n",
            "        LTChar                 197 431 201 440  c\n",
            "        LTChar                 201 431 205 440  a\n",
            "        LTChar                 205 431 208 440  l\n",
            "        LTChar                 208 431 213 440  a\n",
            "        LTChar                 213 431 218 440  b\n",
            "        LTChar                 218 431 220 440  l\n",
            "        LTChar                 220 431 225 440  e\n",
            "        LTAnno                  \n",
            "        LTChar                 229 431 233 440  s\n",
            "        LTChar                 233 431 238 440  o\n",
            "        LTChar                 238 431 241 440  l\n",
            "        LTChar                 241 431 246 440  u\n",
            "        LTChar                 246 431 248 440  t\n",
            "        LTChar                 248 431 251 440  i\n",
            "        LTChar                 251 431 256 440  o\n",
            "        LTChar                 256 431 261 440  n\n",
            "        LTAnno                  \n",
            "        LTChar                 266 431 268 440  t\n",
            "        LTChar                 268 431 273 440  o\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  419 273 429  COVID-19 researchers and policy makers in\n",
            "        LTChar                 89  419 96  429  C\n",
            "        LTChar                 96  419 103 429  O\n",
            "        LTChar                 102 419 110 429  V\n",
            "        LTChar                 110 419 113 429  I\n",
            "        LTChar                 113 419 120 429  D\n",
            "        LTChar                 120 419 123 429  -\n",
            "        LTChar                 123 419 128 429  1\n",
            "        LTChar                 128 419 133 429  9\n",
            "        LTAnno                  \n",
            "        LTChar                 137 419 141 429  r\n",
            "        LTChar                 141 419 145 429  e\n",
            "        LTChar                 145 419 149 429  s\n",
            "        LTChar                 149 419 153 429  e\n",
            "        LTChar                 153 419 158 429  a\n",
            "        LTChar                 158 419 161 429  r\n",
            "        LTChar                 161 419 165 429  c\n",
            "        LTChar                 165 419 170 429  h\n",
            "        LTChar                 170 419 175 429  e\n",
            "        LTChar                 175 419 178 429  r\n",
            "        LTChar                 178 419 182 429  s\n",
            "        LTAnno                  \n",
            "        LTChar                 186 419 190 429  a\n",
            "        LTChar                 190 419 195 429  n\n",
            "        LTChar                 195 419 200 429  d\n",
            "        LTAnno                  \n",
            "        LTChar                 204 419 209 429  p\n",
            "        LTChar                 209 419 214 429  o\n",
            "        LTChar                 214 419 217 429  l\n",
            "        LTChar                 217 419 220 429  i\n",
            "        LTChar                 220 419 224 429  c\n",
            "        LTChar                 224 419 229 429  y\n",
            "        LTAnno                  \n",
            "        LTChar                 233 419 241 429  m\n",
            "        LTChar                 241 419 245 429  a\n",
            "        LTChar                 245 419 250 429  k\n",
            "        LTChar                 250 419 254 429  e\n",
            "        LTChar                 254 419 258 429  r\n",
            "        LTChar                 258 419 262 429  s\n",
            "        LTAnno                  \n",
            "        LTChar                 266 419 268 429  i\n",
            "        LTChar                 268 419 273 429  n\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  407 273 417  their search and discovery for answers to high\n",
            "        LTChar                 89  407 92  417  t\n",
            "        LTChar                 92  407 97  417  h\n",
            "        LTChar                 97  407 101 417  e\n",
            "        LTChar                 101 407 104 417  i\n",
            "        LTChar                 104 407 107 417  r\n",
            "        LTAnno                  \n",
            "        LTChar                 110 407 114 417  s\n",
            "        LTChar                 114 407 118 417  e\n",
            "        LTChar                 118 407 123 417  a\n",
            "        LTChar                 123 407 126 417  r\n",
            "        LTChar                 126 407 130 417  c\n",
            "        LTChar                 130 407 135 417  h\n",
            "        LTAnno                  \n",
            "        LTChar                 138 407 142 417  a\n",
            "        LTChar                 142 407 147 417  n\n",
            "        LTChar                 147 407 152 417  d\n",
            "        LTAnno                  \n",
            "        LTChar                 155 407 160 417  d\n",
            "        LTChar                 160 407 163 417  i\n",
            "        LTChar                 163 407 167 417  s\n",
            "        LTChar                 167 407 171 417  c\n",
            "        LTChar                 171 407 176 417  o\n",
            "        LTChar                 176 407 181 417  v\n",
            "        LTChar                 181 407 185 417  e\n",
            "        LTChar                 185 407 189 417  r\n",
            "        LTChar                 189 407 193 417  y\n",
            "        LTAnno                  \n",
            "        LTChar                 196 407 199 417  f\n",
            "        LTChar                 199 407 204 417  o\n",
            "        LTChar                 204 407 208 417  r\n",
            "        LTAnno                  \n",
            "        LTChar                 210 407 215 417  a\n",
            "        LTChar                 215 407 220 417  n\n",
            "        LTChar                 220 407 224 417  s\n",
            "        LTChar                 224 407 231 417  w\n",
            "        LTChar                 231 407 235 417  e\n",
            "        LTChar                 235 407 239 417  r\n",
            "        LTChar                 239 407 242 417  s\n",
            "        LTAnno                  \n",
            "        LTChar                 245 407 248 417  t\n",
            "        LTChar                 248 407 253 417  o\n",
            "        LTAnno                  \n",
            "        LTChar                 256 407 261 417  h\n",
            "        LTChar                 261 407 263 417  i\n",
            "        LTChar                 263 407 268 417  g\n",
            "        LTChar                 268 407 273 417  h\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  395 273 405  priority scientiﬁc questions. We present a\n",
            "        LTChar                 89  395 94  405  p\n",
            "        LTChar                 94  395 97  405  r\n",
            "        LTChar                 97  395 100 405  i\n",
            "        LTChar                 100 395 105 405  o\n",
            "        LTChar                 105 395 108 405  r\n",
            "        LTChar                 108 395 111 405  i\n",
            "        LTChar                 111 395 114 405  t\n",
            "        LTChar                 114 395 119 405  y\n",
            "        LTAnno                  \n",
            "        LTChar                 124 395 128 405  s\n",
            "        LTChar                 128 395 132 405  c\n",
            "        LTChar                 132 395 135 405  i\n",
            "        LTChar                 135 395 140 405  e\n",
            "        LTChar                 140 395 145 405  n\n",
            "        LTChar                 145 395 147 405  t\n",
            "        LTChar                 147 395 150 405  i\n",
            "        LTChar                 150 395 156 405  ﬁ\n",
            "        LTChar                 156 395 160 405  c\n",
            "        LTAnno                  \n",
            "        LTChar                 165 395 170 405  q\n",
            "        LTChar                 170 395 175 405  u\n",
            "        LTChar                 175 395 180 405  e\n",
            "        LTChar                 180 395 184 405  s\n",
            "        LTChar                 184 395 186 405  t\n",
            "        LTChar                 186 395 189 405  i\n",
            "        LTChar                 189 395 194 405  o\n",
            "        LTChar                 194 395 199 405  n\n",
            "        LTChar                 199 395 203 405  s\n",
            "        LTChar                 203 395 205 405  .\n",
            "        LTAnno                  \n",
            "        LTChar                 217 395 226 405  W\n",
            "        LTChar                 225 395 230 405  e\n",
            "        LTAnno                  \n",
            "        LTChar                 235 395 240 405  p\n",
            "        LTChar                 240 395 243 405  r\n",
            "        LTChar                 243 395 248 405  e\n",
            "        LTChar                 248 395 251 405  s\n",
            "        LTChar                 251 395 256 405  e\n",
            "        LTChar                 256 395 261 405  n\n",
            "        LTChar                 261 395 264 405  t\n",
            "        LTAnno                  \n",
            "        LTChar                 269 395 273 405  a\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  383 273 393  quantitative evaluation and qualitative analysis\n",
            "        LTChar                 89  383 94  393  q\n",
            "        LTChar                 94  383 99  393  u\n",
            "        LTChar                 99  383 103 393  a\n",
            "        LTChar                 103 383 108 393  n\n",
            "        LTChar                 108 383 111 393  t\n",
            "        LTChar                 111 383 114 393  i\n",
            "        LTChar                 114 383 117 393  t\n",
            "        LTChar                 117 383 121 393  a\n",
            "        LTChar                 121 383 124 393  t\n",
            "        LTChar                 124 383 127 393  i\n",
            "        LTChar                 126 383 131 393  v\n",
            "        LTChar                 131 383 136 393  e\n",
            "        LTAnno                  \n",
            "        LTChar                 138 383 142 393  e\n",
            "        LTChar                 142 383 147 393  v\n",
            "        LTChar                 147 383 151 393  a\n",
            "        LTChar                 151 383 154 393  l\n",
            "        LTChar                 154 383 159 393  u\n",
            "        LTChar                 159 383 163 393  a\n",
            "        LTChar                 163 383 166 393  t\n",
            "        LTChar                 166 383 169 393  i\n",
            "        LTChar                 169 383 174 393  o\n",
            "        LTChar                 174 383 179 393  n\n",
            "        LTAnno                  \n",
            "        LTChar                 181 383 185 393  a\n",
            "        LTChar                 185 383 190 393  n\n",
            "        LTChar                 190 383 195 393  d\n",
            "        LTAnno                  \n",
            "        LTChar                 197 383 202 393  q\n",
            "        LTChar                 202 383 207 393  u\n",
            "        LTChar                 207 383 212 393  a\n",
            "        LTChar                 212 383 215 393  l\n",
            "        LTChar                 215 383 217 393  i\n",
            "        LTChar                 217 383 220 393  t\n",
            "        LTChar                 220 383 224 393  a\n",
            "        LTChar                 224 383 227 393  t\n",
            "        LTChar                 227 383 230 393  i\n",
            "        LTChar                 230 383 235 393  v\n",
            "        LTChar                 235 383 239 393  e\n",
            "        LTAnno                  \n",
            "        LTChar                 241 383 246 393  a\n",
            "        LTChar                 246 383 251 393  n\n",
            "        LTChar                 251 383 255 393  a\n",
            "        LTChar                 255 383 258 393  l\n",
            "        LTChar                 258 383 263 393  y\n",
            "        LTChar                 263 383 267 393  s\n",
            "        LTChar                 267 383 269 393  i\n",
            "        LTChar                 269 383 273 393  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  371 275 381  of the system against other leading COVID-\n",
            "        LTChar                 89  371 94  381  o\n",
            "        LTChar                 94  371 97  381  f\n",
            "        LTAnno                  \n",
            "        LTChar                 102 371 104 381  t\n",
            "        LTChar                 104 371 109 381  h\n",
            "        LTChar                 109 371 114 381  e\n",
            "        LTAnno                  \n",
            "        LTChar                 118 371 122 381  s\n",
            "        LTChar                 122 371 127 381  y\n",
            "        LTChar                 127 371 131 381  s\n",
            "        LTChar                 131 371 133 381  t\n",
            "        LTChar                 133 371 138 381  e\n",
            "        LTChar                 138 371 146 381  m\n",
            "        LTAnno                  \n",
            "        LTChar                 150 371 154 381  a\n",
            "        LTChar                 154 371 159 381  g\n",
            "        LTChar                 159 371 164 381  a\n",
            "        LTChar                 164 371 166 381  i\n",
            "        LTChar                 166 371 171 381  n\n",
            "        LTChar                 171 371 175 381  s\n",
            "        LTChar                 175 371 178 381  t\n",
            "        LTAnno                  \n",
            "        LTChar                 182 371 187 381  o\n",
            "        LTChar                 187 371 190 381  t\n",
            "        LTChar                 190 371 195 381  h\n",
            "        LTChar                 195 371 199 381  e\n",
            "        LTChar                 199 371 203 381  r\n",
            "        LTAnno                  \n",
            "        LTChar                 207 371 210 381  l\n",
            "        LTChar                 210 371 214 381  e\n",
            "        LTChar                 214 371 219 381  a\n",
            "        LTChar                 219 371 224 381  d\n",
            "        LTChar                 224 371 226 381  i\n",
            "        LTChar                 226 371 231 381  n\n",
            "        LTChar                 231 371 236 381  g\n",
            "        LTAnno                  \n",
            "        LTChar                 241 371 247 381  C\n",
            "        LTChar                 247 371 254 381  O\n",
            "        LTChar                 254 371 261 381  V\n",
            "        LTChar                 261 371 264 381  I\n",
            "        LTChar                 264 371 272 381  D\n",
            "        LTChar                 272 371 275 381  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     88  359 273 369  19 search platforms. ACS is top performing\n",
            "        LTChar                 88  359 93  369  1\n",
            "        LTChar                 93  359 98  369  9\n",
            "        LTAnno                  \n",
            "        LTChar                 102 359 106 369  s\n",
            "        LTChar                 106 359 110 369  e\n",
            "        LTChar                 110 359 115 369  a\n",
            "        LTChar                 115 359 118 369  r\n",
            "        LTChar                 118 359 122 369  c\n",
            "        LTChar                 122 359 127 369  h\n",
            "        LTAnno                  \n",
            "        LTChar                 131 359 136 369  p\n",
            "        LTChar                 136 359 139 369  l\n",
            "        LTChar                 139 359 143 369  a\n",
            "        LTChar                 143 359 146 369  t\n",
            "        LTChar                 146 359 149 369  f\n",
            "        LTChar                 149 359 154 369  o\n",
            "        LTChar                 154 359 158 369  r\n",
            "        LTChar                 158 359 166 369  m\n",
            "        LTChar                 166 359 169 369  s\n",
            "        LTChar                 169 359 172 369  .\n",
            "        LTAnno                  \n",
            "        LTChar                 179 359 186 369  A\n",
            "        LTChar                 186 359 192 369  C\n",
            "        LTChar                 192 359 198 369  S\n",
            "        LTAnno                  \n",
            "        LTChar                 202 359 204 369  i\n",
            "        LTChar                 204 359 208 369  s\n",
            "        LTAnno                  \n",
            "        LTChar                 212 359 215 369  t\n",
            "        LTChar                 215 359 220 369  o\n",
            "        LTChar                 220 359 225 369  p\n",
            "        LTAnno                  \n",
            "        LTChar                 228 359 233 369  p\n",
            "        LTChar                 233 359 238 369  e\n",
            "        LTChar                 238 359 241 369  r\n",
            "        LTChar                 241 359 244 369  f\n",
            "        LTChar                 244 359 249 369  o\n",
            "        LTChar                 249 359 253 369  r\n",
            "        LTChar                 253 359 261 369  m\n",
            "        LTChar                 261 359 263 369  i\n",
            "        LTChar                 263 359 268 369  n\n",
            "        LTChar                 268 359 273 369  g\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  347 273 357  across these systems yielding quality results\n",
            "        LTChar                 89  347 93  357  a\n",
            "        LTChar                 93  347 98  357  c\n",
            "        LTChar                 98  347 101 357  r\n",
            "        LTChar                 101 347 106 357  o\n",
            "        LTChar                 106 347 110 357  s\n",
            "        LTChar                 110 347 114 357  s\n",
            "        LTAnno                  \n",
            "        LTChar                 118 347 121 357  t\n",
            "        LTChar                 121 347 126 357  h\n",
            "        LTChar                 126 347 130 357  e\n",
            "        LTChar                 130 347 134 357  s\n",
            "        LTChar                 134 347 139 357  e\n",
            "        LTAnno                  \n",
            "        LTChar                 143 347 147 357  s\n",
            "        LTChar                 147 347 152 357  y\n",
            "        LTChar                 152 347 156 357  s\n",
            "        LTChar                 156 347 158 357  t\n",
            "        LTChar                 158 347 163 357  e\n",
            "        LTChar                 163 347 170 357  m\n",
            "        LTChar                 170 347 174 357  s\n",
            "        LTAnno                  \n",
            "        LTChar                 179 347 184 357  y\n",
            "        LTChar                 184 347 186 357  i\n",
            "        LTChar                 186 347 191 357  e\n",
            "        LTChar                 191 347 193 357  l\n",
            "        LTChar                 193 347 198 357  d\n",
            "        LTChar                 198 347 201 357  i\n",
            "        LTChar                 201 347 206 357  n\n",
            "        LTChar                 206 347 211 357  g\n",
            "        LTAnno                  \n",
            "        LTChar                 215 347 220 357  q\n",
            "        LTChar                 220 347 225 357  u\n",
            "        LTChar                 225 347 230 357  a\n",
            "        LTChar                 230 347 233 357  l\n",
            "        LTChar                 233 347 235 357  i\n",
            "        LTChar                 235 347 238 357  t\n",
            "        LTChar                 238 347 243 357  y\n",
            "        LTAnno                  \n",
            "        LTChar                 247 347 251 357  r\n",
            "        LTChar                 251 347 255 357  e\n",
            "        LTChar                 255 347 259 357  s\n",
            "        LTChar                 259 347 264 357  u\n",
            "        LTChar                 264 347 267 357  l\n",
            "        LTChar                 267 347 269 357  t\n",
            "        LTChar                 269 347 273 357  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  335 273 345  which we detail with relevant examples in this\n",
            "        LTChar                 89  335 96  345  w\n",
            "        LTChar                 96  335 101 345  h\n",
            "        LTChar                 101 335 104 345  i\n",
            "        LTChar                 104 335 108 345  c\n",
            "        LTChar                 108 335 113 345  h\n",
            "        LTAnno                  \n",
            "        LTChar                 116 335 123 345  w\n",
            "        LTChar                 123 335 127 345  e\n",
            "        LTAnno                  \n",
            "        LTChar                 130 335 135 345  d\n",
            "        LTChar                 135 335 139 345  e\n",
            "        LTChar                 139 335 142 345  t\n",
            "        LTChar                 142 335 146 345  a\n",
            "        LTChar                 146 335 149 345  i\n",
            "        LTChar                 149 335 152 345  l\n",
            "        LTAnno                  \n",
            "        LTChar                 154 335 161 345  w\n",
            "        LTChar                 161 335 164 345  i\n",
            "        LTChar                 164 335 167 345  t\n",
            "        LTChar                 167 335 172 345  h\n",
            "        LTAnno                  \n",
            "        LTChar                 175 335 178 345  r\n",
            "        LTChar                 178 335 182 345  e\n",
            "        LTChar                 182 335 185 345  l\n",
            "        LTChar                 185 335 189 345  e\n",
            "        LTChar                 189 335 194 345  v\n",
            "        LTChar                 194 335 198 345  a\n",
            "        LTChar                 198 335 203 345  n\n",
            "        LTChar                 203 335 206 345  t\n",
            "        LTAnno                  \n",
            "        LTChar                 209 335 213 345  e\n",
            "        LTChar                 213 335 218 345  x\n",
            "        LTChar                 218 335 222 345  a\n",
            "        LTChar                 222 335 230 345  m\n",
            "        LTChar                 230 335 235 345  p\n",
            "        LTChar                 235 335 238 345  l\n",
            "        LTChar                 238 335 242 345  e\n",
            "        LTChar                 242 335 246 345  s\n",
            "        LTAnno                  \n",
            "        LTChar                 249 335 251 345  i\n",
            "        LTChar                 251 335 256 345  n\n",
            "        LTAnno                  \n",
            "        LTChar                 259 335 262 345  t\n",
            "        LTChar                 262 335 267 345  h\n",
            "        LTChar                 267 335 269 345  i\n",
            "        LTChar                 269 335 273 345  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     89  323 112 333  work.\n",
            "        LTChar                 89  323 96  333  w\n",
            "        LTChar                 96  323 101 333  o\n",
            "        LTChar                 101 323 104 333  r\n",
            "        LTChar                 104 323 109 333  k\n",
            "        LTChar                 109 323 112 333  .\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        72  300 78  312  1\n",
            "      LTTextLineHorizontal     72  300 78  312  1\n",
            "        LTChar                 72  300 78  312  1\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        90  300 155 312  Introduction\n",
            "      LTTextLineHorizontal     90  300 155 312  Introduction\n",
            "        LTChar                 90  300 95  312  I\n",
            "        LTChar                 95  300 101 312  n\n",
            "        LTChar                 101 300 105 312  t\n",
            "        LTChar                 105 300 111 312  r\n",
            "        LTChar                 110 300 116 312  o\n",
            "        LTChar                 116 300 123 312  d\n",
            "        LTChar                 123 300 130 312  u\n",
            "        LTChar                 130 300 135 312  c\n",
            "        LTChar                 135 300 139 312  t\n",
            "        LTChar                 139 300 142 312  i\n",
            "        LTChar                 142 300 148 312  o\n",
            "        LTChar                 148 300 155 312  n\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        71  77  292 291  With the global outbreak of Coronavirus disease\n",
            "(COVID-19) (Guan et al., 2020), the world is in\n",
            "turmoil. Medical researchers are required to work\n",
            "quickly to fully understand and to provide a form of\n",
            "intervention for the virus. Due to a large research\n",
            "focus on the disease, knowledge is published at a\n",
            "rapid rate throughout the world. One such reposi-\n",
            "tory of information is curated through the COVID-\n",
            "19 Open Research Dataset Challenge (CORD-19)\n",
            "(Wang et al., 2020). CORD-19 is a joint challenge\n",
            "put forth by Allen Institute (AI2), National Insti-\n",
            "tutes of Health (NIH), and the United States federal\n",
            "government via the White House. The objective of\n",
            "the challenge is to make sense of and extract useful\n",
            "knowledge across thousands of scholarly articles\n",
            "related to COVID-19. CORD-19 aims to connect\n",
            "      LTTextLineHorizontal     71  280 290 291  With the global outbreak of Coronavirus disease\n",
            "        LTChar                 71  280 82  291  W\n",
            "        LTChar                 82  280 85  291  i\n",
            "        LTChar                 85  280 88  291  t\n",
            "        LTChar                 88  280 93  291  h\n",
            "        LTAnno                  \n",
            "        LTChar                 97  280 100 291  t\n",
            "        LTChar                 100 280 105 291  h\n",
            "        LTChar                 105 280 110 291  e\n",
            "        LTAnno                  \n",
            "        LTChar                 114 280 119 291  g\n",
            "        LTChar                 119 280 122 291  l\n",
            "        LTChar                 122 280 128 291  o\n",
            "        LTChar                 128 280 133 291  b\n",
            "        LTChar                 133 280 138 291  a\n",
            "        LTChar                 138 280 142 291  l\n",
            "        LTAnno                  \n",
            "        LTChar                 145 280 150 291  o\n",
            "        LTChar                 150 280 156 291  u\n",
            "        LTChar                 156 280 159 291  t\n",
            "        LTChar                 159 280 165 291  b\n",
            "        LTChar                 165 280 168 291  r\n",
            "        LTChar                 168 280 173 291  e\n",
            "        LTChar                 173 280 178 291  a\n",
            "        LTChar                 178 280 184 291  k\n",
            "        LTAnno                  \n",
            "        LTChar                 187 280 193 291  o\n",
            "        LTChar                 193 280 197 291  f\n",
            "        LTAnno                  \n",
            "        LTChar                 200 280 207 291  C\n",
            "        LTChar                 207 280 213 291  o\n",
            "        LTChar                 213 280 217 291  r\n",
            "        LTChar                 217 280 222 291  o\n",
            "        LTChar                 222 280 228 291  n\n",
            "        LTChar                 228 280 233 291  a\n",
            "        LTChar                 232 280 238 291  v\n",
            "        LTChar                 238 280 241 291  i\n",
            "        LTChar                 241 280 245 291  r\n",
            "        LTChar                 245 280 250 291  u\n",
            "        LTChar                 250 280 255 291  s\n",
            "        LTAnno                  \n",
            "        LTChar                 258 280 264 291  d\n",
            "        LTChar                 264 280 267 291  i\n",
            "        LTChar                 267 280 271 291  s\n",
            "        LTChar                 271 280 276 291  e\n",
            "        LTChar                 276 280 281 291  a\n",
            "        LTChar                 281 280 285 291  s\n",
            "        LTChar                 285 280 290 291  e\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  266 290 277  (COVID-19) (Guan et al., 2020), the world is in\n",
            "        LTChar                 72  266 75  277  (\n",
            "        LTChar                 75  266 83  277  C\n",
            "        LTChar                 83  266 91  277  O\n",
            "        LTChar                 90  266 98  277  V\n",
            "        LTChar                 98  266 102 277  I\n",
            "        LTChar                 102 266 110 277  D\n",
            "        LTChar                 110 266 114 277  -\n",
            "        LTChar                 114 266 119 277  1\n",
            "        LTChar                 119 266 125 277  9\n",
            "        LTChar                 125 266 129 277  )\n",
            "        LTAnno                  \n",
            "        LTChar                 132 266 136 277  (\n",
            "        LTChar                 136 266 144 277  G\n",
            "        LTChar                 144 266 149 277  u\n",
            "        LTChar                 149 266 154 277  a\n",
            "        LTChar                 154 266 160 277  n\n",
            "        LTAnno                  \n",
            "        LTChar                 163 266 168 277  e\n",
            "        LTChar                 168 266 171 277  t\n",
            "        LTAnno                  \n",
            "        LTChar                 175 266 180 277  a\n",
            "        LTChar                 180 266 183 277  l\n",
            "        LTChar                 183 266 186 277  .\n",
            "        LTChar                 186 266 188 277  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 192 266 197 277  2\n",
            "        LTChar                 197 266 203 277  0\n",
            "        LTChar                 203 266 209 277  2\n",
            "        LTChar                 209 266 214 277  0\n",
            "        LTChar                 214 266 218 277  )\n",
            "        LTChar                 218 266 221 277  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 224 266 227 277  t\n",
            "        LTChar                 227 266 233 277  h\n",
            "        LTChar                 233 266 238 277  e\n",
            "        LTAnno                  \n",
            "        LTChar                 241 266 249 277  w\n",
            "        LTChar                 249 266 255 277  o\n",
            "        LTChar                 255 266 259 277  r\n",
            "        LTChar                 259 266 262 277  l\n",
            "        LTChar                 262 266 267 277  d\n",
            "        LTAnno                  \n",
            "        LTChar                 271 266 274 277  i\n",
            "        LTChar                 274 266 278 277  s\n",
            "        LTAnno                  \n",
            "        LTChar                 282 266 285 277  i\n",
            "        LTChar                 285 266 290 277  n\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  253 291 264  turmoil. Medical researchers are required to work\n",
            "        LTChar                 72  253 75  264  t\n",
            "        LTChar                 75  253 81  264  u\n",
            "        LTChar                 81  253 84  264  r\n",
            "        LTChar                 84  253 93  264  m\n",
            "        LTChar                 93  253 98  264  o\n",
            "        LTChar                 98  253 101 264  i\n",
            "        LTChar                 101 253 104 264  l\n",
            "        LTChar                 104 253 107 264  .\n",
            "        LTAnno                  \n",
            "        LTChar                 110 253 120 264  M\n",
            "        LTChar                 120 253 125 264  e\n",
            "        LTChar                 125 253 131 264  d\n",
            "        LTChar                 131 253 134 264  i\n",
            "        LTChar                 134 253 139 264  c\n",
            "        LTChar                 139 253 143 264  a\n",
            "        LTChar                 143 253 146 264  l\n",
            "        LTAnno                  \n",
            "        LTChar                 149 253 153 264  r\n",
            "        LTChar                 153 253 158 264  e\n",
            "        LTChar                 158 253 162 264  s\n",
            "        LTChar                 162 253 167 264  e\n",
            "        LTChar                 167 253 172 264  a\n",
            "        LTChar                 172 253 175 264  r\n",
            "        LTChar                 175 253 180 264  c\n",
            "        LTChar                 180 253 186 264  h\n",
            "        LTChar                 186 253 191 264  e\n",
            "        LTChar                 191 253 194 264  r\n",
            "        LTChar                 194 253 199 264  s\n",
            "        LTAnno                  \n",
            "        LTChar                 201 253 206 264  a\n",
            "        LTChar                 206 253 210 264  r\n",
            "        LTChar                 210 253 215 264  e\n",
            "        LTAnno                  \n",
            "        LTChar                 217 253 221 264  r\n",
            "        LTChar                 221 253 226 264  e\n",
            "        LTChar                 226 253 231 264  q\n",
            "        LTChar                 231 253 237 264  u\n",
            "        LTChar                 237 253 240 264  i\n",
            "        LTChar                 240 253 244 264  r\n",
            "        LTChar                 244 253 249 264  e\n",
            "        LTChar                 249 253 254 264  d\n",
            "        LTAnno                  \n",
            "        LTChar                 257 253 260 264  t\n",
            "        LTChar                 260 253 265 264  o\n",
            "        LTAnno                  \n",
            "        LTChar                 268 253 276 264  w\n",
            "        LTChar                 276 253 281 264  o\n",
            "        LTChar                 281 253 285 264  r\n",
            "        LTChar                 285 253 291 264  k\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  239 290 250  quickly to fully understand and to provide a form of\n",
            "        LTChar                 72  239 77  250  q\n",
            "        LTChar                 77  239 83  250  u\n",
            "        LTChar                 83  239 86  250  i\n",
            "        LTChar                 86  239 90  250  c\n",
            "        LTChar                 90  239 96  250  k\n",
            "        LTChar                 96  239 99  250  l\n",
            "        LTChar                 99  239 104 250  y\n",
            "        LTAnno                  \n",
            "        LTChar                 106 239 109 250  t\n",
            "        LTChar                 109 239 115 250  o\n",
            "        LTAnno                  \n",
            "        LTChar                 117 239 120 250  f\n",
            "        LTChar                 120 239 126 250  u\n",
            "        LTChar                 126 239 129 250  l\n",
            "        LTChar                 129 239 132 250  l\n",
            "        LTChar                 132 239 137 250  y\n",
            "        LTAnno                  \n",
            "        LTChar                 139 239 145 250  u\n",
            "        LTChar                 145 239 150 250  n\n",
            "        LTChar                 150 239 155 250  d\n",
            "        LTChar                 155 239 160 250  e\n",
            "        LTChar                 160 239 164 250  r\n",
            "        LTChar                 164 239 168 250  s\n",
            "        LTChar                 168 239 171 250  t\n",
            "        LTChar                 171 239 175 250  a\n",
            "        LTChar                 175 239 181 250  n\n",
            "        LTChar                 181 239 186 250  d\n",
            "        LTAnno                  \n",
            "        LTChar                 188 239 193 250  a\n",
            "        LTChar                 193 239 199 250  n\n",
            "        LTChar                 199 239 204 250  d\n",
            "        LTAnno                  \n",
            "        LTChar                 206 239 209 250  t\n",
            "        LTChar                 209 239 214 250  o\n",
            "        LTAnno                  \n",
            "        LTChar                 217 239 222 250  p\n",
            "        LTChar                 222 239 226 250  r\n",
            "        LTChar                 226 239 231 250  o\n",
            "        LTChar                 231 239 236 250  v\n",
            "        LTChar                 236 239 239 250  i\n",
            "        LTChar                 239 239 244 250  d\n",
            "        LTChar                 244 239 249 250  e\n",
            "        LTAnno                  \n",
            "        LTChar                 251 239 256 250  a\n",
            "        LTAnno                  \n",
            "        LTChar                 258 239 262 250  f\n",
            "        LTChar                 262 239 267 250  o\n",
            "        LTChar                 267 239 271 250  r\n",
            "        LTChar                 271 239 279 250  m\n",
            "        LTAnno                  \n",
            "        LTChar                 281 239 287 250  o\n",
            "        LTChar                 287 239 290 250  f\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  226 290 237  intervention for the virus. Due to a large research\n",
            "        LTChar                 72  226 75  237  i\n",
            "        LTChar                 75  226 81  237  n\n",
            "        LTChar                 81  226 84  237  t\n",
            "        LTChar                 84  226 89  237  e\n",
            "        LTChar                 89  226 92  237  r\n",
            "        LTChar                 92  226 98  237  v\n",
            "        LTChar                 98  226 103 237  e\n",
            "        LTChar                 103 226 108 237  n\n",
            "        LTChar                 108 226 111 237  t\n",
            "        LTChar                 111 226 114 237  i\n",
            "        LTChar                 114 226 120 237  o\n",
            "        LTChar                 120 226 125 237  n\n",
            "        LTAnno                  \n",
            "        LTChar                 128 226 132 237  f\n",
            "        LTChar                 132 226 137 237  o\n",
            "        LTChar                 137 226 141 237  r\n",
            "        LTAnno                  \n",
            "        LTChar                 144 226 147 237  t\n",
            "        LTChar                 147 226 152 237  h\n",
            "        LTChar                 152 226 157 237  e\n",
            "        LTAnno                  \n",
            "        LTChar                 160 226 166 237  v\n",
            "        LTChar                 166 226 169 237  i\n",
            "        LTChar                 169 226 172 237  r\n",
            "        LTChar                 172 226 178 237  u\n",
            "        LTChar                 178 226 182 237  s\n",
            "        LTChar                 182 226 185 237  .\n",
            "        LTAnno                  \n",
            "        LTChar                 188 226 196 237  D\n",
            "        LTChar                 196 226 202 237  u\n",
            "        LTChar                 202 226 207 237  e\n",
            "        LTAnno                  \n",
            "        LTChar                 210 226 213 237  t\n",
            "        LTChar                 213 226 218 237  o\n",
            "        LTAnno                  \n",
            "        LTChar                 221 226 226 237  a\n",
            "        LTAnno                  \n",
            "        LTChar                 229 226 232 237  l\n",
            "        LTChar                 232 226 237 237  a\n",
            "        LTChar                 237 226 240 237  r\n",
            "        LTChar                 240 226 246 237  g\n",
            "        LTChar                 246 226 251 237  e\n",
            "        LTAnno                  \n",
            "        LTChar                 253 226 257 237  r\n",
            "        LTChar                 257 226 262 237  e\n",
            "        LTChar                 262 226 266 237  s\n",
            "        LTChar                 266 226 271 237  e\n",
            "        LTChar                 271 226 276 237  a\n",
            "        LTChar                 276 226 280 237  r\n",
            "        LTChar                 280 226 285 237  c\n",
            "        LTChar                 285 226 290 237  h\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  212 290 223  focus on the disease, knowledge is published at a\n",
            "        LTChar                 72  212 76  223  f\n",
            "        LTChar                 76  212 81  223  o\n",
            "        LTChar                 81  212 86  223  c\n",
            "        LTChar                 86  212 92  223  u\n",
            "        LTChar                 92  212 96  223  s\n",
            "        LTAnno                  \n",
            "        LTChar                 99  212 104 223  o\n",
            "        LTChar                 104 212 110 223  n\n",
            "        LTAnno                  \n",
            "        LTChar                 113 212 116 223  t\n",
            "        LTChar                 116 212 121 223  h\n",
            "        LTChar                 121 212 126 223  e\n",
            "        LTAnno                  \n",
            "        LTChar                 129 212 135 223  d\n",
            "        LTChar                 135 212 138 223  i\n",
            "        LTChar                 138 212 142 223  s\n",
            "        LTChar                 142 212 147 223  e\n",
            "        LTChar                 147 212 152 223  a\n",
            "        LTChar                 152 212 156 223  s\n",
            "        LTChar                 156 212 161 223  e\n",
            "        LTChar                 161 212 164 223  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 167 212 172 223  k\n",
            "        LTChar                 172 212 178 223  n\n",
            "        LTChar                 178 212 184 223  o\n",
            "        LTChar                 183 212 191 223  w\n",
            "        LTChar                 191 212 194 223  l\n",
            "        LTChar                 194 212 199 223  e\n",
            "        LTChar                 199 212 205 223  d\n",
            "        LTChar                 205 212 211 223  g\n",
            "        LTChar                 211 212 215 223  e\n",
            "        LTAnno                  \n",
            "        LTChar                 218 212 221 223  i\n",
            "        LTChar                 221 212 226 223  s\n",
            "        LTAnno                  \n",
            "        LTChar                 228 212 234 223  p\n",
            "        LTChar                 234 212 240 223  u\n",
            "        LTChar                 240 212 245 223  b\n",
            "        LTChar                 245 212 248 223  l\n",
            "        LTChar                 248 212 251 223  i\n",
            "        LTChar                 251 212 256 223  s\n",
            "        LTChar                 256 212 261 223  h\n",
            "        LTChar                 261 212 266 223  e\n",
            "        LTChar                 266 212 272 223  d\n",
            "        LTAnno                  \n",
            "        LTChar                 275 212 279 223  a\n",
            "        LTChar                 279 212 283 223  t\n",
            "        LTAnno                  \n",
            "        LTChar                 285 212 290 223  a\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  199 292 209  rapid rate throughout the world. One such reposi-\n",
            "        LTChar                 72  199 76  209  r\n",
            "        LTChar                 76  199 81  209  a\n",
            "        LTChar                 81  199 86  209  p\n",
            "        LTChar                 86  199 89  209  i\n",
            "        LTChar                 89  199 95  209  d\n",
            "        LTAnno                  \n",
            "        LTChar                 98  199 101 209  r\n",
            "        LTChar                 101 199 106 209  a\n",
            "        LTChar                 106 199 109 209  t\n",
            "        LTChar                 109 199 114 209  e\n",
            "        LTAnno                  \n",
            "        LTChar                 117 199 120 209  t\n",
            "        LTChar                 120 199 126 209  h\n",
            "        LTChar                 126 199 129 209  r\n",
            "        LTChar                 129 199 135 209  o\n",
            "        LTChar                 135 199 140 209  u\n",
            "        LTChar                 140 199 146 209  g\n",
            "        LTChar                 146 199 152 209  h\n",
            "        LTChar                 152 199 157 209  o\n",
            "        LTChar                 157 199 163 209  u\n",
            "        LTChar                 163 199 166 209  t\n",
            "        LTAnno                  \n",
            "        LTChar                 169 199 172 209  t\n",
            "        LTChar                 172 199 177 209  h\n",
            "        LTChar                 177 199 182 209  e\n",
            "        LTAnno                  \n",
            "        LTChar                 185 199 193 209  w\n",
            "        LTChar                 193 199 198 209  o\n",
            "        LTChar                 198 199 202 209  r\n",
            "        LTChar                 202 199 205 209  l\n",
            "        LTChar                 205 199 211 209  d\n",
            "        LTChar                 211 199 213 209  .\n",
            "        LTAnno                  \n",
            "        LTChar                 217 199 225 209  O\n",
            "        LTChar                 225 199 230 209  n\n",
            "        LTChar                 230 199 235 209  e\n",
            "        LTAnno                  \n",
            "        LTChar                 238 199 242 209  s\n",
            "        LTChar                 242 199 248 209  u\n",
            "        LTChar                 248 199 253 209  c\n",
            "        LTChar                 253 199 258 209  h\n",
            "        LTAnno                  \n",
            "        LTChar                 261 199 265 209  r\n",
            "        LTChar                 265 199 270 209  e\n",
            "        LTChar                 270 199 275 209  p\n",
            "        LTChar                 275 199 281 209  o\n",
            "        LTChar                 281 199 285 209  s\n",
            "        LTChar                 285 199 288 209  i\n",
            "        LTChar                 288 199 292 209  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  185 292 196  tory of information is curated through the COVID-\n",
            "        LTChar                 72  185 75  196  t\n",
            "        LTChar                 75  185 80  196  o\n",
            "        LTChar                 80  185 84  196  r\n",
            "        LTChar                 84  185 89  196  y\n",
            "        LTAnno                  \n",
            "        LTChar                 92  185 98  196  o\n",
            "        LTChar                 98  185 101 196  f\n",
            "        LTAnno                  \n",
            "        LTChar                 104 185 107 196  i\n",
            "        LTChar                 107 185 112 196  n\n",
            "        LTChar                 112 185 116 196  f\n",
            "        LTChar                 116 185 121 196  o\n",
            "        LTChar                 121 185 125 196  r\n",
            "        LTChar                 125 185 133 196  m\n",
            "        LTChar                 133 185 138 196  a\n",
            "        LTChar                 138 185 141 196  t\n",
            "        LTChar                 141 185 144 196  i\n",
            "        LTChar                 144 185 150 196  o\n",
            "        LTChar                 150 185 155 196  n\n",
            "        LTAnno                  \n",
            "        LTChar                 158 185 161 196  i\n",
            "        LTChar                 161 185 165 196  s\n",
            "        LTAnno                  \n",
            "        LTChar                 168 185 173 196  c\n",
            "        LTChar                 173 185 178 196  u\n",
            "        LTChar                 178 185 182 196  r\n",
            "        LTChar                 182 185 186 196  a\n",
            "        LTChar                 186 185 189 196  t\n",
            "        LTChar                 189 185 194 196  e\n",
            "        LTChar                 194 185 200 196  d\n",
            "        LTAnno                  \n",
            "        LTChar                 202 185 205 196  t\n",
            "        LTChar                 205 185 211 196  h\n",
            "        LTChar                 211 185 214 196  r\n",
            "        LTChar                 214 185 220 196  o\n",
            "        LTChar                 220 185 225 196  u\n",
            "        LTChar                 225 185 231 196  g\n",
            "        LTChar                 231 185 236 196  h\n",
            "        LTAnno                  \n",
            "        LTChar                 239 185 242 196  t\n",
            "        LTChar                 242 185 247 196  h\n",
            "        LTChar                 247 185 252 196  e\n",
            "        LTAnno                  \n",
            "        LTChar                 255 185 262 196  C\n",
            "        LTChar                 262 185 270 196  O\n",
            "        LTChar                 269 185 277 196  V\n",
            "        LTChar                 277 185 281 196  I\n",
            "        LTChar                 281 185 288 196  D\n",
            "        LTChar                 288 185 292 196  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     71  171 291 182  19 Open Research Dataset Challenge (CORD-19)\n",
            "        LTChar                 71  171 77  182  1\n",
            "        LTChar                 77  171 82  182  9\n",
            "        LTAnno                  \n",
            "        LTChar                 85  171 93  182  O\n",
            "        LTChar                 93  171 98  182  p\n",
            "        LTChar                 98  171 103 182  e\n",
            "        LTChar                 103 171 109 182  n\n",
            "        LTAnno                  \n",
            "        LTChar                 112 171 119 182  R\n",
            "        LTChar                 119 171 124 182  e\n",
            "        LTChar                 124 171 128 182  s\n",
            "        LTChar                 128 171 133 182  e\n",
            "        LTChar                 133 171 138 182  a\n",
            "        LTChar                 138 171 142 182  r\n",
            "        LTChar                 142 171 147 182  c\n",
            "        LTChar                 147 171 152 182  h\n",
            "        LTAnno                  \n",
            "        LTChar                 155 171 163 182  D\n",
            "        LTChar                 163 171 168 182  a\n",
            "        LTChar                 168 171 171 182  t\n",
            "        LTChar                 171 171 176 182  a\n",
            "        LTChar                 176 171 180 182  s\n",
            "        LTChar                 180 171 185 182  e\n",
            "        LTChar                 185 171 188 182  t\n",
            "        LTAnno                  \n",
            "        LTChar                 191 171 198 182  C\n",
            "        LTChar                 198 171 204 182  h\n",
            "        LTChar                 204 171 208 182  a\n",
            "        LTChar                 208 171 212 182  l\n",
            "        LTChar                 212 171 215 182  l\n",
            "        LTChar                 215 171 220 182  e\n",
            "        LTChar                 220 171 225 182  n\n",
            "        LTChar                 225 171 231 182  g\n",
            "        LTChar                 231 171 235 182  e\n",
            "        LTAnno                  \n",
            "        LTChar                 238 171 242 182  (\n",
            "        LTChar                 242 171 249 182  C\n",
            "        LTChar                 249 171 257 182  O\n",
            "        LTChar                 257 171 265 182  R\n",
            "        LTChar                 265 171 273 182  D\n",
            "        LTChar                 273 171 276 182  -\n",
            "        LTChar                 276 171 282 182  1\n",
            "        LTChar                 282 171 287 182  9\n",
            "        LTChar                 287 171 291 182  )\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  158 290 169  (Wang et al., 2020). CORD-19 is a joint challenge\n",
            "        LTChar                 72  158 75  169  (\n",
            "        LTChar                 75  158 86  169  W\n",
            "        LTChar                 85  158 89  169  a\n",
            "        LTChar                 89  158 95  169  n\n",
            "        LTChar                 95  158 100 169  g\n",
            "        LTAnno                  \n",
            "        LTChar                 103 158 108 169  e\n",
            "        LTChar                 108 158 111 169  t\n",
            "        LTAnno                  \n",
            "        LTChar                 114 158 118 169  a\n",
            "        LTChar                 118 158 121 169  l\n",
            "        LTChar                 121 158 124 169  .\n",
            "        LTChar                 124 158 127 169  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 130 158 135 169  2\n",
            "        LTChar                 135 158 140 169  0\n",
            "        LTChar                 140 158 146 169  2\n",
            "        LTChar                 146 158 151 169  0\n",
            "        LTChar                 151 158 155 169  )\n",
            "        LTChar                 155 158 158 169  .\n",
            "        LTAnno                  \n",
            "        LTChar                 161 158 168 169  C\n",
            "        LTChar                 168 158 176 169  O\n",
            "        LTChar                 176 158 183 169  R\n",
            "        LTChar                 183 158 191 169  D\n",
            "        LTChar                 191 158 195 169  -\n",
            "        LTChar                 195 158 200 169  1\n",
            "        LTChar                 200 158 206 169  9\n",
            "        LTAnno                  \n",
            "        LTChar                 208 158 211 169  i\n",
            "        LTChar                 211 158 216 169  s\n",
            "        LTAnno                  \n",
            "        LTChar                 218 158 223 169  a\n",
            "        LTAnno                  \n",
            "        LTChar                 226 158 229 169  j\n",
            "        LTChar                 229 158 234 169  o\n",
            "        LTChar                 234 158 237 169  i\n",
            "        LTChar                 237 158 243 169  n\n",
            "        LTChar                 243 158 246 169  t\n",
            "        LTAnno                  \n",
            "        LTChar                 249 158 253 169  c\n",
            "        LTChar                 253 158 259 169  h\n",
            "        LTChar                 259 158 264 169  a\n",
            "        LTChar                 264 158 267 169  l\n",
            "        LTChar                 267 158 270 169  l\n",
            "        LTChar                 270 158 275 169  e\n",
            "        LTChar                 275 158 280 169  n\n",
            "        LTChar                 280 158 285 169  g\n",
            "        LTChar                 285 158 290 169  e\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  144 292 155  put forth by Allen Institute (AI2), National Insti-\n",
            "        LTChar                 72  144 78  155  p\n",
            "        LTChar                 78  144 83  155  u\n",
            "        LTChar                 83  144 86  155  t\n",
            "        LTAnno                  \n",
            "        LTChar                 89  144 93  155  f\n",
            "        LTChar                 93  144 99  155  o\n",
            "        LTChar                 99  144 102 155  r\n",
            "        LTChar                 102 144 106 155  t\n",
            "        LTChar                 105 144 111 155  h\n",
            "        LTAnno                  \n",
            "        LTChar                 114 144 120 155  b\n",
            "        LTChar                 120 144 125 155  y\n",
            "        LTAnno                  \n",
            "        LTChar                 129 144 137 155  A\n",
            "        LTChar                 137 144 140 155  l\n",
            "        LTChar                 140 144 143 155  l\n",
            "        LTChar                 143 144 148 155  e\n",
            "        LTChar                 148 144 153 155  n\n",
            "        LTAnno                  \n",
            "        LTChar                 157 144 160 155  I\n",
            "        LTChar                 160 144 166 155  n\n",
            "        LTChar                 166 144 170 155  s\n",
            "        LTChar                 170 144 173 155  t\n",
            "        LTChar                 173 144 176 155  i\n",
            "        LTChar                 176 144 179 155  t\n",
            "        LTChar                 179 144 185 155  u\n",
            "        LTChar                 185 144 188 155  t\n",
            "        LTChar                 188 144 193 155  e\n",
            "        LTAnno                  \n",
            "        LTChar                 196 144 200 155  (\n",
            "        LTChar                 200 144 208 155  A\n",
            "        LTChar                 208 144 212 155  I\n",
            "        LTChar                 212 144 217 155  2\n",
            "        LTChar                 217 144 221 155  )\n",
            "        LTChar                 221 144 224 155  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 227 144 235 155  N\n",
            "        LTChar                 235 144 240 155  a\n",
            "        LTChar                 240 144 243 155  t\n",
            "        LTChar                 243 144 246 155  i\n",
            "        LTChar                 246 144 252 155  o\n",
            "        LTChar                 252 144 257 155  n\n",
            "        LTChar                 257 144 262 155  a\n",
            "        LTChar                 262 144 265 155  l\n",
            "        LTAnno                  \n",
            "        LTChar                 269 144 272 155  I\n",
            "        LTChar                 272 144 278 155  n\n",
            "        LTChar                 278 144 282 155  s\n",
            "        LTChar                 282 144 285 155  t\n",
            "        LTChar                 285 144 288 155  i\n",
            "        LTChar                 288 144 292 155  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  131 290 142  tutes of Health (NIH), and the United States federal\n",
            "        LTChar                 72  131 75  142  t\n",
            "        LTChar                 75  131 80  142  u\n",
            "        LTChar                 80  131 83  142  t\n",
            "        LTChar                 83  131 88  142  e\n",
            "        LTChar                 88  131 92  142  s\n",
            "        LTAnno                  \n",
            "        LTChar                 95  131 100 142  o\n",
            "        LTChar                 100 131 103 142  f\n",
            "        LTAnno                  \n",
            "        LTChar                 106 131 114 142  H\n",
            "        LTChar                 114 131 118 142  e\n",
            "        LTChar                 118 131 123 142  a\n",
            "        LTChar                 123 131 126 142  l\n",
            "        LTChar                 126 131 129 142  t\n",
            "        LTChar                 129 131 134 142  h\n",
            "        LTAnno                  \n",
            "        LTChar                 137 131 140 142  (\n",
            "        LTChar                 140 131 148 142  N\n",
            "        LTChar                 148 131 152 142  I\n",
            "        LTChar                 152 131 159 142  H\n",
            "        LTChar                 159 131 163 142  )\n",
            "        LTChar                 163 131 166 142  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 168 131 173 142  a\n",
            "        LTChar                 173 131 178 142  n\n",
            "        LTChar                 178 131 183 142  d\n",
            "        LTAnno                  \n",
            "        LTChar                 186 131 189 142  t\n",
            "        LTChar                 189 131 194 142  h\n",
            "        LTChar                 194 131 199 142  e\n",
            "        LTAnno                  \n",
            "        LTChar                 201 131 209 142  U\n",
            "        LTChar                 209 131 214 142  n\n",
            "        LTChar                 214 131 217 142  i\n",
            "        LTChar                 217 131 220 142  t\n",
            "        LTChar                 220 131 225 142  e\n",
            "        LTChar                 225 131 230 142  d\n",
            "        LTAnno                  \n",
            "        LTChar                 233 131 239 142  S\n",
            "        LTChar                 239 131 242 142  t\n",
            "        LTChar                 242 131 246 142  a\n",
            "        LTChar                 246 131 249 142  t\n",
            "        LTChar                 249 131 254 142  e\n",
            "        LTChar                 254 131 258 142  s\n",
            "        LTAnno                  \n",
            "        LTChar                 261 131 264 142  f\n",
            "        LTChar                 264 131 269 142  e\n",
            "        LTChar                 269 131 274 142  d\n",
            "        LTChar                 274 131 279 142  e\n",
            "        LTChar                 279 131 283 142  r\n",
            "        LTChar                 283 131 287 142  a\n",
            "        LTChar                 287 131 290 142  l\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  117 290 128  government via the White House. The objective of\n",
            "        LTChar                 72  117 77  128  g\n",
            "        LTChar                 77  117 83  128  o\n",
            "        LTChar                 83  117 88  128  v\n",
            "        LTChar                 88  117 93  128  e\n",
            "        LTChar                 93  117 96  128  r\n",
            "        LTChar                 96  117 102 128  n\n",
            "        LTChar                 102 117 110 128  m\n",
            "        LTChar                 110 117 115 128  e\n",
            "        LTChar                 115 117 120 128  n\n",
            "        LTChar                 120 117 123 128  t\n",
            "        LTAnno                  \n",
            "        LTChar                 126 117 131 128  v\n",
            "        LTChar                 131 117 134 128  i\n",
            "        LTChar                 134 117 139 128  a\n",
            "        LTAnno                  \n",
            "        LTChar                 142 117 145 128  t\n",
            "        LTChar                 145 117 150 128  h\n",
            "        LTChar                 150 117 155 128  e\n",
            "        LTAnno                  \n",
            "        LTChar                 158 117 168 128  W\n",
            "        LTChar                 168 117 173 128  h\n",
            "        LTChar                 173 117 176 128  i\n",
            "        LTChar                 176 117 179 128  t\n",
            "        LTChar                 179 117 184 128  e\n",
            "        LTAnno                  \n",
            "        LTChar                 187 117 194 128  H\n",
            "        LTChar                 194 117 200 128  o\n",
            "        LTChar                 200 117 205 128  u\n",
            "        LTChar                 205 117 209 128  s\n",
            "        LTChar                 209 117 214 128  e\n",
            "        LTChar                 214 117 217 128  .\n",
            "        LTAnno                  \n",
            "        LTChar                 220 117 227 128  T\n",
            "        LTChar                 227 117 232 128  h\n",
            "        LTChar                 232 117 237 128  e\n",
            "        LTAnno                  \n",
            "        LTChar                 240 117 245 128  o\n",
            "        LTChar                 245 117 250 128  b\n",
            "        LTChar                 250 117 253 128  j\n",
            "        LTChar                 253 117 258 128  e\n",
            "        LTChar                 258 117 263 128  c\n",
            "        LTChar                 263 117 266 128  t\n",
            "        LTChar                 266 117 269 128  i\n",
            "        LTChar                 269 117 274 128  v\n",
            "        LTChar                 274 117 279 128  e\n",
            "        LTAnno                  \n",
            "        LTChar                 281 117 287 128  o\n",
            "        LTChar                 287 117 290 128  f\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  104 290 115  the challenge is to make sense of and extract useful\n",
            "        LTChar                 72  104 75  115  t\n",
            "        LTChar                 75  104 80  115  h\n",
            "        LTChar                 80  104 85  115  e\n",
            "        LTAnno                  \n",
            "        LTChar                 88  104 92  115  c\n",
            "        LTChar                 92  104 98  115  h\n",
            "        LTChar                 98  104 102 115  a\n",
            "        LTChar                 102 104 105 115  l\n",
            "        LTChar                 105 104 108 115  l\n",
            "        LTChar                 108 104 113 115  e\n",
            "        LTChar                 113 104 119 115  n\n",
            "        LTChar                 119 104 124 115  g\n",
            "        LTChar                 124 104 129 115  e\n",
            "        LTAnno                  \n",
            "        LTChar                 131 104 134 115  i\n",
            "        LTChar                 134 104 138 115  s\n",
            "        LTAnno                  \n",
            "        LTChar                 141 104 144 115  t\n",
            "        LTChar                 144 104 149 115  o\n",
            "        LTAnno                  \n",
            "        LTChar                 152 104 160 115  m\n",
            "        LTChar                 160 104 165 115  a\n",
            "        LTChar                 165 104 170 115  k\n",
            "        LTChar                 170 104 175 115  e\n",
            "        LTAnno                  \n",
            "        LTChar                 177 104 182 115  s\n",
            "        LTChar                 182 104 186 115  e\n",
            "        LTChar                 186 104 192 115  n\n",
            "        LTChar                 192 104 196 115  s\n",
            "        LTChar                 196 104 201 115  e\n",
            "        LTAnno                  \n",
            "        LTChar                 203 104 208 115  o\n",
            "        LTChar                 208 104 212 115  f\n",
            "        LTAnno                  \n",
            "        LTChar                 215 104 219 115  a\n",
            "        LTChar                 219 104 225 115  n\n",
            "        LTChar                 225 104 230 115  d\n",
            "        LTAnno                  \n",
            "        LTChar                 233 104 237 115  e\n",
            "        LTChar                 237 104 243 115  x\n",
            "        LTChar                 243 104 246 115  t\n",
            "        LTChar                 246 104 249 115  r\n",
            "        LTChar                 249 104 254 115  a\n",
            "        LTChar                 254 104 259 115  c\n",
            "        LTChar                 259 104 262 115  t\n",
            "        LTAnno                  \n",
            "        LTChar                 264 104 269 115  u\n",
            "        LTChar                 269 104 274 115  s\n",
            "        LTChar                 274 104 278 115  e\n",
            "        LTChar                 278 104 282 115  f\n",
            "        LTChar                 282 104 287 115  u\n",
            "        LTChar                 287 104 290 115  l\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  90  290 101  knowledge across thousands of scholarly articles\n",
            "        LTChar                 72  90  78  101  k\n",
            "        LTChar                 78  90  83  101  n\n",
            "        LTChar                 83  90  89  101  o\n",
            "        LTChar                 88  90  96  101  w\n",
            "        LTChar                 96  90  100 101  l\n",
            "        LTChar                 100 90  104 101  e\n",
            "        LTChar                 104 90  110 101  d\n",
            "        LTChar                 110 90  116 101  g\n",
            "        LTChar                 116 90  121 101  e\n",
            "        LTAnno                  \n",
            "        LTChar                 124 90  129 101  a\n",
            "        LTChar                 129 90  133 101  c\n",
            "        LTChar                 133 90  137 101  r\n",
            "        LTChar                 137 90  143 101  o\n",
            "        LTChar                 143 90  147 101  s\n",
            "        LTChar                 147 90  151 101  s\n",
            "        LTAnno                  \n",
            "        LTChar                 154 90  158 101  t\n",
            "        LTChar                 158 90  163 101  h\n",
            "        LTChar                 163 90  169 101  o\n",
            "        LTChar                 169 90  174 101  u\n",
            "        LTChar                 174 90  179 101  s\n",
            "        LTChar                 179 90  183 101  a\n",
            "        LTChar                 183 90  189 101  n\n",
            "        LTChar                 189 90  195 101  d\n",
            "        LTChar                 195 90  199 101  s\n",
            "        LTAnno                  \n",
            "        LTChar                 202 90  208 101  o\n",
            "        LTChar                 208 90  211 101  f\n",
            "        LTAnno                  \n",
            "        LTChar                 214 90  219 101  s\n",
            "        LTChar                 219 90  224 101  c\n",
            "        LTChar                 224 90  229 101  h\n",
            "        LTChar                 229 90  235 101  o\n",
            "        LTChar                 235 90  238 101  l\n",
            "        LTChar                 238 90  243 101  a\n",
            "        LTChar                 243 90  246 101  r\n",
            "        LTChar                 246 90  250 101  l\n",
            "        LTChar                 250 90  255 101  y\n",
            "        LTAnno                  \n",
            "        LTChar                 258 90  263 101  a\n",
            "        LTChar                 263 90  267 101  r\n",
            "        LTChar                 267 90  270 101  t\n",
            "        LTChar                 270 90  273 101  i\n",
            "        LTChar                 273 90  278 101  c\n",
            "        LTChar                 278 90  281 101  l\n",
            "        LTChar                 281 90  286 101  e\n",
            "        LTChar                 286 90  290 101  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     72  77  290 87   related to COVID-19. CORD-19 aims to connect\n",
            "        LTChar                 72  77  76  87   r\n",
            "        LTChar                 76  77  81  87   e\n",
            "        LTChar                 81  77  84  87   l\n",
            "        LTChar                 84  77  89  87   a\n",
            "        LTChar                 89  77  92  87   t\n",
            "        LTChar                 92  77  97  87   e\n",
            "        LTChar                 97  77  102 87   d\n",
            "        LTAnno                  \n",
            "        LTChar                 105 77  108 87   t\n",
            "        LTChar                 108 77  114 87   o\n",
            "        LTAnno                  \n",
            "        LTChar                 116 77  124 87   C\n",
            "        LTChar                 124 77  132 87   O\n",
            "        LTChar                 131 77  139 87   V\n",
            "        LTChar                 139 77  143 87   I\n",
            "        LTChar                 143 77  151 87   D\n",
            "        LTChar                 151 77  155 87   -\n",
            "        LTChar                 155 77  160 87   1\n",
            "        LTChar                 160 77  166 87   9\n",
            "        LTChar                 166 77  169 87   .\n",
            "        LTAnno                  \n",
            "        LTChar                 172 77  179 87   C\n",
            "        LTChar                 179 77  188 87   O\n",
            "        LTChar                 188 77  195 87   R\n",
            "        LTChar                 195 77  203 87   D\n",
            "        LTChar                 203 77  207 87   -\n",
            "        LTChar                 207 77  212 87   1\n",
            "        LTChar                 212 77  218 87   9\n",
            "        LTAnno                  \n",
            "        LTChar                 221 77  225 87   a\n",
            "        LTChar                 225 77  229 87   i\n",
            "        LTChar                 229 77  237 87   m\n",
            "        LTChar                 237 77  242 87   s\n",
            "        LTAnno                  \n",
            "        LTChar                 244 77  247 87   t\n",
            "        LTChar                 247 77  253 87   o\n",
            "        LTAnno                  \n",
            "        LTChar                 256 77  261 87   c\n",
            "        LTChar                 261 77  266 87   o\n",
            "        LTChar                 266 77  272 87   n\n",
            "        LTChar                 272 77  277 87   n\n",
            "        LTChar                 277 77  282 87   e\n",
            "        LTChar                 282 77  287 87   c\n",
            "        LTChar                 287 77  290 87   t\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        307 389 527 617  the machine learning community with biomedical\n",
            "domain experts and policy makers in a race to iden-\n",
            "tify effective treatments and management policies\n",
            "for COVID-19. In accordance with this initiative\n",
            "our goal is to provide a scalable solution aimed at\n",
            "aiding COVID-19 researchers and policy makers in\n",
            "their search and discovery for answers to high prior-\n",
            "ity scientiﬁc questions. These questions should be\n",
            "understood in their natural language form; exam-\n",
            "ples include: “What do we know about COVID-19\n",
            "risk factors?”, “Which medications were most ben-\n",
            "eﬁcial in the 2002 SARS outbreak?”, and “Which is\n",
            "the most referenced paper doing study for Hydroxy-\n",
            "chloroquine?” To appropriately answer these ques-\n",
            "tions we require a system with a strong biomedical\n",
            "understanding of the natural queries and structured\n",
            "knowledge (Rotmensch et al., 2017).\n",
            "      LTTextLineHorizontal     307 606 526 617  the machine learning community with biomedical\n",
            "        LTChar                 307 606 310 617  t\n",
            "        LTChar                 310 606 316 617  h\n",
            "        LTChar                 316 606 321 617  e\n",
            "        LTAnno                  \n",
            "        LTChar                 323 606 332 617  m\n",
            "        LTChar                 332 606 337 617  a\n",
            "        LTChar                 337 606 342 617  c\n",
            "        LTChar                 342 606 347 617  h\n",
            "        LTChar                 347 606 350 617  i\n",
            "        LTChar                 350 606 356 617  n\n",
            "        LTChar                 356 606 361 617  e\n",
            "        LTAnno                  \n",
            "        LTChar                 363 606 366 617  l\n",
            "        LTChar                 366 606 371 617  e\n",
            "        LTChar                 371 606 376 617  a\n",
            "        LTChar                 376 606 380 617  r\n",
            "        LTChar                 380 606 385 617  n\n",
            "        LTChar                 385 606 388 617  i\n",
            "        LTChar                 388 606 394 617  n\n",
            "        LTChar                 394 606 399 617  g\n",
            "        LTAnno                  \n",
            "        LTChar                 402 606 407 617  c\n",
            "        LTChar                 407 606 412 617  o\n",
            "        LTChar                 412 606 421 617  m\n",
            "        LTChar                 421 606 429 617  m\n",
            "        LTChar                 429 606 435 617  u\n",
            "        LTChar                 435 606 440 617  n\n",
            "        LTChar                 440 606 443 617  i\n",
            "        LTChar                 443 606 446 617  t\n",
            "        LTChar                 446 606 452 617  y\n",
            "        LTAnno                  \n",
            "        LTChar                 455 606 463 617  w\n",
            "        LTChar                 463 606 466 617  i\n",
            "        LTChar                 466 606 469 617  t\n",
            "        LTChar                 469 606 474 617  h\n",
            "        LTAnno                  \n",
            "        LTChar                 477 606 482 617  b\n",
            "        LTChar                 482 606 485 617  i\n",
            "        LTChar                 485 606 491 617  o\n",
            "        LTChar                 491 606 499 617  m\n",
            "        LTChar                 499 606 504 617  e\n",
            "        LTChar                 504 606 510 617  d\n",
            "        LTChar                 510 606 513 617  i\n",
            "        LTChar                 513 606 518 617  c\n",
            "        LTChar                 518 606 522 617  a\n",
            "        LTChar                 522 606 526 617  l\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 592 527 603  domain experts and policy makers in a race to iden-\n",
            "        LTChar                 307 592 313 603  d\n",
            "        LTChar                 313 592 318 603  o\n",
            "        LTChar                 318 592 326 603  m\n",
            "        LTChar                 326 592 331 603  a\n",
            "        LTChar                 331 592 334 603  i\n",
            "        LTChar                 334 592 339 603  n\n",
            "        LTAnno                  \n",
            "        LTChar                 342 592 347 603  e\n",
            "        LTChar                 347 592 352 603  x\n",
            "        LTChar                 352 592 357 603  p\n",
            "        LTChar                 357 592 362 603  e\n",
            "        LTChar                 362 592 366 603  r\n",
            "        LTChar                 366 592 369 603  t\n",
            "        LTChar                 369 592 373 603  s\n",
            "        LTAnno                  \n",
            "        LTChar                 375 592 380 603  a\n",
            "        LTChar                 380 592 385 603  n\n",
            "        LTChar                 385 592 391 603  d\n",
            "        LTAnno                  \n",
            "        LTChar                 393 592 399 603  p\n",
            "        LTChar                 399 592 404 603  o\n",
            "        LTChar                 404 592 407 603  l\n",
            "        LTChar                 407 592 410 603  i\n",
            "        LTChar                 410 592 415 603  c\n",
            "        LTChar                 415 592 420 603  y\n",
            "        LTAnno                  \n",
            "        LTChar                 422 592 431 603  m\n",
            "        LTChar                 431 592 436 603  a\n",
            "        LTChar                 436 592 441 603  k\n",
            "        LTChar                 441 592 445 603  e\n",
            "        LTChar                 445 592 449 603  r\n",
            "        LTChar                 449 592 453 603  s\n",
            "        LTAnno                  \n",
            "        LTChar                 456 592 459 603  i\n",
            "        LTChar                 459 592 464 603  n\n",
            "        LTAnno                  \n",
            "        LTChar                 467 592 471 603  a\n",
            "        LTAnno                  \n",
            "        LTChar                 474 592 478 603  r\n",
            "        LTChar                 478 592 482 603  a\n",
            "        LTChar                 482 592 487 603  c\n",
            "        LTChar                 487 592 492 603  e\n",
            "        LTAnno                  \n",
            "        LTChar                 494 592 497 603  t\n",
            "        LTChar                 497 592 503 603  o\n",
            "        LTAnno                  \n",
            "        LTChar                 505 592 508 603  i\n",
            "        LTChar                 508 592 514 603  d\n",
            "        LTChar                 514 592 518 603  e\n",
            "        LTChar                 518 592 524 603  n\n",
            "        LTChar                 524 592 527 603  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 579 526 590  tify effective treatments and management policies\n",
            "        LTChar                 307 579 310 590  t\n",
            "        LTChar                 310 579 313 590  i\n",
            "        LTChar                 313 579 317 590  f\n",
            "        LTChar                 317 579 323 590  y\n",
            "        LTAnno                  \n",
            "        LTChar                 325 579 330 590  e\n",
            "        LTChar                 330 579 334 590  f\n",
            "        LTChar                 333 579 337 590  f\n",
            "        LTChar                 337 579 342 590  e\n",
            "        LTChar                 342 579 347 590  c\n",
            "        LTChar                 347 579 350 590  t\n",
            "        LTChar                 350 579 353 590  i\n",
            "        LTChar                 353 579 358 590  v\n",
            "        LTChar                 358 579 363 590  e\n",
            "        LTAnno                  \n",
            "        LTChar                 366 579 369 590  t\n",
            "        LTChar                 369 579 372 590  r\n",
            "        LTChar                 372 579 377 590  e\n",
            "        LTChar                 377 579 382 590  a\n",
            "        LTChar                 382 579 385 590  t\n",
            "        LTChar                 385 579 394 590  m\n",
            "        LTChar                 394 579 399 590  e\n",
            "        LTChar                 399 579 404 590  n\n",
            "        LTChar                 404 579 407 590  t\n",
            "        LTChar                 407 579 411 590  s\n",
            "        LTAnno                  \n",
            "        LTChar                 414 579 419 590  a\n",
            "        LTChar                 419 579 424 590  n\n",
            "        LTChar                 424 579 430 590  d\n",
            "        LTAnno                  \n",
            "        LTChar                 433 579 441 590  m\n",
            "        LTChar                 441 579 446 590  a\n",
            "        LTChar                 446 579 452 590  n\n",
            "        LTChar                 452 579 456 590  a\n",
            "        LTChar                 456 579 462 590  g\n",
            "        LTChar                 462 579 467 590  e\n",
            "        LTChar                 467 579 475 590  m\n",
            "        LTChar                 475 579 480 590  e\n",
            "        LTChar                 480 579 486 590  n\n",
            "        LTChar                 486 579 489 590  t\n",
            "        LTAnno                  \n",
            "        LTChar                 491 579 497 590  p\n",
            "        LTChar                 497 579 502 590  o\n",
            "        LTChar                 502 579 505 590  l\n",
            "        LTChar                 505 579 508 590  i\n",
            "        LTChar                 508 579 513 590  c\n",
            "        LTChar                 513 579 516 590  i\n",
            "        LTChar                 516 579 521 590  e\n",
            "        LTChar                 521 579 526 590  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 565 526 576  for COVID-19. In accordance with this initiative\n",
            "        LTChar                 307 565 311 576  f\n",
            "        LTChar                 311 565 317 576  o\n",
            "        LTChar                 317 565 320 576  r\n",
            "        LTAnno                  \n",
            "        LTChar                 323 565 331 576  C\n",
            "        LTChar                 331 565 339 576  O\n",
            "        LTChar                 338 565 346 576  V\n",
            "        LTChar                 346 565 350 576  I\n",
            "        LTChar                 350 565 358 576  D\n",
            "        LTChar                 358 565 362 576  -\n",
            "        LTChar                 362 565 367 576  1\n",
            "        LTChar                 367 565 373 576  9\n",
            "        LTChar                 373 565 375 576  .\n",
            "        LTAnno                  \n",
            "        LTChar                 379 565 383 576  I\n",
            "        LTChar                 383 565 389 576  n\n",
            "        LTAnno                  \n",
            "        LTChar                 392 565 397 576  a\n",
            "        LTChar                 397 565 402 576  c\n",
            "        LTChar                 402 565 407 576  c\n",
            "        LTChar                 407 565 412 576  o\n",
            "        LTChar                 412 565 416 576  r\n",
            "        LTChar                 416 565 421 576  d\n",
            "        LTChar                 421 565 426 576  a\n",
            "        LTChar                 426 565 432 576  n\n",
            "        LTChar                 432 565 437 576  c\n",
            "        LTChar                 437 565 442 576  e\n",
            "        LTAnno                  \n",
            "        LTChar                 445 565 453 576  w\n",
            "        LTChar                 453 565 456 576  i\n",
            "        LTChar                 456 565 459 576  t\n",
            "        LTChar                 459 565 464 576  h\n",
            "        LTAnno                  \n",
            "        LTChar                 467 565 470 576  t\n",
            "        LTChar                 470 565 476 576  h\n",
            "        LTChar                 476 565 479 576  i\n",
            "        LTChar                 479 565 483 576  s\n",
            "        LTAnno                  \n",
            "        LTChar                 486 565 490 576  i\n",
            "        LTChar                 490 565 495 576  n\n",
            "        LTChar                 495 565 498 576  i\n",
            "        LTChar                 498 565 501 576  t\n",
            "        LTChar                 501 565 504 576  i\n",
            "        LTChar                 504 565 509 576  a\n",
            "        LTChar                 509 565 512 576  t\n",
            "        LTChar                 512 565 515 576  i\n",
            "        LTChar                 515 565 521 576  v\n",
            "        LTChar                 521 565 526 576  e\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 552 526 563  our goal is to provide a scalable solution aimed at\n",
            "        LTChar                 307 552 313 563  o\n",
            "        LTChar                 313 552 318 563  u\n",
            "        LTChar                 318 552 322 563  r\n",
            "        LTAnno                  \n",
            "        LTChar                 325 552 330 563  g\n",
            "        LTChar                 330 552 336 563  o\n",
            "        LTChar                 336 552 341 563  a\n",
            "        LTChar                 341 552 344 563  l\n",
            "        LTAnno                  \n",
            "        LTChar                 346 552 349 563  i\n",
            "        LTChar                 349 552 354 563  s\n",
            "        LTAnno                  \n",
            "        LTChar                 356 552 360 563  t\n",
            "        LTChar                 360 552 365 563  o\n",
            "        LTAnno                  \n",
            "        LTChar                 368 552 373 563  p\n",
            "        LTChar                 373 552 377 563  r\n",
            "        LTChar                 377 552 382 563  o\n",
            "        LTChar                 382 552 388 563  v\n",
            "        LTChar                 388 552 391 563  i\n",
            "        LTChar                 391 552 396 563  d\n",
            "        LTChar                 396 552 401 563  e\n",
            "        LTAnno                  \n",
            "        LTChar                 404 552 409 563  a\n",
            "        LTAnno                  \n",
            "        LTChar                 412 552 416 563  s\n",
            "        LTChar                 416 552 421 563  c\n",
            "        LTChar                 421 552 426 563  a\n",
            "        LTChar                 426 552 429 563  l\n",
            "        LTChar                 429 552 434 563  a\n",
            "        LTChar                 434 552 439 563  b\n",
            "        LTChar                 439 552 442 563  l\n",
            "        LTChar                 442 552 447 563  e\n",
            "        LTAnno                  \n",
            "        LTChar                 450 552 454 563  s\n",
            "        LTChar                 454 552 460 563  o\n",
            "        LTChar                 460 552 463 563  l\n",
            "        LTChar                 463 552 468 563  u\n",
            "        LTChar                 468 552 471 563  t\n",
            "        LTChar                 471 552 474 563  i\n",
            "        LTChar                 474 552 480 563  o\n",
            "        LTChar                 480 552 485 563  n\n",
            "        LTAnno                  \n",
            "        LTChar                 488 552 493 563  a\n",
            "        LTChar                 493 552 496 563  i\n",
            "        LTChar                 496 552 504 563  m\n",
            "        LTChar                 504 552 509 563  e\n",
            "        LTChar                 509 552 515 563  d\n",
            "        LTAnno                  \n",
            "        LTChar                 518 552 522 563  a\n",
            "        LTChar                 522 552 526 563  t\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 538 526 549  aiding COVID-19 researchers and policy makers in\n",
            "        LTChar                 307 538 312 549  a\n",
            "        LTChar                 312 538 315 549  i\n",
            "        LTChar                 315 538 320 549  d\n",
            "        LTChar                 320 538 323 549  i\n",
            "        LTChar                 323 538 329 549  n\n",
            "        LTChar                 329 538 334 549  g\n",
            "        LTAnno                  \n",
            "        LTChar                 336 538 344 549  C\n",
            "        LTChar                 344 538 351 549  O\n",
            "        LTChar                 351 538 358 549  V\n",
            "        LTChar                 358 538 362 549  I\n",
            "        LTChar                 362 538 370 549  D\n",
            "        LTChar                 370 538 373 549  -\n",
            "        LTChar                 373 538 379 549  1\n",
            "        LTChar                 379 538 384 549  9\n",
            "        LTAnno                  \n",
            "        LTChar                 386 538 390 549  r\n",
            "        LTChar                 390 538 395 549  e\n",
            "        LTChar                 395 538 399 549  s\n",
            "        LTChar                 399 538 404 549  e\n",
            "        LTChar                 404 538 408 549  a\n",
            "        LTChar                 408 538 412 549  r\n",
            "        LTChar                 412 538 417 549  c\n",
            "        LTChar                 417 538 422 549  h\n",
            "        LTChar                 422 538 427 549  e\n",
            "        LTChar                 427 538 430 549  r\n",
            "        LTChar                 430 538 435 549  s\n",
            "        LTAnno                  \n",
            "        LTChar                 437 538 442 549  a\n",
            "        LTChar                 442 538 447 549  n\n",
            "        LTChar                 447 538 452 549  d\n",
            "        LTAnno                  \n",
            "        LTChar                 455 538 460 549  p\n",
            "        LTChar                 460 538 466 549  o\n",
            "        LTChar                 466 538 469 549  l\n",
            "        LTChar                 469 538 472 549  i\n",
            "        LTChar                 472 538 476 549  c\n",
            "        LTChar                 476 538 482 549  y\n",
            "        LTAnno                  \n",
            "        LTChar                 484 538 492 549  m\n",
            "        LTChar                 492 538 497 549  a\n",
            "        LTChar                 497 538 502 549  k\n",
            "        LTChar                 502 538 507 549  e\n",
            "        LTChar                 507 538 511 549  r\n",
            "        LTChar                 511 538 515 549  s\n",
            "        LTAnno                  \n",
            "        LTChar                 517 538 520 549  i\n",
            "        LTChar                 520 538 526 549  n\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 525 527 536  their search and discovery for answers to high prior-\n",
            "        LTChar                 307 525 310 536  t\n",
            "        LTChar                 310 525 316 536  h\n",
            "        LTChar                 316 525 320 536  e\n",
            "        LTChar                 320 525 323 536  i\n",
            "        LTChar                 323 525 327 536  r\n",
            "        LTAnno                  \n",
            "        LTChar                 329 525 333 536  s\n",
            "        LTChar                 333 525 338 536  e\n",
            "        LTChar                 338 525 343 536  a\n",
            "        LTChar                 343 525 346 536  r\n",
            "        LTChar                 346 525 351 536  c\n",
            "        LTChar                 351 525 356 536  h\n",
            "        LTAnno                  \n",
            "        LTChar                 359 525 363 536  a\n",
            "        LTChar                 363 525 369 536  n\n",
            "        LTChar                 369 525 374 536  d\n",
            "        LTAnno                  \n",
            "        LTChar                 376 525 382 536  d\n",
            "        LTChar                 382 525 385 536  i\n",
            "        LTChar                 385 525 389 536  s\n",
            "        LTChar                 389 525 394 536  c\n",
            "        LTChar                 394 525 399 536  o\n",
            "        LTChar                 399 525 404 536  v\n",
            "        LTChar                 404 525 409 536  e\n",
            "        LTChar                 409 525 412 536  r\n",
            "        LTChar                 412 525 418 536  y\n",
            "        LTAnno                  \n",
            "        LTChar                 420 525 423 536  f\n",
            "        LTChar                 423 525 429 536  o\n",
            "        LTChar                 429 525 432 536  r\n",
            "        LTAnno                  \n",
            "        LTChar                 435 525 439 536  a\n",
            "        LTChar                 439 525 445 536  n\n",
            "        LTChar                 445 525 449 536  s\n",
            "        LTChar                 449 525 457 536  w\n",
            "        LTChar                 457 525 461 536  e\n",
            "        LTChar                 461 525 465 536  r\n",
            "        LTChar                 465 525 469 536  s\n",
            "        LTAnno                  \n",
            "        LTChar                 471 525 474 536  t\n",
            "        LTChar                 474 525 480 536  o\n",
            "        LTAnno                  \n",
            "        LTChar                 482 525 487 536  h\n",
            "        LTChar                 487 525 490 536  i\n",
            "        LTChar                 490 525 496 536  g\n",
            "        LTChar                 496 525 501 536  h\n",
            "        LTAnno                  \n",
            "        LTChar                 503 525 509 536  p\n",
            "        LTChar                 509 525 512 536  r\n",
            "        LTChar                 512 525 515 536  i\n",
            "        LTChar                 515 525 520 536  o\n",
            "        LTChar                 520 525 524 536  r\n",
            "        LTChar                 524 525 527 536  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 511 526 522  ity scientiﬁc questions. These questions should be\n",
            "        LTChar                 307 511 310 522  i\n",
            "        LTChar                 310 511 313 522  t\n",
            "        LTChar                 313 511 319 522  y\n",
            "        LTAnno                  \n",
            "        LTChar                 321 511 326 522  s\n",
            "        LTChar                 326 511 331 522  c\n",
            "        LTChar                 331 511 334 522  i\n",
            "        LTChar                 334 511 338 522  e\n",
            "        LTChar                 338 511 344 522  n\n",
            "        LTChar                 344 511 347 522  t\n",
            "        LTChar                 347 511 350 522  i\n",
            "        LTChar                 350 511 356 522  ﬁ\n",
            "        LTChar                 356 511 361 522  c\n",
            "        LTAnno                  \n",
            "        LTChar                 364 511 369 522  q\n",
            "        LTChar                 369 511 374 522  u\n",
            "        LTChar                 374 511 379 522  e\n",
            "        LTChar                 379 511 384 522  s\n",
            "        LTChar                 384 511 387 522  t\n",
            "        LTChar                 387 511 390 522  i\n",
            "        LTChar                 390 511 395 522  o\n",
            "        LTChar                 395 511 400 522  n\n",
            "        LTChar                 400 511 405 522  s\n",
            "        LTChar                 405 511 407 522  .\n",
            "        LTAnno                  \n",
            "        LTChar                 411 511 418 522  T\n",
            "        LTChar                 418 511 423 522  h\n",
            "        LTChar                 423 511 428 522  e\n",
            "        LTChar                 428 511 432 522  s\n",
            "        LTChar                 432 511 437 522  e\n",
            "        LTAnno                  \n",
            "        LTChar                 440 511 445 522  q\n",
            "        LTChar                 445 511 450 522  u\n",
            "        LTChar                 450 511 455 522  e\n",
            "        LTChar                 455 511 460 522  s\n",
            "        LTChar                 460 511 463 522  t\n",
            "        LTChar                 463 511 466 522  i\n",
            "        LTChar                 466 511 471 522  o\n",
            "        LTChar                 471 511 477 522  n\n",
            "        LTChar                 477 511 481 522  s\n",
            "        LTAnno                  \n",
            "        LTChar                 483 511 488 522  s\n",
            "        LTChar                 488 511 493 522  h\n",
            "        LTChar                 493 511 499 522  o\n",
            "        LTChar                 499 511 504 522  u\n",
            "        LTChar                 504 511 507 522  l\n",
            "        LTChar                 507 511 513 522  d\n",
            "        LTAnno                  \n",
            "        LTChar                 515 511 521 522  b\n",
            "        LTChar                 521 511 526 522  e\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 498 527 509  understood in their natural language form; exam-\n",
            "        LTChar                 307 498 313 509  u\n",
            "        LTChar                 313 498 318 509  n\n",
            "        LTChar                 318 498 324 509  d\n",
            "        LTChar                 324 498 329 509  e\n",
            "        LTChar                 329 498 333 509  r\n",
            "        LTChar                 333 498 337 509  s\n",
            "        LTChar                 337 498 340 509  t\n",
            "        LTChar                 340 498 346 509  o\n",
            "        LTChar                 346 498 351 509  o\n",
            "        LTChar                 351 498 357 509  d\n",
            "        LTAnno                  \n",
            "        LTChar                 360 498 363 509  i\n",
            "        LTChar                 363 498 368 509  n\n",
            "        LTAnno                  \n",
            "        LTChar                 371 498 374 509  t\n",
            "        LTChar                 374 498 380 509  h\n",
            "        LTChar                 380 498 385 509  e\n",
            "        LTChar                 385 498 388 509  i\n",
            "        LTChar                 388 498 392 509  r\n",
            "        LTAnno                  \n",
            "        LTChar                 395 498 400 509  n\n",
            "        LTChar                 400 498 405 509  a\n",
            "        LTChar                 405 498 408 509  t\n",
            "        LTChar                 408 498 414 509  u\n",
            "        LTChar                 414 498 418 509  r\n",
            "        LTChar                 418 498 423 509  a\n",
            "        LTChar                 423 498 426 509  l\n",
            "        LTAnno                  \n",
            "        LTChar                 429 498 432 509  l\n",
            "        LTChar                 432 498 437 509  a\n",
            "        LTChar                 437 498 442 509  n\n",
            "        LTChar                 442 498 448 509  g\n",
            "        LTChar                 448 498 453 509  u\n",
            "        LTChar                 453 498 458 509  a\n",
            "        LTChar                 458 498 464 509  g\n",
            "        LTChar                 464 498 469 509  e\n",
            "        LTAnno                  \n",
            "        LTChar                 472 498 476 509  f\n",
            "        LTChar                 476 498 481 509  o\n",
            "        LTChar                 481 498 485 509  r\n",
            "        LTChar                 485 498 493 509  m\n",
            "        LTChar                 493 498 497 509  ;\n",
            "        LTAnno                  \n",
            "        LTChar                 500 498 505 509  e\n",
            "        LTChar                 504 498 510 509  x\n",
            "        LTChar                 510 498 515 509  a\n",
            "        LTChar                 515 498 524 509  m\n",
            "        LTChar                 524 498 527 509  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 484 526 495  ples include: “What do we know about COVID-19\n",
            "        LTChar                 307 484 313 495  p\n",
            "        LTChar                 313 484 316 495  l\n",
            "        LTChar                 316 484 320 495  e\n",
            "        LTChar                 320 484 325 495  s\n",
            "        LTAnno                  \n",
            "        LTChar                 327 484 330 495  i\n",
            "        LTChar                 330 484 336 495  n\n",
            "        LTChar                 336 484 340 495  c\n",
            "        LTChar                 340 484 343 495  l\n",
            "        LTChar                 343 484 349 495  u\n",
            "        LTChar                 349 484 354 495  d\n",
            "        LTChar                 354 484 359 495  e\n",
            "        LTChar                 359 484 362 495  :\n",
            "        LTAnno                  \n",
            "        LTChar                 365 484 370 495  “\n",
            "        LTChar                 370 484 380 495  W\n",
            "        LTChar                 380 484 386 495  h\n",
            "        LTChar                 386 484 390 495  a\n",
            "        LTChar                 390 484 393 495  t\n",
            "        LTAnno                  \n",
            "        LTChar                 396 484 401 495  d\n",
            "        LTChar                 401 484 407 495  o\n",
            "        LTAnno                  \n",
            "        LTChar                 410 484 417 495  w\n",
            "        LTChar                 417 484 422 495  e\n",
            "        LTAnno                  \n",
            "        LTChar                 425 484 430 495  k\n",
            "        LTChar                 430 484 436 495  n\n",
            "        LTChar                 436 484 441 495  o\n",
            "        LTChar                 441 484 448 495  w\n",
            "        LTAnno                  \n",
            "        LTChar                 451 484 456 495  a\n",
            "        LTChar                 456 484 461 495  b\n",
            "        LTChar                 461 484 467 495  o\n",
            "        LTChar                 467 484 472 495  u\n",
            "        LTChar                 472 484 475 495  t\n",
            "        LTAnno                  \n",
            "        LTChar                 478 484 485 495  C\n",
            "        LTChar                 485 484 493 495  O\n",
            "        LTChar                 492 484 500 495  V\n",
            "        LTChar                 500 484 503 495  I\n",
            "        LTChar                 503 484 511 495  D\n",
            "        LTChar                 511 484 515 495  -\n",
            "        LTChar                 515 484 520 495  1\n",
            "        LTChar                 520 484 526 495  9\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 471 527 481  risk factors?”, “Which medications were most ben-\n",
            "        LTChar                 307 471 311 481  r\n",
            "        LTChar                 311 471 314 481  i\n",
            "        LTChar                 314 471 318 481  s\n",
            "        LTChar                 318 471 323 481  k\n",
            "        LTAnno                  \n",
            "        LTChar                 326 471 330 481  f\n",
            "        LTChar                 330 471 334 481  a\n",
            "        LTChar                 334 471 339 481  c\n",
            "        LTChar                 339 471 342 481  t\n",
            "        LTChar                 342 471 347 481  o\n",
            "        LTChar                 347 471 351 481  r\n",
            "        LTChar                 351 471 355 481  s\n",
            "        LTChar                 355 471 360 481  ?\n",
            "        LTChar                 360 471 365 481  ”\n",
            "        LTChar                 365 471 367 481  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 370 471 375 481  “\n",
            "        LTChar                 375 471 385 481  W\n",
            "        LTChar                 385 471 390 481  h\n",
            "        LTChar                 390 471 393 481  i\n",
            "        LTChar                 393 471 398 481  c\n",
            "        LTChar                 398 471 404 481  h\n",
            "        LTAnno                  \n",
            "        LTChar                 406 471 415 481  m\n",
            "        LTChar                 415 471 419 481  e\n",
            "        LTChar                 419 471 425 481  d\n",
            "        LTChar                 425 471 428 481  i\n",
            "        LTChar                 428 471 433 481  c\n",
            "        LTChar                 433 471 437 481  a\n",
            "        LTChar                 437 471 440 481  t\n",
            "        LTChar                 440 471 443 481  i\n",
            "        LTChar                 443 471 449 481  o\n",
            "        LTChar                 449 471 454 481  n\n",
            "        LTChar                 454 471 458 481  s\n",
            "        LTAnno                  \n",
            "        LTChar                 461 471 469 481  w\n",
            "        LTChar                 469 471 474 481  e\n",
            "        LTChar                 474 471 477 481  r\n",
            "        LTChar                 477 471 482 481  e\n",
            "        LTAnno                  \n",
            "        LTChar                 485 471 493 481  m\n",
            "        LTChar                 493 471 498 481  o\n",
            "        LTChar                 498 471 503 481  s\n",
            "        LTChar                 503 471 506 481  t\n",
            "        LTAnno                  \n",
            "        LTChar                 508 471 514 481  b\n",
            "        LTChar                 514 471 518 481  e\n",
            "        LTChar                 518 471 524 481  n\n",
            "        LTChar                 524 471 527 481  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 457 526 468  eﬁcial in the 2002 SARS outbreak?”, and “Which is\n",
            "        LTChar                 307 457 312 468  e\n",
            "        LTChar                 312 457 318 468  ﬁ\n",
            "        LTChar                 318 457 323 468  c\n",
            "        LTChar                 323 457 326 468  i\n",
            "        LTChar                 326 457 330 468  a\n",
            "        LTChar                 330 457 333 468  l\n",
            "        LTAnno                  \n",
            "        LTChar                 336 457 339 468  i\n",
            "        LTChar                 339 457 344 468  n\n",
            "        LTAnno                  \n",
            "        LTChar                 346 457 349 468  t\n",
            "        LTChar                 349 457 354 468  h\n",
            "        LTChar                 354 457 359 468  e\n",
            "        LTAnno                  \n",
            "        LTChar                 361 457 367 468  2\n",
            "        LTChar                 367 457 372 468  0\n",
            "        LTChar                 372 457 377 468  0\n",
            "        LTChar                 377 457 383 468  2\n",
            "        LTAnno                  \n",
            "        LTChar                 385 457 391 468  S\n",
            "        LTChar                 391 457 398 468  A\n",
            "        LTChar                 398 457 406 468  R\n",
            "        LTChar                 406 457 411 468  S\n",
            "        LTAnno                  \n",
            "        LTChar                 414 457 419 468  o\n",
            "        LTChar                 419 457 424 468  u\n",
            "        LTChar                 424 457 427 468  t\n",
            "        LTChar                 427 457 433 468  b\n",
            "        LTChar                 433 457 436 468  r\n",
            "        LTChar                 436 457 441 468  e\n",
            "        LTChar                 441 457 446 468  a\n",
            "        LTChar                 446 457 451 468  k\n",
            "        LTChar                 451 457 456 468  ?\n",
            "        LTChar                 456 457 461 468  ”\n",
            "        LTChar                 461 457 463 468  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 465 457 470 468  a\n",
            "        LTChar                 470 457 476 468  n\n",
            "        LTChar                 476 457 481 468  d\n",
            "        LTAnno                  \n",
            "        LTChar                 483 457 488 468  “\n",
            "        LTChar                 488 457 498 468  W\n",
            "        LTChar                 498 457 503 468  h\n",
            "        LTChar                 503 457 506 468  i\n",
            "        LTChar                 506 457 511 468  c\n",
            "        LTChar                 511 457 516 468  h\n",
            "        LTAnno                  \n",
            "        LTChar                 518 457 521 468  i\n",
            "        LTChar                 521 457 526 468  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 443 527 454  the most referenced paper doing study for Hydroxy-\n",
            "        LTChar                 307 443 310 454  t\n",
            "        LTChar                 310 443 316 454  h\n",
            "        LTChar                 316 443 320 454  e\n",
            "        LTAnno                  \n",
            "        LTChar                 323 443 331 454  m\n",
            "        LTChar                 331 443 336 454  o\n",
            "        LTChar                 336 443 340 454  s\n",
            "        LTChar                 340 443 343 454  t\n",
            "        LTAnno                  \n",
            "        LTChar                 346 443 349 454  r\n",
            "        LTChar                 349 443 354 454  e\n",
            "        LTChar                 354 443 358 454  f\n",
            "        LTChar                 358 443 362 454  e\n",
            "        LTChar                 362 443 366 454  r\n",
            "        LTChar                 366 443 371 454  e\n",
            "        LTChar                 371 443 376 454  n\n",
            "        LTChar                 376 443 381 454  c\n",
            "        LTChar                 381 443 385 454  e\n",
            "        LTChar                 385 443 391 454  d\n",
            "        LTAnno                  \n",
            "        LTChar                 393 443 398 454  p\n",
            "        LTChar                 398 443 403 454  a\n",
            "        LTChar                 403 443 408 454  p\n",
            "        LTChar                 408 443 413 454  e\n",
            "        LTChar                 413 443 417 454  r\n",
            "        LTAnno                  \n",
            "        LTChar                 419 443 424 454  d\n",
            "        LTChar                 424 443 430 454  o\n",
            "        LTChar                 430 443 433 454  i\n",
            "        LTChar                 433 443 438 454  n\n",
            "        LTChar                 438 443 443 454  g\n",
            "        LTAnno                  \n",
            "        LTChar                 446 443 450 454  s\n",
            "        LTChar                 450 443 453 454  t\n",
            "        LTChar                 453 443 458 454  u\n",
            "        LTChar                 458 443 463 454  d\n",
            "        LTChar                 463 443 469 454  y\n",
            "        LTAnno                  \n",
            "        LTChar                 471 443 475 454  f\n",
            "        LTChar                 475 443 480 454  o\n",
            "        LTChar                 480 443 484 454  r\n",
            "        LTAnno                  \n",
            "        LTChar                 486 443 494 454  H\n",
            "        LTChar                 494 443 499 454  y\n",
            "        LTChar                 499 443 504 454  d\n",
            "        LTChar                 504 443 508 454  r\n",
            "        LTChar                 508 443 513 454  o\n",
            "        LTChar                 513 443 518 454  x\n",
            "        LTChar                 518 443 524 454  y\n",
            "        LTChar                 524 443 527 454  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 430 527 441  chloroquine?” To appropriately answer these ques-\n",
            "        LTChar                 307 430 312 441  c\n",
            "        LTChar                 312 430 317 441  h\n",
            "        LTChar                 317 430 320 441  l\n",
            "        LTChar                 320 430 326 441  o\n",
            "        LTChar                 326 430 329 441  r\n",
            "        LTChar                 329 430 335 441  o\n",
            "        LTChar                 335 430 340 441  q\n",
            "        LTChar                 340 430 346 441  u\n",
            "        LTChar                 346 430 349 441  i\n",
            "        LTChar                 349 430 354 441  n\n",
            "        LTChar                 354 430 359 441  e\n",
            "        LTChar                 359 430 364 441  ?\n",
            "        LTChar                 364 430 368 441  ”\n",
            "        LTAnno                  \n",
            "        LTChar                 372 430 378 441  T\n",
            "        LTChar                 377 430 383 441  o\n",
            "        LTAnno                  \n",
            "        LTChar                 386 430 390 441  a\n",
            "        LTChar                 390 430 396 441  p\n",
            "        LTChar                 396 430 401 441  p\n",
            "        LTChar                 401 430 405 441  r\n",
            "        LTChar                 405 430 410 441  o\n",
            "        LTChar                 410 430 416 441  p\n",
            "        LTChar                 416 430 419 441  r\n",
            "        LTChar                 419 430 422 441  i\n",
            "        LTChar                 422 430 427 441  a\n",
            "        LTChar                 427 430 430 441  t\n",
            "        LTChar                 430 430 435 441  e\n",
            "        LTChar                 435 430 438 441  l\n",
            "        LTChar                 438 430 443 441  y\n",
            "        LTAnno                  \n",
            "        LTChar                 446 430 451 441  a\n",
            "        LTChar                 451 430 456 441  n\n",
            "        LTChar                 456 430 460 441  s\n",
            "        LTChar                 460 430 468 441  w\n",
            "        LTChar                 468 430 473 441  e\n",
            "        LTChar                 473 430 476 441  r\n",
            "        LTAnno                  \n",
            "        LTChar                 479 430 482 441  t\n",
            "        LTChar                 482 430 487 441  h\n",
            "        LTChar                 487 430 492 441  e\n",
            "        LTChar                 492 430 496 441  s\n",
            "        LTChar                 496 430 501 441  e\n",
            "        LTAnno                  \n",
            "        LTChar                 504 430 509 441  q\n",
            "        LTChar                 509 430 515 441  u\n",
            "        LTChar                 515 430 520 441  e\n",
            "        LTChar                 520 430 524 441  s\n",
            "        LTChar                 524 430 527 441  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 416 526 427  tions we require a system with a strong biomedical\n",
            "        LTChar                 307 416 310 427  t\n",
            "        LTChar                 310 416 313 427  i\n",
            "        LTChar                 313 416 319 427  o\n",
            "        LTChar                 319 416 324 427  n\n",
            "        LTChar                 324 416 328 427  s\n",
            "        LTAnno                  \n",
            "        LTChar                 331 416 339 427  w\n",
            "        LTChar                 339 416 343 427  e\n",
            "        LTAnno                  \n",
            "        LTChar                 346 416 350 427  r\n",
            "        LTChar                 350 416 354 427  e\n",
            "        LTChar                 354 416 360 427  q\n",
            "        LTChar                 360 416 365 427  u\n",
            "        LTChar                 365 416 368 427  i\n",
            "        LTChar                 368 416 372 427  r\n",
            "        LTChar                 372 416 376 427  e\n",
            "        LTAnno                  \n",
            "        LTChar                 379 416 384 427  a\n",
            "        LTAnno                  \n",
            "        LTChar                 387 416 391 427  s\n",
            "        LTChar                 391 416 396 427  y\n",
            "        LTChar                 396 416 400 427  s\n",
            "        LTChar                 400 416 403 427  t\n",
            "        LTChar                 403 416 408 427  e\n",
            "        LTChar                 408 416 416 427  m\n",
            "        LTAnno                  \n",
            "        LTChar                 419 416 427 427  w\n",
            "        LTChar                 427 416 430 427  i\n",
            "        LTChar                 430 416 433 427  t\n",
            "        LTChar                 433 416 438 427  h\n",
            "        LTAnno                  \n",
            "        LTChar                 441 416 446 427  a\n",
            "        LTAnno                  \n",
            "        LTChar                 448 416 453 427  s\n",
            "        LTChar                 453 416 456 427  t\n",
            "        LTChar                 456 416 459 427  r\n",
            "        LTChar                 459 416 464 427  o\n",
            "        LTChar                 464 416 470 427  n\n",
            "        LTChar                 470 416 475 427  g\n",
            "        LTAnno                  \n",
            "        LTChar                 478 416 483 427  b\n",
            "        LTChar                 483 416 486 427  i\n",
            "        LTChar                 486 416 492 427  o\n",
            "        LTChar                 492 416 500 427  m\n",
            "        LTChar                 500 416 505 427  e\n",
            "        LTChar                 505 416 510 427  d\n",
            "        LTChar                 510 416 513 427  i\n",
            "        LTChar                 513 416 518 427  c\n",
            "        LTChar                 518 416 523 427  a\n",
            "        LTChar                 523 416 526 427  l\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 403 526 414  understanding of the natural queries and structured\n",
            "        LTChar                 307 403 313 414  u\n",
            "        LTChar                 313 403 318 414  n\n",
            "        LTChar                 318 403 323 414  d\n",
            "        LTChar                 323 403 328 414  e\n",
            "        LTChar                 328 403 332 414  r\n",
            "        LTChar                 332 403 336 414  s\n",
            "        LTChar                 336 403 339 414  t\n",
            "        LTChar                 339 403 344 414  a\n",
            "        LTChar                 344 403 349 414  n\n",
            "        LTChar                 349 403 354 414  d\n",
            "        LTChar                 354 403 357 414  i\n",
            "        LTChar                 357 403 363 414  n\n",
            "        LTChar                 363 403 368 414  g\n",
            "        LTAnno                  \n",
            "        LTChar                 371 403 376 414  o\n",
            "        LTChar                 376 403 380 414  f\n",
            "        LTAnno                  \n",
            "        LTChar                 382 403 385 414  t\n",
            "        LTChar                 385 403 391 414  h\n",
            "        LTChar                 391 403 396 414  e\n",
            "        LTAnno                  \n",
            "        LTChar                 398 403 404 414  n\n",
            "        LTChar                 404 403 408 414  a\n",
            "        LTChar                 408 403 411 414  t\n",
            "        LTChar                 411 403 417 414  u\n",
            "        LTChar                 417 403 420 414  r\n",
            "        LTChar                 420 403 425 414  a\n",
            "        LTChar                 425 403 428 414  l\n",
            "        LTAnno                  \n",
            "        LTChar                 431 403 436 414  q\n",
            "        LTChar                 436 403 441 414  u\n",
            "        LTChar                 441 403 446 414  e\n",
            "        LTChar                 446 403 450 414  r\n",
            "        LTChar                 450 403 453 414  i\n",
            "        LTChar                 453 403 458 414  e\n",
            "        LTChar                 458 403 462 414  s\n",
            "        LTAnno                  \n",
            "        LTChar                 464 403 469 414  a\n",
            "        LTChar                 469 403 475 414  n\n",
            "        LTChar                 475 403 480 414  d\n",
            "        LTAnno                  \n",
            "        LTChar                 483 403 487 414  s\n",
            "        LTChar                 487 403 490 414  t\n",
            "        LTChar                 490 403 493 414  r\n",
            "        LTChar                 493 403 499 414  u\n",
            "        LTChar                 499 403 504 414  c\n",
            "        LTChar                 504 403 506 414  t\n",
            "        LTChar                 506 403 512 414  u\n",
            "        LTChar                 512 403 515 414  r\n",
            "        LTChar                 515 403 520 414  e\n",
            "        LTChar                 520 403 526 414  d\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 389 468 400  knowledge (Rotmensch et al., 2017).\n",
            "        LTChar                 307 389 313 400  k\n",
            "        LTChar                 313 389 318 400  n\n",
            "        LTChar                 318 389 324 400  o\n",
            "        LTChar                 323 389 331 400  w\n",
            "        LTChar                 331 389 334 400  l\n",
            "        LTChar                 334 389 339 400  e\n",
            "        LTChar                 339 389 345 400  d\n",
            "        LTChar                 345 389 350 400  g\n",
            "        LTChar                 350 389 355 400  e\n",
            "        LTAnno                  \n",
            "        LTChar                 358 389 361 400  (\n",
            "        LTChar                 361 389 369 400  R\n",
            "        LTChar                 369 389 374 400  o\n",
            "        LTChar                 374 389 377 400  t\n",
            "        LTChar                 377 389 385 400  m\n",
            "        LTChar                 385 389 390 400  e\n",
            "        LTChar                 390 389 396 400  n\n",
            "        LTChar                 396 389 400 400  s\n",
            "        LTChar                 400 389 405 400  c\n",
            "        LTChar                 405 389 410 400  h\n",
            "        LTAnno                  \n",
            "        LTChar                 413 389 418 400  e\n",
            "        LTChar                 418 389 421 400  t\n",
            "        LTAnno                  \n",
            "        LTChar                 424 389 428 400  a\n",
            "        LTChar                 428 389 432 400  l\n",
            "        LTChar                 432 389 434 400  .\n",
            "        LTChar                 434 389 437 400  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 440 389 445 400  2\n",
            "        LTChar                 445 389 451 400  0\n",
            "        LTChar                 451 389 456 400  1\n",
            "        LTChar                 456 389 462 400  7\n",
            "        LTChar                 462 389 465 400  )\n",
            "        LTChar                 465 389 468 400  .\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        307 102 527 385  AWS CORD-19 Search (ACS) provides an easy\n",
            "to use search interface where researchers query us-\n",
            "ing natural language 1. ACS goes beyond keyword\n",
            "matching by understanding question semantics to\n",
            "efﬁciently ﬁnd relevant answers. As illustrated in\n",
            "Figure 1, we provide the system with a natural lan-\n",
            "guage query inquiring about IL-6 inhibitors and\n",
            "showcase the system response with relevant query\n",
            "components highlighted. This query can confuse a\n",
            "traditional search engine that relies solely on text\n",
            "matching as it can only capture term overlap be-\n",
            "tween a query and a document and may not nec-\n",
            "essarily be relevant to the researcher’s true intent\n",
            "(i.e. learning implicit relations). Our system, how-\n",
            "ever, establishes the relationship between IL-6 and\n",
            "severity of SARS-CoV-2 showing evidence where\n",
            "elevated IL-6 occur in a large number of patients\n",
            "with severe COVID-19 and higher mortality rates.\n",
            "In the following example, if tasked with un-\n",
            "derstanding the epidemiology and transmission of\n",
            "COVID-19 a researcher may ask about salivary\n",
            "      LTTextLineHorizontal     318 374 526 385  AWS CORD-19 Search (ACS) provides an easy\n",
            "        LTChar                 318 374 326 385  A\n",
            "        LTChar                 325 374 335 385  W\n",
            "        LTChar                 335 374 341 385  S\n",
            "        LTAnno                  \n",
            "        LTChar                 344 374 351 385  C\n",
            "        LTChar                 351 374 359 385  O\n",
            "        LTChar                 359 374 366 385  R\n",
            "        LTChar                 366 374 374 385  D\n",
            "        LTChar                 374 374 378 385  -\n",
            "        LTChar                 378 374 383 385  1\n",
            "        LTChar                 383 374 388 385  9\n",
            "        LTAnno                  \n",
            "        LTChar                 391 374 397 385  S\n",
            "        LTChar                 397 374 402 385  e\n",
            "        LTChar                 402 374 407 385  a\n",
            "        LTChar                 407 374 410 385  r\n",
            "        LTChar                 410 374 415 385  c\n",
            "        LTChar                 415 374 421 385  h\n",
            "        LTAnno                  \n",
            "        LTChar                 423 374 427 385  (\n",
            "        LTChar                 427 374 435 385  A\n",
            "        LTChar                 434 374 442 385  C\n",
            "        LTChar                 442 374 448 385  S\n",
            "        LTChar                 448 374 451 385  )\n",
            "        LTAnno                  \n",
            "        LTChar                 454 374 459 385  p\n",
            "        LTChar                 459 374 463 385  r\n",
            "        LTChar                 463 374 468 385  o\n",
            "        LTChar                 468 374 474 385  v\n",
            "        LTChar                 474 374 477 385  i\n",
            "        LTChar                 477 374 482 385  d\n",
            "        LTChar                 482 374 487 385  e\n",
            "        LTChar                 487 374 491 385  s\n",
            "        LTAnno                  \n",
            "        LTChar                 494 374 499 385  a\n",
            "        LTChar                 499 374 504 385  n\n",
            "        LTAnno                  \n",
            "        LTChar                 507 374 511 385  e\n",
            "        LTChar                 511 374 516 385  a\n",
            "        LTChar                 516 374 521 385  s\n",
            "        LTChar                 521 374 526 385  y\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 361 527 372  to use search interface where researchers query us-\n",
            "        LTChar                 307 361 310 372  t\n",
            "        LTChar                 310 361 316 372  o\n",
            "        LTAnno                  \n",
            "        LTChar                 318 361 324 372  u\n",
            "        LTChar                 324 361 328 372  s\n",
            "        LTChar                 328 361 333 372  e\n",
            "        LTAnno                  \n",
            "        LTChar                 336 361 340 372  s\n",
            "        LTChar                 340 361 345 372  e\n",
            "        LTChar                 345 361 349 372  a\n",
            "        LTChar                 349 361 353 372  r\n",
            "        LTChar                 353 361 358 372  c\n",
            "        LTChar                 358 361 363 372  h\n",
            "        LTAnno                  \n",
            "        LTChar                 366 361 369 372  i\n",
            "        LTChar                 369 361 374 372  n\n",
            "        LTChar                 374 361 377 372  t\n",
            "        LTChar                 377 361 382 372  e\n",
            "        LTChar                 382 361 386 372  r\n",
            "        LTChar                 386 361 389 372  f\n",
            "        LTChar                 389 361 394 372  a\n",
            "        LTChar                 394 361 399 372  c\n",
            "        LTChar                 399 361 404 372  e\n",
            "        LTAnno                  \n",
            "        LTChar                 406 361 414 372  w\n",
            "        LTChar                 414 361 420 372  h\n",
            "        LTChar                 420 361 424 372  e\n",
            "        LTChar                 424 361 428 372  r\n",
            "        LTChar                 428 361 433 372  e\n",
            "        LTAnno                  \n",
            "        LTChar                 435 361 439 372  r\n",
            "        LTChar                 439 361 444 372  e\n",
            "        LTChar                 444 361 448 372  s\n",
            "        LTChar                 448 361 453 372  e\n",
            "        LTChar                 453 361 458 372  a\n",
            "        LTChar                 458 361 461 372  r\n",
            "        LTChar                 461 361 466 372  c\n",
            "        LTChar                 466 361 471 372  h\n",
            "        LTChar                 471 361 476 372  e\n",
            "        LTChar                 476 361 480 372  r\n",
            "        LTChar                 480 361 484 372  s\n",
            "        LTAnno                  \n",
            "        LTChar                 487 361 492 372  q\n",
            "        LTChar                 492 361 498 372  u\n",
            "        LTChar                 498 361 502 372  e\n",
            "        LTChar                 502 361 506 372  r\n",
            "        LTChar                 506 361 511 372  y\n",
            "        LTAnno                  \n",
            "        LTChar                 514 361 520 372  u\n",
            "        LTChar                 520 361 524 372  s\n",
            "        LTChar                 524 361 527 372  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 347 526 360  ing natural language 1. ACS goes beyond keyword\n",
            "        LTChar                 307 347 310 358  i\n",
            "        LTChar                 310 347 316 358  n\n",
            "        LTChar                 316 347 321 358  g\n",
            "        LTAnno                  \n",
            "        LTChar                 324 347 329 358  n\n",
            "        LTChar                 329 347 334 358  a\n",
            "        LTChar                 334 347 337 358  t\n",
            "        LTChar                 337 347 342 358  u\n",
            "        LTChar                 342 347 346 358  r\n",
            "        LTChar                 346 347 351 358  a\n",
            "        LTChar                 351 347 354 358  l\n",
            "        LTAnno                  \n",
            "        LTChar                 356 347 359 358  l\n",
            "        LTChar                 359 347 364 358  a\n",
            "        LTChar                 364 347 369 358  n\n",
            "        LTChar                 369 347 375 358  g\n",
            "        LTChar                 375 347 380 358  u\n",
            "        LTChar                 380 347 385 358  a\n",
            "        LTChar                 385 347 390 358  g\n",
            "        LTChar                 390 347 395 358  e\n",
            "        LTAnno                  \n",
            "        LTChar                 398 352 402 360  1\n",
            "        LTChar                 402 347 405 358  .\n",
            "        LTAnno                  \n",
            "        LTChar                 408 347 416 358  A\n",
            "        LTChar                 416 347 423 358  C\n",
            "        LTChar                 423 347 429 358  S\n",
            "        LTAnno                  \n",
            "        LTChar                 432 347 437 358  g\n",
            "        LTChar                 437 347 442 358  o\n",
            "        LTChar                 442 347 447 358  e\n",
            "        LTChar                 447 347 451 358  s\n",
            "        LTAnno                  \n",
            "        LTChar                 454 347 459 358  b\n",
            "        LTChar                 459 347 464 358  e\n",
            "        LTChar                 464 347 469 358  y\n",
            "        LTChar                 469 347 475 358  o\n",
            "        LTChar                 475 347 480 358  n\n",
            "        LTChar                 480 347 486 358  d\n",
            "        LTAnno                  \n",
            "        LTChar                 488 347 494 358  k\n",
            "        LTChar                 494 347 498 358  e\n",
            "        LTChar                 498 347 504 358  y\n",
            "        LTChar                 504 347 511 358  w\n",
            "        LTChar                 511 347 517 358  o\n",
            "        LTChar                 517 347 520 358  r\n",
            "        LTChar                 520 347 526 358  d\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 334 526 344  matching by understanding question semantics to\n",
            "        LTChar                 307 334 316 344  m\n",
            "        LTChar                 316 334 321 344  a\n",
            "        LTChar                 321 334 324 344  t\n",
            "        LTChar                 324 334 329 344  c\n",
            "        LTChar                 329 334 334 344  h\n",
            "        LTChar                 334 334 337 344  i\n",
            "        LTChar                 337 334 343 344  n\n",
            "        LTChar                 343 334 348 344  g\n",
            "        LTAnno                  \n",
            "        LTChar                 351 334 357 344  b\n",
            "        LTChar                 357 334 362 344  y\n",
            "        LTAnno                  \n",
            "        LTChar                 365 334 370 344  u\n",
            "        LTChar                 370 334 376 344  n\n",
            "        LTChar                 376 334 382 344  d\n",
            "        LTChar                 382 334 386 344  e\n",
            "        LTChar                 386 334 390 344  r\n",
            "        LTChar                 390 334 394 344  s\n",
            "        LTChar                 394 334 398 344  t\n",
            "        LTChar                 398 334 402 344  a\n",
            "        LTChar                 402 334 408 344  n\n",
            "        LTChar                 408 334 414 344  d\n",
            "        LTChar                 414 334 417 344  i\n",
            "        LTChar                 417 334 422 344  n\n",
            "        LTChar                 422 334 428 344  g\n",
            "        LTAnno                  \n",
            "        LTChar                 430 334 436 344  q\n",
            "        LTChar                 436 334 441 344  u\n",
            "        LTChar                 441 334 446 344  e\n",
            "        LTChar                 446 334 451 344  s\n",
            "        LTChar                 451 334 454 344  t\n",
            "        LTChar                 454 334 457 344  i\n",
            "        LTChar                 457 334 462 344  o\n",
            "        LTChar                 462 334 468 344  n\n",
            "        LTAnno                  \n",
            "        LTChar                 471 334 475 344  s\n",
            "        LTChar                 475 334 480 344  e\n",
            "        LTChar                 480 334 488 344  m\n",
            "        LTChar                 488 334 493 344  a\n",
            "        LTChar                 493 334 499 344  n\n",
            "        LTChar                 499 334 502 344  t\n",
            "        LTChar                 502 334 505 344  i\n",
            "        LTChar                 505 334 510 344  c\n",
            "        LTChar                 510 334 514 344  s\n",
            "        LTAnno                  \n",
            "        LTChar                 517 334 520 344  t\n",
            "        LTChar                 520 334 526 344  o\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 320 526 331  efﬁciently ﬁnd relevant answers. As illustrated in\n",
            "        LTChar                 307 320 312 331  e\n",
            "        LTChar                 312 320 316 331  f\n",
            "        LTChar                 316 320 322 331  ﬁ\n",
            "        LTChar                 322 320 327 331  c\n",
            "        LTChar                 327 320 330 331  i\n",
            "        LTChar                 330 320 335 331  e\n",
            "        LTChar                 335 320 340 331  n\n",
            "        LTChar                 340 320 343 331  t\n",
            "        LTChar                 343 320 346 331  l\n",
            "        LTChar                 346 320 352 331  y\n",
            "        LTAnno                  \n",
            "        LTChar                 355 320 361 331  ﬁ\n",
            "        LTChar                 361 320 366 331  n\n",
            "        LTChar                 366 320 372 331  d\n",
            "        LTAnno                  \n",
            "        LTChar                 375 320 378 331  r\n",
            "        LTChar                 378 320 383 331  e\n",
            "        LTChar                 383 320 386 331  l\n",
            "        LTChar                 386 320 391 331  e\n",
            "        LTChar                 391 320 397 331  v\n",
            "        LTChar                 396 320 401 331  a\n",
            "        LTChar                 401 320 407 331  n\n",
            "        LTChar                 407 320 410 331  t\n",
            "        LTAnno                  \n",
            "        LTChar                 413 320 418 331  a\n",
            "        LTChar                 418 320 423 331  n\n",
            "        LTChar                 423 320 428 331  s\n",
            "        LTChar                 428 320 436 331  w\n",
            "        LTChar                 436 320 440 331  e\n",
            "        LTChar                 440 320 444 331  r\n",
            "        LTChar                 444 320 449 331  s\n",
            "        LTChar                 449 320 451 331  .\n",
            "        LTAnno                  \n",
            "        LTChar                 455 320 463 331  A\n",
            "        LTChar                 463 320 467 331  s\n",
            "        LTAnno                  \n",
            "        LTChar                 470 320 473 331  i\n",
            "        LTChar                 473 320 476 331  l\n",
            "        LTChar                 476 320 479 331  l\n",
            "        LTChar                 479 320 485 331  u\n",
            "        LTChar                 485 320 489 331  s\n",
            "        LTChar                 489 320 492 331  t\n",
            "        LTChar                 492 320 496 331  r\n",
            "        LTChar                 496 320 501 331  a\n",
            "        LTChar                 501 320 504 331  t\n",
            "        LTChar                 504 320 509 331  e\n",
            "        LTChar                 509 320 514 331  d\n",
            "        LTAnno                  \n",
            "        LTChar                 517 320 520 331  i\n",
            "        LTChar                 520 320 526 331  n\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 306 527 317  Figure 1, we provide the system with a natural lan-\n",
            "        LTChar                 307 306 313 317  F\n",
            "        LTChar                 313 306 316 317  i\n",
            "        LTChar                 316 306 322 317  g\n",
            "        LTChar                 322 306 327 317  u\n",
            "        LTChar                 327 306 331 317  r\n",
            "        LTChar                 331 306 335 317  e\n",
            "        LTAnno                  \n",
            "        LTChar                 338 306 344 317  1\n",
            "        LTChar                 344 306 346 317  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 349 306 357 317  w\n",
            "        LTChar                 357 306 362 317  e\n",
            "        LTAnno                  \n",
            "        LTChar                 364 306 370 317  p\n",
            "        LTChar                 370 306 373 317  r\n",
            "        LTChar                 373 306 379 317  o\n",
            "        LTChar                 379 306 384 317  v\n",
            "        LTChar                 384 306 387 317  i\n",
            "        LTChar                 387 306 392 317  d\n",
            "        LTChar                 392 306 397 317  e\n",
            "        LTAnno                  \n",
            "        LTChar                 400 306 403 317  t\n",
            "        LTChar                 403 306 408 317  h\n",
            "        LTChar                 408 306 413 317  e\n",
            "        LTAnno                  \n",
            "        LTChar                 416 306 420 317  s\n",
            "        LTChar                 420 306 425 317  y\n",
            "        LTChar                 425 306 430 317  s\n",
            "        LTChar                 430 306 433 317  t\n",
            "        LTChar                 433 306 437 317  e\n",
            "        LTChar                 437 306 446 317  m\n",
            "        LTAnno                  \n",
            "        LTChar                 448 306 456 317  w\n",
            "        LTChar                 456 306 459 317  i\n",
            "        LTChar                 459 306 462 317  t\n",
            "        LTChar                 462 306 468 317  h\n",
            "        LTAnno                  \n",
            "        LTChar                 470 306 475 317  a\n",
            "        LTAnno                  \n",
            "        LTChar                 478 306 483 317  n\n",
            "        LTChar                 483 306 488 317  a\n",
            "        LTChar                 488 306 491 317  t\n",
            "        LTChar                 491 306 496 317  u\n",
            "        LTChar                 496 306 500 317  r\n",
            "        LTChar                 500 306 505 317  a\n",
            "        LTChar                 505 306 508 317  l\n",
            "        LTAnno                  \n",
            "        LTChar                 511 306 514 317  l\n",
            "        LTChar                 514 306 518 317  a\n",
            "        LTChar                 518 306 524 317  n\n",
            "        LTChar                 524 306 527 317  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 293 526 304  guage query inquiring about IL-6 inhibitors and\n",
            "        LTChar                 307 293 313 304  g\n",
            "        LTChar                 313 293 318 304  u\n",
            "        LTChar                 318 293 323 304  a\n",
            "        LTChar                 323 293 329 304  g\n",
            "        LTChar                 329 293 334 304  e\n",
            "        LTAnno                  \n",
            "        LTChar                 338 293 343 304  q\n",
            "        LTChar                 343 293 349 304  u\n",
            "        LTChar                 349 293 354 304  e\n",
            "        LTChar                 354 293 357 304  r\n",
            "        LTChar                 357 293 363 304  y\n",
            "        LTAnno                  \n",
            "        LTChar                 367 293 370 304  i\n",
            "        LTChar                 370 293 375 304  n\n",
            "        LTChar                 375 293 381 304  q\n",
            "        LTChar                 381 293 386 304  u\n",
            "        LTChar                 386 293 390 304  i\n",
            "        LTChar                 390 293 393 304  r\n",
            "        LTChar                 393 293 396 304  i\n",
            "        LTChar                 396 293 402 304  n\n",
            "        LTChar                 402 293 407 304  g\n",
            "        LTAnno                  \n",
            "        LTChar                 411 293 416 304  a\n",
            "        LTChar                 416 293 422 304  b\n",
            "        LTChar                 422 293 427 304  o\n",
            "        LTChar                 427 293 433 304  u\n",
            "        LTChar                 433 293 436 304  t\n",
            "        LTAnno                  \n",
            "        LTChar                 440 293 443 304  I\n",
            "        LTChar                 443 293 450 304  L\n",
            "        LTChar                 450 293 453 304  -\n",
            "        LTChar                 453 293 459 304  6\n",
            "        LTAnno                  \n",
            "        LTChar                 463 293 466 304  i\n",
            "        LTChar                 466 293 471 304  n\n",
            "        LTChar                 471 293 477 304  h\n",
            "        LTChar                 477 293 480 304  i\n",
            "        LTChar                 480 293 485 304  b\n",
            "        LTChar                 485 293 489 304  i\n",
            "        LTChar                 489 293 492 304  t\n",
            "        LTChar                 492 293 497 304  o\n",
            "        LTChar                 497 293 502 304  r\n",
            "        LTChar                 501 293 506 304  s\n",
            "        LTAnno                  \n",
            "        LTChar                 509 293 514 304  a\n",
            "        LTChar                 514 293 520 304  n\n",
            "        LTChar                 520 293 526 304  d\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 279 526 290  showcase the system response with relevant query\n",
            "        LTChar                 307 279 312 290  s\n",
            "        LTChar                 312 279 317 290  h\n",
            "        LTChar                 317 279 322 290  o\n",
            "        LTChar                 322 279 330 290  w\n",
            "        LTChar                 330 279 335 290  c\n",
            "        LTChar                 335 279 340 290  a\n",
            "        LTChar                 340 279 344 290  s\n",
            "        LTChar                 344 279 349 290  e\n",
            "        LTAnno                  \n",
            "        LTChar                 352 279 355 290  t\n",
            "        LTChar                 355 279 360 290  h\n",
            "        LTChar                 360 279 365 290  e\n",
            "        LTAnno                  \n",
            "        LTChar                 368 279 372 290  s\n",
            "        LTChar                 372 279 377 290  y\n",
            "        LTChar                 377 279 382 290  s\n",
            "        LTChar                 382 279 385 290  t\n",
            "        LTChar                 385 279 389 290  e\n",
            "        LTChar                 389 279 398 290  m\n",
            "        LTAnno                  \n",
            "        LTChar                 401 279 404 290  r\n",
            "        LTChar                 404 279 409 290  e\n",
            "        LTChar                 409 279 413 290  s\n",
            "        LTChar                 413 279 419 290  p\n",
            "        LTChar                 419 279 424 290  o\n",
            "        LTChar                 424 279 430 290  n\n",
            "        LTChar                 430 279 434 290  s\n",
            "        LTChar                 434 279 439 290  e\n",
            "        LTAnno                  \n",
            "        LTChar                 442 279 449 290  w\n",
            "        LTChar                 449 279 453 290  i\n",
            "        LTChar                 453 279 456 290  t\n",
            "        LTChar                 456 279 461 290  h\n",
            "        LTAnno                  \n",
            "        LTChar                 464 279 467 290  r\n",
            "        LTChar                 467 279 472 290  e\n",
            "        LTChar                 472 279 475 290  l\n",
            "        LTChar                 475 279 480 290  e\n",
            "        LTChar                 480 279 485 290  v\n",
            "        LTChar                 485 279 490 290  a\n",
            "        LTChar                 490 279 495 290  n\n",
            "        LTChar                 495 279 498 290  t\n",
            "        LTAnno                  \n",
            "        LTChar                 501 279 507 290  q\n",
            "        LTChar                 507 279 512 290  u\n",
            "        LTChar                 512 279 517 290  e\n",
            "        LTChar                 517 279 520 290  r\n",
            "        LTChar                 520 279 526 290  y\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 266 526 277  components highlighted. This query can confuse a\n",
            "        LTChar                 307 266 312 277  c\n",
            "        LTChar                 312 266 317 277  o\n",
            "        LTChar                 317 266 326 277  m\n",
            "        LTChar                 326 266 331 277  p\n",
            "        LTChar                 331 266 337 277  o\n",
            "        LTChar                 337 266 342 277  n\n",
            "        LTChar                 342 266 347 277  e\n",
            "        LTChar                 347 266 352 277  n\n",
            "        LTChar                 352 266 355 277  t\n",
            "        LTChar                 355 266 359 277  s\n",
            "        LTAnno                  \n",
            "        LTChar                 362 266 368 277  h\n",
            "        LTChar                 368 266 371 277  i\n",
            "        LTChar                 371 266 376 277  g\n",
            "        LTChar                 376 266 381 277  h\n",
            "        LTChar                 381 266 384 277  l\n",
            "        LTChar                 384 266 387 277  i\n",
            "        LTChar                 387 266 393 277  g\n",
            "        LTChar                 393 266 398 277  h\n",
            "        LTChar                 398 266 401 277  t\n",
            "        LTChar                 401 266 406 277  e\n",
            "        LTChar                 406 266 411 277  d\n",
            "        LTChar                 411 266 414 277  .\n",
            "        LTAnno                  \n",
            "        LTChar                 417 266 424 277  T\n",
            "        LTChar                 424 266 429 277  h\n",
            "        LTChar                 429 266 432 277  i\n",
            "        LTChar                 432 266 437 277  s\n",
            "        LTAnno                  \n",
            "        LTChar                 439 266 445 277  q\n",
            "        LTChar                 445 266 450 277  u\n",
            "        LTChar                 450 266 455 277  e\n",
            "        LTChar                 455 266 459 277  r\n",
            "        LTChar                 459 266 464 277  y\n",
            "        LTAnno                  \n",
            "        LTChar                 467 266 472 277  c\n",
            "        LTChar                 472 266 476 277  a\n",
            "        LTChar                 476 266 482 277  n\n",
            "        LTAnno                  \n",
            "        LTChar                 484 266 489 277  c\n",
            "        LTChar                 489 266 495 277  o\n",
            "        LTChar                 495 266 500 277  n\n",
            "        LTChar                 500 266 504 277  f\n",
            "        LTChar                 504 266 509 277  u\n",
            "        LTChar                 509 266 513 277  s\n",
            "        LTChar                 513 266 518 277  e\n",
            "        LTAnno                  \n",
            "        LTChar                 521 266 526 277  a\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 252 526 263  traditional search engine that relies solely on text\n",
            "        LTChar                 307 252 310 263  t\n",
            "        LTChar                 310 252 314 263  r\n",
            "        LTChar                 314 252 319 263  a\n",
            "        LTChar                 319 252 325 263  d\n",
            "        LTChar                 325 252 328 263  i\n",
            "        LTChar                 328 252 331 263  t\n",
            "        LTChar                 331 252 334 263  i\n",
            "        LTChar                 334 252 339 263  o\n",
            "        LTChar                 339 252 345 263  n\n",
            "        LTChar                 345 252 350 263  a\n",
            "        LTChar                 350 252 353 263  l\n",
            "        LTAnno                  \n",
            "        LTChar                 356 252 360 263  s\n",
            "        LTChar                 360 252 365 263  e\n",
            "        LTChar                 365 252 370 263  a\n",
            "        LTChar                 370 252 374 263  r\n",
            "        LTChar                 374 252 379 263  c\n",
            "        LTChar                 379 252 384 263  h\n",
            "        LTAnno                  \n",
            "        LTChar                 387 252 392 263  e\n",
            "        LTChar                 392 252 397 263  n\n",
            "        LTChar                 397 252 403 263  g\n",
            "        LTChar                 403 252 406 263  i\n",
            "        LTChar                 406 252 412 263  n\n",
            "        LTChar                 412 252 417 263  e\n",
            "        LTAnno                  \n",
            "        LTChar                 419 252 423 263  t\n",
            "        LTChar                 423 252 428 263  h\n",
            "        LTChar                 428 252 433 263  a\n",
            "        LTChar                 433 252 436 263  t\n",
            "        LTAnno                  \n",
            "        LTChar                 439 252 443 263  r\n",
            "        LTChar                 443 252 448 263  e\n",
            "        LTChar                 448 252 451 263  l\n",
            "        LTChar                 451 252 454 263  i\n",
            "        LTChar                 454 252 459 263  e\n",
            "        LTChar                 459 252 463 263  s\n",
            "        LTAnno                  \n",
            "        LTChar                 466 252 470 263  s\n",
            "        LTChar                 470 252 476 263  o\n",
            "        LTChar                 476 252 479 263  l\n",
            "        LTChar                 479 252 484 263  e\n",
            "        LTChar                 484 252 487 263  l\n",
            "        LTChar                 487 252 492 263  y\n",
            "        LTAnno                  \n",
            "        LTChar                 495 252 501 263  o\n",
            "        LTChar                 501 252 506 263  n\n",
            "        LTAnno                  \n",
            "        LTChar                 509 252 512 263  t\n",
            "        LTChar                 512 252 517 263  e\n",
            "        LTChar                 517 252 522 263  x\n",
            "        LTChar                 522 252 526 263  t\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 239 527 250  matching as it can only capture term overlap be-\n",
            "        LTChar                 307 239 316 250  m\n",
            "        LTChar                 316 239 321 250  a\n",
            "        LTChar                 321 239 324 250  t\n",
            "        LTChar                 324 239 329 250  c\n",
            "        LTChar                 329 239 334 250  h\n",
            "        LTChar                 334 239 338 250  i\n",
            "        LTChar                 338 239 343 250  n\n",
            "        LTChar                 343 239 349 250  g\n",
            "        LTAnno                  \n",
            "        LTChar                 352 239 357 250  a\n",
            "        LTChar                 357 239 361 250  s\n",
            "        LTAnno                  \n",
            "        LTChar                 365 239 368 250  i\n",
            "        LTChar                 368 239 371 250  t\n",
            "        LTAnno                  \n",
            "        LTChar                 374 239 379 250  c\n",
            "        LTChar                 379 239 384 250  a\n",
            "        LTChar                 384 239 390 250  n\n",
            "        LTAnno                  \n",
            "        LTChar                 393 239 399 250  o\n",
            "        LTChar                 399 239 405 250  n\n",
            "        LTChar                 405 239 408 250  l\n",
            "        LTChar                 408 239 413 250  y\n",
            "        LTAnno                  \n",
            "        LTChar                 417 239 422 250  c\n",
            "        LTChar                 422 239 426 250  a\n",
            "        LTChar                 426 239 432 250  p\n",
            "        LTChar                 432 239 435 250  t\n",
            "        LTChar                 435 239 441 250  u\n",
            "        LTChar                 441 239 444 250  r\n",
            "        LTChar                 444 239 449 250  e\n",
            "        LTAnno                  \n",
            "        LTChar                 453 239 456 250  t\n",
            "        LTChar                 456 239 461 250  e\n",
            "        LTChar                 461 239 465 250  r\n",
            "        LTChar                 465 239 473 250  m\n",
            "        LTAnno                  \n",
            "        LTChar                 477 239 482 250  o\n",
            "        LTChar                 482 239 488 250  v\n",
            "        LTChar                 487 239 492 250  e\n",
            "        LTChar                 492 239 496 250  r\n",
            "        LTChar                 496 239 499 250  l\n",
            "        LTChar                 499 239 504 250  a\n",
            "        LTChar                 504 239 510 250  p\n",
            "        LTAnno                  \n",
            "        LTChar                 513 239 519 250  b\n",
            "        LTChar                 519 239 524 250  e\n",
            "        LTChar                 524 239 527 250  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 225 527 236  tween a query and a document and may not nec-\n",
            "        LTChar                 307 225 310 236  t\n",
            "        LTChar                 310 225 318 236  w\n",
            "        LTChar                 318 225 323 236  e\n",
            "        LTChar                 323 225 328 236  e\n",
            "        LTChar                 328 225 334 236  n\n",
            "        LTAnno                  \n",
            "        LTChar                 337 225 342 236  a\n",
            "        LTAnno                  \n",
            "        LTChar                 345 225 351 236  q\n",
            "        LTChar                 351 225 357 236  u\n",
            "        LTChar                 357 225 361 236  e\n",
            "        LTChar                 361 225 365 236  r\n",
            "        LTChar                 365 225 371 236  y\n",
            "        LTAnno                  \n",
            "        LTChar                 374 225 379 236  a\n",
            "        LTChar                 379 225 385 236  n\n",
            "        LTChar                 385 225 390 236  d\n",
            "        LTAnno                  \n",
            "        LTChar                 393 225 398 236  a\n",
            "        LTAnno                  \n",
            "        LTChar                 402 225 407 236  d\n",
            "        LTChar                 407 225 413 236  o\n",
            "        LTChar                 413 225 418 236  c\n",
            "        LTChar                 418 225 423 236  u\n",
            "        LTChar                 423 225 432 236  m\n",
            "        LTChar                 432 225 437 236  e\n",
            "        LTChar                 437 225 442 236  n\n",
            "        LTChar                 442 225 446 236  t\n",
            "        LTAnno                  \n",
            "        LTChar                 449 225 454 236  a\n",
            "        LTChar                 454 225 459 236  n\n",
            "        LTChar                 459 225 465 236  d\n",
            "        LTAnno                  \n",
            "        LTChar                 468 225 477 236  m\n",
            "        LTChar                 477 225 482 236  a\n",
            "        LTChar                 482 225 487 236  y\n",
            "        LTAnno                  \n",
            "        LTChar                 491 225 496 236  n\n",
            "        LTChar                 496 225 502 236  o\n",
            "        LTChar                 502 225 505 236  t\n",
            "        LTAnno                  \n",
            "        LTChar                 508 225 514 236  n\n",
            "        LTChar                 514 225 519 236  e\n",
            "        LTChar                 519 225 524 236  c\n",
            "        LTChar                 524 225 527 236  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 212 526 223  essarily be relevant to the researcher’s true intent\n",
            "        LTChar                 307 212 312 223  e\n",
            "        LTChar                 312 212 317 223  s\n",
            "        LTChar                 317 212 321 223  s\n",
            "        LTChar                 321 212 326 223  a\n",
            "        LTChar                 326 212 330 223  r\n",
            "        LTChar                 330 212 333 223  i\n",
            "        LTChar                 333 212 336 223  l\n",
            "        LTChar                 336 212 341 223  y\n",
            "        LTAnno                  \n",
            "        LTChar                 344 212 350 223  b\n",
            "        LTChar                 350 212 355 223  e\n",
            "        LTAnno                  \n",
            "        LTChar                 358 212 361 223  r\n",
            "        LTChar                 361 212 366 223  e\n",
            "        LTChar                 366 212 369 223  l\n",
            "        LTChar                 369 212 374 223  e\n",
            "        LTChar                 374 212 380 223  v\n",
            "        LTChar                 379 212 384 223  a\n",
            "        LTChar                 384 212 390 223  n\n",
            "        LTChar                 390 212 393 223  t\n",
            "        LTAnno                  \n",
            "        LTChar                 396 212 399 223  t\n",
            "        LTChar                 399 212 404 223  o\n",
            "        LTAnno                  \n",
            "        LTChar                 407 212 410 223  t\n",
            "        LTChar                 410 212 416 223  h\n",
            "        LTChar                 416 212 421 223  e\n",
            "        LTAnno                  \n",
            "        LTChar                 424 212 428 223  r\n",
            "        LTChar                 428 212 433 223  e\n",
            "        LTChar                 433 212 437 223  s\n",
            "        LTChar                 437 212 442 223  e\n",
            "        LTChar                 442 212 447 223  a\n",
            "        LTChar                 447 212 450 223  r\n",
            "        LTChar                 450 212 455 223  c\n",
            "        LTChar                 455 212 461 223  h\n",
            "        LTChar                 461 212 466 223  e\n",
            "        LTChar                 466 212 470 223  r\n",
            "        LTChar                 470 212 473 223  ’\n",
            "        LTChar                 473 212 477 223  s\n",
            "        LTAnno                  \n",
            "        LTChar                 480 212 483 223  t\n",
            "        LTChar                 483 212 487 223  r\n",
            "        LTChar                 487 212 492 223  u\n",
            "        LTChar                 492 212 497 223  e\n",
            "        LTAnno                  \n",
            "        LTChar                 500 212 503 223  i\n",
            "        LTChar                 503 212 509 223  n\n",
            "        LTChar                 509 212 512 223  t\n",
            "        LTChar                 512 212 517 223  e\n",
            "        LTChar                 517 212 522 223  n\n",
            "        LTChar                 522 212 526 223  t\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 198 527 209  (i.e. learning implicit relations). Our system, how-\n",
            "        LTChar                 307 198 311 209  (\n",
            "        LTChar                 311 198 314 209  i\n",
            "        LTChar                 314 198 316 209  .\n",
            "        LTChar                 316 198 321 209  e\n",
            "        LTChar                 321 198 324 209  .\n",
            "        LTAnno                  \n",
            "        LTChar                 327 198 330 209  l\n",
            "        LTChar                 330 198 335 209  e\n",
            "        LTChar                 335 198 340 209  a\n",
            "        LTChar                 340 198 344 209  r\n",
            "        LTChar                 344 198 349 209  n\n",
            "        LTChar                 349 198 352 209  i\n",
            "        LTChar                 352 198 358 209  n\n",
            "        LTChar                 358 198 363 209  g\n",
            "        LTAnno                  \n",
            "        LTChar                 366 198 369 209  i\n",
            "        LTChar                 369 198 377 209  m\n",
            "        LTChar                 377 198 383 209  p\n",
            "        LTChar                 383 198 386 209  l\n",
            "        LTChar                 386 198 389 209  i\n",
            "        LTChar                 389 198 394 209  c\n",
            "        LTChar                 394 198 397 209  i\n",
            "        LTChar                 397 198 400 209  t\n",
            "        LTAnno                  \n",
            "        LTChar                 402 198 406 209  r\n",
            "        LTChar                 406 198 411 209  e\n",
            "        LTChar                 411 198 414 209  l\n",
            "        LTChar                 414 198 419 209  a\n",
            "        LTChar                 419 198 422 209  t\n",
            "        LTChar                 422 198 425 209  i\n",
            "        LTChar                 425 198 430 209  o\n",
            "        LTChar                 430 198 436 209  n\n",
            "        LTChar                 436 198 440 209  s\n",
            "        LTChar                 440 198 444 209  )\n",
            "        LTChar                 444 198 446 209  .\n",
            "        LTAnno                  \n",
            "        LTChar                 450 198 458 209  O\n",
            "        LTChar                 458 198 463 209  u\n",
            "        LTChar                 463 198 467 209  r\n",
            "        LTAnno                  \n",
            "        LTChar                 469 198 474 209  s\n",
            "        LTChar                 474 198 479 209  y\n",
            "        LTChar                 479 198 483 209  s\n",
            "        LTChar                 483 198 486 209  t\n",
            "        LTChar                 486 198 491 209  e\n",
            "        LTChar                 491 198 500 209  m\n",
            "        LTChar                 500 198 502 209  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 505 198 511 209  h\n",
            "        LTChar                 511 198 516 209  o\n",
            "        LTChar                 516 198 524 209  w\n",
            "        LTChar                 524 198 527 209  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 185 526 195  ever, establishes the relationship between IL-6 and\n",
            "        LTChar                 307 185 312 195  e\n",
            "        LTChar                 312 185 317 195  v\n",
            "        LTChar                 317 185 322 195  e\n",
            "        LTChar                 322 185 325 195  r\n",
            "        LTChar                 325 185 328 195  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 330 185 335 195  e\n",
            "        LTChar                 335 185 339 195  s\n",
            "        LTChar                 339 185 342 195  t\n",
            "        LTChar                 342 185 347 195  a\n",
            "        LTChar                 347 185 353 195  b\n",
            "        LTChar                 353 185 356 195  l\n",
            "        LTChar                 356 185 359 195  i\n",
            "        LTChar                 359 185 363 195  s\n",
            "        LTChar                 363 185 368 195  h\n",
            "        LTChar                 368 185 373 195  e\n",
            "        LTChar                 373 185 377 195  s\n",
            "        LTAnno                  \n",
            "        LTChar                 380 185 383 195  t\n",
            "        LTChar                 383 185 388 195  h\n",
            "        LTChar                 388 185 393 195  e\n",
            "        LTAnno                  \n",
            "        LTChar                 396 185 399 195  r\n",
            "        LTChar                 399 185 404 195  e\n",
            "        LTChar                 404 185 407 195  l\n",
            "        LTChar                 407 185 412 195  a\n",
            "        LTChar                 412 185 415 195  t\n",
            "        LTChar                 415 185 418 195  i\n",
            "        LTChar                 418 185 423 195  o\n",
            "        LTChar                 423 185 429 195  n\n",
            "        LTChar                 429 185 433 195  s\n",
            "        LTChar                 433 185 438 195  h\n",
            "        LTChar                 438 185 441 195  i\n",
            "        LTChar                 441 185 447 195  p\n",
            "        LTAnno                  \n",
            "        LTChar                 449 185 455 195  b\n",
            "        LTChar                 455 185 460 195  e\n",
            "        LTChar                 460 185 463 195  t\n",
            "        LTChar                 463 185 470 195  w\n",
            "        LTChar                 470 185 475 195  e\n",
            "        LTChar                 475 185 480 195  e\n",
            "        LTChar                 480 185 485 195  n\n",
            "        LTAnno                  \n",
            "        LTChar                 488 185 492 195  I\n",
            "        LTChar                 492 185 498 195  L\n",
            "        LTChar                 498 185 502 195  -\n",
            "        LTChar                 502 185 507 195  6\n",
            "        LTAnno                  \n",
            "        LTChar                 510 185 515 195  a\n",
            "        LTChar                 515 185 520 195  n\n",
            "        LTChar                 520 185 526 195  d\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 171 526 182  severity of SARS-CoV-2 showing evidence where\n",
            "        LTChar                 307 171 312 182  s\n",
            "        LTChar                 312 171 316 182  e\n",
            "        LTChar                 316 171 322 182  v\n",
            "        LTChar                 321 171 326 182  e\n",
            "        LTChar                 326 171 330 182  r\n",
            "        LTChar                 330 171 333 182  i\n",
            "        LTChar                 333 171 336 182  t\n",
            "        LTChar                 336 171 341 182  y\n",
            "        LTAnno                  \n",
            "        LTChar                 344 171 350 182  o\n",
            "        LTChar                 350 171 353 182  f\n",
            "        LTAnno                  \n",
            "        LTChar                 356 171 362 182  S\n",
            "        LTChar                 362 171 370 182  A\n",
            "        LTChar                 370 171 377 182  R\n",
            "        LTChar                 377 171 383 182  S\n",
            "        LTChar                 383 171 387 182  -\n",
            "        LTChar                 387 171 394 182  C\n",
            "        LTChar                 394 171 400 182  o\n",
            "        LTChar                 400 171 407 182  V\n",
            "        LTChar                 406 171 410 182  -\n",
            "        LTChar                 410 171 415 182  2\n",
            "        LTAnno                  \n",
            "        LTChar                 418 171 422 182  s\n",
            "        LTChar                 422 171 428 182  h\n",
            "        LTChar                 428 171 433 182  o\n",
            "        LTChar                 433 171 441 182  w\n",
            "        LTChar                 441 171 444 182  i\n",
            "        LTChar                 444 171 449 182  n\n",
            "        LTChar                 449 171 455 182  g\n",
            "        LTAnno                  \n",
            "        LTChar                 458 171 463 182  e\n",
            "        LTChar                 462 171 468 182  v\n",
            "        LTChar                 468 171 471 182  i\n",
            "        LTChar                 471 171 476 182  d\n",
            "        LTChar                 476 171 481 182  e\n",
            "        LTChar                 481 171 486 182  n\n",
            "        LTChar                 486 171 491 182  c\n",
            "        LTChar                 491 171 496 182  e\n",
            "        LTAnno                  \n",
            "        LTChar                 499 171 507 182  w\n",
            "        LTChar                 507 171 512 182  h\n",
            "        LTChar                 512 171 517 182  e\n",
            "        LTChar                 517 171 521 182  r\n",
            "        LTChar                 521 171 526 182  e\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 157 526 168  elevated IL-6 occur in a large number of patients\n",
            "        LTChar                 307 157 312 168  e\n",
            "        LTChar                 312 157 315 168  l\n",
            "        LTChar                 315 157 320 168  e\n",
            "        LTChar                 320 157 326 168  v\n",
            "        LTChar                 325 157 330 168  a\n",
            "        LTChar                 330 157 333 168  t\n",
            "        LTChar                 333 157 338 168  e\n",
            "        LTChar                 338 157 344 168  d\n",
            "        LTAnno                  \n",
            "        LTChar                 347 157 350 168  I\n",
            "        LTChar                 350 157 357 168  L\n",
            "        LTChar                 357 157 361 168  -\n",
            "        LTChar                 361 157 367 168  6\n",
            "        LTAnno                  \n",
            "        LTChar                 370 157 375 168  o\n",
            "        LTChar                 375 157 380 168  c\n",
            "        LTChar                 380 157 385 168  c\n",
            "        LTChar                 385 157 391 168  u\n",
            "        LTChar                 391 157 394 168  r\n",
            "        LTAnno                  \n",
            "        LTChar                 397 157 400 168  i\n",
            "        LTChar                 400 157 406 168  n\n",
            "        LTAnno                  \n",
            "        LTChar                 409 157 414 168  a\n",
            "        LTAnno                  \n",
            "        LTChar                 417 157 420 168  l\n",
            "        LTChar                 420 157 425 168  a\n",
            "        LTChar                 425 157 428 168  r\n",
            "        LTChar                 428 157 434 168  g\n",
            "        LTChar                 434 157 439 168  e\n",
            "        LTAnno                  \n",
            "        LTChar                 442 157 447 168  n\n",
            "        LTChar                 447 157 453 168  u\n",
            "        LTChar                 453 157 462 168  m\n",
            "        LTChar                 462 157 467 168  b\n",
            "        LTChar                 467 157 472 168  e\n",
            "        LTChar                 472 157 476 168  r\n",
            "        LTAnno                  \n",
            "        LTChar                 479 157 484 168  o\n",
            "        LTChar                 484 157 488 168  f\n",
            "        LTAnno                  \n",
            "        LTChar                 491 157 496 168  p\n",
            "        LTChar                 496 157 501 168  a\n",
            "        LTChar                 501 157 505 168  t\n",
            "        LTChar                 505 157 508 168  i\n",
            "        LTChar                 508 157 513 168  e\n",
            "        LTChar                 513 157 518 168  n\n",
            "        LTChar                 518 157 521 168  t\n",
            "        LTChar                 521 157 526 168  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 144 526 155  with severe COVID-19 and higher mortality rates.\n",
            "        LTChar                 307 144 315 155  w\n",
            "        LTChar                 315 144 318 155  i\n",
            "        LTChar                 318 144 321 155  t\n",
            "        LTChar                 321 144 326 155  h\n",
            "        LTAnno                  \n",
            "        LTChar                 329 144 333 155  s\n",
            "        LTChar                 333 144 338 155  e\n",
            "        LTChar                 338 144 343 155  v\n",
            "        LTChar                 343 144 348 155  e\n",
            "        LTChar                 348 144 352 155  r\n",
            "        LTChar                 352 144 356 155  e\n",
            "        LTAnno                  \n",
            "        LTChar                 359 144 366 155  C\n",
            "        LTChar                 366 144 374 155  O\n",
            "        LTChar                 374 144 382 155  V\n",
            "        LTChar                 382 144 385 155  I\n",
            "        LTChar                 385 144 393 155  D\n",
            "        LTChar                 393 144 397 155  -\n",
            "        LTChar                 397 144 402 155  1\n",
            "        LTChar                 402 144 408 155  9\n",
            "        LTAnno                  \n",
            "        LTChar                 410 144 415 155  a\n",
            "        LTChar                 415 144 421 155  n\n",
            "        LTChar                 421 144 426 155  d\n",
            "        LTAnno                  \n",
            "        LTChar                 429 144 434 155  h\n",
            "        LTChar                 434 144 437 155  i\n",
            "        LTChar                 437 144 443 155  g\n",
            "        LTChar                 443 144 448 155  h\n",
            "        LTChar                 448 144 453 155  e\n",
            "        LTChar                 453 144 457 155  r\n",
            "        LTAnno                  \n",
            "        LTChar                 460 144 468 155  m\n",
            "        LTChar                 468 144 473 155  o\n",
            "        LTChar                 473 144 477 155  r\n",
            "        LTChar                 477 144 480 155  t\n",
            "        LTChar                 480 144 485 155  a\n",
            "        LTChar                 485 144 488 155  l\n",
            "        LTChar                 488 144 491 155  i\n",
            "        LTChar                 491 144 494 155  t\n",
            "        LTChar                 494 144 500 155  y\n",
            "        LTAnno                  \n",
            "        LTChar                 502 144 506 155  r\n",
            "        LTChar                 506 144 511 155  a\n",
            "        LTChar                 511 144 514 155  t\n",
            "        LTChar                 514 144 519 155  e\n",
            "        LTChar                 519 144 523 155  s\n",
            "        LTChar                 523 144 526 155  .\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     318 129 527 140  In the following example, if tasked with un-\n",
            "        LTChar                 318 129 322 140  I\n",
            "        LTChar                 322 129 327 140  n\n",
            "        LTAnno                  \n",
            "        LTChar                 332 129 335 140  t\n",
            "        LTChar                 335 129 341 140  h\n",
            "        LTChar                 341 129 346 140  e\n",
            "        LTAnno                  \n",
            "        LTChar                 350 129 354 140  f\n",
            "        LTChar                 354 129 360 140  o\n",
            "        LTChar                 360 129 363 140  l\n",
            "        LTChar                 363 129 366 140  l\n",
            "        LTChar                 366 129 371 140  o\n",
            "        LTChar                 371 129 379 140  w\n",
            "        LTChar                 379 129 382 140  i\n",
            "        LTChar                 382 129 388 140  n\n",
            "        LTChar                 388 129 393 140  g\n",
            "        LTAnno                  \n",
            "        LTChar                 398 129 403 140  e\n",
            "        LTChar                 403 129 408 140  x\n",
            "        LTChar                 408 129 413 140  a\n",
            "        LTChar                 413 129 422 140  m\n",
            "        LTChar                 422 129 428 140  p\n",
            "        LTChar                 428 129 431 140  l\n",
            "        LTChar                 431 129 436 140  e\n",
            "        LTChar                 436 129 438 140  ,\n",
            "        LTAnno                  \n",
            "        LTChar                 444 129 447 140  i\n",
            "        LTChar                 447 129 450 140  f\n",
            "        LTAnno                  \n",
            "        LTChar                 455 129 458 140  t\n",
            "        LTChar                 458 129 463 140  a\n",
            "        LTChar                 463 129 467 140  s\n",
            "        LTChar                 467 129 473 140  k\n",
            "        LTChar                 473 129 478 140  e\n",
            "        LTChar                 478 129 483 140  d\n",
            "        LTAnno                  \n",
            "        LTChar                 488 129 496 140  w\n",
            "        LTChar                 496 129 499 140  i\n",
            "        LTChar                 499 129 502 140  t\n",
            "        LTChar                 502 129 508 140  h\n",
            "        LTAnno                  \n",
            "        LTChar                 513 129 518 140  u\n",
            "        LTChar                 518 129 524 140  n\n",
            "        LTChar                 524 129 527 140  -\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 115 526 126  derstanding the epidemiology and transmission of\n",
            "        LTChar                 307 115 313 126  d\n",
            "        LTChar                 313 115 318 126  e\n",
            "        LTChar                 318 115 321 126  r\n",
            "        LTChar                 321 115 325 126  s\n",
            "        LTChar                 325 115 329 126  t\n",
            "        LTChar                 329 115 333 126  a\n",
            "        LTChar                 333 115 339 126  n\n",
            "        LTChar                 339 115 344 126  d\n",
            "        LTChar                 344 115 347 126  i\n",
            "        LTChar                 347 115 353 126  n\n",
            "        LTChar                 353 115 358 126  g\n",
            "        LTAnno                  \n",
            "        LTChar                 361 115 364 126  t\n",
            "        LTChar                 364 115 370 126  h\n",
            "        LTChar                 370 115 374 126  e\n",
            "        LTAnno                  \n",
            "        LTChar                 377 115 382 126  e\n",
            "        LTChar                 382 115 387 126  p\n",
            "        LTChar                 387 115 390 126  i\n",
            "        LTChar                 390 115 396 126  d\n",
            "        LTChar                 396 115 401 126  e\n",
            "        LTChar                 401 115 409 126  m\n",
            "        LTChar                 409 115 412 126  i\n",
            "        LTChar                 412 115 418 126  o\n",
            "        LTChar                 418 115 421 126  l\n",
            "        LTChar                 421 115 426 126  o\n",
            "        LTChar                 426 115 432 126  g\n",
            "        LTChar                 432 115 437 126  y\n",
            "        LTAnno                  \n",
            "        LTChar                 440 115 445 126  a\n",
            "        LTChar                 445 115 450 126  n\n",
            "        LTChar                 450 115 456 126  d\n",
            "        LTAnno                  \n",
            "        LTChar                 458 115 461 126  t\n",
            "        LTChar                 461 115 465 126  r\n",
            "        LTChar                 465 115 470 126  a\n",
            "        LTChar                 470 115 475 126  n\n",
            "        LTChar                 475 115 480 126  s\n",
            "        LTChar                 480 115 488 126  m\n",
            "        LTChar                 488 115 491 126  i\n",
            "        LTChar                 491 115 495 126  s\n",
            "        LTChar                 495 115 500 126  s\n",
            "        LTChar                 500 115 503 126  i\n",
            "        LTChar                 503 115 508 126  o\n",
            "        LTChar                 508 115 514 126  n\n",
            "        LTAnno                  \n",
            "        LTChar                 516 115 522 126  o\n",
            "        LTChar                 522 115 526 126  f\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     307 102 526 113  COVID-19 a researcher may ask about salivary\n",
            "        LTChar                 307 102 315 113  C\n",
            "        LTChar                 315 102 323 113  O\n",
            "        LTChar                 322 102 330 113  V\n",
            "        LTChar                 330 102 334 113  I\n",
            "        LTChar                 334 102 342 113  D\n",
            "        LTChar                 342 102 346 113  -\n",
            "        LTChar                 346 102 351 113  1\n",
            "        LTChar                 351 102 357 113  9\n",
            "        LTAnno                  \n",
            "        LTChar                 361 102 366 113  a\n",
            "        LTAnno                  \n",
            "        LTChar                 370 102 373 113  r\n",
            "        LTChar                 373 102 378 113  e\n",
            "        LTChar                 378 102 383 113  s\n",
            "        LTChar                 383 102 387 113  e\n",
            "        LTChar                 387 102 392 113  a\n",
            "        LTChar                 392 102 396 113  r\n",
            "        LTChar                 396 102 401 113  c\n",
            "        LTChar                 401 102 407 113  h\n",
            "        LTChar                 407 102 412 113  e\n",
            "        LTChar                 412 102 415 113  r\n",
            "        LTAnno                  \n",
            "        LTChar                 419 102 428 113  m\n",
            "        LTChar                 428 102 433 113  a\n",
            "        LTChar                 433 102 438 113  y\n",
            "        LTAnno                  \n",
            "        LTChar                 442 102 447 113  a\n",
            "        LTChar                 447 102 452 113  s\n",
            "        LTChar                 452 102 457 113  k\n",
            "        LTAnno                  \n",
            "        LTChar                 461 102 466 113  a\n",
            "        LTChar                 466 102 472 113  b\n",
            "        LTChar                 472 102 477 113  o\n",
            "        LTChar                 477 102 483 113  u\n",
            "        LTChar                 483 102 486 113  t\n",
            "        LTAnno                  \n",
            "        LTChar                 490 102 494 113  s\n",
            "        LTChar                 494 102 500 113  a\n",
            "        LTChar                 500 102 503 113  l\n",
            "        LTChar                 503 102 506 113  i\n",
            "        LTChar                 506 102 511 113  v\n",
            "        LTChar                 511 102 516 113  a\n",
            "        LTChar                 516 102 521 113  r\n",
            "        LTChar                 521 102 526 113  y\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        320 78  426 87   1https://cord19.aws/\n",
            "      LTTextLineHorizontal     320 78  426 87   1https://cord19.aws/\n",
            "        LTChar                 320 81  323 87   1\n",
            "        LTChar                 323 78  329 87   h\n",
            "        LTChar                 329 78  334 87   t\n",
            "        LTChar                 334 78  340 87   t\n",
            "        LTChar                 340 78  345 87   p\n",
            "        LTChar                 345 78  350 87   s\n",
            "        LTChar                 350 78  356 87   :\n",
            "        LTChar                 356 78  361 87   /\n",
            "        LTChar                 361 78  366 87   /\n",
            "        LTChar                 366 78  372 87   c\n",
            "        LTChar                 372 78  377 87   o\n",
            "        LTChar                 377 78  383 87   r\n",
            "        LTChar                 383 78  388 87   d\n",
            "        LTChar                 388 78  393 87   1\n",
            "        LTChar                 393 78  399 87   9\n",
            "        LTChar                 399 78  404 87   .\n",
            "        LTChar                 404 78  409 87   a\n",
            "        LTChar                 409 78  415 87   w\n",
            "        LTChar                 415 78  420 87   s\n",
            "        LTChar                 420 78  426 87   /\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        16  527 36  567  0\n",
            "2\n",
            "0\n",
            "2\n",
            "      LTTextLineHorizontal     16  557 36  567  0\n",
            "        LTChar                 16  557 36  567  0\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  547 36  557  2\n",
            "        LTChar                 16  547 36  557  2\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  537 36  547  0\n",
            "        LTChar                 16  537 36  547  0\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  527 36  537  2\n",
            "        LTChar                 16  527 36  537  2\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        16  478 36  522  t\n",
            "c\n",
            "O\n",
            "7\n",
            "      LTTextLineHorizontal     16  516 36  522  t\n",
            "        LTChar                 16  516 36  522  t\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  508 36  516  c\n",
            "        LTChar                 16  508 36  516  c\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  493 36  508  O\n",
            "        LTChar                 16  493 36  508  O\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  478 36  488  7\n",
            "        LTChar                 16  478 36  488  7\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        16  461 36  468  ]\n",
            "      LTTextLineHorizontal     16  461 36  468  ]\n",
            "        LTChar                 16  461 36  468  ]\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        16  448 36  461  R\n",
            "      LTTextLineHorizontal     16  448 36  461  R\n",
            "        LTChar                 16  448 36  461  R\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        16  413 36  448  I\n",
            ".\n",
            "s\n",
            "c\n",
            "[\n",
            "      LTTextLineHorizontal     16  441 36  448  I\n",
            "        LTChar                 16  441 36  448  I\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  436 36  441  .\n",
            "        LTChar                 16  436 36  441  .\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  429 36  436  s\n",
            "        LTChar                 16  429 36  436  s\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  420 36  429  c\n",
            "        LTChar                 16  420 36  429  c\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  413 36  420  [\n",
            "        LTChar                 16  413 36  420  [\n",
            "        LTAnno                  \n",
            "    LTTextBoxHorizontal        16  237 36  403  3\n",
            "v\n",
            "6\n",
            "8\n",
            "1\n",
            "9\n",
            "0\n",
            ".\n",
            "7\n",
            "0\n",
            "0\n",
            "2\n",
            ":\n",
            "v\n",
            "i\n",
            "X\n",
            "r\n",
            "a\n",
            "      LTTextLineHorizontal     16  393 36  403  3\n",
            "        LTChar                 16  393 36  403  3\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  383 36  393  v\n",
            "        LTChar                 16  383 36  393  v\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  373 36  383  6\n",
            "        LTChar                 16  373 36  383  6\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  363 36  373  8\n",
            "        LTChar                 16  363 36  373  8\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  353 36  363  1\n",
            "        LTChar                 16  353 36  363  1\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  343 36  353  9\n",
            "        LTChar                 16  343 36  353  9\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  333 36  343  0\n",
            "        LTChar                 16  333 36  343  0\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  328 36  333  .\n",
            "        LTChar                 16  328 36  333  .\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  318 36  328  7\n",
            "        LTChar                 16  318 36  328  7\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  308 36  318  0\n",
            "        LTChar                 16  308 36  318  0\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  298 36  308  0\n",
            "        LTChar                 16  298 36  308  0\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  288 36  298  2\n",
            "        LTChar                 16  288 36  298  2\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  283 36  288  :\n",
            "        LTChar                 16  283 36  288  :\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  273 36  283  v\n",
            "        LTChar                 16  273 36  283  v\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  267 36  273  i\n",
            "        LTChar                 16  267 36  273  i\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  253 36  267  X\n",
            "        LTChar                 16  253 36  267  X\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  246 36  253  r\n",
            "        LTChar                 16  246 36  253  r\n",
            "        LTAnno                  \n",
            "      LTTextLineHorizontal     16  237 36  246  a\n",
            "        LTChar                 16  237 36  246  a\n",
            "        LTAnno                  \n",
            "    LTLine                     307 90  367 90   \n",
            "    LTTextLineHorizontal       16  403 36  408  \n",
            "      LTChar                   16  403 36  408  \n",
            "      LTAnno                    \n",
            "    LTTextLineHorizontal       16  408 36  413  \n",
            "      LTChar                   16  408 36  413  \n",
            "      LTAnno                    \n",
            "    LTTextLineHorizontal       16  468 36  473  \n",
            "      LTChar                   16  468 36  473  \n",
            "      LTAnno                    \n",
            "    LTTextLineHorizontal       16  473 36  478  \n",
            "      LTChar                   16  473 36  478  \n",
            "      LTAnno                    \n",
            "    LTTextLineHorizontal       16  488 36  493  \n",
            "      LTChar                   16  488 36  493  \n",
            "      LTAnno                    \n",
            "    LTTextLineHorizontal       16  522 36  527  \n",
            "      LTChar                   16  522 36  527  \n",
            "      LTAnno                    \n"
          ]
        }
      ],
      "source": [
        "url = 'http://export.arxiv.org/pdf/2007.09186v3'\n",
        "\n",
        "response = requests.get(url, allow_redirects=True)\n",
        "memory_file = BytesIO(response.content)\n",
        "\n",
        "pages = extract_pages(memory_file, page_numbers=[0], maxpages=1)\n",
        "show_ltitem_hierarchy(pages)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "U_hikXUD_J8c"
      },
      "source": [
        "### 4.8 Extract font information\n",
        "\n",
        "Pdfminer can also be used to extract font names and sizes used through the document."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "hPwyeVIF_J8c"
      },
      "outputs": [],
      "source": [
        "# https://stackoverflow.com/questions/22329508/pdf-miner-extract-font-size/69962482\n",
        "\n",
        "def show_ltitem_hierarchy(o: Any, depth=0):\n",
        "    \"\"\"Show location and text of LTItem and all its descendants\"\"\"\n",
        "    if depth == 0:\n",
        "        print('element                        fontname             text')\n",
        "        print('------------------------------ -------------------- -----')\n",
        "\n",
        "    print(\n",
        "        f'{get_indented_name(o, depth):<30.30s} '\n",
        "        f'{get_optional_fontinfo(o):<20.20s} '\n",
        "        f'{get_optional_text(o)}'\n",
        "    )\n",
        "\n",
        "    if isinstance(o, Iterable):\n",
        "        for i in o:\n",
        "            show_ltitem_hierarchy(i, depth=depth + 1)\n",
        "\n",
        "\n",
        "def get_indented_name(o: Any, depth: int) -> str:\n",
        "    \"\"\"Indented name of class\"\"\"\n",
        "    return '  ' * depth + o.__class__.__name__\n",
        "\n",
        "\n",
        "def get_optional_fontinfo(o: Any) -> str:\n",
        "    \"\"\"Font info of LTChar if available, otherwise empty string\"\"\"\n",
        "    if hasattr(o, 'fontname') and hasattr(o, 'size'):\n",
        "        return f'{o.fontname} {round(o.size)}pt'\n",
        "    return ''\n",
        "\n",
        "\n",
        "def get_optional_text(o: Any) -> str:\n",
        "    \"\"\"Text of LTItem if available, otherwise empty string\"\"\"\n",
        "    if hasattr(o, 'get_text'):\n",
        "        return o.get_text().strip()\n",
        "    return ''"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b34d562e-a9fe-4965-d96d-a187c242f072",
        "id": "11J9SDUJ_J8c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "element                        fontname             text\n",
            "------------------------------ -------------------- -----\n",
            "generator                                           \n",
            "  LTPage                                            \n",
            "    LTTextBoxHorizontal                             AWS CORD-19 Search: A Neural Search Engine for COVID-19\n",
            "Literature\n",
            "      LTTextLineHorizontal                          AWS CORD-19 Search: A Neural Search Engine for COVID-19\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L A\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L W\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L C\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L O\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L R\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L D\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L -\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L 1\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L S\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L c\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L :\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L A\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L N\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L S\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L c\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L E\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L f\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L C\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L O\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L V\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L I\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L D\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L -\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L 1\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          Literature\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L L\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\n",
            "Ben Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\n",
            "Ramesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\n",
            "Amazon Web Services AI\n",
            "parmib, liuall, arumae, nimpourd, suyogd, snivelyb,\n",
            "monamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\n",
            "zhiheng, bxiang, tahak @amazon.com\n",
            "      LTTextLineHorizontal                          Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L P\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L d\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L B\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L L\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L L\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L K\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L j\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L A\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L N\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L P\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L d\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L S\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L y\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L D\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L p\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L d\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          Ben Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L B\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L S\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L v\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L l\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L y\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L M\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L M\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L C\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L l\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L b\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L W\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L G\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L P\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L c\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L S\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L y\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L R\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L w\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L y\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L X\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L f\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L M\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          Ramesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L R\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L m\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L N\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L l\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L l\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L p\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L Z\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L e\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L H\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L B\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L X\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L g\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L T\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L h\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 UJKCWC+NimbusRomNo9L K\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L -\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L H\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          Amazon Web Services AI\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L z\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          parmib, liuall, arumae, nimpourd, suyogd, snivelyb,\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re p\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re b\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re u\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re u\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re e\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re p\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re u\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re d\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re s\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re u\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re y\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re g\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re d\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re s\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re v\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re e\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re y\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re b\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          monamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re c\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re b\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re y\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re w\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re g\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re w\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re p\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re c\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re e\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re s\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re h\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re y\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re x\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re f\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re e\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re l\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re p\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          zhiheng, bxiang, tahak @amazon.com\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re z\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re h\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re h\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re e\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re g\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re b\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re x\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re i\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re g\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re t\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re h\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re k\n",
            "        LTAnno                                      \n",
            "        LTChar                 MNDVMT+NimbusMonL-Re @\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re z\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re n\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re .\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re c\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re m\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             Abstract\n",
            "      LTTextLineHorizontal                          Abstract\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L A\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L b\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L s\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L a\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L c\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             Coronavirus disease (COVID-19) has been de-\n",
            "clared as a pandemic by WHO with thousands\n",
            "of cases being reported each day. Numerous\n",
            "scientiﬁc articles are being published on the\n",
            "disease raising the need for a service which\n",
            "can organize, and query them in a reliable fash-\n",
            "ion. To support this cause we present AWS\n",
            "CORD-19 Search (ACS), a public, COVID-19\n",
            "speciﬁc, neural search engine that is powered\n",
            "by several machine learning systems to sup-\n",
            "port natural language based searches. ACS\n",
            "with capabilities such as document ranking,\n",
            "passage ranking, question answering and topic\n",
            "classiﬁcation provides a scalable solution to\n",
            "COVID-19 researchers and policy makers in\n",
            "their search and discovery for answers to high\n",
            "priority scientiﬁc questions. We present a\n",
            "quantitative evaluation and qualitative analysis\n",
            "of the system against other leading COVID-\n",
            "19 search platforms. ACS is top performing\n",
            "across these systems yielding quality results\n",
            "which we detail with relevant examples in this\n",
            "work.\n",
            "      LTTextLineHorizontal                          Coronavirus disease (COVID-19) has been de-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          clared as a pandemic by WHO with thousands\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L H\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          of cases being reported each day. Numerous\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L N\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          scientiﬁc articles are being published on the\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          disease raising the need for a service which\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          can organize, and query them in a reliable fash-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L z\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          ion. To support this cause we present AWS\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L T\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          CORD-19 Search (ACS), a public, COVID-19\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          speciﬁc, neural search engine that is powered\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          by several machine learning systems to sup-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          port natural language based searches. ACS\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          with capabilities such as document ranking,\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          passage ranking, question answering and topic\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          classiﬁcation provides a scalable solution to\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          COVID-19 researchers and policy makers in\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          their search and discovery for answers to high\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          priority scientiﬁc questions. We present a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          quantitative evaluation and qualitative analysis\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          of the system against other leading COVID-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          19 search platforms. ACS is top performing\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          across these systems yielding quality results\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          which we detail with relevant examples in this\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          work.\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             1\n",
            "      LTTextLineHorizontal                          1\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L 1\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             Introduction\n",
            "      LTTextLineHorizontal                          Introduction\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L I\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L r\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L d\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L u\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L c\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L t\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L i\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L o\n",
            "        LTChar                 UJKCWC+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             With the global outbreak of Coronavirus disease\n",
            "(COVID-19) (Guan et al., 2020), the world is in\n",
            "turmoil. Medical researchers are required to work\n",
            "quickly to fully understand and to provide a form of\n",
            "intervention for the virus. Due to a large research\n",
            "focus on the disease, knowledge is published at a\n",
            "rapid rate throughout the world. One such reposi-\n",
            "tory of information is curated through the COVID-\n",
            "19 Open Research Dataset Challenge (CORD-19)\n",
            "(Wang et al., 2020). CORD-19 is a joint challenge\n",
            "put forth by Allen Institute (AI2), National Insti-\n",
            "tutes of Health (NIH), and the United States federal\n",
            "government via the White House. The objective of\n",
            "the challenge is to make sense of and extract useful\n",
            "knowledge across thousands of scholarly articles\n",
            "related to COVID-19. CORD-19 aims to connect\n",
            "      LTTextLineHorizontal                          With the global outbreak of Coronavirus disease\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          (COVID-19) (Guan et al., 2020), the world is in\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L G\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          turmoil. Medical researchers are required to work\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L M\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          quickly to fully understand and to provide a form of\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          intervention for the virus. Due to a large research\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          focus on the disease, knowledge is published at a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          rapid rate throughout the world. One such reposi-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          tory of information is curated through the COVID-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          19 Open Research Dataset Challenge (CORD-19)\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          (Wang et al., 2020). CORD-19 is a joint challenge\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L j\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          put forth by Allen Institute (AI2), National Insti-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L N\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          tutes of Health (NIH), and the United States federal\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L H\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L N\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L H\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L U\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          government via the White House. The objective of\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L H\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L T\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L j\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          the challenge is to make sense of and extract useful\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          knowledge across thousands of scholarly articles\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          related to COVID-19. CORD-19 aims to connect\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             the machine learning community with biomedical\n",
            "domain experts and policy makers in a race to iden-\n",
            "tify effective treatments and management policies\n",
            "for COVID-19. In accordance with this initiative\n",
            "our goal is to provide a scalable solution aimed at\n",
            "aiding COVID-19 researchers and policy makers in\n",
            "their search and discovery for answers to high prior-\n",
            "ity scientiﬁc questions. These questions should be\n",
            "understood in their natural language form; exam-\n",
            "ples include: “What do we know about COVID-19\n",
            "risk factors?”, “Which medications were most ben-\n",
            "eﬁcial in the 2002 SARS outbreak?”, and “Which is\n",
            "the most referenced paper doing study for Hydroxy-\n",
            "chloroquine?” To appropriately answer these ques-\n",
            "tions we require a system with a strong biomedical\n",
            "understanding of the natural queries and structured\n",
            "knowledge (Rotmensch et al., 2017).\n",
            "      LTTextLineHorizontal                          the machine learning community with biomedical\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          domain experts and policy makers in a race to iden-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          tify effective treatments and management policies\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          for COVID-19. In accordance with this initiative\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          our goal is to provide a scalable solution aimed at\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          aiding COVID-19 researchers and policy makers in\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          their search and discovery for answers to high prior-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          ity scientiﬁc questions. These questions should be\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L T\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          understood in their natural language form; exam-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ;\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          ples include: “What do we know about COVID-19\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L :\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L “\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          risk factors?”, “Which medications were most ben-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ?\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ”\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L “\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          eﬁcial in the 2002 SARS outbreak?”, and “Which is\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ?\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ”\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L “\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          the most referenced paper doing study for Hydroxy-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L H\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          chloroquine?” To appropriately answer these ques-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ?\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ”\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L T\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          tions we require a system with a strong biomedical\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          understanding of the natural queries and structured\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          knowledge (Rotmensch et al., 2017).\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 0\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 7\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             AWS CORD-19 Search (ACS) provides an easy\n",
            "to use search interface where researchers query us-\n",
            "ing natural language 1. ACS goes beyond keyword\n",
            "matching by understanding question semantics to\n",
            "efﬁciently ﬁnd relevant answers. As illustrated in\n",
            "Figure 1, we provide the system with a natural lan-\n",
            "guage query inquiring about IL-6 inhibitors and\n",
            "showcase the system response with relevant query\n",
            "components highlighted. This query can confuse a\n",
            "traditional search engine that relies solely on text\n",
            "matching as it can only capture term overlap be-\n",
            "tween a query and a document and may not nec-\n",
            "essarily be relevant to the researcher’s true intent\n",
            "(i.e. learning implicit relations). Our system, how-\n",
            "ever, establishes the relationship between IL-6 and\n",
            "severity of SARS-CoV-2 showing evidence where\n",
            "elevated IL-6 occur in a large number of patients\n",
            "with severe COVID-19 and higher mortality rates.\n",
            "In the following example, if tasked with un-\n",
            "derstanding the epidemiology and transmission of\n",
            "COVID-19 a researcher may ask about salivary\n",
            "      LTTextLineHorizontal                          AWS CORD-19 Search (ACS) provides an easy\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L W\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          to use search interface where researchers query us-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          ing natural language 1. ACS goes beyond keyword\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          matching by understanding question semantics to\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          efﬁciently ﬁnd relevant answers. As illustrated in\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ﬁ\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          Figure 1, we provide the system with a natural lan-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L F\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          guage query inquiring about IL-6 inhibitors and\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YNQEGC+NimbusRomNo9L I\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L L\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L -\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L 6\n",
            "        LTAnno                                      \n",
            "        LTChar                 YNQEGC+NimbusRomNo9L i\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L n\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L h\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L i\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L b\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L i\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L t\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L o\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L r\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          showcase the system response with relevant query\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          components highlighted. This query can confuse a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L T\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          traditional search engine that relies solely on text\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          matching as it can only capture term overlap be-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          tween a query and a document and may not nec-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L q\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          essarily be relevant to the researcher’s true intent\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ’\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          (i.e. learning implicit relations). Our system, how-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L (\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L )\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          ever, establishes the relationship between IL-6 and\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L L\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 6\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          severity of SARS-CoV-2 showing evidence where\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L A\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L R\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L S\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 2\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          elevated IL-6 occur in a large number of patients\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L L\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 6\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          with severe COVID-19 and higher mortality rates.\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L v\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L .\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          In the following example, if tasked with un-\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L x\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L ,\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L w\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          derstanding the epidemiology and transmission of\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L p\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L l\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L g\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L d\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L i\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L n\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L f\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          COVID-19 a researcher may ask about salivary\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L C\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L O\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L V\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L I\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L D\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L -\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 9\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L c\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L h\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L e\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L r\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L m\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L s\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L k\n",
            "        LTAnno                                      \n",
            "        LTChar                 YHBHHN+NimbusRomNo9L a\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L b\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L o\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L u\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L t\n",
            "        LTAnno                                      \n",
            "        LTChar                 YNQEGC+NimbusRomNo9L s\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L a\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L l\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L i\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L v\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L a\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L r\n",
            "        LTChar                 YNQEGC+NimbusRomNo9L y\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             1https://cord19.aws/\n",
            "      LTTextLineHorizontal                          1https://cord19.aws/\n",
            "        LTChar                 YHBHHN+NimbusRomNo9L 1\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re h\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re t\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re t\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re p\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re s\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re :\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re /\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re /\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re c\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re o\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re r\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re d\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re 1\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re 9\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re .\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re a\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re w\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re s\n",
            "        LTChar                 MNDVMT+NimbusMonL-Re /\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             0\n",
            "2\n",
            "0\n",
            "2\n",
            "      LTTextLineHorizontal                          0\n",
            "        LTChar                 Times-Roman 10pt     0\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          2\n",
            "        LTChar                 Times-Roman 10pt     2\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          0\n",
            "        LTChar                 Times-Roman 10pt     0\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          2\n",
            "        LTChar                 Times-Roman 10pt     2\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             t\n",
            "c\n",
            "O\n",
            "7\n",
            "      LTTextLineHorizontal                          t\n",
            "        LTChar                 Times-Roman 6pt      t\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          c\n",
            "        LTChar                 Times-Roman 9pt      c\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          O\n",
            "        LTChar                 Times-Roman 14pt     O\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          7\n",
            "        LTChar                 Times-Roman 10pt     7\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             ]\n",
            "      LTTextLineHorizontal                          ]\n",
            "        LTChar                 Times-Roman 7pt      ]\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             R\n",
            "      LTTextLineHorizontal                          R\n",
            "        LTChar                 Times-Roman 13pt     R\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             I\n",
            ".\n",
            "s\n",
            "c\n",
            "[\n",
            "      LTTextLineHorizontal                          I\n",
            "        LTChar                 Times-Roman 7pt      I\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          .\n",
            "        LTChar                 Times-Roman 5pt      .\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          s\n",
            "        LTChar                 Times-Roman 8pt      s\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          c\n",
            "        LTChar                 Times-Roman 9pt      c\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          [\n",
            "        LTChar                 Times-Roman 7pt      [\n",
            "        LTAnno                                      \n",
            "    LTTextBoxHorizontal                             3\n",
            "v\n",
            "6\n",
            "8\n",
            "1\n",
            "9\n",
            "0\n",
            ".\n",
            "7\n",
            "0\n",
            "0\n",
            "2\n",
            ":\n",
            "v\n",
            "i\n",
            "X\n",
            "r\n",
            "a\n",
            "      LTTextLineHorizontal                          3\n",
            "        LTChar                 Times-Roman 10pt     3\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          v\n",
            "        LTChar                 Times-Roman 10pt     v\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          6\n",
            "        LTChar                 Times-Roman 10pt     6\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          8\n",
            "        LTChar                 Times-Roman 10pt     8\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          1\n",
            "        LTChar                 Times-Roman 10pt     1\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          9\n",
            "        LTChar                 Times-Roman 10pt     9\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          0\n",
            "        LTChar                 Times-Roman 10pt     0\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          .\n",
            "        LTChar                 Times-Roman 5pt      .\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          7\n",
            "        LTChar                 Times-Roman 10pt     7\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          0\n",
            "        LTChar                 Times-Roman 10pt     0\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          0\n",
            "        LTChar                 Times-Roman 10pt     0\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          2\n",
            "        LTChar                 Times-Roman 10pt     2\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          :\n",
            "        LTChar                 Times-Roman 6pt      :\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          v\n",
            "        LTChar                 Times-Roman 10pt     v\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          i\n",
            "        LTChar                 Times-Roman 6pt      i\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          X\n",
            "        LTChar                 Times-Roman 14pt     X\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          r\n",
            "        LTChar                 Times-Roman 7pt      r\n",
            "        LTAnno                                      \n",
            "      LTTextLineHorizontal                          a\n",
            "        LTChar                 Times-Roman 9pt      a\n",
            "        LTAnno                                      \n",
            "    LTLine                                          \n",
            "    LTTextLineHorizontal                            \n",
            "      LTChar                   Times-Roman 5pt      \n",
            "      LTAnno                                        \n",
            "    LTTextLineHorizontal                            \n",
            "      LTChar                   Times-Roman 5pt      \n",
            "      LTAnno                                        \n",
            "    LTTextLineHorizontal                            \n",
            "      LTChar                   Times-Roman 5pt      \n",
            "      LTAnno                                        \n",
            "    LTTextLineHorizontal                            \n",
            "      LTChar                   Times-Roman 5pt      \n",
            "      LTAnno                                        \n",
            "    LTTextLineHorizontal                            \n",
            "      LTChar                   Times-Roman 5pt      \n",
            "      LTAnno                                        \n",
            "    LTTextLineHorizontal                            \n",
            "      LTChar                   Times-Roman 5pt      \n",
            "      LTAnno                                        \n"
          ]
        }
      ],
      "source": [
        "url = 'http://export.arxiv.org/pdf/2007.09186v3'\n",
        "\n",
        "response = requests.get(url, allow_redirects=True)\n",
        "memory_file = BytesIO(response.content)\n",
        "\n",
        "pages = extract_pages(memory_file, page_numbers=[0], maxpages=1)\n",
        "output = show_ltitem_hierarchy(pages)\n",
        "output"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ELbgLdFp-A8Z"
      },
      "source": [
        "### 4.9 Functions to separate text, line and figure objects"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def text_LT_elems(LT_elem_text):\n",
        "\n",
        "    text = []\n",
        "\n",
        "    for page in LT_elem_text:\n",
        "      row = []\n",
        "      for elem in page:\n",
        "        if isinstance(elem, LTTextLineHorizontal) or isinstance(elem, LTTextBoxHorizontal):\n",
        "          row.append(elem)\n",
        "      text.append(row)\n",
        "\n",
        "    return text\n",
        "\n",
        "def lines_LT_elems(LT_elem_text):\n",
        "\n",
        "    lines = []\n",
        "\n",
        "    for page in LT_elem_text:\n",
        "      row = []\n",
        "      for elem in page:\n",
        "        if isinstance(elem, LTLine):\n",
        "          row.append(elem)\n",
        "      lines.append(row)\n",
        "\n",
        "    return lines\n",
        "\n",
        "\n",
        "def figures_LT_elems(LT_elem_text):\n",
        "\n",
        "    figures = []\n",
        "\n",
        "    for page in LT_elem_text:\n",
        "      row = []\n",
        "      for elem in page:\n",
        "        if isinstance(elem, LTFigure):\n",
        "          row.append(elem)\n",
        "      figures.append(row)\n",
        "\n",
        "    return figures"
      ],
      "metadata": {
        "id": "erbo4dnCTCgO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Separate text objects."
      ],
      "metadata": {
        "id": "OgVSkv1OMhsT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "text = text_LT_elems(LT_elem_text)"
      ],
      "metadata": {
        "id": "aMlekOKKTKIV"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ab5ebd7e-dd28-443e-82ba-c13c8b4ddac7",
        "id": "Yi5J91V3BCTg"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[<LTTextBoxHorizontal(0) 103.371,746.071,493.659,776.357 'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 77.083,639.527,523.454,734.427 'Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\\nBen Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\\nRamesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\\nAmazon Web Services AI\\nparmib, liuall, arumae, nimpourd, suyogd, snivelyb,\\nmonamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\\nzhiheng, bxiang, tahak @amazon.com\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 158.891,605.891,203.376,617.847 'Abstract\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 88.261,322.925,274.910,595.902 'Coronavirus disease (COVID-19) has been de-\\nclared as a pandemic by WHO with thousands\\nof cases being reported each day. Numerous\\nscientiﬁc articles are being published on the\\ndisease raising the need for a service which\\ncan organize, and query them in a reliable fash-\\nion. To support this cause we present AWS\\nCORD-19 Search (ACS), a public, COVID-19\\nspeciﬁc, neural search engine that is powered\\nby several machine learning systems to sup-\\nport natural language based searches. ACS\\nwith capabilities such as document ranking,\\npassage ranking, question answering and topic\\nclassiﬁcation provides a scalable solution to\\nCOVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high\\npriority scientiﬁc questions. We present a\\nquantitative evaluation and qualitative analysis\\nof the system against other leading COVID-\\n19 search platforms. ACS is top performing\\nacross these systems yielding quality results\\nwhich we detail with relevant examples in this\\nwork.\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 72.000,300.286,77.978,312.242 '1\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 89.933,300.286,154.814,312.242 'Introduction\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 71.182,76.568,292.082,290.715 'With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 307.276,389.247,527.359,616.943 'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 306.883,101.737,527.356,385.118 'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 319.928,77.606,425.632,87.419 '1https://cord19.aws/\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 16.340,526.960,36.340,566.960 '0\\n2\\n0\\n2\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 16.340,478.080,36.340,521.960 't\\nc\\nO\\n7\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 16.340,461.420,36.340,468.080 ']\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 16.340,448.080,36.340,461.420 'R\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 16.340,413.100,36.340,448.080 'I\\n.\\ns\\nc\\n[\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 16.340,237.000,36.340,403.100 '3\\nv\\n6\\n8\\n1\\n9\\n0\\n.\\n7\\n0\\n0\\n2\\n:\\nv\\ni\\nX\\nr\\na\\n'>,\n",
              "  <LTTextLineHorizontal 16.340,403.100,36.340,408.100 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,408.100,36.340,413.100 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,468.080,36.340,473.080 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,473.080,36.340,478.080 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,488.080,36.340,493.080 ' \\n'>,\n",
              "  <LTTextLineHorizontal 16.340,521.960,36.340,526.960 ' \\n'>],\n",
              " [<LTTextBoxHorizontal(0) 82.729,766.636,101.138,774.003 'Query\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 181.035,766.636,201.485,774.003 'Article\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 279.865,766.636,308.101,774.003 'Response\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 81.749,742.699,166.567,758.658 '“Are IL-6 inhibitors key to\\nCOVID-19?”\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 180.744,725.510,266.356,758.710 'SARS-CoV-2 and COVID-\\n19:\\nis interleukin-6 (IL-6)\\nthe ’culprit lesion’ of ARDS\\nonset?\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 278.886,716.915,448.769,758.658 '“...monoclonal antibody against IL-6, is being tested in a\\nclinical trial against COVID-19 (Sarilimumab COVID-\\n19). Another drug that showed potential inhibition of IL-\\n6 related JAK/STAT pathway is glatiramer acetate which\\nshowed potential to downregulate both IL-17 and IL-6”\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 462.055,766.636,486.423,774.003 'Topic(s)\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 462.280,751.292,517.470,758.658 'Clinical Treatment\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 81.749,682.539,167.790,707.093 '“When is the salivary vi-\\nral load highest for COVID-\\n19?”\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 181.297,673.944,266.358,707.093 'Elective, Non-urgent Pro-\\ncedures\\nAesthetic\\nand\\nSurgery in the Wake of\\nSARS–COVID-19 [...]\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 278.886,673.944,448.770,707.144 '“Patients with COVID-19 have demonstrated high viral\\nloads in the upper respiratory tract soon after their infec-\\ntion, with the highest load assumed to be the day before\\nsymptoms appear.”\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 462.280,699.726,517.470,707.093 'Clinical Treatment\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 81.749,648.161,167.790,664.122 '“Is convalescent plasma ther-\\napy a precursor to vaccine?”\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 181.297,648.161,236.745,664.122 'COVID-19\\nplasma transfusion\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 227.566,656.755,265.135,664.122 'convalescent\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 278.886,639.566,448.766,664.122 '“...a passive immunotherapy, has been used as a possi-\\nble therapeutic option when no proven speciﬁc vaccine\\nor drug is available for emerging infections. ”\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 462.052,639.566,519.098,664.122 'Clinical Treatment,\\nImmunology, Lab\\nTrials\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 72.000,592.787,525.547,626.661 'Figure 1: Sample natural language queries and responses using AWS CORD-19 Search. The response ﬁeld is\\ntaken directly from the top result of this service. Also provided are the article titles where the answer is taken from\\nas well as selected topics for the response.\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 72.000,368.424,292.083,569.273 'viral load. We provide a further advantage from\\nan organizational perspective using topics with do-\\nmain speciﬁc relations. The user can select clinical-\\ntreatments to reﬁne search articles to understand\\nthat “...the highest load [is] assumed to be the day\\nbefore symptoms appear.” Similarly if commis-\\nsioned to understand convalescent plasma therapy\\nand selecting the three topics as shown in the last\\nexample the system identiﬁes the most relevant ar-\\nticle and answers the question in a highlight. There\\nare a few search systems have been developed to\\nsupport COVID-19 search leveraging CORD-19\\ncorpus. We evaluate the ACS against two other\\nsearch engines, and ACS is proved to be a top-\\nperforming system.\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 71.727,282.940,292.075,361.595 'In the following sections we will discuss the\\nvarious individual Amazon Web Services (AWS)\\nproducts that allow ACS to provide this function-\\nality, highlighting the value to scientists who can\\nquickly query, validate their research, and advance\\ntheir investigations.\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 72.000,249.561,178.342,261.517 '2 System Overview\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 71.607,76.568,292.172,236.518 'AWS CORD-19 Search (ACS) relies on a deep\\nsemantic search model to return a ranked list of\\nrelevant documents. This system performs docu-\\nment ranking, passage ranking, question answer-\\ning and FAQ matching. Additionally, it leverages\\nknowledge graphs and topic modeling to enrich the\\nbiomedical data and boost searching performance.\\nThis provides a scalable solution to COVID-19 re-\\nsearchers and policy makers. In this section we\\npresent the overall architecture of the system and a\\ncloser look at several individual components. Fig-\\nure 2 provides an overview of this architecture.\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 306.883,418.693,527.452,569.098 '2.1 Amazon Kendra\\nAmazon Kendra2 is a semantic search and question\\nanswering service provided by AWS for enterprise\\ncustomers. Kendra allows its customers to power\\nnatural language based searching across their own\\ndata. As response to the worldwide COVID-19\\npandemic, Kendra has also been tooled to support\\nCOVID-19 related searching and question answer-\\ning using the document corpus from CORD-19.\\nThe end-to-end Kendra system consists of several\\ncomponents.\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 307.276,323.849,527.356,416.053 'To complement the extracted answers, Kendra\\nuses a deep learning based semantic search model\\nto return a ranked list of relevant documents. Ama-\\nzon Kendra’s ability to understand natural language\\nquestions is at the core of its search engine return-\\ning the most relevant passage and related docu-\\nments\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 320.294,220.508,527.359,312.789 '• Document ranking (DR) - Like any tradi-\\ntional search engine, Kendra returns a ranked\\nlist of relevant documents based on the user’s\\nquery to fulﬁll their information needs. A\\ndeep semantic search model is used to under-\\nstand natural language questions in addition\\nto the keyword search.\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 320.294,144.329,527.360,209.511 '• Passage ranking (PR) & Question Answer-\\ning (QA) - Kendra ranks the passages and\\ntries to extract the answer from the top rele-\\nvant passages with a deep reading comprehen-\\nsion model.\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 320.294,95.249,527.351,133.332 '• FAQ Matching (FAQM) - If there exists Fre-\\nquently Answered Questions and their corre-\\nsponding answers, Kendra will automatically\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 319.928,77.606,484.810,87.419 '2https://aws.amazon.com/kendra/\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 237.044,540.631,360.501,550.594 'Figure 2: System Architecture.\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 93.818,492.407,290.280,516.866 'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 72.000,325.745,292.082,472.146 'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 72.000,297.540,197.858,308.449 '2.2 Comprehend Medical\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 71.575,141.116,292.075,289.172 'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 307.276,506.033,470.563,516.942 '2.3 COVID-19 Knowledge Graph\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 307.276,148.089,527.359,497.728 'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 84.653,118.369,217.255,128.182 '3https://docs.aws.amazon.\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 72.000,98.443,245.150,117.372 'com/comprehend/latest/dg/\\nhow-document-classification.html\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 84.653,87.569,271.054,97.382 '4https://aws.amazon.com/comprehend/\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 72.000,77.606,115.039,86.572 'medical/\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 307.276,122.881,394.156,133.790 '2.4 Topic Models\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 306.937,76.568,525.549,114.576 'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 186.115,447.945,411.429,457.908 'Figure 3: Visualization of COVID-19 Knowledge graph.\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 71.607,304.876,292.075,424.179 'for topic modeling is Latent Dirichlet Allocation\\n(LDA) (Blei et al., 2003), a generative model which\\ngroups documents together by observed content,\\noften giving each document a mixture of topics it\\nbelongs to. An extension of this work termed Z-\\nlabel LDA (Andrzejewski and Zhu, 2009) utilizes\\npriors to allow the model to force certain topics\\nwhich the users have manually curated, or wish to\\nsee clustered together.\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 71.607,105.493,292.083,295.072 '2.4.1 Generating Topics\\nFor the purposes of this work we experimented\\nwith 5, 10, and 20 topic models 5. The outputs\\nof each clustering size were manually inspected\\nand topic labels were provided by us when inspect-\\ning the top ten terms for each cluster. The ﬁnal\\ngranularity of the topic models was chosen by man-\\nually deleting and merging topics from the 20 topic\\nmodel. In general we were able to clearly extract\\ngroups which centered around important topics in-\\ncluding virology, proteomics, epidemiology, and\\ncellular biology to name a few. However when\\nfaced with 20 topics the less populated ones tended\\nto be noisy, and captured peripheral information\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 307.003,182.933,527.359,424.179 'present in the input, such as language (e.g. Spanish\\nand French) or provide redundancies with existing\\ntopics (e.g. two topics for Inﬂuenza). As a con-\\ntrol we ran a publicly available implementation of\\nZ-label LDA 6 with no priors which yields topics\\nclose to those extracted using Comprehend Topic\\nModeling. Although similar we observed better\\ndeﬁnition in certain groups (such as pulmonary\\ndiseases, and policy/industry), and decided to use\\nthis as the curation entry-point. Our goal was to\\nlimit these topics to ten, and compile them in ad-\\nvance as much as possible. With the help of medi-\\ncal professionals we eliminated and combined top-\\nics to form the following: Vaccines/immunology,\\nGenomics, Public health Policies, Epidemiology,\\nClinical Treatment, Virology, Inﬂuenza, Healthcare\\nIndustry, Pulmonary Infections, and Lab Trials (hu-\\nman).\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 307.276,105.493,527.349,173.130 '2.4.2 Multi-Label Classiﬁcation\\nHaving to manually feed a topic model and re-train\\non the entire corpus once new data becomes avail-\\nable is largely inefﬁcient. We therefore used the\\ntopic model labels to train a multi-label classiﬁer\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 84.653,86.950,290.268,97.382 '5These models were trained using CORD-19 data available\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 72.000,76.987,148.708,85.954 'as of April 6th, 2020.\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 319.928,85.850,511.709,97.382 '6http://pages.cs.wisc.edu/˜andrzeje/\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 307.276,77.606,414.873,86.572 'research/zl_lda.html\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 114.189,763.315,177.505,774.224 'Search Engine\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 194.622,763.315,216.189,774.224 'P@1\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 236.578,763.315,258.145,774.224 'P@5\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 275.807,763.315,302.829,774.224 'P@10\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 317.764,763.315,429.298,774.224 'P@20 R@10 R@20\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 442.138,763.315,484.302,774.224 'ndcg@20\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 260.421,745.841,337.123,756.750 'Keyword Queries\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 127.525,713.265,164.169,737.723 'ACS\\nCovidex\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 113.245,699.716,220.405,737.723 '0.5250\\n0.3421\\nCOVID-19 RE 0.5750\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 232.359,699.792,262.361,737.799 '0.5650\\n0.2316\\n0.5650\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 274.314,699.716,304.318,737.799 '0.5325\\n0.2079\\n0.4775\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 316.270,699.716,346.274,737.799 '0.4775\\n0.1658\\n0.4412\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 358.220,699.716,388.228,737.799 '0.0260\\n0.0109\\n0.0236\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 400.176,699.716,430.185,737.799 '0.0459\\n0.0173\\n0.0429\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 448.216,699.716,478.220,737.799 '0.4380\\n0.1633\\n0.4022\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 236.361,682.241,361.183,693.150 'Natural Language Questions\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 127.525,649.665,164.169,674.123 'ACS\\nCovidex\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 113.245,636.116,220.405,674.199 '0.8750\\n0.4750\\nCOVID-19 RE 0.6000\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 232.360,636.116,262.361,674.199 '0.7000\\n0.4800\\n0.5300\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 274.317,636.116,304.318,674.199 '0.6400\\n0.4225\\n0.4925\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 316.273,636.116,346.274,674.199 '0.5550\\n0.3625\\n0.4600\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 358.220,636.116,388.230,674.199 '0.0345\\n0.0204\\n0.0267\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 400.175,636.116,430.176,674.199 '0.0582\\n0.0356\\n0.0474\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 448.219,636.116,478.220,674.199 '0.5357\\n0.3229\\n0.4133\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 192.835,613.165,404.401,623.128 'Table 1: Evaluation results on TREC-COVID dataset\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 71.640,497.195,290.269,589.399 '(Read et al., 2011). To evaluate the performance\\nof this model we calculate the average F1 across\\ntest samples by calculating the set overlap between\\nour gold standard (topic model) and system labels\\n(multi-label classiﬁcation). This held-out test set\\ncontains 20% of the CORD-19 data available at the\\ntime.\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 72.000,429.449,290.273,494.555 'Using this metric our trained model achieved an\\naverage F1 of 91.92, with on average 2.37 labels\\nper document. Fewer than 1% of the documents\\nin the test set received no label, using 0.5 as the\\nconﬁdence threshold.\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 72.000,406.223,145.620,418.179 '3 Evaluation\\n'>,\n",
              "  <LTTextBoxHorizontal(28) 71.182,196.446,292.177,397.044 'ACS supports document ranking (DR), passage\\nranking (PR) and question answering (QA) on the\\ntop suggested results. In this section we evaluate\\nthe overall performance of ACS with publicly avail-\\nable datasets. The TREC-COVID track (Voorhees\\net al., 2020) built DR test collections that contain\\ndocument relevance ground truth and are used to\\nassess DR only. For PR and QA, we manually anno-\\ntate questions collected from CovidQA (Tang et al.,\\n2020), the ﬁrst open sourced Covid-19 QA dataset.\\nThe ACS results are compared against Covidex7\\nand Google Covid-19 Research Explorer8 (COVID-\\n19 RE), which are also COVID-19 public search\\nengines that facilitate DR and answer highlighting\\nin passage.\\n'>,\n",
              "  <LTTextBoxHorizontal(29) 72.000,115.151,292.075,193.806 'The TREC-COVID challenge track contains 40\\ntopic sets along with their document relevance\\njudgement. The topic sets are written by its or-\\nganizers with biomedical training, and motivated\\nby search submitted to the National Library of\\nMedicine and social media. Each topic consists\\n'>,\n",
              "  <LTTextBoxHorizontal(30) 84.653,87.569,272.049,108.256 '7https://covidex.ai/\\n8https://covid19-research-explorer.\\n'>,\n",
              "  <LTTextBoxHorizontal(31) 72.000,77.606,136.558,86.572 'appspot.com/\\n'>,\n",
              "  <LTTextBoxHorizontal(32) 307.276,361.703,527.359,589.399 'of three ﬁelds with different levels of granularity, a\\nkeyword-based query (KQ), a more precise natural\\nlanguage question (NQ), and a longer descriptive\\nnarrative. The DR assessments are performed fol-\\nlowing the TREC pooling mechanism. The par-\\nticipants submitted ranked lists of documents for\\neach topic set, based on which a depth of 10 to 20\\ndocuments are pooled and combined as a collection\\nof (q, D) pairs. The pairs are then assessed by an-\\nnotators with in-domain expertise. There are three\\nrounds of judgement available that corresponding\\nto three different versions of CORD-19 corpus. To\\nensure sufﬁcient coverage of annotation, we aggre-\\ngate all rounds results into 33,064 (q, D) relevance\\njudgements. Since the document id may change\\nacross versions, we map ids from each round to the\\nMay 19 release of CORD-19 corpus.\\n'>,\n",
              "  <LTTextBoxHorizontal(33) 306.883,253.109,527.355,358.862 'To collect DR results from the three systems, we\\ncrawled the top 50 articles by querying the engines\\nwith KQ and its NQ variation from the topic sets on\\nJune 15, 2020. The crawled data, which are char-\\nacterized by the article title and link, are mapped\\nto May 19 CORD-19 corpus. Note that articles\\nthat cannot be found in the corpus are removed to\\nensure fair comparison.\\n'>,\n",
              "  <LTTextBoxHorizontal(34) 306.883,103.867,527.356,250.268 'We use the standard DR metrics in our evalua-\\ntion, namely, the precision and recall at k (P@k,\\nR@k) and normalized discounted cumulative gain\\nin the top k documents (NDCG@k). Note that we\\nevaluate with k up to 20 since TREC-COVID has\\na pooling depth with 20 at most. Table 1 presents\\nthe DR performance of the engines over KQ and\\nNQ, respectively. ACS performs consistently better\\nthan the other engines on NQ and mostly on KQ,\\nand all engines perform better on NQ comparing\\nwith KQ.\\n'>,\n",
              "  <LTTextBoxHorizontal(35) 307.276,76.568,525.549,101.026 'In addition, we are able to evaluate how robust\\neach system is against query variation from KQ\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 173.032,756.541,236.348,767.450 'Search Engine\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 251.554,749.766,296.543,774.224 'Top3\\nEM F1\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 313.646,749.766,358.635,774.224 'Top10\\nEM F1\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 375.738,749.766,420.727,774.224 'Top30\\nEM F1\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 173.032,703.642,269.281,741.725 '11.7\\nACS\\nCovidex\\n0.90\\nCOVID-19 RE 10.0\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 281.237,703.642,300.328,741.725 '35.6\\n18.9\\n31.8\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 312.282,703.642,331.373,741.725 '16.0\\n2.10\\n13.5\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 343.329,703.642,362.420,741.725 '42.2\\n24.2\\n36.9\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 374.374,703.642,393.465,741.725 '26.0\\n3.60\\n18.2\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 405.421,703.642,424.512,741.725 '50.4\\n27.6\\n43.8\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 154.912,680.691,442.323,690.654 'Table 2: Top results variation from KQ to NQ on TREC-COVID dataset\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 140.305,644.760,203.621,655.669 'Search Engine\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 221.679,637.986,243.246,648.895 'P@1\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 140.305,591.861,247.463,629.944 '0.4074\\nACS\\n0.4074\\nCovidex\\nCOVID-19 RE 0.4074\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 263.635,637.986,285.203,662.444 'PR\\nP@2\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 259.418,591.861,289.419,629.944 '0.5370\\n0.4444\\n0.3704\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 305.592,637.986,327.159,648.895 'P@3\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 347.545,637.986,369.112,648.895 'P@1\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 301.374,591.861,331.376,629.944 '0.4938\\n0.4074\\n0.3580\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 343.328,591.861,373.328,629.944 '0.3333\\n0.1481\\n0.2963\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 389.501,637.986,411.069,662.444 'QA\\nP@2\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 385.284,591.861,415.284,629.944 '0.2593\\n0.1852\\n0.2407\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 431.458,637.986,453.025,648.895 'P@3\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 427.239,591.937,457.241,629.868 '0.2099\\n0.2099\\n0.2593\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 109.892,568.910,487.345,578.873 'Table 3: Passage ranking (PR) and question answering (QA) performance on CovidQA dataset\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 203.824,478.151,238.804,492.777 'q , KQk\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 71.607,317.448,292.083,545.144 'to NQ. Ideally, the results shall remain unchanged\\nwith query variation that requests the same infor-\\nmation. We deﬁne exact match (EM) and F1 score\\namong top k results to evaluate the robustness. Let\\nQ be the topic sets, and N Qk\\nq denote the\\ntop k searching results of natural language ques-\\ntion and keyword query corresponding to the same\\ntopic q ∈ Q, respectively. As shown in Eq.(1), we\\ntake the top k results KQk\\nq as ground truth, and\\ncompute the average exact match and F1 score of\\nthe top k results N Qk\\nq , and then average over all\\nqueries. The article title string is used as com-\\nparison key, and EM and F1 are standard that the\\nmaximum is taken over all the ground truth arti-\\ncles. Table 2 demonstrates that ACS has the best\\ncapability to provide consistent results with query\\nvariation.\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 109.069,288.073,190.977,298.982 'EM (N Q, KQ, k)\\n'>,\n",
              "  <LTTextBoxHorizontal(28) 100.584,259.761,109.069,270.670 '=\\n'>,\n",
              "  <LTTextBoxHorizontal(29) 110.264,252.278,124.950,278.050 '1\\n|Q|\\n'>,\n",
              "  <LTTextBoxHorizontal(30) 147.349,252.278,153.088,278.050 '1\\nk\\n'>,\n",
              "  <LTTextBoxHorizontal(31) 128.270,265.696,144.027,276.605 '(cid:88)\\n'>,\n",
              "  <LTTextBoxHorizontal(32) 127.963,247.066,144.336,255.036 'q∈Q\\n'>,\n",
              "  <LTTextBoxHorizontal(33) 156.386,265.696,172.143,281.937 'k\\n(cid:88)\\n'>,\n",
              "  <LTTextBoxHorizontal(34) 157.413,247.244,171.117,255.214 'i=1\\n'>,\n",
              "  <LTTextBoxHorizontal(35) 173.962,259.761,215.850,272.805 'em(N Qk\\n'>,\n",
              "  <LTTextBoxHorizontal(36) 211.441,257.634,261.684,272.805 'q (i), KQk\\nq )\\n'>,\n",
              "  <LTTextBoxHorizontal(37) 277.548,258.972,290.268,269.881 '(1)\\n'>,\n",
              "  <LTTextBoxHorizontal(38) 71.607,76.568,292.075,222.969 'Next, we evaluate the performance of PR and QA\\nwith CovidQA dataset. CovidQA contains 27 ques-\\ntions and their answers from 124 question-article\\npairs, which are selected by in-domain volunteers\\nas the most promising COVID-19 literature. Since\\nthe annotation does not possess sufﬁcient answer\\ncoverage over the entire corpus, we leverage our\\ninternal annotation resources to make PR and QA\\njudgement. More explicitly, we crawled the top 3\\nresults characterized by the article, displayed pas-\\nsage and the highlighted text snippet on June 15,\\n'>,\n",
              "  <LTTextBoxHorizontal(39) 307.276,425.841,527.359,545.144 '2020 from ACS, Covidex and COVID-19 RE, re-\\nspectively. After that, the annotators assess PR\\nand QA in terms of whether the displayed passage\\ncontains relevant information and whether the high-\\nlighted text snippet answers the given question. To\\navoid bias, the crawled results are combined and\\nshufﬂed randomly, therefore, the annotators do not\\nhave access to the source engine and the rank posi-\\ntion information during the judgement.\\n'>,\n",
              "  <LTTextBoxHorizontal(40) 306.457,136.037,527.486,404.380 'Table 3 presents precision of top PR and QA\\nresults. We use P@k instead of EM and F1 to eval-\\nuate QA since ACS highlights the answer while\\nCovidex and COVID-19 RE highlight the sentence\\nthat contains the answer. Instead of extracting the\\nanswer, it is more reasonable to judge whether the\\nhighlighted text answers the question and compute\\nprecision. ACS achieves better accuracy on both\\nPR and QA on most metrics. Note that ACS high-\\nlights answer snippet for at most three passages\\n, while Covidex and COVID-19 RE highlight all\\ndisplayed passages. This explains why ACS P@3\\nunderperforms COVID-19 RE. Explicitly with an\\nexample, Figure 4 displays the topmost results of\\nquerying “What is the incubation period of virus”\\nfrom the three systems. ACS highlights the answer\\nin the passage. In contrast, Covidex presents an\\narticle published at 2004 with information of virus\\nwhich is irrelevant with coronavirus, and COVID-\\n19 RE does not answer the question at all.\\n'>,\n",
              "  <LTTextBoxHorizontal(41) 307.276,76.568,525.549,114.576 'As evident from the above results, ACS is one\\nof the top-performing systems that provides high\\nquality informative results over CORD-19 search.\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 142.598,764.340,159.220,773.066 'ACS\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 286.114,764.340,315.423,773.066 'Covidex\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 424.543,764.340,476.695,773.066 'COVID-19 RE\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 226.245,748.431,375.285,757.157 'What is the incubation period of the virus?\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 234.561,730.257,342.408,738.982 'Deadly viral syndrome mimics\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 84.707,699.717,218.564,738.982 'Importance of Social Distancing:\\nModeling\\n2019-\\nof\\nSusceptible-Infected-\\nnCoV using\\nQuarantined-Recovered-t model\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 148.734,720.077,171.507,728.802 'spread\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 128.351,720.077,139.014,728.802 'the\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 83.546,618.280,218.563,688.144 '“Studies on the nature of the virus\\nhave suggested different incubation\\nperiods of the virus, and reports have\\nsuggested a median incubation pe-\\nriod of 5-6 days and a very high symp-\\ntom probability period of 14 days [5]\\n”\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 233.401,577.561,368.418,688.144 '“...The incubation period is typically\\n3 to 14 days with the symptoms of\\nan acute nonspeciﬁc, ﬂulike illness\\ndeveloping suddenly...The incuba-\\ntion period is 7 to 10 days before\\nthe onset of symptoms [37]. This\\nincubation period provides the po-\\ntential for worldwide exposure be-\\ncause a person harboring the virus\\ncan expose the world at large via air\\ntravel...”\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 384.417,709.897,518.274,738.982 'Prediction of the virus incubation pe-\\nriod for COVID-19 and future out-\\nbreaks\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 383.257,618.280,518.274,688.083 '“. . . while minimizing the negative\\nconsequences of the quarantine. 70 71\\nThe length of the incubation period\\nvaries both across and within virus\\nfamilies 4 . To our knowledge, 72\\ngenomic features (if any) that corre-\\nlate with the incubation. . . ”\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 73.917,553.886,523.629,563.849 'Figure 4: Top-1 result of article title and displayed passage by querying ’What is the incubation period of virus?’\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 72.000,504.668,133.115,516.624 '4 Analysis\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 72.000,294.788,292.083,495.386 'In this section we look into a number of sample\\nqueries to shed light on how different components\\nof ACS help in improving search results. We begin\\nby observing how small semantic differences in the\\nquery alter the results. The ﬁrst sample in Figure 5\\nis speciﬁc to medications. While the top result does\\nnot include the term medication the system high-\\nlights ribavirin and corticosteroids. The CORD-19\\nsystem understands that these terms represent med-\\nications with the help of CM NERe engine. In the\\nsecond example we change medications to mea-\\nsures and observe the top result discussing border\\ncontrol, and quarantine. This clearly demonstrates\\nthat Amazon Kendra has a deep comprehension of\\ntoken and query meanings.\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 71.487,145.747,292.173,292.148 'Finally, we take a look at the effects of topic mod-\\neling when grouping and ﬁltering results. The last\\ntwo examples in Figure 5 showcase the difference\\nthis makes in the top result. Without specifying any\\ntopic the resulting article discusses high level pol-\\nicy, speciﬁcally quarantine measures in Singapore.\\nWhen we ﬁlter by clinical treatment the top result\\ninstead focuses on infections which is covered in\\nthe clinical setting. Furthermore the extracted text\\nreturned to the user still focuses on lessons learned\\nstaying true to the query.\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 72.000,122.417,264.515,134.373 '5 Limitations and Future Directions\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 71.607,76.568,292.083,114.576 'AWS CORD-19 is an initial step towards help-\\ning medical researchers ﬁnd relevant content in\\na timely and meaningful way. In order to improve\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 307.276,491.262,525.548,515.720 'the robustness, we see following areas as direction\\nfor future research.\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 306.883,175.521,527.359,484.589 'Feedback Loop - Since ACS is a search engine\\nthe motivation would be to evaluate it as such; us-\\ning well-established methodologies based on test\\ncollections—comprising topics (information needs)\\nand human annotations. Since no designated evalu-\\nation data exist, our initial focus is to capture dif-\\nferent interactions and feedback. Currently, ACS\\nlacks the feedback loop and federated learning ap-\\nproaches where the system would continuously\\nlearn and improve the search. However, the system\\ncaptures feedback from the researchers in the form\\nof implicit and explicit reactions. Implicit feed-\\nback evaluation consists of topics of interests, their\\nclicks as well as the ranking of the results which\\nwere selected by medical researchers. Explicit feed-\\nback evaluation is captured by providing up-down\\nrating associated with each search results. In the\\nfuture results can be personalized based on this\\nfeedback. Now that we have a system in place, our\\nefforts have shifted to broader engagement with po-\\ntential stakeholders to solicit additional guidance,\\nwhile trying to balance between the features and\\nranking.\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 306.883,76.568,525.727,168.848 'Q&A Curation - Curation and normalization of\\nquestions have potential a use-case of presenting\\ntrending questions asked by the medical research\\ncommunity at a particular point. However, curation\\nwould involve capturing the questions asked as well\\nas identifying similar questions that can be later\\nnormalized. Currently, there is no mechanism to\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 83.088,766.219,102.115,773.833 'Query\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 213.583,766.219,234.719,773.833 'Article\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 344.618,766.219,373.802,773.833 'Response\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 82.075,741.477,199.890,758.149 '“What medications were most beneﬁ-\\ncial in the SARS outbreak?”\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 213.853,732.593,330.658,757.974 'Development of chemical inhibitors\\nof the SARS coronavirus: Viral heli-\\ncase as a potential target\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 82.075,679.296,198.627,695.968 '“What measures were most beneﬁcial\\nin the SARS outbreak?”\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 213.853,670.413,330.658,695.793 'Impact of quarantine on the 2003\\nSARS outbreak: A retrospective mod-\\neling study\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 82.075,634.882,198.629,651.378 '“What did we learn from the SARS\\noutbreak?” (no topic)\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 213.853,634.882,329.394,651.378 'Use of quarantine in the control of\\nSARS in Singapore\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 82.075,590.467,198.629,606.964 '“What did we learn from the SARS\\noutbreak?” (clinical-treatment)\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 213.853,590.467,330.658,606.964 'Characteristics of COVID-19 infec-\\ntion in Beijing\\n'>,\n",
              "  <LTTextBoxHorizontal(11) 343.606,705.944,519.196,757.974 '“...spread of SARS, a number of broad-spectrum antiviral\\nmedications were empirically administered to the SARS\\npatients during the SARS outbreak in 2003. These medi-\\ncations include ribavirin, HIV protease inhibitors, cor-\\nticosteroids, and alpha-interferon (IFN-a). In a retro-\\nspective review of treatment...”\\n'>,\n",
              "  <LTTextBoxHorizontal(12) 343.606,661.530,517.931,695.793 '“During the 2003 Severe Acute Respiratory Syndrome\\n(SARS) outbreak, traditional intervention measures\\nsuch as quarantine and border control were found to\\nbe useful in containing the outbreak...”\\n'>,\n",
              "  <LTTextBoxHorizontal(13) 343.606,617.116,517.926,651.378 '“The main lesson to learn from the SARS outbreak is the\\ncapability of an emerging infection to cause a pandemic\\nin a short span of time and the paradigm shift needed to\\nrespond to such a disease.”\\n'>,\n",
              "  <LTTextBoxHorizontal(14) 343.606,581.584,519.190,606.964 '“We compared the epidemic features between COVID-19\\nand 2003 SARS for learn lessons and control the out-\\nbreak.”\\n'>,\n",
              "  <LTTextBoxHorizontal(15) 72.000,546.613,525.547,568.532 'Figure 5: Sample queries demonstrating the semantic understanding of ACS, the use-fullness of Comprehend\\nMedical NERe, and the utility of topic modeling for ﬁltering.\\n'>,\n",
              "  <LTTextBoxHorizontal(16) 72.000,511.939,270.317,522.848 'curate the questions asked by the researchers.\\n'>,\n",
              "  <LTTextBoxHorizontal(17) 307.276,511.796,400.802,523.752 'Acknowledgments\\n'>,\n",
              "  <LTTextBoxHorizontal(18) 72.000,424.903,292.075,503.634 'Summarization - Currently, ACS outputs the\\nrelevant passage based on the query. It would be\\nbeneﬁcial to get the overall summary of the pa-\\nper. A potential future direction would be to gen-\\nerates summaries (Raffel et al., 2019) from paper\\nabstracts and full body.\\n'>,\n",
              "  <LTTextBoxHorizontal(19) 306.763,409.683,526.910,501.888 'We acknowledge the broader collaboration with\\nAI2 team and White house as well as the broader\\nAWS CORD-19 team including Kendra Science\\nteam, Tyler Stepke, Kingston Bosco, Victor Wang,\\nVaibhav Chaddha, Miguel Calvo, Ninad Kulkani,\\nKevin Longofer, Ray Chang, Adrian Bordone,\\nTony Nguyen, and Kyle Johnson.\\n'>,\n",
              "  <LTTextBoxHorizontal(20) 72.000,381.109,147.067,393.065 '6 Conclusion\\n'>,\n",
              "  <LTTextBoxHorizontal(21) 71.607,76.568,292.084,358.461 'This paper describes our efforts in building AWS\\nCORD-19 Search with its capabilities consisting\\nof topic based, knowledge graph, and natural lan-\\nguage search queries. This is further enhanced\\nwith reading comprehension and FAQ matching\\nas well as document ranking providing a scalable\\nsolution to COVID-19 researchers and policy mak-\\ners in their search and discovery for answers to\\nhigh priority scientiﬁc questions. Our solution is\\npowered by Amazon Kendra, Comprehend Medi-\\ncal and Neptune which incorporate the latest neural\\narchitectures to provide information access capa-\\nbilities to the CORD-19 challenge. By comparing\\nwith other public search engines that support sim-\\nilar functionalities over COVID-19 search, ACS\\nis demonstrated to be powerful on its document\\nranking and question answering components. We\\nhope that our solution can prove useful in the ﬁght\\nagainst this global pandemic, and that the capabili-\\nties we have developed can be applied to analyzing\\nthe scientiﬁc literature more broadly.\\n'>,\n",
              "  <LTTextBoxHorizontal(22) 307.276,371.126,362.820,383.082 'References\\n'>,\n",
              "  <LTTextBoxHorizontal(23) 307.276,297.584,527.200,362.342 'David Andrzejewski and Xiaojin Zhu. 2009. Latent\\ndirichlet allocation with topic-in-set knowledge. In\\nProceedings of the NAACL HLT 2009 Workshop\\non Semi-Supervised Learning for Natural Language\\nProcessing, pages 43–48. Association for Computa-\\ntional Linguistics.\\n'>,\n",
              "  <LTTextBoxHorizontal(24) 307.276,253.340,527.200,285.221 'Iz Beltagy, Kyle Lo, and Arman Cohan. 2019. Scib-\\nert: Pretrained language model for scientiﬁc text. In\\nEMNLP.\\n'>,\n",
              "  <LTTextBoxHorizontal(25) 307.276,176.219,527.290,240.977 'P. Bhatia, B. Celikkaya, M. Khalilia, and S. Senthivel.\\n2019. Comprehend medical: A named entity recog-\\nIn\\nnition and relationship extraction web service.\\n2019 18th IEEE International Conference On Ma-\\nchine Learning And Applications (ICMLA), pages\\n1844–1851.\\n'>,\n",
              "  <LTTextBoxHorizontal(26) 307.276,132.204,527.200,163.856 'Parminder Bhatia, Busra Celikkaya, and Mohammed\\nJoint entity extraction and asser-\\narXiv preprint\\n'>,\n",
              "  <LTTextBoxHorizontal(27) 318.185,121.016,428.371,152.897 'Khalilia. 2018.\\ntion detection for clinical\\narXiv:1812.05270.\\n'>,\n",
              "  <LTTextBoxHorizontal(28) 434.110,131.975,451.395,141.938 'text.\\n'>,\n",
              "  <LTTextBoxHorizontal(29) 307.276,76.772,527.290,108.653 'David M Blei, Andrew Y Ng, and Michael I Jordan.\\n2003. Latent dirichlet allocation. Journal of ma-\\nchine Learning research, 3(Jan):993–1022.\\n'>],\n",
              " [<LTTextBoxHorizontal(0) 72.000,722.135,292.014,775.934 'Jeffrey Dalton, Laura Dietz, and James Allan. 2014.\\nEntity query feature expansion using knowledge\\nbase links. In Proceedings of the 37th international\\nACM SIGIR conference on Research & development\\nin information retrieval, pages 365–374.\\n'>,\n",
              "  <LTTextBoxHorizontal(1) 307.276,722.135,527.200,775.934 'Da Zheng, Xiang Song, Chao Ma, Zeyuan Tan, Zi-\\nhao Ye, Jin Dong, Hao Xiong, Zheng Zhang, and\\nGeorge Karypis. 2020. Dgl-ke: Training knowl-\\narXiv preprint\\nedge graph embeddings at scale.\\narXiv:2004.08532.\\n'>,\n",
              "  <LTTextBoxHorizontal(2) 72.000,567.628,292.014,709.097 'Wei-jie Guan, Zheng-yi Ni, Yu Hu, Wen-hua Liang,\\nChun-quan Ou, Jian-xing He, Lei Liu, Hong Shan,\\nChun-liang Lei, David S.C. Hui, Bin Du, Lan-juan\\nLi, Guang Zeng, Kwok-Yung Yuen, Ru-chong Chen,\\nChun-li Tang, Tao Wang, Ping-yan Chen, Jie Xiang,\\nShi-yue Li, Jin-lin Wang, Zi-jing Liang, Yi-xiang\\nPeng, Li Wei, Yong Liu, Ya-hua Hu, Peng Peng,\\nJian-ming Wang, Ji-yang Liu, Zhong Chen, Gang\\nLi, Zhi-jian Zheng, Shao-qin Qiu, Jie Luo, Chang-\\njiang Ye, Shao-yong Zhu, and Nan-shan Zhong.\\n2020. Clinical characteristics of coronavirus disease\\n2019 in china. New England Journal of Medicine,\\n382(18):1708–1720.\\n'>,\n",
              "  <LTTextBoxHorizontal(3) 72.000,500.791,291.924,554.590 'Colin Raffel, Noam Shazeer, Adam Roberts, Katherine\\nLee, Sharan Narang, Michael Matena, Yanqi Zhou,\\nWei Li, and Peter J Liu. 2019. Exploring the limits\\nof transfer learning with a uniﬁed text-to-text trans-\\nformer. arXiv preprint arXiv:1910.10683.\\n'>,\n",
              "  <LTTextBoxHorizontal(4) 72.000,455.873,290.271,487.753 'Jesse Read, Bernhard Pfahringer, Geoff Holmes, and\\nEibe Frank. 2011. Classiﬁer chains for multi-label\\nclassiﬁcation. Machine learning, 85(3):333.\\n'>,\n",
              "  <LTTextBoxHorizontal(5) 72.000,399.995,291.516,442.835 'Maya Rotmensch, Yoni Halpern, Abdulhakim Tlimat,\\nSteven Horng, and David Sontag. 2017. Learning\\na health knowledge graph from electronic medical\\nrecords. Scientiﬁc reports, 7(1):1–11.\\n'>,\n",
              "  <LTTextBoxHorizontal(6) 72.000,355.077,290.271,386.957 'Gaurav Singh and Parminder Bhatia. 2019. Relation\\nextraction using explicit context conditioning. arXiv\\npreprint arXiv:1902.09271.\\n'>,\n",
              "  <LTTextBoxHorizontal(7) 72.000,288.240,291.924,342.039 'Raphael Tang, Rodrigo Nogueira, Edwin Zhang, Nikhil\\nGupta, Phuong Cam, Kyunghyun Cho, and Jimmy\\nLin. 2020. Rapidly Bootstrapping a Question An-\\narXiv preprint\\nswering Dataset for COVID-19.\\narXiv:2004.11339.\\n'>,\n",
              "  <LTTextBoxHorizontal(8) 72.000,210.445,292.014,275.203 'Ellen Voorhees, Tasmeer Alam, Steven Bedrick, Dina\\nDemner-Fushman, William R Hersh, Kyle Lo, Kirk\\nRoberts, Ian Soborof, and Lucy Lu Wang. 2020.\\nTREC-COVID: Constructing a Pandemic Informa-\\narXiv preprint\\ntion Retrieval Test Collection.\\narXiv:2005.04474.\\n'>,\n",
              "  <LTTextBoxHorizontal(9) 72.000,143.609,291.516,197.407 'Lucy Lu Wang, Kyle Lo, Yoganand Chandrasekhar,\\nRussell Reas, Jiangjiang Yang, Darrin Eide, Kathryn\\nFunk, Rodney Kinney, Ziyang Liu, William Merrill,\\net al. 2020. Cord-19: The covid-19 open research\\ndataset. arXiv preprint arXiv:2004.10706.\\n'>,\n",
              "  <LTTextBoxHorizontal(10) 72.000,76.772,292.014,130.571 'Quan Wang, Zhendong Mao, Bin Wang, and Li Guo.\\n2017. Knowledge graph embedding: A survey of\\nIEEE Transactions\\napproaches and applications.\\non Knowledge and Data Engineering, 29(12):2724–\\n2743.\\n'>]]"
            ]
          },
          "metadata": {},
          "execution_count": 55
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "KDFHoYnUBCTh"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-31_2007.09186v3_LTtext.pickle', 'wb') as f:\n",
        "  pickle.dump(text, f)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(text)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ov-_FBbpTOjx",
        "outputId": "4a3ce76a-86ae-4bc2-b408-e8c4d0d5ff31"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "9"
            ]
          },
          "metadata": {},
          "execution_count": 81
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Return only the `LTTextBox` and `LTTextLine` objects for page 1."
      ],
      "metadata": {
        "id": "V_Qw7XL46gsd"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "text[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i-kbCubqwlEs",
        "outputId": "5f4d8a38-c801-4c8d-ad4c-6f8a94581f62"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTTextBoxHorizontal(0) 103.371,746.071,493.659,776.357 'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\n'>,\n",
              " <LTTextBoxHorizontal(1) 77.083,639.527,523.454,734.427 'Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\\nBen Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\\nRamesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\\nAmazon Web Services AI\\nparmib, liuall, arumae, nimpourd, suyogd, snivelyb,\\nmonamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\\nzhiheng, bxiang, tahak @amazon.com\\n'>,\n",
              " <LTTextBoxHorizontal(2) 158.891,605.891,203.376,617.847 'Abstract\\n'>,\n",
              " <LTTextBoxHorizontal(3) 88.261,322.925,274.910,595.902 'Coronavirus disease (COVID-19) has been de-\\nclared as a pandemic by WHO with thousands\\nof cases being reported each day. Numerous\\nscientiﬁc articles are being published on the\\ndisease raising the need for a service which\\ncan organize, and query them in a reliable fash-\\nion. To support this cause we present AWS\\nCORD-19 Search (ACS), a public, COVID-19\\nspeciﬁc, neural search engine that is powered\\nby several machine learning systems to sup-\\nport natural language based searches. ACS\\nwith capabilities such as document ranking,\\npassage ranking, question answering and topic\\nclassiﬁcation provides a scalable solution to\\nCOVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high\\npriority scientiﬁc questions. We present a\\nquantitative evaluation and qualitative analysis\\nof the system against other leading COVID-\\n19 search platforms. ACS is top performing\\nacross these systems yielding quality results\\nwhich we detail with relevant examples in this\\nwork.\\n'>,\n",
              " <LTTextBoxHorizontal(4) 72.000,300.286,77.978,312.242 '1\\n'>,\n",
              " <LTTextBoxHorizontal(5) 89.933,300.286,154.814,312.242 'Introduction\\n'>,\n",
              " <LTTextBoxHorizontal(6) 71.182,76.568,292.082,290.715 'With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n'>,\n",
              " <LTTextBoxHorizontal(7) 307.276,389.247,527.359,616.943 'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n'>,\n",
              " <LTTextBoxHorizontal(8) 306.883,101.737,527.356,385.118 'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n'>,\n",
              " <LTTextBoxHorizontal(9) 319.928,77.606,425.632,87.419 '1https://cord19.aws/\\n'>,\n",
              " <LTTextBoxHorizontal(10) 16.340,526.960,36.340,566.960 '0\\n2\\n0\\n2\\n'>,\n",
              " <LTTextBoxHorizontal(11) 16.340,478.080,36.340,521.960 't\\nc\\nO\\n7\\n'>,\n",
              " <LTTextBoxHorizontal(12) 16.340,461.420,36.340,468.080 ']\\n'>,\n",
              " <LTTextBoxHorizontal(13) 16.340,448.080,36.340,461.420 'R\\n'>,\n",
              " <LTTextBoxHorizontal(14) 16.340,413.100,36.340,448.080 'I\\n.\\ns\\nc\\n[\\n'>,\n",
              " <LTTextBoxHorizontal(15) 16.340,237.000,36.340,403.100 '3\\nv\\n6\\n8\\n1\\n9\\n0\\n.\\n7\\n0\\n0\\n2\\n:\\nv\\ni\\nX\\nr\\na\\n'>,\n",
              " <LTTextLineHorizontal 16.340,403.100,36.340,408.100 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,408.100,36.340,413.100 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,468.080,36.340,473.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,473.080,36.340,478.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,488.080,36.340,493.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,521.960,36.340,526.960 ' \\n'>]"
            ]
          },
          "metadata": {},
          "execution_count": 82
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Return *all* objects on page 1 and notice that an `LTLine` object now appears above the `LTTextLine` objects at the end."
      ],
      "metadata": {
        "id": "So-7IN34L9NW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "LT_elem_text[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9c3223d7-4739-46b7-d325-84291a56cfdd",
        "id": "r2LbN5SIBCRS"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTTextBoxHorizontal(0) 103.371,746.071,493.659,776.357 'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\n'>,\n",
              " <LTTextBoxHorizontal(1) 77.083,639.527,523.454,734.427 'Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\\nBen Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\\nRamesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\\nAmazon Web Services AI\\nparmib, liuall, arumae, nimpourd, suyogd, snivelyb,\\nmonamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\\nzhiheng, bxiang, tahak @amazon.com\\n'>,\n",
              " <LTTextBoxHorizontal(2) 158.891,605.891,203.376,617.847 'Abstract\\n'>,\n",
              " <LTTextBoxHorizontal(3) 88.261,322.925,274.910,595.902 'Coronavirus disease (COVID-19) has been de-\\nclared as a pandemic by WHO with thousands\\nof cases being reported each day. Numerous\\nscientiﬁc articles are being published on the\\ndisease raising the need for a service which\\ncan organize, and query them in a reliable fash-\\nion. To support this cause we present AWS\\nCORD-19 Search (ACS), a public, COVID-19\\nspeciﬁc, neural search engine that is powered\\nby several machine learning systems to sup-\\nport natural language based searches. ACS\\nwith capabilities such as document ranking,\\npassage ranking, question answering and topic\\nclassiﬁcation provides a scalable solution to\\nCOVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high\\npriority scientiﬁc questions. We present a\\nquantitative evaluation and qualitative analysis\\nof the system against other leading COVID-\\n19 search platforms. ACS is top performing\\nacross these systems yielding quality results\\nwhich we detail with relevant examples in this\\nwork.\\n'>,\n",
              " <LTTextBoxHorizontal(4) 72.000,300.286,77.978,312.242 '1\\n'>,\n",
              " <LTTextBoxHorizontal(5) 89.933,300.286,154.814,312.242 'Introduction\\n'>,\n",
              " <LTTextBoxHorizontal(6) 71.182,76.568,292.082,290.715 'With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n'>,\n",
              " <LTTextBoxHorizontal(7) 307.276,389.247,527.359,616.943 'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n'>,\n",
              " <LTTextBoxHorizontal(8) 306.883,101.737,527.356,385.118 'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n'>,\n",
              " <LTTextBoxHorizontal(9) 319.928,77.606,425.632,87.419 '1https://cord19.aws/\\n'>,\n",
              " <LTTextBoxHorizontal(10) 16.340,526.960,36.340,566.960 '0\\n2\\n0\\n2\\n'>,\n",
              " <LTTextBoxHorizontal(11) 16.340,478.080,36.340,521.960 't\\nc\\nO\\n7\\n'>,\n",
              " <LTTextBoxHorizontal(12) 16.340,461.420,36.340,468.080 ']\\n'>,\n",
              " <LTTextBoxHorizontal(13) 16.340,448.080,36.340,461.420 'R\\n'>,\n",
              " <LTTextBoxHorizontal(14) 16.340,413.100,36.340,448.080 'I\\n.\\ns\\nc\\n[\\n'>,\n",
              " <LTTextBoxHorizontal(15) 16.340,237.000,36.340,403.100 '3\\nv\\n6\\n8\\n1\\n9\\n0\\n.\\n7\\n0\\n0\\n2\\n:\\nv\\ni\\nX\\nr\\na\\n'>,\n",
              " <LTLine 307.276,89.599,367.052,89.599>,\n",
              " <LTTextLineHorizontal 16.340,403.100,36.340,408.100 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,408.100,36.340,413.100 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,468.080,36.340,473.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,473.080,36.340,478.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,488.080,36.340,493.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,521.960,36.340,526.960 ' \\n'>]"
            ]
          },
          "metadata": {},
          "execution_count": 83
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "Separate line objects."
      ],
      "metadata": {
        "id": "VZvxB5blMtyQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "lines = lines_LT_elems(LT_elem_text)"
      ],
      "metadata": {
        "id": "7LaN5kUFVcoY"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(lines)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BG0m5tBjVj4A",
        "outputId": "3c6289a7-61a1-436c-8f9d-1fea9d3fb430"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "9"
            ]
          },
          "metadata": {},
          "execution_count": 87
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Return only the lines contained in the `LTLine` objects for page 1."
      ],
      "metadata": {
        "id": "3wkU8QcPLaUc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "lines[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EmfJ7oGvVluM",
        "outputId": "04a4f272-07d4-4b3d-c76c-3c8530b928d4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTLine 307.276,89.599,367.052,89.599>]"
            ]
          },
          "metadata": {},
          "execution_count": 88
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Return only the lines contained in the `LTLine` objects for whole PDF."
      ],
      "metadata": {
        "id": "Bxa5tgGvLrGl"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "lines"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_17QRIgfVoXe",
        "outputId": "74e3a5be-bcfd-4bc2-8dd3-16ec3ebc5ae2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[<LTLine 307.276,89.599,367.052,89.599>],\n",
              " [<LTLine 75.362,778.544,525.543,778.544>,\n",
              "  <LTLine 75.362,762.998,525.543,762.998>,\n",
              "  <LTLine 75.362,635.726,525.543,635.726>,\n",
              "  <LTLine 307.276,89.599,367.052,89.599>],\n",
              " [<LTLine 72.000,130.362,131.776,130.362>],\n",
              " [<LTLine 72.000,99.562,131.776,99.562>,\n",
              "  <LTLine 307.276,99.562,367.052,99.562>],\n",
              " [<LTLine 107.268,778.645,490.276,778.645>,\n",
              "  <LTLine 107.268,761.170,490.276,761.170>,\n",
              "  <LTLine 107.268,741.980,490.276,741.980>,\n",
              "  <LTLine 107.268,697.571,490.276,697.571>,\n",
              "  <LTLine 107.268,678.381,490.276,678.381>,\n",
              "  <LTLine 107.268,632.092,490.276,632.092>,\n",
              "  <LTLine 72.000,110.353,131.776,110.353>],\n",
              " [<LTLine 167.054,778.645,430.489,778.645>,\n",
              "  <LTLine 244.213,761.607,244.213,775.156>,\n",
              "  <LTLine 306.305,761.607,306.305,775.156>,\n",
              "  <LTLine 368.397,761.607,368.397,775.156>,\n",
              "  <LTLine 244.213,748.057,244.213,761.606>,\n",
              "  <LTLine 306.305,748.057,306.305,761.606>,\n",
              "  <LTLine 368.397,748.057,368.397,761.606>,\n",
              "  <LTLine 167.054,745.906,430.489,745.906>,\n",
              "  <LTLine 244.213,729.031,244.213,742.580>,\n",
              "  <LTLine 306.305,729.031,306.305,742.580>,\n",
              "  <LTLine 368.397,729.031,368.397,742.580>,\n",
              "  <LTLine 244.213,715.482,244.213,729.031>,\n",
              "  <LTLine 306.305,715.482,306.305,729.031>,\n",
              "  <LTLine 368.397,715.482,368.397,729.031>,\n",
              "  <LTLine 244.213,701.933,244.213,715.482>,\n",
              "  <LTLine 306.305,701.933,306.305,715.482>,\n",
              "  <LTLine 368.397,701.933,368.397,715.482>,\n",
              "  <LTLine 167.054,699.618,430.489,699.618>,\n",
              "  <LTLine 134.327,666.864,463.216,666.864>,\n",
              "  <LTLine 211.485,649.826,211.485,663.375>,\n",
              "  <LTLine 337.351,649.826,337.351,663.375>,\n",
              "  <LTLine 211.485,636.277,211.485,649.826>,\n",
              "  <LTLine 337.351,636.277,337.351,649.826>,\n",
              "  <LTLine 134.327,634.125,463.216,634.125>,\n",
              "  <LTLine 211.485,617.250,211.485,630.799>,\n",
              "  <LTLine 337.351,617.250,337.351,630.799>,\n",
              "  <LTLine 211.485,603.701,211.485,617.250>,\n",
              "  <LTLine 337.351,603.701,337.351,617.250>,\n",
              "  <LTLine 211.485,590.152,211.485,603.701>,\n",
              "  <LTLine 337.351,590.152,337.351,603.701>,\n",
              "  <LTLine 134.327,587.837,463.216,587.837>,\n",
              "  <LTLine 110.264,264.604,124.949,264.604>,\n",
              "  <LTLine 147.349,264.604,153.372,264.604>],\n",
              " [<LTLine 75.981,778.444,525.546,778.444>,\n",
              "  <LTLine 75.981,762.535,525.546,762.535>,\n",
              "  <LTLine 75.981,744.121,525.546,744.121>,\n",
              "  <LTLine 75.981,573.013,525.546,573.013>],\n",
              " [<LTLine 75.473,778.526,525.542,778.526>,\n",
              "  <LTLine 75.473,762.458,525.542,762.458>,\n",
              "  <LTLine 75.473,577.615,525.542,577.615>],\n",
              " []]"
            ]
          },
          "metadata": {},
          "execution_count": 89
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Separate figure objects."
      ],
      "metadata": {
        "id": "e9Jl0XSUMz_l"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "figures = figures_LT_elems(LT_elem_text)"
      ],
      "metadata": {
        "id": "eo1Rhq6_Vf8p"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(figures)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KLix2E0gVraD",
        "outputId": "f22fb5e6-9fc4-4e2e-b77c-08c7ee8b410e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "9"
            ]
          },
          "metadata": {},
          "execution_count": 91
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Only two `LTFigure` objects appear in the document - on page 3 (table for Sample natural language queries and responses using AWS CORD-19 Search) and page 4 (System Architecture diagram)."
      ],
      "metadata": {
        "id": "P1REyRp5M6eX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "figures"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oW-6sxadVz38",
        "outputId": "55a9ab4b-d2ca-454d-debc-4e8bf90f6120"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[],\n",
              " [],\n",
              " [<LTFigure(Im1) 72.000,559.122,526.860,779.082 matrix=[454.86,0.00,0.00,219.96, (72.00,559.12)]>],\n",
              " [<LTFigure(Im2) 97.924,466.435,499.624,779.085 matrix=[401.70,0.00,0.00,312.65, (97.92,466.44)]>],\n",
              " [],\n",
              " [],\n",
              " [],\n",
              " [],\n",
              " []]"
            ]
          },
          "metadata": {},
          "execution_count": 92
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "M6M6LVIxn2VP"
      },
      "source": [
        "### 4.10 Extract individual characters\n",
        "\n",
        "We can access the individual layout objects, such as the `LTTextBox` objects and the text they contain, but we do not know what the structure looks like. We can do this by breaking the document down into `LTChar` objects, individual characters, contained in a list inside `LTTextLine` objects.\n",
        "\n",
        "\n",
        "The [pdfminer documentation](https://pdfminersix.readthedocs.io/en/latest/topic/converting_pdf_to_text.html#layout-analysis-algorithm) describes how the  layout analysis consists of three different stages: it groups characters into words and lines, then it groups lines into boxes and finally it groups textboxes hierarchically.\n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "![Layout analysis output graphic](images/layout_analysis_output.png)"
      ],
      "metadata": {
        "id": "wjHJlpwQTkmd"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "We will extract individual characters from the document and return as a list of `LTChar` objects."
      ],
      "metadata": {
        "id": "4acZ8j0NJf2N"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "R_AcyqVIK3bc"
      },
      "outputs": [],
      "source": [
        "TEXT_ELEMENTS = [\n",
        "    pdfminer.layout.LTTextBox,\n",
        "    pdfminer.layout.LTTextBoxHorizontal,\n",
        "    pdfminer.layout.LTTextLine,\n",
        "    pdfminer.layout.LTTextLineHorizontal\n",
        "]\n",
        "\n",
        "def flatten(lst):\n",
        "    \"\"\"Flattens a list of lists\"\"\"\n",
        "    return [item for sublist in lst for item in sublist]\n",
        "\n",
        "def extract_characters(element):\n",
        "    \"\"\"\n",
        "    Recursively extracts individual characters from\n",
        "    text elements.\n",
        "    \"\"\"\n",
        "    if isinstance(element, pdfminer.layout.LTChar):\n",
        "        return [element]\n",
        "\n",
        "    if any(isinstance(element, i) for i in TEXT_ELEMENTS):\n",
        "        return flatten([extract_characters(e) for e in element])\n",
        "\n",
        "    if isinstance(element, list):\n",
        "        return flatten([extract_characters(l) for l in element])\n",
        "\n",
        "    return []"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Extract first five characters from 'AWS CORD-19' on page 1."
      ],
      "metadata": {
        "id": "jtEFkO_9Pp9_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# extract characters from LT_elem_text - page 1\n",
        "characters_1 = extract_characters(LT_elem_text[0])\n",
        "characters_1[:5]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-12blCcuxuG1",
        "outputId": "81c1d23d-0753-4a88-fc0b-124c8756cd8e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTChar 103.371,762.011,113.729,776.357 matrix=[1.00,0.00,0.00,1.00, (103.37,765.01)] font='UJKCWC+NimbusRomNo9L-Medi' adv=10.357956399999999 text='A'>,\n",
              " <LTChar 111.864,762.011,126.210,776.357 matrix=[1.00,0.00,0.00,1.00, (111.86,765.01)] font='UJKCWC+NimbusRomNo9L-Medi' adv=14.3462 text='W'>,\n",
              " <LTChar 126.210,762.011,134.187,776.357 matrix=[1.00,0.00,0.00,1.00, (126.21,765.01)] font='UJKCWC+NimbusRomNo9L-Medi' adv=7.9764872 text='S'>,\n",
              " <LTChar 137.773,762.011,148.131,776.357 matrix=[1.00,0.00,0.00,1.00, (137.77,765.01)] font='UJKCWC+NimbusRomNo9L-Medi' adv=10.357956399999999 text='C'>,\n",
              " <LTChar 148.131,762.011,159.292,776.357 matrix=[1.00,0.00,0.00,1.00, (148.13,765.01)] font='UJKCWC+NimbusRomNo9L-Medi' adv=11.1613436 text='O'>]"
            ]
          },
          "metadata": {},
          "execution_count": 96
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Extract first five characters from 'Query' on page 2."
      ],
      "metadata": {
        "id": "T4vl0MazP7Bj"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "O9ywRMxqK8Os",
        "outputId": "3eba8579-081f-435c-cd1c-d4a8ee3918ca"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTChar 82.729,766.636,88.047,774.003 matrix=[1.23,0.00,0.00,1.23, (82.73,768.23)] font='YHBHHN+NimbusRomNo9L-Regu' adv=4.315827199999999 text='Q'>,\n",
              " <LTChar 88.047,766.636,91.731,774.003 matrix=[1.23,0.00,0.00,1.23, (88.05,768.23)] font='YHBHHN+NimbusRomNo9L-Regu' adv=2.9888 text='u'>,\n",
              " <LTChar 91.731,766.636,95.001,774.003 matrix=[1.23,0.00,0.00,1.23, (91.73,768.23)] font='YHBHHN+NimbusRomNo9L-Regu' adv=2.6540544 text='e'>,\n",
              " <LTChar 95.001,766.636,97.454,774.003 matrix=[1.23,0.00,0.00,1.23, (95.00,768.23)] font='YHBHHN+NimbusRomNo9L-Regu' adv=1.9905408 text='r'>,\n",
              " <LTChar 97.454,766.636,101.138,774.003 matrix=[1.23,0.00,0.00,1.23, (97.45,768.23)] font='YHBHHN+NimbusRomNo9L-Regu' adv=2.9888 text='y'>]"
            ]
          },
          "metadata": {},
          "execution_count": 97
        }
      ],
      "source": [
        "# extract characters from LT_elem_text - page 2\n",
        "characters_2 = extract_characters(LT_elem_text[1])\n",
        "characters_2[:5]"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Extract first five characters from 'Figure' on page 3."
      ],
      "metadata": {
        "id": "qjoJTMmVP9Kt"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# extract characters from LT_elem_text - page 3\n",
        "characters_3 = extract_characters(LT_elem_text[2])\n",
        "characters_3[:5]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xJymondmjvHK",
        "outputId": "19432d77-b34a-4745-f0ae-057372c841f8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTChar 237.044,540.631,242.583,550.594 matrix=[1.00,0.00,0.00,1.00, (237.04,542.78)] font='YHBHHN+NimbusRomNo9L-Regu' adv=5.539205600000001 text='F'>,\n",
              " <LTChar 242.583,540.631,245.353,550.594 matrix=[1.00,0.00,0.00,1.00, (242.58,542.78)] font='YHBHHN+NimbusRomNo9L-Regu' adv=2.7696028000000004 text='i'>,\n",
              " <LTChar 245.353,540.631,250.334,550.594 matrix=[1.00,0.00,0.00,1.00, (245.35,542.78)] font='YHBHHN+NimbusRomNo9L-Regu' adv=4.9813 text='g'>,\n",
              " <LTChar 250.334,540.631,255.315,550.594 matrix=[1.00,0.00,0.00,1.00, (250.33,542.78)] font='YHBHHN+NimbusRomNo9L-Regu' adv=4.9813 text='u'>,\n",
              " <LTChar 255.315,540.631,258.633,550.594 matrix=[1.00,0.00,0.00,1.00, (255.32,542.78)] font='YHBHHN+NimbusRomNo9L-Regu' adv=3.3175458000000004 text='r'>]"
            ]
          },
          "metadata": {},
          "execution_count": 98
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "As page_layouts is a nested list of `LTPage` objects in the PDF, we can access them by index position [0]."
      ],
      "metadata": {
        "id": "59-tjo2-yaAL"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "page_layouts[0] # article 1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9BmhlcZ2yNIH",
        "outputId": "3996a5c5-9c82-4915-e424-4d11c92e3813"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTPage(1) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(2) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(3) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(4) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(5) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(6) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(7) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(8) 0.000,0.000,595.276,841.890 rotate=0>,\n",
              " <LTPage(9) 0.000,0.000,595.276,841.890 rotate=0>]"
            ]
          },
          "metadata": {},
          "execution_count": 99
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Assign each page to a variable name."
      ],
      "metadata": {
        "id": "_IcMkhlEzEk5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "page_1 = page_layouts[0][0]\n",
        "page_2 = page_layouts[0][1]\n",
        "page_3 = page_layouts[0][2]\n",
        "page_4 = page_layouts[0][3]\n",
        "page_5 = page_layouts[0][4]\n",
        "page_6 = page_layouts[0][5]\n",
        "page_7 = page_layouts[0][6]\n",
        "page_8 = page_layouts[0][7]\n",
        "page_9 = page_layouts[0][8]"
      ],
      "metadata": {
        "id": "4wIypKgSyt_O"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 4.11 Visualise the PDF structure\n",
        "\n",
        "Using the bounding boxes describe each element in the PDF and visualise them."
      ],
      "metadata": {
        "id": "0gpxhQEs-zaY"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5m_RhVCvMekt"
      },
      "outputs": [],
      "source": [
        "def draw_rect_bbox(bbox, ax, color):\n",
        "    \"\"\"\n",
        "    Draws an unfilled rectangle onto ax.\n",
        "    \"\"\"\n",
        "    x0,y0,x1,y1 = tuple(bbox)\n",
        "    ax.add_patch(\n",
        "        patches.Rectangle(\n",
        "            (x0, y0),\n",
        "            x1 - x0,\n",
        "            y1 - y0,\n",
        "            fill=False,\n",
        "            color=color\n",
        "        )\n",
        "    )\n",
        "\n",
        "def draw_rect(rect, ax, color=\"black\"):\n",
        "    draw_rect_bbox(rect.bbox, ax, color)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Plot page 1 of the PDF with characters inside red rectangles, textboxes inside blue rectangles, and lines in black."
      ],
      "metadata": {
        "id": "r9H4Be5EiVOu"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "xmin, ymin, xmax, ymax = page_1.bbox\n",
        "size = 6\n",
        "num_pages = 2\n",
        "\n",
        "fig, axes = plt.subplots(1,num_pages, figsize = (num_pages*size, size * (ymax/xmax)), sharey=True, sharex=True)\n",
        "\n",
        "# figures, chars and lines\n",
        "ax = axes[0]\n",
        "for figure in figures[0]:\n",
        "    draw_rect(figure, ax, \"green\")\n",
        "\n",
        "for c in characters_1:\n",
        "    draw_rect(c, ax, \"red\")\n",
        "\n",
        "for line in lines[0]:\n",
        "    draw_rect(line, ax, \"black\")\n",
        "\n",
        "# chars and TextBoxes\n",
        "ax = axes[1]\n",
        "for c in characters_1:\n",
        "    draw_rect(c, ax, \"red\")\n",
        "\n",
        "for textbox in text[0]:\n",
        "    draw_rect(textbox, ax, \"blue\")\n",
        "\n",
        "\n",
        "plt.xlim(xmin, xmax)\n",
        "plt.ylim(ymin, ymax)\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 509
        },
        "id": "lHuqBeqOya29",
        "outputId": "c8ab0f58-0f33-43d7-a983-15ae1bd3fa9e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 864x610.971 with 2 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAsUAAAHsCAYAAAA6gNB7AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nOy9b6w131Xf9929N0BCU0yMa7m+1jlUWImiSBD8CKioqhQ3VaBRzAvi0FTUUEvuC2jTUql2+6LkRSuBVNUlimTJilXZVRqgJMhWhaIiA6r6ApTnRxAkUBSH3hPfnwz+hRq3DUrhuZq+OGf5rrvuWnuvPWfOOTN3vh/p6MzM3nvttffM+c6fs/eaMgwDCCGEEEIIWTP/wqUdIIQQQggh5NLwopgQQgghhKweXhQTQgghhJDVw4tiQgghhBCyenhRTAghhBBCVg8vigkhhBBCyOq5vrQDAPA1X/M1w3a7vbQbhBDSzWuvvfZPh2F4y6X9OCfUbELIkol0exYXxdvtFi9fvry0G4QQ0k0pZXdpH84NNZsQsmQi3ebwCUIIIYQQsnp4UUwIIYQQQlYPL4oJIYQQQsjq4UUxIYQQQghZPbwoJoQQQgghq4cXxYQQQgghZPXwopgQQgghhKweXhQTQgghhJDVw4tiQgghhBCyenhRTAghhBBCVg8vigkhhBBCyOrhRTEhhBBCCFk9vCgmhBBCCCGrhxfFhBBCCCFk9fCimBBCCCGErB5eFBNCCCGEkNXDi2JCCCGEELJ6UhfFpZT/tJTyD0sp/6CU8rdKKV9RSvnaUsovllI+U0r58VLKlx3yfvlh/TOH9O0pG0AIIYQQQsixNC+KSylvB/AfA3gxDMOfAnAF4LsB/AiADw/D8HUAvgDg/Yci7wfwhcP2Dx/yEUIIIYQQMluywyeuAfzhUso1gD8C4HMAvg3ATx7SPw7gOw/L7zms45D+7lJKmcZdQgghhBBCpqd5UTwMw+sA/lsA/wT7i+EvAngNwO8Ow/DqkO0OwNsPy28H8NlD2VeH/G+2dkspHyilvCylvHzjjTeObQchhJATQs0mhDx3MsMnvhr7p79fC+BfAfCVAP7csRUPw/DRYRheDMPw4i1vecux5gghhJwQajYh5LmTGT7xbwH4P4dheGMYhj8A8HcAfCuANx2GUwDADYDXD8uvA3gHABzSvwrA70zqNSGEEEIIIROSuSj+JwC+pZTyRw5jg98N4NcA/ByA7zrkeR+ATx6WP3VYxyH9Z4dhGKZzmRBCCCGEkGnJjCn+RewnzP0SgF89lPkogA8C+MFSymewHzP8sUORjwF482H7DwL40An8JoQQQgghZDKu21mAYRh+CMAPmc2/CeCbnLz/HMBfPN41QgghhBBCzgPfaEcIIYQQQlYPL4oJIYQQQsjq4UUxIYQQQghZPbwoJoQQQgghq4cXxYQQQgghZPXwopgQQgghhKweXhQTQgghhJDVw4viU7HdAqXkP9fXj9e329iOzVvb3toWLdd8qJXJ2tbrtbb29EPWF72+3bZtj2nvMb7LcmvfWftj+zg6nmrHWOsYzPhCyIzYXt+lfwb88LO2z/b67tI/0bOQenkHGcFuB+i3W5ey/47eeF2Kn9/a8fLWtre2Rcs1H1plsrZ1Wq2ttbTIZrb+aL+MbcfYfejZyew7bf+YPo76PcrbOgYzvhAyI3b3N08P6eg499Kyv9vW9mM1O/P7m0KzW23NtHmsZp+6jZn9ckzfj8lb80W2Afljp8dvAKXcYA3wSTEhhBBCCFk9vCgmhBBCCCGrhxfFhBBCCCFk9fCimBBCCCGErB5eFBNCCCGEkNXDi2JCCCGEELJ6GJLtVGw2fuipKBzV1dXjtM0mtmPz1ra3tkXLNR9qZbK29XqtrT39kPVFr4v9sX00pv+ydlr7ztof28fR8QTEeVvHYMYXQuZGz2+Dmt2v2WN90eunbOMU55uefjunZkflMn4DAILQhM8MXhSfitvbedk5hnP5MLaeqyvg/j7eZtP1+t2dn3Z//3h5sxnv36n7r8f+HI4nQuZKFJO4hzn8xuaq2aKpLc2OygF7zbYXb9Ts0xNcgz83eFFMlkHtpRNe8PExgdxradFdOSGEkKcco9l2nZpNzgTHFBNCCCGEkNXDi2JCCCGEELJ6eFFMCCGEEEJWDy+KCSGEEELI6uFFMSGEEEIIWT28KCaEEEIIIauHIdnIMqgFMPeCjx8bIN4u84UThDxrondNkLEMTmxb2Wa/vfxR2th85BjWcgrkRTFZBnMJYE4IeZZQYgghHD5BCCGEEEJWDy+KCSGEEELI6uFFMSGEEEIIWT28KCaEEEIIIauHF8Utttv9lGT5XF+3l+22Wj4vz3b7tN5M3VF9221fW1r5vXJZn3v8zvrTsy/G+N+yd4x/U7Qhe2x4fvbayezrbJvG9om0g5CA7fXdk8OTH374ucxne313aUlIw+gTLXY7YBge1kt5WI+Wbd5avigPENuLliNbu11/W2r5o7ozPus8Lb+z/tSWe/ou8j/brjH+TdGGbPu846DXTqZPsm0a2ydSPyEBu/ubh0PtXJptt2Xq7tXUll9R/mM02/Mz+1vP9oe3nvGl5n/GXtRfxy632pBZHnMcTLE/jtVsx1YpN0/7ZabwSTEhhBBCCFk9vCgmhBBCCCGrhxfFhBBCCCFk9fCimBBCCCGErB5eFBNCCCGEkNXD6BMtNpvHs92vrh7Wo2Wbt5bPy7PZ7L8je9FyVN/NTV9bpP4ov1cu67PO0/I7609tOZu35n+rXcf4N0UbMnai46DXjm5v7/E+pq7aviKkxjG/sbGa3asBvZpa86uW/xjN9vycWrO99ZYvPTrk2evp37GafYydMcfBFPvjWM12bTlROWYKnxS3uL3dhxaRg+r+fv99deUvbzYP+WWb5v5+n18utmRd29rtgDsT12+zebio0X54tm3abvf0xxHZAPZ1l/IQwsv6enPztI3WZynj+Sh+vHr12Ib1R2zsdvsYtdofXV6XtX2ZRWx7F1zanu0L7R/w9FjR+bU9u93ut/v7vR1vf3ltjPalbaPdr1k74o+E3REbumzUPpsu695x5JWXMpL/9rbdVkKymi15W5rtlZNtPRplbUeILe/8YxEfTqHZYiej2dqXnv6QftU+1PaX+A/EN8n2XGuRtrc0O/Ijo9mezR7NBur7tWVHa7Z3TvfwNFvK1DTb80/yL4gyzMDhFy9eDC9fvry0G3VKGRd/MioL9MUoFHriIHppU9rI9ElPP1l/tK/H+NCzbPunlp7xaYzfUR+02lhLz+bz0vUxWPM9u6+j78i3Wl/MgFLKa8MwvLi0H+dk7ppdCjAgqQ2yDqTjrrrbsr/1mv1ouff3eynNPrY/5qbZWT90XutPttyUml1rX1RH5liI7DXqj7Jfkki3+aSYEEIIIYSsHl4UE0IIIYSQ1cOLYkIIIYQQsnp4UUwIIYQQQlYPL4oJIYQQQsjqYZziLMfE+PVi+AF+ehRjMYrJWPMl8u0YGxJmJdsnvXETtS3t6zE+ZJcz8SB7ffJiV3r718vXG6+6lp7NF6Vrf3riV3vpNv6mdwx4/aNDEhLSoidGrv2tNeOuOtuyGlWzb5ezv5G5aPax/TFHzc74IcxJszP7PavZLXut+hdE86K4lPLHAfy42vSvAvivAHzisH0L4BbAe4dh+EIppQD4UQDfAeD3AHzvMAy/NK3bF6A3NuqxsVS328exYHe7OIZjJoai9sva9srZb50mMSijumRdBELq0vYkZu7VVVxet1eXlby2HTVseW+5htcP0kbdp1oQvHimUVxPr79s2yL/gcdlWzGsM22P9rvsM6lzipjBjDtMTsEljivRgZZOelri/RZ3u73NDD2abfOLT8dqtsbTqZrfXplTaDbwuJ1jNFuw+l/TbGtvas32oGZ307woHobhNwB8AwCUUq4AvA7gpwB8CMCnh2H44VLKhw7rHwTw7QDeefh8M4CPHL5JD7vd9LEO5cdvbdfKRXVmykh9tXi8tTxe3lp8ylYfZfzP+NLTp73959Uv21pty/ZHa1u2H70nBYSsmYy2arK/RWBcTNqe3/EUmt1abvVRj62xmt1qQ885RD/A6IkPHPlQa+cxcYyp2Wl6xxS/G8A/HoZhB+A9AD5+2P5xAN95WH4PgE8Me34BwJtKKW+bxFtCCCGEEEJOQO9F8XcD+FuH5bcOw/C5w/JvAXjrYfntAD6rytwdtj2ilPKBUsrLUsrLN954o9MNQggh54SaTQh57qQvikspXwbgLwD4n23asH9X9PCkUIVhGD46DMOLYRhevOUtb+kpSggh5MxQswkhz52eJ8XfDuCXhmH47cP6b8uwiMP35w/bXwfwDlXu5rCNEEIIIYSQWdITku3fxcPQCQD4FID3Afjhw/cn1fYfKKX8GPYT7L6ohlmQLL0heKL0TEidnjBDtWX73QqXI+tRnlponDGhb1r+1/ztDVM0pv9qIfNqtnr7IxP+J9OPUichZE9PGLgofexvW/JkNMfWJ2WP1ezacub8k7V1jGZrX3rL2bRaiLIpz2GZ46Snv0hI6qK4lPKVAP4sgP9Qbf5hAD9RSnk/gB2A9x62/zT24dg+g31Itu+bzNs1ccqQJ5cKp7LdxqHgMtt0WBux44UQEjIh1DQ67J0XAsfz7e7uQWyG4XFIJs1UoXAIIfPluf3GezW7tizhwTab/cfTbC8E2RSabbd55w8vLNzNTW6fPrf9vmJSF8XDMPwzAG82234H+2gUNu8A4Psn8Y48L1qheOy2KPRMNqyNbPfy2xA6PSFwaqFuaqHZCCFkSYzV7NYy8NSOzje1ZrdsR22ibq8OvuaZEEIIIYSsHl4UE0IIIYSQ1cOLYkIIIYQQsnp4UUwIIYQQQlYPL4oJIYQQQsjq6YlTTMhxtOJT2m3HxCsWbHkvVmZvXMha/MeojYwPSQhZGmM1u7Zc0+xsfP0xsXxrtqM2UbdXB58UX4Ltdv+Dk8/1tb/cWm/l3W5zdU7pg67T1nt3V++XKMalxI0U9LrEmxyGh09NyCSusJTz4m7aOmy656O1K3GLX716HP5n6n2s12t9f8rjrOd4IGSJzEE/L6EbOg6wR6SfFqvZu92DRmrN9mztdnt/M5rdSrdpd3dPNVt0W2Ipn1o/df+f6zjrOR5WBp8UX4Ja7Ee93FrP5M3UObUPUVunbpuNNZlp6yl9YD+0+4GQJUL9PI0PmTpP7QP7od4PK4NPigkhhBBCyOrhRTEhhBBCCFk9vCgmhBBCCCGrhxfFhBBCCCFk9XCi3SWohbmphR/rzaujMLRC60zlg438EIXRmdKHWp1Tto39cHw/ELJEqJ/T+3CJ8xP7ob8fVgafFJ+T7XZ/4OnQMoAfSsZLy+aV0DK3tw/11uqc0gcJYbPd7tdvbx9C7kzRNm9d6pSwMlFYtFP5sNk8rkP3g/Vpqn1c6wev74+xO+XxQMiSyGi2rGd+H1FeT7Ol3mPsZvJmdOMUPmiNrGm2LTvlOVLXU9PsU/jg9X9Ns0/pQ+08sjL4pPicPPewPmv2wavj1G2bY2ghhmYjzwlq9ml98MpGPiy9rUs/j6wEPikmhBBCCCGrhxfFhBBCCCFk9fCimBBCCCGErB5eFBNCCCGEkNXDi2JCCCGEELJ6GH3inDznWJdr9mHucSYZr5iQcVCzT+uDXT+Xds2hv5d2HlkJvCg+JxKD8pL1brcP8RdPyd3d0x+8hHo5lw+n4ubmaZ9q0bk0tb4nhOShZp/Xh1NR02xgfrpNzb4YvCheG5eMf3gJH6K2z9UuYwQTQjTPWbO9stm2z8UuYwQ/KzimmBBCCCGErB5eFBNCCCGEkNXDi2JCCCGEELJ6eFFMCCGEEEJWDyfarY1LhXq5RLiZU4XdmWM4H4ZDI+RLWIlZNgNQsutLyztHn+aYd9ks6RTEi+K1cakQQ5o5hDlagl1CyCj4kySEjIHDJwghhBBCyOrhRTEhhBBCCFk9vCgmhBBCCCGrhxfFhBBCCCFk9fCimBBCCCGErB5eFBNCCCGEkNXDi2JCCCGEELJ6eFFMCCGEEEJWDy+KCSGEEELI6uFFMSGEEEIIWT28KH4ubLdAKY8/19f1ZfsdLW+3l24dIYQ8O7bXd09ke4rP9vru0k0jZJHwovi5sNsBw/D4c39fX7bf0fJud+nWTY/cREQ3AbWbjNpNRO2mYrvtvznx0rS9mj/eeqa+Wl1eW712tvokc+NW6w/eqJFnwO7+5rFso9SX7XeQvru/uXTTJudUNxD8nO+zBNm+vrQDhFwEuYkoZf8NPCyXsl+X7YJOt2VkWZez9uTmwkuv+eKltdpgfc62w2uDrkvbkHxeO1t94vmX7UtdFyFkFcgNRFon0cgr24C2BvWUzfg3VrMzNmt9YdsTtSPqE+tPp2aXnbEzQ/ikmBBCCCGErB5eFBNCCCGEkNXDi2JCCCGEELJ6eFFMCCGEEEJWT2qiXSnlTQD+BoA/BWAA8B8A+A0APw5gC+AWwHuHYfhCKaUA+FEA3wHg9wB87zAMvzS55+Qxm83TyUdXVw/bvGX7HeXbbM7ThnMi/VVrb9Sftf7S5ay9zQa4u+vbD16atlfzx1tvtcO2wdalbUg+r52tPvH8y/Tlcz4myerYXN2hFB0pYgBKbdl+++mbqzsAzy8CRVOzPZ2KNES2iV2bX58Hespm/Bur2V49kTZmNNu2s9Un1p9ezd5h9mSfFP8ogL87DMOfAPD1AH4dwIcAfHoYhncC+PRhHQC+HcA7D58PAPjIpB4Tn9vbpyHZXr2qL9vvaPn29tKtOx339w8/Zlne7R4iRUja1dVD+v39fttmA9zcPE6P7F1fP7V5f//YRuSfzSMX1rvdgy3ro3BzE19A2ryyTdvb7R7H0ZH8Nzf742OzefAR2Pt1Z2Kk2j7ZbPafnr60tqxfhCyQ21c3T2R7is/tq2d4QSxEGutpIlDX20izgb2OiT2t/aJfNY3SecTWMZpd00L7HWn2/f2DbfFRIkVYvyLN1j5vNg/ngKxmL4Dmk+JSylcB+DcAfC8ADMPw+wB+v5TyHgB/5pDt4wB+HsAHAbwHwCeGYRgA/EIp5U2llLcNw/C5yb0nZCw2bFgrPE4rBE1kwysvZXps2DJjwwa1/Gv5bLfpsHC1dng+9fRDtI0Qsh56NFuWbTlZz2pORsNPGerNtivra9Q2234b2jN7rtPlM/0xDObfjXmSeVL8tQDeAPA/lFL+finlb5RSvhLAW9WF7m8BeOth+e0APqvK3x22PaKU8oFSystSyss33nhjfAsIIYScHGo2IeS5k7kovgbwjQA+MgzDnwbwz/AwVAIAcHgqPDhlQ4Zh+OgwDC+GYXjxlre8pacoIYSQM0PNJoQ8dzIXxXcA7oZh+MXD+k9if5H826WUtwHA4fvzh/TXAbxDlb85bHt+bLf7vwVK6XtFrf54r8vVy9vt09f5Zr+j8pEvNX8z23RdY9IjvyNbhBCSRMsJP+f/bK/vmvuIkEvTHFM8DMNvlVI+W0r548Mw/AaAdwP4tcPnfQB++PD9yUORTwH4gVLKjwH4ZgBffLbjiWvjUoH8GE29HI2bbI1xOnbca83f7LaedkfpvX1ICCEJdjtgQKcG6uUePbPbZb1XRyPNbvmo80Q+TqHZrXarvI+jbBAyT1Ih2QD8RwD+ZinlywD8JoDvw/4p80+UUt6PfaCN9x7y/jT24dg+g31Itu+b1GNCpsCGDfOW9XotNJrd3gqfI2WyNryQOXrZlq+VbfnnbfN8tGF/Wu3wfNLh6XrKMywbIeukR7NlWZfz9LClOTU99PJ79Ryr2VE9GR8j21GYuOy5Lhtq9FH7u0bZXoTURfEwDL8M4IWT9G4n7wDg+4/0i5DT8pzDzJ0T9iMh5FzYiAzkeI7V8J7yC/iDl2+0I+tku93fuUZjurfbeExzbVy1HatdG9891k40fn27fdq2lh9eHdmx5Fk/W2PKW/0bjbUnhKyLmlZE+lfTVk//W3NyWnYymtozf8ZrW68+92q2LGfOKc9Ms7PDJwh5XmTi68p2We8Zg9eymbHTGj8Y2bRxJ2tlNF7dtW12OUpr9V2rf6N+IoSsi5aeAO24uzX97Ek/RlM9v1uaq9vWsp/V7yi/LEsM4to55ZlpNi+Kj6E2LhV4mmbHPsk2vWzL1F412foWH2v2NTV/e8eejknPjMnieFJCyFh6NVsvt+YH1HRO1seO4bc+tXzUeTwfp9LsVrsf5eXQBzJ/OHxiCuxrDW2avOrRvr5R0uyrF+3rIfWrJmvYdP1qRW1Tv6ZxGOJXStr8kmZ9lPzaV91e/ZpH+wrgCGvLvmaTY1kJIWOpabZGv8rWatDNzWMdldcHe9plsa/R1X7pN4TZV5+LT1p/tT2t2bK9V7OlXvHFe3Wy9zrflmYTsgD4pPgYvL/gAf8vF7td1r2/8KM8vX+1t/62jv6S0XnssteOqM2Zv6ayaZ7fhBDSQ0azZb02JMnT7Ohve2svo6uRTc9XXX4qze56dW9SxwlZAHxSTAghhBBCVg+fFJN1omMzAn1jmqPt8i1/UbbGd7fstMYPyrLYieJO2jLHxMGs+RKltcYctvrXK8O/YwlZH7Ux3dm4u5F+ZufktOxoMnl74h7rMfE1+1n9jvLLsgzfqZ1TejR7h9nDJ8WE2DF/IkA75xcsY/Rq4wU97Pg8bScatyi+eOPBvfp3u334GzsG3ZbRY7v1uEA9Bly3V+exvukxj7a9MpZQ2iZjDq1PetmOr5R+snUTQtaN1gKZZ7LdPtVtqxmRnkRzcrw5PVrXbHk9V6c2P0eojcXWmi3aLnaAx2PAtR+RrnqabfPrfN45xTsfeDaifp05fFJ8DN7TRqD/6aP3lpnMnV7rqWLrCV1096nz2GWvHVGba3fZrXbUtk3xlLAnxIxOa61nttvlVtgb7UvGXu/47MxYRdke5bf2bF/WxijWbAhR3YT0kNFsWfc0sqbZ0RNKba/1xK5lU7ZpX3X5qTS75y1lGR2f6ilhS7P1/JhLa3brHDN2W6vd2oeMZtvtUb5In1tl9LYFwCfFU2FnIXt3e9FMX+DhR6bzeHdd0VNF+6ROc3eXi/rg2dBP/OTu185ItneYr17t8+oIEa2IFpl2EELIVESarZELQ/2kzP6TJHlqGmsj8VjsE7fIphd5J/qXRvLXNFtHRhLN3m73ee0/UvynhqwAPik+hmhGcjZ4eO3utvaEsFaHzTvGVs1G5Kt827+uojv31t15q12EENLLWM0GcnpaKytlMk8Bo7xZHc7+06Tt2T7y/KJmk2cOnxQTQgghhJDVs4yL4u12f6eZ+Wy3l/aWEEJWz/b6LifZ18kX+hBCyIlZxvAJ7+USEfybhmSohZiJJqVk1jPbe8Le9Ey8bKW3ttUm8Mj2KH802SYzcadmoxXejsyW3f1NSrZLuWlnIgRoa3b0Gu/a+qk0uxXGbOy2Vrsl3euv2ivEZXuUz/vWZaI6ZRtDshGyALzJJPrV2B52UqAOcyPoUGp24ovNa8Pt6HqiV2PLxBj9enCZMOT5qtd1W/XERx0yTfzX/ZT1zbZtu91P4LGvrLWTfXRd3gRPTsQkhESaXfuXOKvZoqFjNRvYa6KXZifJj9VsvawngmY12762207yFG5v/Qn9eoK8RU/cFF8WptnLeFI8V6IwPdng4bW7W+8uLFOHtTHGlmej5at8e0JTu9MVWn3l+X0MrckkUmcrjxd2JtomZT27LfuZSZJ2PVt3lBb5n51Eam3VfGvth2iCFCE9jNVsIP4nKaNd2fCUshzlzeqwV6+XV/eL7SPPr2M027l+7OYcmu2Fj/TsevZb+cX+JTU7m7flW9THkf2FaDafFE9JdIcI+Hda+u9k+zRMP8GzdZTy9AmdfXJo67Pp1patU794QR/orSeDYvP6ev9t70oBP/C5bUeUTgghU1HTbCH6J8nqqeT1XnDg/aviabZgX3zjabatz4ZXk1BsNc2+unrwr6bZ1q7G+krIguGT4mOInl7V7mDHPA3LPpmLbGRC+dT89uzafNl2Zn2R9JZfhBCSpUezM9poNUzyZrU78imjk73/zLTal3262HphRa2/CJk5y7gotn/btPISQgi5PCndHtpZCCHkDCzjovj+/vE4lhq8IyWEkHmQ0W1KNiFkJizjopiQqWlNJokmqOhlL29tm9Tr2W1NlIz8k3ZkJ2DW2qTTaqF7PPuZyT4131r7IZogRQhZD+fQbBs+ckxYTC//XDTblu3R3YxmR74sJCQbL4qPwTtgAP9glO29B6AtF323DuJjIl14dm35TDszvtg6vdi2U1wQ3d4eb2MO2HbMuV3H+DbndpHl0KPZGW20GiZ5M9p9rGb33oS22te6kKrZbcUjnzr6xCk5tdasRbM9SjvLpWH0iamJYql6sVclfFkUUaJWXs9K1rOHZaKDzBS2dXgzi29u/JnDYnO7fRyzEPB98OIW2tnR4qu2Ke3Tftr+iOyPZbt9qOf6+mHmdbQuy9vt0zcs6nT7kTSbx9suy9vtYx9t3lrZnrSont48UX/Zspm213y0+4GQKWjFv/Y0z+qp1idZjqJCSBzc3e6xzvVodi3ig41qpF80kdFsQWu2xp4LvLbo8lPS0mhPWzKa7a33aLatZ4weR3atL149rTZ7dY05x/X6uEDNLsM57rwavHjxYnj58mWc4fo6/+O6utr/8M9BKfnIEGO/gfzMY/mu5R1jR3wYY6OnLm03U/cU+87WE61HaRlbtXZO0ae1smPsZsrYPLU2T9X2aJ9cmFLKa8MwvLi0H+ekqdk47KLEo6GC4Sy78kv+TPEbA2Kdsmljfy+138cx9sbqfqTBSc0+dj8/+sn3aE5Ls6P1qExP24/V1qittp5MmayeHtt2vW62neu3niHS7WUMn+BEO0IIWR4Z3aZkE0JmAodPEEIIIYSQ1cOLYkIIIYQQsnp4UUwIIYQQQlbPMsYUz5UolM6x4dN6wwEdG5anlVfPXm7ZH9sHQiaW4hQh2aJQStG6rTvyWYj62pYZEyOyVrYnLXOMZI8jr82tfdzroxc7lJAexuqVlJVvG3YMqP8WWuV7fmtj7WV1xWuTLq9t2DbWNHvX3j1NWhpd66UcdBsAACAASURBVKeaZnvrURlvv9h6xuhxZNf60opH7LU5e23R6q9eH61mT3EMnJhlXBR7Fx21vJdGwuUADyHSNFbAbMgy+73Z7MPgbLd7e1Je2mrr0yFybH/YEEEtf+7vH0Ko6XS9TduS+m0bLBLuTdBtvLt78NGGDpK2bbfTxFCs+Sh+6XbJ/oz2l07TZaMQSDc3j/et1CERVzz7Qi2skndM2LB+QH2/6+PIy6PbYOtp2dc2on66uvIvPqY+BsjpSOn2DKajtzQbeKy79hj29K6myV59nq7rZeCxZlt7+rcriJ5KPVnN9tCabX+7otmelmrdnAJPazXe+dJrW9RWTzc13r4TrcqEuouINDu6DvD6QZ+fa2309n3U9kwfi/+SV/eF+DHlMXBClnFRPNfoE7vduJA2Nl8rTI4cTL311ezrMlLOyy9EeXWdtfZGvtny0tZsWKSx9PRl5PuY0E1Rnsifmv2MH1FfRr5F9dTqiuqp2Y/8zPTfVMcAOT1ziz4xVrMlPfMbzehez2/Q5hEurdlZf7707e+SLjL7pPd8qe14fdTKkylzjGbL9qhdUT21urLn5Naxmz1vTXkMnBiOKSaEEEIIIauHF8WEEEIIIWT18KKYEEIIIYSsnmWMKV7aRDtCCFk5V3iFUtqnmCu8wlJORYSQ580ylGiuE+0yIdG89J7wLFdXT0OiZeur2ddlvHA+9kYkCrEidbbaG/lmy+vwQrX+OjYkV60vPb96+tX6aLd7ecaEX2v5UdsnrZA9tp5aXV49mZBAnp82b80Gw7LNmntc5+bZJS6cJ2OsZgM5rW79LrSd7G9Q55mTZmf9+VL55Dm8RmafZM6XkbbYPor00Nt3Gf+q/dOoO2pXVE+trqie7G9gbEjWOUSaabCMi+I544UrkR+IDhe13T4NSVILsaKx4dVs2BttR9vQeaLQXzqflJVvL+yKbbNN13W2wrJZv/SPWcpbO9qvY5hbKK+5+UPIcyXS7N1ury1yJW91TYdF83RR0OEMPc222qq1LdL1rGZ75az/UZr37ZFpv12fKkZt9uEY8Hw1dU7t6vWltLNcnGEYLv5517veNVS5uhqG/c+h/bm6qtuaEqlTr3vbdZrN5323ttk6onqjPL3+RG2JbGVsR/55NiJ/j2GzeThe9PG12TykSXrt29qw21t2W3VE9nSezeZxmyK73javbKZcxjfd5lbeaF/YfWXTZgCAl8MR+rfET1Ozh2G4wh/kJBt/kOzp49if7Q6VPtrofGeWx+j3qTV7ivwn0uxjZftL+6+lYVnNbunYWM3u0d5Is1u2TnU+afVlpk26vEk79hiYkki3l/GkeK7DJ8hy0fEVgYfjy1uXfN63LWO3t+y26siUsfErhchXrw5dtqdcy7dMv9T6yO4rm0ZmyyyHT5BlU9PJMbHeWzpWS8vU1aOh2TqO8S1TptaXmbp0ec/2zGH0CUIIIYQQsnqWcYtuB4+38hJCCLk8Kd1O/gtICCEnZhlPimX4ROYz5p3jY9lsHi7Y7UxLOzNeZml6M0DtDE5v1qa37NnJ5GnVFZVp+W/rrNm2/aRtCF6dul5CyLzJaPY5qWm21RZPs2tamtHvU2t2TZdbaa1vajZZAct4UjxndKQHHS3BhqiJZjuLeEg5EQ87q9ebZaxnOYvw3Nw8jWQR1WVnLEu92g9dxrbl7g64dg4hCR0jtmx/6dnWu51vw2svIYQcS6TZVic93ZE819ePo1EA7WgOXj0ZzfZsAk+jTGgN17rtaXZL/21/eZptbVCzyTOAF8XH4A0mF6KB6bIO9A3ub33bsq2B8da2CPqYyWO2nLXV2x+1dtt6x6Kfvmib2fi6ekiPtmG3t+y26siUsfErhcjX2pOjbLmMb5l4n1E/6j7z4nLyyRMZwzGa3TvpaSrNjmzaCbYtP6L6xmp2ZiLZ1BNjPZ08JtZ7TcfGanamTquPQqaOY3xrlWn1Zauu2vlkqrB8J2YZwycIORX2icbdXS5+p407qu3J9pub/Ulhs9nbvLvbb99sgFev4gs7L6bpMDxc/FrkyY0XK1ojT6Vse+WkGP2LED0B0k+ZxK59KlWLv+31g31Spttl07Zbv52EkOeL1SKrYdFTesDXOE9vrWYDe42zml07P3h6q8vYc42mZUNrq7bp9Y+uV9Da6/WltNv+wxGdu+w/ILrfdNoCWMaTYn1nkslLSIvoiVHmyUftKYktp+vKbI+e6EQ+9zyxiZ4S9faDl27tRuUy/VMr65Un8yW1j4Z2FkKA8ZotaVkbkm+Mznl6W6sja6NXk7PnppofGrnxkHLZ+h/ZxexJPSkupdyWUn61lPLLpZSXh21/rJTyM6WUf3T4/urD9lJK+WullM+UUn6llPKNR3s514l2hBBCYjKaTQghM6Fn+MS/OQzDNwzD8OKw/iEAnx6G4Z0APn1YB4BvB/DOw+cDAD4ylbOEEEIIIYScgmOGT7wHwJ85LH8cwM8D+OBh+ycOr9H7hVLKm0opbxuG4XPHODpLvMHkQjQwXdaBvsH9mYH8YyY2SV4Zt9RrS9DbrK3e/qi1W/tGCCE9HKPZGd2t5R2r2ZFNO8G25Ydss/WN1ezI56idO3+XEDInsk+KBwD/aynltVLKBw7b3qoudH8LwFsPy28H8FlV9u6w7RGllA+UUl6WUl6+8cYbI1yfEXbilZ2k5OUF/AlUdgB9hAz61/XaiQGeb3rylx6Qr8vKoHhrS9cr2EkMXug5W87rq6i/CCGz4Flrtk2L8nqTn2qTnvS6aK+elJrRbOCp3uvQaDLO8+4ur9meVt/ePky8knL2XGHxJngRsmCyT4r/9WEYXi+l/MsAfqaU8n/oxGEYhlJK1+CwYRg+CuCjAPDixYtlDixrvWsdqA+ot4PWbXrrW+xmB/33Tm5qtafHt9bEAqHHh2OInhhlnnzUnpLo5Si8TTbsjbQ1EzYt47f3VKr3H4XWvxutcpn+qZX1ypOz8Kw1G8hrVUbLvDzRJKWxuqvrjuqs6XHkm87r+evVm52kdixjNVvSMjYiPcvqnKe3Xn1eWkazvfyZPtHLWnvtOxVqumv/ncjU+Wh9/rKRuigehuH1w/fnSyk/BeCbAPy2DIsopbwNwOcP2V8H8A5V/OawjZD5cHt7+bp6fTinz+fkubaLEDItnJg5LWO1d2y5ie6NTklz+EQp5StLKX9UlgH82wD+AYBPAXjfIdv7AHzysPwpAP/+IQrFtwD44rMcT0yWzXa7v3MtZf8X5PX14/VS9nkkn5cefXvL2+3jenV6VCZbXvvplbfpUT36U6uv1hctX6L1KE9Ur/hHCFkPLQ3WupXR7No2q4Et3e4p76XV0lv662l9pky2nRl9bvXzQjQ786T4rQB+qpQi+f+nYRj+binl7wH4iVLK+7EfQv/eQ/6fBvAdAD4D4PcAfN/RXnqP8mt5CWlhh6wAub8dbXrrr1hZrsV3FMaWt/575b32RW3VZbJ/nUZ2bP5oPSpTq5fMHu4mMikZDQba8XdrGibbasMbNWPKW331bOn0lv7q8tLuTJlsO6P11vYFanbzongYht8E8PXO9t8B8G5n+wDg+yfxTpA4xRnO2fGt10oC9bFD3ut5s+NYe8c3jRnHGc0itvVmfMvMjO71gRAyazKyfdZzZeb1ta3xnr1RHmQ9Go85Vnd13VGdkR7bcto3GWOaOVfUfLD+79q7h5BLs4zXPMuPK/O55JNimZnsvbrW+nV/vxcf+3peO0M5ehlJa9ay9kXsXV09nbEs5Tw/rS05w9Ve3St12DyZ1xzr7dGrOsX+duv3CyFkNmQk+6JEr1GPXpsr+qMj9VjNljIWrb1W27TGaXvig9V765uU915ZXBv/KeX06321Zou/0WuBa69q13Us5BW/hCzjNc9zfVJc+yt7bGQK768MSev9G1/3hZ1VXPPBrmf/zonKem2wPmX6zrNBCJkts3tSPEazJU+ksS3NbpWv+QPkhzrVhhmNiXjh1ZnVbLfN7d1DyKVZxpNiQgghhBBCTsgynhQTMjV2HDeQG4un01tjyTPjCTVjy9sxjrZ8K16mLmfLZMcTRuMtvTq89ahMVC/HlROyPloanI2/W9Mw2Vab86MZU96LPV9Lz4znlvzST9kx4Jl2tub+RNsXOK6cT4rJurFjo/UYam8sXm0soJdH2O324nB39/SNWHoMYIQIvB6PaMd5e23yxkB64yVlXY9htOMfa2MvvfGWXv6oP4XN5mFMeZSP48oJWS8tDd5uYz20654twWq21jI798bDaqjk1Reuej6RpIu+Sd2Rv57P+u2Ketx3NF7e+unlt3hzl+zbaqO5QAuAT4rJOsmMscvmy44blDzRWEGNNy6wN6xPj09iv1W2Vl9v/lodUbrXfkLIOhijZVktzWh2b0jNjA9R2Za/tp1eva36jvFvlGZj9izjSbE8gs989B3QqbF/bdT+9rDfsmzXbXmd5pWvfeu/mGXWcMYHr46oXbZeb3vNVk/fRW0jhMyOK7zKSTZenc+pMZrd0tqWZme0OvInq7OR7mbPR54fx2i2Z5OQBbCMi2KJPpH51P6CPgc6zJoNMaaxf5nLXw06NI78ZeOFKJOytb/xxZbY80LGRX8reeseUR7714nnF/AQMs67Kxd6wwwRQi7OPa5zkn3pPyyt5lrNjv6ur2m2/NVutS8z9Er/de4NW8pqtujkduuHgYtsyDbtb69m2zbzIQZZCBw+cQytt4vJcu2NNrW/JKK/bGx9vdtqPrX+Umn5otuese/1p+DZ0PUQQkgPrdBhrbeQtb71xWEtDOYcNbvmh26H158CNZssnGVcFPf8/XLO4ROEEEJc9sMn2qeY/fCJZZyKCCHPm2Uo0Vxf3kGWS2uMXe9YPCCfR/4GteU0toyXvxXWp8cnsV8r26qvN3+tjijdaz+ZJTJ8okXmwpkQAG0ty7zSWtJlvUeze0NqtnyslW35a9vp1duq7xj/xmg2ktdxF2QZasQnxeRUtMZj2+NJxFHCjunxctG4PSmj/5oFHvLY/Nqe9tMioXRaYwhr2LHf2hcv3Jr0SeSjrdPzpRV6zvrg9S/Hlc8ePikmJ6E1Hrum2Z7GtjRbjxv35vNYe9ZXi/UxKmvnENntNo+n2ff3D/XpeqKx7dYXbdeix5tbfyLN3j01MzeWoURzfVIcBdYWP7wnZd6dnt0u39HdaevOMPN0LfKpdffY8kXalLlrtk/6Mk8UtS/H4I3P08u1b6E2ltratOlRvTa/TcuUS71yNfAzu601XrLWbk3WL69/yayZ5ZPi2ssQPM216a1vHd9Vv0BiCZpd80PyHKPZU1wQXUKzvTo8HyL/Wuutusb6VdPsWr5aO3p89/p3ASwj+sQc2W4fDjIbTNtye+tfyEV3e4Ke1ew9Add3YDIjOIpUIchLD7RPNhC3N/vae5lD1B4P7aPMypYnj9fXD+3UbfPgPwGEkGOx2qifqll9tGVqESREs6+vH//rYevJaLb+h0Z88vJnNLvWD9Y3QaJevHr1+GnrGM1eyIsbCFnGk+I54s26BeI7ozERJLQ9oXUn3HrZhLZdi4qRmems03Q/1O5Sxba0JdN+m4cCSwgZQzYSkM6beSqpy3lPyvTy2HNDLX+tbbVttbxWs3v+0XH7qr5rCJkDy7go5phiQghZFBxTTAhZGstQormOKSaEEOIyyzHFhBBSgWpE1ok3aUUvtyYH6jc8tcIAeelRvTa/TcuUa00mivzs2abDymXaEf3b0zNhyNsPhJD1cAnN9uqwPvRodrbsMRMjW5od5Wu1w+t7L3+k2QsY+ciLYrJOnnM4r+fcNkLIesn+Y0zmyQL+yGf0ibHI0zj5SPSJ6GlWFBantk1v9+z31KNt2yeKni0vreZ7qy3Wx5722zxTPCXcbvf2ZBa1fJeyT5N0+eh0W0bnt2nRd88260+mXFRGlr2y2bb0lMn2ZU992jYhPUS6VtPHmla1NNtbzmp2Nn9Gs6NtPZrd034vzxTUtMDqTE1TenRujF71aLYsi55Zv3rqGnNOyfbnMecwWV4AfFI8Fu8lCXaSn/67xqbd3DzY0C+CEFuyrvFCnnkvbLC2rH/6pQ+SpgNui5BFeMHCI/+krdquDUVn+8Late2YgtaMb6AeV9Kb/d2KlZmNOjJFbMqoTM2eTcv4nimT6cuoDa18hIzBaonWl+jFPcDjlzp4ulR7yYElo9nC3d3jF/VIuvd3d6SRtXOCTff+are29LCElmZPGae4pgVRTN0xmq3bP0avvHI1TdMvdzom6knvOSXbn8eewxYSgYQXxcfQClOTuWiZKjxZ9ockdsb+UHovirJ2M33BiyFCyDGMebHN2JtGyRfVYe20fInsteqpXdiMtavbmKmfmk0WwjIuinv+fqk94SSEEHIWGJKNELI0lqFEDMlGCCGLgiHZCCFLgxPtCCGEEELI6uEt+jG0ZuTWYgfqGcN2Qp5XFvDzyISIbGxasdOKUVizo7dZ/8bYzfaFNxN7LNG+s/4KmX2ZORay/Vrzp+f4au0vm5ZpS0+ZbF96bWjlY6xi0suYGN5jfx+Sz6sjq7Wedut8PTFuo/QxdnVfZDV717erXFrnEPtPcescMnY/1mzrvuk5h/aeyzPHZFR39pxcO3bGbEPyH/8LsownxdKhmY8cxJfw0Zv1a/2RGbkSAUK/X17n1TN5b26AV68eDlRvprSNYBH5IzOBxZ6sD4Mf31bsRP1q67m52duSH7i0Vc+ytuW9vrC2ovYcizcT++7uwRfrs+fHbrcv46V5M76lv6WP7Cz4zeYhTfsjvng+2xnrd3cPx4S3D2v7NNNHeha83r/Wf+u7ptYfUfvIssho9qWIfk+134o+nmv5RLOH4UFndRnRDF2upnHeOUAiD+n21Npi7cu66Kz2U/+mvXK6L6yf59Jsvexpdq1+3f81+7IOPNVlHXEk0mztR03T7Llc+96j25lzg/ZPa3C0/7022P6I6lugZpchO1b3hLx48WJ4+fJlnKGUvjHF52qT1CXCXpvBq/3S6dmZwGNDtLTqm9q3Xr9s/0S+eHUfQ09fttJqfZ7pK68vPFs1mzV/xu6jVn21dkZ19/RHz764IKWU14ZheHFpP85JU7OR3z3n2o2lAAM69KyVPva3OoVmZ/L0/s5PrNkFw1H7+dFx0tN32W2Z/LW+8HzI1mPzRfVk12v1RD7ptmR972zzTCQbQKzby3hSTAghhBBCyAlZxphiPTYlk5cQQsjlSen2TB4dEUJWzzIuihmSjRBClkdq/MTp3SCEkAwcPkEIIYQQQlbPMp4UzxUdOgV4eEqtQ5LI9kx4lciODlnWsiFpkT+ePR3aqhZmrmYzst3yy/aPFzbGhmqbMiRbrS3ZtKht2f7X6PSofGub9ad2vEV9nqkvamdviKDM8V0rR0iWjGb36F1Ls22dU2m2LGfst+o7l2bv2runSY/2ZvUymz8K+1bT7Gw9Np+1VVvvDf/m5ekNEVirt1pu/kOllnFRPPcxxVEIFx1SRYcp8cLm6LISame7fQgdo38sUVlr38t7dbW3ud0+ruP6+qm/wD7kis5XqwN48FXCbEk520+6jOfjKUL5aLzwc+dmDj7MCfbH8yOl2xc4UUbh1KwWa70TejW7pWk6j2c/0myt2zXN9tpsw8xlNNvrl3NqNjCf0AVkHMnLuEuyjIviuY4p3u3a4bB0ms3TCmcj9m3eqGytTu/bqyNqS5RPaLVByPSFZ8PWcSz6Ih/wbwhkWe6IbUxpHa9R0r2ThF2W+iIb9gQb3UBEflt71mdb1tZXa0/tZsarN3PyjGx77dP7hBfR82duY4ojzbbrnk7JdmC8Znu2s7ruaWqtTE2zM+eETL+cU7PFzqk0W+fJaK6tI2sz0kKrw9pvqdfqn9Zse86wZWp95Z1vonKZfvLSbm4A3GLuLOOimJCp0ScMoC9upJdP0msnwSivt936mIllWTt5t8pGfdJzcvfqrZX3/OqxQwhZF6fUbFs+o7nRes+NhPYP8G9KrN1Is2sa3uqr6IbOlsv0k5dmX/QyUzjRjhBCCCGErB5eFBNCCCGEkNWzjItimWiX+Zxzop3M1pV6ZVmwaTpPbVvvrGJZrtmPZpvaOuxyK19Ury3X0xeeDc8nQsi8yWj2OYk0O6PPNZ3NanZGmyNd9zTVK5PR7Mw5IdMv1GzyzFjGmOK5TrQ79USfc0wkytYx1pcp2sAJVYQsk7lNtKNmn65cy8aZ738IGcMynhTPle12HxJH7oZl+fr6YXm73X90epRXb99uH+qw+SWvtm1tRXXZpzS2/pq/LZ+s/Sjdq6PHtuQ9Bv0UJXpKrp9+1J6e6PTak6HoSY21kXmKX/O75bMt68WtHvPEy6u3Vr5mp/XPB588kTFkNbuVN6PJY3S9ZrumpzV/Wz5527J9om3bMlNrNnBazc7oXE3rsjYjLaw9qW891ffOGZlzUVS2Vi7TT17aUjR7GIaLf971rncNVa6uhmH/zKH9ubqq25oSqVOv2+12uZbXbq99R2VbddX8a9no8a2VL2ur1p5j2GweHzN2ubbt6spPj/LatM3msQ/e9p5yNZ/s92aTb3vNRs2PWp5W32X7UuzPAAAvhxG6t+RPU7OHYbjCH+QkG3+Q7Onj2J/t8Fg/WprUo82RnUzZHt3L6Gqrba06onK180WjrmNl+9H+69XfjC73aLaktTS7prlZ3zJ6O4VmW1+z/nVo9rHHwJREus3hE2Sd1GJMD0N9mxxjmfJRmvbB2+7Zj8pJnqjMsb5HNjIh46I2RvVn2qDzktlyj+uUbJeyjNMQmQE13dLL2e9aGZsGPD1viFZHmi3LXlrWN11H1vcxmm19zfqXaZvOO3M4fIIQQgghhKyeZdyiy9iUbF5CCCEX5QqvUk+Br/AKSzkVEUKeN2klKqVcAXgJ4PVhGP58KeVrAfwYgDcDeA3A9wzD8PullC8H8AkA7wLwOwD+0jAMt0d5OdfhE5vNw3vjgccDzMUXGVyuB55HefX21uB3SbNlW3Xp/vHqr/l7c1P3ydpvhe3pGezvtZsQMltmOXyiR7NreTOa7Glwb7jNmo57Nj1/M+cgu02WW32SPQdtNsAuv5sIuRQ9wyf+CoBfV+s/AuDDwzB8HYAvAHj/Yfv7AXzhsP3Dh3zPF/1OdGC/fH//sP3u7vG7yIWbm/2F/qtXDxebUlbKlbL/jurZ7R7bvrnZ2xuGpzY1kl++tW3x1+a/uXkIs3N7++D7ZvNg377H3fpi+0Rj2+v1Ey+GCSHH0tLs3W4fOeHu7nFeq0WSJt9WkyPN1jpXs6nR9qxd8Vdrqmj2MOz12mq29IP+1v6IL7ZPbB27HTWbPCtSt+illBsA/w6A/wbAD5ZSCoBvA/CXD1k+DuCvAvgIgPcclgHgJwH89VJKOcz2G8dch09kJhHJepTWstMaBC/bdTnPps0X1RPlj95bHr2nXee3kxNsP2QmCNg2HYs8qQHyT09aT/ejvNmnRdknPrac5PF8qj2lz/reCgfU+0SsVn+mDWKXzJpZDp/ITvyMtHGMnUw9ns3ITra+Xs2uTSgbMzHMtmkKarqll6d4Aq/TrJ5Jmv731NNsSfN8z/oW/UPbYyuj2dbXrH+Ztont4JCcE1kl+u8B/OcA/uhh/c0AfncYhleH9TsAbz8svx3AZwFgGIZXpZQvHvL/09FeznX4BFkuc3ghSORDy7c5+A7k/JiLr+TszHL4BFk2Rzxbm4Sxmn3Kus9t4xgWcHnWHD5RSvnzAD4/DMNrU1ZcSvlAKeVlKeXlG2+8MaVpQtpst/sbqFLi4PX6kwmqb5e323rQ/MheJk/2ZS7b7eP2tl4w09NGbZ88e6jZ5OK0NNfT0Z4XlkSa7ZXJaHbr3BHps+dHRrOzdVO3QzK36N8K4C+UUr4DwFcA+JcA/CiAN5VSrg9Pi28AvH7I/zqAdwC4K/tHAF+F/YS7RwzD8FEAHwWAFy9eXPj2j6yOTJxioC+WpbcsNlr59bZMnmib9TmKcWl9q7W15k/0Fy15dlCzycXJaG40NESn92p2LV+U3hO7NzOEMqPZ2bqlPHlC80nxMAz/xTAMN8MwbAF8N4CfHYbh3wPwcwC+65DtfQA+eVj+1GEdh/SfPWo8MSGEEEIIISfmmMFcHwTwY6WU/xrA3wfwscP2jwH4H0spnwHwf2F/IX0cc51ol5lEJOs2TU8UqtmJBrXXJgN4NnW+Wj1RfhnsH/WBt93Lkx2YX2sTJ1kRsgxSun3GZybZiZ+eNmY129qppY/R7Gx9vZqt/fEmIvdODNN2FjLJipCui+JhGH4ewM8fln8TwDc5ef45gL84gW8PLGWinQ6V44XOEXSIG11Of282+0Hx220c4sfWI+F+rA962bNr2yDCaGcRX1/HYYHsNpsfeGrXttn79uwQQpZBaqbd6d0I8TTbw9Miq1fAY43z0qUuWZeIBdF5QvLJhaa2q/PodEH7q+u29vW3549Xt+0rWb+/r7eJkJnDab/HMDaETmubFxqnZjcapyS2rV81u7r8qdrUG+qH46EIIVOQ0WzZBvTpnFeuR1uz5SLfI7vReSGj51O2iZAFwItisk5acYqF1l+ItfKtN0l5doVWnigOpPW59hZCyduKo1nzJ/qLlhBCJmRzdYf9KxMAYFD/METLh3XASa+U/1J+x5abr+WHrTOT5vlhtrnt6rB/gfuUJYx85EUxWSeXjtd4btbWXkLIs+L2FW/Ayenpec0zIc+f7Xb/FDT62NiU2+3TMtn4lXo9EzczypeNg1nzK7Id1e/Fc87EU9b+ZNtaaxchZBW0pJmf+X+WINvLeFK8lOgTUr/e1or04G0rpf6a3Nq37QPrV8uupHllp2pTxm7t+5T/wdReSS3banEwozy1OJrZuJmRrehbl5cJOL1+RvV76bW22PzR2M5WXd74SDJfUvtpaGeZioxmyzagT+fEvpfXs2vrrpVrDcXKDoES5njDVwAAIABJREFUsno+VZsmiD5h3079qC01zc7kqWlhtK5tHKvZtbQxtr08U/lxhGaX3Rl/6yNZxpNiiT6R+USzh8/BZgO8evX4ok3P1L252fuox2F6/upIEnd3j/N5NiI7doayd8Pgbb+7exyZQtdpZx17USO8OiTqhYct481q1nXtdsu45SRk7WQ0+5JozbZaKLr16tXTPDXdjjRbyrewmiraeXv72AftR02zxTero1H0CFnebB72j9xUe9E0pJys63ZozSZkASzjSfFcid5CY5/KZZ+S1e7yhMiGfWIX1W/fcKZtZp7utfJlnli2npratkZPVgkhpIeaZkcaVivb0j3gON3Xmm1tRPmz7crov7ah29M6P7maHe8WQubCMp4UE0IIIYQQckJ4UUwIIYQQQlbPMoZPzHWiHXl+eK9A1ev2WLQTXLw8tTjIrZjA0QQar2xtgs5m8/CmqR4/MxN4JD0TTzma6Flra61dZN6kdHv+k2/IjGlpNlDXGZueXdc2jtVsnVbTSNu+YyZdXkKz1UigubKMJ8VLmWhHlsN2u/+x2lBhGhE++bYTKe3EGtkWvTp2s3k88cXDlpcJNF6anUzjHftyIXx39/SV27Yt3qRL3T47oVLn9SbuiE+63fLqWTvxxuvLmxt/chInWy6HuU+0I8tCa7an255m2wmHvZrtTcKMuLqqa7bVQzthEXis2Rbxx7Mt1DQ72mb74lSavQCW8aR4rtg7p+ipXOaOywtjo7frOiMbUi6ycXX19A1nkh7lz7Sr94ll66mpXtd+eX05ltaEG6m3Z/JKVMab0BLla5Vv+azz1fxptSVrw5two9sZrff2V639hGSpabanXVpnxoTHlHJeeaCt+1qzrY0of6Zd1gfPlqfZOt0ue7Ye5ZngBmisRkyp2a16a/54ZXW+Mf6MaVP23DK5Zj/dpXNjGU+K58h2W7/zie7grq8fh+zxnvp54W30k8Pork3XrbeL7Vev9ttKeXon693NypNEfacod4JRm6MQR7bt9s7Ve9JYe/LIN7QRQnqxuhn946BDicmTSfvkzvunxmq+PDn0ymuNq2n27e3DP1sSds2GQatpdi1UqLRDp9u2S51ag7NPTSU/hzuRhcAnxWOJnijqtN47Kc+u5O25E7XbvZA+Np+3HLXF86lVLmqP1ydeu7z8hBDSQ+ufjZ5/SFqaHdnu1Wyvjqiczutt9/ogasfY807Ul4QsAD4pJoQQQgghq2cZT4rtuKVWXkIIIZcnpdsTjDUlhJAJWMZFsUSfyMC/aUiG1oQboH/yilcmCl+TnaxTs90zEebYiTjZPLIs7bTtzvpXKzfVZEtyejK6TckmWcZqxJSaLWWiemv+eD7rfGP86W1Tz7llas1ewA3wMi6K54h38ZT9AQLtsjZvz4/ObvdmL9t8djlqS+1HWSsXtcfrE69dXh3HEE3Uy0zgu/Qkv+328ZjDY/y5dFsIOSc1reu9GWxptl2vaZqtT2u2V4dXrqXZ2ahIUdu9/vN88fpyqhi12YdjlkvrnGj22ieJL+AGmGOKx3J7+/TirBZfEHiYBaxjJuqZvrqspNkZzoAfb1DQM36FaIa1N9REtkWRMvRkDW/WtfSBzFi29dhZ0nq7zu9FovD8Hct2u/dTZpZ7sS9tHlm26973dluvo2YjWha/ZEKM3q9SV62817ZW/bodPb56fkf96S17banFJyUkg9Vs0TQvIpCgNVhrk9V7q9k2XmsU41uXFWqabbVSl4k0e7vdn7PsxbYup6NMWB9tu2ybvDZY/6YgqzE1bY40bIxmZ/RT+rhHsz2989rWOveMPdeM0f5WHy9Es8sw9s5rQl68eDG8fPkyzlBK3/CJc7VJ6prqu8emzQu0ZzRnImG01ntjHwrH9EdU/hjG7IOePmn5nem7MXXY9vXU1Tq2enxt+Z0pm+2/C1NKeW0YhheX9uOcNDUb+V10rl1ZCjDgSA0G+rW653fl2e31KeNjrU4pX7OV7Qdlq2A4aj9/yd0ere49z+i2j9W5jj4JNdurQ8q16h97bGXrrOlwo4+PPQamJNLtZQyfsH/RtPISQgi5PCndnslZkhCyepZxUcyJdoQ8YMeFEzJXUo+KT+8GIReFmr0YOKaYkKVxezuboQOEEEIaULMXAy+KCSGEEELI6lnG8Im50orxOkXMxFaIM9kGPE3XeXrDB0XrNVuef0JPP7TCuE0Rkq22D7Jh5bx2Z0PX1fZ5lG7brdfHhtpr1d+MO1mpq+cYtstCax8R0sOYuNytkGbHarberpe933cmVJpnt/a7j/yytnr7QZefKiRbS6daYeVq+1wic/RotleP55eQ1Wy73hM3uCdvrV292l/rc7E3xTFwYnhRPJbt9iE8mQ61I+iwNnp7VE4OQPvOexEffYEpZeRHbMtI3vv7x7a1HRt2J1qX5Zub/V9A4v/19WP/vbbqNnpt2mwOP5Sd3w/Wp1Mg/mr/vT732mTDLUkMSukjvR9sWmSj1p93dw/9Djyke/u1hj4Gvfy6321oPeuXbZe1ATyEG6ztR5vW+l0Qcgw1zfZ02Z74RX+9Y79Xs718+nd1d/c0nFukt7Ztuj5PO3SbvfL6t6/Didk6rK+nxLNv+7ymg3b/2n2oy+r+a+lzpNnil073jg+N1VjxwaZ57dQhYaO80bWDRmu/R3Q82mN3YSzjotjejbTyngOJOyiUUg8nZbfbcjZNb6+FdPHsCZnwVq1vKS8/Hml3lC8TniVqS22bl+cY7P7L+FvLo/vZ9pGX5tmyfVjrr0zenjA7mfb3tNkeC6321o5/2e7VS+ZNaj8N7SxT4f3+ZBloH5deWnTs92q2V0bHhe/V0dbvpxYWbIxma1+9dk3BFJptt4nvx+plxp+xutzywWtTtr7MObynba3jYeYs46KY0ScIIWR5ZHSbkk0ImQmcaEcIIYQQQlYPL4oJIYQQQsjqWcbwiTnSmu0P7Ndrszl1udZMz2hGr2dP8OrNzsiNZuZmImfYtkXfnl8ZXxl5gBAyhlr0BeBpWkazM1GIxEZNsz2N05Eeahpty9f0VLdhSs32olLodu/au4eQS7OMi+I5TrTLIrMyZYbu0nkObQD8NwxlbxCitFoopt5QaFFdQD6vl68nXE+mPzJhBqP+tds8v6KTOm+MlkFKt8840a5GFFmlxSk1cUl62/J1irHjU2i23tYKVTc2vFmPX7Ic1ZPxwdYt+TPngcyDrZ62RTdJC7kxWsbwCZlol/mcOhyMILNVdb162a4zpNR8ERHxwrPpb41Nu7p6CD0G7AXAC82jQ9Tc3OyPD3kKH9Wl69Q+21A7m82DrcjO3Z1/LHrtFZuvXj1uj7W93dbD99S210JjkeWT0exzonXbarT9pmbPl5Ye1zRb29CaPQyPdU7SvfCaki+j2Z6eaX0VW9qOXo402/aFfgBnNVsThT215xTP39b3M2AZT4oJmRobgkfIhp+xeWxZve7Z0/kyIZ16wgtl8kZ+ed+2v6I67fKYUD5RH0XbCSHr4ZhwaDaPtQfUQ7PVNClTZ+Sj166ojTVfZN2GT7XlIv9r/TtGsxmS7YQsefgEIYSslSUNnyCErJ5lXBQzTjEhhCwPxikmhCyIZYwpJoQQQggh5IQs40nxHMmGZJN1mQBFCCHk7Gw2QNkN6sm0t6y+AT7FnhAGjCFLgBfFY1lSqBzyFBuCR2bOZsPP2Dy6bC22aU84s7HhhTJ5I79q4XW8cja0j17uDeVT66NoOyFJKNnPgIy2HaPZtTjWNU3qCb3m6Wsmr5B9AJfVbK8/ptDshYZk40UxWSdzOkPOyZeIJfhICHnenDqE33PSuUu3xat/Af+8LGNMsdyFZD6MPkEybLcPx8z19eNvb5tO226flt9uY7tRmZ46ZVnX00qv2fe2R77Wytr6onbW6rXf2kamL8h8yWg2IVky+hnpVEazM1qarfPcmh21xdNXnbdV/5SavQCW8aSY0SfI1HjxKOXb22bTgHZsyFaZ3jpt3ODInvajZqMWu1int8ra+iLbrT7w6sj0BZkvjD5BpiQbgz3SqTHxfKM48pk6z63ZXlukXK2+Wv21fu/WbMyeZTwpJoQQQggh5ITwojhC/w0hfy9EeexfGd7fB156Zrn1l03kGyGErIjt9d2DLF7fNfNc6hP5Rgi5PMsYPnEJotcAe3m8vzJ6X+UYLWf/siGEkBWzu79RsuiHwNR5qnobpdtlr2zDVuQbIeTyLOOimK95JoSQ5ZHS7RNHFCCEkCTNi+JSylcA+N8AfPkh/08Ow/BDpZSvBfBjAN4M4DUA3zMMw++XUr4cwCcAvAvA7wD4S8Mw3B7lJSfakanx4lFmYl5KWeBxXi82ZKtMT51e3GAdAzKKK5yJ0WlvOltxlnWara8Vd7PWB604mlFbyXzhRDsyJRn9lGWdP6vZ2k4rjnyrzktottcWKRfV16q/1u+9mr2AG+DMmOL/D8C3DcPw9QC+AcCfK6V8C4AfAfDhYRi+DsAXALz/kP/9AL5w2P7hQz5C5s/9/dN/GmSbvNxjs/HjL+52+7HeMsNYyog9PfNYX8zpPLYusaMRURV74tOrV4fg6Lt9+t1dzr4uK/XtdvuPlNNlbm4e8kteXd9ut1+2ddYQ2/f3cf/adkT5CCHrwdM3nQbUNVtrl9Wp3W4/Z+f2tq3Zur6MZgN7LdXDa1qaLcuRZnttub9/qOfmJtZsOXdprY/aqM9tvZq9AJpPiodhGAD8v4fVP3T4DAC+DcBfPmz/OIC/CuAjAN5zWAaAnwTw10sp5WCHkHlQC52WGcPdG9LNlgVyYxclX6uOTJs8v2zZqdrgpWW2Z/zRdRBC1kOvvsk2IK/559Bsz69Wm2TZhpKb8rzTakvUX9l2LIBU9IlSylUp5ZcBfB7AzwD4xwB+dxiGV4csdwDeflh+O4DPAsAh/YvYD7GwNj9QSnlZSnn5xhtvHNeKUyB/JdT+prV/N8gnej2jTc8s22+7vJC7L0LIspm7Zm+uHiJLbK78CA86T8HwdNnbVlv2yjZsRb4RQi5PaqLdMAz3AL6hlPImAD8F4E8cW/EwDB8F8FEAePHiRf0p8iUm2mX+muXft4SQldCl2UJKt6f5E/H2lY7q4Ed4eJznUszBB0KIR1ec4mEYfhfAzwH41wC8qZQiF9U3AF4/LL8O4B0AcEj/Kuwn3I1HJtplPnq8JCGEkMuR0WxCCJkJzYviUspbDk+IUUr5wwD+LIBfx/7i+LsO2d4H4JOH5U8d1nFI/1mOJyaEEEIIIXMm86T4bQB+rpTyKwD+HoCfGYbhfwHwQQA/WEr5DPZjhj92yP8xAG8+bP9BAB+a3m1CjkSPGa+N5Y7GcHvlW3Z02ey48szY8ppPLb+icDzHtiGy420Xsv5wLD0h66RX33r08Zya3aN1dvlUmp1py9hzyII0u8zhIe6LFy+Gly9fxhlK6YtTPIM2EULWQSnltWEYXlzaj3PS1GzkpZiSTTLwOFk+c9qHkW53jSleFdvtwx2OfK6v9x9Zlu3b7UMZvV2WdbpN8+zZ5Vp561fLZuS357Ptg2wdtfRaucif3raJncz+bfk/pt7oOLD9WrPl1Vezkd0fLb+zx3vUBs/fVt2147p17BJyYHt99+TQ42dZn+bPOtIGu9yrJdttX3rNZlazW2Uz52+x6eUbo9ljrkdqeReo2anoExfnEtEndExDQXzojQOo07OxEWtxCb08Ub1j4zdam16dmdiKkZ9RXs+f3rZljpWe2JC99WZj+LZs2TzaF2sjuz9afts2R75mtmfbWeurVvxQMm9S+2maR0e7+5unT6Fax1nr+IvK9Pyuxuhjq76e38mxmn1sP3W0o+wax0K2LZl9U9Ozc2h2xq+e48Xms/0V1d/Tr6Xk4iRHthfAMi6K+ZpnQghZHhndpmQTQmYCh08QQgghhJDVs4wnxZcYPkEIIeQ4zjh8ghBCjmUZF8WXGD4h4UU0csFdyuMLdR2S5O7uYXsUskSnea+Etss3N3F565dnO9pW80vSdB+07GXSvf5r9VNv2xYS+oWQZ8+5h094mi3aoNNbmq3zjNW8qHzNVqu+jGbXzhNjNNvrv55+6jkfHYarEnIplnFRfAnGvMK5VebY10Kf6rXSc3td9Tn8sTEVgXhZr2dOgtkTbc2WV5/23drInvBafts2e77W2iDbhWyfZfz0fCBEk314IsxN+7LU/F5qm4D2DVL2At+uZ7SkJ71mM6vZLb+yD7Wi85ROy2r2mId0tbzWhwXc9HBMccR2u9+ZtZAv9hOFZMmGW9luH+quhbaqhfvyfN9u43Bw2ZBv0fLYcGS1dmXD4tTam+X+3h9yI68Lt2k6vyzL9263r//u7nG+m5sHUdrt9ukasWHR27Ufd3cPNqwP9/f7uobhQbw8dN7b232fyaxiXZf1S9oJPK5D7Ol2Wz+8dur+0O3QiJ+19hAC1PW3ptuR3vZodqvuTFjMll2ruWM0W75rdUT+tWxGZTPhwXp0W7B6DDzVZy+/zutp8s3NXrs2Gz89okez9bLUl9Fs4PH5xNYldnW53W7/0TrqabbVWtue+/u9nevr/beXrs97r14t7gEGX97RsqPtWdul7L9r4Uwytmy6tXuMnUw4Iklr2fWWbdmoLq+/onZF7fTqyobEsdT67Zg+j9oT9f+5QyC1+rKnTdnyxxy3rTbPAL68wye7i6aW7FB/ZRmo63i0vfeYbZXJ2KlpSca/mg/aTo8WZGy2ttu6D9sLhvBYeORC67yc6fOx/d+r87bNtbSMZtfsZdreU9ex5z3zXdu/5ybS7WUMn/D+jqjlJYQQcnlSuj2TsyQhZPUs46KYcYoJIWR5pB4Vn94NQgjJwDHFhBBCCCFk9fCimBBCCCGErJ5lDJ+4xJjiVrgRjU3vCeell6OQXlk7NrRWK+yVniWqy0X+ectid0zcz6hd2bA4rfbWqMWMrvkW9ceYsGOtsl4Z2++1+jJxrTMxtDN1tI7nnj7MhB9a2Izm1XLuMcU1/Y38sr+naHvvMR6V6dFsu9377fZqttiJQmvV/Gu1S/sTnROi8GCZkF2t9tV869GeWv/XytZ8rKX1vougJ4Z26zxq68rkGXOOWEhItmVcFF9qTHEUbsSGO6mF0/JCXgG+LQl1IulevToUi/VPylvbOkSYrs+GEru6eshj65I6NPJDlpA19gfs+WfTdB5tT2+3fnh+a3a7fX8/11ieWXraOLY/MuXW0NfE55JjiiNtrqXJBUqvZnuhEQV9DhAtlrp0eY3oqWibtS91Wz31NFuX1W2zmh2FX6y1L2pbFENXQnpZzc5y7tAF59au56iVuk2n+q1PyDIuii+BjsUK7H/cY0LTyPJu54f8yYaeimy3wqFk06K6s+u1+mo+RPXX/Kr1uy0TISc8K/TRjRBQT7fbajcvrfqAdvlM/dGJOHuTVWtzb/tqNzmtdtfaR4gmEyYNqOuOHJc9ml3TNG1P6hlra2xYMJse+VLzoaaznj293qpDl6khNwrHarbNO7Vmt+oao9mybG/aPNvSdmu/5p8to9dbmp0+19xi7vCimKwTOeFlT0JA/0kqUyZzg1LL06q/dSLO3IhM0b7MjV9Ub61+Qsh6mEqzbd6pNTtTV69my7K9aeu5MarZtWmt9cyNkefXzOFEO0IIIYQQsnp4UUwIIYQQQlYPL4oJIYQQQsjqWcaYYh32I5N3CqLQYtaXKCQJsF+OQtpENnVZL08rXEomVIqX5tXthX2J0mv11XzQeWqhenS9tX63ZQghlyOl20M7S099NY0RIt2TiUxAn2bX9FXba51TenTc29YK1ZUJndnSak9na2HXoj73bO1AyEVZxkXxJUKyTT2znTPl50UUf7J1k5M9SR1z8yL+1cpn62+diGv11OrvbV/tJqfV7lr9ZN5kdHvK+TfZ80SG53wOmGvbWsfCVJpt806t2a26xmi2LLfiS0cP1lr+2TS9ntHszLlmATc9HD4Rsd3ud2Qp+ziNtW9v23b7YEPn87br7yhvq76MPW85U0emLZn6Itu6v6O6vPZGPojNDBI+xgtp4+UTJK9mswFevTr8+HePw+VsNvsT9s3N422S33J3ty9v83jhhjQ3N/5M5chnqce2XWKaap91nRKbWveN9jWKD2vr1D6JXWm3iL/XZolFTYimR3s8TRE9aWm2tRvpWUt/s9+Z80hvW3rPaZGdKTRb912LKTRbL9/cTKvZwGP9sj55WqsvOiPNlmW58PTCsYl/Lc3W9UobbT3aR63Ztk/lfQGAf67S9S+AMkx5Zz2SFy9eDC9fvowzlNL3pHiKNmk7shx9R3mAXKgrr4zN2/Kp5UOtfE+eqC0Zf720nn6z2zI+RHjHyZj9XCsXpdn2jbWZ6dden3uOEW9f9Nbt2a21q1b/hSilvDYMw4tL+3FOmpqN/O45hWSP0h6d1+aX9R4tb/1uM+Uyv/tWnqgtmb5o9dNUmn3IVzCEx4J7nByr2dk+se0ba7PmT82H7LYpNTvTv53+zUSyAcS6vYzhE/pRfSYvIYSQy5PS7ZmcJQkhq2cZF8WXGFNMCCHkOFKPik/vBiGEZOCYYkIIIYQQsnp4UUwIIYQQQlbPMoZPXAIdozITZiob3qUWcqonDErGhyg0i17O1JGNH9yqz0uzcZx7Yh9HPnizfi2ZGKSZPq61JUqT2cM2f69Nr19r+ceGTsvG7B5Tt/a9N7wRw7IRS4/21I75KOZuZDf6vUbhu1q/v9rvpuVTti2tvsj001SaLXZ2qDO1Zrf8kfQeza7ZjPxp7Zfstik1O9O/0fkirHf+8weW8aRYOjnzmfJEaUOaRN82DxCHurLbgcehrzQ6LJekR/W1QljZ5QgbluXqyg+TZbdLGC4J8yLrur5aaK3b24cfsg3dottp943O7+Vr4fln+1Yj4Xu8MGQSmkbC3tjjQdK03e22HqrGCz0mPojPGhuux7Yrwob10bOHtc/2uNW/t1r75Ri34dq0XZ1Hh0cii2RzdZeT7Ku7trEe9HGnfwM6zfuda03TZT1N0qGvNLvdPsQYEGu2LHuandHMzeZxXGDvtyltAdqaLXk8zdbYNNFsG+rL+qK3AafT7Ah7frVh1Tz/xU9Ps8cQ6Z+0x+pdrV3eMWFDcdY0W/yQtmvN9vz2fLb1eue/hbKMkGyXQO50xoSwyuSP8rbq9NIjf2W95Zcup9ufCUGUrbvV7mwYmR57NaK8mf3a62OmP716Mv3bKuct9/Zt1q8e2zVbY/rygjAk2zz40uHQ83s99njM6KRXf8u3TJnoNx3ZOpVmjy3j+JwKyZZt99Tn7s62uPuwVxNb7Wztu0z/ZG1n6m3UNxPJBhDr9jKeFBNCCCGEEHJCeFFMCCGEEEJWDy+KCSGEEELI6uFFMSGEEEIIWT0MyRbRCmnSCvMyNpSJV2crXdvQabKeCeEi6UIrbEyt7lbYud72ZcMEWXs1vJA6rTprbe0NgebZ6q2zVS5abvk4NuxQzXYm9I+eHZ7pS4ZkIxbvWJHtsp7V78he5jeij+UezRZb3nbPtveb9vw8pWZny2S0dYc6Nc326tJts2E/e8OQebZOdZ6wy7U075ho+ZU5drx29Gi228aZzLKrwOgThBByBIw+MQ9Kmc/MdjKO2j7k/l0+c9qHjD7Ry3a734PX1/tvvext227jcjrNKyt5pKxXV61uuy2yVVuW9Uxer81RP7R87m1f1Da9XXzI7N8eH3r98Y6HTF1euq4nqr/nuMzUn/Ut42v2+Mn6mNnHZH1kNTKjVa3jOPrtZ7TM02ztS49m9+hJ9JuL2lLThR7N9tpm68z8pnv7ume/Rfu8poe1fqr1tbe959gbc4zV9qHn0xjfWuewBcAnxRGljIuZ2soDjItr2OOP0Ip9qJdrPkTLXrtqfmbalW1fNh5nRManjP3e/ZapK9vHtv4x/Vyr32tX1nbmeGj9LrLHxIXhk+J58KVDYqxmwykL1PNKetZuy05WF2yjj9GTWlvG9EmPFpk6U3GKe/ra+pppa0+ZbD9FZXv8OMdx3TqfZMpX/Knt33PDJ8WEEEIIIYQE8KKYEEIIIYSsHl4UE0IIIYSQ1cOLYkIIIYQQsnquL+3AbKnF++uNWzk2zrC2k419q2MHevW04iD2lPPaNTYG5LGxffX2njjFNZ9smtjv8acVx7PVxz3xRqeKJyrLXrsy+7AW8zTyLRMT3CtLiOYYzdZpst3aHKPZmbRsnNkodm1WT6LfXE/s81qf9MT29WL57lDnlJpt0zJ6mI0d7PV19nqg59zfq+214yITu7jnmM/s3zkwDMPFP+9617uG2bHZDAMwDFdXj7+jbZtNXEa+N5sH2y17PcveNs+fyBdJq/lubbTK1+rK9GV2H0S+io3M/q3tr97jQNvLHCNRe2r7odU/3veYvsvu28x+9frKHk+6vshf77i6MABeDkmtey6fOWr2/myWPA5bx+6YfMfq97Ga3frN95TP+JPR+B7dAfb7MLt/W23MHgdTaXZkK3M+yX7X0qyGZjS7t+7a8ZTQ7Nr+PTeRbjMkW0Qp+91pv7002Qb4ZbxvyVuzl12Otll/Wu2qpbfaasv32Mr6mO0H27+t/Vvbb1Fa5jjw/Mz22xifeo+3Vt9l962XP7MPtV2hdcxH/X5BGJJtHnzpcMjqXe3Yle29+Y7R72M12+bNtKHlX82fmm9jNHsYqj/pJ/u31UZv35xSs3t86tHuHm3Vdff4mK1b243aYutV6zORbACxbjfHFJdS3lFK+blSyq+VUv5hKeWvHLb/sVLKz5RS/tHh+6sP20sp5a+VUj5TSvmVUso3Tt8cQgghhBBCpiMz0e4VgP9sGIY/CeBbAHx/KeVPAvgQgE8Pw/BOAJ8+rAPAtwN45+HzAQAfmdxrQgghhBBCJqR5UTwMw+eGYfilw/L/A+DXAbwdwHsAfPyQ7eMAvvOw/B4AnzgM2/gFAG8qpbxtcs8JIYSjjGghAAAdkUlEQVQQQgiZiK6QbKWULYA/DeAXAbx1GIbPHZJ+C8BbD8tvB/BZVezusM3a+kAp5WUp5eUbb7zR6TYhhJBzQs0mhDx30iHZSin/IoC/DeA/GYbh/y5qcswwDEMppWv49DAMHwXwUWA/aaOn7FmIQlgB7fA8rVBVrRBUeltmuRZ6p5WuwxF5NrMhhGz5Wl1jQ/hk+0H7F6FtR/ttbFgcKWP9jPaJ156o3lr4ptp35FPGfuRj5jirHUfabjaMoFc3OQuz12wgfxwC+fBbvWG6epen0mxr0/rW87tutTur8T2aXQqAxmHV0rUoDFrrfGbtjNHsyCdZz1wjZM+/rfNVzcfe8K8tzW6FEXy0Pk/Z0KQuikspfwj7C+K/OQzD3zls/u1SytuGYfjcYXjE5w/bXwfwDlX85rBteVxdAff38TY5KHa7/Ud+AHq7zn93B1xfP6xb+952seltv7/fH5C3t8B2u7dvt0XoWISRL9a2Zrd7aIvXTzav50utL7Vvnm2v3poPltvbfN65cuo2PIc+IutF9KFXs4HHOqvzWduteq0tWZYLE6nP4umbvciTfDrvzU1Os1ttyKDPD+fQbCAfuoDadT56+rq0s1ya5kVx2T8S/hiAXx+G4b9TSZ8C8D4AP3z4/qTa/gOllB8D8M0AvqiGWSwHEasx4bf09mz+nvBcXl3aX72tN1SWV3erLzz7mXbU+saWr61726OTjSBCrvFuRuTmQPohyqPLt06K0c1NVEevD7Lu2an54N1oSR/VyrfSvRu3qJyt2/NT2yREk9W0Hs3uCVNVC5GV0cPIt952ZM9fLf1u9YnU1QjFFfbRk+2oo28KIh0Dnt50ZG5UPO3M6qTWtUy69cHW6/nm+ZVtc0azI98jHe6p+0t5bjF3Mk+KvxXA9wD41VLKLx+2/ZfYXwz/RCnl/di/p+S9h7SfBvAdAD4D4PcAfN+kHhMyBfqGAaifdMbe5GRugnpOylkfWnYyabqPWuUz9sWe9jVTd+uETQhZD5mHVED9psDbrsuP0claXV56tN56wDT2Rimj2ZHv2Zupnv0zY5oXxcMw/O+I79/e7eQfAHz/kX4RQgghhBByNrqiTxBCCCGEEPIc4UUxIYQQQghZPemQbKsjEzYtCm3SG8KnFeolsqXr0v72hECrhV/JhpCLQrFEfZXpS1u+tu5tb4VkI4Q8P7Ka1qPZLZuCl8ery/rh+VCrW9Jq7egJ+9mj2VFdXvnaephPjXkl5ALwojhiaTPbPX+nasPS+iKDjdVZO+mMvcnJ3ARlT8o9PtTsZE6OtZNd66TWc3OVqbt1wiZEYyNCrJVzavaUdbXmYrW01EZD8PJ722VZ0np10rsBqaVbH2y9nm9jb/iymh35nrmZatUteXaYPRw+4bHd7ndkKfu4jrK83T5N13l0Xluup0xtW9auLGfse+2K2qJtttoty9a+ztNaj9KydbYQQbLbdPzP3W4vajrMDPAQF1QEwYbrubnxw+t4oYHu7uoh4sQHixdiR9dv7ehy1gfJKzFHdX3STt0Gr/+8kHQSH9W2T/uo+6GUx2GAvH1EiEdLe2qa0autU2l2rUyPZtfy955vrK89mp3R7WM0u4XWbFkXIs0G9tu1Hnm6KWnAY63SdWvNFt2zemvPCzXNjvRb1yn11TTb5te27LYezZb+1vG8vfPAQuCTYo8o7mIpT9N1Hps3E4PXK1PblrXrpUc25MCvtbs3bI2tT9vXeWyZWt3ZvtZ1RvSEmhkbls3b3tPmTNtrfkf+1nyKfLb2rO3e49irr9UPXllCNC3tEU59bPZodq1Mj2Zn2pL9ndbK2Xw96xmNrTHmXOnpWmTL255pY0ubvWVgf9F4Ds2u9U+rP2t91Gqjtx9mDp8UE0IIIYSQ1cOLYkIIIYQQsnp4UUwIIYQQQlYPL4oJIYQQQsjq4UQ7jyjuoheqSufJxHvMlMmG0mrF9s2GBJPZpFG7x4St0cu18C62TK3uTF/bOiOyoWa8UD6ZUDTR9p42Z9oebW+FeIt8iHz2jrXWMdAbxqfVDwzJRlq0tEc45bHZq9m1Mj2aHbUl0q6e81aPftXWa3VmQnaNOVdaXbuEZnvl7bZTa3bPMRvVm/1deDYXEpKNF8Uet7f78DA2HImEqhIk5IgNxaXLX13tvzebw0Gxe1pei5wO32Jta8Su/CjkYLU+10LKiF1pl9jSfulwMdIuCftyc/O4rTaPDl1zff3UF+uz107PH53PWxY/ajzH2MuEkMeaEGmbp2+SP6PZgK/1Nc0GHnRRND8610QhuFqabeuWdolm27ZGmm3PL955KqpXf0vZjH5nsFE7yLIo7SyXhhfFEbXwYUA7rIwtr8t5dkRkWmFYtL2afVuuZ91+a3+1n15bbR7bVq+vsqGRovQxoX30iUiLt6xHsR29PPaE6sUOjm6avDzWvi7r1WNvLrx8tTpqPnn94aVlbEU3Qa3vWj2EaDzNBnLaYctnNbsW3jFa9srbvD3rrXbV2iE+2Dy1UHA9Ibh6Q8DV0E8hs1rh5YluaLzykl/3Uyv/qTU765PQc27JbJdloE+zcYu5w4tisk7siQioX7RHeWz5zI2GV39vjMio/sifWh01n+xyzbdM+6J+7vF1QTEvCSETMibmbushTzZmr1d/Lf8pNTvrk9Bzbslsb/kc7YcFwIl2hBBCCCFk9fCimBBCCCGErB5eFBNCCCGEkNXDMcURUfgwwabZkCW2vBceRduRWb3WXrTshTvR5W25nnUvPItu192d31Yvj22r11dR27x+ivrds0sIWReRZre0wwuHldXsWnjHaNkrb/P2rLfapfVYt6um2Z7dzHlKf3t5a77uQMhF4UVxxLEz25/rzHivXXbbEtpuT0RAfGKqCbzY0vlqJy0vbmrvxX50w1Lzp1ZHzSevP2onwpqtbNzMmq+88SE1bNSJc/AczhVz8AFAM2RXSztqadFDnlZ5nT9zM6C3nUqzsz4JPeeWzHZZjnz2+mUhNz0cPhGx3e53ZCn7GLt6ebt9nMemy7reXitj02y92qbe3vIjshH5KzY9e55Nu63VjpYfUT2RT14brS8ZbOgajcQqrcXSvLt7GkpHyr569bi8PI3xQtzosta/q6t9GR1vVPLasD7ij7Wx2Tw86bH+l/JgO6LWF9rH6+vHMU51PVKHDtOj+8iG8fH6g5CIsZodaaEuU9PerGb36llWs6O2tXTS+tMq16PZnk+tc0GWmkb8/+3dX4ht51nH8d/DHG39h2maGA6dZKbFgORCYzO0CfYiVtQ0iLkpYhAaJHBuKlQQNEEwc66sN1aLUiwYakFaFZWGEKgxrXhl2xObpifGmFOZoRPSHpQ2XgjVU18v9lrnvPPO+3fP3rPW2uv7gWH2+ve+z7tmz7PevWevZ7a3Fy+OYrmul8t529sn81yYU0u1lVtztr+f34afH31927Fa0r5Yzo7t51/DwvVhzu7b3d6uy9kTy+PmhnhlHdjb23OXLl0aOozjzJYvQxJu99tIHVMq59W3mWur1F8qrrDt2H6pOEvt15y/1PmuiT11HvwxxbSc62XirSmpU4qj9ty3xBmOrbb0T+q8tp6HMMbTfh8BM3veObc3dBxnaYw5+9hTojVn1+TZ2LrWnJ2KJ9V2ar9cnKXcUZsvS/stkwML5yH3a31i26qvj6nzGxtrrr/U+VumndZryml+trU5O4y94WdtcmNJ28m8zTvFAAAAmD0mxQAAAJg9JsUAAACYPSbFAAAAmD1KsqXkylOV6lNKJ9fnSsDEylj5/abKgbWUlUnFFbbdUl4sXBeruxk7f6k4UrGnYsqV3sndedxyrkt95+IolTrLLbec+9o4a0uipcZXqoVdU2ItjDH1HC59pyQbYpbN2bnf/Vx5wdac7be76pydqq1f+7vkH1OKJ3aOW36vw3FcX1+4C6slZ5f6Tm2rqfkfOw+nydmp/VvGEmuj5Wdbm7PD2Ett9P1tb0+iJBuT4pytrUU5Eb/ciHSj7FRfMqVf14ut948J++i3+cLjYyVg/GLrYcz9RcAvFeZvD/vo1x8d3djut9HHELblC8cYjqE/NjzOXx+LPXZ8+HPxv29vl+tunmVdzrHUAB0zzhFWJcwJ0smcfXCwKBfm5+0w34X5zG932Zzt57Nwfdh/30YpZ5euR2FbIT/n+8fnym3FcnZ/TvvyXv6kKtZWuL6mjm1L6YJNyimbMhYr7zI459zgX/fcc48bncWv3/Hl1PfUfv56f1ttO+E+qWNTx8T2K/WR2r6K43PnMPyei73lPLba2Vkcv7V1o62dnRvb/PWpfcJt/eOtrfTx4X6p9mv68NuM7Rcuh23321PrU/3Vxh2LJXcuU/34bQ9I0iW3ptw41q8x5uzrv/YteSb8PkTO9o+tyXelPDhUzs4dX3lsLnUnt5Vydil/rDJnp/JTro9cu6nrQq7t1LhysZbiLp2Xypyd+/metVTe5p1iwHd4uPg1Nlt8l268G9+/++Kvj+0TO75flzo+3C8VQ00ffpupOP1l/10lf4xhn7HYSzHE4vbj8vtKncvYGML4AMxTKWeX8vEqc3bs2FIfuXZjfcRydrg9tb4mhlLOTp2XcP9o2xo9brQDAADA7DEpBgAAwOwxKQYAAMDs8ZnilJ2d49UdasvZnKZ8lb9/a2m0fn1/TGuJrmVK08TKtbSUnKk5p7lzlmqzVI4NwGZqyS9hmauhcnaYt3L5rpQbh87ZqetmTVs11SeANWNSnLIpJVDQplRHOpXwa2tGS/Hjw/1SMdT0Ed6Alroo9sv+iwh/jGGfuReJqXWxuP24+lhy5zI2hjA+QDp5k2kOOX71TntOl7kRq6b2fy5/rDJnx9ov9ZFrN9ZHLGfXvOjLxVqKOzZe/7zU5uxSHeoRMNeSRNZkb2/PXbp0aegwjtvdTdfHja2LbevljsvV8g1ravr1JGPbw9qUYb9Svu9UrLlz4NenDMeQi7PmXMb6DNuPnb+aOsXAipjZ8865vaHjOEtjzNlmkpN3ES7lLimds3PHhTnvrHN27Niaa1Qq/tT5ivVVk6NjcZdydtefHR4kX9eYtb3mwfiM6WeYytu8U5wSKy3i/0TDdbFtvdJxUnl7riRWGG+q3/74VN+pWHPnwC90X1PmpaacWOochPGH4wpjymlJ9DUXmtwFoOUC2dpXr/XCn7pItezTEl8YT22cmQsoL3xwQkvuSu3fq815Q+TsXKmsdeXsWJulc9ty7v08n+K/85jLv1LdmzulFwOxnC2V38xaJiemllv6L42j9vqSi690jnM5WwcaOybFmKeWRF9zoSldAGovkK199Vov/KWLWM0+LfGV+snFmTtvAOaj5kVPTd4r5S6pri59a7u5nJhrt7b/0jhqry+5+GrOce6YkaP6BAAAAGaPSTEAAABmj0kxAAAAZo/PFKfESovESozktvVyx5XqQpbKgcVKoaT67ftJ9Z2KNXcO+tIwteWzasqJxb73H+IPj42dP+oUA/PUkrtS+/dqc94QOTtXKmtdOTvWZuncxrbnavsfChgUk+KcfjJWKikjlUuHxZal43e3xtqVjlcm6PfzKxb4ycWPw38cKyPkt5k6PhZT2P+5c+W7UMOLQ+wO1lCfvGN32PbL4RhqtST6mgtN6QJQe4Fs7cs/Fy0X/tRFqmWflvjCeGrjLP1zHCBUm8diuTJcH7sz3895tTlbOp7PwlxYE3ef7/rfn/D4PvZUTP76mpzt7xe7dvn6fcKcHXN0lN5WUvOipybv5XJXS136lnZLOTG13NJ/aRy115dcfKVzPPEXPUyKU2rvPA3Xn7aKQdhubl1rWR7/mGVi6pdTJYmWOU+xc9yyPVdyqFaY9EvL/TopfvHM9RO+oPCTVa6vXPvhtlQt01w7pXqkh4eLdp3LtxubSMTqoqbE+k1NAIDQMmXAWkqJ5XJYbl0shlyby15rSmPNtVVznmpzb+s1pbU6QZhzwm1SPF+UXvDE+omVzUy9URPrJyb2YiyWt3O51d831t/h4WJ7ab9UH7lyobGxbEjOZlKMeUq9oCgt16yPXQD8FxDhckupnJr+YnVIS/GG+9fsV5ow5OJZZjISOwbAfIwpZ8faWjZn17zZVdr3tLm99o00v90NzNncaAcAAIDZY1IMAACA2StOis3sSTO7amaXvXU3m9mzZvZq9/0t3Xozs4+a2RUze9HM3rnO4AEAAIBVqPlM8Sck/ZGkT3rrHpP0nHPuw2b2WLf8W5LeJ+nO7uvdkj7WfZ+e2jtPa+5srb0jNdZubl1LWZ5UlYOWmPp9UiWJljlPYdupsdec3zA+APOSygulXNKvy7WRy6ulnB2LIdXmaa41ubG2VpnJ9VPKvbXnOYzvUMCgipNi59w/mtlusPohSfd3j/9M0j9oMSl+SNInnXNO0j+Z2U1mdt459/qqAj5TsTtT+ztP/WoB/j65agSlMkF9eTa/jVg1gNhdpaU7aktVDmLxhvGV7iSNnafYuak9J/24au9gbbnTNfWCorRcsz52ken7W/bFSc0LrP5xrA5pLt6aeqSx/UoThlw8y0xGYrEAoVhuC3NRrFRmqSxbqnpMrEyk366/fmenm/wdntwe5q5cDk3lUz8+/7hwfUwq9pxUzu7biPXhn+cGO1tHMvPf8HDSsfSWWq5ZH3tc+r6KbaX9TtPfMusaxi01xzKFtL1s9YnbvInuNyTd1j1+m6Sve/sddetOTIrN7IKkC5J0xx13LBnGGuXulK+5ozV2TOyxtPydrH6ctf3WxhWW8ImtX7bv0jjDPvvl2nbCiWDMwUF5n1UK+zuL/lv7qN1/2djP+pxjpUafs6X63NZL5Tj/cSw/le7Iz7XV77NMrq7N2bHcnIu7tnJCbc5OjbXUV8bBNf4CiPU79Y123bvCrrjjyeM+7pzbc87t3XrrracNAwCwRuRsAJtu2UnxN83svCR1369261+TdLu333a3DgAAABitZSfFT0l6pHv8iKTPeOs/0FWhuFfSG5P9PDEAAABmo/iZYjP7lBY31d1iZkeSnpD0YUl/aWaPanG/6C91uz8j6UFJVyT9t6RfXUPMAAAAwErVVJ94OLHpZyL7OkkfPG1Qo5C7U76mCkHNnfm15W9i21qqGKTibik3FFu/bN+t5YT65dp2pnCLK4DVq81tvVR5MP9xSwWWUls1ZdFaK9LE2g/jq6kcs8qc3e8TjjXXFyXZMALLVp/YfFO5U37IOKdyjiL29/d18eLFQfp+4okntL+/P0jfwEZzzfd8L2fsuS8V39jjzhegANaOf/Ocsru7ePV67lzdd//x7u6NNvzt/vrwmNJyzfrwcay/lj5r+gr7KMXbHxPun2qn5nFq3Bn7+/tyzsW/dnbktrbUl1UpPm7cd//ixeXGddp9K84LMGmx539puSUvpXJk+BX2U9O/3+c6c3asj1y8qXOTamfZ3HWK/LR77ij745jq1+65o+V/F7AU3ilOSdXgTX2Xjq/r25DSNS79Y3qp5Zr14eO+f7+/lj5r+grHVIrXPw+tdTRTj1PjXlbs57bK2s9D7gtsspa65/1yS17q5fpIravpP6zvG+uvZbmUC3JjSMVUamfZ3HWK/HT43W05ZWIcc07O7Hv8n5XgLPBOMQAAAGaPSTEQ4kY9AJgW8jZWgEkxEBr7zSgAgOPI21gBPlMMxJRK8vmPW8rcDbkv76QA2EA30rWTjn2veTzefXe2jrT4x8A4K0yKU1I1eGvrLfZtHB2dnJSkJly55Zr14ePt7ZP9tfRZ01dNzeKwzVitzpo6mqnHqXGfBu86AJOys3XU3ZgUTDCk/HJ0WzhZ8ffLHZdbV9G/xfYp9Z9bzkzGwjhS8Z6IqdDO0hNCt/Tr9s1N10yIzxqT4pRV/JaNpVbkWfS3qj42N7sBWKODa0wgAJwOnykGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsMSkGAADA7DEpBgAAwOwxKQYAAMDsrWVSbGYPmNkrZnbFzB5bRx8AAADAqqx8UmxmW5L+WNL7JN0l6WEzu2vV/QAAAACrso53it8l6Ypz7t+dc/8j6dOSHlpDPwAAAMBKrGNS/DZJX/eWj7p1AAAAwCidG6pjM7sg6UK3+B0zuzxULCt2i6T/GDqIFWEs47VJ45n6WHaGDuAskLMnY5PGw1jGaRPGEs3b5pxbaS9mdp+kfefcz3fLj0uSc+53M8dccs7trTSQgTCWcdqksUibNZ5NGstcbNLPbJPGIm3WeBjLOG3SWELr+PjElyTdaWZvN7PvlfTLkp5aQz8AAADASqz84xPOuWtm9muSPitpS9KTzrmXVt0PAAAAsCpr+Uyxc+4ZSc80HPLxdcQxEMYyTps0FmmzxrNJY5mLTfqZbdJYpM0aD2MZp00ayzEr/0wxAAAAMDX8m2cAAADM3uCT4qn9S2gze9LMrvrliMzsZjN71sxe7b6/pVtvZvbRbmwvmtk7h4v8JDO73cw+b2b/YmYvmdmHuvWTG4+ZvdnMvmhmX+nGcrFb/3Yz+0IX8190N3/KzN7ULV/ptu8OGX+MmW2Z2ZfN7OlueZJjMbMDM/uqmb1gZpe6dZN7jmGBnD0ccvZ485xEzh7bc2wZg06KbZr/EvoTkh4I1j0m6Tnn3J2SnuuWpcW47uy+Lkj62BnFWOuapN9wzt0l6V5JH+zO/xTH8x1J73XO/YSkuyU9YGb3Svo9SR9xzv2opG9JerTb/1FJ3+rWf6Tbb2w+JOllb3nKY/lp59zdXhmfKT7HZo+cPThy9rjzHDl7XM+xds65wb4k3Sfps97y45IeHzKmyrh3JV32ll+RdL57fF7SK93jP5H0cGy/MX5J+oykn536eCR9v6R/lvRuLQqMnwufb1pUR7mve3yu28+Gjt0bw7YWiee9kp6WZBMey4GkW4J1k36OzfWLnD2uL3L2qPIcOXvEz7Har6E/PrEp/xL6Nufc693jb0i6rXs8mfF1f775SUlf0ETH0/3p6gVJVyU9K+lrkr7tnLvW7eLHe30s3fY3JL31bCPO+gNJvynp/7rlt2q6Y3GS/s7MnrfFf0WTJvocw8b8fCb//CNnjy7PkbMXRvUcazXYv3neVM45Z2aTKulhZj8o6a8l/bpz7r/M7Pq2KY3HOfddSXeb2U2S/lbSjw0c0lLM7BckXXXOPW9m9w8dzwq8xzn3mpn9iKRnzexf/Y1Teo5h80zx+UfOHhdy9uYY+p3i1yTd7i1vd+um5ptmdl6Suu9Xu/WjH5+ZfY8WyfXPnXN/062e7HgkyTn3bUmf1+LPVTeZWf/iz4/3+li67T8s6T/PONSUn5L0i2Z2IOnTWvw57g81zbHIOfda9/2qFhe+d2niz7EZ25Sfz2Sff+TsUeY5cvYNo3yO1Rp6Urwp/xL6KUmPdI8f0eJzXv36D3R3Z94r6Q3vzw+Ds8XbC38q6WXn3O97myY3HjO7tXu3QWb2fVp8zu5lLRLt+7vdwrH0Y3y/pM+57gNRQ3POPe6c23bO7WrxO/E559yvaIJjMbMfMLMf6h9L+jlJlzXB5xgkkbMHRc4eZ54jZ4/zObaUoT/ULOlBSf+mxWeJfnvoeCri/ZSk1yX9rxafnXlUi88CPSfpVUl/L+nmbl/T4k7tr0n6qqS9oeMPxvIeLT479KKkF7qvB6c4Hkk/LunL3VguS/qdbv07JH1R0hVJfyXpTd36N3fLV7rt7xh6DIlx3S/p6amOpYv5K93XS/3v+BSfY3xd/5mSs4cbCzl7hHkuGBc5ewTjWPaL/2gHAACA2Rv64xMAAADA4JgUAwAAYPaYFAMAAGD2mBQDAABg9pgUAwAAYPaYFAMAAGD2mBQDAABg9pgUAwAAYPb+H2BXTX4+yB9VAAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Plot page 2 with figure at the top in the form of a table, and characters, textboxes and lines."
      ],
      "metadata": {
        "id": "02btDtQljYUV"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "xmin, ymin, xmax, ymax = page_2.bbox\n",
        "size = 6\n",
        "num_pages = 2\n",
        "\n",
        "fig, axes = plt.subplots(1,num_pages, figsize = (num_pages*size, size * (ymax/xmax)), sharey=True, sharex=True)\n",
        "\n",
        "# figures, chars and lines\n",
        "ax = axes[0]\n",
        "for figure in figures[1]:\n",
        "    draw_rect(figure, ax, \"green\")\n",
        "\n",
        "for c in characters_2:\n",
        "    draw_rect(c, ax, \"red\")\n",
        "\n",
        "for line in lines[1]:\n",
        "    draw_rect(line, ax, \"black\")\n",
        "\n",
        "# chars and TextBoxes\n",
        "ax = axes[1]\n",
        "for c in characters_2:\n",
        "    draw_rect(c, ax, \"red\")\n",
        "\n",
        "for textbox in text[1]:\n",
        "    draw_rect(textbox, ax, \"blue\")\n",
        "\n",
        "\n",
        "plt.xlim(xmin, xmax)\n",
        "plt.ylim(ymin, ymax)\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 509
        },
        "id": "DU7t_YDoielJ",
        "outputId": "a4113fef-0c34-44ad-d465-56c04973ea91"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 864x610.971 with 2 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAsUAAAHsCAYAAAA6gNB7AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nOy9b6x031Xf992+N5AAqQ3mJ8v1tWeojKiiSPzxI3BFFbW4iQKJYr8AQlQFh1pyX7gtKZWCkxfl96IvQK1EQapcrFiVqdoAokW2IpQWAVFVqaA8P4IgIUL5xb0Tj2WCIcZtg0jyXHZfzKxn1qy71t5rn5l7Z+ae70e6mnPO3nvttfc58z3n7tl7nVJrBSGEEEIIIXPmDad2gBBCCCGEkFPDh2JCCCGEEDJ7+FBMCCGEEEJmDx+KCSGEEELI7OFDMSGEEEIImT18KCaEEEIIIbPn+tQOAMBXfuVX1uVyeWo3CCFkmNdee+13aq2vnNqPx4SaTQi5ZCLdPouH4uVyiefPn5/aDUIIGaaUsjq1D48NNZsQcslEus3pE4QQQgghZPbwoZgQQgghhMwePhQTQgghhJDZw4diQgghhBAye/hQTAghhBBCZg8figkhhBBCyOzhQzEhhBBCCJk9fCgmhBBCCCGzhw/FhBBCCCFk9vChmBBCCCGEzB4+FBNCCCGEkNnDh2JCCCGEEDJ7+FBMCCGEEEJmDx+KCSGEEELI7OFDMSGEEEIImT18KCaEEEIIIbOHD8WEEEIIIWT2pB6KSyn/eSnlH5ZS/kEp5W+VUv5oKeWrSim/XEp5vZTyk6WUL9rm/eLt/uvb9OVDNoAQQgghhJBD6T4Ul1LeBuA/A/Cs1vonAVwB+C4APwTgh2ut7wTweQAf2Bb5AIDPb4//8DYfIYQQQgghZ0t2+sQ1gD9WSrkG8CUAPgvgWwD89Db94wDet91+73Yf2/T3lFLKcdwlhBBCCCHk+HQfimutnwHw3wD4p9g8DH8BwGsAfq/W+mKbbQ3gbdvttwH49Lbsi23+N1u7pZQPllKel1Kef+5znzu0HYQQQh4QajYh5KmTmT7x5diM/n4VgH8TwJcC+LOHVlxr/Wit9Vmt9dkrr7xyqDlCCCEPCDWbEPLUyUyf+A8A/N+11s/VWv81gP8VwDcDeNN2OgUA3AD4zHb7MwDeDgDb9DcC+N2jek0IIYQQQsgRyTwU/1MA7y6lfMl2bvB7APwGgF8E8O3bPO8H8Int9ie3+9im/0KttR7PZUIIIYQQQo5LZk7xL2OzYO5XAPz6tsxHAXw/gO8rpbyOzZzhj22LfAzAm7fHvw/Ahx/Ab0IIIYQQQo7GdT8LUGv9AQA/YA5/CsA3Onn/AMB3HO4aIYQQQgghjwPfaEcIIYQQQmYPH4oJIYQQQsjs4UMxIYQQQgiZPXwoJoQQQgghs4cPxYQQQgghZPbwoZgQQgghhMwePhR3ePXVV1FK4R//+HdGf6+++uqppYEQQsgTIxWneM68+uqrvAETQgghhDxxOFJMCCGEEEJmDx+KR1gugVI2f9fX9z9le7k8tad9pC3Wf/13CW2x58Q7L1HaubftXNF93upjvc2+Jidieb3eu1y9v+X1+tRudsm04xLakm3HJbbtnBnt97n2NadPjLBaAbVutkvZbOtPYH/7nJG2WP81l9AWe06A++3yzs8ltO1c0dcOANzdxd8B2V6tTuMrmT2ru5uNRDQ0oZSbk/qY4WU7gPvapdpT7irOmXvtaOnzBZ6nc2XvewB0Nfvcr6OHgiPFhBBCCCFk9vChmBBCCCGEzB4+FI+wWOwm3Fxd3f+U7cXi1J72kbZY//XfJbTFnhPvvERp5962c0X6XIj6WG+zr8mJWFxt51Ki3v/cbi+uzn/+5Mt2aN9VG2T73Ntyrx3mXFz6eTpX9r4HUb+r7bn2NecUj3B7e2oPjsdTactTacclwT4nF8Tti8w81POfq5prB3Dubcm3w+O823bOjPf7PPuaI8Ud+PIO/vHv/P4YO5wQQsix4UhxB768gxBCCCHk6cOR4hG8+Kw25u1y2Y5dbNNtPhsnWLZ1eVtW7Gvbkr/XFsnv2dR+eOn6L+Nfr96ozlZbpM2236L2RH1sP6U/e+Wi85Ctz543+/mmN+23s2U/8j3yw7tmpFyvz+35al3L2q53njPnrXc9E+JwSEzcx/7rxYV9Km25pHZc2t9yeT7Xz6XGOS61nj4W3bNnz+rz589P7UafUnbbvdi32fRWjONovxXL0eZrtcXmi2xmYzNP+dR2dR9H+147LC1/s30apdvjUZ5Re5l+bOW1PkU+emmt9nh456tXX+ZY9jo8M0opr9Van53aj8fkUjS7FKBioiZnvrvZ71ziu11QU5Ld1ezR+0TGP6++RrlWW/baoX3JtqfVz9F5arUxk+5pl/W7lS/yK6PxvfaqY5P6PXvP8upunLfe9XxqIt3mSDEhhBBCCJk9fCgmhBBCCCGzhwvtRlgsdq+rLWUTf1W2JQ7rYgGs1/l0m0+27f7Nza68Lavti+1eXFhrB/Bj+4ofXrr2U7ev5V+rXtm2dd40QsNIf2pa7Yn62H5Knb1yNo+0M1uf1ONdE6UAb3zjfjtbfkW+223Zj66Z3vWj4xT3+gTY90fw8rXOW+saICTDqCZ735ERzW7pgVf/3UAbMhrnfQo9vfI0W5fz/NGfvbZIvj/8Q+ANbxhrj9fPI5qd1fqensk1IX57vmQ1215rvTZ694PFAljFXb6JU3wDQM1pKNjty7Y+FuXZSzPltp+bOMcXqNu11pP/vetd76oXwWJR62bGTK1XV7ttfUwf1/ve52Jx2nZEviwWuzQ5ro/Zdnn7rW1t89B+sjZaddu22DyLhd83kX86v70mWu2J2mf72NbvXV+9els+9nw91fV5YQB4Xs9ARx/z71I0G6j3v9PZ7+Ujfy+ARHqkpbXu9GOKZmc0z9PP4F7Xasu9djyEZkf2p2h2dD2IrcfU7I6PvWuI7Ih0mwvtRng5U11texPQdR6dZj913sek50vktz5mt1tp3nbL9kg/2XwjPkbHvPp7x7Uv1q/eZ6+PLFEdUb1RGa8u75M04UK786UUbBbaZb4nI5r0AN+LntmX6Z5vwL5WjGp2VvNa2qfyH2XBl5cWHcu2w/O7d0047UvfG6L6pmh2wsdzX9x2TnChHSGEEEIIIQF8KCaEEEIIIbOHC+1GyCw4kG3Jo9Ps56kWD3mL4bQveuFJbwFhZqGV3fYW3el8I/1kbbTqtosUbP7RhRp2AdnUBTDWr6h+7YdXh3dML6SxPrb6XJcl5EJZLICyqthfBKS3G59AJ8/xfe0SaakY0IuvRjQ7o3n2XiDpwH372YV2UbsO1ezI/hTN9tqn++MxNbvnY2axJmnCh+IRbm9P7cFxuL3dvPrGRm1YLjfRNfSXer3evFHs7m5z3FuBK2maq6v94zr/anX/1Tt3d/f3xUYW7aPUt1rtjq9W+23RZTK2tQje3e233fZH5JvNv1rdv3H12qf9sP2st6+u9n28u9uv1/pByBPjqUj2S+z3HdjXctElq3OCjXDg2fU0SuzIPULK6LJ2O8Ohmr1abdokkaFse61uSx6vfba/7D0ospVpX6vf7b7OE2m0vt+Qo8KFdnMls6ihtfhBM2VhmrdwQfxqLUrw2tFa5JBZzOCV6eXr1dlbUJGpK7PgJOrPQ9vDhXZpuNCOPAalOIsGo++63s5odyt/b5Gao3EZyT6KZve2RxbttRa5jdzHptbXOpe9Ora+cqFdHi60I4QQQgghJIAPxYQQQgghZPZwTvFckblS9icbvR8tfgDab2Kz5b1FCd6blMQvu91agaLzez8R2vZEx7wyrXyZOqfaiWxHP4GOtEefz57PXGhHyPmR0Z5o0Zmdi+ppvHzaRWzRIjVXvzu/4R9Ls3vbGQ3N1qHbaO+dx6zPHrOanb0vkEnwoXiu3N3l5ja9eHEa/7LodghTA8H3ArVn7fbsZfyM8ou9Q+ZvR/ntsSe3SomQCyc7x/bUut17TpuipdEnMKZrrTozc4ll4d3IWg7PRy9f6x6Q0X9yMJw+QQghhBBCZg8figkhhBBCyOzh9Im54r3AA/Dnn50z0g4vkDqwvy37klfiPnrz4uxx76UXvTo9e7251RrPd7EXzfvzynj1t869zkcIOQsWV2uUcgNELxjR2yf+Jb0rHy3tizRp9OUa1l6mTk9jrW5bzfXsRTZt23sv/rA+W99svtX9riZj8KF4rjy1+aL6JRVeMHn5tEHUbUB4HXjeC2Qv9dgXeUR1atu6Xu234AXLF6KXmdj69QO/tzCyF3D+5ubpXRuEPAFuX1zAIMUIVru8T6D9ko0pmq3teOWA+35o3Yw0W5ez6fb+pH2wgzSt+1XUTtFsTis+GD4Uk8tmtXqc4Ov6bUIjizQytjO+ZBb9tY6J/yOLBwkh5CHILKCboqN6P9LszEK3lh8tm1MXEGbqjBbs8W2kR4VzigkhhBBCyOzhQzEhhBBCCJk9nD5BLpvWgkGgvWCitaBDsAva7AI5u1DC1pmxLdvZRXCZBXI2X7QgL/KBEEIeAHfBYO8T8I/p49G+PRblb9WVtZmpZ2qdL2dSRAssKd/HgA/F5LLhojBCCLkYntyCQfKk4PQJQgghhBAye/hQTAghhBBCZg8figkhhBBCyOzhQ/EIy+VmMdL19eZTb9vP5XJXpnXMS7P29J8+psvZstfX+2mttnh2vfZ5dUV9EG1nbNoyvXZE9UfHl8v7/ts+9dpn+8VeB9459c6nbZfXhla/9tpp6x1Nz147LZ9a/d277kb9ISRgeb2+99U957/l9fpJtOWptOPS/pbL8/kutK6Bc6bUWtsZSvkaAD+pDv1bAP5LAD++Pb4EcAvgO2utny+lFAA/AuDbAPw+gL9Sa/2VVh3Pnj2rz58/n9iER6SU9osaDn3BwiEBw60vOn+vLZHdqE5dPuqDlq9evoxPUTui/m0dt/Vm2trya8p5bvVRq19tuV69h6R7WB+jvmtdu1GZVrs7WnUqSimv1VqfndqPx+RSNPvlpfqQmi02dbrQ+p5aXwAU1P7XLqt5et8ayeiR53NSG9PtyNY3cvwx7tG2r0buhb1+nXrfwOD106uvlUenBe1u+XIORLrdHSmutf5mrfXraq1fB+Bd2Dzo/gyADwP4+VrrVwP4+e0+AHwrgK/e/n0QwEeO0wRCCCGEEEIehtHpE+8B8E9qrSsA7wXw8e3xjwN433b7vQB+vG74JQBvKqW89SjeEkIIIYQQ8gCMxin+LgB/a7v9llrrZ7fbvwXgLdvttwH4tCqz3h77rDqGUsoHsRlJxjve8Y5BN06EfVEE4L8IoZRdFO3F4v4LH/QxL7+1p9HHdDlbFti9sKHXFs+u4L0AQpeP+sCz6eXzbNoyvXbYvvRekOH1i/Zf59cvwNDtszY9G9457bXda0OrX60PvXpH0zN9HvWdd+5tf3v9ZftyxB/yKFykZgMPr9leuq1b77c0+y7ZFtuuqE77vfHuL57Nu7u4rzKanW1H6x7ntcPWJ/mj+2fUhqjtrfPd08+oLpvXa2dWs73rTJ+XVdzley9NeekDdvuyrY9FefbSTLnt5+JqDeDydDv9UFxK+SIAfwHAX7dptdZaShmaPVJr/SiAjwKb+WkjZc8CuQhvbzez29frnZhcXQGr1W7WuxzTF7Mc02mSX9dxd7f7QqxW9wVrtbr/hRObmTbYvHpf6lks7r8kQ9q9cr6F0i+aKO96vVlI5fms2zqKtqffRqfPmW6n7uP1ev/8aF/0m+xsX11d7bdhbRYaeDa8tut0r132etC2tR19zLsebLouL9ev93IUvjBlllysZsuDBLD/UKe/X4dotvd9kDpWq/vlDtHsHrouT7cjPM0GfN1erTa65ZHVbK8/I80We3Kf1T6LP94/JFJPdL49/baa7Gm26K628RCarY/p9nt+3935dSse/6Upl/dADKC/0O5lxlLeC+BDtdY/s93/TQD/Xq31s9vpEX+31vo1pZQf227/LZsvsn0pizZQyvEXPGUWH3lltT+yLXm8fa8tPX96i5y8ctZHm7fXj72FKaPtyCz4mrK4ISrXSxtZWHKIP5m2Zvw855USZwIX2p0vpQAVie9eZvuYC5HkmMlz8EK7rA5nNLuX32v/dr8n2RUdP7KaFbV56r21l6+X1vIn0xavvKVxvs99cds5MXmhneIvYTd1AgA+CeD92+33A/iEOv7dZcO7AXyh9UBMCCGEEELIqUlNnyilfCmAPw3gP1aHfxDAT5VSPoDNTJbv3B7/WWzCsb2OTaSK7zmat4QQQgghhDwAqYfiWuu/APBmc+x3sYlGYfNWAB86infnRrSQTc9Zk/RScguaPHs2v7dgC7i/LXlkf8oCNc/XyE60aEMvJLF5td/e4hDvuGcv2w5dXuYU63K2jl5/aFteX9n2eNt2/pmtS8+Ba7UnqtMr7/Vnqw29fifkUrAaEX0/DtHsqA5P81qafehCu6wOZzS7lT/Supft6fyGH2m1p9mRZtn7ZfTZOt9Z/e5p5kNotnedRfXL58QlOGTHaPQJIsjEdn1RX9oCJG+Bg4dddLVc7ib868UcclwLmdcneiGGXVxhbWXxFmzohRiSRy+k8BZP6AUR1q7ktf1lF4XYBTT2IVg+WwsmvHK6PXZxh10Y6Nn3Fmza+nXeS7ueCfHQi5v0/jE1+1h2nCmkLp5mt3S8p9myeM5qsOhgVJ+19VJbku0A+potuh1pNnBfGzXeMU+z7SJFXVbn0zo7VbPtYj3dp5Ev0SI73aZVv7tJm/RCu4fkUhZtdCfzn0FfpsksBIgWYnjp+niU1suXqcNrR2RLH4/yZvanLOQYWZQTLcbplWvV2yozct5JFy60O19eXsYXotk9qbvXFqC/4E4KZ47LtudM5r6QWPBVilloZ+u3PuhjPb0f0bnITlZDTZuPslhvyv1DbXOhXZ5jLLQjhBBCCCHkScKHYkIIIYQQMns4p3iE1qKDS3vjVvTmvGgSf/T2Ju+4lPXe3KRpLfaI6vDa0VuwJvVmFpxFix1HF3J47Wotkmj1h+2HXr3RYqFoMWErLyEXzN5bvAr2PwGk5/A+Et2vXeu72/oO9zTbu49Zzc7cFyStN6c4asMUzW5pXq+e0bfW6W3bjhHNzmpz5v6hbaw6/U66cE7xXFku/TfkRW8ZivLIMcDf9xYLaA5d7GLb4X1aX3Q7osVqXnmBi9CIgnOKyWPwci7uVM3WPLBm2+nIzXZE7REizdYL6Hp9MBPNbvU72SfSbY4Uz5XVKrfQq5dHjgHtRR12lFj7ccx29BZMeO0QRhbSEULIY3OoZmsbsi15rP2H0myvnp4uZ9rR0u5j+ExmAecUE0IIIYSQ2cOHYkIIIYQQMns4fWKu2IV2QHsxWZRHjgH+vrdYQHPoAsXW4sfMorZo4YYtL3PTuAiNEHIC7i8ahNqOPs0xIFhoaNJ0Ho/ODLLm2uhw8WNQd7odwWLKpM9PAd6aDocPxXPlqSw6eCrtIISQBrcvLizCUcBTaQd5mnD6BCGEEEIImT18KCaEEEIIIbOHD8WEEEIIIWT28KF4hOVys9Dq+nrzt1zuH9Of9lhm29rz8mm7kr9lL9MWazeq07Yxsh/1ifUz8tt+9tqh7dm+yZ4HqUP3Z+t8Ru3ItCfqI88Hr89b7fJs6DK9Ojxb0fnNnEt7XUfXU6+f3/Sm+BogJGB5vd677M79b3m9fhJteSrtuLS/c+v3lj/nCt9oN0Ip7Rc69F72kNnu5Yv86JVptcXalW1bp9dWz75N9z5bfkcv34jaYWnZ7PVV1NdRu2ydvboz7cz2x8ixQ3z0+txeF62+9NKjMq1+PgOt8uAb7c6Xl5dq7zsVvaxi9Ps4UodzvKB2JXtv51iaLSaxy1dQ9/ZbfttjqXYcSz/1cdtRvfMwos0jfk49NuVeoNIn93vLP6DdX1E/o309n5pItzlSTAghhBBCZg8figkhhBBCyOxhnOIR9IsigN2LJ+SYfMqLHqIXSETb1p63re16L57w7PXaYu3Ktq3TtjWKFJ59oUbUVvvZa4d9IYj2zau31VdiL+qTXjta7bm726V5feT5YOvstUvStQ1dxkuP2tM7v9Intg/str6uhcz1rT+/7Mt8XwjpkdEYT7uzZUfqaL0sCADuEm0R/zz9tnmsP423Oyyu1ih3FfrFFyV8mYY5bo4trtYAGrotPv3hHwJveMO4NkX9qfvG66fWebO2PW1uaXZ0DYxotmf35Qly7hnW78z1k73/yjHpx6h81M9A358zhA/FU7m723wR7+7uC81qtX/hSH79YKTflCbHxZ7+UqxW9/MBm22d33Jzk3+xheeLbGvRsXXZh3GdrtsqffRQL9rQ9erzYR+WW6xWu0VwgtiR89lrn+S1/WTz2AfG29vNQjTdn9F57bFeb9oRlbfnzPo4cr68hwh9fcu29J+tQ/ug+8H2990dF9qR6chDVvR9Fa0VvfD0MLp+ddme9mQ0e5VoT/Tdtf90ix/a30iz7+5we3UFiOm7uwPngyZf0PGGN+z3m6ddLd2w/TxVsyXdIjqodclqlb1vjGq31WzRVF2n1Wzvnpu5flqa3cvvXfu6H2yZxSJ3PZ8ZXGg3Qim5BRoTJsgfLd0ey7TFsyV5su2y6ZHvx2bUh0w7RvP0Fov17B170caxyvRWbNhrpbfAZ7Ttj3H9HAEutDtfStkuHgPGNKuXv6f7rbyN78DQQqmM7rTa1fHxoSV7kmYfcu5adrRjp9DkrM1Oe7rXjyyc3DsRjbb3fIieGRL+nBoutCOEEEIIISSAD8WEEEIIIWT2cE7xCNECMjunWNL0tv20ee1nNLG/V16ONRZU3GuLZwu4P2eu1S7bbpun589UtL9Sp0b2ozxe+mge2wctG5695XI3D6vlb/aY9kn3fe+cZc+X7vNe30f1Rdd79jtGSILFAigr+WkXQLRATPDSese8sq289rhKb13mi6s1SrnxfX/51Uv4lmlPQX+x3CFM1exMuSh/xs5I3dG9Z4ptazPS7N49doU+mfpHnlG8hdcj/pwZfCge4aEWip2Cp9KWu7vD5pbZ/ZH5eHpfFiNEc2298rKtFymMzJ1r+fqQE7m8Prf1y/YoT+W6JGfBU7qcbl880ANqyAPWl1k/0NJhz4aXr6eR9hPIaXb2fjLq66E4krxHz88XL8brbH3Jev6cIZw+QQghhBBCZg8figkhhBBCyOzh9Aly2XgBz3vzZO1LLVqB4j17XjlvDjiQm8PrzduOytg6dIxK699DoeeQAb5vrReuEEJmy97c6Gh+dW9etN6PjntlvHJ7n45Nb7tV50G+HsbQnPTQz+NxiUtB+FBMLpunNGnwUmCfE0Im8vhzownAfs/C6RMdXn31VZRS+Mc//p3R36uvvnpqaSCEEPLE4Ehxh1dffZU3YEIIIYSQJw5HigkhhBBCyOzhQzEhhBBCCJk9fCgmhBBCCCGzhw/FEcvlJrxUKcD19eZP73ufy6VfVo7rNM/eqP0oX6t8VHbKscifyIdse+x2y+9R/6f2Zev86nPX8t1Lj879aN+PXAet63L0XGfOx0j/22slOv/ad0IALK/X976K/OMf/87jb3m9PrVEpCj1IV8Hm+TZs2f1+fPnp3Zjn1L2X/sI+K+BzLwyslXW1jVif8qn+NLyY/RYNk9Ud69PMmWn+jp6LnvnrnfteOm9tEx7jnVd9nyY0l/H8C06/2dAKeW1WuuzU/vxmJyjZu9dEg+h2Z4dW/mxrnO7730Her4dU7Nteqtvo7zH0ohePZm+0Hkz5zij0zptxJfs9de6Lqf036H3zmz/b4+fkWQDiHWbI8WEEEIIIWT28KGYEEIIIYTMHj4UE0IIIYSQ2cOHYkIIIYQQMnv4RruIxWI3SfzqavOp90u5/7lY+GVvbu7blTLanpC1H+XrlffSvHp7xyJ/ojbYujPlJT3ye9T/qX2p/bDnV5+7lu9eenTuR/u+1dbIjnddRj5M6a9WW3rnXveZd/7lGCGah9Rsz44wqjORj3Zf/FgsgPV67Dt2TM32bLT6RNdj/R71P6t5LV91GSHS3YzO9zQ860vUJ1H/RtelZyur2S0/pva/d/5xRqvsGnCkuIWc6Lu7/WN3d7tPYHOhLhbAarU5+ev1riywOS6hpNZr3/5isfnz7Gs79qKT8pJ/sdj/4tjyUlaX0b7a9kvbxBddD7Bpl9jVbdLlpIytS/pF+sSmLxa75aq6Dg/d7labJE2+qKtV+5xqWy+/3Lj/UCZ5FgvgxYtd+nq9a9/V1X4dtq26Pu/c6L6/vd2EJZN+8a4Xi7RJ0NclsOlr22ZB91fLti4rx7z+FXv63Os+0P2zWu3nI6SF/S54340pmq3T9fXaqsdqtv1uiZ1a93VbbK1Wu/CDWc229wGr2fY+JHlubu77EWmB6KZFa7boRctfq0ka6ctIs21ez5an2VJW29BltTbZerQ26/Mmx3v0NNses/dzfV2uVvevQemLq6td3p5me8c836xm6/z2+6brvjDN5khxhFxIU0K86P2oXFTGltN1Czav9S+qe8Q3+Wz1g+eDth21JSpj7Uvdq1UuTExkZyTkTbafemnic9Qnvf5q9aOtQ7ch20cZv6P+i+qI/LTHs33unSNdPyGWQ0JI9cpnv7+6HiH73bN5sr61/Ml+/63eZu4PPRut9mf7JaPvo7ayZXvH9HnL+Ood99pziN+t/hnV7JZvmT7fuwZwEXCkmBBCCCGEzB4+FBNCCCGEkNnDh2JCCCGEEDJ7UnOKSylvAvA3AfxJABXAfwTgNwH8JIAlgFsA31lr/XwppQD4EQDfBuD3AfyVWuuvHN3zh2Zv1STaKze91biy31pRq8vYSf6t1a02r87TizCRieTgrXaN6rE+SF6vHpsn2rd1ZyIs9FZEZ1f3Zvopk6ZX/Oq8UdutzWjbq0O3IdtHh6xU9upoRYhoXdOtPvfOkcDoE8Sjd+229KJXPtLskeu1993zvisj0XckX+v72NP6qdGNIs2O+rfXNntOvf49hma3yraO2fPW8zW6R3ntOcTvqH+maHbLt16f2/N4IdEnsgvtfgTA36m1fnsp5YsAfAmAvw8Q8coAACAASURBVAHg52utP1hK+TCADwP4fgDfCuCrt3/fBOAj28/L4vb21B4Q4RLPxWP4/BB1XGJfEyLUy7jxupzLd+8YfpxLWx6DObX1EEo/yznQnT5RSnkjgD8F4GMAUGv9V7XW3wPwXgAf32b7OID3bbffC+DH64ZfAvCmUspbj+75Q7Ncbv7LkfAny+XumPxJmv60+bzyXrnM53K5802OZctFdbfSdHqmvkyatIEQQo5NpJdWn2Rba1JL86dq9qiGtuq2GhrZzPg9ot/UbDIjMiPFXwXgcwD+h1LK1wJ4DcD3AnhLrfWz2zy/BeAt2+23Afi0Kr/eHvusOoZSygcBfBAA3vGOd0z1/+HwwtIA/bBUvRBmU0PdRKF7Ij9GQ3H1Qq5k6suk9eINE0LOkrPXbCDWyygMlqTrfMfUbF2P2NJ+jGi2lNP+RjZ7fns+RH5Rs8mMyCy0uwbwDQA+Umv9egD/ApupEi+ptVYMThiptX601vqs1vrslVdeGSlKCCHkkaFmE0KeOpmH4jWAda31l7f7P43NQ/I/k2kR28/f3qZ/BsDbVfmb7TFCCCGEEELOku5Dca31twB8upTyNdtD7wHwGwA+CeD922PvB/CJ7fYnAXx32fBuAF9Q0ywIIYQQQgg5O7LRJ/5TAP/TNvLEpwB8DzYP1D9VSvkAgBWA79zm/VlswrG9jk1Itu85qsePhQ1LY8OpANNCmNn8I59e6J7Ij8j3qSFfsiFaemn6Pe6EEHIkFldrlHIDoGKz0t1+OtuCm7+201KfXj3J8vfqdo5HNrt+Z4/ZOgiZxqVE0Uw9FNdafxXAMyfpPU7eCuBDB/p1es45zMo5+0YIISfg9gX/4SaEHAbfaEcIIYQQQmYPH4oJIYQQQsjs4UMxIYQQQgiZPXwoJoQQQgghs4cPxYQQQgghZPbwoZgQQgghhMwePhQTQgghhJDZw4diQgghhBAye/hQTAghhBBCZg8figkhhBBCyOzhQzEhhBBCCJk9fCiOWC6BUoDra/9Tby+X9/Nnyuh6oj9tu2dTtqM0a8v6ELW11abIxkhf9Nrk9W+UV/uUOb9en0Xl7bmy9Y603TsfGRuRvUybCXnCLK/XTSnl3/n/tWSM5/fy/1rn91y4PrUDZ8tqBdS6OZPeJ7B/DGinR2WkHo21J9s9m7ZsJi3T1labIhte3RlbPb97eXX+Frrfo35plYnqzbbdqzdjI7KXaTMhT5jV3c3uq5TR5Fa6ty35bZrQ07Oo/KjejtyPMja8PFP6InOf6PRfWZljinvnN2qLd06i/UPbbo9N1e+ZcAm3KY4UE0IIIYSQ2cOHYkIIIYQQMnv4UEwIIYQQQmYPH4oJIYQQQsjs4UK7iMViMyv86sr/BHbbi8VmP0pvlZF6PKztnk3Z9vJLXTrN+hC1tdWmyIbnV89WVNbr3yivzt9C97vts6i8PVe23pG223ZlbUT2Mm0m5KnjfaenaLa3fXOzq8fT7Za2t3S5pYPWVkZvo7p6/o30Rcu+53er/bpPV2jTa5Olp9mjbfc0W/s2ot/U7LOED8VZ7u42F/Ht7SauyEp9e9frTbrm6io+dnW1KX99vdv3PrVtOSbCLPXrvCI8el+XXa12ebQPuo2S12urtifodgD3/cv0hVefboPtA1uP12fL5cZeRCvtmGUIIach0hWtKRqrpVZXVqv9BzxbRliv7z9Max9a9dg0q6/rdXzfkPaKr9a+YG3aMhbPN6+s57c9Fzqv19Yeo9EaqNlkED4UR0Qhxry0XrgVeyz7GZUFHj5Ujm2rl3ZI+LTIDy/smbXXaqtsZwSWEPK0yOpYlDcbJk3KZ7RftKgVtnLkXmLrzt4DvOMj7cnU16uz5UOtgDMAS8hjwjnFhJwDy+XmxrBcntoTQgghPajZTxI+FJN5IoImf5m302XfFJgpY9+iJ6NIq9X9t/dZ+3q/9Wa/3lsEe28lzL59MOOPbUPGdu/tjISQeXFqzdafWrOBjb1IU0c1O6OHPZtZzR7N0+qrXp0XAKdPkHky5e103jHZBuKfELNTXaxv2oaXt2W/54O1EdV7jJ+To/Zmp9S07BFC5sOpNdvTJ0EejiNN1Wk9zW7V9xDTXkbyZPK30s8cjhQTQgghhJDZw5HiCC/kTS+EGTAWeuyQEGg2Xba9tJ4vXple2J7Ih2z4tEyYINumVmicVvikS0H6gYsECZlGVse8vNkwab1wb5EW9cKoZe3ZujO6OKrZLd96IcqiOnuaXQoANWp6CehrjDwJ+FAccexwXpfKY7d1Tn1rubCfmQg5O2rwUHUOunIOPpw7lyZ90fVGLhZOn4hYLjcPJ/LXWiS1XOYmvHtle3amLELSeZbL++3x8uq6W75n2mvrzLQl8iFbNmo3IWQ+HEuzW1rb08LH0uyRe0+mvSNt0b72NLvVdvtJyInhSHFEayGWfHmnxq20ZTMT1b3Plh3JY2NktvJ69np5W+nRwq2RNmX7I2p3xJS304285SjzliRdRv8EZ39mjd5E2LLf80HXP2Va0MjPr14fZX+mjcrwJ0vicSzN1tu2bJT22JqdaUdWsyNbPb3Vi8t6/nhtv/eJNqfWbE+XhcVi9+KW6E2EWc326mtNRzl02stInlZfWX+svc5t+RzgQzEh8mYl/YYlTeuNU/ZNgcDYG6Jkv/WWKe9NhD2/vTTvjVKyv1zut9frA7uv7Ymt6O2L+g1io/0d5SOEzBNPhzWRZkv+lmZHZTwb8uAnenhzs9Ns0XNPs2Xba5N8Rm8B9PD6IqvZGq3Z9r7T0mxdh7abKXNm8KGYzJPoTX29UDnZUR9J643eHBoyJzNqPvorhW2zZrSdU/oha4MQMi96o+w9zZ6qt9m8LRue35n8rXbrNJvuHRvR7Cl92LNxAXBOMSGEEEIImT18KCaEEEIIIbOH0yciWguxpk5498r2FixNWYSk89gYmVHezMR7L28rPVq4FX16PmT7I2o3IWQ+HEuz9bYtq9NOqdmZdmQ1O7LV01vxtaXZXv+Hds1P/4Q8MhwpbmEnjMsxmQQvi4BkUrrksxPY7YIhmfC+WNxf8ATsH28tINBlb24283ZsWZnwH02sF3Tdth7tu7R3vW733XK5q1PaID7aB1Zt1y7a0n0u9Cbu39wwJighcyVawKT1cr1ua7ZFNFs0VtDRF6J/xCPNtvcBbVM0W/sf+Xh7e/9eYttvtdWyWOw0s6XZto5Is239Pf3Wi8AIOSEcKY7w3mPemwQvZCfM92y2JuhHk9htGDTPnykT/61P2Yn2nv2WjxmfPB+9+ltEb+rzRlC8PKMj/pGNaBRmZOQ+k9ay6aV5fTDazkNCBnn97NVNiGZEo1qLsrxjWr+kLsknx7M6e8wFta26s/qZbVvPx5a/2ftPC0+7PP3wjh+it5lR7xHNbtk5F80e7cPevfFCQrJxpJgQwY6CeKMqXrgbO5Ijo1ASoqdVVterR8b1aPxiAbx44Y8GRaPm2qaMcIlvupyMVLVGkuSYbaf+hcTrC0nX/aDr1W3ybNi+yozoEULmjfdLrXxqfYo0246MW9veiP+oZnu2RzRb0rKaLdueZmt0P3i/WN/c+Jpt23HBWs2RYjJPpr7MRJMZEdE2pIwcGxlp0T5nfWyNAmVsZEZ6en0RtTk7AtU7P4SQ+fBQOtXSJa9cxgdt5yE0+5BfhbP+t7Q8stHq6wuAI8WEEEIIIWT28KGYEEIIIYTMHk6fiNDvMRd6E+6FzOT8aML7SKggvR2FQfP8iWy1Jtlbn1q2em1r+ehtW588H736CSHzYmQxkNbtTF6tX/re0AtBqW3Z7UMX1GbCX1r7Xn3ZtvV8bPmbuf+s/NNKyGPBh+KISw3pdQl+X4KPhJDLw84hfQgi/Tqlrh2r7lO37TKmnZInDKdPRCyXm1iRpWz+7PZyucun06+vd39euVLaZaP6vM/l8r6NyJ6Xt2Wvlc9uZ9ppffHan2lTq6xXR4SMfOiRCzua4R3Xf72REFnBq8sIUb2RHe1z1kevfM/Hlh/2M9MXto6oTS37nk/8NYB49PQq0kFvu6U7rbwtH3r6FunkY2l21EdR+yM/s2VHNFvO70PoVEuXIpun1GzPdq8PMrrq3RciLc/avzDNLvUx/rPu8OzZs/r8+fNTu7FPKZvPXmxJOa4/hezKUPvp1deLbtDKn/G5tTK251/G3xG7U/ou8oGQB6aU8lqt9dmp/XhMzlGzX37lW9oQ6aA+Jtu23JSV+F5+W38rf8vnh9Bs266MvUN8MccKaijblPTL55zOYaTbHCkm82S53P1HmxldyYzSj44a2e3lct+3jB+tctnyrREv3c7RvtJ+tdreG/WK/CKEzIuehhxbs3t5rfZmNXKK1h96P8j0VbafMu24UM2+PrUDhJyEVnxc+WyllbJJnzLS30rXvmX8kDSv3Gj53qjRlL4SvzL2W35E5Qgh8+EQjZqi2Z49ewyYdj/R5bL3nEPuB5m+snky6VFd0Xk6czhSTAghhBBCZk/qobiUcltK+fVSyq+WUp5vj31FKeXnSin/ePv55dvjpZTyo6WU10spv1ZK+YaHbAAhhBBCCCGHMjJ94t+vtf6O2v8wgJ+vtf5gKeXD2/3vB/CtAL56+/dNAD6y/bwsbLxGG1tR3o3urS4V9HFtY2pcy2g1Z1Q2ioeZsdeq1/oX+Tsa1zPyM9N31ocLWelKCDkiPb2KdFAfk22vnI3X6+XV29F9Y0SztX4+tGbrer16vHjFkc5nyjJOMTkzDpk+8V4AH99ufxzA+9TxH68bfgnAm0opbz2gntNxd7f5vLrabOsH3tVq82Ver/fL3NwAL17sxM+W02VXq51tQfLqcpIux+7uNgLSix2pbaxW933VbRN7t7ebOUCLxb5fLdu2fcJ6vZlov1rt6tDzi9brzbEXL3bCqf30+kBEV+apSh7dj5E/hJCnj9ZJT3utDt7dbfS61s2n1Z2rq/uaLWmCrc9qNrCxrTXb889qtuin5iE1++pq0z9Sp/QLsH+/07qt/fT6IGqHrbeVTsgjkR0prgD+91JKBfBjtdaPAnhLrfWz2/TfAvCW7fbbAHxalV1vj31WHUMp5YMAPggA73jHO6Z5/5AcshCrV16TmQTv5c9M1I+2rQ29nWl/5Eu0mKC1AECE0KvP64OoT2w9PYFtjVpMHaXPjvRH6VPfSjjyy0OUpvsxGjUa7aveCJKtO6pL4C8BJ+XsNRs4KBwYgL6e2v1WPTZ/9t5w7po9cv/x+qzVb6ZJ9+hpiE47hmZ79vSxzCj4sbT+0PtBpq90+cz9olWX7ccL+SUgO1L879ZavwGbqREfKqX8KZ1YN8GOq1syoNb60Vrrs1rrs1deeWWkKCEPgzdyA+xGRWrdHx2xD94yqiIjNq1RIA89oqJHvu1outjyRt51PTqP/uUi23YZvYlGqrw+0qNVdgQpKqtH46SuaKSPnAxqNrkYDtXsrH7bUXCtu1oP9a8FLc3Weaxmt34F0NhfU8Uvz4atS9qk/fLKRb+gPAHNTo0U11o/s/387VLKzwD4RgD/rJTy1lrrZ7fTI357m/0zAN6uit9sjxFyPrRGpoH2qLMuq8u3RlWmjgxpn7zQZl59LZ+sz16dvVEv60O2ntZ+5rjXJ4SQ+fBYmm3zTNFvz+6hmi3tbGm1p6W9vuvdH7zyT1SzuyPFpZQvLaX8cdkG8GcA/AMAnwTw/m229wP4xHb7kwC+exuF4t0AvqCmWRBCCCGEEHJ2ZEaK3wLgZ8rmKf8awP9ca/07pZS/B+CnSikfwGamyHdu8/8sgG8D8DqA3wfwPUf3mhBCCCGEkCPSfSiutX4KwNc6x38XwHuc4xXAh47i3SmZuhArG35GyC54svmzocl6k/OlPrtwKeO/Z6d13PPZhrZr+d/qE1u/niNFCJkHUzQbyOup3W/VY/Nn7w3nrtkj9x+vz5r9NrQ0iZCjw9c8R/TCnT10+UPqWS5z4W30pHgdB7M3WV7y2AVdQByWRyblX1/vT9638Ytvbzf+r9f36/AWaeljsm1DHxFC5oGdAz/KY+hGVEdLt70Qb6KnOt3TSMHqqKfZNmybLKSyei66LQt5I8326o0W215IdALytOFDcYQIVOsL7W17D3bA/nEbXSAjIFk/ZJXtyCKu7MT9XhowfXK+LuMtKGuFJYp8adEamda+eKPOuqwuPyU0mld/axQr+xKU1uhUr87eqJftn2w9rf3Mca9PCNHYf+wjXdXHgP0HRE/LvagokX4D7X1bXurJLL6KFuGOanUvLFtm8ZZdFObZGrGb0e3H0mydZ4p+HzLCHmmp7oPMLyG6rVHfZXXbKz9Fsy/gn55SD/3P+gg8e/asPn/+/NRu7DPysBjlB6YJQ6verLBc6kNxtu8kvdc+Qh6YUsprtdZnp/bjMTlHzX75tc9o2F4B+JpyqJ1o/1B9PbeH4patEbuloKCG0k1Zv3zO6RxGun3IG+0IuVyWy803tJTNT5D6M7st+zZ9ubxfx0g98mnteHmXy/ttifKNtFvnz/qr07RfmeOjdYh/hJD5MKoXkWZHmnJszR7R9Z6OZvzx9PVUmm3vRxei2Zw+QebJlLdK2W3ZB/bTvbf0jdSjP7WdKK+t3+57tnr+6Pyevcjf1shb7/joLyeEkHkxqhc23csTaWnGXk+zM3myOpr1R/L39P2hNdsrewFwpJgQQgghhMwejhQ/RXrhzUYXWWXS9PvS5bjdt2Wtf9GCsqhszxdCCLkUMuHNvHQ5rvNndNzT7F5ZwVsU5mn2iN0LWYhFnjZ8KI6IIgiMrDodjRSgt7OftrysZL5kLt1/QshpyD4Q6mPA/gNiKwpA5j4gdqJ9W17quWTdO5bvl/ELO3nCcPpEFi+MjveCCHkQlnBsIooSU3K9btdzc7OxHcVztOnaN13/yET53oIrKZs5nlng0PKp5+tyuWnnchm3RfIQQuaHF3NXQqy9eHH/l6Srq40+izZLOdFsCdPm6TGw02Rdv9iVuOm13s9jNTvS18wipqxmA7l7QaTZUv7Ymq3zEXJCOFIc0VrcJNtAbkHQ6IT1Vp5M+mi9UZle2J5W+227RtqZaX8rlnHmxSVT3iplt2Vf+61jZEaxH7O/EozEtNTH7b5nq+fPoT+LeiNvreOjv5xwigzx6OkRkF/g1FqEdWzNGtV5W29UNvKj114h6j9r/xjt11oZMaoXNt3LE2lpxl5PszN5sjqa8cfT11NptmxL2QuZHsORYkKiUXmb3sqjkbdAATtBsWXtKH/WtrW1Xu9GsrS/Op8d9RK/ZNTM+9VBt2H0ldlXVzu/WvTenKjz2M/ViiNLhMyV6K2p3pv/suiH9sVimmZbXfLy6F+TtT3vFw79C0ZPs3UbJE8W+2tJRFRvlEfvZwarzgCOFJN5EoVLa+0D+8e849kRJFvGs9Eb2bJ5p/x6MPUXkZb9bJ94NqK+b9VDCJkHo5rdGv3u6aKuL6vZNu1YWp0d6c/41vqVpNUnGXu9Pr4AyeZIMSGEEEIImT18KCaEEEIIIbOH0ycieqHTsjEeo0nnIxPldZ5e+sgE+p6vtr7M8cwCh5ZP2fa3YhmPzoElhDwNWno0ssDJS/Psa7uHaNaozsu251NPs1v3toxmR+FGD23/hSzEIk8bPhRPRUKuWbxJ5iIKdkGUIA9xMtHd2rULjOwCA2vDTtq3Zex2VJ93zE6a122zNr0FBF5feAsFbDm7YMza0nnX681ih0uO+0kIOS76gUzjLRjy9BTYbev0SNclvyxatbrq1RHpvKfZmcVk9t4hD5+Lxf7CJ0+rBYl9v1zutFdsi/+6LmtTt//mZmfL3qsuZCEWedrwoTgiM5k9E27M7vcmso9Mitdk6hitT+e17YnaGvWT/ozq9urI2rLpPYGNwqW19m1/eMezI0i2jGejN7Kl846mZV8mE406texn+8SzEfW9Vw/DshHLlAWcsm3Ly36kQZn6vDy6/l4dWX8i3zJ+2f6weUVLM4vPehqvbU1ZiDWq2VZbvOOZ0fCsZmsbx9LqkZF+W79Na2l2r08y9rp9bK65M4QPxWSeXNIo8kP6ekn9QAiZN/ZB/qE4Z108Z9969P7pOQO40I7Mk+Vy859r6w1MrXzRW6akrC3XyhP5kX07oN3O+J851muL9SlqW8//1n7vbYuEkPnQ0r6Wrk7RbE9rDtHsrJ6NanbGRsuvh9Jsb/sCuAwvCTk2vbc69fIB7ekkvdjCOo9nY/SnX+8nykw7Mz/ZZt4Apv2ybev539rv+UoImQ+ZaYGerk7RbK/8IZrdy5f1IzN1JhPbXvtk2xblG9HsqJ/PHI4UE0IIIYSQ2cOR4ojeZPbWJH87Cb/3PvLRifm27kwdkX+9SfNi1wulY+uN+kLbHFl81vOvtWiLYdkImR8Z/RBGNVuXyS4Y9fIIWd23tnr3pJH7jSaqV7Q0s/hM2+zZcv07/4VY5GnDh+KIS57MTgghc+SxFmLNkWPeEyNbl/ELO3nCcPpExHIZL/YZmRgv+Vp5souMdH1TJ+pbu9mJ/1F7vP3Wogavzsg3255e/3EBFiHzJtKfET0+RLMjPbK2WnqaWQjbWzj20Jod+TRVs6nb5EzgSHGEjslYSvwpeTMTzDOT573jXn1iz+bP+tOLJ9mrb3S/1RdR3TamZasdXj+1aE2P0fFvo3y6LZn4v6081saUn371tp46kn171Ugs48w0nilxoL39TNxOQoRoUVW0f2zN9vJkNM/616s3o7HZPmj5MNIXh2i29TuipYstXZ2i2TrfMTS7l2/0jYMjNlp+PZRmez467zs7N/hQTOZN9NY8+9Bl3/Rk33Aln96cO1tev5UPaL8V0PoWvWlQv3XK2re2on6wNmxborLavvf6V/vGQftWw+jth626xFdCyLywuibbq1U82uy9/S/SbE+DRjQ7suMd03Yyfnj9oG1oP1v3C/2ZeUtsT7M9bF03NxfxGm8+FJN50hvJkG1gfAQnO/oS5evVFdWr25Xx0Rvt6tno2bPtjfpPjvdG4Xp1EULmQ2Z0Xuc79JeAKZrt2WkdG/HDarbny6iOZn6B6PmZreMC4JxiQgghhBAye/hQTAghhBBCZg+nT0TomIzRop/MxHhJB3KT573jXn1e3OCsP97ig96kfW+uaLTfWtTg1Rn5ZmNattrh9RMhZF5YXQPu7/f0OJMns/DV6lGkedmYx7a8V0bbe0jNjnyaqtlSZgVCTgofiltEi3/ki68XZC0W8SR0mYxvJ+d7eBPh9UT25XJXPloM4PlvJ/vbRVVTFhfohQre4ocov9dOOaZ99xYe2DZl+pQQMh/0oqpI7xaLzTxH2bd6pDVbl/OItNrW6/mo9dMrr+0D+/cenabLaT2ONFjabO3b9nqa7bVZ27ULvajZ5ILgQ3FE793hwFiIn6jcMRZuZfJoH7yFWT1/Rv3v+TLqe6aOkQn9vZEMyaPryI7gZEdfonp6dXnpI+F8otGuKb80tEavov4bfZMXfw0gWUYWotr9kcWno5p96ALYQxavevm9+rP+ZtqcvS94x1t4Gqm3I83xtiXfiJ5nNNuzEx079ttss3X3dHVUszN1XMgvAXwoJkTwRsr16HWLaHTfhqR56DBix7DPUGeEkHMmGsHOaHZr9NvyGJo9JxL/95waPhSTedIbIRdGR0CeQEgaQgg5S3q/HLbSpoxgU7NnB6NPEEIIIYSQ2cOHYkIIIYQQMns4fSKi9e7wVnibaBK6V2504Va2vJdHiBZm9fwZ9b+Vf4rvmTq4CIuQeRPpi9W7aFFpdvHpqGYfsgA2q6ly/BANPnSxbWQ3s7jsQhZikacNH4oj5jK5fi7tJIQ8feyaAIvVu6z+HaqTl6azp/KXU3jJieH0iYjlcvMf7PX15jOzbT+Xy31bXlnv7/raz2OP621bV8Y/z4ZXfqTNvbRsn2T3vXqljhYyEuKNdug/fczL10pnGDFCHpdRzbZacqg+TdGpzL0mo/stDR7V7JafD6HZCd1eXK135lD97VaabLfSbR5rm38H/V3CLZAjxRHZOI162/vUtgRbFmjHM46Oe3ZG4mG2fLFtmGIzSuv1ych+VG+PSxu5IYT0GdVsvV3K7oUSU/Vpik5l7jXStihN/M5EWsiktfwc7ZNMX+i2BNy+uGmmE3IMOFJMiGa53P1bG42i27TWCElrlKdlxxu1ikbeW36NjBhN/eWh1Rbv1xPtb68Pev4RQmaBlQ3+Xd7fJcg2R4oJ0Xjxi2UbmBYnMyrXslPK/VErSR/xS9K8eqJPXTbzy0OrLd6n9rfXBz3/CCGzwA5Uv6SlH8D02MaRznl57DGpd4ruejqZaVfPXq/to5rdamPgZ1l5J/C84EgxIYQQQgiZPXwoJoQQQgghs4fTJyKycRr1di82pmBjSwL30+0x7/ih8TAjGzaPd2wkTmXLz6hPsvtevZewxJUQcnxGNVtvX10BN9vFXFP1aYpOZe410jabpm2s17EGe8eysYUfQ7PF/gqEnJT0SHEp5aqU8vdLKX97u/9VpZRfLqW8Xkr5yVLKF22Pf/F2//Vt+vJhXD8hV1fA3d39bc1isYtwcHu7/6Cmy0reWnd5tD3J02O93ixEknmoNzf7NsWu9nexAF682OVZrTZ/tn26/GKx+bNtFtseOk0Lt9Rr+9D6WOvuZuX5JPuyvVpdxox+Qsjx8bQo0mzZthrU02ythbpOTzcjP0QL1+vNvtZs24aWZosN7z4kfloNtf3U0m+t2eLfFM3WfZSpl5ATMDJS/L0A/hGAf2O7/0MAfrjW+hOllP8ewAcAfGT7+fla6ztLKd+1zfcXj+jz45BdYBSlr1a+PY2dyO4t8hpZOGDTxGavTC9kXDTBvtX+zCIDsRNN/I98zPowBe8Nf3Jc1x2NdnhpUbmWHW/UqjXy3hthyowYTf3lodWW6NeTbB/0/CNEM6JFUT6hpdmy7dXZqj9Td2Zh7Uh4tMhutj2efzpfpoXbVQAAIABJREFUr77s+TgnzY50x+Zvjaz30nva2vLdS5vyltpWe1uaLbRsWz8v5JeA1EhxKeUGwJ8D8De3+wXAtwD46W2WjwN433b7vdt9bNPfs81PyPmwXG6+rKXsh/wSwZcRDRkNklF0O1JuRz4sNze7h9sobzRqokeBxJYdgVmvd6P7HtIGPbqk0/QIjzeir385uLraz39z44/2e6NCenRM+lGPirXwRpgIIfPjmJot2xbRtUh3olFvYbXaabb366r9hcDj7m5fX710y2q16w/PX2BjrzWC75URzZbyIwMSF6jZ2ZHi/xbAXwPwx7f7bwbwe7XWF9v9NYC3bbffBuDTAFBrfVFK+cI2/+8cxWNCjkFmVD4KjWP3s6MvrZGfEV+mjsD0RooO6YvMKFmvLV7fZUb0CCHz4Jia3bIj+bMj/IeOwGfvHTZPrz1T71+t/tJpUf0tX86c7khxKeXPA/jtWutrx6y4lPLBUsrzUsrzz33uc8c0TQgh5MhQswkhT53M9IlvBvAXSim3AH4Cm2kTPwLgTaUUGWm+AfCZ7fZnALwdALbpbwTwu9ZorfWjtdZntdZnr7zyykGNIIQQ8rBQswkhT53uQ3Gt9a/XWm9qrUsA3wXgF2qt/yGAXwTw7dts7wfwie32J7f72Kb/Qq12tcIFYCfBexPQW+l23o3Y03/RBPlWXa269b62GZXx6m21T8p4fno+t9K0HVtHz8deP+hyhJD5MKJFkS4JLc2eqoWZuqOFtVnN7pXJ3kNa/k3R7F4/ULPJGXBInOLvB/ATpZT/CsDfB/Cx7fGPAfgfSymvA/jn2DxIXx4STu0x7R27zlPWe2wuwUdCyGk55vjLqTTnoTT7WO3J2plS32VMOyVPmKE32tVa/26t9c9vtz9Va/3GWus7a63fUWv9l9vjf7Ddf+c2/VMP4fiDs1xu/nuVVa56237q7eUyZ0O2W2nyOWJT8ko+L0+vjG2LtpXpg1Z/9Hxt9WPkp7ety3tkRuW90aBodKg1Gt8a+emNzkQjNVNGYHojRT073q8RmV9Usm1p+dYa0SNEaOnoiG5mdLmXlrXp6aDNk/W3paktX1v2s5rtpY/2a+b8ZrQlo9mRnYxmZ3WuVf+Ue4fdzvg24otXplW+5duFanY5h5kNz549q8+fPz+1G/uUMn1V/ZTIBF5atHq0Z9Om2zy9MiPtHPFjpC0j/vX6lZAHpJTyWq312an9eEzOUbNfSsAx9OqQfFP06jE1u3fMpgPT7z+DfWCb7J5fcrGc0zmMdHtopJiQJ8NyufmG9kZ1ep9SVsrZ/eyIemTPqzfyVR/P/CLRG+XSbRoZZerlb7VlxJ6UJYTMg1GtjDTQ6o5nZ0S/Pe3t6VxLy1ttifyIbGU1O9N30f0ha+8COGROMSGXi/fWqOh4ZsQDaMeXtPmzI+pevRJIvdWGyJ9WnZ6tEZ+njGrZtnjlW/YIIfNhVCt7Gq01NLI/MqKutbencz0tjz4jPyNbmXYII/cj3Z9Zexcg2RwpJoQQQgghs4cPxYQQQgghZPZw+kSEXfUJxKtGdZoX59LLJ9tAnCZl9bvKezbtqn5B5+mViSIEZPug1R/y3vfI11Y/Rn56des+I4TMgymaDYxpS89eL+pLr27hoTS7d8zaz2q2lz7arziTVVhktvChuMfd3eYLe3eXy68FBNg9+HrIg5vMBdJldH3aptiTdHkAtDEhjxnH8qHjW/bsH6NthJB5MKrZV1cbDdYaG9kQrV0u7+t2ZNPT7Lu7zYOi1jJq9kXMOSVPGz4UR9gJ8VPC3shxIJ6k76Vl8mj7LXEmPr0Rll4sSm90xNvPjqhH9rx65Z+pVhu8f6RGRo5sm0ZGmXr5W23xykf2LiTuJXlEjqXZvQVbvcVMni+efTLGqFZGGhiNaI+OqLe0t6dzLS1vtSXyM7KVaYfQ67vo/pC2d/6/BPChmMwbGRFarTZhY2Q0x47i65EjPYIkZWX77m4jil5+EZLVanfM5tPpuh7tj4juYrG5ucqolfcgbNvpjZ557ZI2AL7ftqzXXvnUvts03e9Z/7xRNkLIPLD6IVrX0mytr94ofqTZ1mbrHnB9vavX+uNNQVmtcvqniTRW7jteHaN29AO215e6v7Qtz769/13A+B0fisk8af0SIMeA9giSPuale59iszfC5I022bKZUausX73ykd/Wx2y7eyN2mXRCyLzIhEXLhob0jrW0K8ov6RnNtscP8Sur2SP2JG8UWm5En6Pzc+Yw+gQhhBBCCJk9fCgmhBBCCCGzh9MnIjJhyHoT2eU44E9ujxYztRYIePYZfowQMnMWV2uUIlpYsYlkEH1Gx0bLBGmAc7xXL+HaWXJq+FAcwUU8hBByMdy+4OAAIeQw+FBM5kkvuH0r5Ew2vForzFlUTqfr+r2yh75wIPPLR1S35NE+jrR7StgjzydCyHxo6UMvTFhGe1raFeWX9Ixm2+NT/RrR7BF7kld+fc7UlU1n9AlCzpin9EvAU2oLIYRE6MgS5PK4gOlCXGg3wnK5+Y/n+nrzWcrmWJQm25k819f+cS9fZN+rw+Zv2Wn57OXt5dH1eXmjNreO2bb2+qN3Lnv19T6jPuud++XyONdMrw7rj1d/5nrxzn2mfKt9UdtGbBCyxcoQ/y7vr/uVthoQbffudZGWtfRRnFsuj6/ZXt1Znc/cv6I2Zcpl68/YuABKPYP/vJ49e1afP39+ajf6lJKPJ9hKi/IAubcgTYlvKGlSh1emV76X1/PV1hf5c2h7Wv54jLZ/9DxH7W7lydQ/xcfIn0zfe3mmnjt7LGr/FLsnpJTyWq312an9eEzOUbPDy2GKhrTKZb6jXvms/Sht1E+vY46h2aP6FNl3fCuo4Vd6z+VMn9p2tHzz2mz91Hm8Nhyi2Z7v0XmYUo9tQ6vfjnG/CWyciWQDiHWbI8WEEEIIIWT28KGYEEIIIYTMHj4UE0IIIYSQ2cOHYkIIIYQQMnsYkm0ELx5sJlZsJk/mzXU9+14dUVzZTExBbc/64bXd81Xq8/L24tx6x2xb1+t2f0R4cYq9+kbiQ+o2il/Rue/FLM5eM5k6tD/Z+Jut8yM2R8+d175snb3rnpCI6HshaZ6GSJpXPqMJ2lYvlmzmO976Lj62Zmf1MKq31d8rtGm1P2pH1reePkrs3t59Z4pme3Yize7d46O0SLN75bL1p2Ion8kquxa11pP/vetd76pnx2JR62ah5O7v6mr3Kdv6uN5eLHZ2enmjz17+xaLtZ1SutR3ZjOzb/C1/W33n9ZvXlp4/1k7v/PbKts6f9TV7zrz6M9dA7zrL5O9dC63+H8mXOedT/B85x48EgOe1nl5HH/PvHDV7czfrXO+975XWs5HvT0Zbr65ymjHy/T8HzW71m7ff+D4DifOb0ZJTaHbPXusaGNXqKRo6qtlT6uv43Tq/j02k2wzJFlHK5nTKfzqybY/pvLbclLz6c9S29d1rS8uubEc2bT94+Vv+tvJn+6LXXq+cR6tPDmlfL8/oZ9bXY+Q/Zj7bZyN9OFr3iWFItvNg73I45HsFjH9/pIx1JGP7nvOBjXPV7NG2NepMhWTL6MJI+3p5Rj6j/s5eX636W31/jHuA12fH0uft9plINoBYtzmnmBBCCCGEzB4+FBNCCCGEkNnDh2JCCCGEEDJ7+FBMCCGEEEJmD0OyRdiQNaVMD48ThaAZCmXi5LchVoTR0ECtsC0Wa2ckzIsc8/JnwtpoH6L2ZsN1ZULoSb5syKaWP1PD2rT60bPXy3+MMFGZfED/nEd9mL1eGZKNWKZoay88V8uepHl1e9s6rFekGZGNc9XsVr95bYvqzIZky2jSKTQ742N0DYxq9RQNHdXsyIdWfVHdFxSSjdEnyDxZLoGVUeCrK+Dubvcpx4Dd/mIB3N4+lpfkAmD0ifOglPNZ2U6m0TqHpQAVzsO1p9n6GDX7bDin7yijT4yyXG7OYCnA9bX/2duW/eVy32Zkp5fPK+fZ9/yx+8vl/Xrs8V6bW2nap1bZXr5Wv7byiN2I1Wrz7dR/d3f7n7Kt9+2DNCHkfMjoVlbXIo1s5ctqlFfuMTTb+petI3M/a9WdPR89spqtP6nZZABOn4iQhyZg9++N/dRp3jawH79QbEZ25Msb5fPK6bq0z9afnn+jMSkzMQzFp5a9Xr4Rv72+JITMhyPEUr2nTSP5JL1VTrDlojzH1OxW/mwd0f2sVXevP16WAyEnhSPFhBBCCCFk9vChmBBCCCGEzB4+FBNCCCGEkNnDOcVkntiQPEAuHBHDgBFCyKOzuFqjlBtzVOYh6/nI9hjnKp8Ll3D75ENxRCvmXyber36Y6sU5lPw2fmEm1qbktz5bf+x+FFcyimE5GkexFzMym6/Vr5lYkBEM0UPI0yOjW710q00j+SS9VU7w4gs/tGZb/7J1ZO5nrbp7/ZGIY3v7oqPphBwBTp/oYWMeyhdcs1hsVs7Kg5iOmQhsVuleX+9HRLD27u52D4WrlV+Pjb1Y6/7DpC13cxP/a7Zeb/60XfFV2qS5udnVZ+NCSnu0rdVq45PUEfXlarUL8aPtRdzcAC9e7Pth23Bzw4deQuaO1hkbx1b0zP7zrPOt134sc0FrmGiv1UWrUYvFTlu1fa1dorOWUc3W9Vk/dJ3W1nrt3++sZt/e7my3yGi2+ErIieFIcUQrLBowFkpN8o2GqGnl0WFxrD+joXcydffC8IzaavWpZ8Pa64UDIoTMj1YIsZZm2zKyH2nLY+j6oTqbPZ6xFWl2FAZ0VLN1PkJOCEeKCSGEEELI7OFDMSGEEEIImT18KCbEslxufsqTuc6yL3/Z11GPlPHqyrwqtvfKVm2z5VPLl1Y/ZOuO2hP5mH0NLyFk9iyv13vSxL/z/Fter/sn88RwTjEhFj33Te8LpeReRz1SxqtLp2VfFWvz2vZkXinrlYv6IVt3qz2ej622Wz8JIbNmdXezkYmMBmc1u1dmxJ6neb37wahmj/hkNXRKWduWRDvuh9Q7P/hQHNEKiwbsLoZMKDXJp7fFxtQ8mVBmUagea79VzqvP8ylTTxTKrReayPOtFQ6Iq5gJmSeR7tg0q9m2jOzb7VGts3l6Yd6m6mwmXNtDaHYUBnRUsyV9BUJOCqdP9LAhxGwYsru7/ZBrOmSahKHRdrRdja5Dyopwe3l1nV6oOMkDbP5T8x4UvXKST7fF1uf57SFpEhouytfz0+YtxQ9NRAghQitUmNVsya/DWHrhwzz9F25u9jXb0+QozFsU7ixqk7btabbkkRBro0Sa3fIzuhfIoA01m1wA3ZHiUsofBfB/APjibf6frrX+QCnlqwD8BIA3A3gNwF+utf6rUsoXA/hxAO8C8LsA/mKt9faB/H84Wj/7jvy0m/05XH96Id6A+OeOno9iS8pl8k9pb+YnmlY/2DZHUwz4szohxGPk5+rez8GenUM1NuNHz571c/SekGlnr+5omtXIz+6RT4SckMxI8b8E8C211q8F8HUA/mwp5d0AfgjAD9da3wng8wA+sM3/AQCf3x7/4W0+QgghhBBCzpbuQ3Hd8P9td//I9q8C+BYAP709/nEA79tuv3e7j236e0rhv4DkgpC5b3YunPzJPLjsdiafV5ednxfZjdKjuXw9H225Vj9k647a03sFumefc8cJIRarCZFuZzW7V6alcb3tx9LsyD89l9v2Xa891nZGq219Z0xqoV0p5QqbKRLvBPDfAfgnAH6v1vpim2UN4G3b7bcB+DQA1FpflFK+gM0Ui98xNj8I4IMA8I53vOOwVhByTOwroh/zldEPUddUm8foh2wZvpb77KFmk7PGRo54LE05Z82eYmsk/6jtC3guTi20q7Xe1Vq/DsANgG8E8G8fWnGt9aO11me11mevvPLKoeYIeTiWy81/ua24vF4MXSnnpfdiEOvyrTpbacvlvv/2r+V/z5dMHOFs3ONeX7TiG5NHg5pNLorlMq+3nl6OaFxG8zPHIs3O+J+9f2T9se1p3QeekGYPhWSrtf5eKeUXAfw7AN5USrnejhbfAPjMNttnALwdwLqUcg3gjdgsuLssbKieUto/CWRC1mTKReGCgF263u752AvdFuUfbW+mL1r9kA3bk2n7sX9W9xY8ih/HWFSTWUgzJd6lXbCZsRv5Zcu0/I/6Iop7nOkLLsohGTzNiMKPtXQoshOVa5XVnxk/MuE2ta1emLdIo1vtlHtQqx+ymp25R4idFY6HXiyY0RhvgfsxFi9mtCxaZC+M3C/s/iE+9mw+Qc3ORJ94BcC/3j4Q/zEAfxqbxXO/CODbsYlA8X4An9gW+eR2///apv9CrfZ3jQvgWD+JHGLnmD/LXMrP05fiJyHk/LjAW03IpWjhMf28jOcm8oTJTJ94K4BfLKX8GoC/B+Dnaq1/G8D3A/i+Usrr2MwZ/tg2/8cAvHl7/PsAfPj4bj8Cy2Xup4rlsp2/99PLlHQvr/hBCCFzpaW9Wc2OtDijyXxNOSEXTXekuNb6awC+3jn+KWzmF9vjfwDgO47i3SkZjRHZi1HpHev9lBGlt/ISQshcyU4BAvKxeEc0uecDNZuQs4ZvtCOEEEIIIbNnaKEdIbNELyQBcotJMotuosUnUqeUby1gafkTLdhs2e0tPpqy+LIX97jXFy2fCCHEQy9UzGiM1cvRBeaZBeW9Y5FmZ/zXdbfuH1kfbXta9wHPH69/Vv6pOic4UkzIISwWm59CRcw8rq6Au7t43x67utr8tLta7YRSBOnubidskU3ZXq02giR2JM3WC2zq0O3Q9azXOxvWbym3WMRpwGYepfxcrVfUR3XadulPQgiZSkazPawW2TSt2YsF8OLFThe1Znta3dNsq/Via7HY1dHTbNm25SxyvxEbtn2LxZPWbI4UE9KjFZKtFcZH8kbzDmU/GypoJPSNXYU/Ms8yG34oU07SojZn6+R8TELICFFItihUZS8kW2aeeM+WLhOVF3qaHe1nQ6Z55XRaL/2JajYfiiNGY0T24gB7xzJxK710Ly9/TiaEzJjF1Rql3ACo2IT2ks/tNmCO62NOfps3U8aro2uTCLyNkVPDh+KIh3w1IiGEkKNy+2Lw53BCCDFwTjEhhBBCCJk9fCgmhBBCCCGzh9MnCOnRCskWhfE5RkgcW0c29I3eFqJ8YteGSfPKtML2eOV68+1H6uQcekLICFFINi/sWSYk20iItciWLhOVF3qaLfb1fsbXVjmdFq2pGg03qjV7FZ+uc4EPxYT0yMwXfypzyqe2o1WuZ/Op9B0h5HyYqjvH1CNq2z4XsLCU0ycI6bFcbv7TLWUTbzfa1vvL5X7ZbJq1Y4/pshlfenVljnnHo0+vzin1ig3vHHjphBCiaWleRrtGNPtYGpzV+RHNtv5E/dHS7YwPT0SzS7Wx8U7As2fP6vPnz0/tBiE+pbRjTMo2MB7jtxeH0h7TZbz6WvmzMTa9uJmRnVG72Xqj+qP9E1JKea3W+uzUfjwm1GxyTrhycKgOAdP0XI5J+Sn3gqwf0TGvrdaf0f7Itjc6KaWgoJ6LbIe6zZFiQgghhBAye/hQTAghhBBCZg8figkhhBBCyOzhQzEhhBBCCJk9fCiO0Csno1WT3irN3qpSL6/UYVdrtsqNrAq9wBWgZ4XEmixlF3PR29b7NgZvNs3ascd02Ywvvboyx7zj0adX55R6bRxibctLJ7Nneb2+J59P4W95vT51114mLc3LaNeIZh9Lg7M6P6LZ1p+oP1q6nfHhiWg24xRHrFb3V1b28mRW63t5dR3e8WOsFLX1kONxdQXc3W225Yu/Wm3+IZHjmlaaZ/fmZhPvcrncLyuB3L0ykrZa7f8j1KtT55Mg9FL3er05vljs+2Prt+3TfaL7SgLcr1a7+iTNBo7X6VLHcsk4oOQlq7ubndR5Gusdb+nnQ2m2t63tmLpKuek3noxhdainr6vV7uGvZa+n2a0y2idNdMzm18f0Czju7vaPiWbberWfWuPFBrBvQ5e9u8tp9gXAh2JCekT//EQ3TyD3z8lIODXtR+YfotZN1/rQ8kWETD49f0bqzz4MZNrGf/AIIRFWL4G2PrX0VcjqW2TTs5Hxydbf8sVqY09vW+W17rds9Hx7+Ymzh9MnCCGEEELI7OFDMSGEEEIImT18KCaEEEIIIbOHc4ojZOWkbPfyyEpMIVpV6uXVddjjUbmW3VZdF7IClBBCRlhcrdWitGj+oj2u92Xbfkb2ZP5kq47k9p6d/boWV2sAXGxHyGPAh+KIzKp2rnyfB9E/P94/Ifqfm9Y/J61/dmz5KKRPy5fIJ03vnyiJPiE2ZHWx9Wek/pZf2X8Ao7BtZNbcvniqD45PtV0PjNVLoK1PLX0VIs0Ecprt2cj4ZOtv+aLr1xEhIr1tlde637LR8+1lmhMR5szgQzEhPc7ln59T+uHVfS79QgghllF9egg9O9TmqTX22PUz+gQhT4DlcvNfbimbOI69F6osl34577OVtlzubPRezDLyEhdtN1s2+7IYz+fROvVnqw2EEBKhdSijm5FGHarZx9DvjH5G2pm5X03VbK+uC9dsjhQT0sN7kUsvNqNXbjTGsK6rVa7lx+jLAqKy1o8pebN1eu2P2kAIIR46RjAw/SUqh2q2l7/nxzFiurf8yOYd7bcnoNkcKSaEEEIIIbOHD8WEEEIIIWT28KE4Qs+lif68+TK2nM4zsh/NK2rVkZnz5NkZ8TGaN5udD+q1odfGVplWHYSQ2bC8Xnclm3/n/UfpJqeGc4ojWvNIhVLG3lU+sm8/M3XYvFk7Iz5qe54Nmz4y3zZqY6tMyzdCyGxY3d3sS7TVMXvM0xCv3KhmZ+qZotlRWs/HqOzo5yFtSfZTWZnzRcgjw4diQnrYeLzA/r7dHo3j24vD2ysXbUfpmViTvTicUd5DYhJ7n1EZxigmhLTQMYKBvm724vhO1Wwvf8uPTKz5TNmWH3r7GHHkdfmozGIBrHD28KGYkCxXV8Dd3WZbBGO12j8ux5bLtq27u105XVaniW05FtUp29YPj8Xi9LEvCSHkMehpttXN29uNdq/U05vOQ80+jAv4EZcPxYT08KaZ6Okh0c+Cki77rZ8O7X7P5sjPozYPIYQ8deTBdkQ/pVxm6oc+HuUZ1ewojTwaXGhHCCGEEEJmD0eKI+w7x4H7+6PvKh/Zn/I+9NS7xx07Iz723u3emw/amm8btTEzR9ergxAyL1o6Zo89lGZn6jlXzW7Nk43qP4ZmS94LmHNKnjZ8KI6YOofnMeb+HKuOY9h5qH7y0p/KvCpCyMNgo01keCxdOaXeHstGq+wxNJszBciJ4fSJiOVy8x9s9B7vKK31TvSe3d67ypdLP35y9h3runwmbnEm/rEtm7GX8dmzkan7+prBLgmZK1M0O6svUzVb28j4EflwLM327iUPrdlWm61m6/KEnBCOFEe0JtsLXlrrs2fXfnp1AOOxN60P2bjFPVutuMMte1mfrQ3re1S3Xjl8DFo/GerjI+F99DFhariz3s+fOg+nlpCnzBTNlm1b/pia3Ypfn9VX8c+zO6LZtr0Zez0/Ixu2DmlHFHN+It5sR0Dse/t1/3hpfUbHMnlG6+ylXTaXcPvhQzEhPThtgxBCzhZKNDkWnD5BCCGEEEJmDx+KCSGEEELI7OFDMSGEEEIImT2cUxzhLa4aWRgVxXps2c3Gu7QrCrLvWO+947wXv9Kzn4lRmX3ve88n63sUC/Pm5v75JIQ8faZotmzb8sfU7FYs5Ky+rtfH0Wxd/rE0W8qINnvxixmnmJwBHCnOot9VrvfttuXmZn+V7Xq92da2Wu9A1/lWq11564fsS336wVC/l13KC/Je9eXy/vvZW30geVar+6K4Wm3C7HhRIGybbVsFewPw3lMveez75tfrw8L7LJcbu2JD9uUvG/YoEwqpF/JItjMhn1rhp/S+9SHyLeNf7zPTj1G5Xp5DzjGZD1Zn5JjVb9EXq9laF6dottbCq6tYs0XntA3RV63bh2q2+AXs6m1pttdfuq2C1my5B9k+jjRb2jqR5fX6npzx7/z+ltfr/sk8MaVOCXZ+ZJ49e1afP39+ajf2KaUfYi0bUmeknN2eaj+yMcW/nm3JC/T7xMsT2WuV6W3rMqOM9uXosUPbk+lbm6+X16sn49+U66hle8q1f2JKKa/VWp+d2o/H5Bw1++XlkPnuZL5r2e+33W6lefaz136m3Khmj9rx+nSKvaCPpn6lw3M/ol2tfLq9U+9BI1rplbH2onqy7R35LmTsJvKciWQDiHW7O1JcSnl7KeUXSym/UUr5h6WU790e/4pSys+VUv7x9vPLt8dLKeVHSymvl1J+rZTyDcdvDiGEEEIIIccjM33iBYD/otb6JwC8G8CHSil/AsCHAfx8rfWrAfz8dh8AvhXAV2//PgjgI0f3mhBCCCGEkCPSfSiutX621vor2+3/F8A/AvA2AO8F8PFtto8DeN92+70Afrxu+CUAbyqlvPXonhNCCCGEEHIkhhbalVKWAL4ewC8DeEut9bPbpN8C8Jbt9tsAfFoVW2+PWVsfLKU8L6U8/9znPjfoNiGEkMeEmk0IeeqkQ7KVUr4MwP8C4K/WWv+fIpOwAdRaayllaPp0rfWjAD4KbBZtjJR9FHqhbqaE1MmUs9s9+5kQaa1yU8IOWf+kPNDvEy+PZ88LI5fts0PDsklfZPty5NjU9kRlopBNNp/ej0L7ecez13nrGhkJSzVy7Ytd8iicvWYD/ndQjst+T4dbYcMO0WzrT1azbblja/aUkHO6T62/UajMVP8dcFlZTfDqz7TNbh9Dsz17I2FfRzQ7295e+7P3gxHNPuT8Pha11u4fgD8C4H8D8H3q2G8CeOt2+60AfnO7/WMA/pKXL/p717veVc+OxaLWzULJzd/VVby9WOyXkfRWmd6xyKaXt1fPof6N2MzYivKOtLlXl9g65rUwxe/Wvi1n9zPXXtQ3U8955hzovj3G9Tlq1+aOsw3UAAAaKElEQVQ9MQCe14a+PcW/c9Tszd0M+389DctqTeZa9r6/me/C1O/hQ/h3Ss0GNudw6rm3LBan0+wp+j31nNuy2TYfcn2OXgPbfFPP70MQ6XY3JFvZDAl/HMA/r7X+VXX8vwbwu7XWHyylfBjAV9Ra/1op5c8B+E8AfBuAbwLwo7XWb2zVcY7hfVDK5lR6+3Yb2OzLcfsZlW8dy9garefQchmbGVvSX4fU2zovto5DOaS/bN5W/pE2jvbXoddKrx+OeX2O9PkZwJBs54F7SfSuLeDw61y2M7YOvQ88tH+RD1LWs3ssza5HCMlmD07x2zM4RbOl7mPenzO2sm3u9cMD6H1BPRfZDnU7M33imwH8ZQC/Xkr51e2xvwHgBwH8VCnlA9i8h+Y7t2k/i80D8esAfh/A9xzoOyGEEEIIIQ9K96G41vp/AihB8nuc/BXAhw70ixBCCCGEkEeDr3kmhBBCCCGzhw/FhBBCCCFk9qRDss2OTPgo2bYhfEbDtnnHDgmXY7cjW6PhtjI2M7YAP+9Im1vn5dCQbJZeG1t+27x635YbCV1m22vrnXrOM+fAC3t0yPU5ale3jxCN/t4AOQ3LaM2xQrllbY5q4bH8i3zWfftQml0KcMyQXTZM5WNqtjBF/0bPuW1Pts2e3SnXWfZZYLHYrD47c7rRJx6Dc1zJTAghGRh94jwo5WwCkpCJTD2HPPeXwTmdp0i3OX0iYrkErq83Z7GU/e3lcpfHptl8ksemefs2v80XfWaPWb+zZVvpkV82Xdet/bF92LObqV/qOua10Guf7d/R66XX1sh29nrz7EZ5sufFa1/Gtyl5Wv1NiNC6poFxzbZ/j6XZ3ve+972135Xoe5PxL9KC6Ds6UkfvHB0LrUet9nnXRdTvI+c2sj313tDru1Y9Lc3O+JZ5Fsqc+wuAI8URcgKnxmbs2Yj2dX6bL1Nvxs/Ij6is+NRqZy+upbYRbR8Q/7DZvkPJtG80fTQuqddnXj+00kZihur6Wn0flWv5lvm+ZNp17PM8EY4UnwcvL4cp36/WtWb3H0Ozre3M99YrE/ne8y/SAqGV5wD9nhrH1pWClo5N+czaytad7aOH1uxM/WI7U3/D/plINoBYtzlSTAghhBBCZg8figkhhBBCyOzhQzEhhBBCCJk9fCgmhBBCCCGzh3GKI6bGOPTyeWnevs1v8x0aU7AXBzcqKz4dEotYx8iUfvXihB473u0x8GL92vbZ/tXXzpSY060Yo73rMtNHQjZ+a9T3tn0Z3zLfF5un1d+ECNH1NVWztV2bt5X/UM2WbS+2sJfP+6604gv3/Iu0QPzxtORQ/T52HFutR7322esiuldm+kzalHl/wbHiPB+i2RnfMs9CXp57ds9klV0DPhS3uLvbfF5dbbblc7XahBiRdEmzZVar/fIauZh0ntVqZ8cro/3S5aO88qWUPNZvW07n99rmYftGf1q78uWI2qXri3y0Ntfrtr1jErUP2JyP29tNaBp73rUwj9Shz/Fqtd9/9nqz15Pd99D1aN+lT72+t+cZ2P8n58WL/T4Q9HnS5fW2V2/U34REeNdXpNn6+o402/s+6vwtzY6+zxFag+VTa4etw353Pd2MNDTSbvFD7K7Xu7ZEbfP60rNp29+7H0zF0+GorkivWv0O+O2356Ol+zZvdH1E/azzH6LZtbb7wNqxmm3v673+PnP4UByxWuVD58g+kA+HFeVv1RWVj/zy8rTsWp9027KhZEZ9bbWh16dCVO5YZK4FfRObEg4q0/5WWma/V4/4PnK+M3Z0HtuOEdsPfZ7J5ZPVX+94S0cHwk5NLi/HJE9Wb4F93cm0P3NvELvW50wfHFLnMcjev0fbo/NE+Xv1t/TS86XVBsl/MZqdO32nhHOKCSGEEELI7OFDMSGEEEIImT18KCaEEEIIIbOHD8WEEEIIIWT2cKFdhA7FkglxMhL+Rux7+Vt1ReUjv0ZDnmXDC7X8HfG114ZWn0pZLwzMQ4Vka/WFFzop0zetY975iPrG5u1dX149rfA8mTZkw17Z/UP7hxAhq7/e8eh6jb6Pxy4P5DTR0+xMyDZ7zNbj+RGFlzzWfcCmr3Acsvfv0fboPF5+73xkz91oSM1jhDR9bM1mSLYL5vb21B6Qc2HkWrjk6+ZYvl9yH5DLpp7/TfdBeKjv3GN/l0s/S4pTa9Cl3TMey4djnd8HhNMnIpbLzX82pWziW15f7+/Ltt23216afC6X9+uJ8mmfovoie9G250crn/3sHbN+j9i0fdyry6ZL3Y95Ldj+HzkHx/aZkDmS1eys9mQ12yvb0/cRbT9Us7NpGb+z94GRe4V8HhPdPyOanb3X2PsceRJwpDjCxhkExuL3tcrpz176SCzBlp/etudHL9+UOLJTYvfqdth+zrSrFSB/lOy14Pk9cg6O6TMhcySr2frYMTQ7KtuypfNktP0Qzc6mZfsgcx/otcG1jeORjf+r06zfmT469sM8OSkcKSaEEEIIIbOHD8WEEEIIIWT28KGY/P/t3V+oNddZx/Hfwzna+g/TNDGEnuTsFgOSC40ltAn2IlbUNIj1oohBMEggNy1UEDRBMOe9st5YLUqxYGgL0qioNIRAjWnBK/vPpukbY8xbyaEnpH1R23ghVN/j8mJm5ayzzlqz1syZfWbPnu8HNnv2mjVr1jPvfp+9zp6ZtQEAABaPQTEAAAAWjxvtcuJ5BqXTr8OL67vmMExt13eOxNq5BGvmOSz1o89cxKk6YVmfORu7jlt8nGvi8vMGj6H2vdA1r3BXX9fRZ2CJanN2WDZGzk5t22de4lJuP2/Orl1X0+/az4FSDMmyYVPqxdM0N/yNe/7muLg8tS6xbXG7kW8Q3GJzmFqeQXGt4+Pm2Sc+z/8rxzMH7Ow028T1Q4eHJ+t9/f39Zs7A1apZn0pQvt3j49P78H3w6+K+xG3H/Qi3S/U/3G9X7H5wd3h4tt9+2/A5FVNXrKn9xzGPaRPmkQTQTy6PlHJ2vByqzdnh67itOGf7QWG8rivPhv2I+5/L2V2xhzk7/hyJj0P8Oowv7n/X/kfM2aRojIVBcU5qGjG/LPWfvia1Llc/3H+p/T59SLVd00aqPN6mNM1Nn/7G26SOV2n/TG8GLE/f6cxS63Pb9M3ZcZ6tyYG5df51TR/XkbO7XqfKh+x/7CnZgAG4phgAAACLx6AYAAAAi8egGAAAAIvHoBiIrVbNNW6r1enXqcfu7unXXduEdf1y/Fwq8/vw7efq1LwOy8I2wxjC9XFZKY6ufuZep45fqn5YF8CirXaPsimax+Y8VrtHU79VirjRLic1jZhflk5e105fk1qXKoun6KmZ1qc0jVDt1G65NlLlcex9pibzumIqHa+u/Z93erPcDTNhn1M3kYTrw23Cdee9ITPuQ65Ozet4X2GbPoauspo4cv2M20jtp/YmUMDL5exUHuyTl4fk7DjP1uTb1LohU2bm2j5Pzq7Jz6XYSrEMnJLt8HivSQ+5PCqVbyqMy8I24vbGzNlx3/rk7FTuzuXzITk71W5pXcc+zDZ/2lEGxTlTz/Gyzv1PHRsArEM8gLhIcV4lz/bH37mYGJdP5KxWzV83Y5zyzp0a7jo9HtZL1S2dto6379PPXL9zfQ/3d974w+c+xyfXXwDLMSRn1+S+rlzcp17pUqC+OTvXZp+c3Sf+Pp8bfXI2eRsbgm+Kc3Jz+Q45fRI+l9r2y2G93PrSKZCuU/g1ccX97uq7X+f7NTT+uG7t8YnbALA8Q3J2V72aXDykXs32Xk3f4zbjH/oofUbUxt/3c6M2Z4d1gAnxTTEAAAAWj2+KgVjuhpmQfx3fZNK1zdAbeeJtizerJPaXeh2WhW2GMRwd5ctKcXT1Mz6GqX131Q/rAkCcE+IcnMvhvm6qzrpzdtyXITk7FXN4s/nQnB32retmzlzOTu5jwmv+KzEoBmJdN8js7EjHx6eXwzKffHZ2Tpfv7zftrlbNKURf7sXt+LLUfo6Ozm4vnSRBf+o01WYcg399eHiS6A4Pm2v9/Pa+bLU62268j7APqX6GsfjjEvY5Tt7h/gAgJ3eTZSpnx8tHR+n6YT4bkrP9ci5nx/kvl7NTbYZ5Wjq97NsMc3bchre3d/LZFPYz9TlxfFzO2am+zwiDYqCk5jrB2mmAStfUxe2lymuvLaztr9/mvHX7tFe6xrK2LQBIqbl3pybX+HKvT85OtTGk/Xhfub7WXsOeqpO7Jyh1fGraTD5r4zEozsnN5Tvk9EntPMHxKYyauZK7ToF0ncKvne8y7Heu77Wn1mvi7zPXZhwfp9WBZRuSs7vq1eTiPvVKlwL1zdl+OW4znKe9lLNrcn+fz8M+n0lxnUMBk2JQXBKeus2dFkidMohPMUinB4yp9XE9v9/UafrwlEfYRuq0SnzKOoyr5lRHeBom7m98rVR8aj2MM3WaJdeH3GUHPr7w+IT9SB1PAMvRJ7dJ+dPCftnn0nh97rKpcF3cj/B0fFgnzmm1OTs+pe7r9cnZoTFydtwGORszwqA4p/YXtWpOn+fqpdrsaj989omkNKVOrk+1+6k5BrVx9onR89vXTmPHaXVgufrkl9Q630af/Fmzvz75subyotocndu21IfScejK2TVtdPUXmBBTsgEAAGDx+KYYKKm5TrB07WDNNXWp9lLlpWvR+7Sfuj49V7dmKrhSndw1i7nj29UW144DyCndu1OTa4bk1K6243W17cf7SvW1K0fW1MndExTHUNNm9jMrOqOwgRgUAyVdU7RN2RYAII1c289FHK8ZXCFTvHzCzB4zs6tmdjkou97Mnjazl9rnN7XlZmYfMbMrZvacmb19nZ0HLsRq1fyVa9bcdBM+l5Zz61arfu2m6sTPcZt9+hu2Ecccx7K7e7ZebdxdsXT1K17v9w8AKbncNHbOzrV73pxd6u/YObsrjgXl7Jpvij8u6Y8lfTIoe1jSM865D5nZw+3r35b0Hkm3tY93Svpo+zw/Nb+o1XU6O3Uaueu0Sd+pbsJfq+mqG7dXE0ef/vaJs9dpFp3evnYau7BPY0nNU+yfw7LUstRvXa6tVJ3Uc9hmn/7GbaTmU/Ztm+Vv9CzF3RVL1/FIrQdiffJL16U8Q09Fl55L+TK1XU1cfl1Yd6qcXdNGqn9jT8nWdXP2mDk7126qvE/OLvV37Jxdc/PkAnJ28Zti59w/SPrPqPi9kj7RLn9C0i8F5Z90jX+UdJ2Z3TxWZyeRm34mZX9funbt5NqceKqZ+BfC/DbhurCNcPqbuB9+qqBwap9UP+J18fQ94b5z+9vZafbj9xW276ftidvxUschdUxS+/TbmzXP4X7j4xnu5+hoNn+VAliT2ty9t3c2l6WcN2f7No6O0jnbuaYvXTk7l6vDep7PiX59Ks7cZ0vXcej6pUz/HLbh9xtuG/Yv3AaY2NBrim9yzr3aLn9T0k3t8lskfSOod9SWvaqImT0k6SFJuvXWWwd2Y41yU51J5b/6Sr8M0/WXZ2mqta79p9ro6kdqXc1f0jV/cXqlbyhT+871J95v17YkWGBUG5+zpWFnYuJv1Pp+k9gnZ9e0kep3Koaa/eS+uevK2aXj1bdfvf4tBEzq3FOyOeec1P+WQufcx5xzdzrn7rzxxhvP2w0AwBqRswFsu6GD4m/5yyLa56tt+SuSbgnq7bVlAAAAwMYaevnEE5IekPSh9vnTQfkHzOxxNTfYvRZcZgHMU2qe4tL8kH5ZqrvJpNRuqk7XTTy5bXL9Tc1ZGbfj2+660bMUd1csXccjdxMRgMWL01XDX44RXpbRVdYuS8PWnWk3VZ54PtVmn/7mnlP9jZar2yzEkjwe+fVzSNvFQbGZfUrSPZJuMLMjSY+qGQz/pZk9qOZ+0V9uqz8l6T5JVyT9t6RfX0OfgYu1xPkua2Ne4rEBsFFIQxhLcVDsnLs/s+pnEnWdpPeft1MbITfVmVT+pq70yzBd3xaWplrr2n+qjZpfqKlps6aPcd2u2LviyfUn3m/Xtr4ugOWozW9hWe00adL5c3aub7U5u+tMUal+zTRpXt8zYn3OYHW23fv2JGBU/KJdznn+9Bzjz9ZNaWPK9te434ODA126dOn8fRng0Ucf1cHBwST7BrZaPIvCRZpDzt50Z07ZAxfr3LNPbK3VqvnL1az8KzSr1eltzvOrObl9lH69JtdO6bkUQ1he+lWduL818eeOVU0svm8DHBwcyDmXfuzvy+3syE+rUlzuWffg0qXy+6Nm/VzqnuPfCeilKz9JJzkqlUdr8nLXez2Xs0vrSv+PUnW7YjhPzq7px9DPmbBvI1vtHiUPEY9hj9VuxzzVW45vinO65sI1a8rjeSfPM09l19yZqfJwX75erp3S3JalGHx5rj9dx6QUf+5YlfYV7nNsfeaZHjqX59LqAhehlJ/ivBnn+VQbufZqc3ZX231yYelzx5eV5jyu+Ywp9WPI50zYt5EdHu/JqdDH3DJ1z9Q1W+7lh3xTDAAAgMVjUAzE5jBvDADgBHkbI2BQDMSWfrMLAMwNeRsj4JpiICU17Vtu+bzTzm17Xb7BAbBGJ+naqfjjHGeWqRvX3d85UvODxMvDoDinay7ceN7F+FfAhsxTGS/n5nbM/eJYbn81c1t2xVCae7mrv10xxn0eMofyOgdbfOsAzMr+zlFwg1D8QV96bpelyoFDZl3X9sl1YXmpTx1tDKpfsT7XbnGbmuN71tCUTroe2zIHxBKD4rwh/8su8n/muvaVa5esA2CDvXxtuR/kAMbBNcUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFi8tQyKzexeM3vRzK6Y2cPr2AcAAAAwltEHxWa2I+lPJL1H0u2S7jez28feDwAAADCWdXxT/A5JV5xz/+ac+x9Jj0t67xr2AwAAAIxiHYPit0j6RvD6qC0DAAAANtLuVDs2s4ckPdS+/K6ZXZ6qLyO7QdK/T92JkRDL5tqmeOYey/7UHbgI5OzZ2KZ4iGUzbUMsybxtzrlR92Jmd0s6cM79fPv6EUlyzv1exzZfcs7dOWpHJkIsm2mbYpG2K55timUptunfbJtikbYrHmLZTNsUS2wdl098UdJtZvZWM/teSb8i6Yk17AcAAAAYxeiXTzjnrpnZByR9RtKOpMecc8+PvR8AAABgLGu5ptg595Skp3ps8rF19GMixLKZtikWabvi2aZYlmKb/s22KRZpu+Ihls20TbGcMvo1xQAAAMDc8DPPAAAAWLzJB8Vz+0loM3vMzK6G0xGZ2fVm9rSZvdQ+v6ktNzP7SBvbc2b29ul6fpaZ3WJmnzOzfzaz583sg2357OIxszea2RfM7KttLJfa8rea2efbPv9Fe/OnzOwN7esr7frVlP1PMbMdM/uKmT3Zvp5lLGb2spl9zcyeNbMvtWWze4+hQc6eDjl7c/OcRM7etPfYEJMOim2ePwn9cUn3RmUPS3rGOXebpGfa11IT123t4yFJH72gPta6Juk3nXO3S7pL0vvb4z/HeL4r6d3OuZ+QdIeke83sLkm/L+nDzrkflfRtSQ+29R+U9O22/MNtvU3zQUkvBK/nHMtPO+fuCKbxmeN7bPHI2ZMjZ292niNnb9Z7rD/n3GQPSXdL+kzw+hFJj0zZp8p+ryRdDl6/KOnmdvlmSS+2y38q6f5UvU18SPq0pJ+dezySvl/SP0l6p5oJxnfj95ua2VHubpd323o2dd+DGPbUJJ53S3pSks04lpcl3RCVzfo9ttQHOXuzHuTsjcpz5OwNfo/VPqa+fGJbfhL6Jufcq+3yNyXd1C7PJr729M1PSvq8ZhpPe+rqWUlXJT0t6euSvuOcu9ZWCfv7eizt+tckvflie9zpDyX9lqT/a1+/WfONxUn6OzP7sjW/iibN9D2Grfn3mf37j5y9cXmOnN3YqPdYX5P9zPO2cs45M5vVlB5m9oOS/lrSbzjn/svMXl83p3icc8eS7jCz6yT9raQfm7hLg5jZL0i66pz7spndM3V/RvAu59wrZvYjkp42s38JV87pPYbtM8f3Hzl7s5Czt8fU3xS/IumW4PVeWzY33zKzmyWpfb7alm98fGb2PWqS65875/6mLZ5tPJLknPuOpM+pOV11nZn5P/7C/r4eS7v+hyX9xwV3NeenJP2imb0s6XE1p+P+SPOMRc65V9rnq2o++N6hmb/HFmxb/n1m+/4jZ29kniNnn9jI91itqQfF2/KT0E9IeqBdfkDNdV6+/NfauzPvkvRacPphctZ8vfBnkl5wzv1BsGp28ZjZje23DTKz71Nznd0LahLt+9pqcSw+xvdJ+qxrL4iamnPuEefcnnNupeb/xGedc7+qGcZiZj9gZj/klyX9nKTLmuF7DJLI2ZMiZ29mniNnb+Z7bJCpL2qWdJ+kf1VzLdHvTN2fiv5+StKrkv5XzbUzD6q5FugZSS9J+ntJ17d1Tc2d2l+X9DVJd07d/yiWd6m5dug5Sc+2j/vmGI+kH5f0lTaWy5J+ty1/m6QvSLoi6a8kvaEtf2P7+kq7/m1Tx5CJ6x5JT841lrbPX20fz/v/43N8j/F4/d+UnD1dLOTsDcxzUVzk7A2IY+iDX7QDAADA4k19+QQAAAAwOQbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWDwGxQAAAFg8BsUAAABYPAbFAAAAWLz/B1i4DMHbb3rFAAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Plot page 3 with figure at the top in the form of a diagram, and characters, textboxes and lines."
      ],
      "metadata": {
        "id": "7TN7kg8fj1vr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "xmin, ymin, xmax, ymax = page_3.bbox\n",
        "size = 6\n",
        "num_pages = 2\n",
        "\n",
        "fig, axes = plt.subplots(1,num_pages, figsize = (num_pages*size, size * (ymax/xmax)), sharey=True, sharex=True)\n",
        "\n",
        "# figures, chars and lines\n",
        "ax = axes[0]\n",
        "for figure in figures[2]:\n",
        "    draw_rect(figure, ax, \"green\")\n",
        "\n",
        "for c in characters_3:\n",
        "    draw_rect(c, ax, \"red\")\n",
        "\n",
        "for line in lines[2]:\n",
        "    draw_rect(line, ax, \"black\")\n",
        "\n",
        "# chars and TextBoxes\n",
        "ax = axes[1]\n",
        "for c in characters_3:\n",
        "    draw_rect(c, ax, \"red\")\n",
        "\n",
        "for textbox in text[2]:\n",
        "    draw_rect(textbox, ax, \"blue\")\n",
        "\n",
        "\n",
        "plt.xlim(xmin, xmax)\n",
        "plt.ylim(ymin, ymax)\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 509
        },
        "id": "Z1Fp2Fkgj38S",
        "outputId": "1a498da5-d92a-4bf4-a80b-61c955c1047a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 864x610.971 with 2 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAsUAAAHsCAYAAAA6gNB7AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nOzdfax92V3f98/qvWAeQj3Y/nU0nWvOcYQVhKJi7J+MLVBE7JJiCzH8AcYoggGNNP3DtNBECqaVipEi1UhVDKiVW6umsSOCcRyQR8gKccdGUaXi5DfGMQaHMrj3xHdke3449pAG8fD7dfWPs9fcddddD991Hu55WO+XdHX22Xs97nvvd69zzjprO++9AAAAgJH9R7tuAAAAALBrDIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGx6AYAAAAwzvddQMk6SUveYmfz+e7bgYAdHvqqaf+2Ht/a9ftuEnEbACHrBS392JQPJ/PdefOnV03AwC6OecWu27DTSNmAzhkpbjN9AkAAAAMj0ExAAAAhsegGAAAAMNjUAwAAIDhMSgGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHoNiAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAzPNCh2zv03zrnfc859yjn3K865r3LOvcw59zHn3NPOuV91zn3llPYF0/Onp+PzbXYAAAAAWFdzUOyce1jSfy3ptvf+r0s6kfRmST8n6R3e+2+U9CVJj01ZHpP0pWn/O6Z0AAAAwN6yTp84lfTVzrlTSV8j6XOSXifpA9Px90j6vmn7kem5puOvd865zTQXAAAA2LzmoNh7/4yk/1HSv9VyMPycpKckfdl7f29KdiHp4Wn7YUmfnfLem9K/OC3XOfe4c+6Oc+7O3bt31+0HAGCLiNkAjp1l+sTXa/nu78sk/aeSvlbSd69bsff+Xd77297727du3Vq3OADAFhGzARw7y/SJ/1zS/+O9v+u9/0tJvybp2yU9ME2nkKQzSc9M289IeqkkTcdfKOmLG201AAAAsEGWQfG/lfQa59zXTHODXy/p9yV9VNL3T2kelfTBafuJ6bmm4x/x3vvNNRkAAADYLMuc4o9p+YW5j0v63SnPuyT9lKS/45x7Wss5w++esrxb0oun/X9H0lu30G4AAABgY07bSSTv/c9I+plk92ckvTqT9s8k/cD6TQMAAABuBne0AwAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGZ1qSbUTzn59r8dxi180AkDF74UznP3m+62YAAI4Ig+KCxXML+Z/hRnzAPnI/63bdBADAkWH6BAAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGx6AYAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeAyKAQAAMDwGxQAAABgeg2IAAAAMj0ExAAAAhsegGAAAAMNjUAwAAIDhMSgGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHoNiAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGx6AYAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeAyKAQAAMLzmoNg599ecc5+Ifv7EOfeTzrkXOec+7Jz7w+nx66f0zjn3i865p51zn3TOvXL73QAAAABW1xwUe+//wHv/Cu/9KyS9StKfSvp1SW+V9KT3/uWSnpyeS9IbJL18+nlc0ju30XAAAABgU3qnT7xe0h957xeSHpH0nmn/eyR937T9iKT3+qXflvSAc+6hjbQWAAAA2ILeQfGbJf3KtP2g9/5z0/bnJT04bT8s6bNRnotp3xXOucedc3ecc3fu3r3b2QwAwE0iZgM4duZBsXPuKyV9r6R/kh7z3ntJvqdi7/27vPe3vfe3b9261ZMVAHDDiNkAjl3PO8VvkPRx7/0XpudfCNMipsdnp/3PSHpplO9s2gcAAADspZ5B8Q/pcuqEJD0h6dFp+1FJH4z2/8i0CsVrJD0XTbMAAAAA9s6pJZFz7mslfZek/zLa/XZJ73fOPSZpIelN0/4PSXqjpKe1XKnixzbWWgAAAGALTINi7/1/kPTiZN8XtVyNIk3rJb1lI60DAAAAbgB3tAMAAMDwGBQDAABgeAyKAQAAMDwGxQAAABgeg2IAAAAMj0ExAAAAhsegGAAAAMNjUAwAAIDhMSgGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHoNiAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGx6AYAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeAyKAQAAMDwGxQAAABgeg2IAAAAMj0ExAAAAhsegGAAAAMNjUAwAAIDhMSgGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHoNiAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGZxoUO+cecM59wDn3b5xzn3bOvdY59yLn3Iedc384PX79lNY5537ROfe0c+6TzrlXbrcLAAAAwHqs7xT/gqR/5r3/JknfIunTkt4q6Unv/cslPTk9l6Q3SHr59PO4pHdutMUAAADAhjUHxc65F0r6G5LeLUne+7/w3n9Z0iOS3jMle4+k75u2H5H0Xr/025IecM49tPGWAwAAABtieaf4ZZLuSvrfnXO/45z735xzXyvpQe/956Y0n5f04LT9sKTPRvkvpn1XOOced87dcc7duXv37uo9AABsHTEbwLGzDIpPJb1S0ju9998q6T/ocqqEJMl77yX5noq99+/y3t/23t++detWT1YAwA0jZgM4dpZB8YWkC+/9x6bnH9BykPyFMC1ienx2Ov6MpJdG+c+mfQAAAMBeag6Kvfefl/RZ59xfm3a9XtLvS3pC0qPTvkclfXDafkLSj0yrULxG0nPRNAsAAABg75wa0/1Xkn7ZOfeVkj4j6ce0HFC/3zn3mKSFpDdNaT8k6Y2Snpb0p1NaAAAAYG+ZBsXe+09Iup059PpMWi/pLWu2CwAAALgx3NEOAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeAyKAQAAMDwGxQAAABgeg2IAAAAMj0ExAAAAhsegGAAAAMNjUAwAAIDhMSgGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHoNiAAAADI9BMQAAAIZ3uusG7KvZC2dyP+t23QwAGbMXznbdBADAkWFQXHD+k+e7bgIAAABuCNMnAAAAMDwGxQAAABgeg2IAAAAMj0ExbsZ8Ljm3/Dk9zT+G7fj5fH6ZN+yP9+Xyp+XEx0P+Vdu0ymOoDwAOyPz04kroHOlnfnqx69OPHeCLdrgZi4Xk/XLbueV2+hiOSfnncfo0Ta7sNF8Qjq3SplK62uNisZlzCAA3aHH/bBkGW/GxFZ/D81b+NF8ubtf21drSGbudO9vIOcRh4Z1iAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4fFFO9yM2ezySw8nJ8vt9DEcky6fz2aXz0O6eF9cXlp2mi9uy6ptKqWrPZ7xhQ0Ah2d2cjF94cxLTpnH3L7wpbfo+PPPG/nTfFHYvn68lidXZqkP+cfZyYUkYvdoGBTjZpyf77oF1+1jmwBgT5zfG3lQOHLfx8X0CQAAAAyPQTEAAACGx6AYAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeAyKAQAAMDwGxQAAABgeg2IAAAAMj0ExAAAAhsegGAAAAMNjULwJ87nk3OXP6en17fTRuWW+OH/6vFRWKX8tba7+WltK6Ur9SM9BSNOTvtV26/kJ6ay/s7idtfy137P1nLd+F7k+Wn4flr+3deur9ddyzoE9MT+9KIYffnbzMz+94He2R+d7VKe7bsBRWCwk7y+fO3f5PGynj+FYnD99XiorrifNX0tbS2Mtq9aPuL50fy197nmp7ZbzE9LVpGXk2lPLU2tnrt3W30Wuj6V+Wuq0/B6t9dX6aznnwJ5Y3D+7Fq6K/zuW+Gf5n+tNG/ZJfbF41X1pfa0+pu2ytrPQZ+fOVHPld1YrMz1/247Zuf6vU7Yl1rfqM2y3zveoeKcYAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeKYv2jnnziX9e0n3Jd3z3t92zr1I0q9Kmks6l/Qm7/2XnHNO0i9IeqOkP5X0o977j2++6XtkNrv6RaOTk8vnYTt9DPni/OnzUllxvXH6Wtpc/bW2lNLV+pF+2erk5HK/JX2r7dbzE9LVpGXE9Zfy137Ptec9v4tcH0v9tNTZytdTX62/lnMO7JNcvMr979TiZG7fpmJ22Be3tdWW2v99K29anyW9JYab+5z5olxkdnIRfTnMSy73mEr3W59b9ifb0obKrtRhrq+9PTu5kMSX7VI9q0/8Te/9H0fP3yrpSe/9251zb52e/5SkN0h6+fTzbZLeOT2O4/795T/7/fv1dBcXy6Wszs+XjxcXV4PF/fvS2dnl8cXiarkhfQhkYX+cJi4n5JGWASouN6fWh7iPi8WyvNnsellx/jh9aHNoi3S1f/G+YLFYLgOW9i08hm2L83NbunXzrOom69pFfcC+sMTqIBezF4vrcTgX00J+a8zOlRHieC5mx/EwJ43Z83k5TVpuGnutMTvuc6mfxvN/fo8B3GZxPnPWWZLtEUnfOW2/R9JvaTkofkTSe733XtJvO+cecM495L3/3DoN3Wvpkmq1ZX3S5VFCAAmPrSWzcsur1JaZScuJ25dre60vpbLDdtz+1vHSsbS91mXFcucUAEpaS4atGrPDdji2bsyuxcXQjtZyadaYbTkfpf6tcs24kv/6rwi4adY5xV7SP3fOPeWce3za92A00P28pAen7YclfTbKezHtu8I597hz7o5z7s7du3dXaDoA4KYQswEcO+ug+Du896/UcmrEW5xzfyM+OL0rXJ8QlPDev8t7f9t7f/vWrVs9WYH1zefLdyjCj+WOcZu4W1x697xa+Wn+3jvozefX85buLFjqW9o+y/nIlWs9fz31hvODG0HMxi5xR7vD/zmEu+iZpk9475+ZHp91zv26pFdL+kKYFuGce0jSs1PyZyS9NMp+Nu0D9scqd4wL2+ljLV1aV9hnLT/OH6dN89by934EaklfOx+5cq3nr7deAEO4dhdCS8wOz6X2tLveOFeqtxT/Sm2zTH2xxHRLPLVcB3L1r3peksdDuIte851i59zXOue+LmxL+luSPiXpCUmPTskelfTBafsJST/ill4j6bmjnk8MAACAg2d5p/hBSb++XGlNp5L+sff+nznn/pWk9zvnHpO0kPSmKf2HtFyO7Wktl2T7sY23et/klqmxLO9zcnL5beLZ7Pq3dJ2zLWdTWyorLSfU0VoOrrQ0T2tJt1o74uOlvGl7LfWXzikAZNiW90ofp23pcv/z241019KkXywrpLlWRmFfMU2p7Fy6UhsN56HYNsv28pElwrAPmoNi7/1nJH1LZv8XJb0+s99LestGWncoNrGcVauMbS2ZlZbL0lwAjhzLe+0jfifYPe5oBwAAgOGts04xcLhWuWPcJu4WV5tGkpZbmzqT5q3lL7W5drzUx1zbLfVaz19PvdxFDxhLb8wOz+O8rfidTgVcNe6vG7Nbx1eJp6tO47ScR1M8j78puZ8YFGNMh3R3Oqa0AMDVlRuwedu+1lybv75/mD6BMc3ny1eulnWD5/Praw2n+1vr/K5TTy5/qQzrOsRpmdayS+W3+lxbazhto6Xt8/nN/80A2K1txOxSmZZ11UMcspS5asy2XE/Wjdm1eF1bP7+n7QcSs3mnGGPK3c5aWj6vrTtZ2h/va63d21tP2I6P1cqwrENcOt4qO21f6bzkzo/1HPa0HcA4thGzw/E0fZquVk9uLflNxexc2p6yS/3K5cmd41Lduduct9p+AHinGAAAAMNjUAwAAIDhMSgGAADA8JhTjDHl7twntZfYKe23LIGzaj1hOz5WKsO65FpapuWuiGkbLEsXxeVbz6Gl7SzHBoxnGzE7HE/Tp+ksy25uI2bn0vaUHaezXmNq/Y/7HC/f1mr7bLa89/GeY1DcYz5fTi4/OZHu31/um82Wy5iEY9Ll8Thdui/N19of1MosHc+1sacsa39jpTZZ2l07B2neWn9rWOYMOCppeMPhaYbufV2SbcTrySp9PoDv2jEo7lH7hmk4FvZZvz2alln7NmtcTque9Hjaxp6yrP2NldpkbXftfFu/EVxjGXAH4RVw/CLg7Oxy4H5xcZlPag/g48F7KLf2wiYts/U8fnERys/1ofXip9a2uN54X1gEPtQTygnnKG3zOvWMeCFCURrenleKGb3xryf2lOJhT8y2xv/cdtwGS1k9z3vOVWff3aIx6A3vQq4Ss+N4E8fsnsdNxezSGztpuy3xsZQ3V09oR+185eqsxfieerh5B7CnLAPuIHeRCQEhXpamdTEqvTCyvEDoed5aYieXbpWlgdLj8XlJyym1ed16AIxjnZidvmFiecGzrZhde5FQytsTs0v15Potla9juTJyMd5azwHgi3YAAAAYHoNiAAAADI9BMQAAAIbHnOIeuWVX0iVZpPoSLq2lXGpLvMTltOqJt3Nt7CnL2t9YqU2WdreWuYn31foLAK0YK60Xs8O+8BgvUxVY4la83ROzc9tSPWaneXqeW64LvTE75OWrAtgxBsU9at92X/Wb8KV82/hm/SbLvIlv/m+zDkvwjtNKV9OFb+DGF8H0YlS6GKQXvtZFMi2z9Ty9UKV15eooXeRq5yg9Hp+XtJx4oBDnXbceoGXTMXYb14GbKO8QYnbr+1jrxOw43sQxu+dxUzG79eKiJw6X8ubqyZWV5m2tv5/G+J56DgDTJ0rm8+Uv8/R0+Zhuz+dX0+XS5LbjfLl9cVrrY7pv1Xa22pMrv9aWsN0qL9SZttHS11p9FqUlZGaz5bdmw6A3XX5msbisJ84XttOlaNJla6Sr5Z6dXX+HOy0z1BWnTdeHjvMuFpfL6sTbabrcuYi/YXx2lj9HaX/u31/2KeSbzS7rTfuQW086ThNL+1jqM1CLG6VY2IozufiVKyuOjT0xcxMxO42hlvot/W7FbGs7LefBGrdry37NZpdxJ8SiNBbn8pXSpftbMbvUxjRml+pKrzVxP9JrS1xGSCtdxt5cX0v9jpeACzE711ZL7C2lOZBPcJ3PLux4s27fvu3v3Lmz62Zc5VzfepJpvlZ655b7V10WxrpOZk87rfW0yi+l72m35fdgaV/JKr8nS99K56iUrvc8lcrv+X309M+aJj6vub6W0vb+bkv17pBz7inv/e1dt+Mm7WPMvvInYfnbymW2xkWp/n/fEzM3EbNXqd/S71J5pf/RUvnG8+Dki//W2d9vrY5NnKNV4/8qv4PS9ip19sZsyzU4fl46V419td/vTSvFbd4pBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB5LspXUluwKx+N0Qc/6k2FZmNIaj9bHuPxV1qgsLTtTOgdx+a21J9Ply0pp0+Vk0nb2rHtpWa6rtM5y7feUpq0tl5Oeo1K6WlmWtKX2t/4OS38vpb/x0jlIl9zJ/b7jY/Fzy3qxrXMAxGr/B6VYWEpfi4e5v/Ha/3mtbZuI2XE+a/3WflvPS62dlvNgWae4FYNacbT3HFniY+v6lKZtnaPW76ZW56ox23INDlpLv9XOQev3uwcYFFvllj85Pb26T7r6PN1O84U80nI5mtIyKCEAhyW10uVR0iVW4vzpP0auzNxSYXFAyPXDKremYXrOQp25enKPYTmvXPtDn+fzzawnehNre+67baznuonygZw0HoZ9uZidiy9pGRcXV/OVxLHx7Gz5dz2f52NbSJ/mXTVmh/6V+t8jrictL43ZaTtz5zIuN5yTtA9x+2v2ZemCfbevMdW1k+wag+KSxcK2jNgqy97kHuO8ufzheC6Ppa5ama32pm2ynAtrPT196Cm3FWBDYLZcGNMA3xq0x+njwXstvVSvN353Kr1Yteqz9CXNa+lnT12WcxQPHmrnLRzf18CP3VklHq4Sw8L/SLhOBGl8jteOzR23xLl1lvWy5rvJmB2fs2vpVRferGm9GVWLc3Ge3hhayld6IZa+EMjFxFYcTa0Ss+P2tOqyXBOtMftanvPr/dkzDIoxptKLnlVeHFgH79YXRaVyahfYVdbOjPenea39tNZlOUdp3aXzluYBMAbrAN4a53pjaClfKSamL55KMbFUX+36YOl/7RrSelHViuPx76R13tI8e4wv2gEAAGB4DIoBAAAwPAbFAAAAGB5ziktKy6NJ7aVS4jS59OljbimUdCmTcDxdHqXVttyxtMxWe9M2Wc6FtZ6ePvSUa1mWDcBxWSUetmJYLra3lh20LI2VtncTMTtXR8+1yVJPz3nOlVtbglHRPGBgBxgU15S+/Zn7Bme8ikD4Zvx8frn0TCx9HsqJ64y/9blYLMucza6uqpBLm6sr14/wrdFSGbVvuPasMpAuR5T7Nm7aznjJovibq3G+2hJuoY6a1prQoVzLiwPr4L2WPqQp1RtfiK0XtJ6+pHkt/eypy3KO4sGDdb1RwCK3RFoau9Ilz6TyNaC0vFt8fD6/GrNStXhXi9lx3lL8TpesTOvMtTm+3qTXo3RlgdDetJzcMmwhX7ocaZw21JE5TVdY34yqxbncwN4aQ0v5Si92etfor7246XmhZLmGtF5U1eK4NWaneVq/3z3AoLik9E3/sB2OpfvT5XhK34DNldmzHE8pf2/ZtfpqdVvLTY9Zvo1bqqOVL01rYV2+JuxLL0Cli2YQLmTpC4tQd7odAm8sfXHk/fUlfnIXrTivpS+li25ufdG4b6UXUKWLZnrOLi6un49c2UCLJSaHdPHxsB0fS4+n+yzxNNTZirGlslvxedWY3UpfiqmtlR1y/WiV27uiTO7FgFR+0yX3Aqj25lEcs3NrTefyBZYXR7WYHJdjidm5FzC5a0Joa64fuXSlmJ2L/XFfT07qL7wOAINijKm2VmbY17qA9i5FFspqvYBJ09UutLXy032r9sWyJmvPwCL3vJSndREHMI5W/LIugWZdiiyUFdedyxeOW94kq71giMuxXH9K7c6VX8tXS1d73orlBxiz+aIdAAAAhsegGAAAAMNjUAwAAIDhMae4pLaUjuVbliFdPI+m9Y3YUnlpnbX8vWW3voFaqttabunbr7Vv4+bqSM9prU5WJgDGZInJlrge1GJSK57WliCzlm1dIcDSl1r9rVUcasuCWuK25fpwIKsT4LgxKC5pLel1U2Ucq1XPzabOaXWtTF1fviZOV8vTuii3Lirxxclyoa2VH5fVuy51vK+1JmvvwCL3vJSnddEHYukqCKldxGSuA3at72K14pflTRfrC43SC6lcvtA2y5tktRcMcTmW60+u3aXyS/lWjdmla1mpzQfyoodBccl8nl8TON1On6dLY6VLbZXWmSyVEafNLetVqsNSRq5P1nbljm+yjCBdnzjX79x5aK1VfOgXKi7uwHXpBdqyRu+qMXFfYnZaTq3+nj7FZZTWls+d59z5yKXL9qXxoqb1oifYZKzaVFnEz/aLnj3AoLiktsZwvJ07JrWXqgrH0sdW2nh/q+5WGbk+WdtVOr6pMuL+tX4fufOQWyw/1vOiJ03TGsxbLoi1i69UvhDm2pd7nrtZgaWuWv40XS6ftP6LpbSs2vnlQoNYT4zoWVKrJ66leWvt2kTM7qm/t0+lJcVyZcZa/c6133v7O8VSO/7G6aR6DCy9AGiVkeaVVo/ZrRdAcf2tF1BxzI/T1l4k1c5RzwvB6vXR+KJmhxgUY0yWFz29L2R6Loi1i2/YF/LWLial5/F6nbU+pHXV8ufSlZ6v82Kp9Lx0fgGMwfoix7J2ruUFQKuMXN5VY3brBVBaf6lP6bnqfeGVO0e9LwSL51d7j9UnAAAAMDwGxQAAABgeg2IAAAAMzzyn2Dl3IumOpGe899/jnHuZpPdJerGkpyT9sPf+L5xzL5D0XkmvkvRFST/ovT/feMu3rbbGcG25ktoaxmF/nCe3tElp+SzLEivhuKWMXJ+s7cod30QZaf9WXfM5LGcDYBw9McISs2vxbF9idlpOrf6ePqVlxEuKpX1K5/f3LvP1/Pb+fxELx63ni3Y/IenTkv7j6fnPSXqH9/59zrn/RdJjkt45PX7Je/+Nzrk3T+l+cINtvlnhW5Pxtylr31IN3/pMv10af8OzJS0jbMff+s/VHddT+hZ//K3U3LdIc6suhHLS5XhK/al9azV3fsJKB+mk/1o/07bFZbaWYwtlW1/0WF7I1C4muXpqF9/QvpC3djEpPY/X68zdrKB1Qc/lT9O11tlc9cVSWlbr/AKp3Dfg4xVVcjE7lYvjJbkVXuL9rZgdp03rbMXsuK1p39OVAdLt2moBcf3x/17cn03F7FBmJYskW8zOxcM4b88LgFYZubyrxuzWC6C4/tYLqPh3FqetvUiqnaOeF4LV6+MBvOjx3jd/JJ1JelLS6yT9hiQn6Y8lnU7HXyvpN6ft35T02mn7dErnauW/6lWv8ntHuvpY2o731fa38rfKXrXuZcjKl1NKU2uftexW/lXT9tRVM5tdtv/k5PIxbEvLNLm0cZrS9myWryOtL1durfzcsbSdrXLSPNY+5MqxPu+pw5pvT0i64xsx9Nh+9jFmX/m3t8bVUpraMWt8buWxpl03Zqf7eq8bq8Tq3pg9bddCtxS1qzdm98SUNI624nfrmmCJfb0xu5S21v9SnF4lZqd1tM7RlK/2+71ppbhtfaf45yX9PUlfNz1/saQve+/vTc8vJD08bT8s6bPTgPuec+65Kf0fd4/YgW2Jl9xJ3+2I9+fSpmlK2610pfS9ZabtLJUfttO+1doZHyuVY3lu7VNvPgDjWDVmW+JiKY7m0qXl5doWl9k61huzS5C6wQsAACAASURBVGlr/Y+16uk5T7UyDzBmN79o55z7HknPeu+f2mTFzrnHnXN3nHN37t69u8miAQAbRswGcOwsq098u6Tvdc6da/nFutdJ+gVJDzjnwjvNZ5KembafkfRSSZqOv1DLL9xd4b1/l/f+tvf+9q1bt9bqBI5EOo8LwN4gZuMaYjaOTHNQ7L3/ae/9mfd+LunNkj7ivf/bkj4q6funZI9K+uC0/cT0XNPxj0zzN4C68/PrH/MAAPYTMRtHZp3bPP+UpPc55/6+pN+R9O5p/7sl/SPn3NOS/p2WA+nDEy8/k/vGaWupsPTbpWl+yzc1N1F37VuzcV/jNLX2WcvuWfonXUUg9KvUz1ZdrEoAjKkWQy3LO1qWDbOuPhNv99RdqyfoidnpsVa+sC+ks6z8sm7MDnkWAnaqa1Dsvf8tSb81bX9G0qszaf5M0g9soG271VrSqzfdNuyy7m1K+7WNfsZBu3TByAV4ywuBOK/lBUNabu0Cmyszt45orZy0b7kl23J9yJVjfW49T735gJjlXUti9uZtKma3ZmKsGrMtcbEUR9N0ufJybQvbpWPWdf8DS/tq/Y9ZXgiu+qZcqYwDedHDHe1K5vPlL/P0dPlY2j49vb6/dmw+v15H2BeeW+rNtcOSr1aXNW/IX9pvOVeWdvS0oZTGIrcmtHQ5WLy4yKdPnZ0tL8qz2fKbuaU1QkvlxGuTzmbLstKbkIT1PONji8Wy72Ed6dpa00Gpb7k+hHxx/rAd+pA+D2lCeffuXZabrsEan6e0jri8nrW+MSZLTMvF5VzsjuNsGq9yMc8SS9eJ2bm6LNeJuO2l/a22xdeu3phda0MujUVvzC5J451z19dXTuNQri3BbLaMdWHgGK8rHeJgvJZwK2aH+mt9C3XGqzyENHHb0/aXYna4xoQy0+tQaEfr2pCWfSDcPkz3vX37tr9z586um3GVc/ZlWaTyMiulY7k6WnWl26uUUUtnzZvrp6WsXJ5121g7nzWl89jbjlYbc2W32mw9t72/w1K9lvaUfn89ZVvOe1y+5RzsAefcU97727tux03ax5h95U/C+ndc2p/uy+3viZWbiNm547l+1Mop9bsVY9aJSZY2TNtOvvhvfe1ffpsxu3UueuuonXfr73+dduTaYG1TqV2l32ul7Nrv96aV4jbvFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAxvnXWKj1vPsizS1f3hee5Y/C3O2tqOljUw03ZY2mtZsqaVN+QP30JN95fKis/HqkvnlNpQSlOyynqhrWPWpWrCOciVF/aFv5Pc+Vn1d1iqN97fKjuktZadrp5hPTe1uqy/Y4zH8v9g+V9oLYeVi3mteoNVY3aurlw/0u3S8os9MbgVk3qW8aotAblQ3U3F7Na5aMWl9BzlrvvW33/rsXb9tYxTcnWV1qGutTGUVyr7QJZkY1DcEi+llW7Hj7k08ZJZQVj2JV6m5OJiuTRNXGYwmy3XfJzP80vFtITyQtmLxbKsuMy4vlb9oe1xmrjMIF0yJ94f+pwGptC/9Dyk51m6bI/lHORsc73QTZZdK+tQ+nAT5QKxUgxJpfvPzq7Gu3A8LHsYpDE7Lqs3ZufifhzzQtxNByrxUltp3+JrTylu5+TaEvaHcxD6GMrP5UuvkdL6MVva3oozm45LxO08106yawyKSxaL/iW70jQhYLSW00m3c0uphPbUyuhdKivXx1x5adq4jjRvLV3uHObaWTvf4ZyWzke6jFJJetFLA7pUfkFUOpZeKEplpy+oao9puaUXZK32hRdpuQu+pc9pneuWVetTWkapvDD4AILeeFOK2ZZ4m5axasyO29e6puT6WCqvdj7iNJZrXXqsJ2ZblnSL21OTDu4t8bMVW3NxrPSCqhZf4/25a0E43nMNaMXZnj6vW1YuZpfOQfEasKUXNRvEoBhjSi9YlhcTqx7rufi12lO72LTaYL3g5/qV1rmJskp9yvW51CcA41j1xUQrzrSuB9Y3mmrp0vZbHq1vAlmObaKs1u8gPp49T+1f8a7xRTsAAAAMj0ExAAAAhsegGAAAAMNjTnHJKkt2pWnCN4DTfGE7HEu348d0iZdaGbXjcf21pcPS8lrLxcXnq5Uudw7jvJbznS4LlEvHcl3AmCzxJhfnwv5WfInzpWWsGrPj9tWuKaU4mSuvFVPjNJZrXXqsJ2an6WvXgYWAnWJQXBN/wzIWL3dT+7bmxcXVY+m3P0PaeP/FxXJ/bjme2az8bdbA8k3YsBxaSFf7Bm5uCbncOUrXniz1N5c3XaYnbW98vnP1pC4uri8Rl2qtFSm1L6jpsfRCUSq7dvFrXQxz6dI6S22wXvBzfU7rXLesWp/SMlov7oBYLi6ky2Omx9M4FsfQ0qoypRVUajE7XUatdo1JY3IcJ0vLvpViaun6lIul8bnKrXQT9zmoxezQ9rSe9PznrhGpVV5MWGJrz9rBoR1p2fH+3LUgbb/l0fImkLXP65bVemFjidmsPnHAepZT6/lGbCi7tAyQpc5cG0rHSs8tSwjl+lTaLu3LlVtrk6WvluWCLAE2yF0w0m2pvMxO2I4vFOlFLzzmLoRxubkXWOmLq1R6LLfc0MXF9Rc4uXylFzRp3fGF3/v6Mj+589nqU659cXksx4acnhVcwv6e1V/SPNY6rfmsMbl0LcnVV9quHW+125Kndr6Da/2SXeuFhVSOrSEGxjExV15u++Qk/6IkThdiYRr3Q7r42pHWmXsTqBSzY7kXcPGx+GZbpZgdp0+lfUrPT659aczuuCzvCoNijCm+2Ei2i4cl8FsvsrkXRdYXYKW2lS7KrfJ7Lsql9rXStepd5TiAsVhja2sgv8ogPvemi2W9/J4XHqW2WGK2pdye2B6sUv4Bx2y+aAcAAIDhMSgGAADA8BgUAwAAYHjMKS6Zza5/OzdsW745nztWWrYs/laopc5wLG5D6Vgpb61dlm/glsqK91mXJmr11bKcT9qWcE4BjKMUvyzfnLfkjfO0rgO1uFvKZ43JpWtJrr7Sdum4pd2tPKvEbOekA1idAMeNQXHJNr/ZXip719+m31b9u+5XTnyxkWwXj00ss5Oro2c961Uu5rUXW70X5TS/NV2r3t7jLMeGnPRLQoeuFTv3Mbauo/VdLGtsrcXs1htJlphkXd4sV3daX3ps1Tfd0ryl9vfE7J52W2L2QnuPQXHJfF5eYireTh9rx+I0uSVd4nTS9fJaeWrLyeSWaomP556X1i9unYvWvrAdlv2Kz3faFsu5S9NYlus6totJzTp9teYd6Xxif4WLcmuJxVKMrcWtnvjbk6e0r5Y32MeYnTu/rZgd0uhcVfv2oueQ494u2t560bMPvPc7/3nVq17l94509bG0XUvXkybdt25due34eXq89rz3XKzSnlXqtpznktnssg3h5+Tk6mNuX3psNrtaXilvqZzwWConLs9aVy69pZxcGXGeUr2tPpfaamlfqYyQZg9IuuNvKFbuy88+xuzn/+1bcSpsbzLG9sSkTexrPd9lzK6lbbSvFrqlqNxNxuxanlrsTuOiJa632tcbs2tx2xrnV43Zlv1JH2q/35tWitu8U4wxxYveB84t94XH3L70WO96laXyQprWDVQsdeXSh/pr5eTKiM9N2od0f6nP8f5cmlr7amUAGMs2YnYtTy12h3xxu1px3dK+kD4tNy6nFftreSx9jctN6yiVWasrbs+eY/UJAAAADI9BMQAAAIbHoBgAAADDY05xSWu5kp5lYVrLZVmXcmnlaS2HEpeZHs89ty7dYjkHuTzxWsJxn1r9sq7hCWAsljgVtkP6+FgtbvXE3548pX21vME+xuzc+bWshX8gS3bhuPFOcUu8RFoQLyNzdracSH52dnUZmtnsMoDklrkJ68ZeXOTLvHdvWUa8P15r1vvl8Xg5nlwbZ7Mp2CyuLutT6lsQgtjp6dVld0rnqHR+4vJCn0KexWJZfnweSvnjY8H5+fI83Lt39dGy1EwIzvFPGqhrF7uwXQr2aZ7ei2euPmtdufSWcnJlxHlK9bb6XGqrpX2lMnjhg5JaXJMuY5ElZsfbIV7FsSpIrwNpzF4sLv+HSvEyjtm5L4blljxL+xXyhDa2zkXahly/emN2WlY4d/N5OWZb4/Y2YnYtTy12p3HREtdb7euN2bW4bY3zq8bsVa9fe453iks29W3R2jfnVymz9xutaTtybWl9u9/ybdRaX0ptzrW7Vk94bA3SVxEH8vgClV4owkVzsbh6kciuuanLtpYG+SFfuHhaLmBx3lzb47pDmXHaeH96QS6dl4uL6+uuhn2hv+maqul5LJ3TNC2wqtZqLJZVT6zxzlpmz0oFllhZ618o35K2FbNzbW6VZ+nXpmwiZsfl1NZSjsV1td7MCS8GSvnD87ju3phdWls6XFPCi6w4NofzEvc37muujppcPw8Ug2KMqXbBqT3vuUiEdKtczFovllptjdtoueD1XOBCG9I64ra1LpytPKU6Dmx5HwAbtOmYnUtnWaqy9qZQrnxL23N118paJWa3XnzV+mxpZ9q2A4zZTJ8AAADA8BgUAwAAYHgMigEAADA85hSXWJc9i791Ga8OkS6BFucJk9l7yyx907PWxtZSbKXtkC9Xf7pd60utzbl2l+qJH+NlgQBA0uzkQs6F2OAlFz/m9oW5jtHx9HkubzZN67gxb/ax0Z5s+Ya0xbpqbTbmMfXnugNZoABHjEFxiWV5mHXSb6LMbdS5bfvS5toam7XntRcZuRcVpQG/9UVS6cVSra2tF2S1ekvpcy9uci8Ca+uQltpZW96n9KKMqycS5/d4sXz0Nh2zc+laS5jFeSwx1HKPgJ71pVv1tZaFW+XNNUs7c9e89PhCe49BMca0L4PzVexD20ttWKVtmywLwPGKV0LYll3GnW3WfRMxu6XxScE+YE4xxjSfL1/Bhp+wIH3P83hfvD2fXy0/TRee5/KHx7gMS3rLsfn8at8t7beUW0rTc056yw99ATCOdWN2Ka60YqM1vm4ibvbG41p87a2z55zUzs8Bx2zeKcaYVl2nOH4u2dalLB1L9+fWkLSsk2k9Fspcdy1Ma93pOaidk1q5tfIBjGPdmG1dA94Szy0xu5XeGs9r5fa20XLNsZyTWt2lthwA3ikGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHl+0w5hWXac4fi5d7qutS9mzzmRubc3WOpnWY9a1MK1rLFvX77Sck1q5tTVFAYxj3Zhdiiut2BjXbblR1jpxs9Q2a8xep864jFXWTq6drwNZp5h3ikvm8+Uv0rnyUiWl9CFPbrt2vLbkSa6c2pIsrXbUlhQ7PT2Y5VM24uREun//cjs2my2/ORvfRS9OX3J+Xh643b9/tZ54OxxbTNEjrTtX1mzWThPKdO6y7LOzfBsvLpY/cd7Q51zf0/7MZtK9e1fbtFhclhkLaWezy3LTeuLyW+cdw5qfXlwLfyP9zE8z/1/Hahsxu6YUB+PHXMwuxcs47tXqk5bl9sTsoBazU6E9cZsWi+VYINSdpg3p7t+/HqMPOGY731gM2zn3VZL+haQXaPnO8ge89z/jnHuZpPdJerGkpyT9sPf+L5xzL5D0XkmvkvRFST/ovT+v1XH79m1/586ddfuyWc7ZlpPKpc/lr6UP26U8ueOtNrbaUetTqaxjYj3Xtd9NbxnpMam8FI61DWG7lj9NH5+DuA2WJXYsfc6l7TlHufa2nu+Qc+4p7/3tXbfjJu1jzL7yJ2GJ2bn9pb/HNO0q8SC3HbcxV1dHXHHy+/IvsRXXfr/rxMveJdFKsc36uG6b45OQtiHd30pX61cpTesc5dqbeb5Pf6OluG15p/jPJb3Oe/8tkl4h6budc6+R9HOS3uG9/0ZJX5L02JT+MUlfmva/Y0oHAAAA7K3moNgv/b/T06+Yfryk10n6wLT/PZK+b9p+ZHqu6fjrnQsvRQAAAID9Y5pT7Jw7cc59QtKzkj4s6Y8kfdl7f29KciHp4Wn7YUmflaTp+HNaTrFIy3zcOXfHOXfn7t276/UCALBVxGwAx840KPbe3/fev0LSmaRXS/qmdSv23r/Le3/be3/71q1b6xYHANgiYjaAY9e1JJv3/svOuY9Keq2kB5xzp9O7wWeSnpmSPSPppZIunHOnkl6o5RfuDku8ZFdpqZJS+pAnzp/OIKktjWI5nqZJl2RptaO2pNjJSX01g2OQ/n5L5zpdliaXJi3DspRNujRb2I7TrLKEmqV98TmI21D72yily9UZ/nYs/xO585trb/qcJdmQmJ1cyLkQt8KXg3T5PDuJL90fnufS+0J5uTzG7SttbLUj3Xf12OzkQstL8RFbN2bn0oe0cR5LzM499sbsNN2uY7blvK0Tsxfae81BsXPulqS/nAbEXy3pu7T88txHJX2/litQPCrpg1OWJ6bn/9d0/CO+tcTFPjo/32567Ld0GbMQZMLyNvGyNdJlUAnL55ye5peiSZfIiZfNka6WG5cT70/zh31hf27JoFja9rhP5+fL5fjSZXhyfTk7u0wfL+F2cZE/Z7n2hLTpuci1J13+aD7n/w7PO7935APCpsH7H+JKKf6k8WQTMTtXrjVmh3SlZd7SOkJZcRuk68fS85GWmWtXGodzbQjbuZgdtGL2AbC8U/yQpPc45060nG7xfu/9bzjnfl/S+5xzf1/S70h695T+3ZL+kXPuaUn/TtKbt9BuYD2LRf/ySelyNFJ7WbR0e5Vya0udWfJZlu+Jy4jPTeuchPQ9dfcci9tTOg5gDD3x1RJPessslZ+Wa43Z1mXZWm2Nj1nan7bLcn2wHAvXgmLM1t5rDoq995+U9K2Z/Z/Rcn5xuv/PJP3ARloHAAAA3ADuaIf1zefLV4Lz+a5bAmBQIQzxc7g/XEKwawyKS9IIW7vFcus2zfF/elyu9VbQq9wGOld/2qdWfuttoMNHJfGcofm8fs7Sfblz1Opv2s5cWQCGsFhIXu7y5+S0/Dy3He+bzeX9Mqz52TxfTlp+q9xWndb6rWWmx+Iy5a4+xvWU2pTWn2uj5ZzHj0mfD2TaKY5Y1+oTQ6nNq0yfh+3cvrCdK7eUPq3DUr6l/rRPpXriPK25QyXpPFNLH+O8lv6m7cyVBWAcvTE73rbE7FL6Wh21+Nx7zai1oVWfJWbX2pTWH5eXy2s5H1K+z8COMCjGmFrLzpS202VpwvHSsmjpdm+5uf2t5YJy5VqWcLMstVNaVij9RnKp7p5jrWWLWJYNGEtPfG3Fk03H7DROWWJ27TFOW2trbjnWWvtz7Wot72Y9li7Hea38zAusPcOgGGNiKa+yfVuOkN8VACn/rnWvbcUT4tSl0rk4gA8DmFOM9aWvvA/BfL5sc27udDofO04bpyvtKz235CvNI6/li+dSh7amfVmn3J59uTa0+lY6Zp3TDqDPIcZsyRbDQmyYz/tjdiuOW+J3T4yP29qKf9a42Rt7S22o9c3S71zaA8A7xVjfIb5Crq1VWXrunH1OYO65JV96vPRYSttq+yrl9u5L27DKHMtWOXwjB1jdIcZsyR7LpPb66fG+WC2OW+J32s5aGbV1fdO218pd9boSl235Lk/ufJbal61fe49BcUnPLQxbt2iMX41b5rKmdbTKt9af9qlUT9gnletr3QY6nWfa6mPuHFnn+oZ2MtcUGFtvzI63LTG7lL5Wh2X+fJq3Vn8rf+mYJWbX2pTWn7s1cM91Mr3GHMhtgHHcGBSXHOO8o5use526DvVdDAC7tYk5p6ldx6Obqn8fYvYBvJOI48ac4pL5fPkK1jpfcz7P57Nsh/y5OuMy0zakZeTqyD1v9SW3P1dX2r7Sj2X+Uu98qdz5yrUNwDjiOFCLNbnjlnhaivm1mJ3WUaqnFbNrfWnFzN6Y3RtvLd9VIGbjAPBOcUnvHJ/S3KCQ1jq3qDQ3KteGtAzr3NZWX1pziErty+XJ9bl0fnvnS5XaHLcNwDhy885LsTw9HvKtMh801F2K2a21jkuxNk5f60vPdxJyfY7LSNuYprHEaGI2DhSDYowpngMnXQbk3Dy38LxnTmDuuSXfKmtY1tayTNu+Srk9+yxrf1rnQ9bKac2PBHB8LDHMun56vC+2zveHcjG3VkZtXd9QjqXcVa8roe5cG2p9s/Q7W84WpjdtGNMnMLb79/P7ZrPLdzIWi+U/dUgbtuPHkC9sz2bSvXuXASfsj+tI94XyZrPrg75afeFxsbhsazh+drYsL+1nWk+pH7lzE46lZYc2SMtzZ/3CY1xX3Kf4WNo+AGNKY0S8P35RfnHRF7OlqzGxFg/Dvjhmp3G2VE+8vVhctjXuw9nZ9baHY/F1JXdOSrFbWpbr/dXry8XFcurK+fllH2plxOcm14ZS/w/kC/C8U4wxtW63Xfq4Mk3T+iiwNE2kVo71o8rWvrAdytrksdZHt6HvpePWj4Bb7QMwDsuUxJCuN2b35M+lreWr1bPKsU1dl8Kx2lSjVn9z5R9wzOadYgAAAAyPd4pLavcxl67PoSnNDQpprXOLSnOjcm1IyyjNk0qfW+aepvtzddXWPy7VX5q/ZJ1nmpZTynMgH9UA2KA4DrTma6bHQz5L/E7LD3NXSzHbutZxKYbX4mYun2W+bS1mp22wfrdi3Zi9ELBTvFPcEub1XFxcfsQQz/EpzQ0K0jlApflJ8XzQeI7o6Wn57l3pvKrcHFXpch5RaEOuL3FbanM34/aFC0E8jzX0uTSnNO5Tbl5Y3O7SvNfZ7Oq6mOn8rtZ8KADHLcTUUqy5f/8yfvXE7HjuZhzz4+tEK2an5YTYlR4P81RDuXGsjfOX2hfXWbqmxN+BiGN2WkZ8zciVG5/X+PyV4nG6n5iNPcE7xSWlJXRy21J9Xk9aXm6+TlpPa05PrR2tNljmSlnqaLUlV1+rzly56fPSUkpx27gNMDCeVjzN7UuP1ZZqS7dzx+J2WGNvyNOK6a0+1dpjibutmF06f5Y+166VIT2wYwyKMabW7bYty9TUPmptTROplbPKMnClj3trS6S1lk+rHUvbWep7bUkky0fArbYDGEctHkn90zxWzR/2hbStfLntUn8sMXtT16VwrDbVKNfmVWP2AbzuYVCMMVlvS7ru7Ut3fYvYXRq57wA2L/0kYFTbui5tO2a77Ra/Ccwpxpjm8+UrWOfqt3S13oa6VMZ83q7LchvZVvssbY+307Jz/Wzd4jvN2+pT/FO7Fbn1VruhDwDGYInZlvi1bswulbnNmF1qm/VW262YX9rXE7Nb5R0A3inGmCzzBcM/sXU+3iprTtbKtMxttNRba3Ptdtu5snPHrX2Kpft6ykjbAGAMlpjdG79yZfTWdRMxu5Q/zVN7jNvWuv6sErOb50h7j0FxSWkJndy2dP1Yaw5PnCZmXdKm1o5WG3rmpPbOKYrbmauvVWdabq7PpaWU4rZxG2BgPK14mtuXHqst1ZZu547F7bDG3pBnnfmtvdeFOK01ZpfOn6XPpetXyH92dhBzTnHcGBSXbHpuzT7Mr7zpNmy7vn04pwD2xybnnG4yvhxKrNp1nw/gnUQcN+YUl8zny1exzvXNA3WuPt+ydw5R6Xg6zyhuc2veU608y1ymdeZ/5s5rrS+Wc5I7r6EcAOOoxYta3IvjjjV+9sbs3ni465idltET24nZOFC8U1zSmnNa2+fccn9t3cn0ea68Wn25cnruh14qr9aOWhpL+0vnNVePZZ3o3PkOx1nzEhiPdY5j2A554riTK8cSc1a5RtTi4a5jdlpGrZ9pX1rnhJiNPcWgGGOyzBe0zsdbdc3JVpmWuY21stO2x2kst9vOlZ2bH9gz3zHombNYOs5axcBYLDHbEr/WjdmlMrcZs0ttS/OUHldZOz9midmtc6QNTm/aEqZPYGwhqJTkblsa8sW3VA230vb+6pf8Fovrt5GN80tXb8Md31o1vXV4XGfcvlw/zs7Kbc4JbYhvqV06Hycn129LXkoXi28lGx+vtavUbwBjssSDEP/SeJPeHjvEJEvMjuWuCSHvtmL2bHZ9nnYuTyqO2XHbajE35EljdmhTGrPj25UfeKzmnWKMKZ5qIq0/9SN+Bd3zMWLYVyq/VFfvlJpcmbnzYF2qLrdtaXdaZ275n1Z/c+ccwPGzxMFSPLbG3ric0vY+xOxWnnVidtguLTEXtzsuJ+wv9l97j3eKAQAAMDzeKS5pzTmt7avNtwz70+fmOTmZcla5H/o6c5nWmf/ZWq85l651TnLnlXWKgfFY5ziG7ZAnjjsXF7b42arPco2oxcNdx+y0jFK6TcbshYCd4p3iTQrzMqXr36QtzWOyzKWUrs7TSeckhSAezxWK5/Tk5mzGc4bu3bucf1UaTF5cXPapNK82bl+6vVgsl90JZZTm1IZ5T3E6i7hNZ2eHsy4ogN2JY3Ycd1Ihbrfm3NdidhwnQ8wO8/Jr5cRtaMXsuM5azM7VlYvfcRnp/NYwl7UUs0tlpuev9Z0C4AbxTnHJqkuylfJa5gbV8uTSSPa5Va05Q9Z+x1rzpXrmOKV1l+qw1MvSPsCYeufDW27LW5sD2huzazF5EzG71pZWmzcds63nL+03sEMMijGmeKqJtP7Uj3h5sJ6PEUP6Uvm5PKtMqUnLTI9b21DbbrXbsiSQpb+5cw7g+FniYCkerztdL1f+LmN2K886MTtsl5aYi9sdlxP2F/udvIDaQwyKMaZVpldY86w6dWMbUz5aZabHb2LaCVNbAKwifVfawhJvDjlmb6sd2yj/AD4MYE4xxjSfL1+5rnLL7Hi7davqWv60rlJ5q94aPNeXOF2tzNajpe1pXa1bzfae+5AewBhWjUWtfZuI2ZZ6V4nZtVhaKrN0nZjP621fN2a38hwA3inGmGrrFKfHe9ahTMtIy7bOZ6zVG8q0tK3W1yBXpuWx1fa4Ta3zt+q5BzCOVWORZV8oI9QTHw/b1jnotTp6Y3YtlpbKrF0nam2Ij1n71nM+DiBkMygu9xiIbQAAHh9JREFUWWVJtnWWHWvVkUsjXU1XKiOky5WRrjZhmVOatsc6h6m17E86r6uVt3SM5diAMbVil3XOZC1OrROzazF5EzG71pZWmzcds3PbtX7PZizJhp1j+kRLvFxMbumYeF+8NFq8FE28lM/JSf5WkGl58ZI3cVnxdqp0u+Bcn0IZoc3h443F4uqScWndcVuttwY+O7t+m8i4LfFybeFYnCbte1peXFbo03xe7j+A41WK2bntdDnLOP6mMSu+/W8uZkuXsSnen8bQsDRnK56V+hXaHOJ2HLPjL5ilt1IPdVmuEevE7NIt3uMlSONzEy/bCewY7xSX9H58nu6T7B95545bP86xfIxUakutH6W6c89zH8+3PrLJ9bf22Dq/aR4CLDAeS8yOt2uPobxS+lJ96X5LG9K6WnX3fvzfukZsMma3zlf1/Od/rcBN4Z1iAAAADI93ijGm0jrFlltmx9utOeO5W8aG7dp8Rsu8vFbbcn2J+xpY1pfMzf9rtd06v3LVc886xcBYVo1FrX2biNmWekMfSvlWWcs9V2bpOtGaFx8fs/St+/tRXvuOd4oxtnReXXo76lQ6VzzM+07nHs7ny7Udw/Gc+PaouXmHufnc8XzE3C1Xa33JzcGO96fz/HL9S9se5ljG0val8wtzc/OlqxejeK46AARpfMl9R6f0fZM0VsbzmUPsscbsuNzSd2DStpRidtwOa8wubYeyc/Fzschfb9L2la49qTRmp9fCA7x9N+8UY0ylJWTS55Z5dqX5eWldpXJLdbbqrs3tXqdvlvmHlvJ65kTmzp1lmR8A4yjFoXifZW5zLbbEx2uxtFT+KjG71P5W30L51j6k5bXOm6UfXTFbe49BcUnvx+fpPin/kUUrfe/HOZaPkeK21D5uyX0kktad61e6LI/lI5s0XesxbmvpI6D4kXcZgfFYYna83ZoeVEufqy8X663TgKwxepWP/3unGqwTs2v1Nvu0/x+v47gxKC7Zl1vRbvN2xJsqZ1/OFYCxpe+IjebQY/YBvJOI49acU+yce6lz7qPOud93zv2ec+4npv0vcs592Dn3h9Pj10/7nXPuF51zTzvnPumce+W2O7EV8/nylavlNr+l2zXmyqiVk3vM5SvdbrJUZq2dufa2yqmdg1KeTbe3dcvMUB+AcZRiy03E7Fa+UpxtlVW71XopZlvKtNS5ifb2XiOAHbK8U3xP0t/13n/cOfd1kp5yzn1Y0o9KetJ7/3bn3FslvVXST0l6g6SXTz/fJumd0+Nh6b3VbG1/nK9WTmluUprPufztJnPpeta+jI/XyrH0NU237faW6gMwjlrs3XbMTtPlyk7XEc6ls14bajHbUmbrHJXq6G1vOJbWR8zGHmoOir33n5P0uWn73zvnPi3pYUmPSPrOKdl7JP2WloPiRyS913vvJf22c+4B59xDUznAfigtIZM+ry1ZU5ufF397uTU/vVRnre7asji99aTlW+YfWsor9cc6x7K1zA9LsgFjKcWheN8q37tZJZbmyl81Zpfa3+pbKN/ah7S81nlr9aM3Zh/AnPGuJdmcc3NJ3yrpY5IejAa6n5f04LT9sKTPRtkupn1pWY875+445+7cvXu3s9nAhoRlY8I/fvo83Y5v41q6Rax0uXTPfH5ZV+n2sbX2xPtyy+OcnFxfxie+lWut7NZtw+Ny41t/9wh1xrc6j+X6mruFbG7ZoX2dF3mkiNnYC7mYLZWXs4xjTxqzU2E5zaAWs+O60kdLzI7bUVpWs9W3nNDHELfTuJu2pbZ8XW45ztwSdBcXV69Bpd/DATB/0c4591ck/VNJP+m9/xMXvfLw3nvnXNdLAO/9uyS9S5Ju3769/y8fcFzS6THWj0RrHwOmH52m6UvPWx+NlurP5Y/TtG75mutb6+PWdPqLtaxSf3J19PwecGOI2dg5a6zunbqRxqOQvvS8Z1pLKX9allSO2bW+Wa9Xcf82GbNbU5IOLGab3il2zn2FlgPiX/be/9q0+wvOuYem4w9Jenba/4ykl0bZz6Z9AAAAwF5qvlPslm8Jv1vSp733/yA69ISkRyW9fXr8YLT/x51z79PyC3bPHeR84p51iqXyHJvcnB7LfMvWvKLc7SZz6VrzgkrrSNbKqZ2DUp5Nt7c1z4l1ioHxxLGlFLu2FbPTdLnnuTibK7NWjyVmW8psnaNSHb3tDcfS+nIxeyFgpyzTJ75d0g9L+l3n3Cemff+tloPh9zvnHtPyT/lN07EPSXqjpKcl/amkH9toi29aekvJWGluTpgzlO4PAab2LdvSfFLpcqC3WFxfoL3Wzly5ab7cXKB4HlDa5rRd5+fLuVildKG9JyfXy8y1Ld6fu01mvG3pP4Bx9MZsaRmjcrdQl/IxOzefNC03jtm1OFu6VXCtf7lrTKvNQS5m5+bGhnaHOfwhfanv8XnJnav0vIV6gT1hWX3i/5SKS2q/PpPeS3rLmu3avdbtF2tzaHL7c3ksZVv3W+a4rlpWqZ+5OayWNuWWJWrV0+pLqQ4A41g1Ztfmm8Z54nylMtI6wvNavlJd1jribeu8VKk/ZlvS95y3OC0xG3uAO9phTOn0GOtHorXbuOamvKQfP1pvd1r7eLOWP66jdcvXtG+146VbelvLsnw0bfmINe0jgHFYY7U1Zsd5gnVidhrHQvpc/ni7FbNLfeu5XgWbjtmlPud+DwfwuodBMcY0wnJe2+jjCOcNwH5K30U/NsceX107ya51rVMMHI35fPkK1rn8bahzt0Odz6/nS7dLt49N60zLjvOtWk8tfauf1jKtt/6unc9WfXH/a7c7D3UAGEMuBqwbs3Px6aZjdi1dri2t8leJ2bV+tMqo9T9+fgB4pxhjSufGpXPmnMuv1Ri2Qz7rGpJpneF4XHbvupa1dTZb8w2ttzFPy2zVX+qzNV/tvOTaAmActe/wrBqz0/J2EbNr6WrzuUv5Sn3qib29ZZT6f6Uc7T0GxSWt2y/W5tDk9ocy42OWsltzo3JllOa4WuqozSsqtddyy8r4MXwjO21brZ5WX0p1ABjHqjG7FGt6lz1L67DG7Fz70vos15hcfLV+R8ASsy3pS32pzWWVWJINe4FBccmxz+3ZBs4ZgF069jmnm9Ybs7cd4w/gnUQcN+YUl8zny1ey4Wc+r8+1KW3XjpfmYrbqmc8v22iZ2xnKK80PSudgxf1vzVNqlZdLm9bROifWcxw/hjoAjCON2VL/HMneeFOK2WmauC2t7yD0xOxSnl3G7J56idnYI7xTXFKbSxSeW+dW9s7dsdQT2phLWyovLauUJu5/XGarX60+hO3cHKlNnOP4kTUvgfHk4knP2rqrxpu47loctsyx743ZcV/3JWb31EvMxh5hUIwxpXPj0jlzufl7pfnV8XbvGr+hvNo8xp51KUvpW/2s5cvV28qX9tmaL+1/2o5cHQDGUIoz68TseHtXMbuWrjafu5Qvt7839vaUUev/lef7P72J6ROAtHyXwrnlwumx9DawuVurhu3Z7DKAxcdD2fE7IeF4ekvt8I5KXM/Z2XL/bHb99qpp3jhNaH+8HRaGD7dvDXlDvpy4vlyas7PLoBjfHnY+v3oeQztK/Qr7F4ur7UvzhTYzhx0Y22JxGbNzMcoSs72//sVsa8yWrt9WuxTbwrEgjtlp+2sxW7oec1OWmJ27XtRYY3bcl7gtB/ImBu8UY0w9y9v0fKwq5T/CDOlyx0sfsYZjPcvF9Sy1VvrotOecpP2qlZu2d5WlhtKyAYyjJ6b1xOzadMFQb0/8K8W2uCzLrbNb7V43ZteuKaXzVGqz5XdxAHinGAAAAMPjneKSdC6RZb3ennlEpbUme9cFbt0nvXbf99ocrNyc01a/cuXl2pWbI1U7J9ZzHD+yTjEwnlzM7llbd5V4Y5lP2jPHvjdmx33dl5jdU28csxcCdop3ii3CnJkwdymep1OaY2OZe5mbyxPm5uTKCWkXi+USNotFeT5RaFc87zJNW5qDFT7amc2ke/cu25zrV27eZ1x26Hd8juL2hzShrtpgNjffKjcXqjQ3FsAYQswOcSbsS+N27jsAsdLcUOnq/PnavNVcW3JxNI7Z5+f1OtP9Ye5sGrNzeS0xO51vm4vZ4fzk6kvntBKzcSB4p7ikZxmfUvrcttReLic9lstbSx/PV0rbVmqfpW+W+UKl5XYs59R63kvnJbcfwDjWnVPZSpc7vkpsTdOnMdtSZ2lfK284T7WYXTtHPWmI2TgwDIoxpp7lbawfq7aWEgtaH5uus1xcz1JrpY9/e85JqDPOX/tYOe7/KksNpWUDGIc1pvXG7NJ0wbjenvhXim1xWZZbZ7favW7MTvtdm74p2WN2rg2z2UFMj2FQjDHt83JepbZZ2rzP/So5xDYDuHnpO+2bsq0YtO8xe9W6V813AB8GMKcYY5rPl69enSvfjtR6e9KQZj6/LLt169NaHZb65/P2rWBL+2p1rXKL2rTsVvvS47k2lZ6HtgAYTytmrxPbdhGzrdeWWtpSe1oxOC2/1b70OmCN2XHeA8A7xRiTZW1IabX5zZbbb9fqsNYfjrfaVFp/0zLHLzdPsVZfXH6pfbn+pW0qPedWsMC4rPOqV4ltufJz27V0tTZa2tIzBz0ts9ae2vmKy7aeK8uc+lzeA8CguMS6jE9t2Z90uzYPKDeXKS0nzltLH89XSttWap+lb9YlfXJziVrntGf5pNJ5ybUdwDhKsSPeJ9nn4PfE99IxS4xPY7alzp7vJZTiZbxtjf89aXpjNq95sWMMikt658zs87zIddu2qb7t+/wqAIetd86pNd7sIi6tU+cm23uTcftw3lDEkWJOccl8vnwF69xq80jn8/Lcm9w8p565qfHxcKxUlnWOUy1/rbzQFgDYNUssXDVm1+JhK2bnyl81Zod6WvlLx4jZQBHvFJessk6x1D/3qLfM0tyeXDtz9VrmI5XamiuPOZ4A9kXvWr5pnlwZubyrzk2Nj7XamTuexv5WP3PHiNlAEYNijMmyNqRUTtOaX9e6/XatDkv9llvBlvpWq6tnLc/aXMla+9LjuTaVnnP7bmBIs5MLORf+/72WUy3CY2WfZEgbpyulNaTLHsvV0WjLtTQrtvtafaV+rdK+QhnZ58vHQ/iqD4NijGmbcwSPdV70sfYLwN47v8cLYmwfc4oBAAAwPN4pLlllSbb4I/PS7RLDdji2ysfwuWV3cu3M1Vv6CL6Wv9YPPs4GsAf6P17PfPScPs/lzabp/Ti7lN/Q5uZH7Y1j19qwPw7h43UcNwbFJfuwJFurTD7OBgBJfLwOYH1MnwAAAMDwGBQDAABgeAyKMab5XFcWtM/dPKWVp7TdWrA/zddTFgAMaH56kQ2j+/wzP73Y9WlDJ+YUY0zpzVmk689beSwL9KfPYyFfb1kAMJjF/bPLMJrGzjhuxkrxNBdz48eQrlVe44Ypl1/8xKHgneKCt73tbXLO3fjP2972tl13HQAAYDi8U1zwtre9jQEqAADAIHinuFeYH1qbB2o5fixpmecKYM8d4nzUbf0wzxUo453iXmFeqTPOLSodP6a0ALDHFvfP5LWHsXMHaZnnCpQxKMaY0jsWSpfPS7dVyt3lMLedKyv34sFyB0JLuwDgyNXvWBjftS9Wuptfz50OK+XVynDLNku8CDkkDIp7pbdWxmFa5W6A3EEQAHbiMO9YeIhtHhtzinudn+eXagEAAMDBYlBcMp9ffjMh9wUzqb7dOn7IacNH/M7xkT6AvVD7Mp0kOfnLfa3tI04r5c+R5Wc+39mvF7gRTJ8oyd2ooePLDEedFgD2TPbmDvH2PsTOXabdgDCoBo4V7xRjTPP55dsf69yy2bp8XVpWqdywP7wlE/andefKq9VZWnovraeUr9W/3L5S2bXH1jHesgKGxLJ6h/1zKEsB8k4xxrSpWzbX8tZuHVqqJ00fLwFYa0utjMztR5/Ps1jkz0ear9W/Wp9rn7pYy0nf7QplAxjCtU8CpHzsCPtzzxXlb8W3tPxSjCqls1wDannTtNayWv2z9K1Vd8+x50/5YXzpkHeKAQAAMDzeKS7JrUlbW092hLR8qQ7APtvX2LmrtMRsoAvvFG/L/fuXN1/YVtpWvk2nvbhYBlvn1r9V9Hxu6y8A3LRjid+tmN0bv4Ej13yn2Dn3S5K+R9Kz3vu/Pu17kaRflTSXdC7pTd77LznnnKRfkPRGSX8q6Ue99x/fTtO3bN3VJ1bNt291bDMtAGzSIcXWm6hj42nrpx84dJbpE/9Q0v8k6b3RvrdKetJ7/3bn3Fun5z8l6Q2SXj79fJukd06PwH7Z1C2brR9rpmXlPuKM04d9oZ1p3bny0jJK++I8Z2f585Hma/Uvty/tQ+m8tMqJjwV8LAyMJ40DudgRp8vF7JCuFd/S8nMxeza7fDe+FLPDc2vMrl0v0thaux7U+mfpmyUeW449L/my455qDoq99//COTdPdj8i6Tun7fdI+i0tB8WPSHqv995L+m3n3APOuYe895/bVIOBjTo5WX7cGG+HoBBWZgj7F4urH2GG/XEZuf3xY3B2trw74ny+DKpx/lD36ellvrju2ewyb9if+/g0riP0JW1jXE/pHIR9uXMTBtWLxfW+pn2I+2j5yLn2uwAwplLMzsXEUsy2lJ2mS+NRGjdzMTtOE8fKXB9a159YfM0o5cu1Wbr6PI3NJyeXZZdidnqtqZ2jUvv32KpftHswGuh+XtKD0/bDkj4bpbuY9l0bFDvnHpf0uCR9wzd8w4rNAFYUL3VWW7Im3l/a7n0MeUM7WnXlHnN9SPOWlluzfFQan4Ncu3uX76mdg1Z7SuXjRhGzsXOtmF2L62n6deNRz7WgN29I12p/7hysGrNLbVzlGpk7diAhe+0v2k3vCne/L+69f5f3/rb3/vatW7fWbcbmhY88ch8NhF96ery03Tq+atqbqGNbafn4Gzgo+x6zZyfRzR3WvW3yqvn2rY4Np5UuQ7n1Zz7f6Z8F0GXVQfEXnHMPSdL0+Oy0/xlJL43SnU37Ds/5+fLVjffSvXuXj9LlK6Cano8MVk3b81HQvqXNfSva8u1nIiyAjPN7Z8+H7PRHkvxyeLf8OTmtb7eOk9acNp25BeyzVadPPCHpUUlvnx4/GO3/cefc+7T8gt1zRz2feG+/IXzkaQFgFYcS444qbfnXAewby5Jsv6Lll+pe4py7kPQzWg6G3++ce0zSQtKbpuQf0nI5tqe1XJLtx7bQZgAAAGCjLKtP/FDh0Oszab2kt6zbqL0WLy+FwxUvkxN+n7XlbmrbvY+h/vAYppJYy88t9ZPrS2m5tVK/S+egtlyPdfmeUr25OizLAzEnHRhPK2bX4nq8vYl41HMt6Mlruf5sI2aXzkGp/lIZxfK7v3q2E9zmudf5+fKRgfFhC7/HXVunHda8+9JXAFhH6/s8xLr9dSBDJm7zXDKfLwe+zuW/CCbVt1vHc6/ELNujp93UO4S532/pdx1vW9LFj/P56vXVbo89n9e/jBjX1yo318ZwrPblx9wXIGv/N630tfOelhHyA5P56cW1P/nwI+lgVng4trSzk4vN/ZJ744Ql/qWPrdi0TsyuxcO0vlodm2hjroxtxuwD4bxlJYUtu337tr9z586um3GVc9v/ggJ2Z9XfX5q/ljcEAsvfQ+++uOxcmfGxVptzbcyVU6vD2uZa+p7fyx79HznnnvLe3951O27SPsbsK38SG4zfe/SnNrTm7zfe3nTMLqVdJf614nCp05tu4w3H7H37PyrFbd4pBgAAwPAYFAMAAGB4DIoBAAAwPAbFJa3bPK/6pTGWkgKAjdvobZ639UUxAHuNJdlKWNrluOXW7o23S+tUBq01HMNjba1gaznxPsv6xuk6ka0259oYjsXllNazjOusrYlcWqc5rdNynnhxicT5vbMtlbytctFj+aIn/C68Lpf4ym3H+2rH0n3hS2K5MjJpq/WW0hfSPl9vrs5ttLEnTWlfKc319MsXl/v/v8SgGGO66Rc9m65vG+3fRJm9ZfDiE4DB9l704GYcxu+P6RMAAAAYHoNiAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGx6AYAAAAw2NQDAAAgOExKAYAAMDwGBQDAABgeAyKAQAAMDwGxQAAABgeg2IAAAAMj0ExAAAAhsegGAAAAMNjUAwAAIDhMSgGAADA8BgUAwAAYHgMigEAADA8BsUAAAAYHoNiAAAADI9BMQAAAIbHoBgAAADDY1AMAACA4TEoBgAAwPAYFAMAAGB4DIoBAAAwPAbFAAAAGB6DYgAAAAyPQTEAAACGx6AYAAAAw9vKoNg5993OuT9wzj3tnHvrNuoAAAAANmXjg2Ln3Imk/1nSGyR9s6Qfcs5986brAQAAADZlG+8Uv1rS0977z3jv/0LS+yQ9soV6AAAAgI3YxqD4YUmfjZ5fTPsAAACAvXS6q4qdc49Lenx6+ufOuU/tqi0b9hJJf7zrRmwIfdlfx9SfQ+/LbNcNuAnE7INxTP2hL/vpGPqSjdvOe7/RWpxzr5X0Nu/9fzE9/2lJ8t7/D5U8d7z3tzfakB2hL/vpmPoiHVd/jqkvozim39kx9UU6rv7Ql/10TH1JbWP6xL+S9HLn3Mucc18p6c2SnthCPQAAAMBGbHz6hPf+nnPuxyX9pqQTSb/kvf+9TdcDAAAAbMpW5hR77z8k6UMdWd61jXbsCH3ZT8fUF+m4+nNMfRnFMf3Ojqkv0nH1h77sp2PqyxUbn1MMAAAAHBpu8wwAAIDh7XxQfGi3hHbO/ZJz7tl4OSLn3Iuccx92zv3h9Pj1037nnPvFqW+fdM69cnctv84591Ln3Eedc7/vnPs959xPTPsPrj/Oua9yzv1L59y/nvrys9P+lznnPja1+VenL3/KOfeC6fnT0/H5Ltuf45w7cc79jnPuN6bnB9kX59y5c+53nXOfcM7dmfYd3N8YlojZu0PM3t84JxGz9+1vbBU7HRS7w7wl9D+U9N3JvrdKetJ7/3JJT07PpWW/Xj79PC7pnTfURqt7kv6u9/6bJb1G0lum83+I/flzSa/z3n+LpFdI+m7n3Gsk/Zykd3jvv1HSlyQ9NqV/TNKXpv3vmNLtm5+Q9Ono+SH35W96718RLeNziH9jwyNm7xwxe7/jHDF7v/7G+nnvd/Yj6bWSfjN6/tOSfnqXbTK2ey7pU9HzP5D00LT9kKQ/mLb/V0k/lEu3jz+SPijpuw69P5K+RtLHJX2blguMn6Z/b1qujvLaaft0Sud23faoD2daBp7XSfoNSe6A+3Iu6SXJvoP+Gxv1h5i9Xz/E7L2Kc8TsPf4bs/7sevrEsdwS+kHv/eem7c9LenDaPpj+TR/ffKukj+lA+zN9dPUJSc9K+rCkP5L0Ze/9vSlJ3N7n+zIdf07Si2+2xVU/L+nvSfr/pucv1uH2xUv65865p9zyrmjSgf6N4Wh+Pwf/90fM3rs4R8xe2qu/sV47u83zsfLee+fcQS3p4Zz7K5L+qaSf9N7/iXPu+WOH1B/v/X1Jr3DOPSDp1yV9046btBLn3PdIetZ7/5Rz7jt33Z4N+A7v/TPOuf9E0oedc/8mPnhIf2M4Pof490fM3i/E7OOx63eKn5H00uj52bTv0HzBOfeQJE2Pz077975/zrmv0DK4/rL3/tem3QfbH0ny3n9Z0ke1/LjqAedcePEXt/f5vkzHXyjpizfc1JJvl/S9zrlzSe/T8uO4X9Bh9kXe+2emx2e1vPC9Wgf+NzawY/n9HOzfHzF7L+McMfvSXv6NWe16UHwst4R+QtKj0/ajWs7zCvt/ZPp25mskPRd9/LBzbvn2wrslfdp7/w+iQwfXH+fcrendBjnnvlrLeXaf1jLQfv+ULO1L6OP3S/qInyZE7Zr3/qe992fe+7mW/xMf8d7/bR1gX5xzX+uc+7qwLelvSfqUDvBvDJKI2TtFzN7POEfM3s+/sZXselKzpDdK+r+1nEv03+26PYb2/oqkz0n6Sy3nzjym5VygJyX9oaT/Q9KLprROy29q/5Gk35V0e9ftT/ryHVrOHfqkpE9MP288xP5I+s8k/c7Ul09J+u+n/X9V0r+U9LSkfyLpBdP+r5qePz0d/6u77kOhX98p6TcOtS9Tm//19PN74X/8EP/G+Hn+d0rM3l1fiNl7GOeSfhGz96Afq/5wRzsAAAAMb9fTJwAAAICdY1AMAACA4TEoBgAAwPAYFAMAAGB4/3+7dSAAAAAAIMjfepCLIikGAGBPigEA2JNiAAD2pBgAgL0Aommt7r7HOQoAAAAASUVORK5CYII=\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The visualisations above appear to accurately represent the structure of the first three pages of the PDF with the `LTTextBox` objects representing groups of text chunks inside the blue rectangles.\n",
        "\n",
        "For the purposes of this project we will extract the paragraph text chunks only and omit other content such as abstract, section headings, tables, figures, footnotes, acknowledgments and references."
      ],
      "metadata": {
        "id": "FE7yEqj31N11"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 5. Extract text from `LTTextBox` objects\n",
        "\n",
        "Function to extract text from `LTTextBox` objects  using get_text() method."
      ],
      "metadata": {
        "id": "NOlfWoz36_n-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def extract_text(text):\n",
        "    \"\"\"Get text from LTTextBox objects\"\"\"\n",
        "\n",
        "    col_texts = []\n",
        "    col_text  = [textbox.get_text() for textbox in text]\n",
        "    col_texts.append(col_text)\n",
        "\n",
        "    return col_texts"
      ],
      "metadata": {
        "id": "j7NfDb1S6rqD"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Extract text from page 3 of the article visualised above."
      ],
      "metadata": {
        "id": "H4gIIRLhLcEU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "col_texts_3 = extract_text(text[2]) # page 3 of article\n",
        "col_texts_3"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F58IXbk26tdh",
        "outputId": "29018d1d-e6f1-4c4b-a34b-43280ca95a71"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['Figure 2: System Architecture.\\n',\n",
              "  'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n',\n",
              "  'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n',\n",
              "  '2.2 Comprehend Medical\\n',\n",
              "  'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n',\n",
              "  '2.3 COVID-19 Knowledge Graph\\n',\n",
              "  'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n',\n",
              "  '3https://docs.aws.amazon.\\n',\n",
              "  'com/comprehend/latest/dg/\\nhow-document-classification.html\\n',\n",
              "  '4https://aws.amazon.com/comprehend/\\n',\n",
              "  'medical/\\n',\n",
              "  '2.4 Topic Models\\n',\n",
              "  'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n']]"
            ]
          },
          "metadata": {},
          "execution_count": 50
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2dfLT0ZK9L3d"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-30_2007.09186v3_extract_textboxes_updated.pickle', 'wb') as f:\n",
        "  pickle.dump(col_texts_3, f)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We will now perform necessary processing in order to extract text from the whole article."
      ],
      "metadata": {
        "id": "nYq3oJ9BP4aR"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 5.1 Flatten list of `LTText` objects\n",
        "\n",
        "Flatten nested list of `LTText` objects for whole article."
      ],
      "metadata": {
        "id": "xj9oMJKVFeh6"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "flat_text = [item for sublist in text for item in sublist]\n",
        "flat_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "t49tdRUeEaK6",
        "outputId": "3b60a02a-5615-43dc-cb8d-f8327cacbef6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<LTTextBoxHorizontal(0) 103.371,746.071,493.659,776.357 'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\n'>,\n",
              " <LTTextBoxHorizontal(1) 77.083,639.527,523.454,734.427 'Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\\nBen Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\\nRamesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\\nAmazon Web Services AI\\nparmib, liuall, arumae, nimpourd, suyogd, snivelyb,\\nmonamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\\nzhiheng, bxiang, tahak @amazon.com\\n'>,\n",
              " <LTTextBoxHorizontal(2) 158.891,605.891,203.376,617.847 'Abstract\\n'>,\n",
              " <LTTextBoxHorizontal(3) 88.261,322.925,274.910,595.902 'Coronavirus disease (COVID-19) has been de-\\nclared as a pandemic by WHO with thousands\\nof cases being reported each day. Numerous\\nscientiﬁc articles are being published on the\\ndisease raising the need for a service which\\ncan organize, and query them in a reliable fash-\\nion. To support this cause we present AWS\\nCORD-19 Search (ACS), a public, COVID-19\\nspeciﬁc, neural search engine that is powered\\nby several machine learning systems to sup-\\nport natural language based searches. ACS\\nwith capabilities such as document ranking,\\npassage ranking, question answering and topic\\nclassiﬁcation provides a scalable solution to\\nCOVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high\\npriority scientiﬁc questions. We present a\\nquantitative evaluation and qualitative analysis\\nof the system against other leading COVID-\\n19 search platforms. ACS is top performing\\nacross these systems yielding quality results\\nwhich we detail with relevant examples in this\\nwork.\\n'>,\n",
              " <LTTextBoxHorizontal(4) 72.000,300.286,77.978,312.242 '1\\n'>,\n",
              " <LTTextBoxHorizontal(5) 89.933,300.286,154.814,312.242 'Introduction\\n'>,\n",
              " <LTTextBoxHorizontal(6) 71.182,76.568,292.082,290.715 'With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n'>,\n",
              " <LTTextBoxHorizontal(7) 307.276,389.247,527.359,616.943 'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n'>,\n",
              " <LTTextBoxHorizontal(8) 306.883,101.737,527.356,385.118 'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n'>,\n",
              " <LTTextBoxHorizontal(9) 319.928,77.606,425.632,87.419 '1https://cord19.aws/\\n'>,\n",
              " <LTTextBoxHorizontal(10) 16.340,526.960,36.340,566.960 '0\\n2\\n0\\n2\\n'>,\n",
              " <LTTextBoxHorizontal(11) 16.340,478.080,36.340,521.960 't\\nc\\nO\\n7\\n'>,\n",
              " <LTTextBoxHorizontal(12) 16.340,461.420,36.340,468.080 ']\\n'>,\n",
              " <LTTextBoxHorizontal(13) 16.340,448.080,36.340,461.420 'R\\n'>,\n",
              " <LTTextBoxHorizontal(14) 16.340,413.100,36.340,448.080 'I\\n.\\ns\\nc\\n[\\n'>,\n",
              " <LTTextBoxHorizontal(15) 16.340,237.000,36.340,403.100 '3\\nv\\n6\\n8\\n1\\n9\\n0\\n.\\n7\\n0\\n0\\n2\\n:\\nv\\ni\\nX\\nr\\na\\n'>,\n",
              " <LTTextLineHorizontal 16.340,403.100,36.340,408.100 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,408.100,36.340,413.100 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,468.080,36.340,473.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,473.080,36.340,478.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,488.080,36.340,493.080 ' \\n'>,\n",
              " <LTTextLineHorizontal 16.340,521.960,36.340,526.960 ' \\n'>,\n",
              " <LTTextBoxHorizontal(0) 82.729,766.636,101.138,774.003 'Query\\n'>,\n",
              " <LTTextBoxHorizontal(1) 181.035,766.636,201.485,774.003 'Article\\n'>,\n",
              " <LTTextBoxHorizontal(2) 279.865,766.636,308.101,774.003 'Response\\n'>,\n",
              " <LTTextBoxHorizontal(3) 81.749,742.699,166.567,758.658 '“Are IL-6 inhibitors key to\\nCOVID-19?”\\n'>,\n",
              " <LTTextBoxHorizontal(4) 180.744,725.510,266.356,758.710 'SARS-CoV-2 and COVID-\\n19:\\nis interleukin-6 (IL-6)\\nthe ’culprit lesion’ of ARDS\\nonset?\\n'>,\n",
              " <LTTextBoxHorizontal(5) 278.886,716.915,448.769,758.658 '“...monoclonal antibody against IL-6, is being tested in a\\nclinical trial against COVID-19 (Sarilimumab COVID-\\n19). Another drug that showed potential inhibition of IL-\\n6 related JAK/STAT pathway is glatiramer acetate which\\nshowed potential to downregulate both IL-17 and IL-6”\\n'>,\n",
              " <LTTextBoxHorizontal(6) 462.055,766.636,486.423,774.003 'Topic(s)\\n'>,\n",
              " <LTTextBoxHorizontal(7) 462.280,751.292,517.470,758.658 'Clinical Treatment\\n'>,\n",
              " <LTTextBoxHorizontal(8) 81.749,682.539,167.790,707.093 '“When is the salivary vi-\\nral load highest for COVID-\\n19?”\\n'>,\n",
              " <LTTextBoxHorizontal(9) 181.297,673.944,266.358,707.093 'Elective, Non-urgent Pro-\\ncedures\\nAesthetic\\nand\\nSurgery in the Wake of\\nSARS–COVID-19 [...]\\n'>,\n",
              " <LTTextBoxHorizontal(10) 278.886,673.944,448.770,707.144 '“Patients with COVID-19 have demonstrated high viral\\nloads in the upper respiratory tract soon after their infec-\\ntion, with the highest load assumed to be the day before\\nsymptoms appear.”\\n'>,\n",
              " <LTTextBoxHorizontal(11) 462.280,699.726,517.470,707.093 'Clinical Treatment\\n'>,\n",
              " <LTTextBoxHorizontal(12) 81.749,648.161,167.790,664.122 '“Is convalescent plasma ther-\\napy a precursor to vaccine?”\\n'>,\n",
              " <LTTextBoxHorizontal(13) 181.297,648.161,236.745,664.122 'COVID-19\\nplasma transfusion\\n'>,\n",
              " <LTTextBoxHorizontal(14) 227.566,656.755,265.135,664.122 'convalescent\\n'>,\n",
              " <LTTextBoxHorizontal(15) 278.886,639.566,448.766,664.122 '“...a passive immunotherapy, has been used as a possi-\\nble therapeutic option when no proven speciﬁc vaccine\\nor drug is available for emerging infections. ”\\n'>,\n",
              " <LTTextBoxHorizontal(16) 462.052,639.566,519.098,664.122 'Clinical Treatment,\\nImmunology, Lab\\nTrials\\n'>,\n",
              " <LTTextBoxHorizontal(17) 72.000,592.787,525.547,626.661 'Figure 1: Sample natural language queries and responses using AWS CORD-19 Search. The response ﬁeld is\\ntaken directly from the top result of this service. Also provided are the article titles where the answer is taken from\\nas well as selected topics for the response.\\n'>,\n",
              " <LTTextBoxHorizontal(18) 72.000,368.424,292.083,569.273 'viral load. We provide a further advantage from\\nan organizational perspective using topics with do-\\nmain speciﬁc relations. The user can select clinical-\\ntreatments to reﬁne search articles to understand\\nthat “...the highest load [is] assumed to be the day\\nbefore symptoms appear.” Similarly if commis-\\nsioned to understand convalescent plasma therapy\\nand selecting the three topics as shown in the last\\nexample the system identiﬁes the most relevant ar-\\nticle and answers the question in a highlight. There\\nare a few search systems have been developed to\\nsupport COVID-19 search leveraging CORD-19\\ncorpus. We evaluate the ACS against two other\\nsearch engines, and ACS is proved to be a top-\\nperforming system.\\n'>,\n",
              " <LTTextBoxHorizontal(19) 71.727,282.940,292.075,361.595 'In the following sections we will discuss the\\nvarious individual Amazon Web Services (AWS)\\nproducts that allow ACS to provide this function-\\nality, highlighting the value to scientists who can\\nquickly query, validate their research, and advance\\ntheir investigations.\\n'>,\n",
              " <LTTextBoxHorizontal(20) 72.000,249.561,178.342,261.517 '2 System Overview\\n'>,\n",
              " <LTTextBoxHorizontal(21) 71.607,76.568,292.172,236.518 'AWS CORD-19 Search (ACS) relies on a deep\\nsemantic search model to return a ranked list of\\nrelevant documents. This system performs docu-\\nment ranking, passage ranking, question answer-\\ning and FAQ matching. Additionally, it leverages\\nknowledge graphs and topic modeling to enrich the\\nbiomedical data and boost searching performance.\\nThis provides a scalable solution to COVID-19 re-\\nsearchers and policy makers. In this section we\\npresent the overall architecture of the system and a\\ncloser look at several individual components. Fig-\\nure 2 provides an overview of this architecture.\\n'>,\n",
              " <LTTextBoxHorizontal(22) 306.883,418.693,527.452,569.098 '2.1 Amazon Kendra\\nAmazon Kendra2 is a semantic search and question\\nanswering service provided by AWS for enterprise\\ncustomers. Kendra allows its customers to power\\nnatural language based searching across their own\\ndata. As response to the worldwide COVID-19\\npandemic, Kendra has also been tooled to support\\nCOVID-19 related searching and question answer-\\ning using the document corpus from CORD-19.\\nThe end-to-end Kendra system consists of several\\ncomponents.\\n'>,\n",
              " <LTTextBoxHorizontal(23) 307.276,323.849,527.356,416.053 'To complement the extracted answers, Kendra\\nuses a deep learning based semantic search model\\nto return a ranked list of relevant documents. Ama-\\nzon Kendra’s ability to understand natural language\\nquestions is at the core of its search engine return-\\ning the most relevant passage and related docu-\\nments\\n'>,\n",
              " <LTTextBoxHorizontal(24) 320.294,220.508,527.359,312.789 '• Document ranking (DR) - Like any tradi-\\ntional search engine, Kendra returns a ranked\\nlist of relevant documents based on the user’s\\nquery to fulﬁll their information needs. A\\ndeep semantic search model is used to under-\\nstand natural language questions in addition\\nto the keyword search.\\n'>,\n",
              " <LTTextBoxHorizontal(25) 320.294,144.329,527.360,209.511 '• Passage ranking (PR) & Question Answer-\\ning (QA) - Kendra ranks the passages and\\ntries to extract the answer from the top rele-\\nvant passages with a deep reading comprehen-\\nsion model.\\n'>,\n",
              " <LTTextBoxHorizontal(26) 320.294,95.249,527.351,133.332 '• FAQ Matching (FAQM) - If there exists Fre-\\nquently Answered Questions and their corre-\\nsponding answers, Kendra will automatically\\n'>,\n",
              " <LTTextBoxHorizontal(27) 319.928,77.606,484.810,87.419 '2https://aws.amazon.com/kendra/\\n'>,\n",
              " <LTTextBoxHorizontal(0) 237.044,540.631,360.501,550.594 'Figure 2: System Architecture.\\n'>,\n",
              " <LTTextBoxHorizontal(1) 93.818,492.407,290.280,516.866 'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n'>,\n",
              " <LTTextBoxHorizontal(2) 72.000,325.745,292.082,472.146 'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n'>,\n",
              " <LTTextBoxHorizontal(3) 72.000,297.540,197.858,308.449 '2.2 Comprehend Medical\\n'>,\n",
              " <LTTextBoxHorizontal(4) 71.575,141.116,292.075,289.172 'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n'>,\n",
              " <LTTextBoxHorizontal(5) 307.276,506.033,470.563,516.942 '2.3 COVID-19 Knowledge Graph\\n'>,\n",
              " <LTTextBoxHorizontal(6) 307.276,148.089,527.359,497.728 'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n'>,\n",
              " <LTTextBoxHorizontal(7) 84.653,118.369,217.255,128.182 '3https://docs.aws.amazon.\\n'>,\n",
              " <LTTextBoxHorizontal(8) 72.000,98.443,245.150,117.372 'com/comprehend/latest/dg/\\nhow-document-classification.html\\n'>,\n",
              " <LTTextBoxHorizontal(9) 84.653,87.569,271.054,97.382 '4https://aws.amazon.com/comprehend/\\n'>,\n",
              " <LTTextBoxHorizontal(10) 72.000,77.606,115.039,86.572 'medical/\\n'>,\n",
              " <LTTextBoxHorizontal(11) 307.276,122.881,394.156,133.790 '2.4 Topic Models\\n'>,\n",
              " <LTTextBoxHorizontal(12) 306.937,76.568,525.549,114.576 'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n'>,\n",
              " <LTTextBoxHorizontal(0) 186.115,447.945,411.429,457.908 'Figure 3: Visualization of COVID-19 Knowledge graph.\\n'>,\n",
              " <LTTextBoxHorizontal(1) 71.607,304.876,292.075,424.179 'for topic modeling is Latent Dirichlet Allocation\\n(LDA) (Blei et al., 2003), a generative model which\\ngroups documents together by observed content,\\noften giving each document a mixture of topics it\\nbelongs to. An extension of this work termed Z-\\nlabel LDA (Andrzejewski and Zhu, 2009) utilizes\\npriors to allow the model to force certain topics\\nwhich the users have manually curated, or wish to\\nsee clustered together.\\n'>,\n",
              " <LTTextBoxHorizontal(2) 71.607,105.493,292.083,295.072 '2.4.1 Generating Topics\\nFor the purposes of this work we experimented\\nwith 5, 10, and 20 topic models 5. The outputs\\nof each clustering size were manually inspected\\nand topic labels were provided by us when inspect-\\ning the top ten terms for each cluster. The ﬁnal\\ngranularity of the topic models was chosen by man-\\nually deleting and merging topics from the 20 topic\\nmodel. In general we were able to clearly extract\\ngroups which centered around important topics in-\\ncluding virology, proteomics, epidemiology, and\\ncellular biology to name a few. However when\\nfaced with 20 topics the less populated ones tended\\nto be noisy, and captured peripheral information\\n'>,\n",
              " <LTTextBoxHorizontal(3) 307.003,182.933,527.359,424.179 'present in the input, such as language (e.g. Spanish\\nand French) or provide redundancies with existing\\ntopics (e.g. two topics for Inﬂuenza). As a con-\\ntrol we ran a publicly available implementation of\\nZ-label LDA 6 with no priors which yields topics\\nclose to those extracted using Comprehend Topic\\nModeling. Although similar we observed better\\ndeﬁnition in certain groups (such as pulmonary\\ndiseases, and policy/industry), and decided to use\\nthis as the curation entry-point. Our goal was to\\nlimit these topics to ten, and compile them in ad-\\nvance as much as possible. With the help of medi-\\ncal professionals we eliminated and combined top-\\nics to form the following: Vaccines/immunology,\\nGenomics, Public health Policies, Epidemiology,\\nClinical Treatment, Virology, Inﬂuenza, Healthcare\\nIndustry, Pulmonary Infections, and Lab Trials (hu-\\nman).\\n'>,\n",
              " <LTTextBoxHorizontal(4) 307.276,105.493,527.349,173.130 '2.4.2 Multi-Label Classiﬁcation\\nHaving to manually feed a topic model and re-train\\non the entire corpus once new data becomes avail-\\nable is largely inefﬁcient. We therefore used the\\ntopic model labels to train a multi-label classiﬁer\\n'>,\n",
              " <LTTextBoxHorizontal(5) 84.653,86.950,290.268,97.382 '5These models were trained using CORD-19 data available\\n'>,\n",
              " <LTTextBoxHorizontal(6) 72.000,76.987,148.708,85.954 'as of April 6th, 2020.\\n'>,\n",
              " <LTTextBoxHorizontal(7) 319.928,85.850,511.709,97.382 '6http://pages.cs.wisc.edu/˜andrzeje/\\n'>,\n",
              " <LTTextBoxHorizontal(8) 307.276,77.606,414.873,86.572 'research/zl_lda.html\\n'>,\n",
              " <LTTextBoxHorizontal(0) 114.189,763.315,177.505,774.224 'Search Engine\\n'>,\n",
              " <LTTextBoxHorizontal(1) 194.622,763.315,216.189,774.224 'P@1\\n'>,\n",
              " <LTTextBoxHorizontal(2) 236.578,763.315,258.145,774.224 'P@5\\n'>,\n",
              " <LTTextBoxHorizontal(3) 275.807,763.315,302.829,774.224 'P@10\\n'>,\n",
              " <LTTextBoxHorizontal(4) 317.764,763.315,429.298,774.224 'P@20 R@10 R@20\\n'>,\n",
              " <LTTextBoxHorizontal(5) 442.138,763.315,484.302,774.224 'ndcg@20\\n'>,\n",
              " <LTTextBoxHorizontal(6) 260.421,745.841,337.123,756.750 'Keyword Queries\\n'>,\n",
              " <LTTextBoxHorizontal(7) 127.525,713.265,164.169,737.723 'ACS\\nCovidex\\n'>,\n",
              " <LTTextBoxHorizontal(8) 113.245,699.716,220.405,737.723 '0.5250\\n0.3421\\nCOVID-19 RE 0.5750\\n'>,\n",
              " <LTTextBoxHorizontal(9) 232.359,699.792,262.361,737.799 '0.5650\\n0.2316\\n0.5650\\n'>,\n",
              " <LTTextBoxHorizontal(10) 274.314,699.716,304.318,737.799 '0.5325\\n0.2079\\n0.4775\\n'>,\n",
              " <LTTextBoxHorizontal(11) 316.270,699.716,346.274,737.799 '0.4775\\n0.1658\\n0.4412\\n'>,\n",
              " <LTTextBoxHorizontal(12) 358.220,699.716,388.228,737.799 '0.0260\\n0.0109\\n0.0236\\n'>,\n",
              " <LTTextBoxHorizontal(13) 400.176,699.716,430.185,737.799 '0.0459\\n0.0173\\n0.0429\\n'>,\n",
              " <LTTextBoxHorizontal(14) 448.216,699.716,478.220,737.799 '0.4380\\n0.1633\\n0.4022\\n'>,\n",
              " <LTTextBoxHorizontal(15) 236.361,682.241,361.183,693.150 'Natural Language Questions\\n'>,\n",
              " <LTTextBoxHorizontal(16) 127.525,649.665,164.169,674.123 'ACS\\nCovidex\\n'>,\n",
              " <LTTextBoxHorizontal(17) 113.245,636.116,220.405,674.199 '0.8750\\n0.4750\\nCOVID-19 RE 0.6000\\n'>,\n",
              " <LTTextBoxHorizontal(18) 232.360,636.116,262.361,674.199 '0.7000\\n0.4800\\n0.5300\\n'>,\n",
              " <LTTextBoxHorizontal(19) 274.317,636.116,304.318,674.199 '0.6400\\n0.4225\\n0.4925\\n'>,\n",
              " <LTTextBoxHorizontal(20) 316.273,636.116,346.274,674.199 '0.5550\\n0.3625\\n0.4600\\n'>,\n",
              " <LTTextBoxHorizontal(21) 358.220,636.116,388.230,674.199 '0.0345\\n0.0204\\n0.0267\\n'>,\n",
              " <LTTextBoxHorizontal(22) 400.175,636.116,430.176,674.199 '0.0582\\n0.0356\\n0.0474\\n'>,\n",
              " <LTTextBoxHorizontal(23) 448.219,636.116,478.220,674.199 '0.5357\\n0.3229\\n0.4133\\n'>,\n",
              " <LTTextBoxHorizontal(24) 192.835,613.165,404.401,623.128 'Table 1: Evaluation results on TREC-COVID dataset\\n'>,\n",
              " <LTTextBoxHorizontal(25) 71.640,497.195,290.269,589.399 '(Read et al., 2011). To evaluate the performance\\nof this model we calculate the average F1 across\\ntest samples by calculating the set overlap between\\nour gold standard (topic model) and system labels\\n(multi-label classiﬁcation). This held-out test set\\ncontains 20% of the CORD-19 data available at the\\ntime.\\n'>,\n",
              " <LTTextBoxHorizontal(26) 72.000,429.449,290.273,494.555 'Using this metric our trained model achieved an\\naverage F1 of 91.92, with on average 2.37 labels\\nper document. Fewer than 1% of the documents\\nin the test set received no label, using 0.5 as the\\nconﬁdence threshold.\\n'>,\n",
              " <LTTextBoxHorizontal(27) 72.000,406.223,145.620,418.179 '3 Evaluation\\n'>,\n",
              " <LTTextBoxHorizontal(28) 71.182,196.446,292.177,397.044 'ACS supports document ranking (DR), passage\\nranking (PR) and question answering (QA) on the\\ntop suggested results. In this section we evaluate\\nthe overall performance of ACS with publicly avail-\\nable datasets. The TREC-COVID track (Voorhees\\net al., 2020) built DR test collections that contain\\ndocument relevance ground truth and are used to\\nassess DR only. For PR and QA, we manually anno-\\ntate questions collected from CovidQA (Tang et al.,\\n2020), the ﬁrst open sourced Covid-19 QA dataset.\\nThe ACS results are compared against Covidex7\\nand Google Covid-19 Research Explorer8 (COVID-\\n19 RE), which are also COVID-19 public search\\nengines that facilitate DR and answer highlighting\\nin passage.\\n'>,\n",
              " <LTTextBoxHorizontal(29) 72.000,115.151,292.075,193.806 'The TREC-COVID challenge track contains 40\\ntopic sets along with their document relevance\\njudgement. The topic sets are written by its or-\\nganizers with biomedical training, and motivated\\nby search submitted to the National Library of\\nMedicine and social media. Each topic consists\\n'>,\n",
              " <LTTextBoxHorizontal(30) 84.653,87.569,272.049,108.256 '7https://covidex.ai/\\n8https://covid19-research-explorer.\\n'>,\n",
              " <LTTextBoxHorizontal(31) 72.000,77.606,136.558,86.572 'appspot.com/\\n'>,\n",
              " <LTTextBoxHorizontal(32) 307.276,361.703,527.359,589.399 'of three ﬁelds with different levels of granularity, a\\nkeyword-based query (KQ), a more precise natural\\nlanguage question (NQ), and a longer descriptive\\nnarrative. The DR assessments are performed fol-\\nlowing the TREC pooling mechanism. The par-\\nticipants submitted ranked lists of documents for\\neach topic set, based on which a depth of 10 to 20\\ndocuments are pooled and combined as a collection\\nof (q, D) pairs. The pairs are then assessed by an-\\nnotators with in-domain expertise. There are three\\nrounds of judgement available that corresponding\\nto three different versions of CORD-19 corpus. To\\nensure sufﬁcient coverage of annotation, we aggre-\\ngate all rounds results into 33,064 (q, D) relevance\\njudgements. Since the document id may change\\nacross versions, we map ids from each round to the\\nMay 19 release of CORD-19 corpus.\\n'>,\n",
              " <LTTextBoxHorizontal(33) 306.883,253.109,527.355,358.862 'To collect DR results from the three systems, we\\ncrawled the top 50 articles by querying the engines\\nwith KQ and its NQ variation from the topic sets on\\nJune 15, 2020. The crawled data, which are char-\\nacterized by the article title and link, are mapped\\nto May 19 CORD-19 corpus. Note that articles\\nthat cannot be found in the corpus are removed to\\nensure fair comparison.\\n'>,\n",
              " <LTTextBoxHorizontal(34) 306.883,103.867,527.356,250.268 'We use the standard DR metrics in our evalua-\\ntion, namely, the precision and recall at k (P@k,\\nR@k) and normalized discounted cumulative gain\\nin the top k documents (NDCG@k). Note that we\\nevaluate with k up to 20 since TREC-COVID has\\na pooling depth with 20 at most. Table 1 presents\\nthe DR performance of the engines over KQ and\\nNQ, respectively. ACS performs consistently better\\nthan the other engines on NQ and mostly on KQ,\\nand all engines perform better on NQ comparing\\nwith KQ.\\n'>,\n",
              " <LTTextBoxHorizontal(35) 307.276,76.568,525.549,101.026 'In addition, we are able to evaluate how robust\\neach system is against query variation from KQ\\n'>,\n",
              " <LTTextBoxHorizontal(0) 173.032,756.541,236.348,767.450 'Search Engine\\n'>,\n",
              " <LTTextBoxHorizontal(1) 251.554,749.766,296.543,774.224 'Top3\\nEM F1\\n'>,\n",
              " <LTTextBoxHorizontal(2) 313.646,749.766,358.635,774.224 'Top10\\nEM F1\\n'>,\n",
              " <LTTextBoxHorizontal(3) 375.738,749.766,420.727,774.224 'Top30\\nEM F1\\n'>,\n",
              " <LTTextBoxHorizontal(4) 173.032,703.642,269.281,741.725 '11.7\\nACS\\nCovidex\\n0.90\\nCOVID-19 RE 10.0\\n'>,\n",
              " <LTTextBoxHorizontal(5) 281.237,703.642,300.328,741.725 '35.6\\n18.9\\n31.8\\n'>,\n",
              " <LTTextBoxHorizontal(6) 312.282,703.642,331.373,741.725 '16.0\\n2.10\\n13.5\\n'>,\n",
              " <LTTextBoxHorizontal(7) 343.329,703.642,362.420,741.725 '42.2\\n24.2\\n36.9\\n'>,\n",
              " <LTTextBoxHorizontal(8) 374.374,703.642,393.465,741.725 '26.0\\n3.60\\n18.2\\n'>,\n",
              " <LTTextBoxHorizontal(9) 405.421,703.642,424.512,741.725 '50.4\\n27.6\\n43.8\\n'>,\n",
              " <LTTextBoxHorizontal(10) 154.912,680.691,442.323,690.654 'Table 2: Top results variation from KQ to NQ on TREC-COVID dataset\\n'>,\n",
              " <LTTextBoxHorizontal(11) 140.305,644.760,203.621,655.669 'Search Engine\\n'>,\n",
              " <LTTextBoxHorizontal(12) 221.679,637.986,243.246,648.895 'P@1\\n'>,\n",
              " <LTTextBoxHorizontal(13) 140.305,591.861,247.463,629.944 '0.4074\\nACS\\n0.4074\\nCovidex\\nCOVID-19 RE 0.4074\\n'>,\n",
              " <LTTextBoxHorizontal(14) 263.635,637.986,285.203,662.444 'PR\\nP@2\\n'>,\n",
              " <LTTextBoxHorizontal(15) 259.418,591.861,289.419,629.944 '0.5370\\n0.4444\\n0.3704\\n'>,\n",
              " <LTTextBoxHorizontal(16) 305.592,637.986,327.159,648.895 'P@3\\n'>,\n",
              " <LTTextBoxHorizontal(17) 347.545,637.986,369.112,648.895 'P@1\\n'>,\n",
              " <LTTextBoxHorizontal(18) 301.374,591.861,331.376,629.944 '0.4938\\n0.4074\\n0.3580\\n'>,\n",
              " <LTTextBoxHorizontal(19) 343.328,591.861,373.328,629.944 '0.3333\\n0.1481\\n0.2963\\n'>,\n",
              " <LTTextBoxHorizontal(20) 389.501,637.986,411.069,662.444 'QA\\nP@2\\n'>,\n",
              " <LTTextBoxHorizontal(21) 385.284,591.861,415.284,629.944 '0.2593\\n0.1852\\n0.2407\\n'>,\n",
              " <LTTextBoxHorizontal(22) 431.458,637.986,453.025,648.895 'P@3\\n'>,\n",
              " <LTTextBoxHorizontal(23) 427.239,591.937,457.241,629.868 '0.2099\\n0.2099\\n0.2593\\n'>,\n",
              " <LTTextBoxHorizontal(24) 109.892,568.910,487.345,578.873 'Table 3: Passage ranking (PR) and question answering (QA) performance on CovidQA dataset\\n'>,\n",
              " <LTTextBoxHorizontal(25) 203.824,478.151,238.804,492.777 'q , KQk\\n'>,\n",
              " <LTTextBoxHorizontal(26) 71.607,317.448,292.083,545.144 'to NQ. Ideally, the results shall remain unchanged\\nwith query variation that requests the same infor-\\nmation. We deﬁne exact match (EM) and F1 score\\namong top k results to evaluate the robustness. Let\\nQ be the topic sets, and N Qk\\nq denote the\\ntop k searching results of natural language ques-\\ntion and keyword query corresponding to the same\\ntopic q ∈ Q, respectively. As shown in Eq.(1), we\\ntake the top k results KQk\\nq as ground truth, and\\ncompute the average exact match and F1 score of\\nthe top k results N Qk\\nq , and then average over all\\nqueries. The article title string is used as com-\\nparison key, and EM and F1 are standard that the\\nmaximum is taken over all the ground truth arti-\\ncles. Table 2 demonstrates that ACS has the best\\ncapability to provide consistent results with query\\nvariation.\\n'>,\n",
              " <LTTextBoxHorizontal(27) 109.069,288.073,190.977,298.982 'EM (N Q, KQ, k)\\n'>,\n",
              " <LTTextBoxHorizontal(28) 100.584,259.761,109.069,270.670 '=\\n'>,\n",
              " <LTTextBoxHorizontal(29) 110.264,252.278,124.950,278.050 '1\\n|Q|\\n'>,\n",
              " <LTTextBoxHorizontal(30) 147.349,252.278,153.088,278.050 '1\\nk\\n'>,\n",
              " <LTTextBoxHorizontal(31) 128.270,265.696,144.027,276.605 '(cid:88)\\n'>,\n",
              " <LTTextBoxHorizontal(32) 127.963,247.066,144.336,255.036 'q∈Q\\n'>,\n",
              " <LTTextBoxHorizontal(33) 156.386,265.696,172.143,281.937 'k\\n(cid:88)\\n'>,\n",
              " <LTTextBoxHorizontal(34) 157.413,247.244,171.117,255.214 'i=1\\n'>,\n",
              " <LTTextBoxHorizontal(35) 173.962,259.761,215.850,272.805 'em(N Qk\\n'>,\n",
              " <LTTextBoxHorizontal(36) 211.441,257.634,261.684,272.805 'q (i), KQk\\nq )\\n'>,\n",
              " <LTTextBoxHorizontal(37) 277.548,258.972,290.268,269.881 '(1)\\n'>,\n",
              " <LTTextBoxHorizontal(38) 71.607,76.568,292.075,222.969 'Next, we evaluate the performance of PR and QA\\nwith CovidQA dataset. CovidQA contains 27 ques-\\ntions and their answers from 124 question-article\\npairs, which are selected by in-domain volunteers\\nas the most promising COVID-19 literature. Since\\nthe annotation does not possess sufﬁcient answer\\ncoverage over the entire corpus, we leverage our\\ninternal annotation resources to make PR and QA\\njudgement. More explicitly, we crawled the top 3\\nresults characterized by the article, displayed pas-\\nsage and the highlighted text snippet on June 15,\\n'>,\n",
              " <LTTextBoxHorizontal(39) 307.276,425.841,527.359,545.144 '2020 from ACS, Covidex and COVID-19 RE, re-\\nspectively. After that, the annotators assess PR\\nand QA in terms of whether the displayed passage\\ncontains relevant information and whether the high-\\nlighted text snippet answers the given question. To\\navoid bias, the crawled results are combined and\\nshufﬂed randomly, therefore, the annotators do not\\nhave access to the source engine and the rank posi-\\ntion information during the judgement.\\n'>,\n",
              " <LTTextBoxHorizontal(40) 306.457,136.037,527.486,404.380 'Table 3 presents precision of top PR and QA\\nresults. We use P@k instead of EM and F1 to eval-\\nuate QA since ACS highlights the answer while\\nCovidex and COVID-19 RE highlight the sentence\\nthat contains the answer. Instead of extracting the\\nanswer, it is more reasonable to judge whether the\\nhighlighted text answers the question and compute\\nprecision. ACS achieves better accuracy on both\\nPR and QA on most metrics. Note that ACS high-\\nlights answer snippet for at most three passages\\n, while Covidex and COVID-19 RE highlight all\\ndisplayed passages. This explains why ACS P@3\\nunderperforms COVID-19 RE. Explicitly with an\\nexample, Figure 4 displays the topmost results of\\nquerying “What is the incubation period of virus”\\nfrom the three systems. ACS highlights the answer\\nin the passage. In contrast, Covidex presents an\\narticle published at 2004 with information of virus\\nwhich is irrelevant with coronavirus, and COVID-\\n19 RE does not answer the question at all.\\n'>,\n",
              " <LTTextBoxHorizontal(41) 307.276,76.568,525.549,114.576 'As evident from the above results, ACS is one\\nof the top-performing systems that provides high\\nquality informative results over CORD-19 search.\\n'>,\n",
              " <LTTextBoxHorizontal(0) 142.598,764.340,159.220,773.066 'ACS\\n'>,\n",
              " <LTTextBoxHorizontal(1) 286.114,764.340,315.423,773.066 'Covidex\\n'>,\n",
              " <LTTextBoxHorizontal(2) 424.543,764.340,476.695,773.066 'COVID-19 RE\\n'>,\n",
              " <LTTextBoxHorizontal(3) 226.245,748.431,375.285,757.157 'What is the incubation period of the virus?\\n'>,\n",
              " <LTTextBoxHorizontal(4) 234.561,730.257,342.408,738.982 'Deadly viral syndrome mimics\\n'>,\n",
              " <LTTextBoxHorizontal(5) 84.707,699.717,218.564,738.982 'Importance of Social Distancing:\\nModeling\\n2019-\\nof\\nSusceptible-Infected-\\nnCoV using\\nQuarantined-Recovered-t model\\n'>,\n",
              " <LTTextBoxHorizontal(6) 148.734,720.077,171.507,728.802 'spread\\n'>,\n",
              " <LTTextBoxHorizontal(7) 128.351,720.077,139.014,728.802 'the\\n'>,\n",
              " <LTTextBoxHorizontal(8) 83.546,618.280,218.563,688.144 '“Studies on the nature of the virus\\nhave suggested different incubation\\nperiods of the virus, and reports have\\nsuggested a median incubation pe-\\nriod of 5-6 days and a very high symp-\\ntom probability period of 14 days [5]\\n”\\n'>,\n",
              " <LTTextBoxHorizontal(9) 233.401,577.561,368.418,688.144 '“...The incubation period is typically\\n3 to 14 days with the symptoms of\\nan acute nonspeciﬁc, ﬂulike illness\\ndeveloping suddenly...The incuba-\\ntion period is 7 to 10 days before\\nthe onset of symptoms [37]. This\\nincubation period provides the po-\\ntential for worldwide exposure be-\\ncause a person harboring the virus\\ncan expose the world at large via air\\ntravel...”\\n'>,\n",
              " <LTTextBoxHorizontal(10) 384.417,709.897,518.274,738.982 'Prediction of the virus incubation pe-\\nriod for COVID-19 and future out-\\nbreaks\\n'>,\n",
              " <LTTextBoxHorizontal(11) 383.257,618.280,518.274,688.083 '“. . . while minimizing the negative\\nconsequences of the quarantine. 70 71\\nThe length of the incubation period\\nvaries both across and within virus\\nfamilies 4 . To our knowledge, 72\\ngenomic features (if any) that corre-\\nlate with the incubation. . . ”\\n'>,\n",
              " <LTTextBoxHorizontal(12) 73.917,553.886,523.629,563.849 'Figure 4: Top-1 result of article title and displayed passage by querying ’What is the incubation period of virus?’\\n'>,\n",
              " <LTTextBoxHorizontal(13) 72.000,504.668,133.115,516.624 '4 Analysis\\n'>,\n",
              " <LTTextBoxHorizontal(14) 72.000,294.788,292.083,495.386 'In this section we look into a number of sample\\nqueries to shed light on how different components\\nof ACS help in improving search results. We begin\\nby observing how small semantic differences in the\\nquery alter the results. The ﬁrst sample in Figure 5\\nis speciﬁc to medications. While the top result does\\nnot include the term medication the system high-\\nlights ribavirin and corticosteroids. The CORD-19\\nsystem understands that these terms represent med-\\nications with the help of CM NERe engine. In the\\nsecond example we change medications to mea-\\nsures and observe the top result discussing border\\ncontrol, and quarantine. This clearly demonstrates\\nthat Amazon Kendra has a deep comprehension of\\ntoken and query meanings.\\n'>,\n",
              " <LTTextBoxHorizontal(15) 71.487,145.747,292.173,292.148 'Finally, we take a look at the effects of topic mod-\\neling when grouping and ﬁltering results. The last\\ntwo examples in Figure 5 showcase the difference\\nthis makes in the top result. Without specifying any\\ntopic the resulting article discusses high level pol-\\nicy, speciﬁcally quarantine measures in Singapore.\\nWhen we ﬁlter by clinical treatment the top result\\ninstead focuses on infections which is covered in\\nthe clinical setting. Furthermore the extracted text\\nreturned to the user still focuses on lessons learned\\nstaying true to the query.\\n'>,\n",
              " <LTTextBoxHorizontal(16) 72.000,122.417,264.515,134.373 '5 Limitations and Future Directions\\n'>,\n",
              " <LTTextBoxHorizontal(17) 71.607,76.568,292.083,114.576 'AWS CORD-19 is an initial step towards help-\\ning medical researchers ﬁnd relevant content in\\na timely and meaningful way. In order to improve\\n'>,\n",
              " <LTTextBoxHorizontal(18) 307.276,491.262,525.548,515.720 'the robustness, we see following areas as direction\\nfor future research.\\n'>,\n",
              " <LTTextBoxHorizontal(19) 306.883,175.521,527.359,484.589 'Feedback Loop - Since ACS is a search engine\\nthe motivation would be to evaluate it as such; us-\\ning well-established methodologies based on test\\ncollections—comprising topics (information needs)\\nand human annotations. Since no designated evalu-\\nation data exist, our initial focus is to capture dif-\\nferent interactions and feedback. Currently, ACS\\nlacks the feedback loop and federated learning ap-\\nproaches where the system would continuously\\nlearn and improve the search. However, the system\\ncaptures feedback from the researchers in the form\\nof implicit and explicit reactions. Implicit feed-\\nback evaluation consists of topics of interests, their\\nclicks as well as the ranking of the results which\\nwere selected by medical researchers. Explicit feed-\\nback evaluation is captured by providing up-down\\nrating associated with each search results. In the\\nfuture results can be personalized based on this\\nfeedback. Now that we have a system in place, our\\nefforts have shifted to broader engagement with po-\\ntential stakeholders to solicit additional guidance,\\nwhile trying to balance between the features and\\nranking.\\n'>,\n",
              " <LTTextBoxHorizontal(20) 306.883,76.568,525.727,168.848 'Q&A Curation - Curation and normalization of\\nquestions have potential a use-case of presenting\\ntrending questions asked by the medical research\\ncommunity at a particular point. However, curation\\nwould involve capturing the questions asked as well\\nas identifying similar questions that can be later\\nnormalized. Currently, there is no mechanism to\\n'>,\n",
              " <LTTextBoxHorizontal(0) 83.088,766.219,102.115,773.833 'Query\\n'>,\n",
              " <LTTextBoxHorizontal(1) 213.583,766.219,234.719,773.833 'Article\\n'>,\n",
              " <LTTextBoxHorizontal(2) 344.618,766.219,373.802,773.833 'Response\\n'>,\n",
              " <LTTextBoxHorizontal(3) 82.075,741.477,199.890,758.149 '“What medications were most beneﬁ-\\ncial in the SARS outbreak?”\\n'>,\n",
              " <LTTextBoxHorizontal(4) 213.853,732.593,330.658,757.974 'Development of chemical inhibitors\\nof the SARS coronavirus: Viral heli-\\ncase as a potential target\\n'>,\n",
              " <LTTextBoxHorizontal(5) 82.075,679.296,198.627,695.968 '“What measures were most beneﬁcial\\nin the SARS outbreak?”\\n'>,\n",
              " <LTTextBoxHorizontal(6) 213.853,670.413,330.658,695.793 'Impact of quarantine on the 2003\\nSARS outbreak: A retrospective mod-\\neling study\\n'>,\n",
              " <LTTextBoxHorizontal(7) 82.075,634.882,198.629,651.378 '“What did we learn from the SARS\\noutbreak?” (no topic)\\n'>,\n",
              " <LTTextBoxHorizontal(8) 213.853,634.882,329.394,651.378 'Use of quarantine in the control of\\nSARS in Singapore\\n'>,\n",
              " <LTTextBoxHorizontal(9) 82.075,590.467,198.629,606.964 '“What did we learn from the SARS\\noutbreak?” (clinical-treatment)\\n'>,\n",
              " <LTTextBoxHorizontal(10) 213.853,590.467,330.658,606.964 'Characteristics of COVID-19 infec-\\ntion in Beijing\\n'>,\n",
              " <LTTextBoxHorizontal(11) 343.606,705.944,519.196,757.974 '“...spread of SARS, a number of broad-spectrum antiviral\\nmedications were empirically administered to the SARS\\npatients during the SARS outbreak in 2003. These medi-\\ncations include ribavirin, HIV protease inhibitors, cor-\\nticosteroids, and alpha-interferon (IFN-a). In a retro-\\nspective review of treatment...”\\n'>,\n",
              " <LTTextBoxHorizontal(12) 343.606,661.530,517.931,695.793 '“During the 2003 Severe Acute Respiratory Syndrome\\n(SARS) outbreak, traditional intervention measures\\nsuch as quarantine and border control were found to\\nbe useful in containing the outbreak...”\\n'>,\n",
              " <LTTextBoxHorizontal(13) 343.606,617.116,517.926,651.378 '“The main lesson to learn from the SARS outbreak is the\\ncapability of an emerging infection to cause a pandemic\\nin a short span of time and the paradigm shift needed to\\nrespond to such a disease.”\\n'>,\n",
              " <LTTextBoxHorizontal(14) 343.606,581.584,519.190,606.964 '“We compared the epidemic features between COVID-19\\nand 2003 SARS for learn lessons and control the out-\\nbreak.”\\n'>,\n",
              " <LTTextBoxHorizontal(15) 72.000,546.613,525.547,568.532 'Figure 5: Sample queries demonstrating the semantic understanding of ACS, the use-fullness of Comprehend\\nMedical NERe, and the utility of topic modeling for ﬁltering.\\n'>,\n",
              " <LTTextBoxHorizontal(16) 72.000,511.939,270.317,522.848 'curate the questions asked by the researchers.\\n'>,\n",
              " <LTTextBoxHorizontal(17) 307.276,511.796,400.802,523.752 'Acknowledgments\\n'>,\n",
              " <LTTextBoxHorizontal(18) 72.000,424.903,292.075,503.634 'Summarization - Currently, ACS outputs the\\nrelevant passage based on the query. It would be\\nbeneﬁcial to get the overall summary of the pa-\\nper. A potential future direction would be to gen-\\nerates summaries (Raffel et al., 2019) from paper\\nabstracts and full body.\\n'>,\n",
              " <LTTextBoxHorizontal(19) 306.763,409.683,526.910,501.888 'We acknowledge the broader collaboration with\\nAI2 team and White house as well as the broader\\nAWS CORD-19 team including Kendra Science\\nteam, Tyler Stepke, Kingston Bosco, Victor Wang,\\nVaibhav Chaddha, Miguel Calvo, Ninad Kulkani,\\nKevin Longofer, Ray Chang, Adrian Bordone,\\nTony Nguyen, and Kyle Johnson.\\n'>,\n",
              " <LTTextBoxHorizontal(20) 72.000,381.109,147.067,393.065 '6 Conclusion\\n'>,\n",
              " <LTTextBoxHorizontal(21) 71.607,76.568,292.084,358.461 'This paper describes our efforts in building AWS\\nCORD-19 Search with its capabilities consisting\\nof topic based, knowledge graph, and natural lan-\\nguage search queries. This is further enhanced\\nwith reading comprehension and FAQ matching\\nas well as document ranking providing a scalable\\nsolution to COVID-19 researchers and policy mak-\\ners in their search and discovery for answers to\\nhigh priority scientiﬁc questions. Our solution is\\npowered by Amazon Kendra, Comprehend Medi-\\ncal and Neptune which incorporate the latest neural\\narchitectures to provide information access capa-\\nbilities to the CORD-19 challenge. By comparing\\nwith other public search engines that support sim-\\nilar functionalities over COVID-19 search, ACS\\nis demonstrated to be powerful on its document\\nranking and question answering components. We\\nhope that our solution can prove useful in the ﬁght\\nagainst this global pandemic, and that the capabili-\\nties we have developed can be applied to analyzing\\nthe scientiﬁc literature more broadly.\\n'>,\n",
              " <LTTextBoxHorizontal(22) 307.276,371.126,362.820,383.082 'References\\n'>,\n",
              " <LTTextBoxHorizontal(23) 307.276,297.584,527.200,362.342 'David Andrzejewski and Xiaojin Zhu. 2009. Latent\\ndirichlet allocation with topic-in-set knowledge. In\\nProceedings of the NAACL HLT 2009 Workshop\\non Semi-Supervised Learning for Natural Language\\nProcessing, pages 43–48. Association for Computa-\\ntional Linguistics.\\n'>,\n",
              " <LTTextBoxHorizontal(24) 307.276,253.340,527.200,285.221 'Iz Beltagy, Kyle Lo, and Arman Cohan. 2019. Scib-\\nert: Pretrained language model for scientiﬁc text. In\\nEMNLP.\\n'>,\n",
              " <LTTextBoxHorizontal(25) 307.276,176.219,527.290,240.977 'P. Bhatia, B. Celikkaya, M. Khalilia, and S. Senthivel.\\n2019. Comprehend medical: A named entity recog-\\nIn\\nnition and relationship extraction web service.\\n2019 18th IEEE International Conference On Ma-\\nchine Learning And Applications (ICMLA), pages\\n1844–1851.\\n'>,\n",
              " <LTTextBoxHorizontal(26) 307.276,132.204,527.200,163.856 'Parminder Bhatia, Busra Celikkaya, and Mohammed\\nJoint entity extraction and asser-\\narXiv preprint\\n'>,\n",
              " <LTTextBoxHorizontal(27) 318.185,121.016,428.371,152.897 'Khalilia. 2018.\\ntion detection for clinical\\narXiv:1812.05270.\\n'>,\n",
              " <LTTextBoxHorizontal(28) 434.110,131.975,451.395,141.938 'text.\\n'>,\n",
              " <LTTextBoxHorizontal(29) 307.276,76.772,527.290,108.653 'David M Blei, Andrew Y Ng, and Michael I Jordan.\\n2003. Latent dirichlet allocation. Journal of ma-\\nchine Learning research, 3(Jan):993–1022.\\n'>,\n",
              " <LTTextBoxHorizontal(0) 72.000,722.135,292.014,775.934 'Jeffrey Dalton, Laura Dietz, and James Allan. 2014.\\nEntity query feature expansion using knowledge\\nbase links. In Proceedings of the 37th international\\nACM SIGIR conference on Research & development\\nin information retrieval, pages 365–374.\\n'>,\n",
              " <LTTextBoxHorizontal(1) 307.276,722.135,527.200,775.934 'Da Zheng, Xiang Song, Chao Ma, Zeyuan Tan, Zi-\\nhao Ye, Jin Dong, Hao Xiong, Zheng Zhang, and\\nGeorge Karypis. 2020. Dgl-ke: Training knowl-\\narXiv preprint\\nedge graph embeddings at scale.\\narXiv:2004.08532.\\n'>,\n",
              " <LTTextBoxHorizontal(2) 72.000,567.628,292.014,709.097 'Wei-jie Guan, Zheng-yi Ni, Yu Hu, Wen-hua Liang,\\nChun-quan Ou, Jian-xing He, Lei Liu, Hong Shan,\\nChun-liang Lei, David S.C. Hui, Bin Du, Lan-juan\\nLi, Guang Zeng, Kwok-Yung Yuen, Ru-chong Chen,\\nChun-li Tang, Tao Wang, Ping-yan Chen, Jie Xiang,\\nShi-yue Li, Jin-lin Wang, Zi-jing Liang, Yi-xiang\\nPeng, Li Wei, Yong Liu, Ya-hua Hu, Peng Peng,\\nJian-ming Wang, Ji-yang Liu, Zhong Chen, Gang\\nLi, Zhi-jian Zheng, Shao-qin Qiu, Jie Luo, Chang-\\njiang Ye, Shao-yong Zhu, and Nan-shan Zhong.\\n2020. Clinical characteristics of coronavirus disease\\n2019 in china. New England Journal of Medicine,\\n382(18):1708–1720.\\n'>,\n",
              " <LTTextBoxHorizontal(3) 72.000,500.791,291.924,554.590 'Colin Raffel, Noam Shazeer, Adam Roberts, Katherine\\nLee, Sharan Narang, Michael Matena, Yanqi Zhou,\\nWei Li, and Peter J Liu. 2019. Exploring the limits\\nof transfer learning with a uniﬁed text-to-text trans-\\nformer. arXiv preprint arXiv:1910.10683.\\n'>,\n",
              " <LTTextBoxHorizontal(4) 72.000,455.873,290.271,487.753 'Jesse Read, Bernhard Pfahringer, Geoff Holmes, and\\nEibe Frank. 2011. Classiﬁer chains for multi-label\\nclassiﬁcation. Machine learning, 85(3):333.\\n'>,\n",
              " <LTTextBoxHorizontal(5) 72.000,399.995,291.516,442.835 'Maya Rotmensch, Yoni Halpern, Abdulhakim Tlimat,\\nSteven Horng, and David Sontag. 2017. Learning\\na health knowledge graph from electronic medical\\nrecords. Scientiﬁc reports, 7(1):1–11.\\n'>,\n",
              " <LTTextBoxHorizontal(6) 72.000,355.077,290.271,386.957 'Gaurav Singh and Parminder Bhatia. 2019. Relation\\nextraction using explicit context conditioning. arXiv\\npreprint arXiv:1902.09271.\\n'>,\n",
              " <LTTextBoxHorizontal(7) 72.000,288.240,291.924,342.039 'Raphael Tang, Rodrigo Nogueira, Edwin Zhang, Nikhil\\nGupta, Phuong Cam, Kyunghyun Cho, and Jimmy\\nLin. 2020. Rapidly Bootstrapping a Question An-\\narXiv preprint\\nswering Dataset for COVID-19.\\narXiv:2004.11339.\\n'>,\n",
              " <LTTextBoxHorizontal(8) 72.000,210.445,292.014,275.203 'Ellen Voorhees, Tasmeer Alam, Steven Bedrick, Dina\\nDemner-Fushman, William R Hersh, Kyle Lo, Kirk\\nRoberts, Ian Soborof, and Lucy Lu Wang. 2020.\\nTREC-COVID: Constructing a Pandemic Informa-\\narXiv preprint\\ntion Retrieval Test Collection.\\narXiv:2005.04474.\\n'>,\n",
              " <LTTextBoxHorizontal(9) 72.000,143.609,291.516,197.407 'Lucy Lu Wang, Kyle Lo, Yoganand Chandrasekhar,\\nRussell Reas, Jiangjiang Yang, Darrin Eide, Kathryn\\nFunk, Rodney Kinney, Ziyang Liu, William Merrill,\\net al. 2020. Cord-19: The covid-19 open research\\ndataset. arXiv preprint arXiv:2004.10706.\\n'>,\n",
              " <LTTextBoxHorizontal(10) 72.000,76.772,292.014,130.571 'Quan Wang, Zhendong Mao, Bin Wang, and Li Guo.\\n2017. Knowledge graph embedding: A survey of\\nIEEE Transactions\\napproaches and applications.\\non Knowledge and Data Engineering, 29(12):2724–\\n2743.\\n'>]"
            ]
          },
          "metadata": {},
          "execution_count": 74
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "J-SVTVLi-wnA"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-31_2007.09186v3_flat_text.pickle', 'wb') as f:\n",
        "  pickle.dump(flat_text, f)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(flat_text)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "C02Hy_iPJd1k",
        "outputId": "c9f5df29-aa4f-445e-cc23-eff8672fbe04"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "212"
            ]
          },
          "metadata": {},
          "execution_count": 23
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "flat_text[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "V0SgYJbmJh8F",
        "outputId": "9d8dcdb0-673a-4059-8fe4-d2a501d1c2bf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<LTTextBoxHorizontal(0) 103.371,746.071,493.659,776.357 'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\n'>"
            ]
          },
          "metadata": {},
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 5.2 Keep `LTTextBox` objects by index position\n",
        "\n",
        "Of the 212 `LTTextBox` objects in the PDF, 43 have been manually identified as objects to keep and extract text from. The PDF document does not distinguish between characters that compose a paragraph, title, table or any other element on a page unlike structured data such as XML."
      ],
      "metadata": {
        "id": "5i_5B9o656Rv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "flat_text_indices_keep = flat_text.copy()"
      ],
      "metadata": {
        "id": "pOq9elK1F1Gy"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(flat_text_indices_keep)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0f2dca73-3fa3-4b80-ab15-800429d0ac30",
        "id": "jeOSTeFiF1Gz"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "212"
            ]
          },
          "metadata": {},
          "execution_count": 34
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Function to keep only the objects at the indices in the list."
      ],
      "metadata": {
        "id": "lM_6mTEhG2mt"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "keep_indices = set([6,7,8,\n",
        "                    40,41,43,44,45,46,47,48,\n",
        "                    51,52,54,56,\n",
        "                    62,64,65,66,67,\n",
        "                    97,98,\n",
        "                    100,101,104,105,106,107,\n",
        "                    134,146,147,148,149,\n",
        "                    164,165,167,168,169,\n",
        "                    170,187,189,190,192])\n",
        "\n",
        "def keep_LT_elems(text):\n",
        "    kept_indices = [i for j, i in enumerate(text) if j in keep_indices]\n",
        "    return kept_indices"
      ],
      "metadata": {
        "id": "PK2FaC1x6AFi"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "flat_text_indices_keep = keep_LT_elems(flat_text_indices_keep)"
      ],
      "metadata": {
        "id": "2kA7DcCyAkHm"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(flat_text_indices_keep)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xS_WdaujBMKm",
        "outputId": "30338ef3-06bc-4a3a-9af3-edde8735ec47"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "43"
            ]
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 5.3 Extract text from `LTTextBox` objects\n",
        "Function to extract text from `LTTextBox` objects using get_text() method."
      ],
      "metadata": {
        "id": "WUgv5vS2H8la"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def extract_text(text):\n",
        "    \"\"\"\n",
        "    Get text from LTTextBox objects\n",
        "    for two column page layout\n",
        "    \"\"\"\n",
        "    col_texts = []\n",
        "    col_text  = [textbox.get_text() for textbox in text]\n",
        "    col_texts.append(col_text)\n",
        "\n",
        "    return col_texts"
      ],
      "metadata": {
        "id": "eUUur14z4TEx"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Extract the text from the 43 objects which were kept."
      ],
      "metadata": {
        "id": "RQ8aPSsIya8V"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "extract_text_indices_keep = extract_text(flat_text_indices_keep)\n",
        "extract_text_indices_keep"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_WbbAVElCSzu",
        "outputId": "145749a1-f69c-4a24-c3cc-26bb52435885"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n',\n",
              "  'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n',\n",
              "  'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n',\n",
              "  'viral load. We provide a further advantage from\\nan organizational perspective using topics with do-\\nmain speciﬁc relations. The user can select clinical-\\ntreatments to reﬁne search articles to understand\\nthat “...the highest load [is] assumed to be the day\\nbefore symptoms appear.” Similarly if commis-\\nsioned to understand convalescent plasma therapy\\nand selecting the three topics as shown in the last\\nexample the system identiﬁes the most relevant ar-\\nticle and answers the question in a highlight. There\\nare a few search systems have been developed to\\nsupport COVID-19 search leveraging CORD-19\\ncorpus. We evaluate the ACS against two other\\nsearch engines, and ACS is proved to be a top-\\nperforming system.\\n',\n",
              "  'In the following sections we will discuss the\\nvarious individual Amazon Web Services (AWS)\\nproducts that allow ACS to provide this function-\\nality, highlighting the value to scientists who can\\nquickly query, validate their research, and advance\\ntheir investigations.\\n',\n",
              "  'AWS CORD-19 Search (ACS) relies on a deep\\nsemantic search model to return a ranked list of\\nrelevant documents. This system performs docu-\\nment ranking, passage ranking, question answer-\\ning and FAQ matching. Additionally, it leverages\\nknowledge graphs and topic modeling to enrich the\\nbiomedical data and boost searching performance.\\nThis provides a scalable solution to COVID-19 re-\\nsearchers and policy makers. In this section we\\npresent the overall architecture of the system and a\\ncloser look at several individual components. Fig-\\nure 2 provides an overview of this architecture.\\n',\n",
              "  '2.1 Amazon Kendra\\nAmazon Kendra2 is a semantic search and question\\nanswering service provided by AWS for enterprise\\ncustomers. Kendra allows its customers to power\\nnatural language based searching across their own\\ndata. As response to the worldwide COVID-19\\npandemic, Kendra has also been tooled to support\\nCOVID-19 related searching and question answer-\\ning using the document corpus from CORD-19.\\nThe end-to-end Kendra system consists of several\\ncomponents.\\n',\n",
              "  'To complement the extracted answers, Kendra\\nuses a deep learning based semantic search model\\nto return a ranked list of relevant documents. Ama-\\nzon Kendra’s ability to understand natural language\\nquestions is at the core of its search engine return-\\ning the most relevant passage and related docu-\\nments\\n',\n",
              "  '• Document ranking (DR) - Like any tradi-\\ntional search engine, Kendra returns a ranked\\nlist of relevant documents based on the user’s\\nquery to fulﬁll their information needs. A\\ndeep semantic search model is used to under-\\nstand natural language questions in addition\\nto the keyword search.\\n',\n",
              "  '• Passage ranking (PR) & Question Answer-\\ning (QA) - Kendra ranks the passages and\\ntries to extract the answer from the top rele-\\nvant passages with a deep reading comprehen-\\nsion model.\\n',\n",
              "  '• FAQ Matching (FAQM) - If there exists Fre-\\nquently Answered Questions and their corre-\\nsponding answers, Kendra will automatically\\n',\n",
              "  'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n',\n",
              "  'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n',\n",
              "  'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n',\n",
              "  'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n',\n",
              "  'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n',\n",
              "  'for topic modeling is Latent Dirichlet Allocation\\n(LDA) (Blei et al., 2003), a generative model which\\ngroups documents together by observed content,\\noften giving each document a mixture of topics it\\nbelongs to. An extension of this work termed Z-\\nlabel LDA (Andrzejewski and Zhu, 2009) utilizes\\npriors to allow the model to force certain topics\\nwhich the users have manually curated, or wish to\\nsee clustered together.\\n',\n",
              "  '2.4.1 Generating Topics\\nFor the purposes of this work we experimented\\nwith 5, 10, and 20 topic models 5. The outputs\\nof each clustering size were manually inspected\\nand topic labels were provided by us when inspect-\\ning the top ten terms for each cluster. The ﬁnal\\ngranularity of the topic models was chosen by man-\\nually deleting and merging topics from the 20 topic\\nmodel. In general we were able to clearly extract\\ngroups which centered around important topics in-\\ncluding virology, proteomics, epidemiology, and\\ncellular biology to name a few. However when\\nfaced with 20 topics the less populated ones tended\\nto be noisy, and captured peripheral information\\n',\n",
              "  'present in the input, such as language (e.g. Spanish\\nand French) or provide redundancies with existing\\ntopics (e.g. two topics for Inﬂuenza). As a con-\\ntrol we ran a publicly available implementation of\\nZ-label LDA 6 with no priors which yields topics\\nclose to those extracted using Comprehend Topic\\nModeling. Although similar we observed better\\ndeﬁnition in certain groups (such as pulmonary\\ndiseases, and policy/industry), and decided to use\\nthis as the curation entry-point. Our goal was to\\nlimit these topics to ten, and compile them in ad-\\nvance as much as possible. With the help of medi-\\ncal professionals we eliminated and combined top-\\nics to form the following: Vaccines/immunology,\\nGenomics, Public health Policies, Epidemiology,\\nClinical Treatment, Virology, Inﬂuenza, Healthcare\\nIndustry, Pulmonary Infections, and Lab Trials (hu-\\nman).\\n',\n",
              "  '2.4.2 Multi-Label Classiﬁcation\\nHaving to manually feed a topic model and re-train\\non the entire corpus once new data becomes avail-\\nable is largely inefﬁcient. We therefore used the\\ntopic model labels to train a multi-label classiﬁer\\n',\n",
              "  '(Read et al., 2011). To evaluate the performance\\nof this model we calculate the average F1 across\\ntest samples by calculating the set overlap between\\nour gold standard (topic model) and system labels\\n(multi-label classiﬁcation). This held-out test set\\ncontains 20% of the CORD-19 data available at the\\ntime.\\n',\n",
              "  'Using this metric our trained model achieved an\\naverage F1 of 91.92, with on average 2.37 labels\\nper document. Fewer than 1% of the documents\\nin the test set received no label, using 0.5 as the\\nconﬁdence threshold.\\n',\n",
              "  'ACS supports document ranking (DR), passage\\nranking (PR) and question answering (QA) on the\\ntop suggested results. In this section we evaluate\\nthe overall performance of ACS with publicly avail-\\nable datasets. The TREC-COVID track (Voorhees\\net al., 2020) built DR test collections that contain\\ndocument relevance ground truth and are used to\\nassess DR only. For PR and QA, we manually anno-\\ntate questions collected from CovidQA (Tang et al.,\\n2020), the ﬁrst open sourced Covid-19 QA dataset.\\nThe ACS results are compared against Covidex7\\nand Google Covid-19 Research Explorer8 (COVID-\\n19 RE), which are also COVID-19 public search\\nengines that facilitate DR and answer highlighting\\nin passage.\\n',\n",
              "  'The TREC-COVID challenge track contains 40\\ntopic sets along with their document relevance\\njudgement. The topic sets are written by its or-\\nganizers with biomedical training, and motivated\\nby search submitted to the National Library of\\nMedicine and social media. Each topic consists\\n',\n",
              "  'of three ﬁelds with different levels of granularity, a\\nkeyword-based query (KQ), a more precise natural\\nlanguage question (NQ), and a longer descriptive\\nnarrative. The DR assessments are performed fol-\\nlowing the TREC pooling mechanism. The par-\\nticipants submitted ranked lists of documents for\\neach topic set, based on which a depth of 10 to 20\\ndocuments are pooled and combined as a collection\\nof (q, D) pairs. The pairs are then assessed by an-\\nnotators with in-domain expertise. There are three\\nrounds of judgement available that corresponding\\nto three different versions of CORD-19 corpus. To\\nensure sufﬁcient coverage of annotation, we aggre-\\ngate all rounds results into 33,064 (q, D) relevance\\njudgements. Since the document id may change\\nacross versions, we map ids from each round to the\\nMay 19 release of CORD-19 corpus.\\n',\n",
              "  'To collect DR results from the three systems, we\\ncrawled the top 50 articles by querying the engines\\nwith KQ and its NQ variation from the topic sets on\\nJune 15, 2020. The crawled data, which are char-\\nacterized by the article title and link, are mapped\\nto May 19 CORD-19 corpus. Note that articles\\nthat cannot be found in the corpus are removed to\\nensure fair comparison.\\n',\n",
              "  'We use the standard DR metrics in our evalua-\\ntion, namely, the precision and recall at k (P@k,\\nR@k) and normalized discounted cumulative gain\\nin the top k documents (NDCG@k). Note that we\\nevaluate with k up to 20 since TREC-COVID has\\na pooling depth with 20 at most. Table 1 presents\\nthe DR performance of the engines over KQ and\\nNQ, respectively. ACS performs consistently better\\nthan the other engines on NQ and mostly on KQ,\\nand all engines perform better on NQ comparing\\nwith KQ.\\n',\n",
              "  'In addition, we are able to evaluate how robust\\neach system is against query variation from KQ\\n',\n",
              "  'to NQ. Ideally, the results shall remain unchanged\\nwith query variation that requests the same infor-\\nmation. We deﬁne exact match (EM) and F1 score\\namong top k results to evaluate the robustness. Let\\nQ be the topic sets, and N Qk\\nq denote the\\ntop k searching results of natural language ques-\\ntion and keyword query corresponding to the same\\ntopic q ∈ Q, respectively. As shown in Eq.(1), we\\ntake the top k results KQk\\nq as ground truth, and\\ncompute the average exact match and F1 score of\\nthe top k results N Qk\\nq , and then average over all\\nqueries. The article title string is used as com-\\nparison key, and EM and F1 are standard that the\\nmaximum is taken over all the ground truth arti-\\ncles. Table 2 demonstrates that ACS has the best\\ncapability to provide consistent results with query\\nvariation.\\n',\n",
              "  'Next, we evaluate the performance of PR and QA\\nwith CovidQA dataset. CovidQA contains 27 ques-\\ntions and their answers from 124 question-article\\npairs, which are selected by in-domain volunteers\\nas the most promising COVID-19 literature. Since\\nthe annotation does not possess sufﬁcient answer\\ncoverage over the entire corpus, we leverage our\\ninternal annotation resources to make PR and QA\\njudgement. More explicitly, we crawled the top 3\\nresults characterized by the article, displayed pas-\\nsage and the highlighted text snippet on June 15,\\n',\n",
              "  '2020 from ACS, Covidex and COVID-19 RE, re-\\nspectively. After that, the annotators assess PR\\nand QA in terms of whether the displayed passage\\ncontains relevant information and whether the high-\\nlighted text snippet answers the given question. To\\navoid bias, the crawled results are combined and\\nshufﬂed randomly, therefore, the annotators do not\\nhave access to the source engine and the rank posi-\\ntion information during the judgement.\\n',\n",
              "  'Table 3 presents precision of top PR and QA\\nresults. We use P@k instead of EM and F1 to eval-\\nuate QA since ACS highlights the answer while\\nCovidex and COVID-19 RE highlight the sentence\\nthat contains the answer. Instead of extracting the\\nanswer, it is more reasonable to judge whether the\\nhighlighted text answers the question and compute\\nprecision. ACS achieves better accuracy on both\\nPR and QA on most metrics. Note that ACS high-\\nlights answer snippet for at most three passages\\n, while Covidex and COVID-19 RE highlight all\\ndisplayed passages. This explains why ACS P@3\\nunderperforms COVID-19 RE. Explicitly with an\\nexample, Figure 4 displays the topmost results of\\nquerying “What is the incubation period of virus”\\nfrom the three systems. ACS highlights the answer\\nin the passage. In contrast, Covidex presents an\\narticle published at 2004 with information of virus\\nwhich is irrelevant with coronavirus, and COVID-\\n19 RE does not answer the question at all.\\n',\n",
              "  'As evident from the above results, ACS is one\\nof the top-performing systems that provides high\\nquality informative results over CORD-19 search.\\n',\n",
              "  'In this section we look into a number of sample\\nqueries to shed light on how different components\\nof ACS help in improving search results. We begin\\nby observing how small semantic differences in the\\nquery alter the results. The ﬁrst sample in Figure 5\\nis speciﬁc to medications. While the top result does\\nnot include the term medication the system high-\\nlights ribavirin and corticosteroids. The CORD-19\\nsystem understands that these terms represent med-\\nications with the help of CM NERe engine. In the\\nsecond example we change medications to mea-\\nsures and observe the top result discussing border\\ncontrol, and quarantine. This clearly demonstrates\\nthat Amazon Kendra has a deep comprehension of\\ntoken and query meanings.\\n',\n",
              "  'Finally, we take a look at the effects of topic mod-\\neling when grouping and ﬁltering results. The last\\ntwo examples in Figure 5 showcase the difference\\nthis makes in the top result. Without specifying any\\ntopic the resulting article discusses high level pol-\\nicy, speciﬁcally quarantine measures in Singapore.\\nWhen we ﬁlter by clinical treatment the top result\\ninstead focuses on infections which is covered in\\nthe clinical setting. Furthermore the extracted text\\nreturned to the user still focuses on lessons learned\\nstaying true to the query.\\n',\n",
              "  'AWS CORD-19 is an initial step towards help-\\ning medical researchers ﬁnd relevant content in\\na timely and meaningful way. In order to improve\\n',\n",
              "  'the robustness, we see following areas as direction\\nfor future research.\\n',\n",
              "  'Feedback Loop - Since ACS is a search engine\\nthe motivation would be to evaluate it as such; us-\\ning well-established methodologies based on test\\ncollections—comprising topics (information needs)\\nand human annotations. Since no designated evalu-\\nation data exist, our initial focus is to capture dif-\\nferent interactions and feedback. Currently, ACS\\nlacks the feedback loop and federated learning ap-\\nproaches where the system would continuously\\nlearn and improve the search. However, the system\\ncaptures feedback from the researchers in the form\\nof implicit and explicit reactions. Implicit feed-\\nback evaluation consists of topics of interests, their\\nclicks as well as the ranking of the results which\\nwere selected by medical researchers. Explicit feed-\\nback evaluation is captured by providing up-down\\nrating associated with each search results. In the\\nfuture results can be personalized based on this\\nfeedback. Now that we have a system in place, our\\nefforts have shifted to broader engagement with po-\\ntential stakeholders to solicit additional guidance,\\nwhile trying to balance between the features and\\nranking.\\n',\n",
              "  'Q&A Curation - Curation and normalization of\\nquestions have potential a use-case of presenting\\ntrending questions asked by the medical research\\ncommunity at a particular point. However, curation\\nwould involve capturing the questions asked as well\\nas identifying similar questions that can be later\\nnormalized. Currently, there is no mechanism to\\n',\n",
              "  'curate the questions asked by the researchers.\\n',\n",
              "  'Summarization - Currently, ACS outputs the\\nrelevant passage based on the query. It would be\\nbeneﬁcial to get the overall summary of the pa-\\nper. A potential future direction would be to gen-\\nerates summaries (Raffel et al., 2019) from paper\\nabstracts and full body.\\n',\n",
              "  'We acknowledge the broader collaboration with\\nAI2 team and White house as well as the broader\\nAWS CORD-19 team including Kendra Science\\nteam, Tyler Stepke, Kingston Bosco, Victor Wang,\\nVaibhav Chaddha, Miguel Calvo, Ninad Kulkani,\\nKevin Longofer, Ray Chang, Adrian Bordone,\\nTony Nguyen, and Kyle Johnson.\\n',\n",
              "  'This paper describes our efforts in building AWS\\nCORD-19 Search with its capabilities consisting\\nof topic based, knowledge graph, and natural lan-\\nguage search queries. This is further enhanced\\nwith reading comprehension and FAQ matching\\nas well as document ranking providing a scalable\\nsolution to COVID-19 researchers and policy mak-\\ners in their search and discovery for answers to\\nhigh priority scientiﬁc questions. Our solution is\\npowered by Amazon Kendra, Comprehend Medi-\\ncal and Neptune which incorporate the latest neural\\narchitectures to provide information access capa-\\nbilities to the CORD-19 challenge. By comparing\\nwith other public search engines that support sim-\\nilar functionalities over COVID-19 search, ACS\\nis demonstrated to be powerful on its document\\nranking and question answering components. We\\nhope that our solution can prove useful in the ﬁght\\nagainst this global pandemic, and that the capabili-\\nties we have developed can be applied to analyzing\\nthe scientiﬁc literature more broadly.\\n']]"
            ]
          },
          "metadata": {},
          "execution_count": 39
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "g0gHE6SmCdzd"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-31_2007.09186v3_extract_text_indices_keep.pickle', 'wb') as f:\n",
        "  pickle.dump(extract_text_indices_keep, f)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "extract_text_indices_keep[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nmBUBGjkCmQI",
        "outputId": "f6787421-997f-46b5-b029-f4af9c419d25"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\n',\n",
              " 'the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\n',\n",
              " 'AWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\n',\n",
              " 'viral load. We provide a further advantage from\\nan organizational perspective using topics with do-\\nmain speciﬁc relations. The user can select clinical-\\ntreatments to reﬁne search articles to understand\\nthat “...the highest load [is] assumed to be the day\\nbefore symptoms appear.” Similarly if commis-\\nsioned to understand convalescent plasma therapy\\nand selecting the three topics as shown in the last\\nexample the system identiﬁes the most relevant ar-\\nticle and answers the question in a highlight. There\\nare a few search systems have been developed to\\nsupport COVID-19 search leveraging CORD-19\\ncorpus. We evaluate the ACS against two other\\nsearch engines, and ACS is proved to be a top-\\nperforming system.\\n',\n",
              " 'In the following sections we will discuss the\\nvarious individual Amazon Web Services (AWS)\\nproducts that allow ACS to provide this function-\\nality, highlighting the value to scientists who can\\nquickly query, validate their research, and advance\\ntheir investigations.\\n',\n",
              " 'AWS CORD-19 Search (ACS) relies on a deep\\nsemantic search model to return a ranked list of\\nrelevant documents. This system performs docu-\\nment ranking, passage ranking, question answer-\\ning and FAQ matching. Additionally, it leverages\\nknowledge graphs and topic modeling to enrich the\\nbiomedical data and boost searching performance.\\nThis provides a scalable solution to COVID-19 re-\\nsearchers and policy makers. In this section we\\npresent the overall architecture of the system and a\\ncloser look at several individual components. Fig-\\nure 2 provides an overview of this architecture.\\n',\n",
              " '2.1 Amazon Kendra\\nAmazon Kendra2 is a semantic search and question\\nanswering service provided by AWS for enterprise\\ncustomers. Kendra allows its customers to power\\nnatural language based searching across their own\\ndata. As response to the worldwide COVID-19\\npandemic, Kendra has also been tooled to support\\nCOVID-19 related searching and question answer-\\ning using the document corpus from CORD-19.\\nThe end-to-end Kendra system consists of several\\ncomponents.\\n',\n",
              " 'To complement the extracted answers, Kendra\\nuses a deep learning based semantic search model\\nto return a ranked list of relevant documents. Ama-\\nzon Kendra’s ability to understand natural language\\nquestions is at the core of its search engine return-\\ning the most relevant passage and related docu-\\nments\\n',\n",
              " '• Document ranking (DR) - Like any tradi-\\ntional search engine, Kendra returns a ranked\\nlist of relevant documents based on the user’s\\nquery to fulﬁll their information needs. A\\ndeep semantic search model is used to under-\\nstand natural language questions in addition\\nto the keyword search.\\n',\n",
              " '• Passage ranking (PR) & Question Answer-\\ning (QA) - Kendra ranks the passages and\\ntries to extract the answer from the top rele-\\nvant passages with a deep reading comprehen-\\nsion model.\\n',\n",
              " '• FAQ Matching (FAQM) - If there exists Fre-\\nquently Answered Questions and their corre-\\nsponding answers, Kendra will automatically\\n',\n",
              " 'match a new coming query with FAQs and\\nextract the answer if a strong match is found.\\n',\n",
              " 'In order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\n',\n",
              " 'Amazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\n',\n",
              " 'Knowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\n',\n",
              " 'Topic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\n',\n",
              " 'for topic modeling is Latent Dirichlet Allocation\\n(LDA) (Blei et al., 2003), a generative model which\\ngroups documents together by observed content,\\noften giving each document a mixture of topics it\\nbelongs to. An extension of this work termed Z-\\nlabel LDA (Andrzejewski and Zhu, 2009) utilizes\\npriors to allow the model to force certain topics\\nwhich the users have manually curated, or wish to\\nsee clustered together.\\n',\n",
              " '2.4.1 Generating Topics\\nFor the purposes of this work we experimented\\nwith 5, 10, and 20 topic models 5. The outputs\\nof each clustering size were manually inspected\\nand topic labels were provided by us when inspect-\\ning the top ten terms for each cluster. The ﬁnal\\ngranularity of the topic models was chosen by man-\\nually deleting and merging topics from the 20 topic\\nmodel. In general we were able to clearly extract\\ngroups which centered around important topics in-\\ncluding virology, proteomics, epidemiology, and\\ncellular biology to name a few. However when\\nfaced with 20 topics the less populated ones tended\\nto be noisy, and captured peripheral information\\n',\n",
              " 'present in the input, such as language (e.g. Spanish\\nand French) or provide redundancies with existing\\ntopics (e.g. two topics for Inﬂuenza). As a con-\\ntrol we ran a publicly available implementation of\\nZ-label LDA 6 with no priors which yields topics\\nclose to those extracted using Comprehend Topic\\nModeling. Although similar we observed better\\ndeﬁnition in certain groups (such as pulmonary\\ndiseases, and policy/industry), and decided to use\\nthis as the curation entry-point. Our goal was to\\nlimit these topics to ten, and compile them in ad-\\nvance as much as possible. With the help of medi-\\ncal professionals we eliminated and combined top-\\nics to form the following: Vaccines/immunology,\\nGenomics, Public health Policies, Epidemiology,\\nClinical Treatment, Virology, Inﬂuenza, Healthcare\\nIndustry, Pulmonary Infections, and Lab Trials (hu-\\nman).\\n',\n",
              " '2.4.2 Multi-Label Classiﬁcation\\nHaving to manually feed a topic model and re-train\\non the entire corpus once new data becomes avail-\\nable is largely inefﬁcient. We therefore used the\\ntopic model labels to train a multi-label classiﬁer\\n',\n",
              " '(Read et al., 2011). To evaluate the performance\\nof this model we calculate the average F1 across\\ntest samples by calculating the set overlap between\\nour gold standard (topic model) and system labels\\n(multi-label classiﬁcation). This held-out test set\\ncontains 20% of the CORD-19 data available at the\\ntime.\\n',\n",
              " 'Using this metric our trained model achieved an\\naverage F1 of 91.92, with on average 2.37 labels\\nper document. Fewer than 1% of the documents\\nin the test set received no label, using 0.5 as the\\nconﬁdence threshold.\\n',\n",
              " 'ACS supports document ranking (DR), passage\\nranking (PR) and question answering (QA) on the\\ntop suggested results. In this section we evaluate\\nthe overall performance of ACS with publicly avail-\\nable datasets. The TREC-COVID track (Voorhees\\net al., 2020) built DR test collections that contain\\ndocument relevance ground truth and are used to\\nassess DR only. For PR and QA, we manually anno-\\ntate questions collected from CovidQA (Tang et al.,\\n2020), the ﬁrst open sourced Covid-19 QA dataset.\\nThe ACS results are compared against Covidex7\\nand Google Covid-19 Research Explorer8 (COVID-\\n19 RE), which are also COVID-19 public search\\nengines that facilitate DR and answer highlighting\\nin passage.\\n',\n",
              " 'The TREC-COVID challenge track contains 40\\ntopic sets along with their document relevance\\njudgement. The topic sets are written by its or-\\nganizers with biomedical training, and motivated\\nby search submitted to the National Library of\\nMedicine and social media. Each topic consists\\n',\n",
              " 'of three ﬁelds with different levels of granularity, a\\nkeyword-based query (KQ), a more precise natural\\nlanguage question (NQ), and a longer descriptive\\nnarrative. The DR assessments are performed fol-\\nlowing the TREC pooling mechanism. The par-\\nticipants submitted ranked lists of documents for\\neach topic set, based on which a depth of 10 to 20\\ndocuments are pooled and combined as a collection\\nof (q, D) pairs. The pairs are then assessed by an-\\nnotators with in-domain expertise. There are three\\nrounds of judgement available that corresponding\\nto three different versions of CORD-19 corpus. To\\nensure sufﬁcient coverage of annotation, we aggre-\\ngate all rounds results into 33,064 (q, D) relevance\\njudgements. Since the document id may change\\nacross versions, we map ids from each round to the\\nMay 19 release of CORD-19 corpus.\\n',\n",
              " 'To collect DR results from the three systems, we\\ncrawled the top 50 articles by querying the engines\\nwith KQ and its NQ variation from the topic sets on\\nJune 15, 2020. The crawled data, which are char-\\nacterized by the article title and link, are mapped\\nto May 19 CORD-19 corpus. Note that articles\\nthat cannot be found in the corpus are removed to\\nensure fair comparison.\\n',\n",
              " 'We use the standard DR metrics in our evalua-\\ntion, namely, the precision and recall at k (P@k,\\nR@k) and normalized discounted cumulative gain\\nin the top k documents (NDCG@k). Note that we\\nevaluate with k up to 20 since TREC-COVID has\\na pooling depth with 20 at most. Table 1 presents\\nthe DR performance of the engines over KQ and\\nNQ, respectively. ACS performs consistently better\\nthan the other engines on NQ and mostly on KQ,\\nand all engines perform better on NQ comparing\\nwith KQ.\\n',\n",
              " 'In addition, we are able to evaluate how robust\\neach system is against query variation from KQ\\n',\n",
              " 'to NQ. Ideally, the results shall remain unchanged\\nwith query variation that requests the same infor-\\nmation. We deﬁne exact match (EM) and F1 score\\namong top k results to evaluate the robustness. Let\\nQ be the topic sets, and N Qk\\nq denote the\\ntop k searching results of natural language ques-\\ntion and keyword query corresponding to the same\\ntopic q ∈ Q, respectively. As shown in Eq.(1), we\\ntake the top k results KQk\\nq as ground truth, and\\ncompute the average exact match and F1 score of\\nthe top k results N Qk\\nq , and then average over all\\nqueries. The article title string is used as com-\\nparison key, and EM and F1 are standard that the\\nmaximum is taken over all the ground truth arti-\\ncles. Table 2 demonstrates that ACS has the best\\ncapability to provide consistent results with query\\nvariation.\\n',\n",
              " 'Next, we evaluate the performance of PR and QA\\nwith CovidQA dataset. CovidQA contains 27 ques-\\ntions and their answers from 124 question-article\\npairs, which are selected by in-domain volunteers\\nas the most promising COVID-19 literature. Since\\nthe annotation does not possess sufﬁcient answer\\ncoverage over the entire corpus, we leverage our\\ninternal annotation resources to make PR and QA\\njudgement. More explicitly, we crawled the top 3\\nresults characterized by the article, displayed pas-\\nsage and the highlighted text snippet on June 15,\\n',\n",
              " '2020 from ACS, Covidex and COVID-19 RE, re-\\nspectively. After that, the annotators assess PR\\nand QA in terms of whether the displayed passage\\ncontains relevant information and whether the high-\\nlighted text snippet answers the given question. To\\navoid bias, the crawled results are combined and\\nshufﬂed randomly, therefore, the annotators do not\\nhave access to the source engine and the rank posi-\\ntion information during the judgement.\\n',\n",
              " 'Table 3 presents precision of top PR and QA\\nresults. We use P@k instead of EM and F1 to eval-\\nuate QA since ACS highlights the answer while\\nCovidex and COVID-19 RE highlight the sentence\\nthat contains the answer. Instead of extracting the\\nanswer, it is more reasonable to judge whether the\\nhighlighted text answers the question and compute\\nprecision. ACS achieves better accuracy on both\\nPR and QA on most metrics. Note that ACS high-\\nlights answer snippet for at most three passages\\n, while Covidex and COVID-19 RE highlight all\\ndisplayed passages. This explains why ACS P@3\\nunderperforms COVID-19 RE. Explicitly with an\\nexample, Figure 4 displays the topmost results of\\nquerying “What is the incubation period of virus”\\nfrom the three systems. ACS highlights the answer\\nin the passage. In contrast, Covidex presents an\\narticle published at 2004 with information of virus\\nwhich is irrelevant with coronavirus, and COVID-\\n19 RE does not answer the question at all.\\n',\n",
              " 'As evident from the above results, ACS is one\\nof the top-performing systems that provides high\\nquality informative results over CORD-19 search.\\n',\n",
              " 'In this section we look into a number of sample\\nqueries to shed light on how different components\\nof ACS help in improving search results. We begin\\nby observing how small semantic differences in the\\nquery alter the results. The ﬁrst sample in Figure 5\\nis speciﬁc to medications. While the top result does\\nnot include the term medication the system high-\\nlights ribavirin and corticosteroids. The CORD-19\\nsystem understands that these terms represent med-\\nications with the help of CM NERe engine. In the\\nsecond example we change medications to mea-\\nsures and observe the top result discussing border\\ncontrol, and quarantine. This clearly demonstrates\\nthat Amazon Kendra has a deep comprehension of\\ntoken and query meanings.\\n',\n",
              " 'Finally, we take a look at the effects of topic mod-\\neling when grouping and ﬁltering results. The last\\ntwo examples in Figure 5 showcase the difference\\nthis makes in the top result. Without specifying any\\ntopic the resulting article discusses high level pol-\\nicy, speciﬁcally quarantine measures in Singapore.\\nWhen we ﬁlter by clinical treatment the top result\\ninstead focuses on infections which is covered in\\nthe clinical setting. Furthermore the extracted text\\nreturned to the user still focuses on lessons learned\\nstaying true to the query.\\n',\n",
              " 'AWS CORD-19 is an initial step towards help-\\ning medical researchers ﬁnd relevant content in\\na timely and meaningful way. In order to improve\\n',\n",
              " 'the robustness, we see following areas as direction\\nfor future research.\\n',\n",
              " 'Feedback Loop - Since ACS is a search engine\\nthe motivation would be to evaluate it as such; us-\\ning well-established methodologies based on test\\ncollections—comprising topics (information needs)\\nand human annotations. Since no designated evalu-\\nation data exist, our initial focus is to capture dif-\\nferent interactions and feedback. Currently, ACS\\nlacks the feedback loop and federated learning ap-\\nproaches where the system would continuously\\nlearn and improve the search. However, the system\\ncaptures feedback from the researchers in the form\\nof implicit and explicit reactions. Implicit feed-\\nback evaluation consists of topics of interests, their\\nclicks as well as the ranking of the results which\\nwere selected by medical researchers. Explicit feed-\\nback evaluation is captured by providing up-down\\nrating associated with each search results. In the\\nfuture results can be personalized based on this\\nfeedback. Now that we have a system in place, our\\nefforts have shifted to broader engagement with po-\\ntential stakeholders to solicit additional guidance,\\nwhile trying to balance between the features and\\nranking.\\n',\n",
              " 'Q&A Curation - Curation and normalization of\\nquestions have potential a use-case of presenting\\ntrending questions asked by the medical research\\ncommunity at a particular point. However, curation\\nwould involve capturing the questions asked as well\\nas identifying similar questions that can be later\\nnormalized. Currently, there is no mechanism to\\n',\n",
              " 'curate the questions asked by the researchers.\\n',\n",
              " 'Summarization - Currently, ACS outputs the\\nrelevant passage based on the query. It would be\\nbeneﬁcial to get the overall summary of the pa-\\nper. A potential future direction would be to gen-\\nerates summaries (Raffel et al., 2019) from paper\\nabstracts and full body.\\n',\n",
              " 'We acknowledge the broader collaboration with\\nAI2 team and White house as well as the broader\\nAWS CORD-19 team including Kendra Science\\nteam, Tyler Stepke, Kingston Bosco, Victor Wang,\\nVaibhav Chaddha, Miguel Calvo, Ninad Kulkani,\\nKevin Longofer, Ray Chang, Adrian Bordone,\\nTony Nguyen, and Kyle Johnson.\\n',\n",
              " 'This paper describes our efforts in building AWS\\nCORD-19 Search with its capabilities consisting\\nof topic based, knowledge graph, and natural lan-\\nguage search queries. This is further enhanced\\nwith reading comprehension and FAQ matching\\nas well as document ranking providing a scalable\\nsolution to COVID-19 researchers and policy mak-\\ners in their search and discovery for answers to\\nhigh priority scientiﬁc questions. Our solution is\\npowered by Amazon Kendra, Comprehend Medi-\\ncal and Neptune which incorporate the latest neural\\narchitectures to provide information access capa-\\nbilities to the CORD-19 challenge. By comparing\\nwith other public search engines that support sim-\\nilar functionalities over COVID-19 search, ACS\\nis demonstrated to be powerful on its document\\nranking and question answering components. We\\nhope that our solution can prove useful in the ﬁght\\nagainst this global pandemic, and that the capabili-\\nties we have developed can be applied to analyzing\\nthe scientiﬁc literature more broadly.\\n']"
            ]
          },
          "metadata": {},
          "execution_count": 4
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "It appears that three headings were included in error (2.1 Amazon Kendra [45], 2.4.1 Generating Topics [66] and 2.4.2 Multi-Label Classiﬁcation [68]). Further text cleaning will be required and these will be removed. The process was rather manual and would not really be scalable had there been more documents."
      ],
      "metadata": {
        "id": "2rotCvOvLfcI"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 5.4 Join list of extracted text strings\n",
        "\n",
        "Join list into a single string of text for further processing."
      ],
      "metadata": {
        "id": "s2KdmXC4zzO3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Flatten list into a single list of strings\n",
        "flattened_list = [item for sublist in extract_text_indices_keep for item in sublist]\n",
        "\n",
        "# Join the flattened list into a single string\n",
        "joined_text_indices_keep = ''.join(flattened_list)\n",
        "joined_text_indices_keep"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "id": "L7Xsjr9oVwK4",
        "outputId": "f2524804-1bbf-4653-bf2c-2cd17bc273e1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'With the global outbreak of Coronavirus disease\\n(COVID-19) (Guan et al., 2020), the world is in\\nturmoil. Medical researchers are required to work\\nquickly to fully understand and to provide a form of\\nintervention for the virus. Due to a large research\\nfocus on the disease, knowledge is published at a\\nrapid rate throughout the world. One such reposi-\\ntory of information is curated through the COVID-\\n19 Open Research Dataset Challenge (CORD-19)\\n(Wang et al., 2020). CORD-19 is a joint challenge\\nput forth by Allen Institute (AI2), National Insti-\\ntutes of Health (NIH), and the United States federal\\ngovernment via the White House. The objective of\\nthe challenge is to make sense of and extract useful\\nknowledge across thousands of scholarly articles\\nrelated to COVID-19. CORD-19 aims to connect\\nthe machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\ntify effective treatments and management policies\\nfor COVID-19. In accordance with this initiative\\nour goal is to provide a scalable solution aimed at\\naiding COVID-19 researchers and policy makers in\\ntheir search and discovery for answers to high prior-\\nity scientiﬁc questions. These questions should be\\nunderstood in their natural language form; exam-\\nples include: “What do we know about COVID-19\\nrisk factors?”, “Which medications were most ben-\\neﬁcial in the 2002 SARS outbreak?”, and “Which is\\nthe most referenced paper doing study for Hydroxy-\\nchloroquine?” To appropriately answer these ques-\\ntions we require a system with a strong biomedical\\nunderstanding of the natural queries and structured\\nknowledge (Rotmensch et al., 2017).\\nAWS CORD-19 Search (ACS) provides an easy\\nto use search interface where researchers query us-\\ning natural language 1. ACS goes beyond keyword\\nmatching by understanding question semantics to\\nefﬁciently ﬁnd relevant answers. As illustrated in\\nFigure 1, we provide the system with a natural lan-\\nguage query inquiring about IL-6 inhibitors and\\nshowcase the system response with relevant query\\ncomponents highlighted. This query can confuse a\\ntraditional search engine that relies solely on text\\nmatching as it can only capture term overlap be-\\ntween a query and a document and may not nec-\\nessarily be relevant to the researcher’s true intent\\n(i.e. learning implicit relations). Our system, how-\\never, establishes the relationship between IL-6 and\\nseverity of SARS-CoV-2 showing evidence where\\nelevated IL-6 occur in a large number of patients\\nwith severe COVID-19 and higher mortality rates.\\nIn the following example, if tasked with un-\\nderstanding the epidemiology and transmission of\\nCOVID-19 a researcher may ask about salivary\\nviral load. We provide a further advantage from\\nan organizational perspective using topics with do-\\nmain speciﬁc relations. The user can select clinical-\\ntreatments to reﬁne search articles to understand\\nthat “...the highest load [is] assumed to be the day\\nbefore symptoms appear.” Similarly if commis-\\nsioned to understand convalescent plasma therapy\\nand selecting the three topics as shown in the last\\nexample the system identiﬁes the most relevant ar-\\nticle and answers the question in a highlight. There\\nare a few search systems have been developed to\\nsupport COVID-19 search leveraging CORD-19\\ncorpus. We evaluate the ACS against two other\\nsearch engines, and ACS is proved to be a top-\\nperforming system.\\nIn the following sections we will discuss the\\nvarious individual Amazon Web Services (AWS)\\nproducts that allow ACS to provide this function-\\nality, highlighting the value to scientists who can\\nquickly query, validate their research, and advance\\ntheir investigations.\\nAWS CORD-19 Search (ACS) relies on a deep\\nsemantic search model to return a ranked list of\\nrelevant documents. This system performs docu-\\nment ranking, passage ranking, question answer-\\ning and FAQ matching. Additionally, it leverages\\nknowledge graphs and topic modeling to enrich the\\nbiomedical data and boost searching performance.\\nThis provides a scalable solution to COVID-19 re-\\nsearchers and policy makers. In this section we\\npresent the overall architecture of the system and a\\ncloser look at several individual components. Fig-\\nure 2 provides an overview of this architecture.\\n2.1 Amazon Kendra\\nAmazon Kendra2 is a semantic search and question\\nanswering service provided by AWS for enterprise\\ncustomers. Kendra allows its customers to power\\nnatural language based searching across their own\\ndata. As response to the worldwide COVID-19\\npandemic, Kendra has also been tooled to support\\nCOVID-19 related searching and question answer-\\ning using the document corpus from CORD-19.\\nThe end-to-end Kendra system consists of several\\ncomponents.\\nTo complement the extracted answers, Kendra\\nuses a deep learning based semantic search model\\nto return a ranked list of relevant documents. Ama-\\nzon Kendra’s ability to understand natural language\\nquestions is at the core of its search engine return-\\ning the most relevant passage and related docu-\\nments\\n• Document ranking (DR) - Like any tradi-\\ntional search engine, Kendra returns a ranked\\nlist of relevant documents based on the user’s\\nquery to fulﬁll their information needs. A\\ndeep semantic search model is used to under-\\nstand natural language questions in addition\\nto the keyword search.\\n• Passage ranking (PR) & Question Answer-\\ning (QA) - Kendra ranks the passages and\\ntries to extract the answer from the top rele-\\nvant passages with a deep reading comprehen-\\nsion model.\\n• FAQ Matching (FAQM) - If there exists Fre-\\nquently Answered Questions and their corre-\\nsponding answers, Kendra will automatically\\nmatch a new coming query with FAQs and\\nextract the answer if a strong match is found.\\nIn order to improve Kendra CORD-19 search\\nand make it clinically more relevant for medical\\nresearchers, we leverage knowledge extracted us-\\ning the Amazon Comprehend Medical (CM) core\\nNERe service as well as topics created from Ama-\\nzon Comprehend Custom Classiﬁcation 3. More\\nexplicitly, a knowledge graph is built using the\\nmedical entities extracted with CM NERe API, and\\nthe topics of the article are produced using a semi-\\nsupervised prior based LDA approach. Both are\\nused to enrich the data when indexing Kendra.\\nAmazon Comprehend Medical 4 (CM) (Bhatia\\net al., 2019), is a HIPAA eligible AWS service for\\nmedical domain entity recognition (Bhatia et al.,\\n2018), relationship extraction (Singh and Bhatia,\\n2019) and normalization. Comprehend Medical\\nsupports entity types divided into ﬁve different cat-\\negories (Anatomy, Medical Condition, Medication,\\nProtected Health Information, and Test, Treatment,\\n& Procedure) and four traits (Negation, Diagnosis,\\nSign and Symptom). These entities are directly\\nused to enrich the Kendra search.\\nKnowledge graphs (KGs) are structural representa-\\ntions of relations between real-world entities in the\\nform of triplets containing a head entity, a tail entity,\\nand the relation type connecting them. KG based\\ninformation retrieval has shown great success in the\\npast decades (Dalton et al., 2014). The COVID-19\\nKnowledge Graph (CKG) (Fig 3) is a directed prop-\\nerty graph constructed from the CORD-19 Open\\nResearch Dataset of scholarly articles. Entities\\nincluding scholarly articles, authors, author institu-\\ntions, citations, extracted topics and comprehend\\nmedical entities are used to form relations in the\\nCKG. The resulting KG continues to grow as the\\nCORD-19 dataset increases and currently contains\\nover 335k entities and 3.3M relations. The CKG\\npowers a number of features on ACS including: ar-\\nticle recommendations, citation-based navigation,\\nand search result ranking by author or institution\\npublication count. Scientiﬁc article recommenda-\\ntions are made possible by a document similar-\\nity engine that quantiﬁes similarity between doc-\\numents by combining semantic embeddings ob-\\ntained from a pre-trained language model (Beltagy\\net al., 2019) with document knowledge graph em-\\nbeddings (Wang et al., 2017; Zheng et al., 2020)\\ncapturing topological information from the CKG.\\nTopic modeling is a statistical discovery paradigm\\nfor generating topics that occur in a collection of\\ndocuments. Perhaps the most widely used model\\nfor topic modeling is Latent Dirichlet Allocation\\n(LDA) (Blei et al., 2003), a generative model which\\ngroups documents together by observed content,\\noften giving each document a mixture of topics it\\nbelongs to. An extension of this work termed Z-\\nlabel LDA (Andrzejewski and Zhu, 2009) utilizes\\npriors to allow the model to force certain topics\\nwhich the users have manually curated, or wish to\\nsee clustered together.\\n2.4.1 Generating Topics\\nFor the purposes of this work we experimented\\nwith 5, 10, and 20 topic models 5. The outputs\\nof each clustering size were manually inspected\\nand topic labels were provided by us when inspect-\\ning the top ten terms for each cluster. The ﬁnal\\ngranularity of the topic models was chosen by man-\\nually deleting and merging topics from the 20 topic\\nmodel. In general we were able to clearly extract\\ngroups which centered around important topics in-\\ncluding virology, proteomics, epidemiology, and\\ncellular biology to name a few. However when\\nfaced with 20 topics the less populated ones tended\\nto be noisy, and captured peripheral information\\npresent in the input, such as language (e.g. Spanish\\nand French) or provide redundancies with existing\\ntopics (e.g. two topics for Inﬂuenza). As a con-\\ntrol we ran a publicly available implementation of\\nZ-label LDA 6 with no priors which yields topics\\nclose to those extracted using Comprehend Topic\\nModeling. Although similar we observed better\\ndeﬁnition in certain groups (such as pulmonary\\ndiseases, and policy/industry), and decided to use\\nthis as the curation entry-point. Our goal was to\\nlimit these topics to ten, and compile them in ad-\\nvance as much as possible. With the help of medi-\\ncal professionals we eliminated and combined top-\\nics to form the following: Vaccines/immunology,\\nGenomics, Public health Policies, Epidemiology,\\nClinical Treatment, Virology, Inﬂuenza, Healthcare\\nIndustry, Pulmonary Infections, and Lab Trials (hu-\\nman).\\n2.4.2 Multi-Label Classiﬁcation\\nHaving to manually feed a topic model and re-train\\non the entire corpus once new data becomes avail-\\nable is largely inefﬁcient. We therefore used the\\ntopic model labels to train a multi-label classiﬁer\\n(Read et al., 2011). To evaluate the performance\\nof this model we calculate the average F1 across\\ntest samples by calculating the set overlap between\\nour gold standard (topic model) and system labels\\n(multi-label classiﬁcation). This held-out test set\\ncontains 20% of the CORD-19 data available at the\\ntime.\\nUsing this metric our trained model achieved an\\naverage F1 of 91.92, with on average 2.37 labels\\nper document. Fewer than 1% of the documents\\nin the test set received no label, using 0.5 as the\\nconﬁdence threshold.\\nACS supports document ranking (DR), passage\\nranking (PR) and question answering (QA) on the\\ntop suggested results. In this section we evaluate\\nthe overall performance of ACS with publicly avail-\\nable datasets. The TREC-COVID track (Voorhees\\net al., 2020) built DR test collections that contain\\ndocument relevance ground truth and are used to\\nassess DR only. For PR and QA, we manually anno-\\ntate questions collected from CovidQA (Tang et al.,\\n2020), the ﬁrst open sourced Covid-19 QA dataset.\\nThe ACS results are compared against Covidex7\\nand Google Covid-19 Research Explorer8 (COVID-\\n19 RE), which are also COVID-19 public search\\nengines that facilitate DR and answer highlighting\\nin passage.\\nThe TREC-COVID challenge track contains 40\\ntopic sets along with their document relevance\\njudgement. The topic sets are written by its or-\\nganizers with biomedical training, and motivated\\nby search submitted to the National Library of\\nMedicine and social media. Each topic consists\\nof three ﬁelds with different levels of granularity, a\\nkeyword-based query (KQ), a more precise natural\\nlanguage question (NQ), and a longer descriptive\\nnarrative. The DR assessments are performed fol-\\nlowing the TREC pooling mechanism. The par-\\nticipants submitted ranked lists of documents for\\neach topic set, based on which a depth of 10 to 20\\ndocuments are pooled and combined as a collection\\nof (q, D) pairs. The pairs are then assessed by an-\\nnotators with in-domain expertise. There are three\\nrounds of judgement available that corresponding\\nto three different versions of CORD-19 corpus. To\\nensure sufﬁcient coverage of annotation, we aggre-\\ngate all rounds results into 33,064 (q, D) relevance\\njudgements. Since the document id may change\\nacross versions, we map ids from each round to the\\nMay 19 release of CORD-19 corpus.\\nTo collect DR results from the three systems, we\\ncrawled the top 50 articles by querying the engines\\nwith KQ and its NQ variation from the topic sets on\\nJune 15, 2020. The crawled data, which are char-\\nacterized by the article title and link, are mapped\\nto May 19 CORD-19 corpus. Note that articles\\nthat cannot be found in the corpus are removed to\\nensure fair comparison.\\nWe use the standard DR metrics in our evalua-\\ntion, namely, the precision and recall at k (P@k,\\nR@k) and normalized discounted cumulative gain\\nin the top k documents (NDCG@k). Note that we\\nevaluate with k up to 20 since TREC-COVID has\\na pooling depth with 20 at most. Table 1 presents\\nthe DR performance of the engines over KQ and\\nNQ, respectively. ACS performs consistently better\\nthan the other engines on NQ and mostly on KQ,\\nand all engines perform better on NQ comparing\\nwith KQ.\\nIn addition, we are able to evaluate how robust\\neach system is against query variation from KQ\\nto NQ. Ideally, the results shall remain unchanged\\nwith query variation that requests the same infor-\\nmation. We deﬁne exact match (EM) and F1 score\\namong top k results to evaluate the robustness. Let\\nQ be the topic sets, and N Qk\\nq denote the\\ntop k searching results of natural language ques-\\ntion and keyword query corresponding to the same\\ntopic q ∈ Q, respectively. As shown in Eq.(1), we\\ntake the top k results KQk\\nq as ground truth, and\\ncompute the average exact match and F1 score of\\nthe top k results N Qk\\nq , and then average over all\\nqueries. The article title string is used as com-\\nparison key, and EM and F1 are standard that the\\nmaximum is taken over all the ground truth arti-\\ncles. Table 2 demonstrates that ACS has the best\\ncapability to provide consistent results with query\\nvariation.\\nNext, we evaluate the performance of PR and QA\\nwith CovidQA dataset. CovidQA contains 27 ques-\\ntions and their answers from 124 question-article\\npairs, which are selected by in-domain volunteers\\nas the most promising COVID-19 literature. Since\\nthe annotation does not possess sufﬁcient answer\\ncoverage over the entire corpus, we leverage our\\ninternal annotation resources to make PR and QA\\njudgement. More explicitly, we crawled the top 3\\nresults characterized by the article, displayed pas-\\nsage and the highlighted text snippet on June 15,\\n2020 from ACS, Covidex and COVID-19 RE, re-\\nspectively. After that, the annotators assess PR\\nand QA in terms of whether the displayed passage\\ncontains relevant information and whether the high-\\nlighted text snippet answers the given question. To\\navoid bias, the crawled results are combined and\\nshufﬂed randomly, therefore, the annotators do not\\nhave access to the source engine and the rank posi-\\ntion information during the judgement.\\nTable 3 presents precision of top PR and QA\\nresults. We use P@k instead of EM and F1 to eval-\\nuate QA since ACS highlights the answer while\\nCovidex and COVID-19 RE highlight the sentence\\nthat contains the answer. Instead of extracting the\\nanswer, it is more reasonable to judge whether the\\nhighlighted text answers the question and compute\\nprecision. ACS achieves better accuracy on both\\nPR and QA on most metrics. Note that ACS high-\\nlights answer snippet for at most three passages\\n, while Covidex and COVID-19 RE highlight all\\ndisplayed passages. This explains why ACS P@3\\nunderperforms COVID-19 RE. Explicitly with an\\nexample, Figure 4 displays the topmost results of\\nquerying “What is the incubation period of virus”\\nfrom the three systems. ACS highlights the answer\\nin the passage. In contrast, Covidex presents an\\narticle published at 2004 with information of virus\\nwhich is irrelevant with coronavirus, and COVID-\\n19 RE does not answer the question at all.\\nAs evident from the above results, ACS is one\\nof the top-performing systems that provides high\\nquality informative results over CORD-19 search.\\nIn this section we look into a number of sample\\nqueries to shed light on how different components\\nof ACS help in improving search results. We begin\\nby observing how small semantic differences in the\\nquery alter the results. The ﬁrst sample in Figure 5\\nis speciﬁc to medications. While the top result does\\nnot include the term medication the system high-\\nlights ribavirin and corticosteroids. The CORD-19\\nsystem understands that these terms represent med-\\nications with the help of CM NERe engine. In the\\nsecond example we change medications to mea-\\nsures and observe the top result discussing border\\ncontrol, and quarantine. This clearly demonstrates\\nthat Amazon Kendra has a deep comprehension of\\ntoken and query meanings.\\nFinally, we take a look at the effects of topic mod-\\neling when grouping and ﬁltering results. The last\\ntwo examples in Figure 5 showcase the difference\\nthis makes in the top result. Without specifying any\\ntopic the resulting article discusses high level pol-\\nicy, speciﬁcally quarantine measures in Singapore.\\nWhen we ﬁlter by clinical treatment the top result\\ninstead focuses on infections which is covered in\\nthe clinical setting. Furthermore the extracted text\\nreturned to the user still focuses on lessons learned\\nstaying true to the query.\\nAWS CORD-19 is an initial step towards help-\\ning medical researchers ﬁnd relevant content in\\na timely and meaningful way. In order to improve\\nthe robustness, we see following areas as direction\\nfor future research.\\nFeedback Loop - Since ACS is a search engine\\nthe motivation would be to evaluate it as such; us-\\ning well-established methodologies based on test\\ncollections—comprising topics (information needs)\\nand human annotations. Since no designated evalu-\\nation data exist, our initial focus is to capture dif-\\nferent interactions and feedback. Currently, ACS\\nlacks the feedback loop and federated learning ap-\\nproaches where the system would continuously\\nlearn and improve the search. However, the system\\ncaptures feedback from the researchers in the form\\nof implicit and explicit reactions. Implicit feed-\\nback evaluation consists of topics of interests, their\\nclicks as well as the ranking of the results which\\nwere selected by medical researchers. Explicit feed-\\nback evaluation is captured by providing up-down\\nrating associated with each search results. In the\\nfuture results can be personalized based on this\\nfeedback. Now that we have a system in place, our\\nefforts have shifted to broader engagement with po-\\ntential stakeholders to solicit additional guidance,\\nwhile trying to balance between the features and\\nranking.\\nQ&A Curation - Curation and normalization of\\nquestions have potential a use-case of presenting\\ntrending questions asked by the medical research\\ncommunity at a particular point. However, curation\\nwould involve capturing the questions asked as well\\nas identifying similar questions that can be later\\nnormalized. Currently, there is no mechanism to\\ncurate the questions asked by the researchers.\\nSummarization - Currently, ACS outputs the\\nrelevant passage based on the query. It would be\\nbeneﬁcial to get the overall summary of the pa-\\nper. A potential future direction would be to gen-\\nerates summaries (Raffel et al., 2019) from paper\\nabstracts and full body.\\nWe acknowledge the broader collaboration with\\nAI2 team and White house as well as the broader\\nAWS CORD-19 team including Kendra Science\\nteam, Tyler Stepke, Kingston Bosco, Victor Wang,\\nVaibhav Chaddha, Miguel Calvo, Ninad Kulkani,\\nKevin Longofer, Ray Chang, Adrian Bordone,\\nTony Nguyen, and Kyle Johnson.\\nThis paper describes our efforts in building AWS\\nCORD-19 Search with its capabilities consisting\\nof topic based, knowledge graph, and natural lan-\\nguage search queries. This is further enhanced\\nwith reading comprehension and FAQ matching\\nas well as document ranking providing a scalable\\nsolution to COVID-19 researchers and policy mak-\\ners in their search and discovery for answers to\\nhigh priority scientiﬁc questions. Our solution is\\npowered by Amazon Kendra, Comprehend Medi-\\ncal and Neptune which incorporate the latest neural\\narchitectures to provide information access capa-\\nbilities to the CORD-19 challenge. By comparing\\nwith other public search engines that support sim-\\nilar functionalities over COVID-19 search, ACS\\nis demonstrated to be powerful on its document\\nranking and question answering components. We\\nhope that our solution can prove useful in the ﬁght\\nagainst this global pandemic, and that the capabili-\\nties we have developed can be applied to analyzing\\nthe scientiﬁc literature more broadly.\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "hCykFpuh5wXk"
      },
      "outputs": [],
      "source": [
        "with open('2023-02-01_2007.09186v3_joined_text_indices_keep.pickle', 'wb') as f:\n",
        "  pickle.dump(joined_text_indices_keep, f)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 6. Clean extracted text\n",
        "\n",
        "Functions specific to this PDF to remove unwanted text, punctuation etc using replace() method and regular expressions."
      ],
      "metadata": {
        "id": "HCjFaoq-0UXk"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "```python\n",
        "def remove_subheadings(text:str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"2.1 Amazon Kendra\\\\n\", \"\")\n",
        "        .replace(\"2.4.1 Generating Topics\\\\n\", \"\")\n",
        "        .replace(\"2.4.2 Multi-Label Classiﬁcation\\\\n\", \"\")\n",
        "    )\n",
        "\n",
        "def remove_footnotes(text:str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"language 1\", \"language\")\n",
        "        .replace(\"Kendra2\", \"Kendra\")\n",
        "        .replace(\"Classification 3\", \"Classification\")\n",
        "        .replace(\"Medical 4\", \"Medical\")\n",
        "        .replace(\"models 5\", \"models\")\n",
        "        .replace(\"LDA 6\", \"LDA\")\n",
        "        .replace(\"Covidex7\", \"Covidex\")\n",
        "        .replace(\"Explorer8\", \"Explorer\")\n",
        "    )\n",
        "\n",
        "def replace_hyphen(text:str):\n",
        "    return (\n",
        "        text.replace(\"COVID-\\\\n19\", \"COVID-19\")\n",
        "    )\n",
        "\n",
        "def join_textbox_text(text: str):\n",
        "    return (\n",
        "        text.replace(\"\\\\n', '\", \" \")\n",
        "    )\n",
        "\n",
        "def remove_soft_hyphens(text: str):\n",
        "    return (\n",
        "        text.replace(\"-\\\\n\", \"\")\n",
        "    )\n",
        "\n",
        "def remove_new_line_chars(text: str):\n",
        "    return (\n",
        "        text.replace(\"\\\\n\", \" \")\n",
        "    )\n",
        "\n",
        "def remove_ligatures(text: str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"ﬀ\", \"ff\")\n",
        "        .replace(\"ﬁ\", \"fi\")\n",
        "        .replace(\"ﬂ\", \"fl\")\n",
        "        .replace(\"ﬃ\", \"ffi\")\n",
        "        .replace(\"ﬄ\", \"ffl\")\n",
        "        .replace(\"ﬅ\", \"ft\")\n",
        "    )\n",
        "\n",
        "def remove_greek_letters(text: str):\n",
        "    greek_codes   = chain(range(0x370, 0x3e2), range(0x3f0, 0x400))\n",
        "    greek_symbols = (chr(c) for c in greek_codes)\n",
        "    greek_letters = [c for c in greek_symbols if c.isalpha()]\n",
        "\n",
        "    for letter in greek_letters:\n",
        "      text = text.replace(letter, ' ')\n",
        "    return text\n",
        "\n",
        "def remove_accents(text: str):\n",
        "    return unicodedata.normalize('NFD', text)\n",
        "\n",
        "def remove_bullets(text: str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"•\", \" \")\n",
        "    )\n",
        "\n",
        "def remove_emails(text: str):\n",
        "    return (\n",
        "      re.sub(r\"([-!#-'*+/-9=?A-Z^-~]+(\\.[-!#-'*+/-9=?A-Z^-~]+)*|\\\"([]!#-[^-~ \\t]|(\\\\[\\t -~]))+\\\")@([-!#-'*+/-9=?A-Z^-~]+(\\.[-!#-'*+/-9=?A-Z^-~]+)*|\\[[\\t -Z^-~]])+\\s?\", 'email', text)\n",
        "    )\n",
        "\n",
        "def remove_urls(text: str):\n",
        "    return (\n",
        "      re.sub(r'((http|https)\\:\\/\\/)?[a-zA-Z0-9\\.\\/\\?\\:@\\-_=#]+\\.([a-zA-Z]){2,6}([a-zA-Z0-9\\.\\&\\/\\?\\:@\\-_=#])*(?=\\.)*(?=\\.)', 'website', text)\n",
        "    )\n",
        "\n",
        "def remove_references(text: str):\n",
        "    return (\n",
        "    re.sub(r'\\s\\([A-Z]\\w+[^\\)]+?, (19|20)[0-9]{2}\\)', '', text)\n",
        "    )\n",
        "\n",
        "def remove_brackets_and_quotes(text:str):\n",
        "    return (\n",
        "          text\n",
        "          .replace(\"[['\", \"\")\n",
        "          .replace(\"']]\", \"\")\n",
        "          .replace(\"[\", \"\")\n",
        "          .replace(\"]\", \"\")\n",
        "    )\n",
        "\n",
        "def remove_ellipsis(text: str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"...\", \"\")\n",
        "    )\n",
        "\n",
        "def remove_special_chars(text: str):\n",
        "    return (\n",
        "    re.sub(r'[“”!*)@#%(&$_?^=:,]', '', text)\n",
        "    )\n",
        "```"
      ],
      "metadata": {
        "id": "1xvhMbr1SYKS"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Function to call the functions above on the extracted text."
      ],
      "metadata": {
        "id": "55-05Nr1wRri"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def clean_text(text: str):\n",
        "\n",
        "    text = remove_subheadings(text)\n",
        "    text = remove_footnotes(text)\n",
        "    text = replace_hyphen(text)\n",
        "    text = join_textbox_text(text)\n",
        "    text = remove_soft_hyphens(text)\n",
        "    text = remove_new_line_chars(text)\n",
        "    text = remove_ligatures(text)\n",
        "    text = remove_greek_letters(text)\n",
        "    text = remove_accents(text)\n",
        "    text = remove_bullets(text)\n",
        "    text = remove_emails(text)\n",
        "    text = remove_urls(text)\n",
        "    text = remove_references(text)\n",
        "    text = remove_brackets_and_quotes(text)\n",
        "    text = remove_ellipsis(text)\n",
        "    text = remove_special_chars(text)\n",
        "\n",
        "    return text"
      ],
      "metadata": {
        "id": "qoCoOLZl9_zX"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "joined_text_indices_cleaned = clean_text(joined_text_indices_keep)\n",
        "joined_text_indices_cleaned"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "fKcufCSk6WvZ",
        "outputId": "a0b37cff-18e3-45ee-f587-88979bf15242"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'With the global outbreak of Coronavirus disease COVID-19 the world is in turmoil. Medical researchers are required to work quickly to fully understand and to provide a form of intervention for the virus. Due to a large research focus on the disease knowledge is published at a rapid rate throughout the world. One such repository of information is curated through the COVID-19 Open Research Dataset Challenge CORD-19. CORD-19 is a joint challenge put forth by Allen Institute AI2 National Institutes of Health NIH and the United States federal government via the White House. The objective of the challenge is to make sense of and extract useful knowledge across thousands of scholarly articles related to COVID-19. CORD-19 aims to connect the machine learning community with biomedical domain experts and policy makers in a race to identify effective treatments and management policies for COVID-19. In accordance with this initiative our goal is to provide a scalable solution aimed at aiding COVID-19 researchers and policy makers in their search and discovery for answers to high priority scientific questions. These questions should be understood in their natural language form; examples include What do we know about COVID-19 risk factors Which medications were most beneficial in the 2002 SARS outbreak and Which is the most referenced paper doing study for Hydroxychloroquine To appropriately answer these questions we require a system with a strong biomedical understanding of the natural queries and structured knowledge. AWS CORD-19 Search ACS provides an easy to use search interface where researchers query using natural language. ACS goes beyond keyword matching by understanding question semantics to efficiently find relevant answers. As illustrated in Figure 1 we provide the system with a natural language query inquiring about IL-6 inhibitors and showcase the system response with relevant query components highlighted. This query can confuse a traditional search engine that relies solely on text matching as it can only capture term overlap between a query and a document and may not necessarily be relevant to the researcher’s true intent i.e. learning implicit relations. Our system however establishes the relationship between IL-6 and severity of SARS-CoV-2 showing evidence where elevated IL-6 occur in a large number of patients with severe COVID-19 and higher mortality rates. In the following example if tasked with understanding the epidemiology and transmission of COVID-19 a researcher may ask about salivary viral load. We provide a further advantage from an organizational perspective using topics with domain specific relations. The user can select clinicaltreatments to refine search articles to understand that the highest load is assumed to be the day before symptoms appear. Similarly if commissioned to understand convalescent plasma therapy and selecting the three topics as shown in the last example the system identifies the most relevant article and answers the question in a highlight. There are a few search systems have been developed to support COVID-19 search leveraging CORD-19 corpus. We evaluate the ACS against two other search engines and ACS is proved to be a topperforming system. In the following sections we will discuss the various individual Amazon Web Services AWS products that allow ACS to provide this functionality highlighting the value to scientists who can quickly query validate their research and advance their investigations. AWS CORD-19 Search ACS relies on a deep semantic search model to return a ranked list of relevant documents. This system performs document ranking passage ranking question answering and FAQ matching. Additionally it leverages knowledge graphs and topic modeling to enrich the biomedical data and boost searching performance. This provides a scalable solution to COVID-19 researchers and policy makers. In this section we present the overall architecture of the system and a closer look at several individual components. Figure 2 provides an overview of this architecture. Amazon Kendra is a semantic search and question answering service provided by AWS for enterprise customers. Kendra allows its customers to power natural language based searching across their own data. As response to the worldwide COVID-19 pandemic Kendra has also been tooled to support COVID-19 related searching and question answering using the document corpus from CORD-19. The end-to-end Kendra system consists of several components. To complement the extracted answers Kendra uses a deep learning based semantic search model to return a ranked list of relevant documents. Amazon Kendra’s ability to understand natural language questions is at the core of its search engine returning the most relevant passage and related documents   Document ranking DR - Like any traditional search engine Kendra returns a ranked list of relevant documents based on the user’s query to fulfill their information needs. A deep semantic search model is used to understand natural language questions in addition to the keyword search.   Passage ranking PR  Question Answering QA - Kendra ranks the passages and tries to extract the answer from the top relevant passages with a deep reading comprehension model.   FAQ Matching FAQM - If there exists Frequently Answered Questions and their corresponding answers Kendra will automatically match a new coming query with FAQs and extract the answer if a strong match is found. In order to improve Kendra CORD-19 search and make it clinically more relevant for medical researchers we leverage knowledge extracted using the Amazon Comprehend Medical CM core NERe service as well as topics created from Amazon Comprehend Custom Classification 3. More explicitly a knowledge graph is built using the medical entities extracted with CM NERe API and the topics of the article are produced using a semisupervised prior based LDA approach. Both are used to enrich the data when indexing Kendra. Amazon Comprehend Medical CM is a HIPAA eligible AWS service for medical domain entity recognition relationship extraction and normalization. Comprehend Medical supports entity types divided into five different categories Anatomy Medical Condition Medication Protected Health Information and Test Treatment  Procedure and four traits Negation Diagnosis Sign and Symptom. These entities are directly used to enrich the Kendra search. Knowledge graphs KGs are structural representations of relations between real-world entities in the form of triplets containing a head entity a tail entity and the relation type connecting them. KG based information retrieval has shown great success in the past decades. The COVID-19 Knowledge Graph CKG Fig 3 is a directed property graph constructed from the CORD-19 Open Research Dataset of scholarly articles. Entities including scholarly articles authors author institutions citations extracted topics and comprehend medical entities are used to form relations in the CKG. The resulting KG continues to grow as the CORD-19 dataset increases and currently contains over 335k entities and 3.3M relations. The CKG powers a number of features on ACS including article recommendations citation-based navigation and search result ranking by author or institution publication count. Scientific article recommendations are made possible by a document similarity engine that quantifies similarity between documents by combining semantic embeddings obtained from a pre-trained language model with document knowledge graph embeddings capturing topological information from the CKG. Topic modeling is a statistical discovery paradigm for generating topics that occur in a collection of documents. Perhaps the most widely used model for topic modeling is Latent Dirichlet Allocation LDA a generative model which groups documents together by observed content often giving each document a mixture of topics it belongs to. An extension of this work termed Zlabel LDA utilizes priors to allow the model to force certain topics which the users have manually curated or wish to see clustered together. For the purposes of this work we experimented with 5 10 and 20 topic models. The outputs of each clustering size were manually inspected and topic labels were provided by us when inspecting the top ten terms for each cluster. The final granularity of the topic models was chosen by manually deleting and merging topics from the 20 topic model. In general we were able to clearly extract groups which centered around important topics including virology proteomics epidemiology and cellular biology to name a few. However when faced with 20 topics the less populated ones tended to be noisy and captured peripheral information present in the input such as language e.g. Spanish and French or provide redundancies with existing topics e.g. two topics for Influenza. As a control we ran a publicly available implementation of Z-label LDA with no priors which yields topics close to those extracted using Comprehend Topic Modeling. Although similar we observed better definition in certain groups such as pulmonary diseases and policy/industry and decided to use this as the curation entry-point. Our goal was to limit these topics to ten and compile them in advance as much as possible. With the help of medical professionals we eliminated and combined topics to form the following Vaccines/immunology Genomics Public health Policies Epidemiology Clinical Treatment Virology Influenza Healthcare Industry Pulmonary Infections and Lab Trials human. Having to manually feed a topic model and re-train on the entire corpus once new data becomes available is largely inefficient. We therefore used the topic model labels to train a multi-label classifier. To evaluate the performance of this model we calculate the average F1 across test samples by calculating the set overlap between our gold standard topic model and system labels multi-label classification. This held-out test set contains 20 of the CORD-19 data available at the time. Using this metric our trained model achieved an average F1 of 91.92 with on average 2.37 labels per document. Fewer than 1 of the documents in the test set received no label using 0.5 as the confidence threshold. ACS supports document ranking DR passage ranking PR and question answering QA on the top suggested results. In this section we evaluate the overall performance of ACS with publicly available datasets. The TREC-COVID track built DR test collections that contain document relevance ground truth and are used to assess DR only. For PR and QA we manually annotate questions collected from CovidQA the first open sourced Covid-19 QA dataset. The ACS results are compared against Covidex and Google Covid-19 Research Explorer COVID-19 RE which are also COVID-19 public search engines that facilitate DR and answer highlighting in passage. The TREC-COVID challenge track contains 40 topic sets along with their document relevance judgement. The topic sets are written by its organizers with biomedical training and motivated by search submitted to the National Library of Medicine and social media. Each topic consists of three fields with different levels of granularity a keyword-based query KQ a more precise natural language question NQ and a longer descriptive narrative. The DR assessments are performed following the TREC pooling mechanism. The participants submitted ranked lists of documents for each topic set based on which a depth of 10 to 20 documents are pooled and combined as a collection of q D pairs. The pairs are then assessed by annotators with in-domain expertise. There are three rounds of judgement available that corresponding to three different versions of CORD-19 corpus. To ensure sufficient coverage of annotation we aggregate all rounds results into 33064 q D relevance judgements. Since the document id may change across versions we map ids from each round to the May 19 release of CORD-19 corpus. To collect DR results from the three systems we crawled the top 50 articles by querying the engines with KQ and its NQ variation from the topic sets on June 15 2020. The crawled data which are characterized by the article title and link are mapped to May 19 CORD-19 corpus. Note that articles that cannot be found in the corpus are removed to ensure fair comparison. We use the standard DR metrics in our evaluation namely the precision and recall at k email email and normalized discounted cumulative gain in the top k documents email. Note that we evaluate with k up to 20 since TREC-COVID has a pooling depth with 20 at most. Table 1 presents the DR performance of the engines over KQ and NQ respectively. ACS performs consistently better than the other engines on NQ and mostly on KQ and all engines perform better on NQ comparing with KQ. In addition we are able to evaluate how robust each system is against query variation from KQ to NQ. Ideally the results shall remain unchanged with query variation that requests the same information. We define exact match EM and F1 score among top k results to evaluate the robustness. Let Q be the topic sets and N Qk q denote the top k searching results of natural language question and keyword query corresponding to the same topic q ∈ Q respectively. As shown in Eq.1 we take the top k results KQk q as ground truth and compute the average exact match and F1 score of the top k results N Qk q  and then average over all queries. The article title string is used as comparison key and EM and F1 are standard that the maximum is taken over all the ground truth articles. Table 2 demonstrates that ACS has the best capability to provide consistent results with query variation. Next we evaluate the performance of PR and QA with CovidQA dataset. CovidQA contains 27 questions and their answers from 124 question-article pairs which are selected by in-domain volunteers as the most promising COVID-19 literature. Since the annotation does not possess sufficient answer coverage over the entire corpus we leverage our internal annotation resources to make PR and QA judgement. More explicitly we crawled the top 3 results characterized by the article displayed passage and the highlighted text snippet on June 15 2020 from ACS Covidex and COVID-19 RE respectively. After that the annotators assess PR and QA in terms of whether the displayed passage contains relevant information and whether the highlighted text snippet answers the given question. To avoid bias the crawled results are combined and shuffled randomly therefore the annotators do not have access to the source engine and the rank position information during the judgement. Table 3 presents precision of top PR and QA results. We use emailinstead of EM and F1 to evaluate QA since ACS highlights the answer while Covidex and COVID-19 RE highlight the sentence that contains the answer. Instead of extracting the answer it is more reasonable to judge whether the highlighted text answers the question and compute precision. ACS achieves better accuracy on both PR and QA on most metrics. Note that ACS highlights answer snippet for at most three passages  while Covidex and COVID-19 RE highlight all displayed passages. This explains why ACS emailunderperforms COVID-19 RE. Explicitly with an example Figure 4 displays the topmost results of querying What is the incubation period of virus from the three systems. ACS highlights the answer in the passage. In contrast Covidex presents an article published at 2004 with information of virus which is irrelevant with coronavirus and COVID-19 RE does not answer the question at all. As evident from the above results ACS is one of the top-performing systems that provides high quality informative results over CORD-19 search. In this section we look into a number of sample queries to shed light on how different components of ACS help in improving search results. We begin by observing how small semantic differences in the query alter the results. The first sample in Figure 5 is specific to medications. While the top result does not include the term medication the system highlights ribavirin and corticosteroids. The CORD-19 system understands that these terms represent medications with the help of CM NERe engine. In the second example we change medications to measures and observe the top result discussing border control and quarantine. This clearly demonstrates that Amazon Kendra has a deep comprehension of token and query meanings. Finally we take a look at the effects of topic modeling when grouping and filtering results. The last two examples in Figure 5 showcase the difference this makes in the top result. Without specifying any topic the resulting article discusses high level policy specifically quarantine measures in Singapore. When we filter by clinical treatment the top result instead focuses on infections which is covered in the clinical setting. Furthermore the extracted text returned to the user still focuses on lessons learned staying true to the query. AWS CORD-19 is an initial step towards helping medical researchers find relevant content in a timely and meaningful way. In order to improve the robustness we see following areas as direction for future research. Feedback Loop - Since ACS is a search engine the motivation would be to evaluate it as such; using well-established methodologies based on test collections—comprising topics information needs and human annotations. Since no designated evaluation data exist our initial focus is to capture different interactions and feedback. Currently ACS lacks the feedback loop and federated learning approaches where the system would continuously learn and improve the search. However the system captures feedback from the researchers in the form of implicit and explicit reactions. Implicit feedback evaluation consists of topics of interests their clicks as well as the ranking of the results which were selected by medical researchers. Explicit feedback evaluation is captured by providing up-down rating associated with each search results. In the future results can be personalized based on this feedback. Now that we have a system in place our efforts have shifted to broader engagement with potential stakeholders to solicit additional guidance while trying to balance between the features and ranking. QA Curation - Curation and normalization of questions have potential a use-case of presenting trending questions asked by the medical research community at a particular point. However curation would involve capturing the questions asked as well as identifying similar questions that can be later normalized. Currently there is no mechanism to curate the questions asked by the researchers. Summarization - Currently ACS outputs the relevant passage based on the query. It would be beneficial to get the overall summary of the paper. A potential future direction would be to generates summaries from paper abstracts and full body. We acknowledge the broader collaboration with AI2 team and White house as well as the broader AWS CORD-19 team including Kendra Science team Tyler Stepke Kingston Bosco Victor Wang Vaibhav Chaddha Miguel Calvo Ninad Kulkani Kevin Longofer Ray Chang Adrian Bordone Tony Nguyen and Kyle Johnson. This paper describes our efforts in building AWS CORD-19 Search with its capabilities consisting of topic based knowledge graph and natural language search queries. This is further enhanced with reading comprehension and FAQ matching as well as document ranking providing a scalable solution to COVID-19 researchers and policy makers in their search and discovery for answers to high priority scientific questions. Our solution is powered by Amazon Kendra Comprehend Medical and Neptune which incorporate the latest neural architectures to provide information access capabilities to the CORD-19 challenge. By comparing with other public search engines that support similar functionalities over COVID-19 search ACS is demonstrated to be powerful on its document ranking and question answering components. We hope that our solution can prove useful in the fight against this global pandemic and that the capabilities we have developed can be applied to analyzing the scientific literature more broadly. '"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 67
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "MavK3a9qDFBX"
      },
      "outputs": [],
      "source": [
        "with open('2023-02-18_2007.09186v3_joined_text_indices_join_text_cleaned.pickle', 'wb') as f:\n",
        "  pickle.dump(joined_text_indices_cleaned, f)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 6.1 Add cleaned arXiv PDF text to combined PMC and arXiv DataFrame"
      ],
      "metadata": {
        "id": "vnDJU2-JHuWZ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-15_pmc_arxiv_full_text_merged_new.pickle', 'rb') as f:\n",
        "    pmc_arxiv_full_text_merged_new = pickle.load(f)"
      ],
      "metadata": {
        "id": "Ha6927EHHyrW"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_new = pd.read_pickle('2023-01-15_pmc_arxiv_full_text_merged_new.pickle')\n",
        "pmc_arxiv_full_text_merged_new"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "1JqQXM3PH5hu",
        "outputId": "06332148-1e6b-4cb2-e401-44b0bfb5da13"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         article_id   published     revised  \\\n",
              "0        PMC9549161  2022-09-26  2022-10-14   \n",
              "1        PMC9539342  2022-09-22  2022-11-12   \n",
              "2        PMC9357751  2022-12-01  2022-12-05   \n",
              "3        PMC9346052  2022-08-03  2022-09-05   \n",
              "4        PMC9775208  2022-12-15  2022-12-25   \n",
              "...             ...         ...         ...   \n",
              "11688  2003.13665v1  2020-03-30  2020-03-30   \n",
              "11689  2003.14258v1  2020-03-30  2020-03-30   \n",
              "11690  2003.12454v1  2020-03-26  2020-03-26   \n",
              "11691  2003.04524v1  2020-03-10  2020-03-10   \n",
              "11692  2003.00751v1  2020-03-02  2020-03-02   \n",
              "\n",
              "                                                   title  \\\n",
              "0           Drug repositioning: A bibliometric analysis.   \n",
              "1      A review on computer-aided chemogenomics and d...   \n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "3      Scope of repurposed drugs against the potentia...   \n",
              "4      Drug Repurposing Using Gene Co-Expression and ...   \n",
              "...                                                  ...   \n",
              "11688  Genomics-guided molecular maps of coronavirus ...   \n",
              "11689  Nanomechanical sonification of the 2019-nCoV c...   \n",
              "11690  A Machine Learning alternative to placebo-cont...   \n",
              "11691  Old Drugs for Newly Emerging Viral Disease, CO...   \n",
              "11692  Regulation of interferon production as a poten...   \n",
              "\n",
              "                                                 journal  \\\n",
              "0                              Frontiers in pharmacology   \n",
              "1                         Chemical biology & drug design   \n",
              "2      Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                                   Structural chemistry   \n",
              "4                                                Biology   \n",
              "...                                                  ...   \n",
              "11688                                               None   \n",
              "11689                                               None   \n",
              "11690                                               None   \n",
              "11691                                               None   \n",
              "11692                                               None   \n",
              "\n",
              "                                                 authors  \\\n",
              "0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                      0   \n",
              "3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                    Mailem RC, Tayo LL.   \n",
              "...                                                  ...   \n",
              "11688                                    Gennadi Glinsky   \n",
              "11689                                  Markus J. Buehler   \n",
              "11690   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc   \n",
              "11691                                Mohammad Reza Dayer   \n",
              "11692             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei   \n",
              "\n",
              "                              doi  \\\n",
              "0       10.3389/fphar.2022.974849   \n",
              "1              10.1111/cbdd.14136   \n",
              "2                               0   \n",
              "3      10.1007/s11224-022-02020-z   \n",
              "4         10.3390/biology11121827   \n",
              "...                           ...   \n",
              "11688                        None   \n",
              "11689                        None   \n",
              "11690                        None   \n",
              "11691                        None   \n",
              "11692                        None   \n",
              "\n",
              "                                                 pdf_url  \\\n",
              "0      https://europepmc.org/articles/PMC9549161?pdf=...   \n",
              "1      https://europepmc.org/articles/PMC9539342?pdf=...   \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...   \n",
              "3      https://europepmc.org/articles/PMC9346052?pdf=...   \n",
              "4      https://europepmc.org/articles/PMC9775208?pdf=...   \n",
              "...                                                  ...   \n",
              "11688           http://export.arxiv.org/pdf/2003.13665v1   \n",
              "11689           http://export.arxiv.org/pdf/2003.14258v1   \n",
              "11690           http://export.arxiv.org/pdf/2003.12454v1   \n",
              "11691           http://export.arxiv.org/pdf/2003.04524v1   \n",
              "11692           http://export.arxiv.org/pdf/2003.00751v1   \n",
              "\n",
              "                                                    text  \n",
              "0      Sir James Black, a winner of the 1988 Nobel Pr...  \n",
              "1      Tight and selective interaction between ligand...  \n",
              "2      The severe acute respiratory syndrome coronavi...  \n",
              "3      The sudden outbreak of SARS-CoV-2 in 2019 took...  \n",
              "4      The 2019 novel coronavirus, now dubbed SARS-Co...  \n",
              "...                                                  ...  \n",
              "11688  \\nCoronavirus pandemic 2020 caused by the newl...  \n",
              "11689  \\nProteins are the building blocks of virtuall...  \n",
              "11690  \\nCurrent and last decades research in drug di...  \n",
              "11691  \\nThe outbreak of coronavirus  in 2019-2020 is...  \n",
              "11692  \\nNovel Coronavirus SARS-CoV-2 is a single str...  \n",
              "\n",
              "[11693 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-a7610e3c-9456-48ff-82d7-6917fcb86382\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11688</th>\n",
              "      <td>2003.13665v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Genomics-guided molecular maps of coronavirus ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Gennadi Glinsky</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.13665v1</td>\n",
              "      <td>\\nCoronavirus pandemic 2020 caused by the newl...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11689</th>\n",
              "      <td>2003.14258v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Nanomechanical sonification of the 2019-nCoV c...</td>\n",
              "      <td>None</td>\n",
              "      <td>Markus J. Buehler</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.14258v1</td>\n",
              "      <td>\\nProteins are the building blocks of virtuall...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11690</th>\n",
              "      <td>2003.12454v1</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>A Machine Learning alternative to placebo-cont...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ezequiel Alvarez, Federico Lamagna, Manuel Szewc</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.12454v1</td>\n",
              "      <td>\\nCurrent and last decades research in drug di...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11691</th>\n",
              "      <td>2003.04524v1</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>Old Drugs for Newly Emerging Viral Disease, CO...</td>\n",
              "      <td>None</td>\n",
              "      <td>Mohammad Reza Dayer</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.04524v1</td>\n",
              "      <td>\\nThe outbreak of coronavirus  in 2019-2020 is...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11692</th>\n",
              "      <td>2003.00751v1</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>Regulation of interferon production as a poten...</td>\n",
              "      <td>None</td>\n",
              "      <td>Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.00751v1</td>\n",
              "      <td>\\nNovel Coronavirus SARS-CoV-2 is a single str...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11693 rows × 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a7610e3c-9456-48ff-82d7-6917fcb86382')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-a7610e3c-9456-48ff-82d7-6917fcb86382 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-a7610e3c-9456-48ff-82d7-6917fcb86382');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 72
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus = pmc_arxiv_full_text_merged_new.copy()"
      ],
      "metadata": {
        "id": "3Dbs4Cr6IqlI"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Row with missing text which we have now extracted."
      ],
      "metadata": {
        "id": "x2a29ndsxvqm"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus.loc[pmc_arxiv_full_text_merged_plus['text'].isna()]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 159
        },
        "id": "CMf-0ARrNpHk",
        "outputId": "71a53e1a-6b9d-4237-bf0f-3a082c237e39"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         article_id   published     revised  \\\n",
              "11566  2007.09186v3  2020-07-17  2020-10-07   \n",
              "\n",
              "                                                   title journal  \\\n",
              "11566  AWS CORD-19 Search: A Neural Search Engine for...    None   \n",
              "\n",
              "                                                 authors   doi  \\\n",
              "11566  Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...  None   \n",
              "\n",
              "                                        pdf_url  text  \n",
              "11566  http://export.arxiv.org/pdf/2007.09186v3  None  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5518aa4c-bc39-4a0b-b151-806e32a0a3e2\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>11566</th>\n",
              "      <td>2007.09186v3</td>\n",
              "      <td>2020-07-17</td>\n",
              "      <td>2020-10-07</td>\n",
              "      <td>AWS CORD-19 Search: A Neural Search Engine for...</td>\n",
              "      <td>None</td>\n",
              "      <td>Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2007.09186v3</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5518aa4c-bc39-4a0b-b151-806e32a0a3e2')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-5518aa4c-bc39-4a0b-b151-806e32a0a3e2 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-5518aa4c-bc39-4a0b-b151-806e32a0a3e2');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 75
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "joined_text_indices_cleaned"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 157
        },
        "id": "tTblcDW0Faot",
        "outputId": "e045cdb5-b89c-467e-c8c1-2d4ea9ef3d43"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'With the global outbreak of Coronavirus disease COVID-19 the world is in turmoil. Medical researchers are required to work quickly to fully understand and to provide a form of intervention for the virus. Due to a large research focus on the disease knowledge is published at a rapid rate throughout the world. One such repository of information is curated through the COVID-19 Open Research Dataset Challenge CORD-19. CORD-19 is a joint challenge put forth by Allen Institute AI2 National Institutes of Health NIH and the United States federal government via the White House. The objective of the challenge is to make sense of and extract useful knowledge across thousands of scholarly articles related to COVID-19. CORD-19 aims to connect the machine learning community with biomedical domain experts and policy makers in a race to identify effective treatments and management policies for COVID-19. In accordance with this initiative our goal is to provide a scalable solution aimed at aiding COVID-19 researchers and policy makers in their search and discovery for answers to high priority scientific questions. These questions should be understood in their natural language form; examples include What do we know about COVID-19 risk factors Which medications were most beneficial in the 2002 SARS outbreak and Which is the most referenced paper doing study for Hydroxychloroquine To appropriately answer these questions we require a system with a strong biomedical understanding of the natural queries and structured knowledge. AWS CORD-19 Search ACS provides an easy to use search interface where researchers query using natural language. ACS goes beyond keyword matching by understanding question semantics to efficiently find relevant answers. As illustrated in Figure 1 we provide the system with a natural language query inquiring about IL-6 inhibitors and showcase the system response with relevant query components highlighted. This query can confuse a traditional search engine that relies solely on text matching as it can only capture term overlap between a query and a document and may not necessarily be relevant to the researcher’s true intent i.e. learning implicit relations. Our system however establishes the relationship between IL-6 and severity of SARS-CoV-2 showing evidence where elevated IL-6 occur in a large number of patients with severe COVID-19 and higher mortality rates. In the following example if tasked with understanding the epidemiology and transmission of COVID-19 a researcher may ask about salivary viral load. We provide a further advantage from an organizational perspective using topics with domain specific relations. The user can select clinicaltreatments to refine search articles to understand that the highest load is assumed to be the day before symptoms appear. Similarly if commissioned to understand convalescent plasma therapy and selecting the three topics as shown in the last example the system identifies the most relevant article and answers the question in a highlight. There are a few search systems have been developed to support COVID-19 search leveraging CORD-19 corpus. We evaluate the ACS against two other search engines and ACS is proved to be a topperforming system. In the following sections we will discuss the various individual Amazon Web Services AWS products that allow ACS to provide this functionality highlighting the value to scientists who can quickly query validate their research and advance their investigations. AWS CORD-19 Search ACS relies on a deep semantic search model to return a ranked list of relevant documents. This system performs document ranking passage ranking question answering and FAQ matching. Additionally it leverages knowledge graphs and topic modeling to enrich the biomedical data and boost searching performance. This provides a scalable solution to COVID-19 researchers and policy makers. In this section we present the overall architecture of the system and a closer look at several individual components. Figure 2 provides an overview of this architecture. Amazon Kendra is a semantic search and question answering service provided by AWS for enterprise customers. Kendra allows its customers to power natural language based searching across their own data. As response to the worldwide COVID-19 pandemic Kendra has also been tooled to support COVID-19 related searching and question answering using the document corpus from CORD-19. The end-to-end Kendra system consists of several components. To complement the extracted answers Kendra uses a deep learning based semantic search model to return a ranked list of relevant documents. Amazon Kendra’s ability to understand natural language questions is at the core of its search engine returning the most relevant passage and related documents   Document ranking DR - Like any traditional search engine Kendra returns a ranked list of relevant documents based on the user’s query to fulfill their information needs. A deep semantic search model is used to understand natural language questions in addition to the keyword search.   Passage ranking PR  Question Answering QA - Kendra ranks the passages and tries to extract the answer from the top relevant passages with a deep reading comprehension model.   FAQ Matching FAQM - If there exists Frequently Answered Questions and their corresponding answers Kendra will automatically match a new coming query with FAQs and extract the answer if a strong match is found. In order to improve Kendra CORD-19 search and make it clinically more relevant for medical researchers we leverage knowledge extracted using the Amazon Comprehend Medical CM core NERe service as well as topics created from Amazon Comprehend Custom Classification 3. More explicitly a knowledge graph is built using the medical entities extracted with CM NERe API and the topics of the article are produced using a semisupervised prior based LDA approach. Both are used to enrich the data when indexing Kendra. Amazon Comprehend Medical CM is a HIPAA eligible AWS service for medical domain entity recognition relationship extraction and normalization. Comprehend Medical supports entity types divided into five different categories Anatomy Medical Condition Medication Protected Health Information and Test Treatment  Procedure and four traits Negation Diagnosis Sign and Symptom. These entities are directly used to enrich the Kendra search. Knowledge graphs KGs are structural representations of relations between real-world entities in the form of triplets containing a head entity a tail entity and the relation type connecting them. KG based information retrieval has shown great success in the past decades. The COVID-19 Knowledge Graph CKG Fig 3 is a directed property graph constructed from the CORD-19 Open Research Dataset of scholarly articles. Entities including scholarly articles authors author institutions citations extracted topics and comprehend medical entities are used to form relations in the CKG. The resulting KG continues to grow as the CORD-19 dataset increases and currently contains over 335k entities and 3.3M relations. The CKG powers a number of features on ACS including article recommendations citation-based navigation and search result ranking by author or institution publication count. Scientific article recommendations are made possible by a document similarity engine that quantifies similarity between documents by combining semantic embeddings obtained from a pre-trained language model with document knowledge graph embeddings capturing topological information from the CKG. Topic modeling is a statistical discovery paradigm for generating topics that occur in a collection of documents. Perhaps the most widely used model for topic modeling is Latent Dirichlet Allocation LDA a generative model which groups documents together by observed content often giving each document a mixture of topics it belongs to. An extension of this work termed Zlabel LDA utilizes priors to allow the model to force certain topics which the users have manually curated or wish to see clustered together. For the purposes of this work we experimented with 5 10 and 20 topic models. The outputs of each clustering size were manually inspected and topic labels were provided by us when inspecting the top ten terms for each cluster. The final granularity of the topic models was chosen by manually deleting and merging topics from the 20 topic model. In general we were able to clearly extract groups which centered around important topics including virology proteomics epidemiology and cellular biology to name a few. However when faced with 20 topics the less populated ones tended to be noisy and captured peripheral information present in the input such as language e.g. Spanish and French or provide redundancies with existing topics e.g. two topics for Influenza. As a control we ran a publicly available implementation of Z-label LDA with no priors which yields topics close to those extracted using Comprehend Topic Modeling. Although similar we observed better definition in certain groups such as pulmonary diseases and policy/industry and decided to use this as the curation entry-point. Our goal was to limit these topics to ten and compile them in advance as much as possible. With the help of medical professionals we eliminated and combined topics to form the following Vaccines/immunology Genomics Public health Policies Epidemiology Clinical Treatment Virology Influenza Healthcare Industry Pulmonary Infections and Lab Trials human. Having to manually feed a topic model and re-train on the entire corpus once new data becomes available is largely inefficient. We therefore used the topic model labels to train a multi-label classifier. To evaluate the performance of this model we calculate the average F1 across test samples by calculating the set overlap between our gold standard topic model and system labels multi-label classification. This held-out test set contains 20 of the CORD-19 data available at the time. Using this metric our trained model achieved an average F1 of 91.92 with on average 2.37 labels per document. Fewer than 1 of the documents in the test set received no label using 0.5 as the confidence threshold. ACS supports document ranking DR passage ranking PR and question answering QA on the top suggested results. In this section we evaluate the overall performance of ACS with publicly available datasets. The TREC-COVID track built DR test collections that contain document relevance ground truth and are used to assess DR only. For PR and QA we manually annotate questions collected from CovidQA the first open sourced Covid-19 QA dataset. The ACS results are compared against Covidex and Google Covid-19 Research Explorer COVID-19 RE which are also COVID-19 public search engines that facilitate DR and answer highlighting in passage. The TREC-COVID challenge track contains 40 topic sets along with their document relevance judgement. The topic sets are written by its organizers with biomedical training and motivated by search submitted to the National Library of Medicine and social media. Each topic consists of three fields with different levels of granularity a keyword-based query KQ a more precise natural language question NQ and a longer descriptive narrative. The DR assessments are performed following the TREC pooling mechanism. The participants submitted ranked lists of documents for each topic set based on which a depth of 10 to 20 documents are pooled and combined as a collection of q D pairs. The pairs are then assessed by annotators with in-domain expertise. There are three rounds of judgement available that corresponding to three different versions of CORD-19 corpus. To ensure sufficient coverage of annotation we aggregate all rounds results into 33064 q D relevance judgements. Since the document id may change across versions we map ids from each round to the May 19 release of CORD-19 corpus. To collect DR results from the three systems we crawled the top 50 articles by querying the engines with KQ and its NQ variation from the topic sets on June 15 2020. The crawled data which are characterized by the article title and link are mapped to May 19 CORD-19 corpus. Note that articles that cannot be found in the corpus are removed to ensure fair comparison. We use the standard DR metrics in our evaluation namely the precision and recall at k email email and normalized discounted cumulative gain in the top k documents email. Note that we evaluate with k up to 20 since TREC-COVID has a pooling depth with 20 at most. Table 1 presents the DR performance of the engines over KQ and NQ respectively. ACS performs consistently better than the other engines on NQ and mostly on KQ and all engines perform better on NQ comparing with KQ. In addition we are able to evaluate how robust each system is against query variation from KQ to NQ. Ideally the results shall remain unchanged with query variation that requests the same information. We define exact match EM and F1 score among top k results to evaluate the robustness. Let Q be the topic sets and N Qk q denote the top k searching results of natural language question and keyword query corresponding to the same topic q ∈ Q respectively. As shown in Eq.1 we take the top k results KQk q as ground truth and compute the average exact match and F1 score of the top k results N Qk q  and then average over all queries. The article title string is used as comparison key and EM and F1 are standard that the maximum is taken over all the ground truth articles. Table 2 demonstrates that ACS has the best capability to provide consistent results with query variation. Next we evaluate the performance of PR and QA with CovidQA dataset. CovidQA contains 27 questions and their answers from 124 question-article pairs which are selected by in-domain volunteers as the most promising COVID-19 literature. Since the annotation does not possess sufficient answer coverage over the entire corpus we leverage our internal annotation resources to make PR and QA judgement. More explicitly we crawled the top 3 results characterized by the article displayed passage and the highlighted text snippet on June 15 2020 from ACS Covidex and COVID-19 RE respectively. After that the annotators assess PR and QA in terms of whether the displayed passage contains relevant information and whether the highlighted text snippet answers the given question. To avoid bias the crawled results are combined and shuffled randomly therefore the annotators do not have access to the source engine and the rank position information during the judgement. Table 3 presents precision of top PR and QA results. We use emailinstead of EM and F1 to evaluate QA since ACS highlights the answer while Covidex and COVID-19 RE highlight the sentence that contains the answer. Instead of extracting the answer it is more reasonable to judge whether the highlighted text answers the question and compute precision. ACS achieves better accuracy on both PR and QA on most metrics. Note that ACS highlights answer snippet for at most three passages  while Covidex and COVID-19 RE highlight all displayed passages. This explains why ACS emailunderperforms COVID-19 RE. Explicitly with an example Figure 4 displays the topmost results of querying What is the incubation period of virus from the three systems. ACS highlights the answer in the passage. In contrast Covidex presents an article published at 2004 with information of virus which is irrelevant with coronavirus and COVID-19 RE does not answer the question at all. As evident from the above results ACS is one of the top-performing systems that provides high quality informative results over CORD-19 search. In this section we look into a number of sample queries to shed light on how different components of ACS help in improving search results. We begin by observing how small semantic differences in the query alter the results. The first sample in Figure 5 is specific to medications. While the top result does not include the term medication the system highlights ribavirin and corticosteroids. The CORD-19 system understands that these terms represent medications with the help of CM NERe engine. In the second example we change medications to measures and observe the top result discussing border control and quarantine. This clearly demonstrates that Amazon Kendra has a deep comprehension of token and query meanings. Finally we take a look at the effects of topic modeling when grouping and filtering results. The last two examples in Figure 5 showcase the difference this makes in the top result. Without specifying any topic the resulting article discusses high level policy specifically quarantine measures in Singapore. When we filter by clinical treatment the top result instead focuses on infections which is covered in the clinical setting. Furthermore the extracted text returned to the user still focuses on lessons learned staying true to the query. AWS CORD-19 is an initial step towards helping medical researchers find relevant content in a timely and meaningful way. In order to improve the robustness we see following areas as direction for future research. Feedback Loop - Since ACS is a search engine the motivation would be to evaluate it as such; using well-established methodologies based on test collections—comprising topics information needs and human annotations. Since no designated evaluation data exist our initial focus is to capture different interactions and feedback. Currently ACS lacks the feedback loop and federated learning approaches where the system would continuously learn and improve the search. However the system captures feedback from the researchers in the form of implicit and explicit reactions. Implicit feedback evaluation consists of topics of interests their clicks as well as the ranking of the results which were selected by medical researchers. Explicit feedback evaluation is captured by providing up-down rating associated with each search results. In the future results can be personalized based on this feedback. Now that we have a system in place our efforts have shifted to broader engagement with potential stakeholders to solicit additional guidance while trying to balance between the features and ranking. QA Curation - Curation and normalization of questions have potential a use-case of presenting trending questions asked by the medical research community at a particular point. However curation would involve capturing the questions asked as well as identifying similar questions that can be later normalized. Currently there is no mechanism to curate the questions asked by the researchers. Summarization - Currently ACS outputs the relevant passage based on the query. It would be beneficial to get the overall summary of the paper. A potential future direction would be to generates summaries from paper abstracts and full body. We acknowledge the broader collaboration with AI2 team and White house as well as the broader AWS CORD-19 team including Kendra Science team Tyler Stepke Kingston Bosco Victor Wang Vaibhav Chaddha Miguel Calvo Ninad Kulkani Kevin Longofer Ray Chang Adrian Bordone Tony Nguyen and Kyle Johnson. This paper describes our efforts in building AWS CORD-19 Search with its capabilities consisting of topic based knowledge graph and natural language search queries. This is further enhanced with reading comprehension and FAQ matching as well as document ranking providing a scalable solution to COVID-19 researchers and policy makers in their search and discovery for answers to high priority scientific questions. Our solution is powered by Amazon Kendra Comprehend Medical and Neptune which incorporate the latest neural architectures to provide information access capabilities to the CORD-19 challenge. By comparing with other public search engines that support similar functionalities over COVID-19 search ACS is demonstrated to be powerful on its document ranking and question answering components. We hope that our solution can prove useful in the fight against this global pandemic and that the capabilities we have developed can be applied to analyzing the scientific literature more broadly. '"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 77
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Add extracted text to text column in row for the arXiv PDF."
      ],
      "metadata": {
        "id": "Kyu5nAZZypa2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus.loc[11566:11567, ['text']] = joined_text_indices_cleaned"
      ],
      "metadata": {
        "id": "-A4cSvNpOdCr"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus.loc[pmc_arxiv_full_text_merged_plus['article_id']  == '2007.09186v3']"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 177
        },
        "id": "S4r1CC6wQXla",
        "outputId": "46bdf0f0-1291-4305-f17c-f232de7bae45"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         article_id   published     revised  \\\n",
              "11566  2007.09186v3  2020-07-17  2020-10-07   \n",
              "\n",
              "                                                   title journal  \\\n",
              "11566  AWS CORD-19 Search: A Neural Search Engine for...    None   \n",
              "\n",
              "                                                 authors   doi  \\\n",
              "11566  Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...  None   \n",
              "\n",
              "                                        pdf_url  \\\n",
              "11566  http://export.arxiv.org/pdf/2007.09186v3   \n",
              "\n",
              "                                                    text  \n",
              "11566  With the global outbreak of Coronavirus diseas...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-347b0773-1164-4f90-aa60-eacf2dd6d566\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>11566</th>\n",
              "      <td>2007.09186v3</td>\n",
              "      <td>2020-07-17</td>\n",
              "      <td>2020-10-07</td>\n",
              "      <td>AWS CORD-19 Search: A Neural Search Engine for...</td>\n",
              "      <td>None</td>\n",
              "      <td>Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2007.09186v3</td>\n",
              "      <td>With the global outbreak of Coronavirus diseas...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-347b0773-1164-4f90-aa60-eacf2dd6d566')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-347b0773-1164-4f90-aa60-eacf2dd6d566 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-347b0773-1164-4f90-aa60-eacf2dd6d566');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 79
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check that there are no rows with missing text."
      ],
      "metadata": {
        "id": "u4EeRGOczE08"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus.loc[pmc_arxiv_full_text_merged_plus['text'].isna()]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 49
        },
        "id": "ns5n9rP3TFTj",
        "outputId": "0a377a3d-169b-48ab-d8ef-65cd6006efac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Empty DataFrame\n",
              "Columns: [article_id, published, revised, title, journal, authors, doi, pdf_url, text]\n",
              "Index: []"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-ba9b29bc-c4fd-4fef-a774-aad0fb5e4c4e\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-ba9b29bc-c4fd-4fef-a774-aad0fb5e4c4e')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-ba9b29bc-c4fd-4fef-a774-aad0fb5e4c4e button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-ba9b29bc-c4fd-4fef-a774-aad0fb5e4c4e');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 80
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-02-18_pmc_arxiv_full_text_merged_plus.pickle', 'wb') as f:  #\n",
        "  pickle.dump(pmc_arxiv_full_text_merged_plus, f)"
      ],
      "metadata": {
        "id": "r39twgF_StHG"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 6.2 Clean sample of articles from full dataset\n",
        "\n"
      ],
      "metadata": {
        "id": "vnuobYaXJVKi"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# copy combined PMC and arXiv data to create clean text version\n",
        "pmc_arxiv_full_text_merged_plus_cleaned = pmc_arxiv_full_text_merged_plus.copy()"
      ],
      "metadata": {
        "id": "-b6JWluYH7wB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus_cleaned.head(10)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "utqyFZfxIAT5",
        "outputId": "babf4f16-3c77-4727-b01f-225322967b2b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   article_id   published     revised  \\\n",
              "0  PMC9549161  2022-09-26  2022-10-14   \n",
              "1  PMC9539342  2022-09-22  2022-11-12   \n",
              "2  PMC9357751  2022-12-01  2022-12-05   \n",
              "3  PMC9346052  2022-08-03  2022-09-05   \n",
              "4  PMC9775208  2022-12-15  2022-12-25   \n",
              "5  PMC9527439  2022-09-25  2022-10-07   \n",
              "6  PMC9729590  2022-12-08  2023-01-03   \n",
              "7  PMC9236981  2022-06-28  2022-12-21   \n",
              "8  PMC9694939  2022-11-10  2022-12-13   \n",
              "9  PMC9556799  2022-10-13  2022-11-01   \n",
              "\n",
              "                                               title  \\\n",
              "0       Drug repositioning: A bibliometric analysis.   \n",
              "1  A review on computer-aided chemogenomics and d...   \n",
              "2  Repurposing Molnupiravir as a new opportunity ...   \n",
              "3  Scope of repurposed drugs against the potentia...   \n",
              "4  Drug Repurposing Using Gene Co-Expression and ...   \n",
              "5  Novel Drug Design for Treatment of COVID-19: A...   \n",
              "6  Repurposing FDA-approved drugs cetilistat, abi...   \n",
              "7  A comprehensive review of artificial intellige...   \n",
              "8  Structural Homology-Based Drug Repurposing App...   \n",
              "9  Rational drug repositioning for coronavirus-as...   \n",
              "\n",
              "                                             journal  \\\n",
              "0                          Frontiers in pharmacology   \n",
              "1                     Chemical biology & drug design   \n",
              "2  Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                               Structural chemistry   \n",
              "4                                            Biology   \n",
              "5  The Canadian journal of infectious diseases & ...   \n",
              "6                   Informatics in medicine unlocked   \n",
              "7  Biomedicine & pharmacotherapy = Biomedecine & ...   \n",
              "8                     Molecules (Basel, Switzerland)   \n",
              "9                                           iScience   \n",
              "\n",
              "                                             authors  \\\n",
              "0  Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1  Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                  0   \n",
              "3  Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                Mailem RC, Tayo LL.   \n",
              "5             Mousavi S, Zare S, Mirzaei M, Feizi A.   \n",
              "6  Shahabadi N, Zendehcheshm S, Mahdavi M, Khadem...   \n",
              "7  Ahmed F, Soomro AM, Chethikkattuveli Salih AR,...   \n",
              "8  Aljuaid A, Salam A, Almehmadi M, Baammi S, Als...   \n",
              "9  Wang J, Liu J, Luo M, Cui H, Zhang W, Zhao K, ...   \n",
              "\n",
              "                            doi  \\\n",
              "0     10.3389/fphar.2022.974849   \n",
              "1            10.1111/cbdd.14136   \n",
              "2                             0   \n",
              "3    10.1007/s11224-022-02020-z   \n",
              "4       10.3390/biology11121827   \n",
              "5          10.1155/2022/2044282   \n",
              "6     10.1016/j.imu.2022.101147   \n",
              "7  10.1016/j.biopha.2022.113350   \n",
              "8     10.3390/molecules27227732   \n",
              "9    10.1016/j.isci.2022.105348   \n",
              "\n",
              "                                             pdf_url  \\\n",
              "0  https://europepmc.org/articles/PMC9549161?pdf=...   \n",
              "1  https://europepmc.org/articles/PMC9539342?pdf=...   \n",
              "2  https://europepmc.org/articles/PMC9357751?pdf=...   \n",
              "3  https://europepmc.org/articles/PMC9346052?pdf=...   \n",
              "4  https://europepmc.org/articles/PMC9775208?pdf=...   \n",
              "5  https://europepmc.org/articles/PMC9527439?pdf=...   \n",
              "6  https://europepmc.org/articles/PMC9729590?pdf=...   \n",
              "7  https://europepmc.org/articles/PMC9236981?pdf=...   \n",
              "8  https://europepmc.org/articles/PMC9694939?pdf=...   \n",
              "9  https://europepmc.org/articles/PMC9556799?pdf=...   \n",
              "\n",
              "                                                text  \n",
              "0  Sir James Black, a winner of the 1988 Nobel Pr...  \n",
              "1  Tight and selective interaction between ligand...  \n",
              "2  The severe acute respiratory syndrome coronavi...  \n",
              "3  The sudden outbreak of SARS-CoV-2 in 2019 took...  \n",
              "4  The 2019 novel coronavirus, now dubbed SARS-Co...  \n",
              "5  Coronavirus disease 2019 (COVID-19), which was...  \n",
              "6  COVID-19 is an infectious disease caused by Co...  \n",
              "7  A novel coronavirus (CoV) first appeared by th...  \n",
              "8  Drug discovery is a time-consuming and costly ...  \n",
              "9  Coronavirus disease 2019 (COVID-19), caused by...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-058e3087-ad0f-4f6f-979f-3d4a7d0e31f8\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>PMC9527439</td>\n",
              "      <td>2022-09-25</td>\n",
              "      <td>2022-10-07</td>\n",
              "      <td>Novel Drug Design for Treatment of COVID-19: A...</td>\n",
              "      <td>The Canadian journal of infectious diseases &amp; ...</td>\n",
              "      <td>Mousavi S, Zare S, Mirzaei M, Feizi A.</td>\n",
              "      <td>10.1155/2022/2044282</td>\n",
              "      <td>https://europepmc.org/articles/PMC9527439?pdf=...</td>\n",
              "      <td>Coronavirus disease 2019 (COVID-19), which was...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>PMC9729590</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>2023-01-03</td>\n",
              "      <td>Repurposing FDA-approved drugs cetilistat, abi...</td>\n",
              "      <td>Informatics in medicine unlocked</td>\n",
              "      <td>Shahabadi N, Zendehcheshm S, Mahdavi M, Khadem...</td>\n",
              "      <td>10.1016/j.imu.2022.101147</td>\n",
              "      <td>https://europepmc.org/articles/PMC9729590?pdf=...</td>\n",
              "      <td>COVID-19 is an infectious disease caused by Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>PMC9236981</td>\n",
              "      <td>2022-06-28</td>\n",
              "      <td>2022-12-21</td>\n",
              "      <td>A comprehensive review of artificial intellige...</td>\n",
              "      <td>Biomedicine &amp; pharmacotherapy = Biomedecine &amp; ...</td>\n",
              "      <td>Ahmed F, Soomro AM, Chethikkattuveli Salih AR,...</td>\n",
              "      <td>10.1016/j.biopha.2022.113350</td>\n",
              "      <td>https://europepmc.org/articles/PMC9236981?pdf=...</td>\n",
              "      <td>A novel coronavirus (CoV) first appeared by th...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>PMC9694939</td>\n",
              "      <td>2022-11-10</td>\n",
              "      <td>2022-12-13</td>\n",
              "      <td>Structural Homology-Based Drug Repurposing App...</td>\n",
              "      <td>Molecules (Basel, Switzerland)</td>\n",
              "      <td>Aljuaid A, Salam A, Almehmadi M, Baammi S, Als...</td>\n",
              "      <td>10.3390/molecules27227732</td>\n",
              "      <td>https://europepmc.org/articles/PMC9694939?pdf=...</td>\n",
              "      <td>Drug discovery is a time-consuming and costly ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>PMC9556799</td>\n",
              "      <td>2022-10-13</td>\n",
              "      <td>2022-11-01</td>\n",
              "      <td>Rational drug repositioning for coronavirus-as...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Wang J, Liu J, Luo M, Cui H, Zhang W, Zhao K, ...</td>\n",
              "      <td>10.1016/j.isci.2022.105348</td>\n",
              "      <td>https://europepmc.org/articles/PMC9556799?pdf=...</td>\n",
              "      <td>Coronavirus disease 2019 (COVID-19), caused by...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-058e3087-ad0f-4f6f-979f-3d4a7d0e31f8')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-058e3087-ad0f-4f6f-979f-3d4a7d0e31f8 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-058e3087-ad0f-4f6f-979f-3d4a7d0e31f8');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Functions to remove unwanted text, punctuation etc from text column using replace() method and regular expressions."
      ],
      "metadata": {
        "id": "msmGD4o01bOG"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "```python\n",
        "def remove_leading_and_trailing_spaces(text: str):\n",
        "    return text.strip()\n",
        "\n",
        "def remove_ligatures(text: str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"ﬀ\", \"ff\")\n",
        "        .replace(\"ﬁ\", \"fi\")\n",
        "        .replace(\"ﬂ\", \"fl\")\n",
        "        .replace(\"ﬃ\", \"ffi\")\n",
        "        .replace(\"ﬄ\", \"ffl\")\n",
        "        .replace(\"ﬅ\", \"ft\")\n",
        "    )\n",
        "\n",
        "def remove_bullets(text: str):\n",
        "    return text.replace(\"•\", \" \")\n",
        "\n",
        "def remove_non_printing_characters(text: str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"\\xa0\", \" \")\n",
        "        .replace(\"\\x0b\", \" \")\n",
        "        .replace(\"\\x0c\", \" \")\n",
        "    )\n",
        "\n",
        "def remove_soft_hyphens(text: str):\n",
        "    return (\n",
        "        text\n",
        "        .replace(\"\\xad\", \"\")\n",
        "        .replace(\"-\\n\", \"\")\n",
        "    )\n",
        "\n",
        "def remove_thin_space(text: str):\n",
        "    return text.replace(\"\\u2009\", \" \" )\n",
        "\n",
        "\n",
        "def remove_slash_apostrophe(text: str):\n",
        "        return text.replace(\"\\'\", \"'\")\n",
        "\n",
        "\n",
        "email_regex = re.compile(r\"([-!#-'*+/-9=?A-Z^-~]+(\\.[-!#-'*+/-9=?A-Z^-~]+)*|\\\"([]!#-[^-~ \\t]|(\\\\[\\t -~]))+\\\")@([-!#-'*+/-9=?A-Z^-~]+(\\.[-!#-'*+/-9=?A-Z^-~]+)*|\\[[\\t -Z^-~]])+\\s?\")\n",
        "\n",
        "def remove_emails(text: str):\n",
        "    return email_regex.sub('email', text)\n",
        "\n",
        "\n",
        "url_regex = re.compile(r'((http|https)\\:\\/\\/)?[a-zA-Z0-9\\.\\/\\?\\:@\\-_=#]+\\.([a-zA-Z]){2,6}([a-zA-Z0-9\\.\\&\\/\\?\\:@\\-_=#])*(?=\\.)*(?=\\.)')\n",
        "\n",
        "def remove_urls(text: str):\n",
        "    return url_regex.sub('website', text)\n",
        "\n",
        "\n",
        "references_regex = re.compile(r'\\s\\([A-Z]\\w+[^\\)]+?, (19|20)[0-9]{2}\\)')\n",
        "\n",
        "def remove_references(text: str):\n",
        "    return references_regex.sub(' ', text)\n",
        "\n",
        "\n",
        "def remove_newline_characters(text: str):\n",
        "    return text.replace(\"\\n\", \" \")\n",
        "\n",
        "\n",
        "square_brackets_regex = re.compile(r'\\[[0-9,\\s]*\\]')\n",
        "\n",
        "def remove_square_brackets(text: str):\n",
        "    return square_brackets_regex.sub(' ', text)\n",
        "\n",
        "\n",
        "parentheses_regex = re.compile(r'\\([0-9,.;%\\-\\s]*\\)')\n",
        "\n",
        "def remove_parentheses(text: str):\n",
        "    return parentheses_regex.sub('', text)\n",
        "\n",
        "\n",
        "figures_tables_regex = re.compile(r'\\s*\\(((t|T)able|(f|F)ig)[^)]+\\)')\n",
        "\n",
        "def remove_figures_tables(text: str):\n",
        "    return figures_tables_regex.sub(' ', text)\n",
        "\n",
        "\n",
        "def remove_brackets(text: str):\n",
        "    return (\n",
        "          text\n",
        "          .replace(\"[\", ' ')\n",
        "          .replace(\"]\", ' ')\n",
        "          )\n",
        "\n",
        "\n",
        "special_chars_regex = re.compile(r'[“”\\\"!*@#$_?^=:;]')\n",
        "\n",
        "\n",
        "def remove_special_chars(text: str):\n",
        "    return special_chars_regex.sub(' ', text)\n",
        "\n",
        "\n",
        "def remove_greek_letters(text: str):\n",
        "    greek_codes   = chain(range(0x370, 0x3e2), range(0x3f0, 0x400))\n",
        "    greek_symbols = (chr(c) for c in greek_codes)\n",
        "    greek_letters = [c for c in greek_symbols if c.isalpha()]\n",
        "\n",
        "    for letter in greek_letters:\n",
        "      text = text.replace(letter, ' ')\n",
        "    return text\n",
        "\n",
        "\n",
        "def remove_accents(text: str):\n",
        "    return unicodedata.normalize('NFD', text)\n",
        "\n",
        "\n",
        "def remove_symbols(text: str):\n",
        "    return (\n",
        "         text\n",
        "         .replace('\\u2122', '')\n",
        "         .replace('\\u00a9', '')\n",
        "         .replace('\\u00AE', '')\n",
        "         )\n",
        "\n",
        "\n",
        "extra_whitespace_regex = re.compile(r'\\s{2,}')\n",
        "\n",
        "def remove_extra_whitespace(text: str):\n",
        "    return extra_whitespace_regex.sub(' ', text)\n",
        "\n",
        "\n",
        "def remove_preceding_space(text: str):\n",
        "      return (\n",
        "         text\n",
        "         .replace(\" .\", \".\")\n",
        "         .replace(' ,', ',')\n",
        "         )\n",
        "\n",
        "\n",
        "thousand_separators_regex = re.compile(r'\\,(?=\\d{2})')\n",
        "\n",
        "def remove_thousand_separators(text:str):\n",
        "    return thousand_separators_regex.sub('', text)\n",
        "\n",
        "def remove_ellipsis(text: str):\n",
        "    return (text\n",
        "            .replace(\"...\", \" \")\n",
        "            .replace(\",.\", \".\")\n",
        "            .replace(\"..\", \".\")\n",
        "            .replace(\",,\", \",\")\n",
        "    )\n",
        "\n",
        "\n",
        "add_space_regex = re.compile(r'(?<!\\s[a-zA-Z]|\\.[a-zA-Z])\\.(?!(?<=\\d\\.)\\d)(?=[A-Z])')\n",
        "\n",
        "def add_space_after_full_stops(text: str):\n",
        "      return (add_space_regex.sub('. ', text))\n",
        "```"
      ],
      "metadata": {
        "id": "qIf76qwqS5Rp"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Function to call the functions above on text column."
      ],
      "metadata": {
        "id": "-lanw-Ed12MS"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def clean_text_all(text):\n",
        "\n",
        "    text = remove_leading_and_trailing_spaces(text)\n",
        "    text = remove_ligatures(text)\n",
        "    text = remove_bullets(text)\n",
        "    text = remove_non_printing_characters(text)\n",
        "    text = remove_soft_hyphens(text)\n",
        "    text = remove_thin_space(text)\n",
        "    text = remove_slash_apostrophe(text)\n",
        "    text = remove_emails(text)\n",
        "    text = remove_urls(text)\n",
        "    text = remove_references(text)\n",
        "    text = remove_newline_characters(text)\n",
        "    text = remove_square_brackets(text)\n",
        "    text = remove_parentheses(text)\n",
        "    text = remove_figures_tables(text)\n",
        "    text = remove_brackets(text)\n",
        "    text = remove_special_chars(text)\n",
        "    text = remove_greek_letters(text)\n",
        "    text = remove_accents(text)\n",
        "    text = remove_symbols(text)\n",
        "    text = remove_extra_whitespace(text)\n",
        "    text = remove_preceding_space(text)\n",
        "    text = remove_thousand_separators(text)\n",
        "    text = remove_ellipsis(text)\n",
        "    text = add_space_after_full_stops(text)\n",
        "    return text"
      ],
      "metadata": {
        "id": "WsnMSuSAVQGw"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Test clean text functions on a sample sentence."
      ],
      "metadata": {
        "id": "xLUY6TtZnRJy"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "test = \"For the helicase (Nsp13) of SARS-CoV-2, a multifunctional protein that helps in the proliferation and replication of the virus, drugs such as vapreotide, atazanavir, daclatasvir, and bismuth potassium citrate commonly block this function of the target. Likewise, dinucleoside 123 and raltegravir affect the pathways of methyltransferases (guanine-N7-methyltransferase, 2’-O-methyltransferase), while gliclazide and memantine affect the SARS-CoV-2 E protein, which is the envelope protein of the virus. Arbidol and chloroquine target the spike protein that helps in the viral entry, along with ACE2 protein in the host, thereby affecting the spike protein pathway [4] (Table 6).\""
      ],
      "metadata": {
        "id": "GrZXHU1zVC7G"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "test = clean_text_all(test)\n",
        "test"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        },
        "id": "ZqNbis3yVsaW",
        "outputId": "9f7846ce-340f-4ab8-ab3a-8c9771a9812a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'For the helicase Nsp13 of SARS-CoV-2, a multifunctional protein that helps in the proliferation and replication of the virus, drugs such as vapreotide, atazanavir, daclatasvir, and bismuth potassium citrate commonly block this function of the target. Likewise, dinucleoside 123 and raltegravir affect the pathways of methyltransferases guanine-N7-methyltransferase, 2’-O-methyltransferase, while gliclazide and memantine affect the SARS-CoV-2 E protein, which is the envelope protein of the virus. Arbidol and chloroquine target the spike protein that helps in the viral entry, along with ACE2 protein in the host, thereby affecting the spike protein pathway 4.'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We can see that parentheses and square brackets have been removed."
      ],
      "metadata": {
        "id": "ZowwW13gnok7"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The first 10 rows of the dataset below before running the clean text function still have parentheses."
      ],
      "metadata": {
        "id": "HJDE1zgJLTy4"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus_cleaned.loc[0:9,['article_id', 'text']] # before running function"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 363
        },
        "id": "Ix5LaVGHpGKw",
        "outputId": "16f801aa-7146-4464-b5b0-736cef914506"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   article_id                                               text\n",
              "0  PMC9549161  Sir James Black, a winner of the 1988 Nobel Pr...\n",
              "1  PMC9539342  Tight and selective interaction between ligand...\n",
              "2  PMC9357751  The severe acute respiratory syndrome coronavi...\n",
              "3  PMC9346052  The sudden outbreak of SARS-CoV-2 in 2019 took...\n",
              "4  PMC9775208  The 2019 novel coronavirus, now dubbed SARS-Co...\n",
              "5  PMC9527439  Coronavirus disease 2019 (COVID-19), which was...\n",
              "6  PMC9729590  COVID-19 is an infectious disease caused by Co...\n",
              "7  PMC9236981  A novel coronavirus (CoV) first appeared by th...\n",
              "8  PMC9694939  Drug discovery is a time-consuming and costly ...\n",
              "9  PMC9556799  Coronavirus disease 2019 (COVID-19), caused by..."
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-70691ad3-f872-4e11-a382-8ac11f7ada8a\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>PMC9527439</td>\n",
              "      <td>Coronavirus disease 2019 (COVID-19), which was...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>PMC9729590</td>\n",
              "      <td>COVID-19 is an infectious disease caused by Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>PMC9236981</td>\n",
              "      <td>A novel coronavirus (CoV) first appeared by th...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>PMC9694939</td>\n",
              "      <td>Drug discovery is a time-consuming and costly ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>PMC9556799</td>\n",
              "      <td>Coronavirus disease 2019 (COVID-19), caused by...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-70691ad3-f872-4e11-a382-8ac11f7ada8a')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-70691ad3-f872-4e11-a382-8ac11f7ada8a button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-70691ad3-f872-4e11-a382-8ac11f7ada8a');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-c526af6e-86f1-47c6-a7f7-c2fcadf71dd1\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c526af6e-86f1-47c6-a7f7-c2fcadf71dd1')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-c526af6e-86f1-47c6-a7f7-c2fcadf71dd1 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create dataset of first 10 rows"
      ],
      "metadata": {
        "id": "ZftSEWBT1rzA"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus_cleaned_test = pmc_arxiv_full_text_merged_plus_cleaned.loc[0:9, :].copy()\n",
        "pmc_arxiv_full_text_merged_plus_cleaned_test"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "d94b4d3c-5c11-49bd-df1e-f09f0bc89ae0",
        "id": "FDEKxT3pss5b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   article_id   published     revised  \\\n",
              "0  PMC9549161  2022-09-26  2022-10-14   \n",
              "1  PMC9539342  2022-09-22  2022-11-12   \n",
              "2  PMC9357751  2022-12-01  2022-12-05   \n",
              "3  PMC9346052  2022-08-03  2022-09-05   \n",
              "4  PMC9775208  2022-12-15  2022-12-25   \n",
              "5  PMC9527439  2022-09-25  2022-10-07   \n",
              "6  PMC9729590  2022-12-08  2023-01-03   \n",
              "7  PMC9236981  2022-06-28  2022-12-21   \n",
              "8  PMC9694939  2022-11-10  2022-12-13   \n",
              "9  PMC9556799  2022-10-13  2022-11-01   \n",
              "\n",
              "                                               title  \\\n",
              "0       Drug repositioning: A bibliometric analysis.   \n",
              "1  A review on computer-aided chemogenomics and d...   \n",
              "2  Repurposing Molnupiravir as a new opportunity ...   \n",
              "3  Scope of repurposed drugs against the potentia...   \n",
              "4  Drug Repurposing Using Gene Co-Expression and ...   \n",
              "5  Novel Drug Design for Treatment of COVID-19: A...   \n",
              "6  Repurposing FDA-approved drugs cetilistat, abi...   \n",
              "7  A comprehensive review of artificial intellige...   \n",
              "8  Structural Homology-Based Drug Repurposing App...   \n",
              "9  Rational drug repositioning for coronavirus-as...   \n",
              "\n",
              "                                             journal  \\\n",
              "0                          Frontiers in pharmacology   \n",
              "1                     Chemical biology & drug design   \n",
              "2  Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                               Structural chemistry   \n",
              "4                                            Biology   \n",
              "5  The Canadian journal of infectious diseases & ...   \n",
              "6                   Informatics in medicine unlocked   \n",
              "7  Biomedicine & pharmacotherapy = Biomedecine & ...   \n",
              "8                     Molecules (Basel, Switzerland)   \n",
              "9                                           iScience   \n",
              "\n",
              "                                             authors  \\\n",
              "0  Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1  Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                  0   \n",
              "3  Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                Mailem RC, Tayo LL.   \n",
              "5             Mousavi S, Zare S, Mirzaei M, Feizi A.   \n",
              "6  Shahabadi N, Zendehcheshm S, Mahdavi M, Khadem...   \n",
              "7  Ahmed F, Soomro AM, Chethikkattuveli Salih AR,...   \n",
              "8  Aljuaid A, Salam A, Almehmadi M, Baammi S, Als...   \n",
              "9  Wang J, Liu J, Luo M, Cui H, Zhang W, Zhao K, ...   \n",
              "\n",
              "                            doi  \\\n",
              "0     10.3389/fphar.2022.974849   \n",
              "1            10.1111/cbdd.14136   \n",
              "2                             0   \n",
              "3    10.1007/s11224-022-02020-z   \n",
              "4       10.3390/biology11121827   \n",
              "5          10.1155/2022/2044282   \n",
              "6     10.1016/j.imu.2022.101147   \n",
              "7  10.1016/j.biopha.2022.113350   \n",
              "8     10.3390/molecules27227732   \n",
              "9    10.1016/j.isci.2022.105348   \n",
              "\n",
              "                                             pdf_url  \\\n",
              "0  https://europepmc.org/articles/PMC9549161?pdf=...   \n",
              "1  https://europepmc.org/articles/PMC9539342?pdf=...   \n",
              "2  https://europepmc.org/articles/PMC9357751?pdf=...   \n",
              "3  https://europepmc.org/articles/PMC9346052?pdf=...   \n",
              "4  https://europepmc.org/articles/PMC9775208?pdf=...   \n",
              "5  https://europepmc.org/articles/PMC9527439?pdf=...   \n",
              "6  https://europepmc.org/articles/PMC9729590?pdf=...   \n",
              "7  https://europepmc.org/articles/PMC9236981?pdf=...   \n",
              "8  https://europepmc.org/articles/PMC9694939?pdf=...   \n",
              "9  https://europepmc.org/articles/PMC9556799?pdf=...   \n",
              "\n",
              "                                                text  \n",
              "0  Sir James Black, a winner of the 1988 Nobel Pr...  \n",
              "1  Tight and selective interaction between ligand...  \n",
              "2  The severe acute respiratory syndrome coronavi...  \n",
              "3  The sudden outbreak of SARS-CoV-2 in 2019 took...  \n",
              "4  The 2019 novel coronavirus, now dubbed SARS-Co...  \n",
              "5  Coronavirus disease 2019 (COVID-19), which was...  \n",
              "6  COVID-19 is an infectious disease caused by Co...  \n",
              "7  A novel coronavirus (CoV) first appeared by th...  \n",
              "8  Drug discovery is a time-consuming and costly ...  \n",
              "9  Coronavirus disease 2019 (COVID-19), caused by...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1036dd87-7a25-4e6c-846e-da8a83fc96b6\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>PMC9527439</td>\n",
              "      <td>2022-09-25</td>\n",
              "      <td>2022-10-07</td>\n",
              "      <td>Novel Drug Design for Treatment of COVID-19: A...</td>\n",
              "      <td>The Canadian journal of infectious diseases &amp; ...</td>\n",
              "      <td>Mousavi S, Zare S, Mirzaei M, Feizi A.</td>\n",
              "      <td>10.1155/2022/2044282</td>\n",
              "      <td>https://europepmc.org/articles/PMC9527439?pdf=...</td>\n",
              "      <td>Coronavirus disease 2019 (COVID-19), which was...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>PMC9729590</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>2023-01-03</td>\n",
              "      <td>Repurposing FDA-approved drugs cetilistat, abi...</td>\n",
              "      <td>Informatics in medicine unlocked</td>\n",
              "      <td>Shahabadi N, Zendehcheshm S, Mahdavi M, Khadem...</td>\n",
              "      <td>10.1016/j.imu.2022.101147</td>\n",
              "      <td>https://europepmc.org/articles/PMC9729590?pdf=...</td>\n",
              "      <td>COVID-19 is an infectious disease caused by Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>PMC9236981</td>\n",
              "      <td>2022-06-28</td>\n",
              "      <td>2022-12-21</td>\n",
              "      <td>A comprehensive review of artificial intellige...</td>\n",
              "      <td>Biomedicine &amp; pharmacotherapy = Biomedecine &amp; ...</td>\n",
              "      <td>Ahmed F, Soomro AM, Chethikkattuveli Salih AR,...</td>\n",
              "      <td>10.1016/j.biopha.2022.113350</td>\n",
              "      <td>https://europepmc.org/articles/PMC9236981?pdf=...</td>\n",
              "      <td>A novel coronavirus (CoV) first appeared by th...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>PMC9694939</td>\n",
              "      <td>2022-11-10</td>\n",
              "      <td>2022-12-13</td>\n",
              "      <td>Structural Homology-Based Drug Repurposing App...</td>\n",
              "      <td>Molecules (Basel, Switzerland)</td>\n",
              "      <td>Aljuaid A, Salam A, Almehmadi M, Baammi S, Als...</td>\n",
              "      <td>10.3390/molecules27227732</td>\n",
              "      <td>https://europepmc.org/articles/PMC9694939?pdf=...</td>\n",
              "      <td>Drug discovery is a time-consuming and costly ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>PMC9556799</td>\n",
              "      <td>2022-10-13</td>\n",
              "      <td>2022-11-01</td>\n",
              "      <td>Rational drug repositioning for coronavirus-as...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Wang J, Liu J, Luo M, Cui H, Zhang W, Zhao K, ...</td>\n",
              "      <td>10.1016/j.isci.2022.105348</td>\n",
              "      <td>https://europepmc.org/articles/PMC9556799?pdf=...</td>\n",
              "      <td>Coronavirus disease 2019 (COVID-19), caused by...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1036dd87-7a25-4e6c-846e-da8a83fc96b6')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-1036dd87-7a25-4e6c-846e-da8a83fc96b6 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-1036dd87-7a25-4e6c-846e-da8a83fc96b6');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-399da648-2d61-479a-b0ab-4ab331bedb96\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-399da648-2d61-479a-b0ab-4ab331bedb96')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-399da648-2d61-479a-b0ab-4ab331bedb96 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Run clean text function on test dataset"
      ],
      "metadata": {
        "id": "O_8DCgH911x_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus_cleaned_test['text'] = pmc_arxiv_full_text_merged_plus_cleaned_test['text'].map(clean_text_all)\n",
        "pmc_arxiv_full_text_merged_plus_cleaned_test"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "ZDGUkj9P_xiF",
        "outputId": "a8368aed-d435-48cf-dc69-f2d98f2b4cc9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   article_id   published     revised  \\\n",
              "0  PMC9549161  2022-09-26  2022-10-14   \n",
              "1  PMC9539342  2022-09-22  2022-11-12   \n",
              "2  PMC9357751  2022-12-01  2022-12-05   \n",
              "3  PMC9346052  2022-08-03  2022-09-05   \n",
              "4  PMC9775208  2022-12-15  2022-12-25   \n",
              "5  PMC9527439  2022-09-25  2022-10-07   \n",
              "6  PMC9729590  2022-12-08  2023-01-03   \n",
              "7  PMC9236981  2022-06-28  2022-12-21   \n",
              "8  PMC9694939  2022-11-10  2022-12-13   \n",
              "9  PMC9556799  2022-10-13  2022-11-01   \n",
              "\n",
              "                                               title  \\\n",
              "0       Drug repositioning: A bibliometric analysis.   \n",
              "1  A review on computer-aided chemogenomics and d...   \n",
              "2  Repurposing Molnupiravir as a new opportunity ...   \n",
              "3  Scope of repurposed drugs against the potentia...   \n",
              "4  Drug Repurposing Using Gene Co-Expression and ...   \n",
              "5  Novel Drug Design for Treatment of COVID-19: A...   \n",
              "6  Repurposing FDA-approved drugs cetilistat, abi...   \n",
              "7  A comprehensive review of artificial intellige...   \n",
              "8  Structural Homology-Based Drug Repurposing App...   \n",
              "9  Rational drug repositioning for coronavirus-as...   \n",
              "\n",
              "                                             journal  \\\n",
              "0                          Frontiers in pharmacology   \n",
              "1                     Chemical biology & drug design   \n",
              "2  Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                               Structural chemistry   \n",
              "4                                            Biology   \n",
              "5  The Canadian journal of infectious diseases & ...   \n",
              "6                   Informatics in medicine unlocked   \n",
              "7  Biomedicine & pharmacotherapy = Biomedecine & ...   \n",
              "8                     Molecules (Basel, Switzerland)   \n",
              "9                                           iScience   \n",
              "\n",
              "                                             authors  \\\n",
              "0  Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1  Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                  0   \n",
              "3  Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                Mailem RC, Tayo LL.   \n",
              "5             Mousavi S, Zare S, Mirzaei M, Feizi A.   \n",
              "6  Shahabadi N, Zendehcheshm S, Mahdavi M, Khadem...   \n",
              "7  Ahmed F, Soomro AM, Chethikkattuveli Salih AR,...   \n",
              "8  Aljuaid A, Salam A, Almehmadi M, Baammi S, Als...   \n",
              "9  Wang J, Liu J, Luo M, Cui H, Zhang W, Zhao K, ...   \n",
              "\n",
              "                            doi  \\\n",
              "0     10.3389/fphar.2022.974849   \n",
              "1            10.1111/cbdd.14136   \n",
              "2                             0   \n",
              "3    10.1007/s11224-022-02020-z   \n",
              "4       10.3390/biology11121827   \n",
              "5          10.1155/2022/2044282   \n",
              "6     10.1016/j.imu.2022.101147   \n",
              "7  10.1016/j.biopha.2022.113350   \n",
              "8     10.3390/molecules27227732   \n",
              "9    10.1016/j.isci.2022.105348   \n",
              "\n",
              "                                             pdf_url  \\\n",
              "0  https://europepmc.org/articles/PMC9549161?pdf=...   \n",
              "1  https://europepmc.org/articles/PMC9539342?pdf=...   \n",
              "2  https://europepmc.org/articles/PMC9357751?pdf=...   \n",
              "3  https://europepmc.org/articles/PMC9346052?pdf=...   \n",
              "4  https://europepmc.org/articles/PMC9775208?pdf=...   \n",
              "5  https://europepmc.org/articles/PMC9527439?pdf=...   \n",
              "6  https://europepmc.org/articles/PMC9729590?pdf=...   \n",
              "7  https://europepmc.org/articles/PMC9236981?pdf=...   \n",
              "8  https://europepmc.org/articles/PMC9694939?pdf=...   \n",
              "9  https://europepmc.org/articles/PMC9556799?pdf=...   \n",
              "\n",
              "                                                text  \n",
              "0  Sir James Black, a winner of the 1988 Nobel Pr...  \n",
              "1  Tight and selective interaction between ligand...  \n",
              "2  The severe acute respiratory syndrome coronavi...  \n",
              "3  The sudden outbreak of SARS-CoV-2 in 2019 took...  \n",
              "4  The 2019 novel coronavirus, now dubbed SARS-Co...  \n",
              "5  Coronavirus disease 2019 COVID-19, which was f...  \n",
              "6  COVID-19 is an infectious disease caused by Co...  \n",
              "7  A novel coronavirus CoV first appeared by the ...  \n",
              "8  Drug discovery is a time-consuming and costly ...  \n",
              "9  Coronavirus disease 2019 COVID-19, caused by s...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-7540e7b9-ec3a-421d-9426-be7a70c77167\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>PMC9527439</td>\n",
              "      <td>2022-09-25</td>\n",
              "      <td>2022-10-07</td>\n",
              "      <td>Novel Drug Design for Treatment of COVID-19: A...</td>\n",
              "      <td>The Canadian journal of infectious diseases &amp; ...</td>\n",
              "      <td>Mousavi S, Zare S, Mirzaei M, Feizi A.</td>\n",
              "      <td>10.1155/2022/2044282</td>\n",
              "      <td>https://europepmc.org/articles/PMC9527439?pdf=...</td>\n",
              "      <td>Coronavirus disease 2019 COVID-19, which was f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>PMC9729590</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>2023-01-03</td>\n",
              "      <td>Repurposing FDA-approved drugs cetilistat, abi...</td>\n",
              "      <td>Informatics in medicine unlocked</td>\n",
              "      <td>Shahabadi N, Zendehcheshm S, Mahdavi M, Khadem...</td>\n",
              "      <td>10.1016/j.imu.2022.101147</td>\n",
              "      <td>https://europepmc.org/articles/PMC9729590?pdf=...</td>\n",
              "      <td>COVID-19 is an infectious disease caused by Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>PMC9236981</td>\n",
              "      <td>2022-06-28</td>\n",
              "      <td>2022-12-21</td>\n",
              "      <td>A comprehensive review of artificial intellige...</td>\n",
              "      <td>Biomedicine &amp; pharmacotherapy = Biomedecine &amp; ...</td>\n",
              "      <td>Ahmed F, Soomro AM, Chethikkattuveli Salih AR,...</td>\n",
              "      <td>10.1016/j.biopha.2022.113350</td>\n",
              "      <td>https://europepmc.org/articles/PMC9236981?pdf=...</td>\n",
              "      <td>A novel coronavirus CoV first appeared by the ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>PMC9694939</td>\n",
              "      <td>2022-11-10</td>\n",
              "      <td>2022-12-13</td>\n",
              "      <td>Structural Homology-Based Drug Repurposing App...</td>\n",
              "      <td>Molecules (Basel, Switzerland)</td>\n",
              "      <td>Aljuaid A, Salam A, Almehmadi M, Baammi S, Als...</td>\n",
              "      <td>10.3390/molecules27227732</td>\n",
              "      <td>https://europepmc.org/articles/PMC9694939?pdf=...</td>\n",
              "      <td>Drug discovery is a time-consuming and costly ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>PMC9556799</td>\n",
              "      <td>2022-10-13</td>\n",
              "      <td>2022-11-01</td>\n",
              "      <td>Rational drug repositioning for coronavirus-as...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Wang J, Liu J, Luo M, Cui H, Zhang W, Zhao K, ...</td>\n",
              "      <td>10.1016/j.isci.2022.105348</td>\n",
              "      <td>https://europepmc.org/articles/PMC9556799?pdf=...</td>\n",
              "      <td>Coronavirus disease 2019 COVID-19, caused by s...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7540e7b9-ec3a-421d-9426-be7a70c77167')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-7540e7b9-ec3a-421d-9426-be7a70c77167 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-7540e7b9-ec3a-421d-9426-be7a70c77167');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-da95e3ab-c498-4242-b055-ff57f2d1bbd4\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-da95e3ab-c498-4242-b055-ff57f2d1bbd4')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-da95e3ab-c498-4242-b055-ff57f2d1bbd4 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_c9021d78-e173-4943-879b-b668e1b1969f\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('pmc_arxiv_full_text_merged_plus_cleaned_test')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_c9021d78-e173-4943-879b-b668e1b1969f button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('pmc_arxiv_full_text_merged_plus_cleaned_test');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "pmc_arxiv_full_text_merged_plus_cleaned_test",
              "summary": "{\n  \"name\": \"pmc_arxiv_full_text_merged_plus_cleaned_test\",\n  \"rows\": 10,\n  \"fields\": [\n    {\n      \"column\": \"article_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"PMC9694939\",\n          \"PMC9539342\",\n          \"PMC9527439\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"published\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"2022-11-10\",\n          \"2022-09-22\",\n          \"2022-09-25\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"revised\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"2022-12-13\",\n          \"2022-11-12\",\n          \"2022-10-07\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Structural Homology-Based Drug Repurposing Approach for Targeting NSP12 SARS-CoV-2.\",\n          \"A review on computer-aided chemogenomics and drug repositioning for rational COVID-19 drug discovery.\",\n          \"Novel Drug Design for Treatment of COVID-19: A Systematic Review of Preclinical Studies.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"journal\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Molecules (Basel, Switzerland)\",\n          \"Chemical biology & drug design\",\n          \"The Canadian journal of infectious diseases & medical microbiology = Journal canadien des maladies infectieuses et de la microbiologie medicale\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Aljuaid A, Salam A, Almehmadi M, Baammi S, Alshabrmi FM, Allahyani M, Al-Zaydi KM, Izmirly AM, Almaghrabi S, Baothman BK, Shahab M.\",\n          \"Maghsoudi S, Taghavi Shahraki B, Rameh F, Nazarabi M, Fatahi Y, Akhavan O, Rabiee M, Mostafavi E, Lima EC, Saeb MR, Rabiee N.\",\n          \"Mousavi S, Zare S, Mirzaei M, Feizi A.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"doi\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"10.3390/molecules27227732\",\n          \"10.1111/cbdd.14136\",\n          \"10.1155/2022/2044282\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pdf_url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"https://europepmc.org/articles/PMC9694939?pdf=render\",\n          \"https://europepmc.org/articles/PMC9539342?pdf=render\",\n          \"https://europepmc.org/articles/PMC9527439?pdf=render\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Drug discovery is a time-consuming and costly process that involves a lengthy assessment of clinical trials and has a success rate of only 2%, with an average cost of USD 2\\u20133 billion, and also fails to deliver at the time of pandemics. On the other hand, drug repurposing has several advantages over drug discovery, and one is that it can be delivered in times of pandemics. At the beginning of February 2020, the World Health Organization WHO declared the novel coronavirus disease 2019 COVID-19 a pandemic due to its rapid spread as well as its high mortality and morbidity rates. COVID-19 first appeared in Wuhan city, China, in late 2019 and is still spreading all over the globe creating a disastrous effect on global health and the economy. Due to its highly contagious nature, as of May 2022, the pandemic has spread to more than 200 countries and territories with over 5.5 million confirmed positive cases and mortality of over 350K across the globe. The etiological agent of COVID-19 is severe acute respiratory syndrome coronavirus 2 SARS-CoV-2, and it has been reported to originate from the animal coronavirus. SARS-CoV-2 belongs to the genus Betacoronavirus of the family Coronaviridae which already has two other members, i.e., SARS coronavirus SARS-CoV-1 and Middle East respiratory syndrome coronavirus MERS-CoV which caused pandemic onslaughts in 2002 and 2012, respectively, resulting in severe human losses both in terms of morbidity and mortality. Coronaviruses are positive-sense single-stranded RNA with a genome size of ~30 kb containing a 5\\u2032 cap-structure and 3\\u2032-polyadenylated tail. Their genome encodes for ORF1a and ORF1ab polyprotein which cleaves into multiple subunits and performs a diverse range of functions, i.e., binding and invading the host cell, viral replication, and evading the host immune system. The proteins of coronaviruses perform their function by aggregating multiple subunits into an heteromeric complex, i.e., replication-transcription complex and several other structural proteins that assemble into new virions. One of these subunits is a non-structural protein 12 NSP-12 which involves RNA-dependent RNA polymerase RdRp activity. NSP-12 of SARS-CoV-2 is highly conserved by sharing 96% similarity with SARS-CoV-1 and 70% with MERS-CoV. The recently resolved structure has also revealed conserved structural architecture by comparison with non-structural protein 5 NS5 of flaviviruses and polymerases of picornaviruses. However, NSP-12 has a length of ~930 amino acids in contrast to the polymerases of Human Poliovirus PV and NS5B of Hepatitis C Virus HCV which usually comprises ~500 amino acids. NSP-12 is responsible for viral replication and transcription once the virus invades into the host cellular environment by aggregating with cofactors such as NSP-7 and NSP-8 along with some additional subunits to initiate viral replication. Its tertiary structure comprises three highly conserved domains N-terminal nucleotidyltransferase also known as nidovirus RdRp-associated nucleotidyltransferase NiRAN, interface, and C-terminal RdRp domain. The RdRp domain of NSP-12 is highly conserved and responsible for viral replication and has been reported in several families of RNA genome viruses, while the NiRAN domain is only conserved among the members of nidoviruses by possessing kinase-like fold and involves in attachment with NSP-7 and NSP-8. RdRp domain contains seven conserved motifs motif A-G that involve attachment with nucleotide triphosphate NTP and RNA primer during the viral replication. Due to these conserved evolutionary features of the RdRp domain, it has been suggested to be an important therapeutic target. Currently, many studies including single-ligand docking as well as screening of the compounds library have been conducted to search for potential inhibitors against NSP-12 of SARS-CoV2. In our study, we employed the structural-based homology method to identify structurally-conserved proteins and then identified US Food and Drug Association FDA -approved inhibitors of the homologous structure. The selected FDA-approved inhibitors were Dasabuvir Sodium Dasabuvir and Sofosbuvir which were previously authorized for inhibiting the NS5B of HCV. Furthermore, the conserved active site of NSP-12 was characterized and the affinity of the inhibitors were evaluated using in silico assessments, i.e., molecular docking and simulations. The DALI server was employed for the identification of homologous protein structures to NSP-12 of SARS-CoV-2 that were already submitted in the PDB repository. The submitted query resulted in a total of 1105 homologous structures with a default cutoff of Z-score \\u22652 as shown in. For screening significant homologs, a cutoff Z-score value \\u226519 was considered, resulting in 611 structures. These resulting homologous structures were involved in the RdRp activity and belonged to four viral families Coronaviridae, Flaviviridae, Picornaviridae, and Caliciviridae and infected a diverse range of hosts. The representatives belonging to different viral genera were selected from the filtered homologous structures and their similarities were evaluated by using the similarity matrix and a cladogram by the neighbor joining method. The resulting matrix was visualized as a heatmap as shown in A. The heatmap of Z-score-based structural similarity revealed the close relationship of coronaviruses with the members of Flaviviridae as well as Caliciviridae and Picornaviridae. Based on structural similarity, the unrooted cladogram of homologous structures to NSP-12 was clustered into four viral families as shown in B. The clade of the coronaviruses, i.e., SARS-CoV-1 and SARS-CoV-2, shares a close relationship with the NS5B of HCV and also with the other representatives of Flaviviridae, i.e., the Zika Virus ZIKV, Dengue Virus DENV, Bovine Viral Diarrhea Virus BVDV, Japanese Encephalitis Virus JEV, and Classical Swine Fever Virus CSFV. The members of Picornaviridae included PolV, Human Coxsackievirus CoxV, Human Rhinovirus HRV, Human Enterovirus HEV, Human Enterovirus 71 HEV71, Foot and Mouth Disease Virus FMDV, and Encephalomyocarditis virus. The representatives of Caliciviridae were the Sapporovirus SapV, Rabbit Haemorrhagic Disease Virus RHDV, Murine Norovirus MNV, Norwalk Virus NarV, and Human Norovirus HNV. The clade of Flaviviridae and coronaviruses shared significant relationships as compared to the members of Caliciviridae and Picornaviridae. The conserved motifs in NSP-12 were analyzed by multiple structures and sequence alignments implemented at the PROMALS3D server. The PROMLS3D server aligns multiple structures based on similarity at the amino acid sequence and secondary structure level. The identified conserved motifs in NSP-12 have been displayed in A. NSP-12 has seven highly conserved motifs motif A-G which are responsible for NTP catalysis, binding to the 3\\u2032 of RNA template, and viral replication by agglomerating with NSP7 and NSP8. To display the conserved orientation of these motifs at the active site of NSP-12, multiple structures belonging to the family Coronaviridae, Flaviviridae, and Picornaviridae were superposed structurally aligned by using the MatchMaker algorithm implemented in UCSF Chimera. The conserved motifs identified from multiple sequence and structural alignment have been shown in B. Motifs A\\u2013E are located in the palm domain while motifs F and G are present in the fingers domain of NSP-12. Motifs A and C have highly conserved aspartic acid residues which bind to the divalent ions Mn2+ or Mg2+ and are responsible for executing the catalysis of nucleotide-binding. Motifs B, D, E, and G are involved in nucleotide recognition and coordination. Motif F is enriched with positively charged basic residues which interact with the triphosphate moieties of the NTP. Motif G has been also reported to involve in adjusting the orientation of the primer and template. Pharmacophore modeling is a very useful technique for the detection of ligand-binding or a druggable site in protein that could be targeted to either trigger or to block the biological activity. Pharmacophore modeling was computed by using the CAVITY search module of the CavityPlus server with a selection of No Ligand mode and default parameters. The CAVITY search module evaluates the pharmacophore or druggable cavity based on surface energy and site geometry. The analysis resulted in the prediction of a total of 27 cavities in NSP-12 and their information has been shown in. The cavities were ranked based on factors, i.e., drug score, druggability, and predicted average PKD and the top-scored cavity as shown in C was selected for further analysis. The top-ranked cavity has a higher drug score revealing significant druggability of the active site with a predicted maximum pKd score of 6.99. The predicted maximum pKd score \\u22656 indicates a suitable ligand ability of a cavity binding site. There was a total of 154 amino acid residues constituting the predicted cavity and they were compared with the evolutionarily conserved residues of the RdRp domain of NSP-12 from multiple sequence alignments. The comparison has been plotted as a Venn diagram as shown in A. The highly conserved residues were K545 and R555 of motif F, D618, and D623 of motif A, S682, G683, T687, and N691 of motif B, D760, and D761 of motif C and W800 of motif D. These residues were further represented as a target in the RdRp domain of NSP-12 for molecular docking with the FDA-approved antivirals. Before molecular docking, the ligands, i.e., Dasabuvir, Sofosbuvir, Ribavirin, GTP, and UTP, were prepared using AutoDock Tools. The grid box was centered at the active site or druggable cavity and the residues were considered rigid for both receptors, i.e., NSP-12 of SARS-CoV-2 and NS5B of HCV. The molecular docking was performed by AutoDock Vina with the parameters that are suggested for a single-docking experiment and the results for each abovementioned ligand were stored. AutoDock Vina predicts the docking score in kcal/mol and the lower energy means higher binding affinity between the ligand and the receptor. The results score of ligand bindings with the active site of NSP-12 and NS5B has been compared and visualized as a bar chart as shown in B.The docking score of NSP-12 with the ligands was in the range of \\u22126.2 to \\u22128.1 kcal/mol, while for NS5B, it was \\u22126.5 to \\u22129.1 kcal/mol, and both had an average difference of 0.84 kcal/mol. The average docking score of NSP-12 and NS5B for Sofosbuvir, Dasabuvir, Ribavirin, GTP, and GTP were \\u22128.2, \\u22128.6, \\u22126.35, \\u22128.25, and \\u22127.6 kcal/mol, respectively. NSP-12 showed the lowest docking score \\u22128.1 kcal/mol with Dasabuvir and higher \\u22126.2 kcal/mol with Ribavirin. In general, the ligands showed a higher docking score for NS5B as compared to NSP-12. To examine the interactions between ligand and the conserved residues present at the active site of NSP-12, LigPlot + v2.1, standalone software was used. The predicted interactions of Sofosbuvir and Dasabuvir with NSP 12 have been shown in A,B, respectively. The conserved residues involved in forming hydrogen bonds with Sofosbuvir were C622 of motif-A, W800 of motif-D, G811, and S814 of motif-E, while Dasabuvir interacted by the formation of hydrogen bonds with D761 of motif-C and W800 of motif-D. In addition, several conserved residues of motif-A, -C, and -E were also involved in hydrophobic interactions with Sofosbuvir and the residues of motif-A, -B, -C, -E, and -F with Dasabuvir. On the other hand, the native competitors such as GTP interacted with NSP-12 by forming hydrogen bonds with A554, R555, and T556 of motif-F, Y619, K621, and C622 of motif-A, while UTP interacted with S759, D760, and D761 of motif-C, W800 of motif-D, and G811 and S814 of motif-E. In the case of HCV as a receptor, Sofosbuvir binds with motif-B and -C, while Dasabuvir interacted with residues of motif-A, -B, -C, and motif-F, and their receptor\\u2013ligand interaction plots have been shown in C,D, respectively. By binding to a protein\\u2019s functional binding cleft, each small molecule has the potential to cause significant conformational changes in the protein\\u2019s structure. To investigate the structural stability of 6M71, 100 ns of molecular dynamics MD simulations were run with the protein in both its apo and ligand-bound forms. The stability of the systems was evaluated using root mean square deviation RMSD, root mean square fluctuation RMSF, radius of gyration, and Hydrogen bond analysis. The average RMSD for 6M71, 6M71-Dasabuvir, Ribavirin, and Sofosbuvir complexes was calculated as 0.23, 0.23, 0.27, and 0.27 nm, respectively. All three systems\\u2019 RMSD values, which quantify conformational changes over time, indicated that the simulation was stable for up to 100 ns. The backbone RMSDs plot analysis revealed that the complex was stable throughout MD simulation despite a small deviation from its primary conformation. displays the RMSD plots of all complexes vs. simulated time. The resulting RMSD graph showed increasing trends with increasing RMSD values ranging from 0.10 to 0.33 nm between 0 and 60 ns in the case of 6M71/Ribavirin and 6M71/Sofosbuvir, suggesting the compounds were becoming accustomed to a new conformation inside the binding pocket after that, the plateau continued and finally, settled at 0.29 nm, which does not exceed the 0.3 nm threshold. However, 6M71/Sofosbuvir has the same evolution to apo 6M71. Lower RMSD values for all complexes investigated shows that Dasabuvir inhibitor is stable inside 6M71 and gives a strong foundation for our research. The root mean square fluctuation RMSF method was used to investigate the effect of ligand binding on the flexible structure of protein as well as the behavior of essential amino acids. Throughout the simulation, a lower RMSF value indicates more rigidity, whereas a higher RMSF value indicates more flexibility. In, RMSFs of the Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes are displayed. When compared to the apo form of CAII, it ise shown that the fluctuations of the ligand-bound complexes\\u2019 residues are relatively stable, particularly at the region where the residues are engaged in ligand binding. Furthermore, the average RMSF values of Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes were 0.10, 0.10, 0.11 and 0.11 nm, respectively. This finding implies that Dasabuvir, Ribavirin, and Sofosbuvir binding helped to maintain NSP-12 structurally stable. We examined how binding of different ligands affected the protein\\u2019s overall compactness in its structure. For this goal, the radius of gyration Rg was calculated as a function of time. Any ligand with a high enough Rg value will be more likely to be flexible, making it unstable. On the other hand, lower Rg values point to confirmation of a dense and closely packed conformation. The average Rg values Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes were found to be 2.99, 3.00, 2.97, 2.98 nm, respectively, suggesting that the binding of Dasabuvir, Ribavirin, and Sofosbuvir to NSP-12 packing did not cause significant change. Moreover, based on the graph, it appears that the Rg of NSP-12/Dasabuvir reached a stable equilibrium during the 100 ns simulation than NSP-12/Ribavirin and NSP-12/Sofosbuvir complexes. Further understanding of the examined chemicals\\u2019 molecular recognition, molecular interactions, and selectivity within the receptors can be attained using hydrogen bond analysis. For all MD trajectories of the complex, we calculated the number of hydrogen bonds created. displays the number of hydrogen bonds and pairs within 0.35 nm generated between NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes during the MD simulation. It was noticed that the Dasabuvir binds to the active pocket of NSP-12 with an average of 3.46 hydrogen bonds and 1.13 pairs within 0.35 nm. Similarly, Ribavirin acting as a ligand interacted with NSP-12 as a target within the binding site with an average of 3.97 hydrogen bonds, although the average of pairs within 0.35 nm is 1.66. However, the average of both hydrogen bonds and pairs within 0.35 nm for the NSP-12/Sofosbuvir complex was 1. On the other hand, the H-bonding plot showed that Dasabuvir was able to maintain a more robust interaction with the binding pockets of NSP-12 throughout the simulation time than Ribavirin and Sofosbuvir. Before MM/PBSA calculation, each complex between the receptors or inhibitors/ligands should reach the equilibrium state. From, it is clear that the temperature of each complex quickly reaches the target value 300 K and remains stable over 100 ns. After that, we utilized a python script MmPbSaStat.py to calculate average free binding energy of the selected complexes, as provided in the g mmpbsa package. This script calculates the average free binding energy and its standard deviation/error from the output files, which were obtained from g mmpbsa. The energy liberated during the process of bond formation, or alternatively, the interaction between a ligand and protein, is shown in the form of binding energy. The lesser the binding energy, the better the binding of the ligand and protein. The final binding energy is a cumulative sum of van der Wall, electrostatic, polar solvation, and SASA energy. Except for the polar solvation energy, all other forms of energy contributed favorably to the interaction between different molecules Dasabuvir, Ribavirin, and Sofosbuvir and NSP-12. The bioactive molecule Dasabuvir showed the least binding free energy \\u221242.151 KJ/mol among all the selected molecules. A comparison of the binding free energies of all the complexes was made by plotting the binding energy versus time graphs. Further, we examined the contribution of each residue of NSP-12 in terms of binding free energy to the interaction with the selected. The contribution of each residue was calculated by decomposing the total binding free energy of the system into per residue contribution energy molecules using python script MmPbSaDecomp.py provided by MM-PBSA package. From the overall results, it was observed that the amino acid residue Trp 800 showed the lowest contribution binding energy for all the tree complexes, suggesting their significant role in binding indeed, contribution energy analysis of amino acid residues Tyr619, Cys622, Trp800, Glu811, Ser814, and Asp761 revealed that these amino acids played an important role during interaction, which confirms the docking result. Finally, the Sofosbuvir and Dasabuvir could be established as a repurposed drug against the COVID-19 main protease. Drug repurposing or repositioning is a term coined for the study of therapeutic uses of existing drugs that are either previously approved or currently under the investigational phase. Drug repurposing offers potential advantages over the discovery of novel drugs, and one is that it can be delivered quickly without lengthy assessments and time-consuming clinical trials. Due to this significance, drug repurposing offers an effective way to save lives and time during pandemics. Many drugs have been already suggested for the current pandemic caused by SARS-CoV-2, of which the most prominent are FDA-approved Chloroquine and its derivatives\\u2014an anti-malarial drug and Remdesivir. Remdesivir, a nucleoside analog, has been shown to significantly reduce the SARS-CoV-2 pathogenicity in in vitro study and is still under investigation. However, there is still a need for the identification of more drugs that could be further investigated for efficiency in reducing viral pathogenicity by using in vitro studies. These drug searches can be improved by employing computational methods such as in silico screening and structure-based drug targeting methods. In this study, we searched for homologs of NSP-12 of SARS-CoV-2 and evaluated its binding affinity with the FDA-approved antivirals, i.e., Sofosbuvir and Dasabuvir, which specifically target the evolutionarily conserved site of NS5B of HCV. The initial search for a homologous protein identified a total of 1105 structures by employing the DALI server. Further screening of these structures based on the quality of structural alignment Z-score \\u2265 19 resulted in 611 structures that belonged to four viral families, i.e., Coronaviridae, Flaviviridae, Picornaviridae, and Caliciviridae, responsible for RdRp activity during viral replication. The NSP-12 of SARS-CoV-2 displayed significant homology to the NS5 of Flaviviridae or NS5B of HCV as compared to the polymerases of Picornaviridae and Caliciviridae. Based on the pairwise alignment of amino acid sequences, NSP-12 shares 96% identity with SARS-CoV-1, 11% with HCV and HNV, and 10% with Poli. Despite sharing a lower identity based on sequence, NSP-12 has seven highly conserved motifs in the RdRp domain that play a key role in viral replication. These motifs have been previously reported to play essential roles in the viral replication of several viral families. Motifs A\\u2013C have highly conserved residues while motifs D-G have conserved structural orientation at the active site of the RdRp domain. The active site of RdRp was also identified as a suitable site for drug targeting based on the highest druggability score evaluated by using the CavityPlus webserver. The conserved residues and residues predicted from CavityPlus that were participating in ligand binding showed significant overlap when compared. The overlap included the conserved residues of motifs A, B, C, D, and F of the RdRp domain of NSP-12. These conserved residues have been also described previously to involve interactions with nucleoside, RNA primer attachment, and the RdRp activity. The active site of the RdRp domain of NSP-12 was further evaluated for binding affinity with FDA-approved antivirals by the molecular docking method. Molecular docking was performed using AutoDock Vina which revealed a significant binding affinity of NSP-12 with Sofosbuvir and Dasabuvir as compared to Ribavirin. Despite that, NS5B showed higher binding affinity lower binding energy as compared to NSP-12 however, the differences were still insignificant. Further, MD simulations were carried out to determine the time dependent structural stability and intermolecular interactions. The profiling of NSP-12 and ligand interaction revealed firm binding due to the formation of hydrogen bonds and hydrophobic interactions. The participating conserved residues of motif-A, -B, -C, -D, -E, and -F of the RdRp domain were involved in these interactions with the ligands, i.e., Dasabuvir, Sofosbuvir, as well as with their native competitors, i.e., GTP and UTP. Similarly, in the case of HCV as a receptor, the residues of motif-A, -B, -C, and -F were involved in interactions with the above-mentioned ligands. In general, both Sofosbuvir and Dasabuvir showed higher binding to the active site of NS5B of HCV and NSP-12 of SARS-CoV-2. We retrieved NSP-12 of SARS-CoV-2 with PDB ID 6M71 at 2.9 A\\u030a resolution from the Protein Data Bank PDB repository that was recently submitted by Gao et al. For the identification of homologous structures in the PDB repository, 6M71 chain A was submitted to the DALI server website.fi/dali/ accessed on 16 March 2020. The DALI server identifies and ranks homologous protein structures based on the quality of structural alignment in the form of a Z-score. The Z-score is a quantitative expression for comparing the intramolecular similarity between two homologous protein structures and its value \\u22652 is suggested to be significant. In this study, we used a Z-score cutoff of \\u226519. To obtain a similarity matrix and a cladogram based on structural similarity, the same server was employed by using an all against all analysis for a total of 53 representative homologs. The heatmap was visualized by using the Pheatmap v1.0.12 package in R and the cladogram in FigTree v1.4.4 software. The selected homologous proteins responsible for RdRp activity were analyzed for conserved sequences motifs and secondary structure by employing the PROMALS3D server. The resulting alignments were exported into ESPript 3.0 server for visualization website.fr accessed on 5 April 2020. To further study the conserved residues at the active site of NSP-12, multiple structures were aligned by using the MatchMaker algorithm implemented in the University of California, San Francisco UCSF Chimera v.14 and visualized. For pharmacophore modeling or assessing druggable cavities in 6M71-A, the CavityPlus server website.php accessed on 12 April 2020 was employed by submitting the structure file. The CAVITY search module of the CavityPlus server was used to detect the ligand-binding cavity and druggability was predicted based on the energy of residues and site geometry with default parameters. Those residues that were reported to be evolutionarily conserved in the RdRp domain as well as identified from druggable cavity search were visualized in the Venn diagram using Venny 2.1\\u2014an online server for plotting Venn diagram. AutoDock Tools v1.4.5 was employed for the ligand and receptor preparation. In general, we followed the Single-docking experiment with AutoDock Vina method by following the guidelines as suggested by Forli et al. for ligand-receptor docking. Initially, the Canonical SMILES IDs of the ligand compounds, i.e., Sofosbuvir CID 45375808, Ribavirin CID 37542, and Dasabuvir CID 56640146 were retrieved from the PubChem library and then imported in Chimera v.14 for converting into PDB files. Subsequently, the structures were prepared by minimizing the free energy and addition of hydrogen atoms. On the other hand, the receptor structures were acquired from PDB and then processed in AutoDock Tools by removing water molecules, isolation of a single chain, addition of charges, merging non-polar hydrogens, and assigning appropriate atom types. AutoDock Vina v1.1.2 was employed for molecular docking between ligand molecules and receptor. A grid box of size 48 \\u00d7 44 \\u00d7 54 A\\u030a was selected and centered at the druggable cavity of 6M71-A oriented at 117.4, 115.3, 129.5 A\\u030a, while the NS5B of HCV PDB ID 1YUY-A was considered as a control with the same size of the grid box used at the origin of 0.6, 80.0, and 64.2 A\\u030a. The parameters for molecular docking were used as described for running the single-docking experiment. Furthermore, we chose the default scoring function implemented in AutoDock Vina. The resulting score of molecular docking of NSP-12 of SARS-CoV-2 and NS5B of HCV were compared and plotted using a bar plot. For depicting protein\\u2013ligand interaction, LigPlot + v2.1 was used for characterization and visualization. The MD simulation of the apo and ligand-bound complex of NSP-12 protein was done using the GROMACS v.2021 and GROMOS96 54a7 force field. The systems were solvated in a cubic box of TIP3P water model and neutralized with the addition of counter ions. The neutralized systems were minimized using the steepest descent algorithm with a maximum force of 1000 KJ mol\\u22121 nm\\u22121 to adjust the geometry. To obtain further convergence, the minimized systems were equilibrated for in an NVT ensemble followed by NPT ensemble. Finally, a 100ns production run was executed, with coordinates and energies being saved in the output trajectory file at 10ps intervals. The RMSD, RMSF, radius of gyration Rg, and hydrogen bond contacts were used to determine the degree of convergence for all of the simulated systems. The molecular mechanic/Poisson\\u2013Boltzmann surface area MM-PBSA methods were used to calculate of binding free energy NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes. This calculation validates the thorough ligand\\u2013protein interaction investigation. The binding energy constitutes potential energy, polar and non-polar solvation energies, and they were calculated by the \\u2018g mmpbsa\\u2019 packages using GROMACS. The binding energy calculations in this method were calculated using the following equation The G binding represents the total binding energy of the complex, while the binding energy of the free receptor is the G receptor, and that of unbounded ligand is represented by the G ligand. Structural homology-based drug repurposing is a significant approach to exploring the homologous proteins which have been already targeted by either approved or investigational drugs. In this study, we searched for structural homologs of NSP-12 of SARS-CoV-2 and identified their relationship with the members of three viral families, i.e., Flaviviridae, Picornaviridae, and Caliciviridae. The RdRp domain of NSP-12 is highly conserved in terms of both sequence and architecture and assessed for binding affinity with FDA-approved antivirals, i.e., Dasabuvir and Sofosbuvir. These antivirals were initially approved for targeting the NS5B of HCV and serve as competitive inhibitors for binding to the active site that results in reducing viral pathogenesis. The results of this study revealed that the six conserved motifs motif A-F were involved in the formation of hydrogen bonds and hydrophobic interactions with the ligands. However, there is still a further need for in vitro analysis of these drugs to assess their efficiency before entering into clinical trials.\",\n          \"Tight and selective interaction between ligands and target proteins is of cardinal importance in drug discovery that various stages of identification and validation of targets and potential drug leads, as well as manifold preclinical and clinical trials, should be carried out until obtaining the final approval from the Food and Drug Administration FDA Abel et al., Campillos et al., Olayan et al. Accurate prediction of drug\\u2013target interactions DTIs plays a vital role for in silico and modern drug discovery in developing new drug candidates Olayan et al. Unfortunately, despite the promising advancements in genomics, proteomics, and systems biology, formidable scientific and regulatory obstacles such as high attrition rates, excessively time\\u2010consuming, and high\\u2010priced procedures have taken a heavy toll on proposing effective biologically active agents, and the success of drugs to pass clinical trials stage is about only 13% Fotis et al., Liu et al., Murray & Rees. Such attempts enable scientists to design energy harvesting materials and systems for the sake of higher efficiency and enlarging practical window Massetti et al., Safaei et al. Accordingly, computer\\u2010aided drug discovery CADD has revolutionized the history of drug discovery, especially regarding its substantial advantages Schneider & Fechner. These favors are a giving more information about the beneficial interaction between the drug and target and the availability of three\\u2010dimensional 3D structure, b providing a cost\\u2010efficient way to reduce failures for high\\u2010throughput screening HTS, c producing new ideas for rationally drug design, and d leading to rationally anticipate the targeted protein and candidate hits Bajorath, Bleicher et al., Ou\\u2010Yang et al., Sliwoski et al. These strategies are currently classified into three major categories ligand\\u2010based, structure\\u2010based, and chemogenomic approaches Imam & Gilani, as summarized in Figure. Most importantly, recent studies have illuminated the potential roles of CADD in designing new drug candidates for COVID\\u201019 treatment. Coronaviruses CoVs are cross\\u2010species viruses and can spread expeditiously into new host species, which in turn cause epidemic diseases. The whole members of the CoV family have a similar genome sequence, and the severe acute respiratory syndrome CoV SARS\\u2010CoV and Middle East respiratory syndrome CoV MERS\\u2010CoV are considered the two prototypical well\\u2010known CoVs Aghamirza Moghim Aliabadi et al., Seidi et al., Yin & Wunderink. The new CoV, identified as COVID\\u201019 and caused by SARS\\u2010CoV\\u20102, has a very similar genome sequence to SARS\\u2010CoV C. Wang et al. The lack of effective drugs led to considerable unwholesomeness and death rate of SARS\\u2010CoV\\u20102, especially in 2020 and 2021, when a novel coronavirus strain brought about a fatal respiratory illness. Since the outgrowth of the SARS\\u2010CoV\\u20102, more than 528 million people have been infected with up to 6.3 million deaths, as reported by the WHO. Although effective vaccines have amazingly reduced the COVID\\u201019 cases recently, they will not change the world right away, and importantly, their permanent efficacy, short\\u2010term and long\\u2010term side effects, and long\\u2010term protection against different variants of SARS\\u2010CoV\\u20102 are still under debate. These issues are accompanied by mass vaccination campaigns Wen et al. Furthermore, several studies have reported that vaccinated people can also be infected with novel coronavirus variants Largent & Miller. Therefore, the design and introduction of new drugs, either the current drugs or synthesized agents, for eradicating the SARS\\u2010CoV\\u20102 is of extreme concern. To accelerate this process, repositioning broad\\u2010spectrum antiviral drugs can be a potential strategy concerning the presence of the pharmacokinetic, pharmacodynamic, and toxicity data of these drugs Serafin et al. For instance, the efficacy of some antiviral drugs with previously known treatment of hepatitis, HIV, influenza, and Ebola viral diseases has been investigated. These agents, including adefovir, foscarnet, tenofovir, sofosbuvir, and uprifosbuvir, have phosphoramide and phosphate with various azole rings in their structures. By assessing their effectiveness against COVID\\u201019, remdesivir was the one that grabbed much attention and is believed to be the most successful drug for the therapy for combating COVID\\u201019 Brown et al., Gholivand et al. Further potential macromolecular targets of SARS\\u2010CoV\\u20102 are depicted in Figure. Therefore, finding the drugs that can target this enzyme may decrease the possibility of mutation\\u2010mediated drug resistance and show significant antiviral activity Pachetti et al. Thus, it is likely that CADD might be able to provide us with effective small molecules targeting the main receptors of SARS\\u2010CoV\\u20102 with fully therapeutical and preventional\\u2010oriented outlooks Muratov et al. The more recent update of FDA approval for COVID\\u201019 drug discovery shows that authorization of remdesivir inhibits the RNA\\u2010dependent RNA polymerase RdRp of the virus, molnupiravir inhibition of viral reproduction through enhancing the vast mutations in the replication of viral RNA and paxlovid 3c\\u2010like protease inhibitor for the treatment of mild\\u2010to\\u2010moderate COVID\\u201019. These drugs are used for various viral diseases such as hepatitis C, Ebola virus, influenza, etc. Vangeel et al. Molnupiravir and remdesivir acted on the RdRp enzyme employed by the virus to increase the transcription and replication capability of the viral RNA genome. While some groups reported the promising results of remdesivir, others showed a lack of efficacy, and therefore, the WHO did not suggest using it Mahase. Our focus in this paper is to explore the role of CADD on COVID\\u201019 most specifically, chemogenomics and DR, with potential examples of COVID\\u201019 drug discovery. Unfortunately, with boundless energy and time to discover a successful small molecule by utilizing various databases to inhibit the main protease Mpro of SARS\\u2010CoV\\u20102 via several virtual screening VS approaches, there are no effective drugs that have been approved. Only safety protocols such as wearing face masks, face shields, social distancing, and washing hands frequently are the best approaches to minimize the illness's danger and prevent transmission. Basically, the process of bringing an FDA\\u2010approved medicine to market for COVID\\u201019 drug discovery is extremely challenging, starting from the initial research of scientists for target identification and lead compound generation. After lead discovery and development of drug\\u2010like candidates and investigation of in vitro and in vivo results, the process will go into three phases, i.e., preclinical and, if successful, into clinical studies, and ultimately a marketed medicine after FDA approval Schaduangrat et al. It is where CADD is evolved as a valuable and powerful method to aid the challenging tasks of identifying small bioactive molecules and predicting targets. In 1981, an influential article appeared in Fortune magazine titled Next Industrial Revolution Designing Drugs by Computer at Merck. This article led to a realization of the importance of in silico studies in drug discovery and manifestation of CADD Sliwoski et al. Since then, high\\u2010throughput screening HTS has been increasingly used in the pharmaceutical sphere and academic institutions to the rapid discovery of hit and lead compounds. This potential and automated approach enables us to screen the entire compound library of either number of chemicals or biological compounds in a target\\u2010based assay or even a cell\\u2010based assay to obtain the desired biological response Bittker & Ross, Maghsoudi et al. It can spring to the mind that the stringent requirement of large resources of targets and ligands in HTS makes it expensive and time\\u2010consuming, and we should overlook the use of the useless ligands having no probability of showing success. But, the CADD technologies are very powerful and able to address this issue easily by employing virtual high\\u2010throughput screening vHTS that uses virtual compound libraries and allows experimentalists to focus on ligands which are more likely to have any activity of interest Leelananda & Lindert. In the first days of COVID\\u201019 outbreak, vHTS was employed and developed efficiently to investigate a group of approved drugs to identify the possible anti\\u2010SARS\\u2010CoV\\u20102 agents at the National Center for Advancing Translational Sciences NCATS of the National Institutes of Health NIH, embracing a broad spectrum of the SARS\\u2010CoV\\u20102 life cycle, such as viral entry, viral replication, in vitro infectivity, and live virus infectivity Xu et al. Many possible drug candidates were selected for targeting various stages of the SARS\\u2010CoV\\u20102 life cycle. For instance, cepharanthine, an anti\\u2010inflammatory compound, effectively showed ability to rescue the cytopathic effects CPEs of SARS\\u2010CoV\\u20102 Rogosnitzky et al., or corilagin, a possible antifibrosis, revealed potential against the angiotensin\\u2010converting enzyme ACE2 receptor\\u2010binding domain RBD Yang et al. Therefore, it is urgent to build a high bio\\u2010safety SARS\\u2010CoV\\u20102 infection model with a high ability for HTS of drugs against SARS\\u2010CoV\\u20102. This strategy just relies on ligand information\\u2014one of the broadly known and widely used ligand\\u2010based approaches in QSAR modeling. The advantages accruing from the QSAR methods in drug discovery and development are noteworthy. This computational technique attempts to construct mathematical models of compounds' physical and chemical properties associated with their chemical structure. QSAR studies are extensively employed to predict pharmacokinetic properties such as ADME and toxicity by building a training set of measured properties of known compounds and making a mathematical model of the encoded chemical structure, and also is useful in sensory applications as well Alam & Khan, Farshchi et al., Kholafazad\\u2010Kordasht et al., Nosrati et al., Seidi et al. Generating molecular descriptors from compound structures is also significant in assessing compounds' different physical, chemical, and topological features. After computing molecular descriptors, machine learning techniques play indispensable roles in the quality of QSAR predictions Popelier & Smith, Rad & Maghsoudi, Sliwoski et al. Using machine learning and QSAR, Muratov et al. identified 32 drugs and their 73 selected binary combinations for testing against SARS\\u2010CoV\\u20102. They found that the combinations of four of these drugs, including nitazoxanide, remdesivir, amodiaquine, and umifenovir, could synergize against SARS\\u2010CoV\\u20102. Interestingly, the combination of nitazoxanide and remdesivir showed a great synergistic interaction, and the combination of remdesivir and hydroxychloroquine exhibited a powerful antagonist role against SARS\\u2010CoV\\u20102 Bobrowski et al. The two proteins pp1a and pp1ab are quickly translated after entry into the host cells and further separated by two viral proteases, 3 C\\u2010like proteases 3CLpro and papain\\u2010like protease PLpro Amin et al. The latter is a critical virus enzyme responsible for producing a functional replicase complex and stimulating viral spread Shin et al. Amin et al. used chemical\\u2010informatics approaches to build a Monte Carlo optimization based on a QSAR data of various PLpro inhibitors, b VS to choose the best in\\u2010house molecules among 67 with the high likelihood of targeting against PLpro, and c eventually verification of these inhibitors by using receptor\\u2013ligand interaction investigation. Out of 67 compounds, 56 were anticipated as active from a QSAR model. Afterward, these 56 similar compounds were screened through SwissADME, and eventually, 13 compounds were approved for the ADME criteria. Furthermore, molecular docking has proved the potentiality of these candidates against SARS\\u2010CoV\\u20102 PLpro. In an interesting work, Tropsha et al. developed open\\u2010source data QSAR models of 42 inhibitors from DrugBank by employing QSAR models, docking, and similarity searching and utilized these models for virtual screening of all drugs in the DrugBank database. The 42 virtual hits were finalized for availability and cost using in\\u2010house ZINC Express software. The National Center for Advancing Translational Sciences NCATS analyzed 11 of total 42 hits which were potential compounds, and three of them, cenicriviroc AC50 of 8.9 M, proglumetacin tested twice independently, with AC50 of 8.9 M and 12.5 M, and sufugolix AC50 12.6 M. While molecular docking was unsuccessful to accurately distinguish between experimentally active and inactive compounds, QSAR was able to identify three inhibitors of SARS\\u2010CoV\\u20102 replication through Mpro inhibition. Overall, these results show that QSAR models proposed using SARS\\u2010CoV\\u20102 Mpro data can be useful to introduce compounds active against SARS\\u2010CoV\\u20102 Alves et al. This strategy relies more on target information. The structure\\u2010based drug design, also called target\\u2010based drug design, rests on the assumption that the bioactivity of a ligand comes from the physical interaction of the ligand with the corresponding target protein. The prerequisite of this approach is the 3D structure of the biological target, obtained from crystal structures, NMR data, or molecular modeling. Molecular docking is an established structure\\u2010based approach frequently used to predict the binding affinity and analyze the interactive mode between a ligand and a target/receptor. In other words, docking enables us to identify novel drug candidates by predicting ligand\\u2013target interactions at a molecular level Pinzi & Rastelli, Sargazi et al. There are several references in which the significant role and applications of docking in drug design and development have been discussed De Vivo & Cavalli, Keretsu et al., Kitchen et al., Meng et al. Furthermore, in the current spread of SARS\\u2010CoV\\u20102, molecular docking is important in screening potential drugs Keretsu et al., Lokhande et al., Marinho et al., Yu et al. The recent work of Khelfaoui et al. could be a good example of using molecular docking combined with MD to find promising candidate drugs against COVID\\u201019. They selected a library of 18 drugs that three of which include ramipril, delapril, and lisinopril, bound to angiotensin\\u2010converting enzyme 2 ACE2 receptor and SARS\\u2010CoV\\u20102/ACE2 complex well. ACE2 is one of the receptors in the human body with which the S proteins from coronaviruses have a great binding affinity and serve as the entry point into cells for SARS\\u2010CoV\\u20102. In fact, due to the long time required to discover specific antiviral agents or a vaccine, the best strategy would be to repurpose existing drugs to treat COVID\\u201019, which will be discussed in the DR section. Using VS and molecular docking, Abdellatiif et al. identified some promising antimicrobial and chemotherapeutic agents having coumarin and pyridine derivatives in their structures. In order to identify the most potential drug\\u2013target interactions, a molecular operating environment MOE program was carried out against the human ACE2, SARS\\u2010CoV\\u2010main peptidase, serine protease hepsin, human coronavirus papain\\u2010like proteases, and SARS\\u2010Coronavirus NSP12. As shown in Figure, among compounds a, b, c, d, and e with binding score energy values of \\u22126.08, \\u22126.48, \\u22126.78, \\u22127.08, and \\u22126.63, respectively, compound d had the binding score energy against the aforementioned targets. Thus, compound d needs to be additionally evaluated in clinical trials as a candidate for potential inhibitor of the SARS\\u2010CoV\\u20102 Abdellatiif et al. Hosseini et al. evaluated a virtual screening using docking of 1615 FDA\\u2010approved drugs to identify novel potential inhibitors for protease protein of COVID\\u201019. They showed that among all investigated FDA\\u2010approved drugs, simeprevir, a Hepatitis C virus HCV NS3/4A protease inhibitor, manifested high affinity to protease binding pocket and placed effectively into the binding pocket more beneficially than the lopinavir\\u2010ritonavir. This study suggested this compound as a potential inhibitor of COVID\\u201019 Hosseini & Amanlou. In another docking study, Motiwale et al. worked to identify potential inhibitors against Mpro of SARS\\u2010CoV\\u20102 from previously reported SARS\\u20103CL protease inhibitors where among 61 previously reported potential inhibitors, 4\\u2010benzoic acid and 4\\u2010 4\\u2010methoxyphenyl \\u20106\\u2010oxo\\u20102\\u2010 2\\u2010phenylethyl sulfanyl \\u20101,6\\u2010dihydropyrimidine\\u20105\\u2010carbonitrile revealed a minimum and maximum binding energy, respectively Motiwale et al. Table presents a selection of recent studies that researched the docking strategy of different pharmacological classes as treatments of SARS\\u2010Co and SARS\\u2010CoV\\u20102. Although many studies are carried out and many drugs are being tested to open a new road for effective treatment of COVID\\u201019, there are no fully beneficial antiviral drugs against the disease. Many of these studies are now in clinical trials to assess their safety and efficacy against infected patients. Chemical genetics, chemical biology, and chemogenomics chemical genomics are some cutting\\u2010edge applied fields in target and drug discovery Rognan, Ura\\u0301n Landaburu et al. Among the presented strategies, chemogenomics is an emerging discipline that principally investigates all molecules that can interact with any biological target. In humans, the number of the gene is about 26000, whereas the amount of various proteins is approximately more than 1 million Andersson et al. However, due to the extensive number of available molecules and biological targets, chemogenomics is limited to screening congeneric compound libraries against a selected target family. This alternative proposes an unprecedented chance to seek drug\\u2010like compounds with a noteworthy target or significant subtype Jones & Bunnage, Kubinyi. Chemogenomics is different from the traditional pharmaceutical methods due to linking the widespread software library directly to the methodology of modern genomics. In order to achieve this purpose, chemogenomics, along with a contribution of several chemical disciplines, including synthetic chemistry, combinatorial chemistry, cheminformatics, bioinformatics, and genomics, screen synthetic compound libraries for their impact on biological targets Bredel & Jacoby, Quinlan & Brennan. The ultimate goal of chemogenomics is the rapid screening as well as the rational development of target\\u2010specific chemical ligands to decrease the cost and enhance the drug discovery period, especially in the case of COVID\\u201019 drug investigation Jones & Bunnage. Basically, the main role of chemogenomics is unifying the discovery of ligands and targets by taking advantage of active compounds as probes to identify proteome functions. Similar to genetics, and if the operation of investigation arises from the target to phenotype, or vice\\u2010versa, there are two concepts a forward and b reverse. In \\u2018forward chemogenomics\\u2019, the ligand is discovered through their phenotypic condition that could affect the entire biological system instead of the base of their mechanism of suppression of a certain desired protein. In other words, the molecular basis of the phenotype interest is unknown and phenotypic screening is carried out in cells of organisms having a single\\u2010cell or multicellular utilizing a group of ligands. On the other hand, in reverse chemogenomics, the desired sequence of genes is cloned and explicit at the first step as target proteins afterward, they are preselected in a high\\u2010throughput screening HTS \\u2018target\\u2010based\\u2019 way by a panel of compounds. Here, the assays are normally classified as i organismal assays, i cell\\u2010free, and ii cell\\u2010based organismal assays. Cell\\u2010free assays are generally straightforward but precise and extremely automated. They are comprehensive binding trials in which manifold compounds are examined concurrently for their binding affinity toward a variety of particular goals. Here, DTIs are clearly recognized in the absence of confounding variables. However, in cell\\u2010based and organismal assays, chosen compounds are delivered right to cells or organisms in vitro to select hits within a relevant cellular context. However, the hits need further mechanistic characterization due to simultaneously interacting with multiple targets. The former approach is usually employed in reverse chemogenomics, whether target information is at hand, while organismal assays and cell\\u2010based are mainly employed in forwarding chemogenomics to test a wide variety of compound effects on whole biological systems Bredel & Jacoby. In forward chemogenomics, cells or organisms are distributed in multi\\u2010well cell culture plates for incubation. Subsequently to the incubation step, an aliquot from donor plates is taken to the recipient and mock plates mock treatment, in which the ligand\\u2013target binding assay is performed. The endpoint of most cell\\u2010based HTS assays is a spectrophotometric measurement in a spectrophotometer equipped with suitable software that calculates the final data, including SAR analysis and in silico QC. Then, active compounds that obtain the intended changes in the phenotype are chosen to elucidate their targets at the molecular level. This strategy could be performed using affinity chromatography, microarray, phage display, or transcriptional or proteomic profiling. In reverse chemogenomics, the main focus is to elucidate the protein families and the gene. In this regard, the sequencing of the target gene demonstrates a specific level of homology that is transfected in a host cell. These family targets are sequentially gathered, purified, and subjected to trial design. Finally, similar to forwarding chemogenomics, the measurement data on the spectrophotometer are directed to a suitable data analysis system. The biological target is extracted from the existent information in the biologically annotated chemical database in the target fishing approach Huang et al. However this procedure needs an abundant effort from a time, cost, and resource perspective to identify all plausible chemical target associations thus, computational target prediction techniques are strongly being pursued with the contribution of advanced data\\u2010mining algorithms Anwar et al. This current mode of action or the computational target approach is classified into four classes data\\u2010mining, pharmacophore searching analogy searching, analysis of bioactivity spectra, and molecular docking. In the past, these methods were applied to investigate new compounds of known targets, but recently, they have been applied to predict novel targets for the known compounds via similar approaches Campbell & Marchant, Ekins et al. Target\\u2010based or receptor\\u2010based chemogenomics refers to the discovered biologically active compounds with phenotypically readout information. Nevertheless, no existing target data thus, by applying the chemogenomics strategy, the hypotheses can be generated of the target by comparing the new biological compounds to the same known target Fourches et al., Weston et al. For example, the target receptors or proteins that are necessary for replication and survival of SARS\\u2010CoV\\u20102 are viral proteases involving papain\\u2010like protease PLpro, the 3\\u2010chymotrypsin\\u2010like protease 3CLpro, and Mpro, RNA\\u2010dependent RNA polymerase RdRp complex, the spike S receptor binding glycoprotein, the nucleocapsid N protein, nonstructural protein NSP, and the membrane M protein including transmembrane TM as well as small envelope E protein Cavasotto et al. As an example of this part, it can be referred to the work of Mishra and his coworkers that reported an inclusive computational approach for the identification of drug molecules that are multi\\u2010targeted struggling with the SARS\\u2010CoV\\u20102 proteins. These drugs are essentially included in virus replication inside the host body, the viral\\u2013host interaction, disease development, and transmission of Sars\\u2010CoV\\u20102 infection. They screened 72 potential antiviral drugs approved by the FDA against the S glycoprotein, hACE2, 3CLpro, Cathepsin L, N protein, RdRp, and NSP6 proteins. Furthermore, they investigated the molecular interaction by applying free energy landscape, simulation of Molecular Dynamics, and binding free energy estimation using MM\\u2010PBSA. Their finding showed favorable results for seven drugs, among which catechin flavan\\u20103\\u2010ol can effectively bind to 3CLpro, Cathepsin L, RBD of S protein, NSP\\u20106, and Nucleocapsid protein to be used as a potential multi\\u2010targeted agent in combating SARS\\u2010CoV\\u20102. However, due to the limited timeframe of combating this pandemic, it is not realistic to be assured of finding a definitive therapy for combating the pandemic. Hence, vigorous attempts are still continuing for the DR as a clinical treatment strategy to fight COVID\\u201019 Mishra et al. In another research, Omotuyi et al. investigated molecular docking protocols for fast screening of the FDA database for high\\u2010affinity interaction of the resolved 3D structure of SARS\\u2010CoV\\u20102 receptor\\u2010binding domain RBD in complex with its receptor hACE\\u20102 interface. Ubrogepant, as one of the preferable candidate drugs, was studied utilizing the atomistic molecular dynamics simulation method. They reported that UBR breaks the interaction at the RBD/ACE\\u20102 interface residues of SARS\\u2010CoV\\u20102 RBD/ACE\\u20102 complex that might result in protein function losses with direct inference on the oligomerization in RBD and function loss in ACE\\u20102. Therefore, leading to a binding and cellular receptor recognition unattainable. Therefore, UBR exhibits a novel therapeutic candidate in combating SARS\\u2010CoV\\u20102 since it binds to proportionally high affinity with free RBD, ACE\\u20102 receptor, and SARS\\u2010CoV\\u20102 RBD/ACE\\u20102 complex based on only binding affinity calculations Omotuyi et al. The other studies in this area are demonstrated in Table. A primary example of the beneficial application of chemogenomics is DR, which accelerates drug discovery by identifying novel clinical targets for the available approved drugs. This method possesses the distinctive merit of rapid clinical development without bearing the long\\u2010lasting and expensive procedures included in drug discovery. Therefore, according to the advantages of these proposed drug discovery methods, many COVID research orientations have been focused on DR Galindez et al., R. Liu et al. Computational DR applies available drugs for new or existing diseases that do not have an absolute cure, while the available drugs were not originally designed for the novel disease treatment. Furthermore, utilizing the enormous volumes of available omics information in digital form in conjunction with in silico screening intrigued the researcher to use this appropriate ground to significantly increase the speed of shortlisting promising candidates in reply to spreading diseases like SARS\\u2010CoV\\u20102. To be exemplified, Lucchetta, M. and M. Pellegrini reported a methodology for preclinical computational DR called DrugMerge, and it is based on combining several drug rankings gained via a set of Disease Active Subnetwork construction algorithms. DrugMerge utilizes differential transcriptomic data from drug perturbation assays and cell lines/tissues of patients affected by the illnesses on the subject of a considerable gene co\\u2010expression network. Application of DrugMerge to COVID\\u201019 shows that DrugMerge can mimic human expert judgment Lucchetta & Pellegrini. Rima Hajjo and Alexander Tropsha developed a systems biology workflow for drug and vaccine repurposing. They reported that the identification of Bacillus Calmette\\u2013Gue\\u0301rin BCG and small molecules of BGC mimics involving antiviral drugs raltegravir, emetine, and lopinavir as high confidence hits are capable of boosting immunity and preventing emerging new viruses. Their results approved that BGC impacted the generation and maturation of nai\\u0308ve T cells, resulting in increasing long\\u2010lasting trained intrinsic immune replies to combat novel viruses, including SARS\\u2010CoV\\u20102 in vitro as a protective measurement from the lethal consequences Onozuka et al. Due to the lengthy development cycle of novel drugs, DR turns into an efficient approach to screening drugs of potential therapeutic use for SARS\\u2010CoV\\u20102. Che et al., in another attempt, screened potential therapeutic drugs for COVID\\u201019 based on a knowledge\\u2010graph\\u2010based DR method. In their experiment, five drugs predicted by the models exhibited efficiency in clinical treatment. The experimental outputs prove that the model can predict drug\\u2013disease interaction effectively for normal diseases and SARS\\u2010CoV\\u20102 Lv et al. Therefore, DR presents a constant source of novel knowledge in virus biology and molecules with previously undefined antiviral properties, which can be applied as molecular tools in unknown molecular mechanisms of virus replication and pathogenesis Mercorelli et al. Further investigation of SARS\\u2010CoV\\u20102 in related research is summarized in Table. A fundamental role of chemogenomics is predicting the activity profile of the compounds for a group of targets that can be implemented by using ligand\\u2010based or target\\u2010based approaches. The ligand\\u2010based philosophy is underlying that molecules sharing enough similarity to available biologically defined ligands have an increased probability of sharing the same biological profile. Therefore, the efforts are devoted to organizing chemical libraries with biological information, including in vitro affinity, targets, and toxicology Harris & Stevens. As a result, designing chemical libraries is of great interest for a certain individual target or target family, which is attempted via utilizing machine\\u2010learning methods. Herein, concentration is on protein\\u2013ligand interaction via applying physics\\u2010based protein\\u2010ligand docking on machine learning. Computational chemogenomics or quantitative multiple structure\\u2013activity relationships QMSPR modeling consists of the compound\\u2013protein interaction space modeling of large datasets at once, usually for drug discoveries, where the available techniques chiefly incorporate escalating numbers of bioactive samples or concentrate on individual subfamilies of proteins and ligands Cheng et al., Xue et al. However, the 3D information of protein targets is needed most of the time. Thus, the application of precise target data in chemogenomics is confined to the explicitly known X\\u2010ray structures or structures that homology models can evaluate. As a result, many QMSAR is centered on ligand\\u2010based studies or similarity principles Gawriljuk et al. In this context, compounds of phenotype\\u2010based preselection possess merits in discovering target\\u2010based drugs due to unclimbable and missing perceptions of the mechanism of action of the drugs. Therefore, a handful of methods can carry out new chemical compound screening which is phenotype\\u2010based. In this regard, Thai\\u2010Hoang Pham et al. extended a sound learning framework for a high\\u2010throughput mechanism\\u2010driven neural network\\u2010based method for phenotype compound screening. They applied this framework for COVID\\u201019 DR. Their proposed novel data escalating technique extracts beneficial information from untrustful assays in the dataset of L1000. Furthermore, they showed the value of DeepCE by applying it to the DR of COVID\\u201019 and produced novel lead compounds that represented clinical evidence. Therefore, DeepCE can provide a potentially strong framework for rapid predictive modeling via using noisy omics data and screening new chemicals to modulate an organized response to illnesses Wang et al. A computational approach for increasing the rate of development of drugs and comprehension of the mechanism of action of small molecules is the identification of drug\\u2013target interaction DTIs. Lee and his coworkers represented an AI\\u2010DTI, a new method for predicting activation and inhibition of DTIs, by combining the mol2vec and genetically perturbed transcriptomes. Their finding showed that this methodology was a success in discovering about half of the DTIs for drugs that are employed for treating COVID\\u201019 Lee et al. A myriad of compounds represent potential drug candidates, but merely a few of them will be formulated, approved, and registered as a drug for individual health issues which investigation for that compound will be a long, intricate, and costly process, while sometimes there is an urgent need for developing a drug like COVID\\u201019 pandemic Muratov et al. Table demonstrates chemogenomics related to COVID\\u201019 research. To conclude, chemogenomics is an emerging interdisciplinary field at the interface of chemistry, biology, and informatics which decreases the cost and time of drug discovery dramatically via three main approaches, including target fishing, predicting the bio\\u2010active profile of drugs, and DR. Herein, of much current importance is chemogenomics, is in COVID\\u201019 drug discoveries, which this review paper highlighted specifically. Drug repositioning, also known as drug repurposing, drug redirecting, drug reprofiling, and therapeutic switching, is defined as identifying new indications and therapeutic targets for existing drug compounds Lotfi Shahreza et al. DR has received significant attention caused of its crucial role in finding and developing new therapies for complex, orphan, and chronic diseases with unknown or less effective treatments Shameer et al. Since developing a new drug in the traditional drug discovery process is time\\u2010consuming and takes approximately between 10 to 15 years to enter the market, repurposing existing drug compounds is capable of overcoming these shortcomings, and this type of research has been growing drastically Kumar et al. Therefore, DR plays a versatile role in developing new drugs because it can reduce the costs of the whole process and shorten the development pathways of approved or investigational drugs Pushpakom et al. DR works with two main approaches. The first is that a drug can bind to various targets simultaneously, which is good for searching novel target sites of action for the known biological agents, and the second is that targets related to a disease are often associated with different biological processes of pathogenesis, which is important for choosing of a new indication for the known target Badary, Kamel et al. These are indispensable in proposing new drug candidates for unprecedented diseases like COVID\\u201019. It has emerged as an alternative way of accelerating the process of developing new treatments against COVID\\u201019 Yu et al. Based on data brought from PubMed, the number of works of literature on DR has been growing dramatically, and this number has increased since the outbreak of SARS\\u2010CoV\\u20102 Galindez et al. It should be noted that polypharmacolgy and DR are inextricably interrelated with each other Cheng. DR has become an influential tactic from both academia and pharmaceutical industries as an alternative strategy to traditional drug discovery that is related to when new therapeutic applications are pinpointed for existing drugs Karakida et al., Turanli et al. Compared to traditional de novo drug discovery and development, this approach provides a new and fast way for drug development and reduces risk, particularly for drugs that cannot pass phase I clinical trials. It is able to reprocess the information on known drugs such as pharmacokinetics and safety for humans and manufacturing pathways, resulting in reduced time and expenses in traditional de novo drug discovery Sawada et al. There are a wide variety of examples of repositioned drugs. Sildenafil is considered the most prominent one, beginning with cardiovascular treatment followed by accidentally treating erectile dysfunction clinical trial and later in low dosages, treating pulmonary hypertension Ghofrani et al., Prasad et al. Furthermore, as repurposed drugs pass the preliminary clinical trials phase I or safety level, the time needed for drugs to reach the market can be reduced significantly. Hence, scientists can identify the unintended secondary targets through CADD. The history of discovering the effect of an established drug on new indications dates back to World War II, since considerable improvement had been observed for cutaneous rashes and arthritis in soldiers who were under treatment with an antimalarial medicine, Chloroquine Schlitzer. Chloroquine is a well\\u2010known antimalarial drug that was synthesized in 1934 and later on demonstrated successful various therapeutic effects, including anti\\u2010inflammatory/immunomodulating, anti\\u2010infective, antithrombotic, and metabolic. Furthermore, antitumoral properties make it available for the treatment of a number of cancers Plantone & Koudriavtseva. In 1942, Marcel Jonbon used sulfonamide antibiotics for typhoid treatment, but he faced hypoglycemia as an unexpected side effect in some patients therefore, sulfonamides presented the role of an anti\\u2010diabetic drug Quianzon & Cheikh. Aspirin is one of the most famous DR examples with an original indication as an analgesic drug that for the first time repositioned as an antiplatelet aggregation drug in 1980 and a second indication still is commonly used for cardiovascular problems Jourdan et al. Furthermore, several types of research have been conducted on the anticancer effect of aspirin such as colorectal cancer Drew et al., lung, and breast cancer Li, Hu, et al. The notable potential and extensive applications of DR in novel therapies are entirely apparent and accordingly, the number of acceptable research studies which have been done or are under processing are growing every day. There are various classifications for DR methods which typically can be categorized into computational approaches and experimental approaches Park. Experimental approaches refer to testing drugs in evaluations based on available comprehensive clinical compound databases in which the possible indications of drugs can be discovered by target\\u2010based or cell\\u2010based analysis Luo et al. There are some hurdles for experimental identifications because they are mostly expensive and time\\u2010consuming, requiring a set of marketed drugs, technical equipment, and also screen tests. In contrast to experimental methods, computational drug repositioning methods which are mainly data\\u2010driven utilizes databases and bioinformatics tools to find out interactions between drugs, targets, and diseases Pillaiyar et al. Analyzing a large type of data like chemical structures, gene expression, genotype or proteomic data, and electronic health records EHRs provides the possibility of understanding novel and unexpected functional gene interactions and unclear drug response or mechanisms of disease Pushpakom et al. Since there is no specific and registered therapy to cure the infection of this virus, finding effective therapeutics by applying existing drugs is a pressing exigency and established strategy Galindez et al. Concerning the time\\u2010consuming regulatory process of the newly approved drugs, the DR strategy strongly utilizes the identification of new treatment methods by presenting FDA\\u2010approved drugs. In addition, many repurposed drugs with acceptable results are accomplished by computational methods Wang & Guan. There are myriad reasons why we need to go through DR for different diseases like COVID\\u201019 if we intend to accelerate the process of developing a new drug. These reasons can be i availability of marketed drugs and their excipients and APIs for formulation and distribution, ii the likelihood of taking advantage of a synergistic effect of various drugs with each other, iii cheap and less time\\u2010consuming process for drugs to be marketed and IV emergence of new therapeutic pathways for former drugs and new categories of medicine Chung et al. Therefore, repositioned drugs have very important impacts on COVID\\u201019 drug discovery. Therefore, different categories of drugs are being repositioned in in vivo and in vitro studies to treat SARS\\u2010CoV\\u20102 Wu et al. These potential agents are namely Ebola virus remdesivir, antimalarial chloroquine, anthelmintic and antiprotozoal drugs niclosamide, diarrhea nitazoxanide, antiviral lopinavir and ritonavir, antineoplastic carfilzomib and bortezomib, and antibacterial azithromycin. In this regard, the WHO supported repositioning remdesivir, lopinavir/ritonavir, and chloroquine/hydroxychloroquine as a potential for COVID\\u201019, although most might be only supportive symptomatic Won & Lee. Nevertheless, they will likely be approved for COVID\\u201019 therapy if they pass clinical trials and are published in literature and guidelines from leading health organizations Nain et al. In order to do this, many groups have been attempting enthusiastically to discover the first approved drug for COVID\\u201019 therapy Erlanson, Low et al., Muratov et al. As a result, we identified several clinical trials, in which 18 studies were at clinical phases 2, 3, or 4. Luo et al. investigated the free energy perturbation\\u2010absolute binding free energy FEP\\u2010ABFE method and bioassay validation to reposition 25 potential therapeutic agents to inhibit Mpro. By using the viral proteinase crystal structure, Mpro PDB ID 6 LU7, up to 2500 FDA\\u2010approved drugs were first screened via Glide's molecular docking program. They found that Cys145\\u2010His41/Ser144\\u2010His163 could be the nucleophilic agent and acid responsible for hydrolysis of the substrate proteins, and Gly143 and Gln166 can form a hydrogen bond with the CO\\u2010NH\\u2010C \\u2010CO\\u2010NH\\u2010C structure of the backbone of the substrate protein. Of 25 possible agents, 15 showed strong inhibitory properties against Mpro. Among these agents, additionally, dipyridamole passed its first\\u2010round clinical trials with considerable potencies Li, Hu, et al., Li, Li, et al. The network\\u2010based methods have also given satisfying results in investigating the DTIs and CoV\\u2013host interactions. Zho et al. introduced 16 possible anti\\u2010HCoV drugs such as melatonin, mercaptopurine, toremifene, mesalazine, and sirolimus from more than 2000 FDA\\u2010approved drugs by applying a combination of network\\u2010based approaches and systems pharmacology\\u2010based network medicine platform with the ability to quantify the interactions between the drug targets and virus\\u2013host interactome in the human protein\\u2013protein interactions network. Zhou et al. The docking screening method was applied for a set of 2201 approved drugs to find inhibitory activities at SARS\\u2010CoV\\u20102 main protease, and based on that, carfilzomib, eravacycline, valrubicin, lopinavir, and elbasvir have nominated as an inhibitor for SARS\\u2010CoV\\u20102 main protease Wang. Table includes recent studies in terms of computational DR for coronavirus drug discovery and various diseases, respectively. Various transcriptomic datasets in relation to SARS\\u2010CoV\\u20102 were reported since the COVID\\u201019 outbreak. Basically, the idea is that for a specific disease signature composing of a class of up and downregulated genes, if there is available drug candidate, where those same classes of genes are alternatively downregulated and up\\u2010regulated, respectively, this drug might be potential for that disease Hoffmann et al., Le et al. Le et al. investigated existing computational drug repositioning with remarkably reversed differential gene expression in comparison with manifold input signatures for SARS\\u2010CoV\\u20102 effects on human cells. They tried to predict the potential drugs toward differentially expressed gene sets from both cell line and organoid disease models and human samples. They found 102 compounds, and among them, 25 were selected in at least two of the signatures, and some of them have been already analyzed in clinical trials. They eventually tested 16 of their shortlisted candidates in live SARS\\u2010CoV\\u20102 antiviral assays. If a drug could notably FDR < 0.05 reverse the disease signature, then the drug could be a potential inhibitor for the SARS\\u2010CoV\\u20102. In the end, they found that 16 of their high predicted hits showed antiviral assay in live SARS\\u2010CoV\\u20102. Four potential inhibitors were tested for inhibition of SARS\\u2010CoV\\u20102 in a human lung cell line, Calu\\u20103, infected with SARS\\u2010CoV\\u20102. Three drugs revealed in vitro antiviral efficacy\\u2014bacampicillin, clofazimine, and haloperidol with no toxicity effects Le et al. Ge et al. used an integrative drug repositioning analysis through machine learning to integrate and mine large\\u2010scale knowledge graph, literature together with transcriptome data to identify the potential compounds against SARS\\u2010CoV\\u20102. They chose to work on CVL218, a poly\\u2010ADP\\u2010ribose polymerase 1 PARP1 inhibitor. Their in vitro data showed an efficient inhibitory effect against SARS\\u2010CoV\\u20102 replication without a clear cytopathic effect. They further revealed that CVL218 can effectively bind to the nucleocapsid N protein of SARS\\u2010CoV\\u20102 and could suppress the LPS\\u2010induced production of many inflammatory cytokine, which are highly related to the prevention of immunopathology induced by SARS\\u2010CoV\\u20102 infection Ge et al. Another PARP1 inhibitor, PJ\\u201034 was previously studied for acute inflammatory including inflammatory diseases of the lung. It was observed that PJ\\u201034 can interact with the N\\u2010terminal domain of the coronavirus nucleocapsid N protein and decrease the RNA binding ability to inhibit virus replication Ge et al. Beclabuvir is an antiviral drug for the treatment of hepatitis C virus HCV infection studied for inhibitory activity against SARS\\u2010CoV\\u20102. It was reported that beclabuvir could be a potential compound for the treatment of SARS\\u2010CoV\\u20102 through virtual screening and molecular docking in order to bind to RdRp of the SARS\\u2010CoV\\u20102 Talluri. Raltegravir belongs to a class of HIV integrase inhibitors used in the treatment of HIV\\u20101 infection. It obstructs the insertion of HIV\\u20101 DNA into the host cell genome. Raltegravir was selected as a highly effective drug against Mpro and has been repurposed for the potential treatment of COVID\\u201019 in many studies Nabi et al. There has been considerable progress in the practical development and the rational application of computational drug discovery, giving rise to a paradigm shift in both industry and academics. CADD is a powerful approach for proposing new drug candidates, particularly when used in tandem with current chemogenomic methods. Even though CADD takes advantage of manifold dilemmas and approximations, this knowledge\\u2010driven approach plays a significant role in the drug discovery process because of its ability to fast\\u2010track drug discovery. Since CADD has provided several achievements, there would be a promising future to aid drug discovery of many more medicines in the future. Chemogenomics is a superior interdisciplinary field at the interface of chemistry, biology, and informatics, which dramatically decreases the cost and time of drug discovery. This technique allows sequential genetics that is further interpreted into potential biological targets. Over the last years, there has been great attention to DR in pharmaceutical and biomedical companies, highlighting the paramount importance of repurposing. This interest has led to victorious launches in new, highly economical, and appealing indications compared to traditional drug discovery with high\\u2010priced drug development. Over the last 3 years, the COVID\\u201019 pandemic has killed many people globally since there are no effective drugs against it. In this sense, we need to highlight that DR attempts the search for effective drugs in an urgent situation such as the COVID\\u201019. Utilizing DR is strongly recommended to continue investigating the repositioning of existing drugs against different diseases. However, how effectively CADD, including DR and chemogenomics, will produce feasible treatments across different disease areas remains to be seen. Therefore, it should not be surprising that success is limited at times. Several basic computational problems still need to be resolved. To ameliorate drug design and development dilemmas, rapid developments in chemical and structural biology, bioinformatics, clinical experiments, and computational technology are essential. For example, high\\u2010throughput screening HTS has been increasingly used in the pharmaceutical sphere and academic institutions to discover hit and lead compounds rapidly. This potential automated approach lets us screen the entire compound library of the number of chemicals or biological compounds in a target\\u2010based assay or even a cell\\u2010based assay to obtain the desired biological response. In order to use HTS against SARS\\u2010CoV\\u20102, it is important to take advantage of a similar homology of virus in genomics, specifically, chemogenomics, as a versatile way of drug screening model that provides better bio\\u2010safety. Hitherto, CADD is the only way for screening drugs against COVID\\u201019 and is based on a single viral protein or small\\u2010scale drug screening of formerly FDA antiviral drugs utilizing wild\\u2010type SARS\\u2010CoV\\u20102. Nevertheless, this approach is not well\\u2010suited to HTS when focusing on the bio\\u2010security requirements of SARS\\u2010CoV\\u20102. Moreover, developing new drugs takes too much time without having a precise crystal structure of the viral target Liu et al. Thus, the best way to rapidly find a drug for the treatment of COVID\\u201019 would be the use of CADD methods in order to discover lead compounds for clinical use for instance, we can build a program that takes advantage of structure\\u2010based drug design, HTS, and virtual drug screening to identify new leads targeting the Mpro. More recently, a library of 10000 compounds involving approved drugs, drug candidates in clinical trials, and biologically active compounds to inhibit Mpro was examined. As a result, six compounds were found with a great IC50 against Mpro and strong antiviral activity in cell\\u2010based assays. Moreover, using CADD, the crystal structure of Mpro in complex with Michael acceptor inhibitor known as N3 can be determined Jin et al. It is also possible to identify drug leads against SARS\\u2010CoV\\u20102 by targeting the 3\\u2010chymotrypsin\\u2010like cysteine protease 3CLpro enzyme performing a vital role in the life cycle of CoVs, including MERS\\u2010CoV and SARS\\u2010CoV Ul Qamar et al. Scientists employed vHTS tools to predict the 3D structure of the SARS\\u2010CoV\\u20102 3CLpro enzyme by constructing a 3D homology model and screening it against 32297 potential antiviral phytochemicals/traditional Chinese medicinal compounds of a medicinal plant library. This effort led to the discovery of nine hits that need further optimization to be identified as anti\\u2010SARS\\u2010CoV\\u20102 leads. In addition, an interesting investigation based on an HTS assay employing a compound library of 2000 approved drugs and pharmacologically active compounds has also been carried out to find broad\\u2010spectrum inhibitors against the replication of CoVs Shen et al. From their HTS data, 56 hits were screened, and 36 compounds were validated in vitro using genetically engineered human CoV OC43 HCoV\\u2010OC43. Consequently, seven effective compounds were selected as broad\\u2010spectrum inhibitors in which lycorine protected BALB/c mice against lethal human coronavirus OC43 HCoV\\u2010OC43, and emetine blocked MERS\\u2010CoV entry following pseudo\\u2010virus entry assays. These studies prove the efficacy of screening strategies in the rapid discovery of drug leads, resulting in new drugs for infectious diseases having no effective drugs or available vaccine. No funds were received for this article.\",\n          \"Coronavirus disease 2019 COVID-19, which was first identified in December 2019, and shortly after, declared a pandemic by World Health Organization WHO. As of January 18, 2022, there have been more than 326 million confirmed cases and 5.54 million deaths globally. Coronaviruses belong to the family of Coronaviridae, RNA viruses with crown-like spikes on the surface of the coronavirus particles. According to a meta-analysis of Macedo et al., the mortality rate of COVID-19 was 17.1% for patients admitted to hospitals, whereas WHO estimated a fatality rate of 6.73%, which was much lower than that calculated from published studies. Among the critical cases of COVID-19, the mortality rate reaches 40%. Substantial efforts have been made in the treatment of patients with COVID-19. The WHO recommendations in the treatment of COVID-19 are as follows molnupiravir conditional, baricitinib strong, ruxolitinib and tofacitinib conditional, sotrovimab conditional, casirivimab and imdevimab conditional, IL-6 receptor blockers tocilizumab and sarilumab strong, remdesivir conditional, and systemic corticosteroids strong. The WHO recommends not to use ivermectin, lopinavir/ritonavir, hydroxychloroquine, and convalescent plasma. The pathogenesis of COVID-19 was explained by cytokine storm, reduction in ACE2 expression, and activation of complement pathway-induced microvascular injury and thrombosis. The mechanisms of the recommended agents are focused on the mentioned pathogenesis to improve the clinical outcome of COVID-19, and antiviral therapies are missing. The anticoronaviral strategies include preventing the synthesis of viral RNA, inhibiting virus replication, blocking the virus binding to human cell receptors, or inhibiting the viruses' self-assembly process. The SARS-CoV-2 contains at least four structural proteins spike S protein, envelope E protein, membrane M protein, and nucleocapsid N protein, and 16 nonstructural proteins NSPs. Among the translated NSPs, the main protease, also called chymotrypsin-like protease 3C-like protease, and the papain-like protease are two essential proteases for proteolytic processing of the coronavirus replicase polyprotein, therefore generating functional replication complex of the virus, whereas RNA-dependent RNA polymerase is the central enzyme for RNA synthesis. These three NSPs play crucial roles in coronavirus replication, making them attractive targets for anticoronaviral drug design. The S protein, a surface-located trimeric glycoprotein of coronaviruses, promotes the attachment of viruses to host cells through binding to angiotensin-converting enzyme 2 ACE2 and virus-cell membrane fusion during viral infection. Thus, the S protein has been considered as a major target for the development of vaccines and drug. The development of a new therapeutic agent is a complex, lengthy, and expensive process, which can take 2\\u20134 years of preclinical development and 3\\u20136 years of clinical development and over 500 million dollar cost. There are three critical steps to develop a new drug including discovery and development, preclinical research, and clinical development. Drug discovery involves screening hits, medicinal chemistry, and optimization of hits to reduce potential drug side effects. For drug discovery, two different complementary approaches can be applied classical pharmacology, also known as phenotypic drug discovery, which is the historical basis of drug discovery, and reverse pharmacology, also known as designated target-based drug discovery. Screening methods based on phenotypic drug discovery have been used to discover new natural products mainly from the terrestrial origin. These two strategies have advantages and disadvantages and promote very different screening assays. The frequent re-discovery of the same compounds, the technical difficulties associated with the isolation of compounds from extracts, and the incompatibility of natural product extracts with high-throughput screening HTS campaigns were the disadvantages of phenotypic drug discovery. On the other hand, natural product structures have the characteristics of high chemical diversity, biochemical specificity, and other molecular properties that make them favorable as lead structures for drug discovery, which serve to differentiate them from libraries of synthetic and combinatorial compounds. Overly simplified assays, acting of drug on more than one target, the multifactorial nature of diseases, and challenges to identify a single molecular target are some limitations of target-based drug discovery. Therefore, a comprehensive screening strategy will incorporate both targeted and phenotypic assays, with one format designated as the primary screen and the other as a secondary or follow-up assay. During the spread of COVID-19 outbreak, great efforts have been made in therapeutic drug discovery against the virus. Because COVID-19 is a new, acute, severe infectious disease, the anti-SARS-CoV-2 drug development strategies are to screen existing drugs to identify potentially effective drugs, to expand indications, and to develop a vaccine. The safety of conventional drugs has been mostly verified if effective, they can be quickly applied in clinical practice repurposing of existing drugs. The recent rises of several high transmissible strains sounded alarms for currently used vaccines and drugs. Therefore, developing broad-spectrum antiviral drugs not only to combat COVID-19 but also to provide protective arsenals against future viral outbreaks is a requirement. Scientists continue the development of broad-spectrum antiviral drugs from natural or chemical sources, which have the potential advantages of broad-spectrum therapeutic effect and insensitivity to viral evasion. Given the urgency of the SARS-CoV-2 outbreak, here we discuss the discovery and development of new therapeutics for SARS-CoV-2 infection based on the strategies from preclinical drug discovery. We report this systematic review based on the Preferred Reporting Items for Systematic Reviews and Meta-Analysis PRISMA guidelines. Studies published in PubMed, Scopus, Web of Science, EMBASE, Google Scholar, and DrugBank were searched from December 2019 to March 2021 using the following search terms Coronavirus, Covid-19, SARS-CoV-2, Drug design, Drug development, In silico, In vitro, and In vivo alone or in combination without language restrictions. The keywords were selected using expert opinions, mesh, and related article titles. All articles with full text or in the absence of full text with abstract are included in the screening of this study. Studies were excluded if studies were comments, editorial, letters, review, and preprints. Two researchers independently extracted data from included studies using a predefined data extraction form. All disagreement was discussed and solved after rechecking the source data with a third investigator. The data extracted, including the last name of the first author, type of the study in silico, in vitro, in vivo, and name of the agent chemical compound, drug, herb, etc., studied the mechanism and efficacy of the agent according to study design that specified according to the following definition. Computer-aided drug design can be divided into three different categories. All are based on ligands and receptors, which are briefly described as follows Dock Receptor-Based Approach. Once the three-dimensional structure of the ligand molecules and their receptor is known, the receptor-based method is a good candidate for identifying or optimizing drugs. Due to the presence of three-dimensional structures of compounds and receptors, the nature of the interaction between the ligand and the receptor and the type of structure that the ligand can have to interact with them in favorable conditions can be identified using this method. The compound is simulated on the active site of the dock meaning anchoring and on the interaction of the ligand with the receptor by molecular mechanics and molecular dynamics. In this method, due to the ligation of the ligand in the active position, the ligand changes in terms of conformity and changes its position in different conditions and shows interaction with the receptor in different types of situations. To determine the type of ligands that can be docked into the receptor site, the matching of the shape and the complementarity of the hydrophobic, hydrophilic, and charged parts must be considered. Various software packages such as AUTODOCK-Glide-LUDI and LigandFit are used to design the drug based on the structure of the receptor. Ligand-Based Approach. This method is used where the three-dimensional structures of the receptor are unknown and instead the structure of the ligands is known, which is one of the common methods. In this method, by indirectly studying compounds that react with biomolecules, they seek to design compounds that are pharmacologically active. In ligand-based drug design methods, in the absence of biomolecule structure, by studying specific ligands, it seeks to identify the structural and physicochemical properties of the compounds so that the desired compound can be designed based on data extracted from the study of previous compounds. This method is a kind of drug design based on pharmacophore pharmacophore refers to the part of the drug to which the effect of the drug depends on that part of the molecule, and by studying the quantitative relationship between structure and their activity, drugs can be designed by this method. It can be said that it is a method for designing the pharmacophores of drugs. Denovo Design-Based Approach. This method is used when the structure of the ligand is unknown but the structure of the receptor is known. In this method, there is information about the structures of the receptor or quasi-receptors, but there is no structure of the main composition that can interact with the active site of the receptor. One of the functions of drug design based on this method is to suggest and present the main composition that is complementary to the active site. The basis of the method is that the database of existing 3D structures is used to find small molecules that can interact with the active site of the receptor in terms of size, geometry, and functional groups. Software packages such as GROW and LEGEND are used to design drugs by this method. Drug design methods in the computer include quantitative structure-activity relationship RASQ, docking, molecular dynamic simulation, and computational modeling. In these studies, the efficacy is evaluated based on the function of the drug or compound agent and the mechanism of action. shows the methods of computer-aided drug design CADD. Quantitative Structure-Activity Relationship QSAR. QSAR provides studies on the relationship between chemical structure and biological activity or other biological activities that are important in selecting or removing a compound before synthesis and testing. QSAR is especially important to predict the result, especially when it is not possible to experiment with a compound. Molecular descriptors, which are the most important components of QSAR, can be obtained experimentally or through mathematical formulas from various theories such as quantum mechanics, chemical graph theory, and study theories. QSAR seeks to establish a statistically significant relationship between structure and performance. It also explains the specific effect of a drug and can ultimately predict the effect of newly synthesized chemical compounds. QSAR model is also an equation that predicts a property through molecular descriptors and their coefficients. Evaluation of the effectiveness of new compounds that have been studied using this method can be reported as a percentage of enzyme inhibition if the modeling has been done and mentioned in the article. In this technique, to achieve a combination with a pharmacological effect and increase the pharmacological activity of the drug, different formulations of a drug interact with the receptor, and the structure that has the best interaction with the receptor and the lowest energy level is selected for laboratory steps. In this way, possible structures that have a stronger interaction with the receiver can be isolated at this stage. The same issue is considered and reported as an effectiveness measure. In this technique, which is based on the simulation of drug-receptor interaction in the body, docking problems are solved and in fact play a complementary role in this technique. Due to the time-consuming work with this technique, effective structures cannot be achieved directly through it, and the final stages of the study of drug-receptor interactions should be evaluated before starting laboratory work by having effective compounds from the previous stages. Molecular dynamic simulations produce information at the microscopic level position and velocity of atoms. The conversion of these data into macroscopic values pressure, energy, etc. is done using statistical mechanics. In fact, molecular dynamics and statistical mechanics link microscopic concepts and macroscopically observable quantities. Molecular dynamic simulations are only able to predict the thermodynamic behavior and stability of the ligand binding mechanism at the active site of the target enzyme. This is reported as a criterion. Study of drug in cell culture medium effectiveness in these studies means inhibition of the replication of COVID-19 by the compound or drug under study. The half-maximal inhibitory concentrations IC50 are a measure of the effectiveness of a compound in inhibiting biological function. In vivo studies are those in which the effects of drugs are tested on whole living organisms or cells usually animals as opposed to a tissue organism or dead organism. In vivo testing is better studied for observing the overall effects of an experiment on living subjects. In this review, we reported a significant number of articles with in silico, in vitro, and in vivo approaches for drug development of COVID-19. We retrieved a total of 2538 articles from the initial database search. After the removal of duplication and screening, 317 studies were selected for inclusion in this review. shows the PRISMA diagram. The analysis of article contents indicated that 266 studies performed in silico approaches against viral targets 34 studies used in vitro approaches against SARS-CoV-2 and 15 studies used in vivo animal models. From 267 studies used in silico approaches, 98 article studies repurposed approved drugs with a new mechanism of action and 91 studies evaluated natural products e.g., herbal medicine on COVID-19. The characteristics of these studies are summarized in Tables and. Also, shows the characteristics of the remaining studies N 87. In silico studies used the following component of novel coronavirus as targets main protease N 154, spike glycoprotein N 62, nonstructural protein N 45, RNA-dependent RNA polymerase N 21, papain-like protease N 19. About 260 drugs were repurposed by the computational methods for COVID-19 therapy such as about 120 drugs candidate against the main protease, 52 drugs against the spike glycoprotein, 14 drugs against RNA-dependent RNA polymerase, and 28 drugs against other nonstructural proteins. Among the studied repurposed drugs, the best results regarding efficacy were observed with saquinavir N 9 study, ritonavir N 8 study, lopinavir N 6 study, remdesivir N 3 study, and amikacin, danoprevir, favipiravir, and telaprevir. shows target-based synthesis of data for COVID-19 drug repurposing. As presented, at least two studies show the efficacy of aprepitant, cobicistat, dipyridamole, and dihydroergotamine against the main protease and tegobuvir N 2 against spike protein. The following list of drugs had multitarget action avapritinib, famotidine, bictegravir, ziprasidone, capmatinib, pexidartinib, amprenavir, zafirlukast, cilostazol, paromomycin, lopinavir, and remdesivir. A total of 91 studies used in silico methods to evaluate the effects of natural products including herbal medicine against SARS-CoV-2. Among them, 54 studies used the main protease as main target, in which eucalyptus N 3, Tinospora cordifolia N 3, and flavonoids e.g., hesperidin, rutin, and herbacetin were the most studied and effective. Other studied targets were as follows spike N 22 and multitarget N 20. From the plant metabolites, oleanolic acid, hesperidin, epigallocatechin gallate, jensenone, tinosponone, and anistone show promising results in computational methods against COVID-19. Aloe, green tea, eucalyptus, curcumin, and many Chinese, Indian, and African plants were also effective on COVID-19 in silico. The derivatives of pyrazoles, oxadiazoles, phenyltriazolinones, triazoles, benzoylpinostrobin, benzoic acid, benzylidenechromanones, coumarin, and selenium show efficacy in computational methods and could be considered as lead molecules for drug design and synthesis against COVID-19. Melatonin, ebselen, phenyl furoxan, thimerosal, isatin, romidepsin, phenyl mercurin, pleconaril, and tyrosine kinase inhibitors such as nilotinib and imatinib are examples of other drugs that are effective in in silico methods. Screening of in vitro studies leads to finding 34 studies. Different compounds were evaluated and most studies focused on the inhibition of viral replication, which was assessed by the quantification of viral RNA by PCR and IC50 value reported. Most of the studies use the Vero E6 cell line for the assessment of replication of SARS-CoV-2. In particular, main protease, papain-like protease, RNA-dependent RNA polymerase, NSP-14, NSP-15, spike protein, and TMPRSS2 were evaluated in vitro as targets. Also, three studies focused on in vitro inhibition of inflammatory markers such as interleukins and the effects of the cytopathic effect on infected cell. Nitric oxide, ginkgolic acid, anacardic acid, troxerutin, bisindolylmaleimide derivatives, small molecules GRL-172, and 5 h, baicalin and baicalein phytochemicals, and bepridil were effective drugs in vitro against the main protease of COVID-19. The following drugs could inhibit the spike/ACE2-mediated cell entry of COVID-19 in vitro romidepsin, panobinostat, givinostat, sirtinol, saquinavir, lipopeptides, hydroxyzine, azelastine, heparin, and glycyrrhizic acid. Alkenyl sulfonylurea derivatives, an IMU-8381 inhibitor of human dihydroorotate dehydrogenase, pyrazole derivatives, and phillyrin, regulated the expression of inflammatory cytokines e.g., IL, TNF-, and NF- B induced by SARS-CoV-2 markedly and could be considered as adjuvant treatment of COVID-19 severe disease. Doxycycline, chlorpromazine, azithromycin, heparin, bepridil, tannic acid, and glycyrrhizic acid are well-known drugs that show both in silico and in vitro inhibitory effects against SARS-CoV-2 and should be considered for this purpose. Also, in vitro studies show that lopinavir/ritonavir, sofosbuvir, and favipiravir have no antiviral effects against SARS-CoV-2 huge gap between in vitro IC50 and free plasma concentration. We included 16 in vivo studies in our final analysis. Most of them use a combination of in vitro and in vivo methods for the evaluation of novel drugs. shows the detail of these studies. Human monoclonal antibodies were the most evaluated drugs N 4, promote the reduction in viral load in vitro, and prevent infection in animal models of SARS-CoV-2. A prodrug of hydroxycytidine molnupiravir improved pulmonary function and reduced viral titer in vivo and was introduced as a potential broad-spectrum antiviral agent against SARS-CoV-2. Chloroquine and chlorpromazine did not inhibit viral replication in mouse lungs, but protected them against clinical disease. Dalbavancin shows significant inhibitory ability in both in vitro and in vivo models of COVID-19. The drug binds directly to ACE2 and blocks its interaction with the SARS-CoV-2 spike protein. Drug development is a multistep process, typically requiring more than five years to assure the safety and efficacy of the new compound. There are several strategies in antiviral drugs for coronaviruses including empirical testing of known antiviral drugs, large-scale phenotypic screening of compound libraries, and target-based drug discovery. To date, an increasing number of drugs have been shown to have anticoronavirus activities in vitro and in vivo, but only remdesivir and several neutralizing antibodies have been approved by the US FDA for treating COVID-19. However, remdesivir's clinical effects are controversial and new antiviral drugs are still urgently needed. Given the urgency of the SARS-CoV-2 outbreak, here we discuss the discovery and development of new therapeutics for SARS-CoV-2 infection, which have been conducted in basic research in silico and preclinical study in vitro, in vivo. Our database search identified about 3000 studies, which means a global effort for drug development in the current COVID-19 pandemic. Although the chance of successful drug development is very low less than about 10% and till today, there are no approved drugs. Many potential candidates at least 420 drugs are in clinical trials. As summarized in the results, many compounds are in the development process for COVID-19 disease. Some of these compounds are completely new and could serve as seeds or leads for developing antiviral drugs against COVID-19, but as we need therapeutics as soon as possible, half of the studies focused on drug repurposing repositioning, which is a process of investigation of existing drugs for new therapeutic purposes. With the emergence of a growing COVID-19 pandemic, the drug repurposing process was being accelerated. Clinical trials using repurposed drugs may take less time and have a lower overall cost of manufacturing and could have a wide distribution of drugs. According to our results, 260 drugs repurposed by the computational methods for COVID-19, among them saquinavir, ritonavir, and lopinavir, showed the best efficacy in in silico environment. These drugs can be rapidly repurposed for clinic application for treating COVID-19 patients given their proven safety. Many trials are performed using a combination of ritonavir-lopinavir. The results of a systematic review and meta-analysis showed that this drug combination has no more treatment effects than other therapeutic agents in COVID-19 patients and is currently not used anymore. We could not find any clinical trial on saquinavir, which is the most studied drug in silico and show high potency against COVID-19. Saquinavir could be a suggestion for further clinical research. Given that the development of synthetic chemicals for therapeutic use is a random process that might result in serendipitous discovery, many pharmaceutical companies are now focused on the development of plant-derived drugs. Natural products and their structural analogs have historically made a major contribution to pharmacotherapy, especially for cancer and infectious diseases. Nevertheless, natural products also present challenges for drug discovery, such as technical barriers to screening, isolation, characterization, and optimization. In recent years, several technological and scientific developments\\u2014including improved analytical tools, genome mining and engineering strategies, and microbial culturing advances\\u2014are addressing such challenges and opening up new opportunities. Consequently, interest in natural products as drug leads is being revitalized. Medicinal plants have attracted significant attention to treat infectious diseases. Complex molecular structures and a wide variety of natural compounds make medicinal plants an excellent biological resource for drug discovery. Our results show that various plants have potential antiviral activities and could use or be a basis for drug development against COVID-19. Some of the studied plants are among what used by people on a daily basis, such as green tea, aloe, curcumin, and eucalyptus. A systematic review and meta-analysis of RCTs on herbal medicine in the management of COVID-19 show the significant effects of the combined therapy of herbal medicine in treating COVID-19 without any significant side effects. Our results retrieved 91 in silico studies of natural products including herbal medicine. The results of this in silico approach showed that some of these studied active ingredients have a high affinity for each of the four important viral proteins compared with the inhibitors previously reported for each of these proteins. They could possibly have an inhibitory effect on the SARS-CoV-2 and COVID-19. Most evaluated chemical compounds had inhibitory effects against one or two proteins of SARS-CoV-2. In addition to having a great affinity to attach to the viral proteins, these herbal compounds have antioxidant, vasoprotective, anticarcinogenic, and antiviral properties. Thus, they can be applied as extremely safe therapeutic natural compounds and clinical assessments might have notable outcomes for controlling COVID-19. Due to the nature of preclinical in vitro and in vivo studies, the number of drug development studies in this area was less than in silico studies. Again, repurposed drugs were the most studied drugs in vitro both herbal and synthetic. Some of them such as saquinavir, heparin, glycyrrhizic acid, and chlorpromazine show efficacy in both in vitro and in silico environments. Chlorpromazine is the single agent that was found to have efficacy in in silico, in vitro, and in vivo areas. In terms of mechanism of action, different targets from the structural and nonstructural proteins of COVID-19 were evaluated. Most of the studies focused on the main protease, papain-like protease, and spike glycoprotein. Apart from the specific protein that leads to viral replication, SARS-CoV-2 causes a surge of pro-inflammatory cytokines and chemokines, which cause damage to lung tissue and deterioration of lung function. Therefore, the design of a drug with multitarget of action against different proteins of COVID-19 and also anti-inflammatory potential could be valuable. Currently, there is no highly efficacious and specific treatment for SARS-CoV-2. Herein, we provided data on novel compounds in therapeutic drug discovery and development. Due to the nature of SARS-CoV-2 and the rises of several high transmissible strains, repurposing existing drugs has demonstrated power by bringing several drugs to approval for treating COVID-19 patients, such as remdesivir. Our results confirmed that a large number of repurposed agents are currently being explored for treating SARS-CoV-2 infection. However, these drugs still suffer from suboptimal therapeutic effect or known strong side effect. To accelerate drug discovery and development, especially during the current pandemic, natural products capture attention again. Our results showed that various natural bioactive compounds are being investigated in the preclinical step of drug development for COVID-19. In addition to having high affinity, these herb active ingredients have antioxidant, vasoprotective, anticarcinogenic, and antiviral properties. Therefore, they can be used as extremely safe therapeutic compounds in drug design studies to control COVID-19. However, the pharmacological effects and adverse reactions of some drugs under development are still unclear, and hence, well-designed high-quality studies are needed to further study the effectiveness and safety of these potential drugs to accelerate drug development targeting SARS-CoV-2 and thus promote progress towards ending the pandemic.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-03-05_pmc_arxiv_full_text_merged_plus_cleaned_test.pickle', 'wb') as f:\n",
        "  pickle.dump(pmc_arxiv_full_text_merged_plus_cleaned_test, f)"
      ],
      "metadata": {
        "id": "1OVgZPiq7TDf"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_arxiv_full_text_merged_plus_cleaned_test.loc[0:9,['article_id', 'text']] # after running function"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 363
        },
        "id": "UC_J2FEK6mOe",
        "outputId": "f4733d92-fa10-4265-a373-ef6840222eb4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   article_id                                               text\n",
              "0  PMC9549161  Sir James Black, a winner of the 1988 Nobel Pr...\n",
              "1  PMC9539342  Tight and selective interaction between ligand...\n",
              "2  PMC9357751  The severe acute respiratory syndrome coronavi...\n",
              "3  PMC9346052  The sudden outbreak of SARS-CoV-2 in 2019 took...\n",
              "4  PMC9775208  The 2019 novel coronavirus, now dubbed SARS-Co...\n",
              "5  PMC9527439  Coronavirus disease 2019 COVID-19, which was f...\n",
              "6  PMC9729590  COVID-19 is an infectious disease caused by Co...\n",
              "7  PMC9236981  A novel coronavirus CoV first appeared by the ...\n",
              "8  PMC9694939  Drug discovery is a time-consuming and costly ...\n",
              "9  PMC9556799  Coronavirus disease 2019 COVID-19, caused by s..."
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5709a03d-9b42-48ab-bf00-e7ef3551d9f2\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>article_id</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>PMC9527439</td>\n",
              "      <td>Coronavirus disease 2019 COVID-19, which was f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>PMC9729590</td>\n",
              "      <td>COVID-19 is an infectious disease caused by Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>PMC9236981</td>\n",
              "      <td>A novel coronavirus CoV first appeared by the ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>PMC9694939</td>\n",
              "      <td>Drug discovery is a time-consuming and costly ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>PMC9556799</td>\n",
              "      <td>Coronavirus disease 2019 COVID-19, caused by s...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5709a03d-9b42-48ab-bf00-e7ef3551d9f2')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-5709a03d-9b42-48ab-bf00-e7ef3551d9f2 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-5709a03d-9b42-48ab-bf00-e7ef3551d9f2');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-3b6aa878-53a8-4c2f-85cd-0b756f57cfdd\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-3b6aa878-53a8-4c2f-85cd-0b756f57cfdd')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-3b6aa878-53a8-4c2f-85cd-0b756f57cfdd button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"pmc_arxiv_full_text_merged_plus_cleaned_test\",\n  \"rows\": 10,\n  \"fields\": [\n    {\n      \"column\": \"article_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"PMC9694939\",\n          \"PMC9539342\",\n          \"PMC9527439\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Drug discovery is a time-consuming and costly process that involves a lengthy assessment of clinical trials and has a success rate of only 2%, with an average cost of USD 2\\u20133 billion, and also fails to deliver at the time of pandemics. On the other hand, drug repurposing has several advantages over drug discovery, and one is that it can be delivered in times of pandemics. At the beginning of February 2020, the World Health Organization WHO declared the novel coronavirus disease 2019 COVID-19 a pandemic due to its rapid spread as well as its high mortality and morbidity rates. COVID-19 first appeared in Wuhan city, China, in late 2019 and is still spreading all over the globe creating a disastrous effect on global health and the economy. Due to its highly contagious nature, as of May 2022, the pandemic has spread to more than 200 countries and territories with over 5.5 million confirmed positive cases and mortality of over 350K across the globe. The etiological agent of COVID-19 is severe acute respiratory syndrome coronavirus 2 SARS-CoV-2, and it has been reported to originate from the animal coronavirus. SARS-CoV-2 belongs to the genus Betacoronavirus of the family Coronaviridae which already has two other members, i.e., SARS coronavirus SARS-CoV-1 and Middle East respiratory syndrome coronavirus MERS-CoV which caused pandemic onslaughts in 2002 and 2012, respectively, resulting in severe human losses both in terms of morbidity and mortality. Coronaviruses are positive-sense single-stranded RNA with a genome size of ~30 kb containing a 5\\u2032 cap-structure and 3\\u2032-polyadenylated tail. Their genome encodes for ORF1a and ORF1ab polyprotein which cleaves into multiple subunits and performs a diverse range of functions, i.e., binding and invading the host cell, viral replication, and evading the host immune system. The proteins of coronaviruses perform their function by aggregating multiple subunits into an heteromeric complex, i.e., replication-transcription complex and several other structural proteins that assemble into new virions. One of these subunits is a non-structural protein 12 NSP-12 which involves RNA-dependent RNA polymerase RdRp activity. NSP-12 of SARS-CoV-2 is highly conserved by sharing 96% similarity with SARS-CoV-1 and 70% with MERS-CoV. The recently resolved structure has also revealed conserved structural architecture by comparison with non-structural protein 5 NS5 of flaviviruses and polymerases of picornaviruses. However, NSP-12 has a length of ~930 amino acids in contrast to the polymerases of Human Poliovirus PV and NS5B of Hepatitis C Virus HCV which usually comprises ~500 amino acids. NSP-12 is responsible for viral replication and transcription once the virus invades into the host cellular environment by aggregating with cofactors such as NSP-7 and NSP-8 along with some additional subunits to initiate viral replication. Its tertiary structure comprises three highly conserved domains N-terminal nucleotidyltransferase also known as nidovirus RdRp-associated nucleotidyltransferase NiRAN, interface, and C-terminal RdRp domain. The RdRp domain of NSP-12 is highly conserved and responsible for viral replication and has been reported in several families of RNA genome viruses, while the NiRAN domain is only conserved among the members of nidoviruses by possessing kinase-like fold and involves in attachment with NSP-7 and NSP-8. RdRp domain contains seven conserved motifs motif A-G that involve attachment with nucleotide triphosphate NTP and RNA primer during the viral replication. Due to these conserved evolutionary features of the RdRp domain, it has been suggested to be an important therapeutic target. Currently, many studies including single-ligand docking as well as screening of the compounds library have been conducted to search for potential inhibitors against NSP-12 of SARS-CoV2. In our study, we employed the structural-based homology method to identify structurally-conserved proteins and then identified US Food and Drug Association FDA -approved inhibitors of the homologous structure. The selected FDA-approved inhibitors were Dasabuvir Sodium Dasabuvir and Sofosbuvir which were previously authorized for inhibiting the NS5B of HCV. Furthermore, the conserved active site of NSP-12 was characterized and the affinity of the inhibitors were evaluated using in silico assessments, i.e., molecular docking and simulations. The DALI server was employed for the identification of homologous protein structures to NSP-12 of SARS-CoV-2 that were already submitted in the PDB repository. The submitted query resulted in a total of 1105 homologous structures with a default cutoff of Z-score \\u22652 as shown in. For screening significant homologs, a cutoff Z-score value \\u226519 was considered, resulting in 611 structures. These resulting homologous structures were involved in the RdRp activity and belonged to four viral families Coronaviridae, Flaviviridae, Picornaviridae, and Caliciviridae and infected a diverse range of hosts. The representatives belonging to different viral genera were selected from the filtered homologous structures and their similarities were evaluated by using the similarity matrix and a cladogram by the neighbor joining method. The resulting matrix was visualized as a heatmap as shown in A. The heatmap of Z-score-based structural similarity revealed the close relationship of coronaviruses with the members of Flaviviridae as well as Caliciviridae and Picornaviridae. Based on structural similarity, the unrooted cladogram of homologous structures to NSP-12 was clustered into four viral families as shown in B. The clade of the coronaviruses, i.e., SARS-CoV-1 and SARS-CoV-2, shares a close relationship with the NS5B of HCV and also with the other representatives of Flaviviridae, i.e., the Zika Virus ZIKV, Dengue Virus DENV, Bovine Viral Diarrhea Virus BVDV, Japanese Encephalitis Virus JEV, and Classical Swine Fever Virus CSFV. The members of Picornaviridae included PolV, Human Coxsackievirus CoxV, Human Rhinovirus HRV, Human Enterovirus HEV, Human Enterovirus 71 HEV71, Foot and Mouth Disease Virus FMDV, and Encephalomyocarditis virus. The representatives of Caliciviridae were the Sapporovirus SapV, Rabbit Haemorrhagic Disease Virus RHDV, Murine Norovirus MNV, Norwalk Virus NarV, and Human Norovirus HNV. The clade of Flaviviridae and coronaviruses shared significant relationships as compared to the members of Caliciviridae and Picornaviridae. The conserved motifs in NSP-12 were analyzed by multiple structures and sequence alignments implemented at the PROMALS3D server. The PROMLS3D server aligns multiple structures based on similarity at the amino acid sequence and secondary structure level. The identified conserved motifs in NSP-12 have been displayed in A. NSP-12 has seven highly conserved motifs motif A-G which are responsible for NTP catalysis, binding to the 3\\u2032 of RNA template, and viral replication by agglomerating with NSP7 and NSP8. To display the conserved orientation of these motifs at the active site of NSP-12, multiple structures belonging to the family Coronaviridae, Flaviviridae, and Picornaviridae were superposed structurally aligned by using the MatchMaker algorithm implemented in UCSF Chimera. The conserved motifs identified from multiple sequence and structural alignment have been shown in B. Motifs A\\u2013E are located in the palm domain while motifs F and G are present in the fingers domain of NSP-12. Motifs A and C have highly conserved aspartic acid residues which bind to the divalent ions Mn2+ or Mg2+ and are responsible for executing the catalysis of nucleotide-binding. Motifs B, D, E, and G are involved in nucleotide recognition and coordination. Motif F is enriched with positively charged basic residues which interact with the triphosphate moieties of the NTP. Motif G has been also reported to involve in adjusting the orientation of the primer and template. Pharmacophore modeling is a very useful technique for the detection of ligand-binding or a druggable site in protein that could be targeted to either trigger or to block the biological activity. Pharmacophore modeling was computed by using the CAVITY search module of the CavityPlus server with a selection of No Ligand mode and default parameters. The CAVITY search module evaluates the pharmacophore or druggable cavity based on surface energy and site geometry. The analysis resulted in the prediction of a total of 27 cavities in NSP-12 and their information has been shown in. The cavities were ranked based on factors, i.e., drug score, druggability, and predicted average PKD and the top-scored cavity as shown in C was selected for further analysis. The top-ranked cavity has a higher drug score revealing significant druggability of the active site with a predicted maximum pKd score of 6.99. The predicted maximum pKd score \\u22656 indicates a suitable ligand ability of a cavity binding site. There was a total of 154 amino acid residues constituting the predicted cavity and they were compared with the evolutionarily conserved residues of the RdRp domain of NSP-12 from multiple sequence alignments. The comparison has been plotted as a Venn diagram as shown in A. The highly conserved residues were K545 and R555 of motif F, D618, and D623 of motif A, S682, G683, T687, and N691 of motif B, D760, and D761 of motif C and W800 of motif D. These residues were further represented as a target in the RdRp domain of NSP-12 for molecular docking with the FDA-approved antivirals. Before molecular docking, the ligands, i.e., Dasabuvir, Sofosbuvir, Ribavirin, GTP, and UTP, were prepared using AutoDock Tools. The grid box was centered at the active site or druggable cavity and the residues were considered rigid for both receptors, i.e., NSP-12 of SARS-CoV-2 and NS5B of HCV. The molecular docking was performed by AutoDock Vina with the parameters that are suggested for a single-docking experiment and the results for each abovementioned ligand were stored. AutoDock Vina predicts the docking score in kcal/mol and the lower energy means higher binding affinity between the ligand and the receptor. The results score of ligand bindings with the active site of NSP-12 and NS5B has been compared and visualized as a bar chart as shown in B.The docking score of NSP-12 with the ligands was in the range of \\u22126.2 to \\u22128.1 kcal/mol, while for NS5B, it was \\u22126.5 to \\u22129.1 kcal/mol, and both had an average difference of 0.84 kcal/mol. The average docking score of NSP-12 and NS5B for Sofosbuvir, Dasabuvir, Ribavirin, GTP, and GTP were \\u22128.2, \\u22128.6, \\u22126.35, \\u22128.25, and \\u22127.6 kcal/mol, respectively. NSP-12 showed the lowest docking score \\u22128.1 kcal/mol with Dasabuvir and higher \\u22126.2 kcal/mol with Ribavirin. In general, the ligands showed a higher docking score for NS5B as compared to NSP-12. To examine the interactions between ligand and the conserved residues present at the active site of NSP-12, LigPlot + v2.1, standalone software was used. The predicted interactions of Sofosbuvir and Dasabuvir with NSP 12 have been shown in A,B, respectively. The conserved residues involved in forming hydrogen bonds with Sofosbuvir were C622 of motif-A, W800 of motif-D, G811, and S814 of motif-E, while Dasabuvir interacted by the formation of hydrogen bonds with D761 of motif-C and W800 of motif-D. In addition, several conserved residues of motif-A, -C, and -E were also involved in hydrophobic interactions with Sofosbuvir and the residues of motif-A, -B, -C, -E, and -F with Dasabuvir. On the other hand, the native competitors such as GTP interacted with NSP-12 by forming hydrogen bonds with A554, R555, and T556 of motif-F, Y619, K621, and C622 of motif-A, while UTP interacted with S759, D760, and D761 of motif-C, W800 of motif-D, and G811 and S814 of motif-E. In the case of HCV as a receptor, Sofosbuvir binds with motif-B and -C, while Dasabuvir interacted with residues of motif-A, -B, -C, and motif-F, and their receptor\\u2013ligand interaction plots have been shown in C,D, respectively. By binding to a protein\\u2019s functional binding cleft, each small molecule has the potential to cause significant conformational changes in the protein\\u2019s structure. To investigate the structural stability of 6M71, 100 ns of molecular dynamics MD simulations were run with the protein in both its apo and ligand-bound forms. The stability of the systems was evaluated using root mean square deviation RMSD, root mean square fluctuation RMSF, radius of gyration, and Hydrogen bond analysis. The average RMSD for 6M71, 6M71-Dasabuvir, Ribavirin, and Sofosbuvir complexes was calculated as 0.23, 0.23, 0.27, and 0.27 nm, respectively. All three systems\\u2019 RMSD values, which quantify conformational changes over time, indicated that the simulation was stable for up to 100 ns. The backbone RMSDs plot analysis revealed that the complex was stable throughout MD simulation despite a small deviation from its primary conformation. displays the RMSD plots of all complexes vs. simulated time. The resulting RMSD graph showed increasing trends with increasing RMSD values ranging from 0.10 to 0.33 nm between 0 and 60 ns in the case of 6M71/Ribavirin and 6M71/Sofosbuvir, suggesting the compounds were becoming accustomed to a new conformation inside the binding pocket after that, the plateau continued and finally, settled at 0.29 nm, which does not exceed the 0.3 nm threshold. However, 6M71/Sofosbuvir has the same evolution to apo 6M71. Lower RMSD values for all complexes investigated shows that Dasabuvir inhibitor is stable inside 6M71 and gives a strong foundation for our research. The root mean square fluctuation RMSF method was used to investigate the effect of ligand binding on the flexible structure of protein as well as the behavior of essential amino acids. Throughout the simulation, a lower RMSF value indicates more rigidity, whereas a higher RMSF value indicates more flexibility. In, RMSFs of the Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes are displayed. When compared to the apo form of CAII, it ise shown that the fluctuations of the ligand-bound complexes\\u2019 residues are relatively stable, particularly at the region where the residues are engaged in ligand binding. Furthermore, the average RMSF values of Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes were 0.10, 0.10, 0.11 and 0.11 nm, respectively. This finding implies that Dasabuvir, Ribavirin, and Sofosbuvir binding helped to maintain NSP-12 structurally stable. We examined how binding of different ligands affected the protein\\u2019s overall compactness in its structure. For this goal, the radius of gyration Rg was calculated as a function of time. Any ligand with a high enough Rg value will be more likely to be flexible, making it unstable. On the other hand, lower Rg values point to confirmation of a dense and closely packed conformation. The average Rg values Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes were found to be 2.99, 3.00, 2.97, 2.98 nm, respectively, suggesting that the binding of Dasabuvir, Ribavirin, and Sofosbuvir to NSP-12 packing did not cause significant change. Moreover, based on the graph, it appears that the Rg of NSP-12/Dasabuvir reached a stable equilibrium during the 100 ns simulation than NSP-12/Ribavirin and NSP-12/Sofosbuvir complexes. Further understanding of the examined chemicals\\u2019 molecular recognition, molecular interactions, and selectivity within the receptors can be attained using hydrogen bond analysis. For all MD trajectories of the complex, we calculated the number of hydrogen bonds created. displays the number of hydrogen bonds and pairs within 0.35 nm generated between NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes during the MD simulation. It was noticed that the Dasabuvir binds to the active pocket of NSP-12 with an average of 3.46 hydrogen bonds and 1.13 pairs within 0.35 nm. Similarly, Ribavirin acting as a ligand interacted with NSP-12 as a target within the binding site with an average of 3.97 hydrogen bonds, although the average of pairs within 0.35 nm is 1.66. However, the average of both hydrogen bonds and pairs within 0.35 nm for the NSP-12/Sofosbuvir complex was 1. On the other hand, the H-bonding plot showed that Dasabuvir was able to maintain a more robust interaction with the binding pockets of NSP-12 throughout the simulation time than Ribavirin and Sofosbuvir. Before MM/PBSA calculation, each complex between the receptors or inhibitors/ligands should reach the equilibrium state. From, it is clear that the temperature of each complex quickly reaches the target value 300 K and remains stable over 100 ns. After that, we utilized a python script MmPbSaStat.py to calculate average free binding energy of the selected complexes, as provided in the g mmpbsa package. This script calculates the average free binding energy and its standard deviation/error from the output files, which were obtained from g mmpbsa. The energy liberated during the process of bond formation, or alternatively, the interaction between a ligand and protein, is shown in the form of binding energy. The lesser the binding energy, the better the binding of the ligand and protein. The final binding energy is a cumulative sum of van der Wall, electrostatic, polar solvation, and SASA energy. Except for the polar solvation energy, all other forms of energy contributed favorably to the interaction between different molecules Dasabuvir, Ribavirin, and Sofosbuvir and NSP-12. The bioactive molecule Dasabuvir showed the least binding free energy \\u221242.151 KJ/mol among all the selected molecules. A comparison of the binding free energies of all the complexes was made by plotting the binding energy versus time graphs. Further, we examined the contribution of each residue of NSP-12 in terms of binding free energy to the interaction with the selected. The contribution of each residue was calculated by decomposing the total binding free energy of the system into per residue contribution energy molecules using python script MmPbSaDecomp.py provided by MM-PBSA package. From the overall results, it was observed that the amino acid residue Trp 800 showed the lowest contribution binding energy for all the tree complexes, suggesting their significant role in binding indeed, contribution energy analysis of amino acid residues Tyr619, Cys622, Trp800, Glu811, Ser814, and Asp761 revealed that these amino acids played an important role during interaction, which confirms the docking result. Finally, the Sofosbuvir and Dasabuvir could be established as a repurposed drug against the COVID-19 main protease. Drug repurposing or repositioning is a term coined for the study of therapeutic uses of existing drugs that are either previously approved or currently under the investigational phase. Drug repurposing offers potential advantages over the discovery of novel drugs, and one is that it can be delivered quickly without lengthy assessments and time-consuming clinical trials. Due to this significance, drug repurposing offers an effective way to save lives and time during pandemics. Many drugs have been already suggested for the current pandemic caused by SARS-CoV-2, of which the most prominent are FDA-approved Chloroquine and its derivatives\\u2014an anti-malarial drug and Remdesivir. Remdesivir, a nucleoside analog, has been shown to significantly reduce the SARS-CoV-2 pathogenicity in in vitro study and is still under investigation. However, there is still a need for the identification of more drugs that could be further investigated for efficiency in reducing viral pathogenicity by using in vitro studies. These drug searches can be improved by employing computational methods such as in silico screening and structure-based drug targeting methods. In this study, we searched for homologs of NSP-12 of SARS-CoV-2 and evaluated its binding affinity with the FDA-approved antivirals, i.e., Sofosbuvir and Dasabuvir, which specifically target the evolutionarily conserved site of NS5B of HCV. The initial search for a homologous protein identified a total of 1105 structures by employing the DALI server. Further screening of these structures based on the quality of structural alignment Z-score \\u2265 19 resulted in 611 structures that belonged to four viral families, i.e., Coronaviridae, Flaviviridae, Picornaviridae, and Caliciviridae, responsible for RdRp activity during viral replication. The NSP-12 of SARS-CoV-2 displayed significant homology to the NS5 of Flaviviridae or NS5B of HCV as compared to the polymerases of Picornaviridae and Caliciviridae. Based on the pairwise alignment of amino acid sequences, NSP-12 shares 96% identity with SARS-CoV-1, 11% with HCV and HNV, and 10% with Poli. Despite sharing a lower identity based on sequence, NSP-12 has seven highly conserved motifs in the RdRp domain that play a key role in viral replication. These motifs have been previously reported to play essential roles in the viral replication of several viral families. Motifs A\\u2013C have highly conserved residues while motifs D-G have conserved structural orientation at the active site of the RdRp domain. The active site of RdRp was also identified as a suitable site for drug targeting based on the highest druggability score evaluated by using the CavityPlus webserver. The conserved residues and residues predicted from CavityPlus that were participating in ligand binding showed significant overlap when compared. The overlap included the conserved residues of motifs A, B, C, D, and F of the RdRp domain of NSP-12. These conserved residues have been also described previously to involve interactions with nucleoside, RNA primer attachment, and the RdRp activity. The active site of the RdRp domain of NSP-12 was further evaluated for binding affinity with FDA-approved antivirals by the molecular docking method. Molecular docking was performed using AutoDock Vina which revealed a significant binding affinity of NSP-12 with Sofosbuvir and Dasabuvir as compared to Ribavirin. Despite that, NS5B showed higher binding affinity lower binding energy as compared to NSP-12 however, the differences were still insignificant. Further, MD simulations were carried out to determine the time dependent structural stability and intermolecular interactions. The profiling of NSP-12 and ligand interaction revealed firm binding due to the formation of hydrogen bonds and hydrophobic interactions. The participating conserved residues of motif-A, -B, -C, -D, -E, and -F of the RdRp domain were involved in these interactions with the ligands, i.e., Dasabuvir, Sofosbuvir, as well as with their native competitors, i.e., GTP and UTP. Similarly, in the case of HCV as a receptor, the residues of motif-A, -B, -C, and -F were involved in interactions with the above-mentioned ligands. In general, both Sofosbuvir and Dasabuvir showed higher binding to the active site of NS5B of HCV and NSP-12 of SARS-CoV-2. We retrieved NSP-12 of SARS-CoV-2 with PDB ID 6M71 at 2.9 A\\u030a resolution from the Protein Data Bank PDB repository that was recently submitted by Gao et al. For the identification of homologous structures in the PDB repository, 6M71 chain A was submitted to the DALI server website.fi/dali/ accessed on 16 March 2020. The DALI server identifies and ranks homologous protein structures based on the quality of structural alignment in the form of a Z-score. The Z-score is a quantitative expression for comparing the intramolecular similarity between two homologous protein structures and its value \\u22652 is suggested to be significant. In this study, we used a Z-score cutoff of \\u226519. To obtain a similarity matrix and a cladogram based on structural similarity, the same server was employed by using an all against all analysis for a total of 53 representative homologs. The heatmap was visualized by using the Pheatmap v1.0.12 package in R and the cladogram in FigTree v1.4.4 software. The selected homologous proteins responsible for RdRp activity were analyzed for conserved sequences motifs and secondary structure by employing the PROMALS3D server. The resulting alignments were exported into ESPript 3.0 server for visualization website.fr accessed on 5 April 2020. To further study the conserved residues at the active site of NSP-12, multiple structures were aligned by using the MatchMaker algorithm implemented in the University of California, San Francisco UCSF Chimera v.14 and visualized. For pharmacophore modeling or assessing druggable cavities in 6M71-A, the CavityPlus server website.php accessed on 12 April 2020 was employed by submitting the structure file. The CAVITY search module of the CavityPlus server was used to detect the ligand-binding cavity and druggability was predicted based on the energy of residues and site geometry with default parameters. Those residues that were reported to be evolutionarily conserved in the RdRp domain as well as identified from druggable cavity search were visualized in the Venn diagram using Venny 2.1\\u2014an online server for plotting Venn diagram. AutoDock Tools v1.4.5 was employed for the ligand and receptor preparation. In general, we followed the Single-docking experiment with AutoDock Vina method by following the guidelines as suggested by Forli et al. for ligand-receptor docking. Initially, the Canonical SMILES IDs of the ligand compounds, i.e., Sofosbuvir CID 45375808, Ribavirin CID 37542, and Dasabuvir CID 56640146 were retrieved from the PubChem library and then imported in Chimera v.14 for converting into PDB files. Subsequently, the structures were prepared by minimizing the free energy and addition of hydrogen atoms. On the other hand, the receptor structures were acquired from PDB and then processed in AutoDock Tools by removing water molecules, isolation of a single chain, addition of charges, merging non-polar hydrogens, and assigning appropriate atom types. AutoDock Vina v1.1.2 was employed for molecular docking between ligand molecules and receptor. A grid box of size 48 \\u00d7 44 \\u00d7 54 A\\u030a was selected and centered at the druggable cavity of 6M71-A oriented at 117.4, 115.3, 129.5 A\\u030a, while the NS5B of HCV PDB ID 1YUY-A was considered as a control with the same size of the grid box used at the origin of 0.6, 80.0, and 64.2 A\\u030a. The parameters for molecular docking were used as described for running the single-docking experiment. Furthermore, we chose the default scoring function implemented in AutoDock Vina. The resulting score of molecular docking of NSP-12 of SARS-CoV-2 and NS5B of HCV were compared and plotted using a bar plot. For depicting protein\\u2013ligand interaction, LigPlot + v2.1 was used for characterization and visualization. The MD simulation of the apo and ligand-bound complex of NSP-12 protein was done using the GROMACS v.2021 and GROMOS96 54a7 force field. The systems were solvated in a cubic box of TIP3P water model and neutralized with the addition of counter ions. The neutralized systems were minimized using the steepest descent algorithm with a maximum force of 1000 KJ mol\\u22121 nm\\u22121 to adjust the geometry. To obtain further convergence, the minimized systems were equilibrated for in an NVT ensemble followed by NPT ensemble. Finally, a 100ns production run was executed, with coordinates and energies being saved in the output trajectory file at 10ps intervals. The RMSD, RMSF, radius of gyration Rg, and hydrogen bond contacts were used to determine the degree of convergence for all of the simulated systems. The molecular mechanic/Poisson\\u2013Boltzmann surface area MM-PBSA methods were used to calculate of binding free energy NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes. This calculation validates the thorough ligand\\u2013protein interaction investigation. The binding energy constitutes potential energy, polar and non-polar solvation energies, and they were calculated by the \\u2018g mmpbsa\\u2019 packages using GROMACS. The binding energy calculations in this method were calculated using the following equation The G binding represents the total binding energy of the complex, while the binding energy of the free receptor is the G receptor, and that of unbounded ligand is represented by the G ligand. Structural homology-based drug repurposing is a significant approach to exploring the homologous proteins which have been already targeted by either approved or investigational drugs. In this study, we searched for structural homologs of NSP-12 of SARS-CoV-2 and identified their relationship with the members of three viral families, i.e., Flaviviridae, Picornaviridae, and Caliciviridae. The RdRp domain of NSP-12 is highly conserved in terms of both sequence and architecture and assessed for binding affinity with FDA-approved antivirals, i.e., Dasabuvir and Sofosbuvir. These antivirals were initially approved for targeting the NS5B of HCV and serve as competitive inhibitors for binding to the active site that results in reducing viral pathogenesis. The results of this study revealed that the six conserved motifs motif A-F were involved in the formation of hydrogen bonds and hydrophobic interactions with the ligands. However, there is still a further need for in vitro analysis of these drugs to assess their efficiency before entering into clinical trials.\",\n          \"Tight and selective interaction between ligands and target proteins is of cardinal importance in drug discovery that various stages of identification and validation of targets and potential drug leads, as well as manifold preclinical and clinical trials, should be carried out until obtaining the final approval from the Food and Drug Administration FDA Abel et al., Campillos et al., Olayan et al. Accurate prediction of drug\\u2013target interactions DTIs plays a vital role for in silico and modern drug discovery in developing new drug candidates Olayan et al. Unfortunately, despite the promising advancements in genomics, proteomics, and systems biology, formidable scientific and regulatory obstacles such as high attrition rates, excessively time\\u2010consuming, and high\\u2010priced procedures have taken a heavy toll on proposing effective biologically active agents, and the success of drugs to pass clinical trials stage is about only 13% Fotis et al., Liu et al., Murray & Rees. Such attempts enable scientists to design energy harvesting materials and systems for the sake of higher efficiency and enlarging practical window Massetti et al., Safaei et al. Accordingly, computer\\u2010aided drug discovery CADD has revolutionized the history of drug discovery, especially regarding its substantial advantages Schneider & Fechner. These favors are a giving more information about the beneficial interaction between the drug and target and the availability of three\\u2010dimensional 3D structure, b providing a cost\\u2010efficient way to reduce failures for high\\u2010throughput screening HTS, c producing new ideas for rationally drug design, and d leading to rationally anticipate the targeted protein and candidate hits Bajorath, Bleicher et al., Ou\\u2010Yang et al., Sliwoski et al. These strategies are currently classified into three major categories ligand\\u2010based, structure\\u2010based, and chemogenomic approaches Imam & Gilani, as summarized in Figure. Most importantly, recent studies have illuminated the potential roles of CADD in designing new drug candidates for COVID\\u201019 treatment. Coronaviruses CoVs are cross\\u2010species viruses and can spread expeditiously into new host species, which in turn cause epidemic diseases. The whole members of the CoV family have a similar genome sequence, and the severe acute respiratory syndrome CoV SARS\\u2010CoV and Middle East respiratory syndrome CoV MERS\\u2010CoV are considered the two prototypical well\\u2010known CoVs Aghamirza Moghim Aliabadi et al., Seidi et al., Yin & Wunderink. The new CoV, identified as COVID\\u201019 and caused by SARS\\u2010CoV\\u20102, has a very similar genome sequence to SARS\\u2010CoV C. Wang et al. The lack of effective drugs led to considerable unwholesomeness and death rate of SARS\\u2010CoV\\u20102, especially in 2020 and 2021, when a novel coronavirus strain brought about a fatal respiratory illness. Since the outgrowth of the SARS\\u2010CoV\\u20102, more than 528 million people have been infected with up to 6.3 million deaths, as reported by the WHO. Although effective vaccines have amazingly reduced the COVID\\u201019 cases recently, they will not change the world right away, and importantly, their permanent efficacy, short\\u2010term and long\\u2010term side effects, and long\\u2010term protection against different variants of SARS\\u2010CoV\\u20102 are still under debate. These issues are accompanied by mass vaccination campaigns Wen et al. Furthermore, several studies have reported that vaccinated people can also be infected with novel coronavirus variants Largent & Miller. Therefore, the design and introduction of new drugs, either the current drugs or synthesized agents, for eradicating the SARS\\u2010CoV\\u20102 is of extreme concern. To accelerate this process, repositioning broad\\u2010spectrum antiviral drugs can be a potential strategy concerning the presence of the pharmacokinetic, pharmacodynamic, and toxicity data of these drugs Serafin et al. For instance, the efficacy of some antiviral drugs with previously known treatment of hepatitis, HIV, influenza, and Ebola viral diseases has been investigated. These agents, including adefovir, foscarnet, tenofovir, sofosbuvir, and uprifosbuvir, have phosphoramide and phosphate with various azole rings in their structures. By assessing their effectiveness against COVID\\u201019, remdesivir was the one that grabbed much attention and is believed to be the most successful drug for the therapy for combating COVID\\u201019 Brown et al., Gholivand et al. Further potential macromolecular targets of SARS\\u2010CoV\\u20102 are depicted in Figure. Therefore, finding the drugs that can target this enzyme may decrease the possibility of mutation\\u2010mediated drug resistance and show significant antiviral activity Pachetti et al. Thus, it is likely that CADD might be able to provide us with effective small molecules targeting the main receptors of SARS\\u2010CoV\\u20102 with fully therapeutical and preventional\\u2010oriented outlooks Muratov et al. The more recent update of FDA approval for COVID\\u201019 drug discovery shows that authorization of remdesivir inhibits the RNA\\u2010dependent RNA polymerase RdRp of the virus, molnupiravir inhibition of viral reproduction through enhancing the vast mutations in the replication of viral RNA and paxlovid 3c\\u2010like protease inhibitor for the treatment of mild\\u2010to\\u2010moderate COVID\\u201019. These drugs are used for various viral diseases such as hepatitis C, Ebola virus, influenza, etc. Vangeel et al. Molnupiravir and remdesivir acted on the RdRp enzyme employed by the virus to increase the transcription and replication capability of the viral RNA genome. While some groups reported the promising results of remdesivir, others showed a lack of efficacy, and therefore, the WHO did not suggest using it Mahase. Our focus in this paper is to explore the role of CADD on COVID\\u201019 most specifically, chemogenomics and DR, with potential examples of COVID\\u201019 drug discovery. Unfortunately, with boundless energy and time to discover a successful small molecule by utilizing various databases to inhibit the main protease Mpro of SARS\\u2010CoV\\u20102 via several virtual screening VS approaches, there are no effective drugs that have been approved. Only safety protocols such as wearing face masks, face shields, social distancing, and washing hands frequently are the best approaches to minimize the illness's danger and prevent transmission. Basically, the process of bringing an FDA\\u2010approved medicine to market for COVID\\u201019 drug discovery is extremely challenging, starting from the initial research of scientists for target identification and lead compound generation. After lead discovery and development of drug\\u2010like candidates and investigation of in vitro and in vivo results, the process will go into three phases, i.e., preclinical and, if successful, into clinical studies, and ultimately a marketed medicine after FDA approval Schaduangrat et al. It is where CADD is evolved as a valuable and powerful method to aid the challenging tasks of identifying small bioactive molecules and predicting targets. In 1981, an influential article appeared in Fortune magazine titled Next Industrial Revolution Designing Drugs by Computer at Merck. This article led to a realization of the importance of in silico studies in drug discovery and manifestation of CADD Sliwoski et al. Since then, high\\u2010throughput screening HTS has been increasingly used in the pharmaceutical sphere and academic institutions to the rapid discovery of hit and lead compounds. This potential and automated approach enables us to screen the entire compound library of either number of chemicals or biological compounds in a target\\u2010based assay or even a cell\\u2010based assay to obtain the desired biological response Bittker & Ross, Maghsoudi et al. It can spring to the mind that the stringent requirement of large resources of targets and ligands in HTS makes it expensive and time\\u2010consuming, and we should overlook the use of the useless ligands having no probability of showing success. But, the CADD technologies are very powerful and able to address this issue easily by employing virtual high\\u2010throughput screening vHTS that uses virtual compound libraries and allows experimentalists to focus on ligands which are more likely to have any activity of interest Leelananda & Lindert. In the first days of COVID\\u201019 outbreak, vHTS was employed and developed efficiently to investigate a group of approved drugs to identify the possible anti\\u2010SARS\\u2010CoV\\u20102 agents at the National Center for Advancing Translational Sciences NCATS of the National Institutes of Health NIH, embracing a broad spectrum of the SARS\\u2010CoV\\u20102 life cycle, such as viral entry, viral replication, in vitro infectivity, and live virus infectivity Xu et al. Many possible drug candidates were selected for targeting various stages of the SARS\\u2010CoV\\u20102 life cycle. For instance, cepharanthine, an anti\\u2010inflammatory compound, effectively showed ability to rescue the cytopathic effects CPEs of SARS\\u2010CoV\\u20102 Rogosnitzky et al., or corilagin, a possible antifibrosis, revealed potential against the angiotensin\\u2010converting enzyme ACE2 receptor\\u2010binding domain RBD Yang et al. Therefore, it is urgent to build a high bio\\u2010safety SARS\\u2010CoV\\u20102 infection model with a high ability for HTS of drugs against SARS\\u2010CoV\\u20102. This strategy just relies on ligand information\\u2014one of the broadly known and widely used ligand\\u2010based approaches in QSAR modeling. The advantages accruing from the QSAR methods in drug discovery and development are noteworthy. This computational technique attempts to construct mathematical models of compounds' physical and chemical properties associated with their chemical structure. QSAR studies are extensively employed to predict pharmacokinetic properties such as ADME and toxicity by building a training set of measured properties of known compounds and making a mathematical model of the encoded chemical structure, and also is useful in sensory applications as well Alam & Khan, Farshchi et al., Kholafazad\\u2010Kordasht et al., Nosrati et al., Seidi et al. Generating molecular descriptors from compound structures is also significant in assessing compounds' different physical, chemical, and topological features. After computing molecular descriptors, machine learning techniques play indispensable roles in the quality of QSAR predictions Popelier & Smith, Rad & Maghsoudi, Sliwoski et al. Using machine learning and QSAR, Muratov et al. identified 32 drugs and their 73 selected binary combinations for testing against SARS\\u2010CoV\\u20102. They found that the combinations of four of these drugs, including nitazoxanide, remdesivir, amodiaquine, and umifenovir, could synergize against SARS\\u2010CoV\\u20102. Interestingly, the combination of nitazoxanide and remdesivir showed a great synergistic interaction, and the combination of remdesivir and hydroxychloroquine exhibited a powerful antagonist role against SARS\\u2010CoV\\u20102 Bobrowski et al. The two proteins pp1a and pp1ab are quickly translated after entry into the host cells and further separated by two viral proteases, 3 C\\u2010like proteases 3CLpro and papain\\u2010like protease PLpro Amin et al. The latter is a critical virus enzyme responsible for producing a functional replicase complex and stimulating viral spread Shin et al. Amin et al. used chemical\\u2010informatics approaches to build a Monte Carlo optimization based on a QSAR data of various PLpro inhibitors, b VS to choose the best in\\u2010house molecules among 67 with the high likelihood of targeting against PLpro, and c eventually verification of these inhibitors by using receptor\\u2013ligand interaction investigation. Out of 67 compounds, 56 were anticipated as active from a QSAR model. Afterward, these 56 similar compounds were screened through SwissADME, and eventually, 13 compounds were approved for the ADME criteria. Furthermore, molecular docking has proved the potentiality of these candidates against SARS\\u2010CoV\\u20102 PLpro. In an interesting work, Tropsha et al. developed open\\u2010source data QSAR models of 42 inhibitors from DrugBank by employing QSAR models, docking, and similarity searching and utilized these models for virtual screening of all drugs in the DrugBank database. The 42 virtual hits were finalized for availability and cost using in\\u2010house ZINC Express software. The National Center for Advancing Translational Sciences NCATS analyzed 11 of total 42 hits which were potential compounds, and three of them, cenicriviroc AC50 of 8.9 M, proglumetacin tested twice independently, with AC50 of 8.9 M and 12.5 M, and sufugolix AC50 12.6 M. While molecular docking was unsuccessful to accurately distinguish between experimentally active and inactive compounds, QSAR was able to identify three inhibitors of SARS\\u2010CoV\\u20102 replication through Mpro inhibition. Overall, these results show that QSAR models proposed using SARS\\u2010CoV\\u20102 Mpro data can be useful to introduce compounds active against SARS\\u2010CoV\\u20102 Alves et al. This strategy relies more on target information. The structure\\u2010based drug design, also called target\\u2010based drug design, rests on the assumption that the bioactivity of a ligand comes from the physical interaction of the ligand with the corresponding target protein. The prerequisite of this approach is the 3D structure of the biological target, obtained from crystal structures, NMR data, or molecular modeling. Molecular docking is an established structure\\u2010based approach frequently used to predict the binding affinity and analyze the interactive mode between a ligand and a target/receptor. In other words, docking enables us to identify novel drug candidates by predicting ligand\\u2013target interactions at a molecular level Pinzi & Rastelli, Sargazi et al. There are several references in which the significant role and applications of docking in drug design and development have been discussed De Vivo & Cavalli, Keretsu et al., Kitchen et al., Meng et al. Furthermore, in the current spread of SARS\\u2010CoV\\u20102, molecular docking is important in screening potential drugs Keretsu et al., Lokhande et al., Marinho et al., Yu et al. The recent work of Khelfaoui et al. could be a good example of using molecular docking combined with MD to find promising candidate drugs against COVID\\u201019. They selected a library of 18 drugs that three of which include ramipril, delapril, and lisinopril, bound to angiotensin\\u2010converting enzyme 2 ACE2 receptor and SARS\\u2010CoV\\u20102/ACE2 complex well. ACE2 is one of the receptors in the human body with which the S proteins from coronaviruses have a great binding affinity and serve as the entry point into cells for SARS\\u2010CoV\\u20102. In fact, due to the long time required to discover specific antiviral agents or a vaccine, the best strategy would be to repurpose existing drugs to treat COVID\\u201019, which will be discussed in the DR section. Using VS and molecular docking, Abdellatiif et al. identified some promising antimicrobial and chemotherapeutic agents having coumarin and pyridine derivatives in their structures. In order to identify the most potential drug\\u2013target interactions, a molecular operating environment MOE program was carried out against the human ACE2, SARS\\u2010CoV\\u2010main peptidase, serine protease hepsin, human coronavirus papain\\u2010like proteases, and SARS\\u2010Coronavirus NSP12. As shown in Figure, among compounds a, b, c, d, and e with binding score energy values of \\u22126.08, \\u22126.48, \\u22126.78, \\u22127.08, and \\u22126.63, respectively, compound d had the binding score energy against the aforementioned targets. Thus, compound d needs to be additionally evaluated in clinical trials as a candidate for potential inhibitor of the SARS\\u2010CoV\\u20102 Abdellatiif et al. Hosseini et al. evaluated a virtual screening using docking of 1615 FDA\\u2010approved drugs to identify novel potential inhibitors for protease protein of COVID\\u201019. They showed that among all investigated FDA\\u2010approved drugs, simeprevir, a Hepatitis C virus HCV NS3/4A protease inhibitor, manifested high affinity to protease binding pocket and placed effectively into the binding pocket more beneficially than the lopinavir\\u2010ritonavir. This study suggested this compound as a potential inhibitor of COVID\\u201019 Hosseini & Amanlou. In another docking study, Motiwale et al. worked to identify potential inhibitors against Mpro of SARS\\u2010CoV\\u20102 from previously reported SARS\\u20103CL protease inhibitors where among 61 previously reported potential inhibitors, 4\\u2010benzoic acid and 4\\u2010 4\\u2010methoxyphenyl \\u20106\\u2010oxo\\u20102\\u2010 2\\u2010phenylethyl sulfanyl \\u20101,6\\u2010dihydropyrimidine\\u20105\\u2010carbonitrile revealed a minimum and maximum binding energy, respectively Motiwale et al. Table presents a selection of recent studies that researched the docking strategy of different pharmacological classes as treatments of SARS\\u2010Co and SARS\\u2010CoV\\u20102. Although many studies are carried out and many drugs are being tested to open a new road for effective treatment of COVID\\u201019, there are no fully beneficial antiviral drugs against the disease. Many of these studies are now in clinical trials to assess their safety and efficacy against infected patients. Chemical genetics, chemical biology, and chemogenomics chemical genomics are some cutting\\u2010edge applied fields in target and drug discovery Rognan, Ura\\u0301n Landaburu et al. Among the presented strategies, chemogenomics is an emerging discipline that principally investigates all molecules that can interact with any biological target. In humans, the number of the gene is about 26000, whereas the amount of various proteins is approximately more than 1 million Andersson et al. However, due to the extensive number of available molecules and biological targets, chemogenomics is limited to screening congeneric compound libraries against a selected target family. This alternative proposes an unprecedented chance to seek drug\\u2010like compounds with a noteworthy target or significant subtype Jones & Bunnage, Kubinyi. Chemogenomics is different from the traditional pharmaceutical methods due to linking the widespread software library directly to the methodology of modern genomics. In order to achieve this purpose, chemogenomics, along with a contribution of several chemical disciplines, including synthetic chemistry, combinatorial chemistry, cheminformatics, bioinformatics, and genomics, screen synthetic compound libraries for their impact on biological targets Bredel & Jacoby, Quinlan & Brennan. The ultimate goal of chemogenomics is the rapid screening as well as the rational development of target\\u2010specific chemical ligands to decrease the cost and enhance the drug discovery period, especially in the case of COVID\\u201019 drug investigation Jones & Bunnage. Basically, the main role of chemogenomics is unifying the discovery of ligands and targets by taking advantage of active compounds as probes to identify proteome functions. Similar to genetics, and if the operation of investigation arises from the target to phenotype, or vice\\u2010versa, there are two concepts a forward and b reverse. In \\u2018forward chemogenomics\\u2019, the ligand is discovered through their phenotypic condition that could affect the entire biological system instead of the base of their mechanism of suppression of a certain desired protein. In other words, the molecular basis of the phenotype interest is unknown and phenotypic screening is carried out in cells of organisms having a single\\u2010cell or multicellular utilizing a group of ligands. On the other hand, in reverse chemogenomics, the desired sequence of genes is cloned and explicit at the first step as target proteins afterward, they are preselected in a high\\u2010throughput screening HTS \\u2018target\\u2010based\\u2019 way by a panel of compounds. Here, the assays are normally classified as i organismal assays, i cell\\u2010free, and ii cell\\u2010based organismal assays. Cell\\u2010free assays are generally straightforward but precise and extremely automated. They are comprehensive binding trials in which manifold compounds are examined concurrently for their binding affinity toward a variety of particular goals. Here, DTIs are clearly recognized in the absence of confounding variables. However, in cell\\u2010based and organismal assays, chosen compounds are delivered right to cells or organisms in vitro to select hits within a relevant cellular context. However, the hits need further mechanistic characterization due to simultaneously interacting with multiple targets. The former approach is usually employed in reverse chemogenomics, whether target information is at hand, while organismal assays and cell\\u2010based are mainly employed in forwarding chemogenomics to test a wide variety of compound effects on whole biological systems Bredel & Jacoby. In forward chemogenomics, cells or organisms are distributed in multi\\u2010well cell culture plates for incubation. Subsequently to the incubation step, an aliquot from donor plates is taken to the recipient and mock plates mock treatment, in which the ligand\\u2013target binding assay is performed. The endpoint of most cell\\u2010based HTS assays is a spectrophotometric measurement in a spectrophotometer equipped with suitable software that calculates the final data, including SAR analysis and in silico QC. Then, active compounds that obtain the intended changes in the phenotype are chosen to elucidate their targets at the molecular level. This strategy could be performed using affinity chromatography, microarray, phage display, or transcriptional or proteomic profiling. In reverse chemogenomics, the main focus is to elucidate the protein families and the gene. In this regard, the sequencing of the target gene demonstrates a specific level of homology that is transfected in a host cell. These family targets are sequentially gathered, purified, and subjected to trial design. Finally, similar to forwarding chemogenomics, the measurement data on the spectrophotometer are directed to a suitable data analysis system. The biological target is extracted from the existent information in the biologically annotated chemical database in the target fishing approach Huang et al. However this procedure needs an abundant effort from a time, cost, and resource perspective to identify all plausible chemical target associations thus, computational target prediction techniques are strongly being pursued with the contribution of advanced data\\u2010mining algorithms Anwar et al. This current mode of action or the computational target approach is classified into four classes data\\u2010mining, pharmacophore searching analogy searching, analysis of bioactivity spectra, and molecular docking. In the past, these methods were applied to investigate new compounds of known targets, but recently, they have been applied to predict novel targets for the known compounds via similar approaches Campbell & Marchant, Ekins et al. Target\\u2010based or receptor\\u2010based chemogenomics refers to the discovered biologically active compounds with phenotypically readout information. Nevertheless, no existing target data thus, by applying the chemogenomics strategy, the hypotheses can be generated of the target by comparing the new biological compounds to the same known target Fourches et al., Weston et al. For example, the target receptors or proteins that are necessary for replication and survival of SARS\\u2010CoV\\u20102 are viral proteases involving papain\\u2010like protease PLpro, the 3\\u2010chymotrypsin\\u2010like protease 3CLpro, and Mpro, RNA\\u2010dependent RNA polymerase RdRp complex, the spike S receptor binding glycoprotein, the nucleocapsid N protein, nonstructural protein NSP, and the membrane M protein including transmembrane TM as well as small envelope E protein Cavasotto et al. As an example of this part, it can be referred to the work of Mishra and his coworkers that reported an inclusive computational approach for the identification of drug molecules that are multi\\u2010targeted struggling with the SARS\\u2010CoV\\u20102 proteins. These drugs are essentially included in virus replication inside the host body, the viral\\u2013host interaction, disease development, and transmission of Sars\\u2010CoV\\u20102 infection. They screened 72 potential antiviral drugs approved by the FDA against the S glycoprotein, hACE2, 3CLpro, Cathepsin L, N protein, RdRp, and NSP6 proteins. Furthermore, they investigated the molecular interaction by applying free energy landscape, simulation of Molecular Dynamics, and binding free energy estimation using MM\\u2010PBSA. Their finding showed favorable results for seven drugs, among which catechin flavan\\u20103\\u2010ol can effectively bind to 3CLpro, Cathepsin L, RBD of S protein, NSP\\u20106, and Nucleocapsid protein to be used as a potential multi\\u2010targeted agent in combating SARS\\u2010CoV\\u20102. However, due to the limited timeframe of combating this pandemic, it is not realistic to be assured of finding a definitive therapy for combating the pandemic. Hence, vigorous attempts are still continuing for the DR as a clinical treatment strategy to fight COVID\\u201019 Mishra et al. In another research, Omotuyi et al. investigated molecular docking protocols for fast screening of the FDA database for high\\u2010affinity interaction of the resolved 3D structure of SARS\\u2010CoV\\u20102 receptor\\u2010binding domain RBD in complex with its receptor hACE\\u20102 interface. Ubrogepant, as one of the preferable candidate drugs, was studied utilizing the atomistic molecular dynamics simulation method. They reported that UBR breaks the interaction at the RBD/ACE\\u20102 interface residues of SARS\\u2010CoV\\u20102 RBD/ACE\\u20102 complex that might result in protein function losses with direct inference on the oligomerization in RBD and function loss in ACE\\u20102. Therefore, leading to a binding and cellular receptor recognition unattainable. Therefore, UBR exhibits a novel therapeutic candidate in combating SARS\\u2010CoV\\u20102 since it binds to proportionally high affinity with free RBD, ACE\\u20102 receptor, and SARS\\u2010CoV\\u20102 RBD/ACE\\u20102 complex based on only binding affinity calculations Omotuyi et al. The other studies in this area are demonstrated in Table. A primary example of the beneficial application of chemogenomics is DR, which accelerates drug discovery by identifying novel clinical targets for the available approved drugs. This method possesses the distinctive merit of rapid clinical development without bearing the long\\u2010lasting and expensive procedures included in drug discovery. Therefore, according to the advantages of these proposed drug discovery methods, many COVID research orientations have been focused on DR Galindez et al., R. Liu et al. Computational DR applies available drugs for new or existing diseases that do not have an absolute cure, while the available drugs were not originally designed for the novel disease treatment. Furthermore, utilizing the enormous volumes of available omics information in digital form in conjunction with in silico screening intrigued the researcher to use this appropriate ground to significantly increase the speed of shortlisting promising candidates in reply to spreading diseases like SARS\\u2010CoV\\u20102. To be exemplified, Lucchetta, M. and M. Pellegrini reported a methodology for preclinical computational DR called DrugMerge, and it is based on combining several drug rankings gained via a set of Disease Active Subnetwork construction algorithms. DrugMerge utilizes differential transcriptomic data from drug perturbation assays and cell lines/tissues of patients affected by the illnesses on the subject of a considerable gene co\\u2010expression network. Application of DrugMerge to COVID\\u201019 shows that DrugMerge can mimic human expert judgment Lucchetta & Pellegrini. Rima Hajjo and Alexander Tropsha developed a systems biology workflow for drug and vaccine repurposing. They reported that the identification of Bacillus Calmette\\u2013Gue\\u0301rin BCG and small molecules of BGC mimics involving antiviral drugs raltegravir, emetine, and lopinavir as high confidence hits are capable of boosting immunity and preventing emerging new viruses. Their results approved that BGC impacted the generation and maturation of nai\\u0308ve T cells, resulting in increasing long\\u2010lasting trained intrinsic immune replies to combat novel viruses, including SARS\\u2010CoV\\u20102 in vitro as a protective measurement from the lethal consequences Onozuka et al. Due to the lengthy development cycle of novel drugs, DR turns into an efficient approach to screening drugs of potential therapeutic use for SARS\\u2010CoV\\u20102. Che et al., in another attempt, screened potential therapeutic drugs for COVID\\u201019 based on a knowledge\\u2010graph\\u2010based DR method. In their experiment, five drugs predicted by the models exhibited efficiency in clinical treatment. The experimental outputs prove that the model can predict drug\\u2013disease interaction effectively for normal diseases and SARS\\u2010CoV\\u20102 Lv et al. Therefore, DR presents a constant source of novel knowledge in virus biology and molecules with previously undefined antiviral properties, which can be applied as molecular tools in unknown molecular mechanisms of virus replication and pathogenesis Mercorelli et al. Further investigation of SARS\\u2010CoV\\u20102 in related research is summarized in Table. A fundamental role of chemogenomics is predicting the activity profile of the compounds for a group of targets that can be implemented by using ligand\\u2010based or target\\u2010based approaches. The ligand\\u2010based philosophy is underlying that molecules sharing enough similarity to available biologically defined ligands have an increased probability of sharing the same biological profile. Therefore, the efforts are devoted to organizing chemical libraries with biological information, including in vitro affinity, targets, and toxicology Harris & Stevens. As a result, designing chemical libraries is of great interest for a certain individual target or target family, which is attempted via utilizing machine\\u2010learning methods. Herein, concentration is on protein\\u2013ligand interaction via applying physics\\u2010based protein\\u2010ligand docking on machine learning. Computational chemogenomics or quantitative multiple structure\\u2013activity relationships QMSPR modeling consists of the compound\\u2013protein interaction space modeling of large datasets at once, usually for drug discoveries, where the available techniques chiefly incorporate escalating numbers of bioactive samples or concentrate on individual subfamilies of proteins and ligands Cheng et al., Xue et al. However, the 3D information of protein targets is needed most of the time. Thus, the application of precise target data in chemogenomics is confined to the explicitly known X\\u2010ray structures or structures that homology models can evaluate. As a result, many QMSAR is centered on ligand\\u2010based studies or similarity principles Gawriljuk et al. In this context, compounds of phenotype\\u2010based preselection possess merits in discovering target\\u2010based drugs due to unclimbable and missing perceptions of the mechanism of action of the drugs. Therefore, a handful of methods can carry out new chemical compound screening which is phenotype\\u2010based. In this regard, Thai\\u2010Hoang Pham et al. extended a sound learning framework for a high\\u2010throughput mechanism\\u2010driven neural network\\u2010based method for phenotype compound screening. They applied this framework for COVID\\u201019 DR. Their proposed novel data escalating technique extracts beneficial information from untrustful assays in the dataset of L1000. Furthermore, they showed the value of DeepCE by applying it to the DR of COVID\\u201019 and produced novel lead compounds that represented clinical evidence. Therefore, DeepCE can provide a potentially strong framework for rapid predictive modeling via using noisy omics data and screening new chemicals to modulate an organized response to illnesses Wang et al. A computational approach for increasing the rate of development of drugs and comprehension of the mechanism of action of small molecules is the identification of drug\\u2013target interaction DTIs. Lee and his coworkers represented an AI\\u2010DTI, a new method for predicting activation and inhibition of DTIs, by combining the mol2vec and genetically perturbed transcriptomes. Their finding showed that this methodology was a success in discovering about half of the DTIs for drugs that are employed for treating COVID\\u201019 Lee et al. A myriad of compounds represent potential drug candidates, but merely a few of them will be formulated, approved, and registered as a drug for individual health issues which investigation for that compound will be a long, intricate, and costly process, while sometimes there is an urgent need for developing a drug like COVID\\u201019 pandemic Muratov et al. Table demonstrates chemogenomics related to COVID\\u201019 research. To conclude, chemogenomics is an emerging interdisciplinary field at the interface of chemistry, biology, and informatics which decreases the cost and time of drug discovery dramatically via three main approaches, including target fishing, predicting the bio\\u2010active profile of drugs, and DR. Herein, of much current importance is chemogenomics, is in COVID\\u201019 drug discoveries, which this review paper highlighted specifically. Drug repositioning, also known as drug repurposing, drug redirecting, drug reprofiling, and therapeutic switching, is defined as identifying new indications and therapeutic targets for existing drug compounds Lotfi Shahreza et al. DR has received significant attention caused of its crucial role in finding and developing new therapies for complex, orphan, and chronic diseases with unknown or less effective treatments Shameer et al. Since developing a new drug in the traditional drug discovery process is time\\u2010consuming and takes approximately between 10 to 15 years to enter the market, repurposing existing drug compounds is capable of overcoming these shortcomings, and this type of research has been growing drastically Kumar et al. Therefore, DR plays a versatile role in developing new drugs because it can reduce the costs of the whole process and shorten the development pathways of approved or investigational drugs Pushpakom et al. DR works with two main approaches. The first is that a drug can bind to various targets simultaneously, which is good for searching novel target sites of action for the known biological agents, and the second is that targets related to a disease are often associated with different biological processes of pathogenesis, which is important for choosing of a new indication for the known target Badary, Kamel et al. These are indispensable in proposing new drug candidates for unprecedented diseases like COVID\\u201019. It has emerged as an alternative way of accelerating the process of developing new treatments against COVID\\u201019 Yu et al. Based on data brought from PubMed, the number of works of literature on DR has been growing dramatically, and this number has increased since the outbreak of SARS\\u2010CoV\\u20102 Galindez et al. It should be noted that polypharmacolgy and DR are inextricably interrelated with each other Cheng. DR has become an influential tactic from both academia and pharmaceutical industries as an alternative strategy to traditional drug discovery that is related to when new therapeutic applications are pinpointed for existing drugs Karakida et al., Turanli et al. Compared to traditional de novo drug discovery and development, this approach provides a new and fast way for drug development and reduces risk, particularly for drugs that cannot pass phase I clinical trials. It is able to reprocess the information on known drugs such as pharmacokinetics and safety for humans and manufacturing pathways, resulting in reduced time and expenses in traditional de novo drug discovery Sawada et al. There are a wide variety of examples of repositioned drugs. Sildenafil is considered the most prominent one, beginning with cardiovascular treatment followed by accidentally treating erectile dysfunction clinical trial and later in low dosages, treating pulmonary hypertension Ghofrani et al., Prasad et al. Furthermore, as repurposed drugs pass the preliminary clinical trials phase I or safety level, the time needed for drugs to reach the market can be reduced significantly. Hence, scientists can identify the unintended secondary targets through CADD. The history of discovering the effect of an established drug on new indications dates back to World War II, since considerable improvement had been observed for cutaneous rashes and arthritis in soldiers who were under treatment with an antimalarial medicine, Chloroquine Schlitzer. Chloroquine is a well\\u2010known antimalarial drug that was synthesized in 1934 and later on demonstrated successful various therapeutic effects, including anti\\u2010inflammatory/immunomodulating, anti\\u2010infective, antithrombotic, and metabolic. Furthermore, antitumoral properties make it available for the treatment of a number of cancers Plantone & Koudriavtseva. In 1942, Marcel Jonbon used sulfonamide antibiotics for typhoid treatment, but he faced hypoglycemia as an unexpected side effect in some patients therefore, sulfonamides presented the role of an anti\\u2010diabetic drug Quianzon & Cheikh. Aspirin is one of the most famous DR examples with an original indication as an analgesic drug that for the first time repositioned as an antiplatelet aggregation drug in 1980 and a second indication still is commonly used for cardiovascular problems Jourdan et al. Furthermore, several types of research have been conducted on the anticancer effect of aspirin such as colorectal cancer Drew et al., lung, and breast cancer Li, Hu, et al. The notable potential and extensive applications of DR in novel therapies are entirely apparent and accordingly, the number of acceptable research studies which have been done or are under processing are growing every day. There are various classifications for DR methods which typically can be categorized into computational approaches and experimental approaches Park. Experimental approaches refer to testing drugs in evaluations based on available comprehensive clinical compound databases in which the possible indications of drugs can be discovered by target\\u2010based or cell\\u2010based analysis Luo et al. There are some hurdles for experimental identifications because they are mostly expensive and time\\u2010consuming, requiring a set of marketed drugs, technical equipment, and also screen tests. In contrast to experimental methods, computational drug repositioning methods which are mainly data\\u2010driven utilizes databases and bioinformatics tools to find out interactions between drugs, targets, and diseases Pillaiyar et al. Analyzing a large type of data like chemical structures, gene expression, genotype or proteomic data, and electronic health records EHRs provides the possibility of understanding novel and unexpected functional gene interactions and unclear drug response or mechanisms of disease Pushpakom et al. Since there is no specific and registered therapy to cure the infection of this virus, finding effective therapeutics by applying existing drugs is a pressing exigency and established strategy Galindez et al. Concerning the time\\u2010consuming regulatory process of the newly approved drugs, the DR strategy strongly utilizes the identification of new treatment methods by presenting FDA\\u2010approved drugs. In addition, many repurposed drugs with acceptable results are accomplished by computational methods Wang & Guan. There are myriad reasons why we need to go through DR for different diseases like COVID\\u201019 if we intend to accelerate the process of developing a new drug. These reasons can be i availability of marketed drugs and their excipients and APIs for formulation and distribution, ii the likelihood of taking advantage of a synergistic effect of various drugs with each other, iii cheap and less time\\u2010consuming process for drugs to be marketed and IV emergence of new therapeutic pathways for former drugs and new categories of medicine Chung et al. Therefore, repositioned drugs have very important impacts on COVID\\u201019 drug discovery. Therefore, different categories of drugs are being repositioned in in vivo and in vitro studies to treat SARS\\u2010CoV\\u20102 Wu et al. These potential agents are namely Ebola virus remdesivir, antimalarial chloroquine, anthelmintic and antiprotozoal drugs niclosamide, diarrhea nitazoxanide, antiviral lopinavir and ritonavir, antineoplastic carfilzomib and bortezomib, and antibacterial azithromycin. In this regard, the WHO supported repositioning remdesivir, lopinavir/ritonavir, and chloroquine/hydroxychloroquine as a potential for COVID\\u201019, although most might be only supportive symptomatic Won & Lee. Nevertheless, they will likely be approved for COVID\\u201019 therapy if they pass clinical trials and are published in literature and guidelines from leading health organizations Nain et al. In order to do this, many groups have been attempting enthusiastically to discover the first approved drug for COVID\\u201019 therapy Erlanson, Low et al., Muratov et al. As a result, we identified several clinical trials, in which 18 studies were at clinical phases 2, 3, or 4. Luo et al. investigated the free energy perturbation\\u2010absolute binding free energy FEP\\u2010ABFE method and bioassay validation to reposition 25 potential therapeutic agents to inhibit Mpro. By using the viral proteinase crystal structure, Mpro PDB ID 6 LU7, up to 2500 FDA\\u2010approved drugs were first screened via Glide's molecular docking program. They found that Cys145\\u2010His41/Ser144\\u2010His163 could be the nucleophilic agent and acid responsible for hydrolysis of the substrate proteins, and Gly143 and Gln166 can form a hydrogen bond with the CO\\u2010NH\\u2010C \\u2010CO\\u2010NH\\u2010C structure of the backbone of the substrate protein. Of 25 possible agents, 15 showed strong inhibitory properties against Mpro. Among these agents, additionally, dipyridamole passed its first\\u2010round clinical trials with considerable potencies Li, Hu, et al., Li, Li, et al. The network\\u2010based methods have also given satisfying results in investigating the DTIs and CoV\\u2013host interactions. Zho et al. introduced 16 possible anti\\u2010HCoV drugs such as melatonin, mercaptopurine, toremifene, mesalazine, and sirolimus from more than 2000 FDA\\u2010approved drugs by applying a combination of network\\u2010based approaches and systems pharmacology\\u2010based network medicine platform with the ability to quantify the interactions between the drug targets and virus\\u2013host interactome in the human protein\\u2013protein interactions network. Zhou et al. The docking screening method was applied for a set of 2201 approved drugs to find inhibitory activities at SARS\\u2010CoV\\u20102 main protease, and based on that, carfilzomib, eravacycline, valrubicin, lopinavir, and elbasvir have nominated as an inhibitor for SARS\\u2010CoV\\u20102 main protease Wang. Table includes recent studies in terms of computational DR for coronavirus drug discovery and various diseases, respectively. Various transcriptomic datasets in relation to SARS\\u2010CoV\\u20102 were reported since the COVID\\u201019 outbreak. Basically, the idea is that for a specific disease signature composing of a class of up and downregulated genes, if there is available drug candidate, where those same classes of genes are alternatively downregulated and up\\u2010regulated, respectively, this drug might be potential for that disease Hoffmann et al., Le et al. Le et al. investigated existing computational drug repositioning with remarkably reversed differential gene expression in comparison with manifold input signatures for SARS\\u2010CoV\\u20102 effects on human cells. They tried to predict the potential drugs toward differentially expressed gene sets from both cell line and organoid disease models and human samples. They found 102 compounds, and among them, 25 were selected in at least two of the signatures, and some of them have been already analyzed in clinical trials. They eventually tested 16 of their shortlisted candidates in live SARS\\u2010CoV\\u20102 antiviral assays. If a drug could notably FDR < 0.05 reverse the disease signature, then the drug could be a potential inhibitor for the SARS\\u2010CoV\\u20102. In the end, they found that 16 of their high predicted hits showed antiviral assay in live SARS\\u2010CoV\\u20102. Four potential inhibitors were tested for inhibition of SARS\\u2010CoV\\u20102 in a human lung cell line, Calu\\u20103, infected with SARS\\u2010CoV\\u20102. Three drugs revealed in vitro antiviral efficacy\\u2014bacampicillin, clofazimine, and haloperidol with no toxicity effects Le et al. Ge et al. used an integrative drug repositioning analysis through machine learning to integrate and mine large\\u2010scale knowledge graph, literature together with transcriptome data to identify the potential compounds against SARS\\u2010CoV\\u20102. They chose to work on CVL218, a poly\\u2010ADP\\u2010ribose polymerase 1 PARP1 inhibitor. Their in vitro data showed an efficient inhibitory effect against SARS\\u2010CoV\\u20102 replication without a clear cytopathic effect. They further revealed that CVL218 can effectively bind to the nucleocapsid N protein of SARS\\u2010CoV\\u20102 and could suppress the LPS\\u2010induced production of many inflammatory cytokine, which are highly related to the prevention of immunopathology induced by SARS\\u2010CoV\\u20102 infection Ge et al. Another PARP1 inhibitor, PJ\\u201034 was previously studied for acute inflammatory including inflammatory diseases of the lung. It was observed that PJ\\u201034 can interact with the N\\u2010terminal domain of the coronavirus nucleocapsid N protein and decrease the RNA binding ability to inhibit virus replication Ge et al. Beclabuvir is an antiviral drug for the treatment of hepatitis C virus HCV infection studied for inhibitory activity against SARS\\u2010CoV\\u20102. It was reported that beclabuvir could be a potential compound for the treatment of SARS\\u2010CoV\\u20102 through virtual screening and molecular docking in order to bind to RdRp of the SARS\\u2010CoV\\u20102 Talluri. Raltegravir belongs to a class of HIV integrase inhibitors used in the treatment of HIV\\u20101 infection. It obstructs the insertion of HIV\\u20101 DNA into the host cell genome. Raltegravir was selected as a highly effective drug against Mpro and has been repurposed for the potential treatment of COVID\\u201019 in many studies Nabi et al. There has been considerable progress in the practical development and the rational application of computational drug discovery, giving rise to a paradigm shift in both industry and academics. CADD is a powerful approach for proposing new drug candidates, particularly when used in tandem with current chemogenomic methods. Even though CADD takes advantage of manifold dilemmas and approximations, this knowledge\\u2010driven approach plays a significant role in the drug discovery process because of its ability to fast\\u2010track drug discovery. Since CADD has provided several achievements, there would be a promising future to aid drug discovery of many more medicines in the future. Chemogenomics is a superior interdisciplinary field at the interface of chemistry, biology, and informatics, which dramatically decreases the cost and time of drug discovery. This technique allows sequential genetics that is further interpreted into potential biological targets. Over the last years, there has been great attention to DR in pharmaceutical and biomedical companies, highlighting the paramount importance of repurposing. This interest has led to victorious launches in new, highly economical, and appealing indications compared to traditional drug discovery with high\\u2010priced drug development. Over the last 3 years, the COVID\\u201019 pandemic has killed many people globally since there are no effective drugs against it. In this sense, we need to highlight that DR attempts the search for effective drugs in an urgent situation such as the COVID\\u201019. Utilizing DR is strongly recommended to continue investigating the repositioning of existing drugs against different diseases. However, how effectively CADD, including DR and chemogenomics, will produce feasible treatments across different disease areas remains to be seen. Therefore, it should not be surprising that success is limited at times. Several basic computational problems still need to be resolved. To ameliorate drug design and development dilemmas, rapid developments in chemical and structural biology, bioinformatics, clinical experiments, and computational technology are essential. For example, high\\u2010throughput screening HTS has been increasingly used in the pharmaceutical sphere and academic institutions to discover hit and lead compounds rapidly. This potential automated approach lets us screen the entire compound library of the number of chemicals or biological compounds in a target\\u2010based assay or even a cell\\u2010based assay to obtain the desired biological response. In order to use HTS against SARS\\u2010CoV\\u20102, it is important to take advantage of a similar homology of virus in genomics, specifically, chemogenomics, as a versatile way of drug screening model that provides better bio\\u2010safety. Hitherto, CADD is the only way for screening drugs against COVID\\u201019 and is based on a single viral protein or small\\u2010scale drug screening of formerly FDA antiviral drugs utilizing wild\\u2010type SARS\\u2010CoV\\u20102. Nevertheless, this approach is not well\\u2010suited to HTS when focusing on the bio\\u2010security requirements of SARS\\u2010CoV\\u20102. Moreover, developing new drugs takes too much time without having a precise crystal structure of the viral target Liu et al. Thus, the best way to rapidly find a drug for the treatment of COVID\\u201019 would be the use of CADD methods in order to discover lead compounds for clinical use for instance, we can build a program that takes advantage of structure\\u2010based drug design, HTS, and virtual drug screening to identify new leads targeting the Mpro. More recently, a library of 10000 compounds involving approved drugs, drug candidates in clinical trials, and biologically active compounds to inhibit Mpro was examined. As a result, six compounds were found with a great IC50 against Mpro and strong antiviral activity in cell\\u2010based assays. Moreover, using CADD, the crystal structure of Mpro in complex with Michael acceptor inhibitor known as N3 can be determined Jin et al. It is also possible to identify drug leads against SARS\\u2010CoV\\u20102 by targeting the 3\\u2010chymotrypsin\\u2010like cysteine protease 3CLpro enzyme performing a vital role in the life cycle of CoVs, including MERS\\u2010CoV and SARS\\u2010CoV Ul Qamar et al. Scientists employed vHTS tools to predict the 3D structure of the SARS\\u2010CoV\\u20102 3CLpro enzyme by constructing a 3D homology model and screening it against 32297 potential antiviral phytochemicals/traditional Chinese medicinal compounds of a medicinal plant library. This effort led to the discovery of nine hits that need further optimization to be identified as anti\\u2010SARS\\u2010CoV\\u20102 leads. In addition, an interesting investigation based on an HTS assay employing a compound library of 2000 approved drugs and pharmacologically active compounds has also been carried out to find broad\\u2010spectrum inhibitors against the replication of CoVs Shen et al. From their HTS data, 56 hits were screened, and 36 compounds were validated in vitro using genetically engineered human CoV OC43 HCoV\\u2010OC43. Consequently, seven effective compounds were selected as broad\\u2010spectrum inhibitors in which lycorine protected BALB/c mice against lethal human coronavirus OC43 HCoV\\u2010OC43, and emetine blocked MERS\\u2010CoV entry following pseudo\\u2010virus entry assays. These studies prove the efficacy of screening strategies in the rapid discovery of drug leads, resulting in new drugs for infectious diseases having no effective drugs or available vaccine. No funds were received for this article.\",\n          \"Coronavirus disease 2019 COVID-19, which was first identified in December 2019, and shortly after, declared a pandemic by World Health Organization WHO. As of January 18, 2022, there have been more than 326 million confirmed cases and 5.54 million deaths globally. Coronaviruses belong to the family of Coronaviridae, RNA viruses with crown-like spikes on the surface of the coronavirus particles. According to a meta-analysis of Macedo et al., the mortality rate of COVID-19 was 17.1% for patients admitted to hospitals, whereas WHO estimated a fatality rate of 6.73%, which was much lower than that calculated from published studies. Among the critical cases of COVID-19, the mortality rate reaches 40%. Substantial efforts have been made in the treatment of patients with COVID-19. The WHO recommendations in the treatment of COVID-19 are as follows molnupiravir conditional, baricitinib strong, ruxolitinib and tofacitinib conditional, sotrovimab conditional, casirivimab and imdevimab conditional, IL-6 receptor blockers tocilizumab and sarilumab strong, remdesivir conditional, and systemic corticosteroids strong. The WHO recommends not to use ivermectin, lopinavir/ritonavir, hydroxychloroquine, and convalescent plasma. The pathogenesis of COVID-19 was explained by cytokine storm, reduction in ACE2 expression, and activation of complement pathway-induced microvascular injury and thrombosis. The mechanisms of the recommended agents are focused on the mentioned pathogenesis to improve the clinical outcome of COVID-19, and antiviral therapies are missing. The anticoronaviral strategies include preventing the synthesis of viral RNA, inhibiting virus replication, blocking the virus binding to human cell receptors, or inhibiting the viruses' self-assembly process. The SARS-CoV-2 contains at least four structural proteins spike S protein, envelope E protein, membrane M protein, and nucleocapsid N protein, and 16 nonstructural proteins NSPs. Among the translated NSPs, the main protease, also called chymotrypsin-like protease 3C-like protease, and the papain-like protease are two essential proteases for proteolytic processing of the coronavirus replicase polyprotein, therefore generating functional replication complex of the virus, whereas RNA-dependent RNA polymerase is the central enzyme for RNA synthesis. These three NSPs play crucial roles in coronavirus replication, making them attractive targets for anticoronaviral drug design. The S protein, a surface-located trimeric glycoprotein of coronaviruses, promotes the attachment of viruses to host cells through binding to angiotensin-converting enzyme 2 ACE2 and virus-cell membrane fusion during viral infection. Thus, the S protein has been considered as a major target for the development of vaccines and drug. The development of a new therapeutic agent is a complex, lengthy, and expensive process, which can take 2\\u20134 years of preclinical development and 3\\u20136 years of clinical development and over 500 million dollar cost. There are three critical steps to develop a new drug including discovery and development, preclinical research, and clinical development. Drug discovery involves screening hits, medicinal chemistry, and optimization of hits to reduce potential drug side effects. For drug discovery, two different complementary approaches can be applied classical pharmacology, also known as phenotypic drug discovery, which is the historical basis of drug discovery, and reverse pharmacology, also known as designated target-based drug discovery. Screening methods based on phenotypic drug discovery have been used to discover new natural products mainly from the terrestrial origin. These two strategies have advantages and disadvantages and promote very different screening assays. The frequent re-discovery of the same compounds, the technical difficulties associated with the isolation of compounds from extracts, and the incompatibility of natural product extracts with high-throughput screening HTS campaigns were the disadvantages of phenotypic drug discovery. On the other hand, natural product structures have the characteristics of high chemical diversity, biochemical specificity, and other molecular properties that make them favorable as lead structures for drug discovery, which serve to differentiate them from libraries of synthetic and combinatorial compounds. Overly simplified assays, acting of drug on more than one target, the multifactorial nature of diseases, and challenges to identify a single molecular target are some limitations of target-based drug discovery. Therefore, a comprehensive screening strategy will incorporate both targeted and phenotypic assays, with one format designated as the primary screen and the other as a secondary or follow-up assay. During the spread of COVID-19 outbreak, great efforts have been made in therapeutic drug discovery against the virus. Because COVID-19 is a new, acute, severe infectious disease, the anti-SARS-CoV-2 drug development strategies are to screen existing drugs to identify potentially effective drugs, to expand indications, and to develop a vaccine. The safety of conventional drugs has been mostly verified if effective, they can be quickly applied in clinical practice repurposing of existing drugs. The recent rises of several high transmissible strains sounded alarms for currently used vaccines and drugs. Therefore, developing broad-spectrum antiviral drugs not only to combat COVID-19 but also to provide protective arsenals against future viral outbreaks is a requirement. Scientists continue the development of broad-spectrum antiviral drugs from natural or chemical sources, which have the potential advantages of broad-spectrum therapeutic effect and insensitivity to viral evasion. Given the urgency of the SARS-CoV-2 outbreak, here we discuss the discovery and development of new therapeutics for SARS-CoV-2 infection based on the strategies from preclinical drug discovery. We report this systematic review based on the Preferred Reporting Items for Systematic Reviews and Meta-Analysis PRISMA guidelines. Studies published in PubMed, Scopus, Web of Science, EMBASE, Google Scholar, and DrugBank were searched from December 2019 to March 2021 using the following search terms Coronavirus, Covid-19, SARS-CoV-2, Drug design, Drug development, In silico, In vitro, and In vivo alone or in combination without language restrictions. The keywords were selected using expert opinions, mesh, and related article titles. All articles with full text or in the absence of full text with abstract are included in the screening of this study. Studies were excluded if studies were comments, editorial, letters, review, and preprints. Two researchers independently extracted data from included studies using a predefined data extraction form. All disagreement was discussed and solved after rechecking the source data with a third investigator. The data extracted, including the last name of the first author, type of the study in silico, in vitro, in vivo, and name of the agent chemical compound, drug, herb, etc., studied the mechanism and efficacy of the agent according to study design that specified according to the following definition. Computer-aided drug design can be divided into three different categories. All are based on ligands and receptors, which are briefly described as follows Dock Receptor-Based Approach. Once the three-dimensional structure of the ligand molecules and their receptor is known, the receptor-based method is a good candidate for identifying or optimizing drugs. Due to the presence of three-dimensional structures of compounds and receptors, the nature of the interaction between the ligand and the receptor and the type of structure that the ligand can have to interact with them in favorable conditions can be identified using this method. The compound is simulated on the active site of the dock meaning anchoring and on the interaction of the ligand with the receptor by molecular mechanics and molecular dynamics. In this method, due to the ligation of the ligand in the active position, the ligand changes in terms of conformity and changes its position in different conditions and shows interaction with the receptor in different types of situations. To determine the type of ligands that can be docked into the receptor site, the matching of the shape and the complementarity of the hydrophobic, hydrophilic, and charged parts must be considered. Various software packages such as AUTODOCK-Glide-LUDI and LigandFit are used to design the drug based on the structure of the receptor. Ligand-Based Approach. This method is used where the three-dimensional structures of the receptor are unknown and instead the structure of the ligands is known, which is one of the common methods. In this method, by indirectly studying compounds that react with biomolecules, they seek to design compounds that are pharmacologically active. In ligand-based drug design methods, in the absence of biomolecule structure, by studying specific ligands, it seeks to identify the structural and physicochemical properties of the compounds so that the desired compound can be designed based on data extracted from the study of previous compounds. This method is a kind of drug design based on pharmacophore pharmacophore refers to the part of the drug to which the effect of the drug depends on that part of the molecule, and by studying the quantitative relationship between structure and their activity, drugs can be designed by this method. It can be said that it is a method for designing the pharmacophores of drugs. Denovo Design-Based Approach. This method is used when the structure of the ligand is unknown but the structure of the receptor is known. In this method, there is information about the structures of the receptor or quasi-receptors, but there is no structure of the main composition that can interact with the active site of the receptor. One of the functions of drug design based on this method is to suggest and present the main composition that is complementary to the active site. The basis of the method is that the database of existing 3D structures is used to find small molecules that can interact with the active site of the receptor in terms of size, geometry, and functional groups. Software packages such as GROW and LEGEND are used to design drugs by this method. Drug design methods in the computer include quantitative structure-activity relationship RASQ, docking, molecular dynamic simulation, and computational modeling. In these studies, the efficacy is evaluated based on the function of the drug or compound agent and the mechanism of action. shows the methods of computer-aided drug design CADD. Quantitative Structure-Activity Relationship QSAR. QSAR provides studies on the relationship between chemical structure and biological activity or other biological activities that are important in selecting or removing a compound before synthesis and testing. QSAR is especially important to predict the result, especially when it is not possible to experiment with a compound. Molecular descriptors, which are the most important components of QSAR, can be obtained experimentally or through mathematical formulas from various theories such as quantum mechanics, chemical graph theory, and study theories. QSAR seeks to establish a statistically significant relationship between structure and performance. It also explains the specific effect of a drug and can ultimately predict the effect of newly synthesized chemical compounds. QSAR model is also an equation that predicts a property through molecular descriptors and their coefficients. Evaluation of the effectiveness of new compounds that have been studied using this method can be reported as a percentage of enzyme inhibition if the modeling has been done and mentioned in the article. In this technique, to achieve a combination with a pharmacological effect and increase the pharmacological activity of the drug, different formulations of a drug interact with the receptor, and the structure that has the best interaction with the receptor and the lowest energy level is selected for laboratory steps. In this way, possible structures that have a stronger interaction with the receiver can be isolated at this stage. The same issue is considered and reported as an effectiveness measure. In this technique, which is based on the simulation of drug-receptor interaction in the body, docking problems are solved and in fact play a complementary role in this technique. Due to the time-consuming work with this technique, effective structures cannot be achieved directly through it, and the final stages of the study of drug-receptor interactions should be evaluated before starting laboratory work by having effective compounds from the previous stages. Molecular dynamic simulations produce information at the microscopic level position and velocity of atoms. The conversion of these data into macroscopic values pressure, energy, etc. is done using statistical mechanics. In fact, molecular dynamics and statistical mechanics link microscopic concepts and macroscopically observable quantities. Molecular dynamic simulations are only able to predict the thermodynamic behavior and stability of the ligand binding mechanism at the active site of the target enzyme. This is reported as a criterion. Study of drug in cell culture medium effectiveness in these studies means inhibition of the replication of COVID-19 by the compound or drug under study. The half-maximal inhibitory concentrations IC50 are a measure of the effectiveness of a compound in inhibiting biological function. In vivo studies are those in which the effects of drugs are tested on whole living organisms or cells usually animals as opposed to a tissue organism or dead organism. In vivo testing is better studied for observing the overall effects of an experiment on living subjects. In this review, we reported a significant number of articles with in silico, in vitro, and in vivo approaches for drug development of COVID-19. We retrieved a total of 2538 articles from the initial database search. After the removal of duplication and screening, 317 studies were selected for inclusion in this review. shows the PRISMA diagram. The analysis of article contents indicated that 266 studies performed in silico approaches against viral targets 34 studies used in vitro approaches against SARS-CoV-2 and 15 studies used in vivo animal models. From 267 studies used in silico approaches, 98 article studies repurposed approved drugs with a new mechanism of action and 91 studies evaluated natural products e.g., herbal medicine on COVID-19. The characteristics of these studies are summarized in Tables and. Also, shows the characteristics of the remaining studies N 87. In silico studies used the following component of novel coronavirus as targets main protease N 154, spike glycoprotein N 62, nonstructural protein N 45, RNA-dependent RNA polymerase N 21, papain-like protease N 19. About 260 drugs were repurposed by the computational methods for COVID-19 therapy such as about 120 drugs candidate against the main protease, 52 drugs against the spike glycoprotein, 14 drugs against RNA-dependent RNA polymerase, and 28 drugs against other nonstructural proteins. Among the studied repurposed drugs, the best results regarding efficacy were observed with saquinavir N 9 study, ritonavir N 8 study, lopinavir N 6 study, remdesivir N 3 study, and amikacin, danoprevir, favipiravir, and telaprevir. shows target-based synthesis of data for COVID-19 drug repurposing. As presented, at least two studies show the efficacy of aprepitant, cobicistat, dipyridamole, and dihydroergotamine against the main protease and tegobuvir N 2 against spike protein. The following list of drugs had multitarget action avapritinib, famotidine, bictegravir, ziprasidone, capmatinib, pexidartinib, amprenavir, zafirlukast, cilostazol, paromomycin, lopinavir, and remdesivir. A total of 91 studies used in silico methods to evaluate the effects of natural products including herbal medicine against SARS-CoV-2. Among them, 54 studies used the main protease as main target, in which eucalyptus N 3, Tinospora cordifolia N 3, and flavonoids e.g., hesperidin, rutin, and herbacetin were the most studied and effective. Other studied targets were as follows spike N 22 and multitarget N 20. From the plant metabolites, oleanolic acid, hesperidin, epigallocatechin gallate, jensenone, tinosponone, and anistone show promising results in computational methods against COVID-19. Aloe, green tea, eucalyptus, curcumin, and many Chinese, Indian, and African plants were also effective on COVID-19 in silico. The derivatives of pyrazoles, oxadiazoles, phenyltriazolinones, triazoles, benzoylpinostrobin, benzoic acid, benzylidenechromanones, coumarin, and selenium show efficacy in computational methods and could be considered as lead molecules for drug design and synthesis against COVID-19. Melatonin, ebselen, phenyl furoxan, thimerosal, isatin, romidepsin, phenyl mercurin, pleconaril, and tyrosine kinase inhibitors such as nilotinib and imatinib are examples of other drugs that are effective in in silico methods. Screening of in vitro studies leads to finding 34 studies. Different compounds were evaluated and most studies focused on the inhibition of viral replication, which was assessed by the quantification of viral RNA by PCR and IC50 value reported. Most of the studies use the Vero E6 cell line for the assessment of replication of SARS-CoV-2. In particular, main protease, papain-like protease, RNA-dependent RNA polymerase, NSP-14, NSP-15, spike protein, and TMPRSS2 were evaluated in vitro as targets. Also, three studies focused on in vitro inhibition of inflammatory markers such as interleukins and the effects of the cytopathic effect on infected cell. Nitric oxide, ginkgolic acid, anacardic acid, troxerutin, bisindolylmaleimide derivatives, small molecules GRL-172, and 5 h, baicalin and baicalein phytochemicals, and bepridil were effective drugs in vitro against the main protease of COVID-19. The following drugs could inhibit the spike/ACE2-mediated cell entry of COVID-19 in vitro romidepsin, panobinostat, givinostat, sirtinol, saquinavir, lipopeptides, hydroxyzine, azelastine, heparin, and glycyrrhizic acid. Alkenyl sulfonylurea derivatives, an IMU-8381 inhibitor of human dihydroorotate dehydrogenase, pyrazole derivatives, and phillyrin, regulated the expression of inflammatory cytokines e.g., IL, TNF-, and NF- B induced by SARS-CoV-2 markedly and could be considered as adjuvant treatment of COVID-19 severe disease. Doxycycline, chlorpromazine, azithromycin, heparin, bepridil, tannic acid, and glycyrrhizic acid are well-known drugs that show both in silico and in vitro inhibitory effects against SARS-CoV-2 and should be considered for this purpose. Also, in vitro studies show that lopinavir/ritonavir, sofosbuvir, and favipiravir have no antiviral effects against SARS-CoV-2 huge gap between in vitro IC50 and free plasma concentration. We included 16 in vivo studies in our final analysis. Most of them use a combination of in vitro and in vivo methods for the evaluation of novel drugs. shows the detail of these studies. Human monoclonal antibodies were the most evaluated drugs N 4, promote the reduction in viral load in vitro, and prevent infection in animal models of SARS-CoV-2. A prodrug of hydroxycytidine molnupiravir improved pulmonary function and reduced viral titer in vivo and was introduced as a potential broad-spectrum antiviral agent against SARS-CoV-2. Chloroquine and chlorpromazine did not inhibit viral replication in mouse lungs, but protected them against clinical disease. Dalbavancin shows significant inhibitory ability in both in vitro and in vivo models of COVID-19. The drug binds directly to ACE2 and blocks its interaction with the SARS-CoV-2 spike protein. Drug development is a multistep process, typically requiring more than five years to assure the safety and efficacy of the new compound. There are several strategies in antiviral drugs for coronaviruses including empirical testing of known antiviral drugs, large-scale phenotypic screening of compound libraries, and target-based drug discovery. To date, an increasing number of drugs have been shown to have anticoronavirus activities in vitro and in vivo, but only remdesivir and several neutralizing antibodies have been approved by the US FDA for treating COVID-19. However, remdesivir's clinical effects are controversial and new antiviral drugs are still urgently needed. Given the urgency of the SARS-CoV-2 outbreak, here we discuss the discovery and development of new therapeutics for SARS-CoV-2 infection, which have been conducted in basic research in silico and preclinical study in vitro, in vivo. Our database search identified about 3000 studies, which means a global effort for drug development in the current COVID-19 pandemic. Although the chance of successful drug development is very low less than about 10% and till today, there are no approved drugs. Many potential candidates at least 420 drugs are in clinical trials. As summarized in the results, many compounds are in the development process for COVID-19 disease. Some of these compounds are completely new and could serve as seeds or leads for developing antiviral drugs against COVID-19, but as we need therapeutics as soon as possible, half of the studies focused on drug repurposing repositioning, which is a process of investigation of existing drugs for new therapeutic purposes. With the emergence of a growing COVID-19 pandemic, the drug repurposing process was being accelerated. Clinical trials using repurposed drugs may take less time and have a lower overall cost of manufacturing and could have a wide distribution of drugs. According to our results, 260 drugs repurposed by the computational methods for COVID-19, among them saquinavir, ritonavir, and lopinavir, showed the best efficacy in in silico environment. These drugs can be rapidly repurposed for clinic application for treating COVID-19 patients given their proven safety. Many trials are performed using a combination of ritonavir-lopinavir. The results of a systematic review and meta-analysis showed that this drug combination has no more treatment effects than other therapeutic agents in COVID-19 patients and is currently not used anymore. We could not find any clinical trial on saquinavir, which is the most studied drug in silico and show high potency against COVID-19. Saquinavir could be a suggestion for further clinical research. Given that the development of synthetic chemicals for therapeutic use is a random process that might result in serendipitous discovery, many pharmaceutical companies are now focused on the development of plant-derived drugs. Natural products and their structural analogs have historically made a major contribution to pharmacotherapy, especially for cancer and infectious diseases. Nevertheless, natural products also present challenges for drug discovery, such as technical barriers to screening, isolation, characterization, and optimization. In recent years, several technological and scientific developments\\u2014including improved analytical tools, genome mining and engineering strategies, and microbial culturing advances\\u2014are addressing such challenges and opening up new opportunities. Consequently, interest in natural products as drug leads is being revitalized. Medicinal plants have attracted significant attention to treat infectious diseases. Complex molecular structures and a wide variety of natural compounds make medicinal plants an excellent biological resource for drug discovery. Our results show that various plants have potential antiviral activities and could use or be a basis for drug development against COVID-19. Some of the studied plants are among what used by people on a daily basis, such as green tea, aloe, curcumin, and eucalyptus. A systematic review and meta-analysis of RCTs on herbal medicine in the management of COVID-19 show the significant effects of the combined therapy of herbal medicine in treating COVID-19 without any significant side effects. Our results retrieved 91 in silico studies of natural products including herbal medicine. The results of this in silico approach showed that some of these studied active ingredients have a high affinity for each of the four important viral proteins compared with the inhibitors previously reported for each of these proteins. They could possibly have an inhibitory effect on the SARS-CoV-2 and COVID-19. Most evaluated chemical compounds had inhibitory effects against one or two proteins of SARS-CoV-2. In addition to having a great affinity to attach to the viral proteins, these herbal compounds have antioxidant, vasoprotective, anticarcinogenic, and antiviral properties. Thus, they can be applied as extremely safe therapeutic natural compounds and clinical assessments might have notable outcomes for controlling COVID-19. Due to the nature of preclinical in vitro and in vivo studies, the number of drug development studies in this area was less than in silico studies. Again, repurposed drugs were the most studied drugs in vitro both herbal and synthetic. Some of them such as saquinavir, heparin, glycyrrhizic acid, and chlorpromazine show efficacy in both in vitro and in silico environments. Chlorpromazine is the single agent that was found to have efficacy in in silico, in vitro, and in vivo areas. In terms of mechanism of action, different targets from the structural and nonstructural proteins of COVID-19 were evaluated. Most of the studies focused on the main protease, papain-like protease, and spike glycoprotein. Apart from the specific protein that leads to viral replication, SARS-CoV-2 causes a surge of pro-inflammatory cytokines and chemokines, which cause damage to lung tissue and deterioration of lung function. Therefore, the design of a drug with multitarget of action against different proteins of COVID-19 and also anti-inflammatory potential could be valuable. Currently, there is no highly efficacious and specific treatment for SARS-CoV-2. Herein, we provided data on novel compounds in therapeutic drug discovery and development. Due to the nature of SARS-CoV-2 and the rises of several high transmissible strains, repurposing existing drugs has demonstrated power by bringing several drugs to approval for treating COVID-19 patients, such as remdesivir. Our results confirmed that a large number of repurposed agents are currently being explored for treating SARS-CoV-2 infection. However, these drugs still suffer from suboptimal therapeutic effect or known strong side effect. To accelerate drug discovery and development, especially during the current pandemic, natural products capture attention again. Our results showed that various natural bioactive compounds are being investigated in the preclinical step of drug development for COVID-19. In addition to having high affinity, these herb active ingredients have antioxidant, vasoprotective, anticarcinogenic, and antiviral properties. Therefore, they can be used as extremely safe therapeutic compounds in drug design studies to control COVID-19. However, the pharmacological effects and adverse reactions of some drugs under development are still unclear, and hence, well-designed high-quality studies are needed to further study the effectiveness and safety of these potential drugs to accelerate drug development targeting SARS-CoV-2 and thus promote progress towards ending the pandemic.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The cleaning function has removed parentheses. We will check a few more sentences to ensure that punctuation etc has been removed."
      ],
      "metadata": {
        "id": "30CaBUPj66XP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_0 = pmc_arxiv_full_text_merged_plus_cleaned_test['text'][0]\n",
        "article_0[:1000]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "id": "x_CPGZEp7YGS",
        "outputId": "b1b719e0-dbeb-45fd-9d1b-65dff5f46da3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Sir James Black, a winner of the 1988 Nobel Prize, clearly recognized well before the 21st century that drug repurposing strategies would occupy an important place in the future of new drug discovery. In 2004, Ted T. Ashburn et al. summarized previous research and developed a general approach to drug development using drug repurposing, retrospectively looking for new indications for approved drugs and molecules that are waiting for approval for new pathways of action and targets. These molecules are usually safe in clinical trials but do not show sufficient efficacy for the treatment of the disease originally targeted. The definition of the term drug repurposing has been endorsed by scholars and used by them. It should be pointed out that the synonyms of drug repurposing often used by academics also include drug repositioning, drug rediscovery, drug redirecting, drug retasking, and therapeutic switching. After the research study by Ashburn et al., Allarakhia et al. expanded the startin'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 30
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "article_1 = pmc_arxiv_full_text_merged_plus_cleaned_test['text'][1]\n",
        "article_1[:1000]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "id": "6i8bRZFH7vTm",
        "outputId": "58581f35-4b06-429f-ce36-4b484fb3b39a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Tight and selective interaction between ligands and target proteins is of cardinal importance in drug discovery that various stages of identification and validation of targets and potential drug leads, as well as manifold preclinical and clinical trials, should be carried out until obtaining the final approval from the Food and Drug Administration FDA Abel et al., Campillos et al., Olayan et al. Accurate prediction of drug–target interactions DTIs plays a vital role for in silico and modern drug discovery in developing new drug candidates Olayan et al. Unfortunately, despite the promising advancements in genomics, proteomics, and systems biology, formidable scientific and regulatory obstacles such as high attrition rates, excessively time‐consuming, and high‐priced procedures have taken a heavy toll on proposing effective biologically active agents, and the success of drugs to pass clinical trials stage is about only 13% Fotis et al., Liu et al., Murray & Rees. Such attempts enable sci'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "article_2 = pmc_arxiv_full_text_merged_plus_cleaned_test['text'][2]\n",
        "article_2[:1000]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "id": "IFkAbDfT8B-l",
        "outputId": "0a691de1-4ea9-409c-cc5d-cc71d6bfbc35"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'The severe acute respiratory syndrome coronavirus-2 SARS-CoV-2 -caused Coronavirus Disease 2019 COVID-19 has resulted in a significant increase in morbidity and mortality around the world. Finding therapies for COVID-19 has taken a significant amount of time and effort. But besides advancements in technology and increased understanding of human related illness, therapeutic developments have become much slower than anticipated. Drug discovery can take decades and is complex and expensive. To bring a medicine to market, it ends up taking an average of 10 years and at least 1 billion. The COVID-19 pandemic has prompted the researchers and doctors to repurpose antiviral medications to combat SARSCoV-2 infection. Drug repurposing DR, also known as drug rescuing, drug redirection, drug repositioning, therapeutic switching, drug reprofiling, drug recycling and drug re-tasking, is a method of recognizing novel therapeutic evidence from Investigational/pro-drugs/old/already marketed/existing/FD'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "There is a risk that performing too much text cleaning could have unwanted consequences for some of the 11,693 rows of data so the cleaned sample will be used for proof of concept."
      ],
      "metadata": {
        "id": "N0indKh46gz9"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 7. Extracting tabular data from PDFs\n",
        "\n",
        "We successfully extracted text from the arXiv PDF after using pdfminer's layout analysis algorithm to identify objects and their position using bounded box coordinates. However, this library does not provide tools for extracting tabular information in a structured way as all elements of the table were `LTTextBoxHorizontal` objects and indistinguishable as rows and columns.\n",
        "\n",
        "For table extraction we will try the [pdfplumber](https://github.com/jsvine/pdfplumber) library which is built on pdfminer and provides methods for extracting tables.\n",
        "\n",
        "The following code checks whether text is within a table's bounding box and extracts and appends table text to one list and article text to another."
      ],
      "metadata": {
        "id": "ai6onYKPPotg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def get_text(pdf_url):\n",
        "\n",
        "    def not_within_bboxes(obj):\n",
        "        \"\"\"Check if the object is in any of the table's bbox.\"\"\"\n",
        "\n",
        "        def obj_in_bbox(_bbox):\n",
        "            \"\"\"See https://github.com/jsvine/pdfplumber/blob/stable/pdfplumber/table.py#L404\"\"\"\n",
        "            v_mid = (obj[\"top\"] + obj[\"bottom\"]) / 2\n",
        "            h_mid = (obj[\"x0\"] + obj[\"x1\"]) / 2\n",
        "            x0, top, x1, bottom = _bbox\n",
        "            return (h_mid >= x0) and (h_mid < x1) and (v_mid >= top) and (v_mid < bottom)\n",
        "\n",
        "        return not any(obj_in_bbox(__bbox) for __bbox in bboxes)\n",
        "\n",
        "    def within_bboxes(obj):\n",
        "        \"\"\"Check if the object is in any of the table's bbox.\"\"\"\n",
        "\n",
        "        def obj_in_bbox(_bbox):\n",
        "            \"\"\"See https://github.com/jsvine/pdfplumber/blob/stable/pdfplumber/table.py#L404\"\"\"\n",
        "            v_mid = (obj[\"top\"] + obj[\"bottom\"]) / 2\n",
        "            h_mid = (obj[\"x0\"] + obj[\"x1\"]) / 2\n",
        "            x0, top, x1, bottom = _bbox\n",
        "            return (h_mid >= x0) and (h_mid < x1) and (v_mid >= top) and (v_mid < bottom)\n",
        "\n",
        "        return any(obj_in_bbox(__bbox) for __bbox in bboxes)\n",
        "\n",
        "\n",
        "    pdf_list = []\n",
        "\n",
        "    tables_list = []\n",
        "\n",
        "    text_list = []\n",
        "\n",
        "    if isinstance(pdf_url, str):\n",
        "        url = pdf_url\n",
        "        response = requests.get(url, allow_redirects=True)\n",
        "        if response.status_code != 200:\n",
        "          raise Exception(response.text)\n",
        "        pdf_memory_file = BytesIO()\n",
        "        pdf_memory_file.write(response.content)\n",
        "        pdf_list.append(pdf_memory_file)\n",
        "        for pdf in pdf_list:\n",
        "            with pdfplumber.open(pdf) as pdf:\n",
        "                for page in pdf.pages:\n",
        "\n",
        "                    # Get the bounding boxes of the tables on the page.\n",
        "                    bboxes = [\n",
        "                        table.bbox\n",
        "                        for table in page.find_tables(\n",
        "                            table_settings={\n",
        "                                \"vertical_strategy\": \"lines\",\n",
        "                                \"horizontal_strategy\": \"lines\",\n",
        "                            }\n",
        "                        )\n",
        "                    ]\n",
        "\n",
        "                    text = page.filter(not_within_bboxes).extract_text(x_tolerance=1)\n",
        "                    text_list.append(text)\n",
        "                    table = page.filter(within_bboxes).extract_text(x_tolerance=1)\n",
        "                    tables_list.append(table)\n",
        "\n",
        "    return tables_list, text_list"
      ],
      "metadata": {
        "id": "IcYk5iMx1d8F"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "We will try this code for a table in the arXiv PDF we extracted text from."
      ],
      "metadata": {
        "id": "wNwfkLu798WQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pdf_url = 'http://export.arxiv.org/pdf/2007.09186v3'\n",
        "tables_list, text_list = get_text(pdf_url)"
      ],
      "metadata": {
        "id": "V5iQlta714rK"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(text_list)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3dcd4974-2d9c-440c-9c01-9883661ac95f",
        "id": "7KpFUgfa14rK"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "9"
            ]
          },
          "metadata": {},
          "execution_count": 62
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "tables_list"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "13dd678a-c4d0-471b-fbf3-57cf756a6dd3",
        "id": "dcJWE5u-14rK"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['', '', '', '', '', '', '', '', '']"
            ]
          },
          "metadata": {},
          "execution_count": 63
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "It appears that the code has been unable to identify the table at the top of page 2 which we identified earlier when visualising page layout.\n",
        "\n",
        "![image.png](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAc8AAAE+CAIAAABk1AzAAACAAElEQVR4XuydB3hUxdfGsVfUP4qgIAJ2rFRRmgIioIAKioB0A9Kr9BJaKAGkBJEWIMCSQFg6C5GlSEcpSlFqSIBggNCWEknZ73fuWW42C/ohRGHxznOffe7OPXPm3Jn3vHPmlrmZ3FaykpWsZKV/PmXyzbCSlaxkJSv9A8liWytZyUpW+jeSxbZWspKVrPRvpAxj2+jo6E2bNm22kpWuI/3000+xsbEpKSm+8Pon08WLF3/++WdfU6xkpWtKvvDyShnGtsOGDQsPD19gJStdR5owYUJkZOQff/zhC69/Mh0/frxPnz6+pljJSn8/1alTxxdeXinD2HbIkCHHjh3zzbWSlf5O2r1795IlS/59tg0NDfXNtZKV/n5q2LChb5ZXstjWSjdRstjWSn6dLLa1kt8ki22t5NfJYlsr+U2y2NZKfp0strWS3ySLba3k18liWyv5TbLY1kp+nSy2tZLfJIttreTXyWJbK/lNstjWSn6dLLa1kt8ki22t5NfJYlsr+U2y2NZKfp2unW3Pnj179OhRl8uVlJTEzqlTp5KNBKuy7/Myu8W2Vrr+ZLGtlfw6XSPbXrx4sU+fPv37958wYcKCBQs6deo0fPjw/fv3b968uXPnznBrdHS0t7zFtla6/mSxrZX8Ol0j2545c6ZatWrr1q0bO3Zs2bJl4+LicIMVK1Z06NDh559/joyMXLhwoUrCy/Bs7969Lba10nWmf59twfnOnTvHjBnje8BKVvr76RrZFsR37969RYsWBK1lypRJSUlZuXJlVFRUzZo1T5w4AdVCuHoxASIm/q1aterRo0d9tVjJSn8n/ftsu2zZsl69egUFBfkesJKV/n66RrY9cOBA8+bNYdLx48eXLFkyPj4eql2+fHmbNm2IBWbNmjV//vzU1FRT3rqSYKXrT/8+27qtKwlWyrh0jWzLDKtt27ZdunQZOXLktGnTCHL79eu3a9euVatWsR8YGMi+t7zFtla6/mSxrZX8Ol0j25IuXLgA5yYmJqakpLBz7ty5FCO5XC72vQNbt8W2VsqIZLGtlfw6XTvb/q1ksa2Vrj9ZbGslv04W21rJb5LFtlby62SxrZX8JllsayW/Tv7HtidOnNiwYcMqK/lb2rp1a2Jiom93/p1ksa2V/Dr5H9vGxcXNnj3bZiV/S4sXLz537pxvd/6dZLGtlfw6+R/betLMme5mzdxNm7qbNJFf9lu2dC9Z4it2zen0aXfv3qJZlevWv787NtZX8trSyZPuXr3EeD0LrcXc6dDB/fPPvkVueJo92928uafBvbdWrdybN/sK/wPplmHbTesvtmxykYY0t8F9L7jOpHuM53rSj2t99bdokrxk3nVNLLzThlXoT/LWb27NmqRMm/hH4oUMO5eMSrEHkls3Tbzc4D5dzrvOpFvU5Z9Lfsu2nTq5N250162LaneDBu5HH3UvXuzu1s1X7JrT4cPu1q3dX3zh7tPHffvtsj3xhHviRPeGDb6S15Zg7Xbt3OvWuXPkcNer5/7uO/ezz0oO55Inj/BaeLhvkRuevvrKPX68NEXVqrLR+G+95S5Txl22rHvqVF/hfyDdMmw7d07qpDrOmHFLYqq1janTbX/V9r0DU+J/9xW75mSflTolLDXms/Yxt+eOyV4kps03i2+vMLxbhr3MOSMiddrU1Ji63WMGTo954s2YLt/FNOob0zcs5oX31r3X/dtRqWdO+xa54Wn7ttQBn/4U82zpmLlbYh7NH9MgMOapYjF9Jrd4avbvGdfyf538lm07dnRv3equX989fLj7yy/dWbK4ly1zd+niK3bN6dAhd5s27jp13EFB7kyZZMue3R0W5l6/3lfy2lJMjPvrryUkfPJJYdgJE4RtCWkDAty5c7vnz3dPn+5b5IanRo3ckyZJU3z6qWw0PmwL1b73nnvKFF/hfyDdMmw7Z07qtHpR8ZMWxX/WPL5eh7hqLfr0Ts1An581K3X69NT46i3jMz0en+21+PbBzkylh3XNSLZli6/fMX7IlPjsr8V3HxnfpGf8gND4F0pseq/T6G9TT9+MbOsO/mxj/DNvxzs2xf/vxfgvu8bnLBDff0LLHJEZ2PJ/nfyWbQmmmNczh+3cWYJQwkNocc4cX7FrTsz0+/UT/dA6ES4b5DJihHv/fl/Ja0sJCaKfDbXYT1ROTxDbsg/5Dhzo/vFH3yI3PNHmPXq4a9Vyt2ghG43TuLFcSWCfIP2fT7cM225Ym9SrVUJwt1PBLWKCWx4c1Dxm2KA/zpzKsNn3+tVJvXokifJaW4LrbQ9ud6RvrZ0LZlzXRXPvtGZlUu8eycGtDgZ3Oi76v44Pbn04uOOx4Ia/9m8SOz3s4oXzGXYuGZViolO6tzgZ3GBncOC54Dq/BLc6FFxvBzb3a3HkdMa1/F8n/2PbPXv2DB06NLB798CuXX23Hj0CMzB16+arn5yePX3Frjldrv8fqiijEs17uam6XYW1Y8eOPXXqlG93/p10y7BtSor74sV0W1KSr8z1pMv1s6Uk+4pdc7qifnNLzriKMjClpvraqVvGtvxfJ/9j28TExN9//z1u1qy4Xr3iAgPjeveO69s3bvnyuJEjZZ+N/IkT42bOjOvTx/NX83WfzMmT4yIj03IGDYobOFCULFsWd/hwnKa9e+NGjJCj3hvyWqP+1f158+LGjk2rKDw8bto0XxktOGBA3O7dHv179oh+0wZTv2mtmcMWHBzXr5+vDL/YjPGmvFkKYYqYfznllSvjvvkmrn9/2bxrQSwoKF1dlFWdgweLwZxdbKzH5jlz0lWkvyin9SjFCXrr4a/msE9rHz4MAJKvzxFvDbaF9cLGJo4d687Abcq4C95V7P0tacLY5MvFrmdbNOu8qX/Prxmv/5/YZs9Oa5PjR1Mmjk26XOavt8nj/jjryrDI1//Y1pO+/NJ9220ymc2WzV2tmlxRffppd4kS7pdfdn/7rbt4cZmPM8PNn1923n1XNuSZpz/4oLtcObkgO2GCyHfu7H7uOfcjj7g//NAdGZk2Lh8+LFVUqiR3q9BD8TFjpIpRo+QC68iR7ooVpepBg+RZCJR8/rlcCmjfXq459Ovnfucdt93uDgwU+Weecb/5ptzNy5nTfeCAR39srDtXLjHpoYfcffu6P/jAPXmyu1QpufT85JPuBx6QnKAgKVi7tvuee+QssLlQIffDD7s/+khKccpY9eqrcu/u9dc9J0jVVau6n39einBG5cvLkw/Ic2r/+5/7rrvEzuzZ5dZfwYLuokXd997rrl5dLET5o4/Koa+/dt99t1w/yZJFLsiOG+c+f8nNmjaV1kADFXGayHPi6ES+a1f3nXfKJd3PPnM//ri7ZElpmQEDZAf9GJ/+Wx7Xlm4Ntk1MdH9w2wJn0c6yvd7aWT/MWTvUmbehs9q3zj6rnW92chZs78xU2jk93lmmnzNPA+fEGPlbtItsb7R1fjjEWTbI2TjcWTXE2W+tyBdoX+q2ld5VRM5IGVRltRwdvMlZb7Kz1gRn4wjnJyOdQetFSbuFzldbOe+p4Czbz5m7vnPCfmfJns6Ws53v9na2X+j8eqGz61LnZ6Odj3ziHLDxknzLQq+nNXu4LXnwR6ucmas4C3dwzjju/HyM86sIZ5VhzkE/Ot8yjKTgvRWc7wU5c9VzPljZ2WO58+MRzsyVnUU6OWuOE+FPDf2PV3f2XeusMMgZstN5b0Xne/2duerKyT5UxTl4s/P9gdJEKCFHtyGbnW91db7exhm8STRXHiaZD3/kbGF3Fu8hG0dpNDJfaur8YHCp5+NMm7f9ktq+3C/Op+o4//eJs/ksZ4keUm/LOSL8Ti/ZvrQ5s9d0PlHLOexn5xttnBP2Ocv1/yLn8oPRGRb9+i3b1q8vvj1woFBYo0ZySZEdGKpwYXdUlPBIvXpCPdAujADdsCEPBcBTZcq4a9Z0r1wpfAQbvvCCZNaqJZd9TbY9dMjdtq27bl25F9e/vxRftkxYbPFiIbhFi0T+qafcM2bIA1vQYvPmQpRDh4rmb75xV6ni/vVX4cElS2QAoEbGWeSjoz36Y2LkaQRMggSnTBFKXbdOyH30aLlLljmz0JnNJvegMOP224X+2rUTOoYEIS+Ghw0b3JUruwsUcK9aJXer9ARpB1oDCykCgcKM2IA81M8wA0s2aSJmwNSoQvkdd0gOFqKcwYNDGE9ZhhOqQCeSJtuiecsWOVMqwirkaT1IFnlGOGiXfFqD0YKGnTlTqJkdcmhGi20vJdi2bqZJMpKxvf22DFQ9ekgXfPWVtDaZOnDS1AycL73kXrNGGpYQgQ08A2yGtJ495aL5tGnSwmXLVso0z7sK+6xUR4MZAoZZswQ5bdrIwE/3gSj008U4yH33ya1O9P/wgwCPofGTTyQEGT5cnpCh1x57zB0RIZUi/+abr72Wpn9GRGpUwwhBCKZu2ybhCIM6MKPT1Ui6/v77xU6cC8SOHy+xC/JAjghJn33MmlXwhpG4jMPhkSf0AU4M5FgOerGWTH0oiA0nIoCg0TiKM1IjDYUwHorvs+EROA6ZxCV165bOvc+0efu21MHVNx57tuixrC8d6zfm2Ad1j9VseWzA+GOZHjtWpaFs3YYfy1Xg2NOFjs1be6xYlWMrdx6r3qzlMwsOWWwraKD1oRU6iWAT54e8nnhCmp7OJqStUEGO0nlAjZ5mozM6dBAGAdxACnhx8lAk8CX+InBbvjyNFFavdhcpIsRHKZidGA16gtDpS1BSrJhEplQNzeESIAwbAO4bbwieiLWJfD/+WLocYwAZxIQkQZ/5uC5sDlw4BYJQRggqIhxGP2WBINEicTQG8IsfEuq+8orIo4FolFCUCBegIw+1sYNjcPpoQ1XevPKXpsAG1GIDOZ06SctQFiPRhhhngVVQMDmo5Rf34xD0zXnRgAwDFMQrTp702Ex0z3CCa9Em+AzOSevhqLQATUomNcLF/KVh8Sg8nB3kR4yQy2bXnW4dtr1jinQQ7U9PQXyMRvQO3Uq/kA+0GLoYoWE4yJSQQkHIxgjK/IwmZXTEBWhkiKZnz0rZ0j2YaA9zOXI3EfwAVLANYTH7wSPYmL6wDwMyTIIu/gI8UISzADBmNgAGHwFUoAUNKv/oo+nYNvRMVO5GglKQA+ABf7584mvM80Av0GJOBrA5I/YBEhOpPHkkh1NGEoUcIsTh9KmC88VNOGXKEmeASYrgceRTO7/kAE5+4VnaRxsNF4DKyWQH7yCfDSeFjqkId3j++dJ5o02b9+9N6dooftSny0d9FDWq/f5RDX4a1WjrqK+jR1VcMKrhJtna7h1V1Tmq2rJRfRJG1V43asCZUY22tPs05tjv13X5yzv5LdsCi2HDPFEqTEGX0zEgjy7H8198UdgT4qBrT5zw3Vwu97lzaX9PnRJCYcf7vVKUMLRCxHTb9OkSdwBQwhCCUDoYrCu3EhviCcAUigEi+ANdDgExmAMFMhmcMYyC4ABMm1cSDh4UVUAcbRjPb9++QpeYjWYGZyxHP8SHh9SoITgjk94CnWAX8BFX4m8wGpxL/I7PMORQhKAb1qMFuncXZ4D+aCVQjqmgGXnMgHypiCIMRaCTRiNQwucDAuQsNAChCGWxnzBcE0Hu5Y3JRutpA/7Zdn2vkJnp1mHbzHZpWBqfjmDQAsOA7f33ha2AGUxHPpwCbJjx/HVrnz0LniuVv+hdhT080VEmWFTRoSEhEqvCZQS5DKIgBGhRNXBiNIUWCTtgeaAIvcLjjPf0O/aARjAJvxMoPP98OradmhhVdqCAllLQIgLQH1UQckLBaMZBKE5F6Mcr+SVoQD8OhUDp0hKbs0/UjEmgGvJlLAeo1M4YQA6MSWSDJAMG0EUbhxjsVZ4pII4JaPEL/ItzRHjIEHEW2hMXwDUqV/aObZOT3CdPpF6Ozb/eKJIRszJP8lu2BY779wsJDh4sRAY46KRffpF+ZeYFeug84gIo79oSUVufPtKLMNScOcJNkBQTKyiSAHD+fKkUUgMEdDn0B3oQYEIN9xEPIgnUyKQgIz9RM0TMKO19JYEq4DVcQqn8p59kwoi/Ic8OVAh00K/vsHGCZOI2MDj8DqCPHJGJISdLRUuXyqyKeRYm0SD4EgRK4EONzOmwEA6lOIhfu1aIeMkSmYTS95GR4mB40rhxcvWAQYKzQJ584mJkqNdk2xudbhm2rf2/+YmNWybOXpT4/Q+JXXolduuTGDQkseXXiTPmJubIm9iwaeKSFYm5nkucND1x03bk/98NpHsnu93taDxH+AgvoGf1KWm4WyNc2BwJcghj4aZt22QeBvszhAMwQItzMQAgsGmTzAKJjt9+Ox3bznBHNbELqCjFII0AG/IOhyCTnXnz5JfIAM2bNwtRwpuwMw6FAAwObpnx7NkjYKZS3BZuA3iAlkCVShGDWxs18ljChmF794o8JwWZAOMJEyQMJ76JixNhtKEBog8OlgsUHTt6s+3NkPyWbRnHaGsohsYFa3RJiRIy4kE3xLzEa3AZxEQ3X1saNMhzJQj2AQ2QLBMWIgV0wrYMnnrpgFEXyAJrEMzMiKiWkITxFsqD6MkEcEwSGcYJtBmETeYitgXoDOlEr0AN5QAFbWBXrzAw82KHfM4LUiaIAH9gF9RTL6WI7ol5v/xSglBKoZ+/FKEFKIJmppyQ+KefytiDPDqRhNkJeDkdToQiOAlRCQMVxtCMNBdnQQ6qKMtfBpKMesT4utOtwbaY36D4roj6iyL67Y4IPhDRdEVEsxURrddGfLkkovdvEW8Pi6i7IGJQdETx4RE9tkWExDNw/78b8PdO9mnnHYW6CXgADKErGxElA79eLiCkJRIkB9wyVJPPL3iGbZGHFiEFdgAMaAFOlMqVKx3bTj4fVbiLgAqYgX/iZSgShWPHSkxKQQwCzNRFMZQwzFMp8lWqiABOAdrBG1MoZPAaYhQoHqACdfbJIRDGi1GOBr1oS+SLs+NWHO3QQa5RQLUcBc/t24urAng0AH5sBt4lSlhsm0EpNlYGXjZCQjZz3+f3t998C15lYkCm1xm6H3lEOlLJtFw5me7RzfQ9aABe8BFTG0ZXjtLN4AOIE1cCCOLKypWFl+E44kcKgkjzui1sy1GwQhV33ikMTgwLFsE6oEE5h1DFBncT5+IP5AA17IEB4VBkgC/wIsQgAqVGIhdIE6swCT0wKYeohb84UubMcnWMHdwA2wgNOB30cy4U5BBTQsYD7EQznkBFlMWRLLbNULZNTZU7rwrbjNp8FtWwTzjheKaZdDH9CNPBaMbVABmV4UcGVwUPuAVRQALgMfTS4xQBDIz08AKQgLMAAPivUSMd2046F5W/gygHP7AnZRm/QY46CFiFxwk70AAOAR5q4Uq9mIArAUt2QDWWaD4OQlACS7KPPL/YySFqZx9Ig3PMw00YFfAyvXrLUXwHG5DkLHAlvVRCEI07FC1qsa2fpEOHBDFMrul7EAOrEuTqjWDIlHkNkSbb3Lky2QFtULM+70VP0/fAFHlm+j/8IKFBfLxcZABMZmzLTufOEvMSias2amTmTlC5fbtwd/36chGDfEKPoCChToTRDCixYfp0QR5HmzcX34iKktvZxLmzZkkmSjp2lPfTYFVs0CsJMCnxvk70IFYsZFqHAH+R5xCbPuKGDEFK7twyHaNe60pChrLtv5CEbSuOkKGXaRm9edddMoQTA9rtciUBmlP0MmwTwzKJUZCwQ2ipl1YnTZLr/gpv9pcuTX8lITXqq1nyzKJCFKpds0YuwlKcX8Z+pkfQIlEtLgMzAjZm+jAgwr17i2exg348i3zgh4MwGIwfL+EqtRNMQMrMX5lNEiZDzS1bSpQDt0L9TGEhXxico6AXj0AShUwosRZtMO+UKcS/Ftv6SYL7oBv6GJ5iwASgYBEGBElM5MlkaKWb6WD4CNQyFEOLzIbgL4SJCkFD376CDAbbefOEWxmTCWk1QWENGkhOI+PeLliJjJQJEXilCHGHDuBwPf5AJ5EPzt58U4iYSAQyBVKgEI9CbPhwmVuR37WrqNLnZghkACj5GIMerKVGvaTAAICR+tSXynNe+AyM3Lat2EOsgbdwOt4jxI1OFtteZbKHnnS83UfGZvoRkMC24IHOHThQCI5JGMCAfJmnQ74ECgAVADBsAzMNFOBE+Dc4WOANY44cmY5tbRejqoQIVGDbTMbTaZAsAz+O0KqV1MsOaiHNL74QjyhVSrQxxgMtnAhLMAnNRNx4DTbo85HgH5qGUgEnTsdsDy+gLIEF9qOQQxTBK4l4cEPoGzDjEdA6hzAAdwDMhNtff80ZWWzrJwm2ZVCl52gguInpPN0JEOFENkJICHTOnLS/bLqzcKFAh6Ps6FFzI9N8dhUKA4jM2oiFgRS/jM8AF3YDi+y88oqEzHpbDPcARqCzSRMhTZWH8YEgFAxjEthiIcTNGMDECnqFgjmKHniWyR3yOBhxBM7ADpyLXxHMso88NI2D4Rt4IDBFHnwT5IJsHCAhIV3L3Lhkse1VJnv4H473hwkbggdgAxiABGOtCcX5890LFqSh18Qn+fzlEBhOj970zyT8EVV+iEARmgNjxYvLBn7AMyM0bAt9E6z07y8UCWUDTsrjSjgU7FmsmGwU7NFDqJYhgdkbAS/A03keR3E6DMZ+cvhL6EAvIMMIkS+fZDIp1Od82MdB8BfYltpxK1wDOi5b1mJbP0mwLT3HgDxypBATkNU3pjIqwbbM9AkKCJkZw/n95RfBDeM2YTI7xCAwKWJUCu/DjIQbEyYIfFUenAUGCnvmzSur8WIh1EwcoTcNQBtH0QNeCR/AH+jftk0uDjz9tDAs1Mx4wD5kDcMyjwP6/GWyqfL794v/TJuWNkLc6GSx7VUmeSah6Xx5dAA89OolYAASTICuI/k+k9B8rkBx7FjPlSg2QmDwDF9/+aVcWGBKt369RKBbtsjAjxOFhHgeNCSO1mdsvv9eKJJZ/7p1nqc2MxmPH3AUyh4xwhM7g21ikaNHRYZ4FpIlc+dOebbhhRdkHwd59115kYcBg/id0YVZYECAxbZ+ko4flwtYUB6BJDMgpjaMn5Mn+4pdc4LNGaUJYx95RLiPwFlfT4BGATE5+q4BaIP4iE/12gUMiySl2GB/MpG823gVgugVAaiWiRUEykbcqrNIcvT9Wqog827joXR9IYIqOKT3H6iXfAqqPI5KfAG//7vs9hfJYturTFEL/vi49KlqFc9WK3qw2nsnq711qFqZE9XeP12tmtyAuLYNJjTTjHEnox6rIWhhu+MOwadCFCABKvADUMEev/qigQqAPQTAKmwIXEE7MAbAKFFJfsnXN8jJV6yqfmCp70egnKPs8Bdt7CBA1aCXDTGKU4qd8uVL5zuSZvRNkCy2vUHp4EHP01dVq8r43KeP4IOxmhzgQrhaoIAc0hc3wCWTLPgRLDJjgqbBE6E3mVWqyBXh+++XgrrOL3BkbACpH34oZE0owUQMMkVM331A5tlnJW7Nn19CWjQQFyDAL5oZY1SeUJcw4WZKFtveJGlG2IWo4r1kzg7qwCdIA3L6RCO4JZJVZixYUJgRLMHW/EKaBK2Qb8WKEpASCIN8feWSiVfHjp74A7Sj5PXXBaJvvy2ZAPWtt4S+CY3xDuJWpoDscFTfP8KVcAHia+hMHyIiNjp2jGDppkoW296gpFcS4MdRowR2CQmCvJkzBcHgBqwAHX18uGtXgaa+A8YOMyk2dnbvlsxhw9z79kkgEBAgl9t8riQQR0RGeq4M7N0r8NUrCfgGc0CUg+NvvpErCYTMX30l1Ox9JaF3b1+zb2iy2PYmSZ4rCTaboA58Miq3aydIfvllYd7x44UHgXHt2p5byiAZxBIrfPedwLhvX/nbq5cw9ciRIgZr//qr7Lz4YrorCWCSTPPtiago4fEGDeQaBfRNdDJrlvAykhzVKwmQry7ndPSoxbZWMtKRI3LLC96E4AAFzKiPfH/0kSCvVCm5e8C4DRyJFIAgaEOAcR5qphRzfGJVMsFily7yl8xmzSSaIIAlKEYY4kYPeCV6fe01z0MRuAdKYNv69SUEeO+9tDcd9J4bVql8z57iNjdTstj2JknzZ19s/dH+oID9QR+uCfp0c1DNbUGfbAyq82vQm7ODyiwNqr87qMTCoJenBFX8IajI7KBXbUF1fwt6YWLQy1OD6v0WlD8i6POf5W/1rUHvLAmqszOowg9BxRcEdTplyNvlUPkVQe8sDqq9M6jyOsnkLxv5LQ4HFYwM+nB1UNtjhv5VQV8dCCo6V5RwtOGeoKaxor/GL0HFFgT1vODz0scNTxbb3qCUkiJvH8K5Ptvvv8vdgMvz9VB8vOzwy/7lAn8m/2cbR00B732zeAatb5BRyWLbmySdP5f6+5HUywF1s23Hj/tafmOTxbZW8ptksa2V/DpdI9umpKQkJCRAoCdOnEhKSmL/1KlTycnJ5JPDfkr6lXMynm03bpQrjHPmyDZjhjweOG2a2+mUCzo2m6xCu2GDZM6dK0fZX7dO5NnCwuTdGHamTJE3zLdskW9lzJwp+2g4e9aj//x5WWJj+nTPhvzkyVIkPFy2ixflu2GLFnkOUZxDkZHyS6XUiFVo++UXubqKMZRFPwVN/YSNag+ZKMGkBQtkn1L8nTVLLEfb1KmiSh+BRD+nZtpDDkdRyw6/HKUIh9hZu1Y2zCB/4UIRJn/+fKmFTNSqPdp0HEWP98OVCHMWiFHv5s1pa/7S5pwyJ4sefmlJxFCFVfPmSabmaythDBs75ND41oqLl1JKsts+PdEx65xjYYoj/JQj0uWY6XLMOOOYddYx9bjDdsJhP+ewJTjCjjpmX3BMPyli9vOSzzbxiAhzaOoxx4zTjvlJjjmJjohTjsm/L5qTtoJd9N6U+TPOiRgFveUnHXFEnAbXompBstROFWHxUosq5xd5TMIS6uUvdoafdExLWDw3Tf++PcnzZ5w3yh6VUghEnhXNkwwNVDotwTHpd9E/54JUjeTcP8QATg1JPVnkKX7JfseCJI889qCQFkAPR1VeqnA5phwzqjsl5s276Jh2XCxHHv2i/LwoRH7KUSkyOX7hvLTFEhOOp86NOC8aJsfLSVEQPeygkP3pxilQnLLz/hD754s9s6e4bvy3G86fPz9v3jxQ+O23365du3bEiBE2m+3AgQPR0dHkhIWFseMtn/Fs26GD3GLq2VPe/6tSRf4+84wsJcP5ZM4sr8k2bix33nWZ4d695SYmh5o397zw2qyZ3NMsUUIu7UN8r70mV+XvuittRcTDh+VifOHCcjOUjYKofeopEXv4YfeZM/IAIHRTqJBnRdcnn5TLrI8/LrdWK1eWa6no79VLLIGPqLR8ebl+7/3thqeflvcpS5WSC7WYNG6c2PPCC3Izt3hxWUkEbblze95AnzBBHifIn18u/7NRELUcLV1abg5Q14cfymlyqEYNuRaMMRUrykVemJSTffttuS12m7GuTbZsYg9Hkc+VS26d5c0rTyxSEadTrJgoaddO7g5/8YU0rPe3Gzhl5G8zXg7OkkVur+mzZZ06yX028qtVk8chsD8oSG6GsEOO9e0Gr5SY6K741glH8/lCFmWCHbWnOb6Y4qg23tHI7nimmaNwd0fLRY7XOzoer+PosMxRop94ftsoR5EesmUqL8LZ6zryNnF89J1j1D5H7w2OCsMcD3xS4OU0NoyckRJUaa3jsVqONksctac6stV1PNPUUWW047YKjg9GCtuiKjTOUXqQo+sqR5aajuJ9HQW7iPICnR01J0uRZ5sJwRXq6mg23/HOAMfL7Qq/nqY/3JY8oPIaR6fljqy1pRQCAbMcH4Y4bq/geLOno/ViR762jnuqOMoMcnRb5Xi6saPzSkfQZkelUY4WCx31wh1Vx4k8llA8cL3j/aGO+z92jDsordFttePRWo5S/R1vdHK0i3L02uCoH+H4ZKzk1LE5nmwg1ZUNdrzQyjFit+Ol1o5ivcX+jsscVb4V+axfOBrOcGSv5/hkjOPh6um+3fBzauuy2xw5Gzoe+dzx1RzH270d5YY6ms4ThTQyGxXRUE/UcwRvc7zWwTE62lF2cM0cK2/8txtSU1PPnDmzbt26CRMmNGrU6Mcff3Q4HMuWLRs5kr50REZGwsXe4W3Gsy3uDbcSPRGawaRwyhtvSCxWp45QCcEsHp4njzw+De3CjG3aCH/BfQ89JLcvQ0Lkob9PPhH6IN7UD9KQbw4S+nYDlASVs1EQcnnpJc+Ki6dPyx0kqBMehK/hJpi9Xz8hL4JT+JeYEa7s1k3ulhLeUiljg8+KixA0DPv558LOVP3bb0JM8Dv7jBDE6ci/8orn7YY9e2RcKVvW80wCtaPq5ZeFgr2fSYBzIXoy2Ro1Es7lRLDwo4/kaXA00w4+bzeQw2CDcsaGd97xrHwGg9N6ZLZtm8a2VMEpw7Do4dRoQFpPP5DDjn67oX17eeyR4ScqSiYH7JBDp1hseynBtnUrnxBEMaLTMox5gYHSqrp4K5AbM0aexqNrwBJjW1ycvEal/ZLJWHiQrnn9dRlxmZnRifTLww/ny5dWheftBkZ3Bmm6lVEZ78BNdFkvEjgBNoymzDzAId2k73G9954HCVgSHy8hAuBhZC1e/ApvNzCnMdfZGDzYs8C/zzMJTJVAKRMpfZHhu+9kGAYSuKR+RoRZEa5KKKPPJODCmK3PJABCII3bAsLq1aWhdIkSGg2HZXJJmKJvrtNQLVqIPAoHDpToAVRnzZru2w3b3aPrrJVohmhj2DAJO6gXHshkfN6JjciMhqLsDz+IIzAxrVu3w/P2m+LbDcnJyUuXLl2wYEHp0qUhXyLchQsXtmjRYv/+/StWrJg6dap6xa5du5o2bVqoUKF4Oi8Dky6URTClD06zc999whpwK60JM9KXUNUdd4gYaKCH2NG1uwge4QjQAG/SSdDc48Y3ER57LG0dgyNH5AVBJCERlacKfaGgSBEJh2FY6iWfTDYiQfqJ3kKYjRyUQ0m4EGoxA0mCPlM/nqaLG6l+drQWtgIFRJIiqpm41XzMG2H9Ne253fj6Aw6DMebKZKAWJLGPmMqbtbDDias9/FJQ34+gLIc4C6zVv7QnRQC6efUDHAN3TpNphPlLDiMEbQUF8NfM56/mMA5BEH54JQFUjx8/vkKFCl0z8B1CZdvbJnu6UnucBqezdEl4egeuYYBn505jgStCh/BweXzK7EftSkC+cqUAzHgbJR3bTjjhuLOSB0JakQIMSWjRbXxoysT2PcZHFhS9dLpKMinZvFlmdZcwlo5tv0uIuqOCmGcao/hE/+fGsug648EFIG4GBvb1TBWNCjb1TarGVXFARTKZDDN4mSo0T0GrUNtMnOvDvNqANFdwsDycY7rko4+WfjZtiY8dW5NeyXtOUXn12xt5z8Qdugm+3XDu3LlJkyZt3ry5Vq1ahw8fXrlyJZ7Qs2dP4tzFixfPmDEjKSltTMj42JbhkVGUztB17+mAVq2kXwk/2aEzoABdX46+p+mhFUIDepFhHOAwdONCjO0MyIyH+vECZMwVWAgowB8jJ9wH9NEM7itUkDFT79QzAMJEzz8vYSPTefzBbbyzQHWEsYSWdD+8T3HCFpgR0qFS7xUX0a8r2ukynWjDoyjFKPyAsbIch/gl8i1lfP8DAxiB8TFomoGEv0QiaMYknITR/pFHxIt0DCAK5ijm8Rdhtg4dRCFju9qj63pkNlarQT8tSaQAWGFbfAwzGE70Dcj0F4VuYPqX2VbTPxLb3jVNmlrbnwYnjqOp+QWKik9dEkgX82SsnTJF5CFU5InyADxdxhSnY0fpdGLbfPnSsa3tguOtXnKI0LhcOdGAPEAyH0AlHzwATlwAkGhiEoNPYQY4L1lS6sJfACdoqV/fl23ztfI8awjgGST07QOQTyn98hO4AlHohxNxUqJdoKuX1Dhl1Q/7oBwz9A1y/TwPMThixYp55DGmRAkxjCJMmzgLilARIMdfmArgtmTCyAp1fTMC+TffvHXe3I2Li+vbt++JEycIaXv06AGfwryrVq3q3r17//794dxUr3Am49lWryQwC2BGwMYOsyrIkV7fuFHaneBu7lzpvz595ChdwpQENA8fLpgj2iV6Zdqybp08owoxLV8uEPG+ksCMhokPcBk5UiCuT2XDzsq2yM+fLwX1SsL06ZIJWSPAJJq5FljHAyi+e7cYgCUgwHvFRfNKAtMZUKvrJLDpjq6TAMZHjJChpV8/kWdiCLGCV870SWOdBCRhZ72SoPMyVUI4yYYS9sErPqNvNxQq5LmkAB2DVICr6yTcabzdoC+h6dsNzOzwcPzEWgMsw9k206S0KwPwKXMgOlov8owd61lvkGZnzIY+6HHvKwljxsh1M7rsm28ErhDQ0qUw719dSTDXSYCgNXlfSfBmW+ige3fPZJ94M1cugTeVLlzoy7aVR4oTYS1OAXGzn8n4dgO/mH35lYTmzQVUHEU5MRCxBUhbskRy9O0Gc52ETZuEqZHHg/irVxKGDBF8qv9mMhZYUJyjzXy7IZPx7g+1462E/LfSOgmEridPnoRSL168ePToUWiXHTJh1YSEBPa9hTOebY3rMjKgEc3RW0Sv9ASDIXglZmS6Qdfq69gMpHqDC1oBuAjoVOVp431twKrvFAJNBHyuJGhZ/b3bWF4AlCvb6pduMxtvfzN6E9umpMj1AYJH/iKJbQzUCADZO40LF/CjqR82J7Qx9WMnYnoNBPYEbSqPeRwFduTr1Amb2Uf+buOyBjEplqOfSdzdxkoICOjpI4Zyiqg8qvR1cgYVSumFFI7q6+3kQNlana6TgCrsR9u+mwWytw7b3hfh6ZTsxkIZ2t1AVDFAR3AIfHKIfPAMunS1NjqF7qOUXl962lgX3FiaIB3bTjzleOgzwUNW4yOM4EFrAbFu41nvOnWkOhRSHWxLDrERwGYqo8hBXufvuhJC9uy+bPvARx7oIkwR1YYlejEKs3Uf/ViIEo6Sr/KoVf0I3GEsg6DLh6oSYM8hlcdmNjUehTmMRcopoiDXhRooyFEFM9DNbHxHkpyHHiqdZ3+a0TdBuna2/Vsp49mWcJIwjTkLc65njE/IZWyKi5MpmF6UpGuBON3JXAaK3LFDogmADrHGxno+Q8Kkj3AADgXQRJR3GxetmAHhBASJDMWmvCZoF+XAonhxyQdDQI2Ig53bb/d8M6Ko8WHRJ43vobLDGE4OlmASbgmeOIQrMo0CZxoE6YU/KtVHCzAG+ZeM7+kixj6VKujZedf4ai+W4wAoB8r6oVMyKQiaUT54sPvChXQtc+PSrcO2OZd6vjVLDAh6M+JaTTq2nXLW8XxLgR8Ao2fBRp48giL6F4h+YHyLhKObN8u8+wPjc1DLlskUnjkTG2BDnX51BqA2aQJp+rJtzgaCWDCjiALzSOhVV+gb8BBmssNfUA0hAnIwiT2KN+D9xhviWe8Y3yUBwMzAdPx4wFhXobDx9Wi8A2GaiFMA/Err4Bn/Qi2+z3kxRUOSuqBaWpXiVIGq118vnffSI0A3R/JbttUrCczBiW0//vg6V5O7Qrp8xUUQybSIOHrXLnmEVq8MwJ46ne/TRx4GMK8kZDFWREReryQsWnTlKwn42/TpnhUUoUJ4XJ9JWLBAfkePTrfuga64yDSTs8YlMhlfddRnEvAifZF83jyPPJNNZnkTJ8o1CurNZKxQB2SRh3kfvtIzCUB/0iRpTH0mARlrxcV/iG31mYSPPpLH+Gj2vXt9hf5+usKVBHPFxa5dpd/5hQFBLMw4f76AYeNGgdC2bQI/IA2rAkvmZ7AYEAL5upb8uHG+zyTolQTG7/ff99xws9nEEfSZBKCeyVjHo7axTsKWLaK/eXPBJPYw92c4z3RpnQTzmYRNmzzPGMC8CQlSOwG4LrCg6yToMwl5jHUSkDefSdi61bPiIogdNUoGAF0n4Va6kvC3UsazLTN64i+GPrqTYRNmydhEbFu+vFxLokfpPwZPhmVgquiETyFHjhKV9O8vEx8CbcJtoEm0QlAM/h40PpkDSIltiYUZnylrXknQ2BbQlColp0CgCgRxv6zGR88Y54EOsQAhAN5IFMBfRmzOF0vIRxuSYJdz/+wzGfD1c0ygH53sa/CCcmohqHnAWP2eUjglkzsYmR2EYVvCAf7C8igB/cgAaI2RUT50qBXbZizb/pGYWjDPiYbFfm2Yx9nwnb0NH4xo+NkZ3PA6N3rbTBLbvthGOhEwABvCW7qVX4SAKBzK5AkUQVgc1c/WAtEWLYTUwAlgAzMIoIGwg/ycOX3ZNldDAQ+gBSSgV78JRmyLX+gz4/AgDoIM+hngATkxgYafQBf+1egeZwGuRB5qBnYiQ4xC7S8Y31ElcGYHGYxnH/3YBjh18TzgTXzADngm8sVBMB7NqCpYsPQzVmybIenECXn2UzcA5HL5ClxnOnJEehcQ0HMgAD4qWFBQAmiY5jBtoYMZP+lsQEYmdPaa8YpEdmPxTX2CqmNHse3iRSEsdggrzPeyADH6wQRlYXMCDQCEftTq3Vs0k0M+iIRhsxmftEEzKJwyRc4avkah2QjsU4X513szD1GEHX7Z9JC3Ej3qXZC/J09myMNbGZJuDbalOWMPpFzeS9e5/f57WhX2yWccTzaQ0R1igp4UQsCM0IEhlv077xSYATZCFoAKDuEpvWB1u/GBW0bfzZsl5gCx8fFUkI5tJ5yOerKukOkTxlMrVAE+KYWPgFj04wX4BdrYIRP3gQoBMMH1AQNXJgIv/+u9mYdAqQLVRKlPqcvRGxtbumSGPSqbIclv2fafTuaVBPpYn0kgpGU4DQuTmwkgVd9uAGc9e8rVAH27gRAb2tXbo3ffLfea/yzFXFpxUa8kREZKgACPw6To1xUXUcKMr2VLeZCQuRJ/cRumh2T+J9Otwbb/QvJcSQAnRKx6JQEIAaSff5aLVAEBEj+CZyh1+3bBcFXjk6C6kqdu+naDV0rHtt4rLh4+nPZMwo4dnm83oH/LFiFiNFOFfrshk/EhqH8xWSsu+kliVK9TRyAybJg8twu99u8v6PnqK7mMRVAApL75RoZu8ARdEgUDKTKZ7Nxz6e2JPn181ZqJoVjXWmzeXCZTeILeFmvSRHicaZE+e9iunVzQgGoJPZimEfwSLzscvtr+G8li26tMkRHJnd77KaztlrBOO8KqzgmrEhlWY1FY+Wlh/WLD2m0Nezc07MEmYUEHw57tFdb/UFi29mFvhoTVXRpWcFjYHfU8W54eYSGnCS3MTR9n0BQ+LalLuR/Dmq0PKz0xbMTJsEozwt4eLaXQRqXof+CrsL4xYc/1DisyUjLf+jbs1UEikH+ot85/envnnTSbb4Zkse2fpPPn5fo9kab2m+74/NUd70O6vAtjPn/5Zaj/s0QAq+vOeCv3VhsRIWGvt37dQf+Rm+v7H/9astj2KlP0nuRpYckKpYzaiJLNtG93Uobr/ye2RYvSbL4ZksW2VvKbZLGtlfw6+S3bJiT4Xh2PjpbN+zK5uf3ZZXjyva+sx8bK6ouakpPlgtTlRbzLmrfmCFS9b1j92eaj/9ChNFVXNA95vb12+aErbpw+Np865flLwbNnfe+k+fwlhNdG0EOXnwiHTJtPnLiyMXoX7i/aP4NWdb412NZzlyw61Wib1LQt+tKm+56j7sty0strZnTqkcNpr/OfdaXGeMQudYJKmkU8Bc2jqsS7Ll/53731nzH0m9b6FDGtNTPTLEkv72PPFcvqX+98nyKmgFmXJzM17mCazX8kYrPR7D7y3n99lBs5h2NvgnUS/lbKeLZt106upd55p9zuLFBALoDef79chs+XT27o6xsyCGTOLDlPPSXXVfXVLzJz5pRrso8ZL2s/+KA8qnLPPXKDNXt2YQ1NcXHyGBZla9aUxwB0aTEqKl5cbn9VqSI3cKdN8yxtNWuWGMDRl16SHX3RAM2FC8v+I4/Ik4kc8tYPvT77rOfRBX3Mu3p1z7MH1aqJPcizv2SJ3D3DwoIF5ZFJjCxbVu41q35dCQF7KlUSeX2UPThYLKFq2sFmkzvRyJcrJ2boKw8PPyw3pmkEijDXQgzhp5+WUwsPl4ZS+bvukqo5zf37PTa3bi2P5lDF7cZ3+qju3XdF7KGHRCGZt90mR2lkTqRiRWlAdsixVlz0Srh9/twnAnI5Akr8GvD88oCCmwIK/hTw2vqA+6YEZBob8MC0gBd/CLh7UkANV8CzywLK7Au4f2pA7qiAp1R+hQgX2uSRf9AWUD5W5d9+5bRZRcTUi5UrXAx4aVVAyd+kVKldAY/OCvjifECWyIBciyXniXkBAXKLi61ePfeL9+wPeHl1vnv21si5MuCFlQG3jxf9b28PyDLTI599Xok30vTbJv1R5eU9AfdMDsi5KCD/j3IKj9kD8q0WYzgFzM42N+CT47JfaLPUm29VQLkDAe/uEZPe2CiZWWeL/G3jAp5eLEZmmxPw8TGPPEYW2yGn/NzygOI7Ax6YGlD0l4BX1wXcGxbwweGAp5cEPDE/4INDAY/MCHhnt4jRPpjKDo1DwSJbJefxOTRLsTyHTZt/3pxc9vmYgIcjpN3e2ibynGmx7dLgn58JyLs0oPCWgMy2gIemB1Q6IlVUTZDeeTii0JNpSq4z+S3bmuskwB264mLWrPL3q6+EIosUEX4cNkx4tnFj+ftk+nUSiLZee02eIqDITz8JQdAQPuskeD+TQCgN93XtKrduod01a0R+/nwPiUyfLsZ06SLg1RUX9e2GPn3kWcXnnnOvXn3lFRepHUl9u8FccfGXX9JWaKQKhpPNm+XJB11yd8YMjzzFZ870LH+zcqVnnQROirNu316qhsEnThS2bdpUlv1+5hn5jK6+jD9njmedhLlzPeskwNocnTRJ1CKPwL33+q6TYK24mBHJ83bDBx/IEyzeKy7SETQg/dWokaA6xlgnge6mnT/6yLMSY926IkwRlacr/2zFRYdbfGHCBM9LLozo8fECD+rVlZQvJWZoEBygapIlPOY949ldxm99u4FY4cMPr/B2gz6TAAgZZXEK7CEaAJbe6yRs2SL74JB6mzQRROGkOFTnzrK0Ajgn8447JFDAyKJF5e0G5HFJUKdrhOrbDTTIkCGedRWcTpEvVUpeziRWmDzZs+LiiBHy1C3jBgUHDvQ8l5b+7YY/XXER/cQTf7LiIn5R9InotDO/vuS3bEtsS//j6rQd7V6okLADvZUnjyA1i7FgAgIwIzm0vr7/qk/Fwl9gl1gPiMCztCzw0mcGYy+9WUtsW6aMlCXKowjyhJN0FZEatEJQDIeCHiUR0A1cOIo8lsBrBKf6Bi37VAG4ffTD5hCcrjKnzy1Cf3Q8luuLv8hjp76WhoXIgGx9bdfUzygCw2IP+qkaCzW2pUH0YW+GAX27F/Bhhi6PREzKX5QTXOMe6Nenejmqsa3K32ks94eRZmwL4glXOXHM5mSpjgbUV9rxLo1tOUojcyLkMJlgBw2AzGLbS0nYNvtimejAgDQ7ncVGuymA6Va6EgAAPDqX7qad6RSVB9Uqz/aksVqmvpx9993p2DbM5SjaS/Tkzy8bPQsYypf3vCxDDl18KZ11pb56z64qeX7Offehco9vqfL0liq3z6vy7PYqr+6rktlZJdv6Km8cqJLlh1KvnzSLzAh1ReVtLCDEKtxk8GDZQafCALOZOIJnKsXpqFcRhWtgEnjQCSiZuuIB9jCLYp4EREEURpJTo4awIZbTILpeGueL3z1hLIrGLI0NKtcXfMaPF2YEZhTUmWUWeTmzdO5L0IVtN10YnaufZ00GPBpttBvugH7miIQITOlwDTa8Ce9411hiOHNmi22NVWloL5qefqJR2M9hLCtDK9OadCo7wJeNQ/Q9kg8bq64ABX1xC0AjCSAgLyiDsoQAJikcOSIEquuzqDz+wF+goPL6yjaAoyI6TNfIQIzp9tUk2FYZEIXYTx+jFguhKlCo+kHSPcbioahlA9Cciy5AozyIAfxyFkiiSs8OMX5xAAiaNkES+RzGEjN5jWVndfkP5MnHcpXXgohRI+eLAfr+O2GI+S4Z8Q7FVSzHpeVvchkrNqg854INZFavLpELf58yFgOisyy2vZSEbe+NkLala2hPWox+ATw0FN1BpzxprFZMM3KINsxurPOiCKeILsWijwPyl/YHgVmzpmPbiaccWWoKcijCL8MqSnIZS97cY6xYRH+BN8CGBo7eIUvTNnk80pzG/HWaMe5UVJbqolnxAACyG4sfoZweV/1KptSIDVStDqg5uh5CXmNxZAWnLirSvbuoArf6QjPa+EWzro6EmL5p9qCxZA+OoD5o6v+fsRAVVWij5crlvb7t9g1nR9/fVk6WuiDoJ4zVnbzlDYIWsgausDanRkU5chTNeen9z+tOfsu2Pusk6BQsKMhzJcH8dgON+913EpQNHSoC9A0dyeyeoE/7ldnKr796riTMm5fuXS/9dgNd2Ly5BH3R0VJLJmNpR7pcryTQzcy7GbTRrG83MJ+6mhRjvN1A/FK5su+3G8wrCQsWiH6fdRLuu7TiIlMjRuMff5RfLCHEYJLIWO1zJUHXSYAvrnglAVR9880VriRw1oS0PuskMGOdPVuaXZ+vYdamqiBW5DkjpmM2m3AExjDtJRCOjBRjmNxZbHspCdvqiovmOgkgAQB07Cjdcfm3G5Qf6RQ4l/zAQOkp5JmSM63+sysJ5oqLdOju3Z63GygFUL3fbgByu3ZJfqVKTaonXC3b6pUEOhrAZ/L6doNe+zLfbtArCU8aj6KPHCl3OzIZayQCic8/l9cumOqZVxJwRkCl6yTgnjgd2kaPlguAz1xaodH7SgKkifPqtx5q1pSLCXolIZOxrsJlKy4K2741WbyMKOf4cbmi6P3tBuQxe/lyGQOgFNxq3ToJ1T/+uOhTh7xO/bqS37ItHcbMGsgy19DZB40OgwAg6Ak8kc9gBaXSPbSarmEMPuhOmhuQ0Vv0H7Mz+pXBk1k5fWySQlycSMJBhBvIA3p4CjH+QujsoJ8+pl4KUjU71A6ewPTVJHANCQIjhlk2upwxAI/JYVxmxcFUv0a7+i4vnoZ+hnSGECZWiBEmNGok5wvcNVIAqZwp8jgtZfFnJDkLCuKxwJSJJ6aCdW95NDMy8cu49ZDxJjunSZDOhAtQmrEtrUpBpl26pjCNjyqqps2Rp32ICyB3MqkOBuG8wH3x4nJpz2LbSwnz37p/S+BzUwNfnB748dbAB4MDSy4LzNQzsPj3gU98J/mVfgrMMynw4aGBVX8JfHlG4B29A+/rH1h4QeD9AwNzTwx8Z3lgKUO+xPeBL9gCa+0OLDgv8J4gQGqmmbaL9QrvCHxoSGCVzYGZBwe2SAgstybwzUVSCrUF5iIf2DQ+MMfYwHwRgS2OS/4LttL5TxyMuapumjH5fFSRrgIVff2MmT5xAzAAcoAW6AJg3CSTsUwzoQwBAYCET8lhJAafQAW8gRlAAlpyGkt9Aiq92Yt76u1cMKZXJMAtjlC3rsgTgcKzhM84L4DEZUCshg741G23iXJQXaxY6bzRps3Ctk/1FS9DnoGNUEDXPzPloRF4FiMZNgC/3jjJn9+KbY0nwPR9f4ZH81d3DhyQyNR8ksk8GnNpiQD9q0fZ1ye9yPFebIEgF8LVB56QQSFiMcbDUiqvmbqjYqocw64mUYThGvqDRhmTYahVqwQrAAgY4d5McBjVQST4YB84EoHS/Uztb79dAkywCPIYBijICA/KNZaBggk2CYGBEXErKASvDN3kwOYMIWvWCGWjBP1Qp5ptbnpG/HKa7Jw6lbZOAmyrhlGccQJXwSofedVw9Ki0A+eoHXGVbfL/pVuDbWlOSC0u5mJcTFLcwZS4A3/ExSbHRRu/By5K/sFk+SWfo8hwSGSS5Jd8xEx5/h5K0UPer7ycdclzS4YG4/ewPAslYqYqdg6lGtUlydFo0Xw4NiX56tYVmBGeElXfJuHkF19IsNKli3CrXt8gBipWTFh19WrPejEAlXgWdtNPn4HS7t0lSujfX5h37Vq5CQbTEUNAdh06CN8hjyqieL1QAPL5266dXLclIga6EREywCMGwEzoKvyYg17ydO91ErZvSx1dbanYQ4yFE4FhnMJb3nRz4Irvs2/kFy18dY1yFclv2dbfEx0JsBj5ib5hMUbvDRtkkAd2sGR4uECK2RlsCMh0AsjMEXYDmhwi5GSohz1h282b5ZoDoz24ZO4JmHCAu+8WPfryOyhnHnefsQ4T8lu3ikKUwMXmYv5Xk/CQKVOkbCbjbiTTC6IMr+8h/dPp1mDbWyDNiEid9eWihHGRCXVaJbTrm9BzeEKz7gmPPZ9w+2MJ79dIKF45IVOWhB/3Jnz2VcLzRRMezpMwODShQi3ZyH8iX0LvkQlNuyUMnpBQumrC1tiEEVMTCpVLeO7NhIZfJ3QdkpCvWMKg8SI5YExCjldFvttQ+ds5OKFElYT+Y2TfviLh/pyiIUG48c82YhUzbd/uHvr5hoRniyRkfSEh6DtREmK7vMjlW9GiaUquM1lse4PS8eMyStP6tWvLh9SYgA8cKM/HwGhM//v0kTl4p04iwKQGFuavfnmUGTq/RBMUYbJDGBscLPJNm8rlp549hU9bthQZdvShNJTA1AQaxL9UgTwKESbEQP7q07hxEm5A0JUrywwOw5hzmVe6//lkse1NklY6k/o2iwvpGhfSZHtIq90h7faHtPgtpOqykMqLQxpsCqm7MaTS4pBB50Ia/xLy+aqQj50hHWNDGm6Wjfxqy0PaR4s8mfV/Cgk+H9LpUMgX60Wy2Y6QtvtCaqwJ6RAjkh0OhHy6QuTJ5G/bvaL56wOy3+t4yEffi4YQiSj+bCOeMVP0vpRuXx2VWrBTlXQ+dHmRyzemcBmV/I9td+zY0blz5yb+nho3blKzZpPq1Zt8/rn8sn32Wbq/PvneOxzyFmbfR4ZfDqmYuamAKaNla9XyNewvUt26vvZwCledBgwYcOLECd/u/DvJYtubJCVeSE04LgHDTb6dOpVmc3ISNqdeLvP/bt5KrjP5H9teuHDh0KFDMVbyt3TkyJHk6wuELba1kl8n/2NbK/1nk8W2VvLr5G9su2CB3AgKM9Y21C3MWI0wPFwuaPNLDr9JSXLTiXyEyeGXTRct1L979nheJVQZVWiqNVc7JEc1XL6jSubP9+ybyidP9uxwaMcOz1FVi2EzZngq1U2r1h0KmvqXLZPH/VSPtzA69+4V21RMl3Zkhxxs5q9aotXxV8W8TwTJ06dFWGW0du8qZs6UG7IbN0pxNR7DIiPl/WZa7HJ5s9EQczjc69eLcs0kBwHOmhyEqfr6MODXbHvmdGr4lCTblGTbpD9sdB07k/mbIju6TUu1Tb5oC0uyTU2Rjb+6b3PLr+6HGRrYwi4VQRv5qKLsVEODZpp6EJOChrweNXdUBmPEEsMeT36ypyIxyTASmUmGbWL/RY9m/qqwaY/YoKamevRP0hq97Ddt5rw89qQYO26jolTZMY23Gfuq0NQv53vpRMLSt+fkSyZJQePE1X7v8/XUZZxjmv3aAl72aHsibzTm6hVJSRcvPZxzrcnf2LZkSXlb4bnn5ImTZs3kkZFSpeQpvHfflVvt1arJE8uPPCLPcgUHy0sHI0fKPaI+feSx58qV5RZ/kybyUBRc8OqrcgEchYMHi8JGjeR5gK5dRebll+WGUpUq8oQKGurVk8f9Ro8W+RIl5KmAwEB5rApionZs6NtX7kFVry63rR59VJ4BaNlS9EM3CAwdKjejmjeXh/uoCG0ffih3mSgyYoQ8e0jO2LHyEHjr1vJ06jffiA0DBog9HTrIk4C1asmtM06BInBWx45yvlAAmNLlIMjhZENCpBRm0DJsHKX2EOMFkMKF5R5Xq1byEvD+/fKMTpcuopwqateWG180HRv25MwppElrjB8v8v36iWE0IPb37i16vvhC5KtWlbtwyJcuLQ/otG8vz0IgQF00VJEicl8uSxZ5dgKbaSuE2das8e3Qv5P8mm0PRKfW+yLJ9qnddl9D27iztvpRtkJDbE1+sH0801Y10lZzvm30KVvuzrbS42yt1tu67rCNOGYrFmJruV74ou2PtvdCbW022kqOFmGUfDjNVnuRbfhRW5bmti7bbAUG23J+beu7z5azg23QQdsDASLffpMc+tJpa7Za5Os4bAWCbSOP275cZnuxl+2beFuxUVLXWJetyDe2elGi5NUg26gTtsrhUjWGjThuyz/IFrDMNuG8LVsb2/uTpOoc7aVU51+kCgQ+irDVmGf7YKqc0av9be03294dawvcZRt6xFb8WxHGyJbrbOUmiRllxov+bK1tzdcKIb7WX442Wm57rodtYqK0xuDDYg+aOWU2ZFDCWffZa2uw1PZyX1u7TbbuO22v9LMFH7I1+N5WJdz2v6a2MS5b3cW2QkPlBEMM+ylYJUIkCw4W/c90k2YpMVraMHcXka+z2FZqjLQGLVN3ia3CFDnNfH1sw+JtjVbY8nQVkziLNj/asjQbWzysZ9ekM2f+a2wLyxw9Kg9lTJwocZNSCS4NH23ZIh6Oqz/wgPvMGXnyFHqNjRV2Js49fFgoCao13yV77TV5TFXfJXvzTSEXSGTpUuFxOAWW8flSzu+/p31z11yVBlqEBH/8UR4V0HfJnnpKiIkYNl++tG/url4t1jZtKjyOtnbt5GFDisTECO1yFgTmL7wg9I080SucDjnCzgsXer6UAzmaq9JQL7WfPSsNAtXu2+d5l+zgQXkPDTM02CTB44cOedb4OHlSytJQBw7IQPL996KcJtJ3yXr1ko1DcCsTCAqeOiXyhLQY5nR6vtGLHgYnaLpHD6Fg5Blg5hsfZcmVS2JYRinM0FVpUBUd7VmVBmFGslWr0nfn30t+zbbgiJaTdnjoIVm70vzmrj4ZMmiQACx/fnmGhGZnchNzaVWaxESZVQAe4Oq9Kg0dR/MyOgJaoAvewOErr8iMSt890y/lgEnAQC/gJpWMb+7qqjTABrABObBnfikHPAMz/Ih+xDB9Bwd55kPPPiuOQ9UvvWS8TBYlVSBAtw4cKJAGZuXKed4lw0HMb+5iJMDQb+7iUOh/5hlxQBLyQEjfJQPPTIlwRuzRdZTYSCgBjbgwjszQPm+ee8UKGb937pQ2xNQnnpCbWbowDacA4Mkk4TJIgnD06zeBwKHdLi81HL/0LhmtQUtSlkCK0yQWoVnobmIXTOIskM+R43jt1ohDKteZ/JBtASXkSIswOdXAE8YBHLAt/EiXm2xbyfjkHPzLrBwKgCNgUmiajmfsg22BGn0JOlFIWwMsMAQg6Bu8AnaGYtBAS3/2mcyvlQ0BBOxJV9HxoBOYAgUwRxEAAfqJK5k+K9viML/9JqsKYS1cD9vStW3byoPZFAGI8Dg5NA6kpmwITHEGnIQYHOxiLa7IqXEKyrZz58oODpCSIoHznj2SA9tiKgE+gXaY8WYtR3FOHS2gV0CsbKhsu2SJKKeJCNVxRXyYjbDXZFsYgbPYuFEMw2lpCvIpCPUDUFwRdka+ofHGMwbAtsjrmmGffCJfzEYV2MUPaWSEQf9/nG2DU6UdYFu8HdjQmJMmyaCrbAvATLalwWlbGpMmPX9efB7wACf4AmGQRpfR9TQveAO0yrbgEIAp2yJPp3BozBjBHr0AO4Nwep8cZVvABiQIX+gvQIgSWAyYgfwaNcQw8KlrlTFUw7YM/NgPgVJKv4iOAN3KjApIAzPcATDAtnQ0sMdZEKYi/fgebAuK0A/bTjNWK4VtOUqgA7Xhs7QGxmMPmjllNmRQAoxxYbgVtuWkGIpAO+yJs+MXsC1FIAFGL2VbMikI5yJZyXgzWF+IB4eMWDSyyhOo0Rq0DA7OtFLZFm/CHuZ8mMRZYEPOnMe/aD18WOp/j21BCV2bJ49nxIMEmeNDWEyNFy0SZBAqgmb6DC57/XXBJbgBPbAbUKA4eKXPaGtwSd8z54U48uaVfPiX0btkSeEXRvv335d5PRqohdYnGkWe6gjr9LUWuBWapnZ4XF97xz0AOn1Gs774ovgDfYZ7gEUAhHdhOdrACvvly0scigHkEB1kzy4wBRbIYyeIefVVKYU25vgqzC9eipFvvy0syT60S4MQC+gCkkAfeIFOGodhBmshXyykKfBnrEIDBMrQDYmjnFKcKT6GHxIF6OoKBFxAnAEJezg1DINe0YmHowfPxEmohY1Ih3agdoaxLFmkWT79VBqK8AHPpCkQoEaaEf3Ir13r26F/J/k32+5PGvzxammH+++XvtBOoVVpdhoQb6fdnn5acAgb0uB63YYugMUAHu0MEQMn0AKeGYlBBdxHIwMV8uk4Oosxb/ly6Qvk6RSgAiaphS6m4BtvSBdDsvQ7cR/wABIYA94ADJmQIyiifwEYhmEDIAQMDN74C8bQiVREKfADMhWTIB/9nBHFAQm+iUm4ia50CgKRx+n0bRo88bHHhOlAL8oVtFiCs3DuOCO+oJenwCcyKOHs+Itr6wU9vAnn0gfPcdJHHvG0J2dXoYLAm5ZBjEOtWkkjQwUo4ddYhlF8kIgBGUgDY/QyF7/sI0bMQXPpojk6Icid+3iRCsM7HP7vsS1tx1wJJAFEuh9a1NUMyKHbHn1Uhk3akUx+Qd7DDwvImN1DwQWMRceBCL8ggMysWT3rJ+hihsjDyFTBX1Ci4xuNTh9wFD69+25BJPT9+OOez0eDD+TpP7qfnOeeE/JCDwU5BKlRCuhDryhBOdBnB0mMQRXKCUk4iv2Ahl/ggv8w/FKcfCIXoJAjhxiDPPkgnlpAg3I3IQOnjCQ5nBF1ceL8gjzOCy8i0se2e+8Vl6A68If9+A8W0kqYipHYg2aVpzHxbeYQRB9UBFFmziy+TRGQijyWoBwK5sT1KFVjJ2qhEjRgMPZQ11PGyns4CSinOGe3fr1vh/6d5N9su/ePwXlCpBF0kU9AqLgFhHQi7QawaUlYDwgBmzvvlK5hAw9wH41JPgEmvcMQy998xqdC6RGgQm+ilu7Q2TSsp6CFnth0nQ3gqgsJcpQdyIX+xQAQAuBxE+CEMJ2LJApBBf34lLE+HHRGdRrB3HeflIKe8hkLigJLIIEwGEMezQ2NpaLBJzZwCK9EHmGgSxV4LvvkA1SImzPivDAJV4X+aAFGI7wS2OiSclSBDH6BSbQYdnKa/OU0OVTKePedE+Rc2Bgh9FbHU8byeNhAOyOjNyeQp+VpcDwCM5DnXGh5zMN+mpEiNCDGox+1VMQZ1a17/NMmw5v8+t9jW3qLAIoZBwMgxIH/E8bSxKBh2zZpPnyewY2+ZJYBR5DDjIyWpV8JBJijyfVzm0xbIAK9IgnRMD1RpmMeROAwcaKQBaMcITBjJiMhxMS04rbbBOt6KYN8/lIpExZwA8VAKLjKr79KvRTkEJ3KXAYv0o870N87d4o2Ig5dQJq+pNTs2TLXw1QiCAITOh70M82fOdMz1QLijP8YBmj0HhqnvGWLEChFwBAzO2BEKfxk7175JbhgwGCE55dWArg//yxIQp42ZLbFoI2peDUhFS2JfpqIaB3GJMrgl3ZDnlkViMcYCtJonAKjGoEYuATWzDEZgQA3GxNh+oV2QC1zRk4NOxlUsIcQ6dQpuWz5X76SEOMe/Ol6mTfoJRc6lxiKMJZm37RJ4NSmjXRHZKSgaMECIQK93ETfMYmmhXftktGaXqCb4DgiYvqdTqeX6X14irkw4xlkATyY+uid1REjRAB5QkWqxgXQD7WBByqlWzEAAINYQE4sScxI8Eg4AtSZR1MvbMXEEVBRNZ2O43AWc+YIdOFcDcOBNPngHJrDePI3b/Zc4s+WTYDRuLFgiaMgn3kPKALYOAjOQqV4KIiiNbBc5VHIiXN20C7BOH6BhTgsZ4EYbvXjj+LsIA3/wqdAJmKcNY1A+6CQ08QGmhqvofiGDSJPDEHsgk4Yg7pw4QkT5HQY8JhDwLw0CyS7cKG4obrDb78dj3YNH5J8I9k2NTX1woULQF93EhMTU42UaKRUcykTI2Uk2xLeAxHaF58HEL/9JiMY/QqRQVjA8aefJCIAYfoMwIEDHrZlDgWUaX22s2c9bEs3wLawCWxLxwMUeIrWh23BEEilY+hCht81a6RvAM3Ro8K25POXSvEHKIYhl05lfoRLUC9cxiHAtHKlwBQ+VbbFWrTBbiiHPQElkIXrNZAEOsjTzUCc/ga7oC06WsINasEwhl80Q8GcssmesC2ui79RirqQx7WUPcElbEsrER/B/rgTtI43woxga/duARPjFi2p32KAPTFSL/DBtshDNHgIxmAbjQZAoQB4Ad/AfpB9110ySrExCGEM7UALMJ7BttgJ26KHlne5hMT/y2wb6x78mfFUIrQI5dG50AENSLNv3SpwguNodgiO0BV2g20HDJCZNaSD29O5e/YIodALdCtdDEHQ7ybb0q0QBxykbEsnAg8OwWUqDzvDtrgAKIVtwQOV0q3KtuAKkMPO1IiL0ctAHUqlZ2FbAAC0qBoQ4jicBXEJ0GVAhU/pXwiUmBGcUwXGkw8+gTogIdoFGF995cESQQ/4Z4zHMZVtGUuIgkEUrQGwYVvklW05a52K4RdYiMNyFoitWCFDFM4O0jCMijhlxDhrGoHzRSGniQ00NV7zww9CC8hPn+5hW31/HRcmtKK65s3l3CGB7duFbR0OcUP04A67dx+PPTd86A1l2x07dsyaNWvt2rV79uyZOnXqggULjhhp+vTpc+fOjcOfvVKGsS1NT3P36OG5rEPURlfRrASYNBbIIF6gV2hH6AZhcmAQvb4DRwALvflO+9LN9BCNzi+MA47pDEBAuyNAXAD6wRYKqQsmhSzoDCAOaMAxVTMSUik4ozuZJYFO/AR79EqxBtfox4U0HoQxcRvQieUYQ0WohQo5yoCPqfwybEDWGAzc+xirJWD/p8Z6j8hzpiihOqyFLilOdAPIkEQGDFEXEOEXQPOL50DomIo34nUqz2kSZXBqaiquhT0qT8NyFOP1YQaqQwOeQ7P07CmNxlHmFngCLcCv6scYqgblyFAFarGHHTJpW04Z4xnA6BE8/DqSf7NtdPLgsg5pIgYhxlqaCLTQ4DQ7XQnMADbdQT4+SYODLpqaDWDQdDQm1Em/0AuUgh/BG3rIpzdpYRBOdxAAAn5dTAtgUBDX0Kv5QFcvzQM5vQGg99zoQYCnfxEGCVgC+IENyEQ/ICRkpkZlpXfflVIcQg+9jFpqh0zJJyJWtOiCtjiOBuwYjDbMwwDwhjDnBVAZzqkU83A3TpMhHywxTjPMUFyvJ+hNCPwCWCKPAbgAf3FzfUKDBsRsTpkNGWqhOmymGVGiTyVCxyqPa9MUeCtmkI9f6DMh+gkfHW8AOZEvaqmI/W+/PT4qYniX328Y2yYlJbVv337Lli2HDx/u2bOnw+GAeX/44YeJEyeGh4dPmTJl8eLF3uFthrEtrUmL0Ad0gF6f1Qc/2Wg7uoq/ZIJjZNjXjaZHWPc5yl9iSciCUpqpwaleEEAPAvwlHyiAJ1OP3h5VJQhocbWBv1oEJfQf03wm1LoRz7IdPCgRJTtmPhvRJZsp5rNjbgQ1BIy6zw5/VYZfAm1iBB95PWSKmTqvXh6rLl702Ha5PaYNWsq0zVsbZX0KImYulXtNyb/Zdu8fg1+dJGjB1YlJwZgJTnDLmAQfVTWwDeRMuOogqvuKNDYg6g1L0A4sKaIRA/KUQq2uya0ygBNXh80VosiY2tg4pNeLyScYvLy7iWrpaLNzdccbvX+2KVTMvz5o4e8V8b93rzyl413wcnlzQx6d3njz9pHLN1PJFe03HcGr+PF1u4f3P3fD2JYYtmLFiq1atQKI77zzDuS7Zs2ahQsXkkOou3z58mnTpl3EXd20296OHTsWL148Pj7eV8s1JMZhhiliMeYdTCWYNwEvhlnglTmzDKREqcwymOzAjMRTRKNEl8jrpSVgOmKEwDo8XCZ0xJuMokxSmHcQujK6AjVdz1uvLjG66l1UoMl0j0H4nnvkqRqdRjHUly4tT5YgDKwZG5nXMIQS0iZkzIquVvJO/zLbEi4A4+rVq3dn+LzuFBuTKs8kMFN57jnBD1N4GHPDBsHk/fcLtIAxYSDwBmxM3jduFFQj/8wzMtuFUpk9QJcM5Gh44w2ZJoNYJvV1jE8Y5M0rOCek5eiOHRJBa12EgcRr4DNnTqmUEJgqihUTRtOn0DZvlugYIiDYzJNH6rJS+qSP594wtj169CgkC5OOHTu2QIECp0+fXrVqFREuxLp9+/bvv/+eCBcKNuUzLLZltjJunLAq2NKVqutfervhrrs8bzcANZgXdM6dK/Lml3J83m544gmBOED8Nf3bDVmzijwY1S/lAEfz7QYmSpmML+V4v91AtIgwkya8xeEQsJJpse0/kP5lttWUYbFtrHtwzU0ynOfOLchcsEBGfeYZTYxPz4IfIArY9MIRuDXfbgDq3m83MEOCE19/Xb5lW+nSl3KIMMgkemBS/Oyz8qw0RIwv5MqV9nbD449LpR9c+ubuuXOST5x7+LBczNWHxJHHEayUPt1gtk1OTmbAj4yMHDVq1IABA0aMGDF+/PiNGzcuWrQIYh05ciTk+49cSShpPN358ccSmTLUg5syZWRqRizw4IMyD2rcWKLXLMa3lyFffWS1cmUhX0IGYlUiVjAKztCgF78g0xdeELQVLiwhcLZschUJEmdi9dZbcskcNPMLjYL1u+8Wr+jfX6JsyJoQY9IkmYVhEr7Ro4foB+jXt7Sgla6Y/JptD0Snfllyl739GvvjjeydN9q7/mR/b5R9zFF7+e/st1e1fzTZXn+uvUh/e5Mo+9vB9le7y6GS34h8tq/snTbYy4+xd1hnf3eEfcIJe7bG9txt7BNP2Qv1M/R8a++7XTI/nGCvPt3+RBN7mMteoLe98WJ71gB70yh765X2KpPsD9cT4YJ9JSdfV3t4or31Cvvbg+wTEuwd1torjLW3Wy3yzb7XFwuszdyYxPbqdePY1i1jdazT6dy2bdvJkyeXLl26fv36M2fOnDt3bvny5evWrXN5f3UmA9mWkTwiQgZ8NpqB0Z7JO2PytGmSzxBt/qrY1EtLzCDDL8J6aM8euX+qi7kgz3SMHZSwTyl9boGjM2d6/nKIujiqMuSjSvWTr8LhxqIzbE6nfHzKShmd/Jptz5xOjZyWaI+4aJ98xj4jSbZp5+2zUuy28/ZJp+y2C0J/U8/Zw/+Q3yln7ZEp9mnnLslfFOEI43dWquTAp+xMPevRMzNJMlEyPdEedsZzCFVk8ktBDk0+LcKaj357quRTF8Lotxn6Vf4yurG2tWsz4Csl1862fytlGNta6T+c/JptrWQli22t5DfJYlsr+XWy2NZKfpMstrWSXyeLba3kN8liWyv5dbLY1kp+kyy2tZJfJ4ttreQ3yWJbK/l18je2PXRIlg4yt7175Tc6WpY+YV//eudfuJB2SDMPHEiX41PKW4bN5ZJ9b53eOz76ryhj7vOLvLc2cztyRBY/9k5ImqViYtKeJ/v9d9+6TCXedfmc4xXFqPT6voD77ye/ZtuLF93796aY/XDLbKDp+PG00zxzOnXf3tTLxfx9UzK4zuRvbNu0qbyz0KSJvHuji6RxAl9+KS8afPKJLMnRtau8yzBmjOQ0aCCv27ZtK+vIhITI+w6dOsk7C7qa3JAh8jpjhw6yblC/fvIiua74NXq05z2Fnj3l8V6VqVtXXmqoXVv0jx0rSnQhjE2bxAxd/h1J/ZYE8ijBDDToq24DBsiKGwhUN5Ye5yz4iwYs+e47sc3nyWksx6QKFcSk3r2ltzX16SMvdFAqMFDe+dE1//V89esmGEDVNWqI/Zw1jTNqlNRILYhxCtRbsaLIf/vtdX6T8d9Pfs22Rw6ndmp3MbTVltDSU0IDY0J7x4Y2XB3aZU9ouemhLTeHdt0b2ubn0MHHJb/v4dAvloZ+GBkafCy0x/7Q8cmh3yWGdtgZ+oVT5Dv+Ftpld2iV2aHfng+t+X1oyNnQTr+FfrowdIQrtN12UTX8dGitpSJfe1lo592hnXaFdtsb2mRj6Hu20CEJRr2rQvsfCa04Q/KHngytbA/tGR3a51BowNrQkWdDm28KHXY6tO6K0Bab5W/NKNEw5mLoBzNFvrYztOy00NGJoV/vEKtC3d82/Gn0YOOjTW6JGZxRyQOCkkM/XyJK6q0UzeNTEJNTZmOn2U9iMzvo73c4tNG60K57QgceDR12SmzmaDmbIX8gtHx4aMdf5Wj3fbJh2DenQnvFhDZYJfKhAvB/bcPzli9P16HXkPyNbd99VwgF2tJ1An/9VdYr0KW8ChSQN77hkRdflLcbYZkPPpAWgmsYlXbvlld1IdZcueRNBNh23TrP50Z++knepCxRQhZbK11a4spChYSeYFW0wVyDBwu9btjgebEyPl7e9IUioa2oKHnTLChIXlpnANi5U167RJ6AFDOmThX6hvERmzBBKs2XT462bCl/sXDOHHmjk0HC503f8uXlJcucOcUk5Lds8eTDwkePysPW06ZJkbAwWeThpZfk5UuaheEEAwg2XntN7OesaZxdu+QcqQV7OAU4mhZAHrY9eDBdpTd98mu2jY1xB/VLdXUb6Mr8pCt8oWvmYle95q4xNlf2Z11dglzjprt6DXWt3uqKXOKat8z14eeu5wu4ftjsCrO7jv/hij3pGjrWVbmm67upruGh8vvqW659R10Vq7t+O+QaOdFVvLxr2wHXgFGi6ud9rko1RP7j2q5vw+To+AhXq66urHld67a5Zjqk3sVrXHledY0Pd23c6cpXxDVljmv2UteXrV07Y10d+7i27nN91kB2dh50la/mCpnkOnzGlfc1MbJKLddjuV2xJ1yDx8gvjlW7V7d6HiDBtvPmSojieu9j15a9rs8DXFPnuBIuIuYKXyAbO+16unYdlp1O/eRMG7cTtc6Nrk27xOave7myPSNHp8935XzR9c1419jpromRsmEPMhELXbWbupb/KDL/4qavN11n8kO21XUSYCjYTddJ+PhjiR8LF5ZleyZPFkY7c+krkMuWSTRHgl9y5BCKgWtmz5aAFH4010mADd95R3i2XDmhM1RNnCiq4CwIS9dJ2LZNOBeCO3dO1oqHtj40vmNGoAoh7tkj2iBZDTChwr/+CiR/sXDJEpEMMb4C6Z0YJ5xOMRWTCG8ZRTTBtufPy2oM9DykM3OmrMuDztOnZRigFAYwGLzxhmdtEV04kXM8cULs4RT0leVTp0TeYturSBnGtt5fgTTXSQBjBAHMVKZMkQkQHQ1Xff+9vDvO6EgcMMP4CiTdRycye5s0SSIAOlfXSQDhdCJdWaqUzHUJJlAFFPEI5Bn4wZj3OglgXuv94QdZfoF8hvNXXhGg6nfGwE/37qKh5pW+AomRuEDWrAJXBTmOVb9r93qxeo6w7fz54nMSLuw2vgKJZu0vAgs2EpMwMEli+sWZEjmhltAHN8Rm5o7Zs8tRXSqXiRoep+92Yg8y8+aJF1zfR0CuIeFw/z22BVUdOwr3tWolgCDKg2LKlBH+IsSz2YRb8+QREqRTIVBiW84QsqOnQQmYe+opiYsBDcCtU8dDfMA0f34hryJFBI7EhrpKJjirVEnInSrodV0Nc+lSz8cRMIZ9MA1iQIMuxfTeeyK/aJGYwZSfeJNDsDyz/s8+kwUZOEq8zF8sJOpEEjGfdRVKlpTxI1s2MYmKiI41wZ6w8JAhcp2BU9MLC7gNrE0IzxiAAeAdYsVOzproG9T+H3t3Aq7XdPYN/L1aakqJxhBpUK25r1ZLDY2YQw1BlddcxNSaaYwV8YaU6uk5KGJqDSWJ4RhqCKExVKgSxBDEkPA6ig8VcxOV73ev+zk7zzkJ3yfxVk/P2de+9rWfte99r7XX+t//dd/r2Xutww8Pgv7a12pTozJv8iohQd9xto7NtpP/0XDAc8F0CywQxJpDQJqjV6+AqPZCdjISpfEkAAAIx40LZGpozQeoG28cLQgPKGm55aKh11knTIBFQC9eFjOJn3AluJ58csRVx5YlP3La4q98JTAvX5DGa+wliRh+IBAUcXTOcQ6x3A4nfvbpExoABqgQPXene/cAbYL87rtbtvrJoD1n+LbXX/XBmDMejVVtRJCbbRbGVeYCjNLCLUyiYJh0wk/ypCzRs9NMElAxKXy6Kl6EfwY7qKw3alceMqpOvWHnf+7WKdlWK2qDhReurUm3/PLR9j16RAvNP394bTrGeeeNSeqcEMNr5EET9809d5zzLBAuiOvS4U+Hn8tquZ0MS3AvnRxhO7aiB01/+csBO5CVqJMnnPoB2lV4VRLFoIF+OimZZ55Qu/XWAVAn5KUvtFAcFQPtSslll1Zfvf3yiFwSPQcsYlt7VXUcH0h1iZE4cWQYnN+ry8wZOJ0wULAiMjoYJ5wLDpREYgySH4HHyeuQco30jrN1bLad+H7DIifHXKBf/GK4rqCVMAMtEAJgl3hzCTyJILr77sFx0AtgMEOApwkznFyggpxu3QJygJTopSfXEgXs7343bIRaNzq6i87Un+t9kcTpODcFMp0SMNYxgzElGJ9OiIV5MuQzne1IVKrllmvpsUobtr383TEn3h1QBDauAEch/wHmegMhTMI2TDpRq1BKBiZzjhFABWMBpauwDf9SXHVudyOZDAr/6Y5Cp2Rb3qVOG9EcdFB4Z7lmgejpoYcCl9oJjwiytBZHUrCv89dJwp8e8ic/CVLTl5JHgkiWZ6q/nTAhUKtvB25tyUEQJXEf+Bp8hO22i5aWAmFJYUAMBGuvHT3whhuGAyJy1wkfcECQuyz0wzxuUZK+XRkQ66RJcS9Acy504IgPUu+7L8haL82PAKD6bdq0GfuHH06vZlNzXn/pE3bhpxizpSU8i5mvTitqO9rWsdn2+Y8afv5GtP6f/hTwAzwAAAl+JQd2v/0iLIMoNAqx0PXkk+GoiuFQJM4CdeE5tOjyd901ghtuoN5X2aAIW+W0uZxWLEw+9bACGnjHCJrJjB0biBUDwdtdd4UhcFHRK3LMhZbZlBLuv384xQ88EFwsYoNe2m6+OTxoqGY4JLGk/eKLWwY2DjqmNl9LsO0fpo+5tQ6l1cs2MDwzCD9unzIlBjEcZ76Ue938gv+crVOybY7bwpx4Skx05pm1P/1fey0oT30gu+99L3A8y/ltX3gh0IPs0KIO/Iknoi/l4un2daHgiH+FXYKgan5b8YtEaM7hrT//OTAHeTRPnhwDBa++Gs4pYwBQcSKOQ81Q/vjjwey5vKNtzJjakk1DhgSncxZQoW6AvMiIE1q/CaxYET3ImmEwBicMSYqfw4bVju4ixladeMyUTzHlz3Wq/STpat7lxFGNySJzoVw1uppiWMDJU0/V/iRUV3Q6p9NPN7qFmHOaXcpM3ZLpWkdVV/LO6U/lLrVdP+nTbh2bbatxW+SlxVULAGAuJ0CS89tCVL4xAl3vvRdUCI065mrclgAKRnbAf+ih4WRU89vyNu6/P7y/+vltczGeiRPjNRsOBArjByBccf2zz4bzOHhwsG01bqsp4fOYY2JA45VXAvOY16ZU48fHOBtUMxx9A+Oy33xzy0m/qyZbD7Yd8faYI0fVYOmYIPFEMJawSZA4uuoSRCUmE8YEnHBchJIcFN6Jp0DxbNAl547ElNztKU9JrnTpSFuaRgK+0k+nTNNk0hAcE7ppCJleXx6GkBguD9L8k9HNI+YUeF1s+y/JtkI2bk6OozGMnOlZSQSSOQ5Lj7hSdsRYI0znWmpK6AHxLHk2w66o+tnP4oRhK4zYEIj79g28sih+Fl8+/xLhnnOvRI6KLRdBpYeiKkchuGPkaVOeFVeM6lV42fXvH3WlMKNHx6OB6ZJL1lbHki78VHhq7fQr7RxsXWzbAdg2x201NNg7iheFg3SKEUWlqgAS3ACHN9wQ0BJ9yu6WWyIE9OB8dr6544EHhm3mghSeVARAQ/7pwr7YOOUprxhQSgY4aZOdEoL0kUdGpJv6OekqGQIZUe/eUWnkVQUBFbVVWT2PvfvJNBgyMM83X1wSH7j3xhubVxvafNGcTnDb0dgW3eAUVbZjWfwO/rhvElEGCthhh7gEdihY30jm29+ujZwmyEBHjEZ+/fWDnQGXBpgW0KGY9dYLHPfrF+0KXhttFINf9Evcfvu4XYOhGLWvLXPJaMwrayhEOhtvHE0rC+dS2MBaa0V/QGGuIwk6bkTQuN6NyD3l11wzyLF+U1rbT35Se8cA9J9+OjDKeLAznczpK1+JSND2zW/Ge2A0sAHysMK0GJjAMwvGUFUaCD76aHQMEyZEnfAXYE5EputWPI/GOLkSvXpFkUAW2zJdBabzj3+MuiJPz8svhxGyGcTB+FUI01144XhN5vLLY1yFmemfFGbUqGATBiZKtTOSzrzm7rNTG9a/Plpf9WpZrQkAIMHUYQ+6NLpLIKd14BYs4RZ4sM8hh4QYiiGAU3S3+MvVvcpSY1AEXcsvH/AAVGwLkwDmFlZAg7agH+xTP+XuUgYp4KE1OQcwg2i0rxJCPvwrHrT4CR7u0tAsC6oZDnZjXDvG2uMt/XYfdNDf8hlnvJNgI2aDE50KYGBPoOIlQMKFF8azcFN0QQqgh37rrciCK4N/V1klUjwmpwS9nnJKPC9PU4H9/MMfomthazZeF3kBE5RyCHgntL34YtSq2uPB0K8SaJPdY48Fwrlc2DYXY1BdBFQF0uClcePS9dHcytOtW3QV+FclvPZa83qnN1/Y2diWX6becwXNjFZ4BNhHXTuR7lh/kjthHkQK565Xt1eSlbzuFFhIZi55L7ZyklF8dbvEFGssBbj99loQ3S5rwFLOdolupLYSPr0sh16/fSZsqyf/y1+iO+li2znYPjO2fW5aQ/87ay2ecW4CKZHZDrr1CMyhnnr8ZPCbwpWY6tUQWrCCtL3Sn3tqhocK/AlFhZFSj9I8P7eMF6VYpZZ8nSm1DD530KE1GvrM2PaysjgApHWx7Wxsnxnb6vE8f678rEIFsOpUopZg5ACnV9ckWit7fjDlqIopcI0QGwVIRxC6cSyg3rUlX4B8rpTDY3WLS+7iO+vYaZaOhREHPeS1JZpbbbXwHOVOJ0ZzC4dan69U/IJ81Yw8N6F79/ifTSERFo+Dd4DsOCPcEGjmmHCB29mzqAp0ZMQ/xbzwRzPnHQOKbuCYV7LYYgFQbpFIkHKFSXliSy0VJ54I2/JHiKkcNEqJp/BQzuXLbZcLT4db5BmhynH++SN3yJMXnZ6CKtXlJ3lBgMLTr6JodkntiQkUBkzJ86EWXDCUKwzHCmRJ0mz3mHP2mmTHZttJHzbs/kigVx2CkwrhPApUtQWD572CBE5BT5ppgQVCUi8LSNlYwKMCwTuX1FXnmh5+CLMFgbaqhkAKdZy5jJNdj+h2wZnAOQcTsqH13JBw6qnR0SrMoosGovIlM8rJ53uKomnN6hwq0C6D4gUvtFAwJhhDxf77t+w8cNDA9/IZg21/9+qYvseHRYC9o0AQYgEDToSVICF3QJK7HGGbDTIlLKnDFnGK89zooa66Kv4nBGb4xIasLF+xoATaF188oEtendBPG8tSsR6WTqDlEIC9S1LoZyMezQkxSshQC6JOZAHJ6pM8PKe88nzpS0Ea9HNr+vdvXvKQ5ovn9NPdjsa2uNKWq91koMHzmuXXDbhVdy0FfPl9ahZR6uercdsHHoiK1j2qXPIXlFUgpegAkRRc6pNz1TLt9Prrsxi3BRd9IxSCKeDmuK1S6cZ5hdXXDZA6ZUp0vyQx3W/K1w0nnxwFU3jGM/O4rXtJytcD7rBDPJpiQyqrw3rILsdJ4APidTlQBXkK7xagJI+R4dK9UqSzNCBWZnmBNVShVJf8THm9CxR6IoliT/YgayhnJ8zPCQNwiW9LTA1APCDCqJ/41C3yUrb1ygJxCi8vJO7o8fUWyulSZ2bbHLedXrpSoQk75+Jx02b+ugGxYjRbft1AQLXnuC3XdfDg4Eq4OuaYwA+Igjc96jlXjaSc18bJtAsy0Acu1t8PHVpbBTLHbeEZzq+8MrCNUPK/gXzzF2WD8ZAh07/whQh0BJR6UNEVw3G1+rqBEZ13Xvtx24teG7PpKcGhSqjRERaCJgkt3/hG4Iq5wQNWBWBI9hOWlAosybNlecGkRwZOAlAHTqAO8zDGNIjNM08b/TRId4LcqQXdfF/YszvmCXm4ZUE8A+VRAMLuolN2cK5dkrjJqEkELUXfAP/LLdf8tcM737htJ2HbLcq3ZGyDm5wjCVOnhr8pJtL2yqm0AihW6hnpr0YSWBEviTzlrlLCpHNVSlhXqvvvD5DdfnttZOChh8JzUfinngqrli5TO4DC8fRouSikbfTokB83LtwKPjuHyOM4wvFPfxr5Kg+4s0yWxsg9O2w5pnzXSEInYdvqW7J33w14cMDZUcor3jnnhHIOO1y9+mpAsfqWjOmRf+KJAJ5Mr7kmMMPALynfkh1+eNyrEnn30r/85fhz24k6efHF2jCC2nBjjiTkHyGeNDcn5MWpKhlx50iCWwRq5AUZjY0R8t5114yRBF2ChvjlLyPT3XbrlCMJ+jcxEQ6yIxEg8ACCAizWs2eMszjRKe1fvpHl7uFWnVgGvzpAlY4EEaWmhRWSejY3ks+/IzCvlgNlfZ3a57VR6BaEAmSQihzRB8oDFOG53OnUb2stJKiFMtJXvHwX3bkb6ZcjSUiSu1z0nwrmdrZnZ0X1Wxfbzmrr2Gz77NSGvtcGeqGO2WOE/K9J/AvV6AkkdMmIFYrmnTckQRQS1LAU4IE3twBb+mUgilVBFEigAgFpU5rtmBTm7VqEfj4g/XKRIqKHZ6EMWMqdfvJMgBNHjxspVxKY4YeKhzJUVx6QyEh/gQXiljSijTdu+f52s/iXrIttZ7V1NLblwGJYcYS4AxQ0xpe+FFgRwzoXw8KQWGCuueKE2S9RPtEBOFf95DI4AXfkC0DS8zOe666rfeKlpSUSSEl50eYkd/rnmy/Qj+DI+ymRfgXIb8PsWB5/QW3qJ5afqLmkqMqg5LCSJ46yAP2XX27zmLxpzoVWh2kODq+ZZwEo2n6L8qWy8/vuC4Pk6SgGTx+a2So63nPP8HqA2NX8UxHUxo8PtsWPf/hD2NuIEWHJ9CNQCCCPrz0XjP6uTHnESj34pEnh448dGydMAnEjYrUEtdB54olxdNduu0W+HCXdgGKoT56UEEFP5qiGySsG+c78BtjE9xu6DQ48gIE6zM8a/bRDHe8SnlVmpudwjRZUe/CDp1SmdD/hzdHulgS2o8aCN0eqnKQSRzjn2LKa1C9F7nKBGWKZHWBoJgKEkRfCclKZgJKALgDn12jk3ajMEgn07t2yxGqDBtS+7Aq2vWbamOF/jVGjCRMCHphaOKhTzzcU+de6XiEgXAE2q8k3OK++OopBXj8tI4jKvwGd5996iM+999wTpA9Lyk+/E7TOEDyjjADVjfwJNXbmmQFaHYZjnpB39eijwzbFEAzHLXwpxeOl8c/1PdwRXhfn47HHag4ZeZluu23zWqd2vpEEYNIF6W9F4jkrDWRoqull6ltY0d3l+7NOeHmaU5eoPTQSjzXn3dAT6tw4mPm+rWpV4yoCE0lJahDs50iCLHSA+XakjlrfDjrVrDTVSAIKg4ZRo4K4cySBV/v440FSOnntmu/banjtihaxZzWSMPOsNCSze4AbpIziE/QSlTYNjNkkmzPL5H0PK5F8flXJLCGY2eTRc7EoBqavUj+p3y3sh3z63dnBEF5++VBLQ6rKL5Xlwr1ium50yS4v1ZgFm3vuwLG7soQuZZHoz49KPZSGmIOtY7PtC9Mbdh4XwFDP9bPSgNaHH4aF66SrkQRoQZ35vu2bb7YZSUBD+SUrJxEU/UTEU6ZEJecsM1xXFoE0cyQBzaE24Q6vUIpGEfGAbjWSoK3POy96yv/5n9pIAnCSB+NHH41byHN7JbIgwQqH1FPkfEzPP99+VppzW8Ys+eMaVIAWnDwvvAGJFHiALqByXLx8MczT5FYzH8Xwk6Tz7EXAVWL+jcaR91Mi9AISMapUUZoAC/1q+fLYkfvv0Zzkl81O9ArKoH7sQgE9hyzkzhYUKctGzFFp3aUY0olBsnLKZeGFmxfdr/P9S9ZJ2JZr8JvyPZjbBYkwd1yZD5cXoKfV53t8yJOjzpkBAGL2/IDCLVUA3q4nUjO8Y4Ebd/iZZwKIJ5wQMuuWWW88Jncg/w3neFKobKJFuTNyZoYOBFliOvZw1FHx1BTKxVUxrKueixh5aoH1tNNi6EaKGkYQyklYYbCAmnRLZ/ZtOwnbXvXBmDMfCwv1CAoPfqI08Dv88PiEAR4ghGZuI/vVc8O55xU2pTz4McMhQwJCkE9GLvl1Q/49gyJVhRupUmyVgEClbFDeqec4sxrG7mFvuin6fhXrYcVeiuGqu0Sf7FTdSqcfzyob/bxsde4BVZd6Uz+EPbLW32yz5tV+0eXb/puybTVu+8gjtTfVwfH88+MlWVvOuKj7ZSpizGefjd47B6rUT47bTpwYsRL9ynxWed8W24LvpWXGxZnft2WTWJgxw3T+gfPSSwH6HLfl+bJqJjd5cuRySZlLDEGjadUiXYqnQ/EejdFW79vSrzAoQOUAcWcet+0kbDvzuK0i8QCwak50lx+qITgPvvHGtRkU7703zJa8Z4SoESOi5Hr6YcPavG8rU4/Z7n3bb3+7Nm7LxtVSznytp3/11fADVDX9njTft7Ux4ep9W6aHbc8oEzYdfXQ0DWvKcVskY0PQ99/fWcdt9T8qS1hhd6JGcuS+Ssnz/CcBa0BVpmgeeyVZyeSfAHwuDVPdm5cqhVxCV+v150l1zlWkhEyfPoFLwAKytdeeIek885KSLwxWt/uJrP9W+5+htgEr2LIEeAIyvXd+1g06cMCW2JWQBzpZGkbGtuzQJZSnR9FRs0PoQfdblwlQ2BK2hcgLymziagbKWQ4r1W8haDbJa9AzkZS7E3zNSrHkhAnxaNCpHiiUC+OkXG+hq8iP1qQwY10Ltn3yyVCupVgC/Qqjt8MaOL0zs+3THzQsdfoM4M0SaYnMPEd2WEC/7lzooPYSRZV8wknT9OtX++nYt2/tltzXWSdAki88VZm2yw7BXXttDZPpG6ZNpQBI2/Mux0p/KlljjZa+Owz6ac20g22v+8eYG96NIr38cqAC/LS+LDCsDgd48u3g/GOKQwrkOAhDp/xWW8Uftjp7+Id8rgBk6nXcnt/i64Q4rUpIlW6AIWBbpgG3PA869VuqTlf03HPx+LwQPgGCRrhiL3cJ1DgNxPIFZ2xLuW5PwbAta9INcDU8IGHEfeedcN68TmPnY1v2z8h1d3Y0IbxVF3pj5zwFRAMuql61HntshMm8M03rrpRXrbo+l/w84IC4xAHURWtmtJL9bcrbQdw5hSnvUr65BQQpryPVN+IRiTLSrrJWJBDJmRzkpSTuYirkaTiszIOOLqFBsaXo6sm3exFVq5BP11J2wKoMx5SVdcREnEpl26puJR6+D1C6lH8xy1EufBnmuvvu8QgCAr6/o9rTA8EZVfQjX2XDtuIs1OmpBXpy91AeTTpVZ5QFhDwONDuXCzHKAZEBnFVeelcSaj0+EOsYFF5z5LQJNHgKwp4dWc/B1rHZ9rkPGzb/Y0CC188h1YiqNCeftQOhWnUJJPKvJK4ZkLikelNexUrPpnEjtKtYSoATy9CgdRCQFHVOfuDAuKQR6QQPEIJ/KJLYVKaAgVtN5kb6Uw+1dKIzkjJqLF+ySUHBMiUzpkzPrzOmH3LOPrvlZ78edGBtduZg2xFvjznipuh06SQJTm7fo3X5KHhAhYonU3npBqD3tPJhJ/mEnyIxDZUAyWR+Vpahys+IgBbMVIsbqVLgNAG9uyxUlGfkQyinZ4dMvogHB3uwbCwzpbhru+3ioRR+r/L/sJ/ZCsxf8JfU4QG3L3OxYwn37rVX86bnNv/+3bZN+qm3jsa2HDrtpD2SUnV0LEF1Q0+PHvF/DmISbuirVTRS0/NrNg9JnkxO+YOAtN+KKwYCnC+7bERMzGCjjYKgta4OWQu56l5ExnfTzAjUiTYTy5DX0jw18Q4006bZ7BzbnKFRTC2g06jXXFN7q4Y8D1qfnLYEEGAnRXTG+9DD128S9dV2lNpuVzZQk4uTSibfZXGSKZlIRg3sWlYS8hTqrZJhDBlJVTqBlT+e54TtqcSRBjlWBagvGLFk7ZTPz5CcUyWLKrsUFk5yhOdg69hs+/xHDVvdGZzVu3dtBkXBFj9RJ8TRg0YodQnMsAl3kv+FdFxdcMGIVFSySyoW3lCeFp9rrsDzzmWtkJGtMy4OHx5Bxqabhjw0Ik0hy+23Bzg1MfrILhM+hSwsIqeDoT+RyWqQDqyCh6x5fBqRXagBSmQqI8DOdwy06d13t+xwWJuRhN++MmbtY9u0ez2Mmaeefpvy5UJe8kSZUi8vU+VXD1uX99iU2Z6XSKbDVMnTAMkJNjs2SCQTUGBKkGkaSybWlwe5e+Q8J6N49Ks05lnJl0vNfU9rvuz9tk36qbeOxrZcfbip/7oBDvhZCG7RRQM3qo8DNfMcYFAizhWe5Ny4I0ZEeA5e+5c3YauvGwQ7mpkrl27sOWWlnK3rxm0x4/LLR4BMnj2ImvWutOXXDegYptF99XWDkoh0GM/558cA6GabhW2k8hy3VQCIJ1m/bVpWfIB1Bqk8o0bFI+iuuRjaXi5OdDyKd8MNMY7mkRWAMINkBjffHFmQp4G8x9Q96IrIwxMZ2fGMVKCMuMkKoKtnXczP87rrK18JneS5OeqHQ8GMPbvyH18WONFhcFfZJIWycPLAA5GIIBZYIB52ww1jwCuHttN5UYyuf8ly3PaNNwJRAhHegPaFBGhU/y4JQc4tc4DBhq5da3bvXpsDDAI5B9pLp6WZ/uM/gl6rrxtWbp0DbIklgmXOal0pB9uSz3FbaOEwgutdd8W4ao7z8gZefz2QoEg5blt93QB4WlBT3nhjm3Hb6uuGmcdtr5k2ZuQrcctTTwXxgcSDDwZC9C6ggijzz1jFlouQjp0CD1PF5grMhx03rvYXK1NiPrxLJkaeyfipnGC5zjrR5dDAlqGXZqXFmwQYlCrad9+wiBzEUBXf/W6YKnJgpH/5Sxhy9l7ctUFl+hFZ6ALz23f1v/baAfJ82Xb11aePH9/8/YbP81+yiy666Iorrhg3bhw4Xn755TfffPPrr78+ZcqUq6++etSoUX9rOxD5mbGt6tC3n19mMhaYLLVUsK2g1fkii4SXyoHicqIbnSFi1WMjpoz9gQm8UAm/VYVCp8hFRXMKck0nDZnDnVoIGbGBnDCBj6DlHnkkkHdWma352WdDHizuuSd4irbBZeUbZISpOarp2yqbkqyySgx4QQ/YcRNOLR+na2MEKgUiWUg737b+64Yzyrpk9GdgXn3dkP+SzbwuGV6DG4CzX3hh7V+y6usG1aKTqL5uYAYeZPz4UI5GL7ooKlP6wgvHnzPk86VI1ptfNzADdYK7GY/6wb+6ovyX7KWXws9l1fPNFzbZr1+wg9qzXdw1K03xbbcdGw2HbtAfS95kk2ANnhckQMtPyqwr2kLo3bdvNJNKdpVvm/IuQRfwICkV+4UvBJ4xiKYXzGkmBIF5OchahHz2gtiWPPzkh44oDFzRNKiwCC2CPelPZIIHfkw/Q9a0AZtL+lrMBXsJbExKBioeeijmSfi4f8lEbx7hlVdq/5Ll1w3KU/1LhsuqrxueeCKc3Obm6X/9azwmeY9GIAc3qq8b+pdZbASLfPMciIP2nJXGoxE4sqyifdhMs9JUXzfoz3QbakYNqHCFyUFkBIVAtipfN5vvCTAAAIAASURBVGBY9q6ippf1Yj73WWm22GKL+++/v6Wl5fTTT7/00kuHDx9+zz33XHnlleedd94FF1zwxz/+8aOPPqqEPzO21UdpLZViRx+ApdPrXdab6dkzXMjFFw92Awg9vF5uubJGjvOU55ZKIeMnEIO7ZuCLUZtvtq66ahB67/JZhJbIjyZ2LtO/ckN4fJTMP39NntPKJSFM29Jllg0NowOALU0uBeY4KZzufOeUPD/FkankS4WycynnnK3futh2VlvHZtunP2jo8Yto93nnDfxwY/MdavjJzxPUGGiBBKhABTpIkMw9d00+v0uUAjAanf+1UlnnhpgT90rXeS9Zvt8h77hhmWXUvkxZUAf+83MG8qutFieMBf6d50umWEaToTkWlK63/uA73wkMUO5efEFzlod/s+qqLb3XaLNSThfbfvw2+2y7/fbbn3nmmffee+/mm2/+7rvv3n333TfddNPAgQOfeOKJ2267bcSIEdOmTSP2yiuvIOKddtrp1Vdfba9iNjZ9foYwiFLgoE41Q87gCYuq6fgy+7Ij1tDzq0rVx1FdbLFoA+DTQjhI0C0MoQGNgu+wMu8yHhFVpX5wwWvwypW++upYeAba3IKk4JIGIMivMBGQVqSKJfA7uMZuAQLULBElaXUMm/JKmP9Q4TU5wqtm1rTt7HnlullpNLZHyIlgKCTMVDYoi/Hlp5/KAG05jwxTAcSUX7Msw56eNaCzNI/PwChRXerHib5hjTXidhWrAh1XKLPVYAS3yAW40YEc5aWekQLgIl+XlIQ1Jttm2Rihq9iBsOyUjU2yAWo75qw0UP2rX/3ql/zEOd5iDrAd74+wCcxUVM7jgcuk5H+2AwcGQkRjzqEIQuAKVNDTwQeHgwntgHdamak9/2tyVaJb8rUq7Qi0TrAnuO6+ezQi6OoLEa5+lzAMUKKt3ctShg4N2OBoRcpxJEClB9UqZ66vSjkAu0uRKMFTcAXGcLXnni0/OmjQ4bUF7oJt62elARtqAQOPAxIMUIj0IRMsmZjikZEuO6BCdnBFHkpB1Indg0iEUgjH2kxDeeCN3elvPAgk00+no8enmTBDJg+B1Ho65q88K64YoKXE40CjE7ZMjE63EM7vIDwyv41+l6BXH7Pzzp/zrDSTJk0aP378+eefv+aaa37wwQdjx47FtocffvhTTz3FscWwU6dOnR4rbE17/fXXTzzxxM/Gt1XvWBtDYTSBz3XXBRXmuO2qbddu4EzZ8rXWkWVWGn0a2uUG2nKFc0C/pLxvm+O2k1pXOM/3bfMLAh3vtWVlZvyV7+fiERv3UD+OrapxW1ZR/76tBv5RWbuhet+WPDxi83HjQoMu1+2zXLsh2Q00XeKzQKpMgYa3eP75oQ0IsC2w5l9YEAwZ25XJ7dcoM4irDUh1TokdZO+6K7C1ZVlGUPGgE6H3L5NJq40EtwLLiDyb/PKXA5HqDSj3LJNJdusWAqqXCaED5MueaaDTCaQ65nRiPCm3UIUI4Lti23vnaKnUfz7bvv32208++eS56ePM2RbjtjOv3YBzRRUC83zfFtJwh9aBlvfei8BfGAEM0tGlWMcmhuAinDrT2g3tVjjfrHXtBlu+nzu0zEqDwaeXWflp/l6Z9RUAzmv7vu0vylQGgJpvIub7trm5Ws17MMtZaS58bcwmJwfGkklBBX4QlnBHweABMCABID04uoQ0OOTxYDf0jQdVjhvz9U1QVEhP4fFXKpM3ejT2pdenP52DW2+NXPLvbvkm2/L9WQTyBT8eEtvkDHkKBJL/5ap5nQrHYr3y4qarO5QpU5QHyN3CW9IXym6xmDMs2Pbz+pYM4l944YWHH34Y2x599NEjR45sbm7mCDgRdl188cWjR4/+XxlJ6CRsW/91Az0PPhj0uvfecen6MreTrf7rBl09DVSxmWokgVm6C8rJw6vIi1193Kw0ogHykyfXRhLazSY+vcxKwyRUEc7I95AQx/SyDLA9y2PjWCmGB59e3mOTe0PXbOKxdRa2neVIAubCiW+/HZ26jb3Uf90AjeyOcZ11VvyVAvP7l1lp7Ap8St3XDUpSjSRUs9JA+0EHhVpod7XdSAKimF5mpUHByAFdVCMJqEDFDh8ebjveR1Dom/3yS2xYmy8/ceLnP5LwzjvvnHbaaYKsP/3pT1zXU0899cILL3z11VenTJly+umnQyeM1st/ZmyLhuBGY+iI1D5sMXudJ1dLHwWmKloY5ajSVZwa1+lpwhXKPFtYAPNK17/BIv7CDtAmERHnSMIZZT7GaiQBlQPHG2/EpVmOJORfasqz8MLB0dIVz4mfvDxZU64nqEYStHSOqLIEt+s5tDRSbveYXWw709ax2XbShw17PhZ1giy4bFgAILXXXuV12mOPjSb77/8Oosw3yjUuVCMgbpoeHZbwMuiqSc1KXqKIXqJbnOdIgvNf/nLGSILoGHTFQ7McSRCwgzR/kL/JmWA1kE8G8tErbwMbAjDocjmV1o1bbx0Aqx9J2O7gNiMJXWz78dtssu2n3T4ztmW6arbaNWd1jjI0G0ZzDrUwhyjrBdrtmoE2ksTQJa8ZZy2ySLA2av7a1+In31YkksOvuEZsgrAE1FpOY/BMc3al75dFDex9yxoHEMmzoFAbO8LQggvW5IU8jjn5BXebPO7m6Tz5ZJvHlO5x2AM7xNo6hk3LKhI2+OCk4D5GwsAgDFzy/SEnQAOp55wTThOWd1fSIl8GmoVaCuOYqKWH5nwbJt+r05MxM/nmh7ye2iPwsGQH9CpHr8Bg2KRbZCSdsD3L44jEc5VJ5/m6mBQGYFfhY8e2ecxPuXVstn12akOf8vbhLPdsa1A8snzNhV9mCd0qMamWvHO0mDPNi9I0rvO55w7oIlCtDL1qXhe7dFngAPxAMf8cFqXlfwAifSdaHDv36xfIB2nA0I5y1LI5sgT2aDfX40DoVPXt27JMnzb/kl35/phfPRD68e7W5T1ufbY4HeTwI/2AoavgISk/tcDpeWlTyZ5Fh3Fe+aIs/+ZVLbp/5Ye3/mW9UaXyvEqLptXV+usH2+qKqN2kTEfpeZmzFBYBuvkCfk6WhGHpxPV6FxXIlPKjD478yitH96ZmpHCA8hNh+hGxSyL375z4eY7bfqrtM2NbTp+qzwpl8wCqH8YaRxwR9f6b8sksysjvmngQieB8tYVvCDpblrXw0JAOUOKB5ftITS5Fs0GesM7PAw6IIAtWMJf2vuCCQNi99wY9AfSoUZFOYXNz+J6QpDCAMn58FADRP/NMvK+qFZ3rHhdaKEqirrVoTpubH/zkaIaStJsnAUpwlh0yOCb57q2e2caT9ROCFemmm8JbVAZIzU8qgAMhErjkkvipc6Yqtb35ZiihkCT301FiSiJBO51+uiWzpt+NfmZ2UrIk9EtXV1wP6ak/y+PoahbDeeYuJSNZWXNz5mDr2GzLt13v+mh3HKdmAA/XQBGM4Z1ttglXVIoelC/p6JLuirwGAlr9Vr5hTdJPmGEC5DUKZ9OJAIghuFf40r17gE2TceV69owUJKLbS6JxC5+RP6H/ph9WERmg8osBEmg1Lscl31HjgWIuQQ8Xm3+q5MqvQqBOFuPHt+xyRJs3wC5/d8zgO0IAAOAtcaj1//rX6dOmBcElZqCCD8uRTHCST2iRV0g3Vrh1L5DffXckklGw226r6SfgKtsh5ueFZQ1AGugh7FIajkvSecQ0X1ZWPKONkkR1KoFe6b8tk9ektsQ/DRKvuKL50LuaR8YfUXOydTS23aCscH5Y+f41Z6VBvhogx22FRQCkC9UJqyM9JIFzyqw0x5QPFs+uW7sBsHRrfD1On64MWbcbt81ZabQcw1Dp9eO2uEmO0t2IbaF5j7Igh6A7v+2BALfzDeWFbTmzWg7L33xz3MXDhf78umGWs9LoTvnXMurVK7xp3awUvgnEcyj4FzwX/oJEl3QPTjLRLayoekPOUaKn0Hm41yUOtROugW7cJU5N6t+0LCohkYbUI33psnSVOue2Z3ncqADkqSIgXy4DnW50QkCO+pJZyjvPAevZ3To22878dQPexGI5bpsvA3AGb2ydlQY2dilrMegmUSR+VKu5wjmkIUSEC+E6Ts6ggKx+3FY8tH/rrDQf93VDvrpOHva4CEDIiZHpoEGRC+7OcVsGwlgE3ZqVHf0/Z6U576Ux39g73GTyy5aZy2FsuTLXIiTAHtg4kbsjVQlF6GVHlbyUfBHTT08HgcS+UeZCVKSf/zx82JRXJxQmFFNzmonb/XSXfN2YOsmrjcSknyuVBQEk5u1udARaSNZFpbyUMlloWZesNmAy21sX2xa21WmvuWYcBVA6UhEEQFMr3tHDAz3+Ve/gSwlrQZ3Sdy6zd2tIHivL4RoAN+8bGvihblcYlMqW+LZoq/yrEHdRhXNdJZ8von/cHGCbt67d8HRZc5cx5HjCXuW1R953/tHBN7HtV9Zu4OCkLymv6n3bbcocYCieQbIl5dGT71tWpXz55dq4rf32snbD/mXN3ZdeirjBk4Jm9b4tutQPKTBVBIScSqhW0YdI87/K2g3vvReZkhcB6IfSwRcu/LxrDrD/Nbbl26pwjqpuVbVrX2DT7vCJlcRAWkEWw8oU7+JxToM+lTeNgmGVkwGoXGmATFjmxIP9+sUtXGkWhLb4HOzuG2Xab2Yoi9tui3HberZtNwfYhAmBMWBjXDCvFnghuXYD3/yWW6I2oM62YuvaDXyR/BeOqQ4rc4CxUwVgyB4z5wz785+DK3Pctt3aDYxX/NdY5gBDplvVvW9LnhgM52ud6FUfM7x13FbQycpy3JbvRT7HbZWwk84B1rdvOIM4lz1vtFHYvHo56KBoGI2ElXTyjhJPPjl6VwI7lE+/4War8qk+IhYZrbZagBvIEIq2xCC77RbhBkZT9eDodj9POinGepC45sTXEOanGzW8kxz3dIt0bGJ3C+PR0trJOZZxIzo7pkxNO7RMZOPop0t2OCCPrN+rrVpa27rYdlZbx2bb56Y1rDY8eLNHj0CXuoVPrqUKxHT5zqkU6MKAPAmXVl895NENV05jde8eiX361CbrQjT8L/hRt1gDllCYphQCAzwYSNQ06l8KSIOxFOkwOays9S0RHmCV2vxKmJKEJfA0lgXY7703rABPAe255wYJ4nTHs8vsOb/7XcugYYMOejOfsYttP3nraGwLZ/PME38C5FFvv8gi8Sq+nwssEGGCk/nnD8tfrCxJMt98kUJAOCDF+VxzxTHXBSGQN+a9AgcdeL7tnMG4PWON/A5NvAMKy5Z55vNv3zxPAZIZvNuJab88z/nrKm1+AjGBvMXt/JF8L63auth2VlvHZtun3muY66iA4he+EHhTXZoVDBKx0u3AmcvhuErGT4mufrUsQODGxDMvlXvrnAAxJpBAok2E56TCYYJz2fKxGWDn32g4GrrgM7GdxwrDbtFSCC5janAVmLtkX7lMtIS/qluWW65llU0H7Vcbju9i20/eOhrb5khCzmeollVlhvw5ktDcHC30+OMRCnFy60cS9mg7m7ggCILJ31vW3K2fTRzWuQY5VH9OmZUmx2H/8z+jk8eMYAS4ojxNJS8NzEfQTjBh01p2nIi+P2FWGkgFOw2MCrnY7UYSllgirItdsT2Gx66csLovfjHSFywrs7G0bt3CUQKXxctqUWnDZKQTyNule1gZuaR3oURiz57hKPmZAnYWxRho69271g+5JC83qtjskOy5npXyZA9HXjWmTjfqz1zKjFKehopKFl444Fv3Fvan3To229a/byuE16tdc03tfdsTyvS1CEJHqPfKWWmwT665++abtXFYt2CywW3X3CWvW83ZxA8t69SBVs5iw0A0pausA8FpnfHjQ8k++9TGbS9tnZUGUG++ObpJVgCWNOfX6opKAwzoJqtZaQBbvw7kgLfPPu3Hbc9+YcwCWwa64LPCsD37CdiwVxAisEJZBRVgUlgiYf1KIg3AlFxVZDox56LbJcoaPAkzGS3eus4brCb4u5UlBBFldl2JZ1DMS46yo8S5RN2Vh1JmDhxhKU7I51Xn3bo1dx/QWcdt/+3ZVk+rAP36hbAyyOv3vw9XVNnyvQL1wHg4R8ojBVwo0atzBIDG0eNvtFHczq1mgQDKoRaQolTyuJXJcTrSg+hfJku0eyL+KePnC/AIWK/wk1mus048ndjTzy23rH0hyq1QnrPLxFFiVeHCLmVmPEpkmvIMcpcyVx7hHXaIjD78sM2Tfpqti207ANuOfGfMcX8MiGI6R74khAA/A4SiHNwAIarAW42oEFYJtHriTcrn+PxZlMe/5iCTB2OeZq4LqagqSuXke2NCK6YqI/pVaVYFya23DgGPCd4cWHGYG+lXh4ANw0pCPj9sy/fq+PvuZRennx5Px5FXeDfS/42Y5bX5B+d31ncS/u3Zdost4kiJW+pXOAf9HOGtRhI8uPLoqGGLIT38cBiVo+fFzs/PaqUcGuAMtjYvK980ta6UY/fzh2WyPnid5aw09SMJxx0XUPYU1UjCSWVWmvy6IUcSPior5egGCLuqMrvY9t+bbXMkwcYebdq9+vaMybQbSZgwoTYy8OWyYrlnHDo03hXj0uZKOS/E6uJBwb8uK6LXjyS4V+3lSjmvvVableasslKO8hPw4MqvWu68M26sH0nYtXzdMKCslFONJCAodpEr5ay+ehR+7NgYcHB+ySXNB98pqzncOhrb8s74cUKAXNLVibYXF6PIeVsXdkYBolohQK5EK9GRmH2hWJO5tib5YmVx6SXKMp9LlS8OFi8rQjv2KMuCLlpWk3bVcamyIjo9TtAc5bQtXRaUXrAsUtu7rFjuJ80uZVhtz+LRSYlEgY/+E2UvWlauXrCsca2Tf6GG19rWxbaz2jo22058v+ErQwMkvctS4bCkGjGFmEAi5w6KJDqBjYx585MZlZz4WbwsUusEbQlBcg3alAeMSt6JW4i5KoJepCycnPCTnsai9f3sWdaaJb9YWbHcVUDCO2hou+1Cf2atYPJNhItXdAOgu9xycdeCC7Ys/p1BA+pWOO9i24/fOhrb9u0b9aWduK4aHvnqgfW9wIcyND+GypEgoetKK4Wdb1mWJhM7cIedgJGa1X7Qo2nt+W9AJY9otDqHV6C9xhqRIv791rfCdaVqvvkCBHKRIwckZ1ESp9tpVjZ3uRev4Vm+BjMgzxcQwgDHIYdE7jxcJVckymFFFKO967eNN47An9iQMoM9T5zjAxyKykFyadNN4xlBGclKWXnlKJuMFFKRHDG4QMztAJTBWr6/gYLJey4xFI+bwK5ltYUcSSC5X1mPRIx26qnxmCoN2njfnuXggwPHIgblIfyjMoWNx+epnXJK1B5VfCJKtI4nJa+oUrYpqzmoMTd2WrZ9blrDZrdFuwAqhKg0u9rT1mIpsElIf7V8ssgsYSMnRZSoYgEGfkAdIZKHjRyrAeP8hweFITg1DBU69ZSHGZhU85oeY2p0UJdy/PG1j84VwDn3VrPShpS1OzgxEAVDuPnyw9fK/I305NfqNGd2227b0m/3QYfU/uMNth3+1pgDr4p2ZzKO7gUn8ECaG20UePNz67Icg84+/5vNLz70ylJkoUiyO+yw8DnIO3pY8gxhzbJcWw6aMXwlZKoyOvnkqKL8sknZPKBn8bCg+N3vhrAb6VdLgA3DSkKeHsIsAs4ZIyXsQpyh8j1avmNK/6oxzVhzv2HNI+YUeB2NbduNJHAS//a32kiCfgwB5aw09e/bqvEdytyMOSuN/u2gsnYDKOfmRGNUazcMGxazu+sntcR+Zc1d3TINL78cQPmP/4i5BaqvG+Byevnu244xzyuv08qXklXLmrtKwtO8++5oaf0+b9Rd+I4YS3i+fN2w1Uyz0mBGLZ27c52zxwEguJFSHaV4HDBieCpEopTf/z5S6sXAF9zb3Zs7l8HDpnymw9zQspZPptDMTchQt15DZs2hSLGUzDK0k8yM8oRD0fUvWTWScEOZZab+fVuXsG39+7Z4UGI1K410pDlxYnCioMclbFvNSrNLeT9XnFH/dUP9SAINyIsFTZkS7jNCmR4TndW+bhCUcA6UZ9ll410UJ1hPMcgjI4mgBdIS9bW/+ZhZaS55Y8yO58xAxc9/Huh1rAdGdUkApEJ+VVZLSzSeVxYYrcfPMWUNququejhxQVK+UvibsjBgJQn2qi5vzxvzXnj2LMeXlUmPLHbULovq2HrSvM0lzcM/qFpz9rYutv2XZFsdMhPCTfxK8uPHB5KUASb0w66ee26AnsmJ4nnH2c9DBnipGQ8LuHxtGjIwJC8qzJ6fjXF/OJ78LO6tAI3DQpi98YDce0T5I6KpKRwNDsLkyVGlyq+ieFVMmp6mshjqrmVZHZZs9/h0qiJXOVaOPAia+5dl0Dyy8nNyO61v20nYduQ7YzY8MYaPPS3waHS3CLx+9rOw1p49wwZzqge3exyOvOfNUTKI4gKDdEZpXFr05EFS3lPo3UEx5fPvL/X2y1+GJNwyjfwkcpuyKKpq4cmqkFxuCjkg2cPKYmtHFHeBNdHJ+s46Ky5Re3FZedfzSF9mmSiqpxANDBvWvPKgzjdPQt++0R5qSpiMWbADs0cBe5VFarWljggD2tEKGsrXyGFFjac8Gt1ww+AOrXXOObELgevlQVYzICmUobWkIERtrAemat55g6rkgsI0bX66Lnc7E9L2a5aloZ0vuWRgTpuR17rKAG26U0SG2jyIYgiOvlemi203kuBq/qsGpmfUrd1w660h6dlZ1De/WRu3/frXw32oxm2F8IrKIN1SjduSz44h37fVP838vu1LL0Wxr7469Kgl1jukLPpLiaLia4HeNdeE8PQylSXi0BbTpoVV2z/4IHTmt7k/qpsDLN+3ffLJzj5u+9y0hh/dE7YNqNClE7UDjxT1DGyaFR6wLZSCtPN8ZdsJs1+/zDIuvXv3kFfDGiJn8FpvveAIagFPm/L4Flgg5PEUclHnsEo569Adiru1GhjkOyTnlO90IBDIQR0ja6wVy5KUCrbIIpEFecVwaUBZv2vRRSOFjfAkdt21ZeufDjqsNvFrbdx270vDHYFADA78UKr8OgxYRan5Uo39+LLAXc7p9dZbtfdtt902BnZdzXFbXovEkSNjj0psXbuBfPW+bc4BBrdMwyNzqqpxWzVz3XVRbP5TNQfY1KlRElXh3MY2mYBLLA44u+YAm7HdeGPU/sVlBr+Ly5wX7NnuRNVk4uVljchKJqeZqH7mXQilStS0oKDq6zVXe6XW1evKdLp5rsFovvLK9nf5ObxMpdFOD0nlrH7mpBt53twcTFe/dbHtrLaOzbaTytoN7VCh+XIKldwvKasvV+BxKd+Nqd8Te4m6ZKLhZYHO/HlxQezMSKaK5pz2JW/XWO2Uu/HSQpT1QM17R42aUSr3wmGrJbb8pnnQ4bWlv7vY9pO3jsa2glm1wE8Up4tB8m8f9YVKWDJnUxCh2excSKGKqEeN//CHcTW//kIcwnPVrbWc8F7Fy3hEKLH88tHbk9eELvEs9P/CYezmRuGwyvKTZyoXIR6dmI68kFyH76ey0cARkLj44lEeWQv9hG88BdzEg8akyEjba8UNynQb+LHtopmRnSgM05UR+ogTOTXu8qR7lynEziwLZfM7cJ8QCbxyECDfUQNu6bIWB3kQmSJcThP0ZFCfow2eS51MmBAhmCc6rSzcArguKSReVmA7AuX1419HGBUbgnUO9aoo5+QPPTTAylkTwUmRqaO6lSLKU8P8JvXWmf8le2Zqw3LnhIOpCTYsSxYlftQ5n1TY1KdPpAiGdIHQQmylsoQH1GkpEIUuN4INgAFnRvcaSHr+dZnyuszllgsEugr/dAKbVhPMLVYWjAHdrbcOZza/FOjRIzRoJrvefeGFQ7lYUKngNr+iRIuOmhuk5QhyGtftO+/cslr/Nu8kjHh7zLqDY+BLF0vSLZhX+SFnv/0iL3iGz5+WSRH5Gd8rSzNg1V69AsbkuRcwLH7NxUEYeMpDFLdd5eTImNiRPJBz6knCLdNgnnDrKbgaxHKEkGkcX2ZQQ/3EpLA4vAzA+NpT8+4JqGcNfUhZ5kpe9FMoIFaNZ57ZvNLPO99IAvSogsbWtRuQDlZSWThOl1it3ZDjtrwzoMn3bV3VfvXjtkjw2NZZaWYetwWX+rUb3Pjoo8F3JB96KF5ORJScROUZXNZuyPdnISnXYqBk5boVzm+6KeSBgJ/4m7J2w/Fl2lkMNcs5wFgLo/IgMA1e4kRPxJCWKjPkSneyQZmai5Uyyy99KdKZk+fFm2xp2fLVJmtxo0KiP5cYCY+YxYoQyVBOQG1IWaFM2bVkmc+XfkWlDejRgXKSxMtSXJXCzEhCtmpnP4yzd5kCFZVkusITljVG+EZZc5B+FaKSO/O/ZDuPC16bNCm6PTUPMEuXFbbV8PjxgeqZx22vuCKCBgjM2WrcCG/ggSmqWWlyTu76cdt8ja9/WS5XyMKTAFpUCwnKoaW4oNXaDRLhlgb76NHBer+Y1Qrn1Rxgr746Y9x25jnAfvvKmLWOiadbaKFofUhQGPwOOXbmw4Gw43QQZZJ8JmDW00jxOIQROi9HipOENMgBD0qFW2WGt3yz09PxJCCQjSxTVln1LPkOg1sUGHe7y1Wq8kszVs+TkLWMVijTWFPi3pSnn1nRQ175K/kll2zuuX8X2/6bsq3GFtGInu6+O0oiJqKfE8ox0evKjtfwwAPBmDp5+UKS5+V+PvJIFFsopEL8ZHiEtbGn8JhcWo4Aw2YGHlx5BKFKoopc0qUrzO/L9LhuhD8n7CrnIZUvZIsJUDDr5SvtFdObhnGKSf1UHkbFYWHA11wTubP8P/4xapv8sWXO5s48T0InYdvr/jHmD29FGTwpj5i5UQgqOXSmeGefHWVLF1W8D59QLV9m4kj+ySeDQD2yGiCTHxAJ2gT1en2IVfj8+oOVKRJVDETlUCsjWXtkPKtiPWN+T+SpyednbCJFwRanVf1oBfrd+8Oy/gW7yKkmybNfCgloix12aO7z6843ksBRwggiAiSLHcAUBYhZOHRaDmtcfXXUPkcPFyMd1YdftJxWES9rV+E5IrjkkgAB4qDhttuiB1bFUIustStycYl+bAvT2jIBoQmxLep0u74XFbIWcbTYB2RlAX806E5lyiSgQdY8O1SVbOuqWBtcBFbYJ5tcaD/zt2TVSjnKA6a0MU7gg1e3azbgQ22Xtn7dgBlFas+3rnCeXzdMLyvr7F/WHQHB/26dlab6usGTYmcMm183VOO2tmXqVspRmCuvDJukJ79uQPfUUqUamaKfypPjLexflSpM16w0ddsLz3/UsPVdYfMqQYsDm8rh1qlPVaSuoBqisK3KzBmv9aYM/qmnorFACLrcCG+cMmDGJpB2773hfPEVoJ0VkNf6emsVjmqhFCwPPTRAi23hOeU90S23hKWMLotu0Ew/JZDAZzzyyEDR4YcHuaNaSuBEVATMIM1X2KK8CYvrx45t2eGwj/2WjEnmt2c5bmtXDHwqL0T/07qvG5SKzZJvLiuc59cNU6bEY774YpSk/usGdKn2ctwWNXOZGYhOPcdtbe+9F90/G3yhdYXzt96K8l/c9usG52qYfvfmYFf1dQP5HLclo5k66aw0GUTADZrjfIk+mDdUIbsFFggULlmWBNdPCgq0R86Wkt+95Cdeiy4a6e5ab724S8zirh/8IH5qD/Iog3KgBzLhsy7OuYykq3r4dgu/eJUy4bFLUEIzMffKiE48JXdqAZfAXHOF0wco+gl6SPYqUxovVJZTBiwc127cNl904/MC64EHBlvpZlBeDsXqGPA+XG+ySUCEOXXrFr0Lj4PTzYXHktuWeQl4TIRzkgS351gVGVgHX0B0VZTAl3HJLgUdy9eNvGlUzq6YZU7Zx7WnR2kZfP+yBAtVnoXL4CeO9mjkGTB20BNI57ngffKeAv9ikDnYOjbbPv1BwxINgRyE2717IBlKgUofLEU6ZMKM7lzrr1ymac+vucASVADJXTkmo24TosQkSnEkL4XC+eYLcObHimDTs2d8LeZIQDpJPAh1JGmTBSVOchxWup+El18+jAWkKZGOGZEa/YuXNXXglkG5d6WVWpZee9BeLfmMwbbNU8dc8Gzgx13ww2GHEGyO3ew6A96DnoBFIDj8C4FQjeXlm/LYmvmAHNeEDAc5P6ZAsmJWno3m4EXllzXZDfBadi6f1STmxaZwqI8ZXVYJ4EZgc1Uha1yhJ3P1qDLhKnOm3708DyhlF1wHHQl5tpzyMNy/f/Map3S+kQScqEsXManfk8tCympHlak7lc4vU1NMWnwNtfjIkUvFFdV/5qdQO5X1dWBRS/zqV1HvulO8AEz4AlPr+mjWnerheXN6YycaJuV1j4ACmpTkPC86Ye2k1aEKmMDFzh5yURz3Qj+/Q7EVFYOTh4OUF7xwQ2Qn/KnfoDnnyktqdpJ0r9go3k8nnhfc/WQ8c88dJ3oa2AVHnbky4E1hF38fEXNJlFwixOux1RIvlUfgKYhVxVarJN3iRnQ8dmwQPXnsqYqcu9pQPmZTZmrpzPEQNxLD2mQkUuvEMWmavFxk/XxZUHJ2t47NtpP/0XBYeeMbLMFY7em92LMwHx70eVDNsHVgGgLeoFEUrBrvvDMo4+ijaxE9GGMZfRgC8hPvcIq5BdRyk6EXrSAmWWhl7cU9BAz+so6WTufahXurceXOSdSbaiAF0OK8YwXYtywul4OqLsGDXDAjDSyCADDnv8c33tjywwPar92w3L5RnvnnD8IFy+wqcs5G0F2mLOTBN1q0LD6SYM6eAFoUGPMquXPkmxhjd8DDiqUrDDSqlqFltmj+LCMCTjK8BPLDy+I3nsVP9w4t8/xSdfzxIe8n64BeCt3SUJYodZLy9INxO3klOeWU5j3+0Fm/bsBuOih1CjoMWN+o7vRLwgp19847tVlp6sdtj2lduwFYeXn6cAFLzmLTbqWc6eVrhfzjvt3XDTxNwU4Out9eVjhH9AKchjIrTa/WFc5zVprHHw8zwH1gJybKN7rYzyuvzJD/uHFbUIYtqFIqknbFZh5InMJzy2psAhznvFG+gD4fLFijqPP99yOScsyTdudVCs6aOrV9ev3+wQfxdtfMAu3UVsrrz9uJVSfMcQ62js22M69wjtrAFXPhi1PLCucPPRSB7bllVppq3PbNN2tDBOKVyZODnZFCjts+3zorTaJdyvSyzJLOG9IwEUPg2XHTUDMbkTWTgX8O9bvvBqdAVEtLAPuAA2Iwaq8yKw2iqVbKub51hXMghLdBZYXzfE14luO2V08b8/sXwx155JF4YNSsA+A8olE75lISUVQ6Rowo/+nlmiDoeozNDLz6fWawtZNvB8t2t3zCpXZXW8+bR0xtvqq+PWdn62hsi32ARmvlMBPoPPpohLH6OhE9AtKi0IDv9MZiivz8Aeb4qmCqB0Ogf/hDsC1iwmjITtzB0SMPH1AO3Ho/3b6+XSf/2GNhCYA+cWLIUyUAROJQCJ3koZAwdGJP9D28LKG4557By0xCwbAtU4NURcXO+aZqykvM8Pzjxm2hPDcG2W7c1i3tVjh315w5j//iW8dm2+c/ajjx/Yi0wEyznlEmwEZbqE3ikLKGsRhowoQgYt0ttKAhrpY21bKYC84lHnts4Efvy0Hzk3/AIsRM0C4FevmJWBXS6Lm2fLfKqTzooPA2ZM1k4J+78NJLgZmttw5v4/dl8T15cYH9hHwONdjDObeGvFwgFphBmgDEKhLoPvZYy65HzmLctn42cdRfP24rAPK8TFXxlJOfW43b/gtvLLvzzUqj/4ceyNDH6rRxjYbXltov/0NwwgsAO1F8vjHKRfUTOsVZYKQPP7l8HgbuKCxndQHfXXeNCI6rywAoP/LIcDGwM/3OVVO+lOfe/APqpLIMOP05V4C8dOAczBPKysyKJDwRnbkXLnmvik1m991Dj+495XNaGT3E27WvcWpbF9vOauvYbBvfkt0bAAYznbH2giXNqs+GJZCAagQHHhkdQwgMg1y+eAO6Wr+prPOIAQ8rH1gnokAr0SsFercv8xTzKvK9F5J8Bcrzv69dyvQuOV0W+AGSRPnmcJC7/Dz00CBWOVIO5IrnhC0cWZapJ8BARG+g29TUss/gQT+tmXYX237y1tHYFg7yNWw0ZBcQOccyW5SX7PzM/1IzMSWrnXx1wk2o0jHvHXdEsFYv5qjrtm9e5iXihlS354sj9ZJ5lHuWCpjAt5JvpxYQ6+WpguZ32k4LLxgkwz1ZaqnAov0734lyrrBCDBdId8JsnC+7bFjXEkuEb86peaHtzI3/XlvHZttn/t6w4vkzwIAB7U522CGi8sTb6NHRmnkpcWKvTvKcsFsSaSAEPLzderTnJSmbF5yjPM5BlZ6oqwBZQVG+jpSTR6P1Au32ev2bbday5b6D9n8tnxHbDr9o6labTRvQ+5YBP542YN1nBmz7twG7/H3AimMHbPx87P3+Z8Bq4wd8e9yA/v9nwPefGLD92wO6XTFg+T8N2PKVAT2uS6T/a+4qnm8zh1tHY9v1148eWOAv7BKqc0XHjo3eHlfqNkHhggtiuH3y5Oj/necQbb4B1rNnDBG4JNJfeOEYQOAIULXqqtG14jLyt90WbkK6nPRjtHvuiZrmtCI4Dogcv/71GF7Q+Qv3hGA6gDPKGpHcB/41Vbxa2kRtt94a3ofI66qrwmfJV68cTy9LSQruFB6+G2Z6AwxpErPnvfxZuxOlzXNPV3/ME3BQWk+auwfMk6efDifCzyrloYfCuWgnVp1XR/uLL8YAdCXwCfIPPxyv3M4sUP10tV2n8im3js22+QYYj1L4v9128ce6PR1bIOFULrZYjICBnHYX6TtyFMlPnBiOsCiKvER4E6VlyAXP+BSc1lqrFuDfckuMFaBjl+ygzr0F3dPK/IE07Fpmx4eWpZeORPzu6rDygaxLvATlISkjXgsAgyiLU2YeCUOTni+iQj5buOee+jfAPvpo+nPPfnTVlR8lJP+ddo7tnHsyHY1t81+y+jnAxM6nl68bMAiP4OKLAw024IDdJ5+MfzzvK183fPWrAV9b/b9kM6/dIEQSNAG0W/Jfsq23jqinWrth+eXjXy/UzBcA1vy6AUy5GPl1AzRzh//2t+ApoBRS3X13/Kdn0x/YnIsicfrzZQ6wmf8lqzaBmI6EGeS+ZJnpJv/L5s8KHjndXGkhJ5s5vkxUKjGFlUGftPPOUX5HYvqVXcucSerE8669du3P5dVXD/M+oSyQI2jlJuvNF188ugRlU58iQbEkeVeZtyw2L98652eXffpEARZZJIrqEuWiRb4SgW9/O7oTPrgCeFKdwRxsHZttqxXOgaeaAwxKQQtIEJwK5xaAkE1N2nLthvy6Yf8yB5gt/yW79NKoZIjdvPVfsvfei8T8WkGjXF8mr6j+JXOJx0ADeIPrLNdu0MpQTTkgTS+TkwHw82UOMGXOf8ls2lSpIF+RZvqXDDHxIuINl1P+3rjzA41HvdL41YbGk95tXH5Y405/adxnQuPO9zfu9mDsZFY4p/G4Nxp3uK9x6dMbj36l8Ud3R+KJ7zZ+55LGH/6p8fg3Gr9/ZeNql8bPre5oHPJ243JnN+71WOMaIxsPeb5xycbGgyc37vFI40InNK5zdeOej8Tuqp8/f61x6zsa17q8cf9nGte9JvI99IXGpZoaD54UMr/6sHGXBxrXvbpxx/sa+zQ3bvCHxi3/2Pi10xtPfKfxuNcbe5zUuMaIxsNeaDzipcbt723c8PrG7v/t0j4r3HXlpe/nY8721sW2/9psy6c4qXzdYD+/zEpHP/mclcbGPXn22XgupOkSKrRd2rrmrie98MLIdJu2azcs1LrmLldFTaLg+llp9i9r7q64Yrjnl5U1d+u/bji1zErDYyLP4bq2fGinALl2w/Ty2fGbb9Y+cuPE/aqs3aBdVEtn/rqhc7BtbdzWVr/mLnhvvHFAMb9uGFFmpbFtskkgkA+evjmX2UaYUeQHXUq728es3cBFYLxMmF3nrDR2V/3MNXf3KROMHHJI/Pv3l1nNSiNf5nxEmad07bJ2gxBQtXApRH7AD8Ogy8X53Gel+eijj6ZMmfLGG29MnTp18uTJL7/88rSyvfDCC84//LDN5COfGdtyo+xbbBEvn2K0lVYKBllzzfjXlQ+FZ/Ntrb3KRB4ICP9iAeAW43frFk2+xx7hx803X9yOu6kSUgm4YA5lE8MyGsZPeghwyqSsu2683Ao3zlEVhX37Bg2BS866z8uDQg4mtSIspSLDZtyIy5gEGby27LJRAOconqdJTMmlIMpZbl1sW7d1bLZ95u8NXz8rhpIgShWBol3TsHMgWWWVABgKg1vWDpNQoTXxi5+OoJ7ozU8lc+aKfv0CWhIxgjCFcg294ILhEAh6gB+8Bw6M7FKehvxIh6MAAxLJyxfOIVkWkKmxco0GuWxTFvVgcSDN/1BUefFUZJTy22/f8u3N2sxK08W2H7/NPtu+//77w4YNO++8866//vrBgwefddZZzzzzzJ///GfnTU1NzuuFP0u21WwaQ93hO14nRNq1aM6BpNPWWpgCXOxQK+5W446QjZJIOse2GgPmqOrVq3b7sGG16WCq/xNw2bplXU/MiJtSPhcLkalE1QcBcKwwSN9PWW9WFsLp2TPkqSW/zDKBmHPOiXT+Mi8mv2PJbwc9FBab5ZYjCbuVubjsvXuHa4PC+DWKrXg0MDPPwsBc0klITGE5ApMuJEcSiHlM6eiPvy/2//73wzFXn4D7g7Ksjlr1COusE1eV/5hjaiMJ9IAgRO5Vpj09rbwARP57ZZZx8gqwyCJRGJcWWyw8dxbLq1JFct+ozF+TnzXPwdax2fbZqQ19r4s6Rxa6WLBRaerWESrgxyXxGQpIsiOG+/CaitWfpbwuljBJTeaS5pACZlof9ejvtUtOJa7JsK1bsCTlFILZLmX2jL3KXMzUahHyGI05aDu3gyX0uhfONTfD0XzwppziM1fXKgvwyA5B07PKKu3nAKtn26amGL/KyeogQabArww0/LjMas9YdPOKwUCATUYS+eCInpHmxAVZM6xAXy5f9OfpxJGcgKyl7t3DSPcqc0x7Xj9Bnbz60bsrM/AzYVSQ8oibe+EW+eY00yrKAw4tiwYxbYZM/qijwoJAV33+4hef8xxgDz744BFHHDF8+PD+/ftj0tGjR99xxx2HH374448/3tzcfENGPa3bZ8a29SMJnKzryoSzdj2V7muFsnbD9PJ1QG75dYO6Gzky4rWZv27IkQTEd/vt0THqx9R49r0XlzV3Dzww9IB79XWDjftwaVm7QferQwboHEkQdmnaHEno1brm7vQygACDkMitfuWV2vTh7OqWW2pzMlzV9s1pWNmmTOUFZ/l1by7Yl58gK2R+kQwHPcoClFII8C7xuERkOs88NTEp+cWOTptFARnYQSQ9CywQ9/YoS17yDojl6oSLlFUv7X4iaPacS2ouWlYbJOyWRcsSmY6UpDzhnExSIf1cpKxCmAtxkudAKb82moPtc2FbuP1s2Hbmrxv4/khHiFB93fDgg/F52K23xg2wxEHTY2msN8oauuQR1id83ZBfQyzbuuYut5RAzkpjG1omJLqhdc1dHbPmELLwTrRO/UjCMa1r7iZ9tvu6gQewfZmVRiw185q7F/6fMRuUKRqALR32/Fw+P3PPWeJAlHVACFwtXRamXKqsYkmMzLzzBsa43n46IuIEXh6ZAzN3ooegJD/iX6wscwnq9kwBOTfqk8iwgi9+MW5cYonQrwC5tKuCfelLoTaNK7URk9KrLBpLRuUXDc2L7Nt8cdvXND/9Nptsq8M/7LDDrr766nPPPXeTTTb5xz/+ceeddyLcnXba6Y033rjxxhuvuuqqf5QPh95+++2HHnqI8KsCgTnfOgnbblH3vm2uuUstL5J+ELm+dVYaVlQ/kuAuJ4xK2VAeM+PI0CxRl/7/s+YuY2bSZ5XZx91O/rLLQl7/z+9wy7hxtVlLBInEHHNQmzwDuPzyKCcjrB9JUG8ddlaaSZMmjRo16owcr5yzrbOwbfW+bf1s4vR7KGikx1NU79vmSAI0jh4dfi44PfFEWAfrY1wMMGelYVnsCAm6OvNIggdkzjmY4GoiXNUhnPVaZxPPNX1pO+WUUJiz0qgc9j58eO09H5SCam1ss34k4XOflYYDe9JJJ5188slHHXUUtn3xxRfTtx04cOAjjzyCahHuRx99NP0zZ1uumThClXH+871uDr9dMAWvvLBkQxFusrDgKIMCkuIIgQxJCEaXHJYjyzpx0CnMgUXIc44osafdjcDHWaMHerQBecThp9YlkECnE+Mwm+QjJYRmANJbVvIAJA5CfHoLevLbHnhiZseWpXzxVP3Wxbaz2jo22076sGGPR6N+1MwaawRsdLcawlHEqua3LcsQXHNN8B3MENtzzxj54WFxFPbdN9AIP4R3LssYa1MtwldgEQCmfTWTpuTB5VQDO+4Yrgn9TijExdr65JNDHqI4gEyGvEZERpwMIM+hp4EDozwywmX5ndvaa8e3lz8pS02rjT59AuSU9+vXsumeg/as/evQxbafvM0m206dOnXKlCn33HPPZZdddsMNN9BywgknPP300/fdd98+++xz3HHH/W+N2+Ig7an/V4lqNv/Ytft5003hKub0LlKc2zWAZn788TiX6JZry1ShiHXUqEhxlwYj5ienQ6tTlTpdAgV9vnvdVcn7SY+TTFQeJ1Lc6Cdrub78HZzFS/lKv8I4kWPKZEZOnm/7DVgX285q++ez7fTPcNx20j8a9n0qUFGBJE9ybo1Ei3rTxFJgxlEirDoCYcpXSHOSKaCVWHrrrRqi8gWDfJ/aSeqvQCuGw+lwSMxP8gSYg4xSp52L/cADIZ9W41LqT7tzV1X+a69tueS2QQPfy2fsYttP3maTbT/t9pmxrT52scWCcbRfjhvmGBAKWGutIJQf/jAGdJZZJsRyMIhkz56xk8mBISjJUXaXpKjxVVYJX1WP/fWvRwr9ev6cYUtTobmFFgq3V3+OXCTOP3/o79YtxFI+x6dAeeTIaHvNxunQokoy77yhHxRk6l5Qg7mvlMn0qM0xJs5I/fatb0U57TnaZee22xUVFiXKS6Jjnny9zDyfs/BVl3qVGSarnxxqrIpAKclEZcsTjyYw7Nu3zb3V7fV68rkcs1RVGchQi+7FEELU+qsu5fnyy3fEcdvPjG0nvt/w5RMCh6o9B8FRQLY+KOb7zuoQJMioNJdEYDlfYg4m5qj6r38dQAV+ifRoeg6pCEz7qm3yUhx5uHpcaOxdwIn+Bg2KrKXnpFwyTf305OSfyE4sCNhu1Dt+t6wG0r17XMKg9Gy1VZiAq+Tpsffq1dJjlUF711bVC7Yd9j9jFtluBpDyCDOyBmAoTTjR3Lv836AYEl11njipTghIz/OvlfVHALVS63kr+dSfP79aRmDzJGUyo6+WCSpXWGGGhgRtdddKK8UjVxpSSUoutVTzUoc2XzJH3+ZM73hs227cVud5yCHBcTq0ag4wdaof/m1Zu0Ej6V1zFbkct9W6emYAHT9+xtoN1fu2H35YW7tBByiLF16ofXsjqsq1GyDy3Xfjz6t83zbHbfV+cuFU4nEdY79+MRa2ct3aDfm+rQiOr3FgWeGccp08Hs+1HriN9Rtn0KP9qizmyHvF2vpqJxJFgsrjWdiDn55IoKdO1IauGC8Qk8hjtSvtSSfFvR6KsbnEMXEkrwy5XoCfHkEgSVi+55wTen5ZJk7kCNidKB4NsjinLNFK3s4BIeYu5fHsxJTNyWuvBSPk1YvLooGyuOiiONJcRvNnb+vYbFv/vm01BxjPbkhZ4TzHbfN9WzL1azdgzxyH3Xbb2toNmvLUssL5VmXthhy3tcEhefWTsRRXjhUIWXTwsHTwwW3GbYVEsJ3jtvuUtRvYCOsAbFgdODCaVRmAlvcnKFm2rHDOBCZMiGGQ68pcdzffPIsVzg+4KpQoKgDAD4tI0AIDM/EgrvKaM13BpFxaZjhUJxL1OjI644zYPZHnBTmIcqLMCT8yf/1rJIKlczfCs3RPCmbSPbh00WSl3+1OlEqF5B8zDWWq6LPLWhJaQZUq2NAyJaOrlZiTs89u3ufm5pHTZjTnbG1dbPsvybZrrBHgEGEhRwU4/PCQVGYNL/hSbNbCu5FCD5cHHSib2mBsQqpf/CJyYWyy0zPvtFMwr278pz8NF57ZKDwn1/Pq/HM2cRV1221REpKelwsvaw4Xv4NYsvOPynRlSMGT5l8u/fvHXSDO2uVFnmOrh3DiqjhDhSCOnM9XkbbbrvOuAtlJ2Lb572N+eEaYIc3f+U7YgsLAA6BiWK607h+WROgKJtzZcMOgxXvuCed0aFlmhY95550h73l1z27JL+YBXrXstlto4Fw//niglLvAXcXaO5fZxAkDG1tGCOTp2WijyIj+nE1cmQWa6sHu8XMiVk3ASGWnuhRDJbsqJcWcDB3avEtz8xWz7yjk1gHZlnuFvLSWKtb7IZdbbw0STLZVU1oLOkGQ2QtYcFN+HZDzBFds++CD0aIghUbHjo0W0tIffBCwxubMQBb4WpPIBTc99FBADYzefDMGNM4rc24RoAFZH1vmwcM4ctHALCHZVjMjLK2ebHvjjbWpngT1z5dll3S/WLsd2/YtK5zTn7NHMxX6YQIati8zPHlA4WGO284/f5sVznPcVmEGlxXOse3+ZaUcaOM+MGl1on/ad99Q7vzkslIO3FdfN9zQusK5vMRoarXd1w1q224DfX3G2WW5J3n9139FLKbaf9S6wrlqzHHbrhXOse2O94evhN1UF9jgymeeiWZCHGxe0zQ3R7dKZvTouLTjjnHO/l99NUCIndNlw545/qi3g59k22nTArfk33knGksj5h9x9MO5JgCkceMC5FCHtlC8bhjDaiAtpWWTbWmDBwQnR4yMsv1MttXoWvaRRwIJV5d/oW+8sWXIBYOOi//Dp1fjtnuXFc7Zy5prhvxjjwUwflO+VliqdYVzDgTYMNX+bVc4z3FbVUQew3J9AJivSn7RRQOHnoIGHgbSJ6DGsOFlrSvlcCk86TFlhXPyd9wRlZz6c6UcOFSq3PAyU9KXuJFFq3l2oW/LFc7XWqsm5uSSS5oPvrPzzQGmprp3D36Zb744LrBAnKj6bt2itVgFr3CuuULGJR2g9AUXjGPeIl2Ph7b4qgsvHALzzhvQ5/RBwFFHBR1gkF3LPANOsM8FZT557Qqj9HMDdysLlecXBJW8o6qs3ORdy9dZ3MM99wxh+h94IGzAJRSMECVW99Lc7rV/vrOIiUdAA5tRYGiDoTQ55IsKe/QIK8KVnpRpUXj55bVFIngEDNLjMAxgBSMGw2j1H0zaCUdm9/LxKJdWOcmzJYVnfr8vq0DyFBC3vLgnPH0egV7kx2UVSJ6CDs+O8VUj0HNVFFVenNlVVw225XS46tG4BqwaQVDC/d+pE69wPvH9hvmOCxBC3TzzRKs5OodMu3rWTAlmmHT8SlnDqXfvWJgj8UyepL5zxbKITqriqemYcVPCqR69CEhoRW36g8hRo7eTqdALhMCjjZyn6yoMqoCNMWmDW1VBMm8s97Zsd3D9HGCxCuRuvwu3gzYOELKDbcBAgtwaj4Nk0R8255bmp0BQjYiXK+v4ZSyI5ckDIZj16lX7Yh576iSwPzFGDXh6IwQtnDqvdRVIaAc2IOeBJdtuvnmN6Fcsq0DCIQcCON8o80CeWqaadKOiehx9BbblhbgqJcWcDBvWvN8tzVfVOpXZ3joa2667bhg8AGlFRKa+nGBAjcc7u76seAis4ikBMqLkuwnHsIxbhLE6scUWCybasiw3q9n0eDr8f7VtmbJcM0cAzqAkvxhme7gMnhjYD34Q9ubEJSbXp0886TrrROCWg/0p368sCg3Hq68eJ+S5qwQ4CBLRt7v8pIR5E16yrHCuS5M1Dz3/4xLrCSkkEoY8Pjt5O6+E/CabBHyVauON4wjiGY5tWdZIp4SAW2hgFWym047bTvqwYcDjwV+/LR+qgi5/yjl85sdO0CudK6pxQZojJsLQmy6xRLgC5HkS4iTpbiGMOgnn++Cf3ya2aTOS8NtXxiy1ewAyX4LUnXscKNL6EJL9CmdCv0tm8cUDJFBNBnJgD050LX7CKmTizVwqEITgHAXTox4kok7M6Bby+dcxPXogd8E2WyDDLxHSAS1O91PW9DMc0IXPvEVhpOe/1oqEH0SWrvJmHO0uffe7zcsd1Xxp7dWL2d46GtuqLFFVY90K5/XjtjYEyguYedzWLXy9jGVyJOG118Kbg29e57/ahtr0EJwLEaJn1JFwN3QVOhslFwFJ96QsDT7EWYhV3wN5cMZ0+cK8CTuG5Ykgbr039AAWq1YngiI8yFZzsq5c4dkOo1wYl9Rq+hR6L44DwEl0L8cW6KWLEPVzbgT075X1U+knLGsdmKpWNqXl9uZkrGJP3HHLLe2f9NNsHZttZ/66oRq3zf+FbGKCG8tK5rMct91ii8D/S2XtZFvOSgMYn+vWnm1HvD3mqJuDHwVM/NB8iQh0dfB2EOL0eEzGC7EoFZ/CCSNFmjnDHHSxSigSS+m2PaAdh0IgzENRvr+IdlWU3oiMjPgKPCoywMYE9i9LmiqDPimHH7kOQMgzUHsMQd0yHBSvM+PVCs7kjiXwMj1MQ7THLcsJzfr0af7moM/zy91PtXWx7afb2r1vq++FV9uA/9faDYj1tLLC+QllCQndNcABEHOF5nzftn7c9qGHauO21fu2V880K03OYpPy+b6tkBBrqNh831YhnSxV3rclgPFvuy0GTHLctmtWmrJ1FrZtt3aDXhnb2q6d1doNvN18f/bee8Ns86UCqNtxx+DNM86I4IkS4OfPolQGe/rpwcIMfKWPWeEc2JqawuSPPbb2Sg9Uz3KF83zflgMuI6R//vnxl6OTC8v7tphh7NgwBy3XSd+3BcecvUJftG1ZxNvJT8pEQYKCHJDFJtpf/zZwYHRlhPNDL6SgJfiJOjHhLT16Wl4k7vhX27rYdlZbx2bbyR/t9KO/D9nqgSFHvDVk7ZuG/NejQ74/Ks63un9I//uH7PoU8A5Ze9SQvZ4fstEdQ/Z7ccgRbw9Z97YhP35myAqXDvn5B0P+67Eha904ZPXrhhz1btxCeJtxQwZMGrLRnXH++e3MCBZy62LbT946Gtu++GL8VzvLHem0O7Gzdh5ufXr90e7qP9d6/782kQ6s6C3E6UAJKJxxnLjhhrUF1pwIjvj1ffuG205A+CNqW3nlwOLQodG72DGjBqaEcM+ete+e+fuoXC91xBEhCdkSEbF9vfUCbboi/RAgImL5ihLoSXlRXo8etSm+9HY5mKi0sE7GkYCQjQkqm0I2leXXuHIklbATzwE2der05575RzvM/nvslWXX3rc9ZnT4BxdcEI2en8yALkTZQQjeUB6eRrLf+lYNzDwDQb1In78PXX36xCUsCVSUAKGfYCmched+/YIZ9e75N5fenQB+5+aTAbb8h5Aq0YRcJJLnZOBrcAVRpaIcES+xROjJj9zkjoKd0KMD4W0MHhzmIHG99Zq/dULnG0moNvEUD0s3OGFCvMKi03Oeu5R33ol/UZ2PHz8jPfcqJU+Ivf/+DPl2t9SfP/10wEqmmeiEdymxXkai7lFnWKW0U2LXG8sxy5+XdAnt/qnnZuqicz/qqNp0ClVKtQ8us5mgNle5kPnW19FHR0p1L2cB5gjkm0N5F98TD3JO/XTE13lLyju2y8i9WQDHSnmVxfHH1yasyKzrb2dd1bl7cxaL2d06NNsqNYjlG5z/Tjv8trTUnjHY9pK/jdnhnBlND07AUw+YaoeWBGSeEANFsASnSgbz/vKXgRyJcEUnSTIYMwXgLZVkXuIn2upzce+gsqir2523w3baxVFlrhXkyzQyXUaVZDlv3n5E8/AP2rTop986LNviI06TPk3fOKS8IKU7Va0cKN2XQEAf5aoUnZ5Lujh914EHRvuJIzbYIBpM3yjKRn9/+UuE4eTXXDPcPfJ8N32jXOjJ6Wz0wCNGRFtqbF4kx1NklJ8D0KzHXnbZGKNI/04uX/96eH/5KhjvjxL63agMzE4T8hbpp9bxzTfbPJ0IizzJY8sCfzvtFO4A35P+9dePrpt+kRRweHAZgSmoedh11omSbF4WGUS+vAAZwZMj//fYsrrECivECXl3kefYAvSqq0bQ56oTbqkCuEuOKplvcsop0f+rZzXDdwYawtuXtXC4CVKUh1pZS1x++ZDnJu9ZVopVRdKVTZU66bRvgL0Q7lR+4td0wMSmzW5p+unEpo2ubzrx7aaVzm3a69GmHe5pOvqvTQc+07TpTU2HTm5a7qymfR5rWu+apsF/a1rnqqbdxjUNfbfpsMlNP/pT7Me80rTLX5pOmNL0/Sua9ni4afVLm06d1rR3kT/+jaalm5oOerbpBzeHwLZ3NW1/T9NWY5p2ub+p8aOmA59uOmRy3DX4jUhRmP88v2n3h5v6XNW0/rVNe4xv2vPhpr0fb+o/pmnXB5rWHtl0wNNRmO9fHmrJ//Cupp+/1rT0aU0HPxeP8N9TTtrkzhOPmJLPGGx72ZQx+18ZgOcHMEYY45ZCSM4kC+qQDM/77htHYGMU4ComI7/llvFPjHulgKXbuZZAvssuIQ/SjEJ8ptDgl/ohnG9LM0MjcOmlcULPzmUB7O99L/TzbY8snA7ttMHwL34RYu5lSjDPuhn7xRfHjfKVDrerrBI4lx3Y771388ZnN4+YU+B1WLY9um7GRcF1u2/J7rsv6PW6slKOqsS/F14Y4cz118cCNmecES2dMy7CQf23ZBttFP/CUaWpji7fkqE8nuDIkbUFRdI7Q5S33x7aNIxtRFkpp1evGDzducy4yO1dbLGAmrwmTmwz46Kr48YFeSnh8x+zUk41bstz+LivG4Asx23rv24AF5ISZV3NSpNfN6gW3nS7cdvsLXAxitQrXHRRPEX1dYMcf/3reKsmZ5l58cUYhcivG44vq2fzyutnpcHgOiHsTJ49iCV1e1K6vm6o/iXLDTL19Brlx2XGRZUPnLqonHEx/yXTTJddFs397LPBBWedFaGVdHyBsx5+OBru+boZF3OlHPLirYUXrn1LRkBLnVRWUBcRo0ONC1pwQi1vzpbvq/bvH/95aLL6b8l4KtfPasbFr5RlJvb/+JVyqnHbXLth2LDACViqBd4oPDMZBVtyydq4bc4aQyG7q77HubjMSoP4gJABylpJ8msIXhF5Ahh27rnD6JRtj7JsO7Ah5Xaz0uRSpHyd/LrhtdeiNqTn1w1XXFEbt9UlkM+1G7Atc9A3gH0n/bqh2rrYtott/ylbF9t2sW2nZ1sEwXRzHAcyRArf+EZUsXYS0d97by2IliIoliIdbxJjOWqzT5/aLONikMcfj39vUl4VQx55JAgfGg9uMJfg+tvfDvSIQTS8CEgQjew0p58IEZJAELOLTVwCiIUWCijIC6ZzlEPubjzooHgTADnKmn4lcemNMlFktYGFRDhDbQrgSbW6SDxfYIQS+pdYIvjOebdutb+zGNVqq4XYRmVBNg+4/vpRP8BHGxKEcpWw6KKBaSfkV189aoYM295uu1AC6/jajb8pc6J7nHnmiUpTYDaTfykQI+wqsOqZYEh5qM2/zhZZJORhWm2oLubBGDQEY0buXWxrQyU6YLhVM3r9b34zmsM5wtUozjWQ+gQVzY1HspLRAYZS8ygVp+y+e/RzKhz50gCcKQ9vyAupgbQGIq8RtalWw3FHlpF6YthTJJ5f6+J3SAAb/EKMHnQMWuBKflBZY4laTc8ukm1hhutw1lkt/XZvM7/tyHfGDBoTN6JCqhDlMstERh5BIA/w25Q1pJkMjwfb0q+cPXqEvDpRAwxZ7K9y3A51A8skCbnYlZLQA8bsLvXT6WEd858uxkvYiaNzHgaj3mCDqE828q1vxVNwU/Qo5KV7RiVhIFDKn2CMKQ+3qoVpcNRU8gEHNG92fvPIqa3tN5tbR2NbVZ8LkWmSXL6le/cYMdSZf+lLUZvQAKZqPz/tXXDBIAu9X37m64gEXSLsJOdsVLlqOeVJusVx3nlnfEZJuap3xMVS3Ejgy2VhGMXwMz8g/uIX42SpspiHxC98IRLtUtgGPLk611yhB5eBoIJl8SjJ6fqrDZN6NPI5KV9OIpO55OeevE7sufjiMWirMHTqITwOtuU05TOmcoYBXo6yps0zKifWQ9y8b/Xg0oorhqQnrb6Elk4bS6bQjVS5UTorxaF+ErZnXWWtUpsTCWLe3v+XvTsB02vK9sd/f//bbaZNQcxtJgiSGINu1EUbus0ubWxDE0PQIoYYQoQogjaEppMYYogiFKINQRK0Ia1NoYgpqiTcNiaCjP/POuvNm0plQtKXulXnOc/7nHefdfbeZ+/v+q619jln72INkrxEzpmnX9htsmw74tvK9f4aLW9HGVpDc4kkcFZ+jA4h+lqz60qdknDSjOU21NrZQYlJHZFP/HFW4rYMYNgjmV8A5wv8qSOZQ37IzjQ6pWj966+CpG+ySSkf12ZuauVyFXA260xfUrgAcF2L1l3+UHpMFmx7zchBC+86rT4gykj7dQl59YFGBKcyqY95R3alqBLVVqUszim3tloxwWlWSYrGQYgyyfwzvdw+6u9CKZtvHo7O/FO/kJa43HJBDnwmNclr86pyHWi3fsEwqXfZLFnnhRaqmv+/m947CWl29Ep+3aAz3i9WOBcaMKc5MrBwva8byivllL9uOKFYKYeRR80sHvMlJCGP0d59tzRBPYM8YsR0IwnQn+4nZ0H+Tz0VUZtTjKeAhdPCU8hnUDmnF4TV1YU1FiFCxtChYd5FWa++GhyX3wrfOOuVcnJWGrfGxeDaCDCZ4iOK2cRVb2AxHzl3ZsrUNXfrjyRIgZh8PyFHEqqnzibOCRJ+5vzNLL+q8t+lvPRS0HeOJMjngWJi6fJs4qxFeSSBi50TON1UzCaOHdxmvm9Lnl/G75CtGxdpqqeD5llpim3OXzdw67TnA8VKOU6Vv27Q+LgpRxjIE37zzWhJ3qt8bPzTJ56YtsI5TJbnAJPJkcVXfLnB/C67BLS++CKoJJ/C60EgAUJBvV5jAIA5w6BXXole5oRyX2yCcZLaP2eNmdVKOTkrzYyziR9XrCxVHkmYMsOau3xwwMZ6t02dTZyCa5YzitnEKbKm46YcXswmzsXWhnL75z9LIwnvFzOWAefbU9fcVYfySIJqIIe7i4VU0AUV6N07cK4y+cEYgsqRhMGD4yxaaOojCc1s28y2zWzbzLbNbDubrZltm9m2mW2b2baZbRsm1dt+emy7/fah/MstF8NVqxSrjKy7boyToh4sBknaWg/5u9pqcWrZZaN7pJDxC4h+0d/qq4e8v2T0bll+nXUiZ8KOFUTAX2fXXjse3ZJ0arHF4iGyHFLe2bL8CsV6Gy1bRqGudSBbwtiQ/BprhHwONMvKKTVZo1h7nFLV3+SzdLFCOGZv3TqgRnOo3ErF8uNKBPqNNy7VR4mKWKFY8Ib6wY0MNU7WJ8vNiuWYsgOwPumkuCM1lKJWKxdLmq9WLMxD3g6ULNPee5dKzDsCPuh0TNKv9peYq7M4UKhf+pPzMNnln+lS/G63XdNl27e+qWxZWWqKRIjf7BSJ2YCAoeWz3ZxCfLpP42Mr6aBIhnxehbwZSxggAwB604GubNGi1IMpTzhBbtdfqREJ6czN7+rFKjtZNBmZQIUUB5mDLna5S0gqKPEgfYUV6tbcdrqVcnrVDlps99IyM4lh6F2yWK0K8mWIyHLuPRmqjEzcO2Vs2zYGZLkLjrM+0vORiUwUnQhs1y7Id7PNIqtUbacUl8DOUWxi++8fRK+UBD8ZFyIHFKF9pOczD2dXKdbsSQ5RNxTvBp31SyABvNpqVcse0/RWymGL+vcPh4sz5fe2YikLO9cAYbHtmT5qVOks80VeIsvGajno1y+E8xTOuuOOMNQSHTsrk8w/f3kNmZgpxHSGSzJdJqxflitzjkD+JUxMhowkfzPlXZtZEXY2K5wpjnkT9TcdzDLbVylmgdtzz4ALeHF2MGBT3Ro32747sfLgl6L39XjfYkZtO2BIsQNJwsZB4greHEhJROVVQJvCmU+i1AGwqaTfTM9M8lpFOM78E5+g6MJ806u+fKLRrwtFXfxB6akyMpF//mbi1PrXXXVPl1PqrQLZ55NBu1QGdAWgbEBFRbjh2BOMN900/MTGuXFsm55vu802AQiOHidLjwqNhUJ2kRdMCBnYRvb5mWcivHV2o41CntUSzDq7225hPwkzj6AAUh07xhtgAhnHPIUE39FHB+CYRA6gRN4cewtkXbrE0ATAib532iliQA6gdCHeYYdFEQw4BAvKtt024iMGdostIgLib3buHI6n/DEmF2+//aLCO+8c8oBIuP7GAuc205GEpro1crYt1tw95JCgKhx0/vmx80+RkUQggbott4xvI2EMbsX1MAPwoMgdu/DCiLsFOuKSfCEP8NhpflmPHoFehtmvs0gQyF0IhASgDpy4k6JsOIdAqL7nnnhwf8wxUTocKpQ8z448YMPnaaeFanTvHiqjGhkwgaJSZCXDnMbowgvrDj2jy6lf5z1Oe992SvRWVIYB+Oij0toN+b5t49yaJNs2mHExv27IDxzq6gKRyGi9el83pDyj6hQAgRfsHn98ILJNm8jw6mJdsvxIQTo0kM9xW2xLB8rjtqNHB3QAqKamtLJOft2Q47anFjO8iHo++KD0pioS32uvKPqf/5y2Lhm67FLMT9Z5tuuSNbPtzLbGzbb1x21z4Qz7E08EHiS+VKzIybRDwj77BG7HjQvM5NcNOW6br4jmuG2PYl0yWWFb+JxSxEM2Zz/8MIw3lnRtedz2xRfj8YZE9Zhx3Lb8dQN5wIbPHLcdODBAm+uSbbhhqJVS4PnoWa9L1sy2s96a2fYnybbt28daI/bNNourVJJfw+ngenCcm+rWzLY/fbatunNi5cWTArw31j60/cUPdRny0O2fPbRtt4eOvfeh7S566OaPE9eNbteuDZ5k/4CtsbFtrrGBwlq1imHv7bePqOc3v4ljBNe2bekxlyiMgLPLLx/hvF+o8tfZrbcOxAiyRGcdihkr2OK11gpErr9+CNvlD6ktW0Z4RZinudxyYZlhfc01g9BBc+21I+JTVsrnUgi0woXitV/+Mg7Ad4stSjPAKnHddSMRy2+8cYSKzipXtVVVVFh/y2Fle58+sZcP7M89N51kU9oaN9uOGF/Z7vYYetp770AROw0/gAFjbCqQQJE4XTgPDyw9uw4zgER+mWVi0MlVkAw8e+4ZgAdIEIJS+IReEOUuyF8ioIK62B/eIM0l8s8FkMCVLXftfPMFUJ0ir1DgXH31kM8nwGCZY1/SVUM9lywWmFl11aiPzAFe/Ssq6tru3qVD6bvzyZOnvPn6xJt6T0zw/l/a+TlM21xujY1tt9oqdnAEnUUWiS7HnrACPfAECjklFYTBx2KLxYc6rVsH+H73u+Dl/ADMAQxxPOEGfTP+UCUdUsvs6azLcTFhpeBxaHaMfLEzHVANl5NRFjhCHiz6S3lkrkrQKRHbUpv6bEttOOYQr2L+QrY++LD0VLd5m83WyNn22xLbzj9/gG2llYLRmGchF7qEXs4mHIIchKDdPfYIh0A6GRBClLCKcF2FIp1ddtkQzifp4h6+AicziRK66EI+p+Koula5VEYKdnaJImiE/P0CKvXJ/MnDM4eACnDF1ROPJ9sutVRU5uKLQwWUqOgCz3Ub7Njl8Mb67Ot/eWtsbIs3xUEnnRRBEPsvlmF3km3FSmx+377xohx+FKPhxEceiRdBbr89duYJxAnkSjliN/K5Us7f/x5nuQaMdv2RhAcfLM05f8wx4VaURxLg8oYb4lUnmObS5vNZNQFZ0Z8Q7M47A5QiRHl+9dV0Iwl//nM8lOjePSr2/ixmpWneZrY1brYtjyQwyWIUDCuEX6VYZyRHErBnzkqT79u+/nqw3iuvBKl9+mnwXa6ylSvlCOfJ50o5+b4ttN9SzEpTHkmorY3xNH9xdM5Kwz2jAtyLwYMjZw7yq68GTYMfv+Saa8Lwq8+FxbKqgKpQW/2RBCqgPuA9s1lpmrfZb81s+5NnW0An/MwzceHZZ5fewbDP9Lh8cO658YTaXc94qsGFlJDuUbAZZa6/PlrgggvihSGNJs9ML//26hXNmF9SZGK3bkENF100TaZ+nuQnNdU1d5se2476YGKvP48/p4BAY9/PP2f8qy82tVlpECvqUWnaK7THvCCMgn//++AF8dRll8VHJsJ8EAQs+MN3cGNPgsakUHvwwTH0vffeEelXVweVuPCww2I0FuLx4JNPRv4IFOPYoRzbXnttjDDIH0wRCg5dccV4UebKK2O/5JKQgWnCCDQhDu7UBkalY+qnngo+kn766VExf2mFW5gN27qQerhrBSF6dbCrOeg7JWcFSafA7ov5YYrUJOf9cpv0SlvtsENMIke+qiqEF100WmO//WLqxXOK5R7Iq49TMrE70HoO1FlTVFSUSFlDLb10tNiJJ4YGikzvvjt6QYwpK5m4L7wiku3aNQZVNLtIWfhJM50lP3Fiwxv8zlvjZtv3J1ee+Wl81qjx9fh224WNhx+qrP0HDoyGvf/+gBzQitMffTTe/4fSNdYIbs2vKLPfdTF5bItVc4ZPaCfmWiROQINzOFhoDe7v6qsHKetoiNWt//3fQZeMcevWQZowL3+dxayCBBh06hQvNV5RLBuDoB0wtxwR6OJVcCOuvjq8kwLPdfudNCu2ffbvk6/vNenVNX9X8+xnNV1urnnui5rrn6w5+bqaP/asueuVmnP71TzxYc3vu8Ja7L/cqabHgJqBb9est0fNZQ9E+tWP1Fz3eE2Pe+Ov/ZjL4+yZN9X8rlPNsLE16+9Vc8eLNX/+W806u9dcel9N37/XvPBlzQHnRlY9H6w5rHvN+XfUtPt9zbWP1dz8XM3mB9e8OK7m4ntqduhQc89rNYtvVnPQBTUX3lVz4Hk1T46uOfKSmttfrLmgf81mB9U8OrKm7YE1N/29ZqN9a24bVnPVwzXr76liVx387F39Jze8w++5/UC2nTx58ujRoz/88MMxY8ZMmDDBAVBOLLZRo0Y5njS9CzPP2JYryn9caqmYdMqvffHFgwugcMEFY9SJBeYvYD2YRpfolStBzN/8ZsxxeUqqPfYI+l5ggRjMIpafx2Bqx9Qg50xq0SIOllkmrl1kkQAr1DqmGwstFBfm2SWXjJo4lr9ypcgfDwK9v8SkqDMNsfPEVVLOMlGEv5991vBOyxsex844Oh/9YrrqeqtA4tBWxSe8lJOuUhuS1LX85S4HJ6e5UwQdI29Tk3xNmHue89vm65O2fFZu0xrkKTMHHDXkl7tM0bJTV4HkNNE3jj93zAGx8pe77vq6YvVJxgCziCcYsCb+5e6bX1cu0S0Qkl2fE1yBDXsMmVI0LFjm54Jkfv7zwEyK/b//FyhNICXM9KC4yk6APLDpIzET5K9YTA5HuJi8Ks4S85fiADAdUah8qIar/MrZqRReuZjrTomKVjHAUJC6qQ8x6TCD1lMHXbLiinXLt+3yh6lL5Uy/sSb6/9tWG5e+9OWhN1gFMr+Mzw2b33RTuCabbBKcLp1pB0WJGZvCf/1VINu2DXPCe2AzBI6DBsUnzoz6lOJTZgbp+mLaXxaCu6Bxcj7cPYpVIFX+hBPCVolxoZpD8PTT4TRgg7eLNVW5DuVVIBX04365i0wfeOCB2267ze+jjz569dVXOx45cuRLL73k+Oabbx45/Yt184xtNR/11kZ8zD/+MdwlPIsv9B9q45+6H1zD+2OW8Y6u4jzqA82NemAln9uiAxjif2Eitl3Hk4eDXxdTYfIE9R8SdK2cFUGY46AbzjgjMIdfAHHLLYNKZAsHKoNKMHX6qucU69HrXbEzNgQ1dIPyevQIbeFBnFWsCSYTNcFcDea3rb81s+3UrXGz7XuTKo8ZERRGazmzgiqNI36Xopvg6phjgtT0KdLUBfhRo9F/gTy2nW++kNFZhNENDwM+0QpSAI/8iscp3oZEvcO9Feq1aRNdryPgE3kRFiTtumtwlptCcFC92moxIrHZZmH4ZSJz+FGZffYJ/3frrQMeupiDTK2ARCXJC2708h131O12dHl+2wZbM9s22H4g29p4ta+//vpf/vKXnXfe+b333nvwwQeHDBly5plnPv/887fffvtAkU69bZ6xbYP3bQXsDjQNfJRnpXnttWnv2+baDZdeGgRH1WFIb2l6wRSrTl6HQWT9tRtQ8P77l2alca2IScdzB1AMitf3vIMvvohxW6gViJ1TjNuqDGWAb4iXCLXqoyY3T52VJsdtBYa8Burkr87WwXMct8Vf668fnAUKdGzjjWPnv/hLtVJJpOeLZSCiuPykXf1lnu9IoMiyPOGcC5jZoEualAr5yzyU83fAadq3mL/csRtn2OQgfw4RpiAsZztky5NLJSuJ7lrsmV/9M1E0ofwJf17SZNm2PG7LSpXHbcG1/qw0nFDKz1AhjlXqrd3wH/8RO2bMtRvwTs5K8369tRveeSf4V1Nr/5yVBhTh+cgjIx2KPvooCsVfsxq31bP5Pnh5VhqZAHm+n1t/3DbZkxLNdtw22HbX3t+ut1HwLH2kd6AFn+ABM/KBGbcJYHaesprwYLgCqu3uUh6YFW0HIWeRMs0FWl6LvwwAdiaDEyVyumWVYKZWvJ/f/jaYFIDllhMgwLC4Yc1iTnHtBpMci003DcQCNv9DU2gZPr5fAFbQBhtUrdyx6uavGt7h99x+ONuOHj16zz337NmzJ7bl6g4ePJgm7L///p999hnm7d+/fw4mjBgx4rTTTmvfvv1HOnvuNw2qOeASc/GV0C7rBBOYK636IYeE7WLN4BjlvfhidDN7njPYY9tLionfgRhB8BowtR7160JuRbItYVjEiVpHl2AcxvC556LnIE9HjhoVDiOXAUEjWWqgMlyA//qvUAOJSFl91IQ5RTccTxrLl+QJ4iAWwl+2XSbKYgBmw7a5oRiWw+3wXmmI6g0fHsevFAsPv1wsHmwnkKv8ZroUjO+YJHkH/vJPiaW8v8Tofx6oMBnp5KVk/i7MtVXL+adwlkJGj9BwfocD1CD/rGFmUpaUOHau5vX4X2bbyZMn33rrrfvtt1+X8uv7c7EF2174bXAZQ45PGWakJnTgwIIQmCFibPvmm0FkeoQtBJI994x2Q7XcW1fxDc8oFo5lR7m02hw1oJX27aOpYQ+h6C/mn2P4+2K108MOC2ERjz5SKLrEwjkNAgvNx8QywhHx3EILBcZkrr/yQyGaBeTyB3g5UysGgCEnz4OhRG+8UXdw59mxbTFlWJhYd52QkHmiQj3BSa3sDspQSeClQKLulamrU+cliUkwkzJ6dBxrB4kplnB1FTDLKi93VepLZpIHziYyUy/yqvr557UOhg+v6vl+1Z0/3FHI7QeybQ7R1tbW3njjjdtttx37/+ijjw4aNOj444+nEnffffd9990HrGX5eebbcjZhDg86OPnkCMYdIFMg3n33CLoFUMwdLwLI8KYUd0ge+QKQHScm87KBfAo+qQtRMFIGdySOKPMxl2PXUgYczSm4/PLwEfyFP7rB9rLtMjz++ChdZfC+UtQHsjkgu+0WpacBUAoZPgVSpgBw7O8ZxWrhLqFdX85pWvj3i/n3uANbFKv2Im53TVvcJu+AZcbgKszI0wG3Jp08h9R9MfssijY54ohweFu2DIstkiKfAyNu6qhiDdQli8V+pCtFa8jf5YwBC8GxdYMbFUvzktdovCq3JsXffaau3SJIlInM3SzPwv3iC0bIKcrPy5iL7X+ZbXObZ77tuxMr//uF6Cb41EoY0AHi01b2HFkS9ecsgoy6RksfFhR1B9icfXZ0OmEA0J66g7WWD3mdQi+gEZwkCla0vMz1C7DpC/js1SuEdStU66yePcN7gGreK+9Y7+drCTIHyMQzRpah/DOSoyZUJr1ptS3KCt/26Jk7UtPYNt+pUDpnFuOr3l7ForlCNPGTe+ftUhkClMtfiBWM0iMCQKvC3KADimV0c6YIbhYYaxOZwLZqw636cKQ0ghtRPc6y2rrqV8VSKe6XG6uFCeSrx5rIqXxhn41ZbrkwYJk/YVXyV7b0q7Kyar2zfrS1Gz799NNrrrmmR48eNxbbmWeeeckllwwfPvzhhx/u3Llzt27dXmMr6m3zjG01Cu0t7/oM7MDLbfgLMfoGmdYXKB+QwUHllPJZOTDpuSM1yKipmZZi95dlrp8yq523Yq+f8vHHkeeMkuVd5oLHOb4XxW3sUsyJQ4Vyflvujzu95ZaABWbPcdvy/LbSqcRjj4Xfna+yl8dtUTY1e+SRGIxjycVWLJBIlvIDliD0/qmrQO5TzG87blwEd7lqpHyw9ivFuC2tYJl+W6wCqVy3CcHChfyWlJ5pcNWmoqDPx6fDQ4Y0vK/vszVuth3xbeUGfRpiT58yZtrZ8d7FUm9lgfrotTtFIFP22KN0IIVFTyBxioWPGrwBwGbE5Ez3GUEuw6++muO1dUPf6XL6hIZ3W2zT2FboBrrMBvLC3QriGlMNUNytWNUx57elyOwQrxPtQotwgCuQ47ZpLaSkPO/Etdwg1IkfbXAL6hwF2qQIZXMR3AJ1oODt2kVcxfAIiwmwYZodYlkUpAHt/ADygweHG6G75V9/3PZf/6ra9oqq3j8S23JssedHH300ZswY6B89evQnn3wyYcKE8ePHS3Q8cfrhuXnGtqIY3kG+paRvVl450KYnGDqBg25gEp94IniBZ9enTwicdFL0Ez+XE8qu5usvmIV8up/5lOknvjWzbWNn2/cnV+4xNDqFn8UX4X/lI9YTivUQtRiP79lnw6XVYrwwYfsmm0QP6i/ymU5SZw0dGn7u0cVMdT/2vBl1daXHtzNuzWzbYPuBbPt9t3nGtquuGk15xx2l+TpzbAtel1oqun3Gp2QZJuutXLsBKWjZXLtBo/MO9OU66zQs5Se4QQZYMDaCICATG4IjbXS/KBKY4AaUN9ggvB5hl3SQZWBAjZh4X8ylTRCxiGnDDUMe+PDgCisElJ3ClUsuGcGBfEANTNu0icuRgual/+RdKH9UqyWZLqVsumloCM4V5SkRCyyzTESjMhT/CuUQNE2DbzFdUx5JGDml8oBhYfBgWPy0yy7BvBpQb8LkS8WsNIxTfsUgRp7V1w3lWWk6dix93fCjbt+JbbP+ALbVVmE2UGQOJS+/fCSecUZ4DFDEujtLwd04T4KHBJziUcDjMbiEQYJwQHL7YLztttEaUlAn3HKwtt46rpIJm7ToolEz7hfo0hG2qn37AOoFF4S+gLHcfvWrqBIBegG3FAF3y0f+jCJHTbY8tu7df8yRhO+7NbPt3G6UFWfxQ7VAvhHMG4VRuOFLrr56UCpELrJIPM2QyGFfb70gxDXWCEhxToFJDk4BnOOUB3fYOuWUoFFaDXYsectiyQnNu/HGkS4TzSidPPTnt/NbbhlnczoeisEj0Lbolc6kr+2sfNiGtdYKfVAZjE//5mJrZttGzLYJidWKJT/cC9cBYHKKBmHT0kvHwwacyMwvvnicghZ/YdUlgEceyMkAHpRShHwPh7vw38VK6VgSUBdeOBAuEykyaVEsv73mmvFCAvDnGw4SnSIvW8eoI4vDBhic+sg233mA+Xz/YdVVq1Y6sarPmIZ3+D23xsa2iIZB4y7Z9cRKxYcxbBRE6rM9i7mRYJcnlTPLEEh55g5N4A4CzJ1ryWtiINABP/2NHjIzeJNhZ1TAlMrtsUc8TgFfkLryyniQyg7zLiWyMZ07x0NVrgHIL7FEKPkzz8Sp0aOjKbgD9JnOf/FF7FiSNbr//hgW0FbS33knMuGAyAS+Ux4uu3cPSUyRX+VpScIO1Ef733JL1AG1cI3pA5PAFuog+XM3mvBIwnvvTtqtTV3H373bccX+HY/9tuP6j3b8TU3HNkM6bv5Mx7ZDOh7yaceK1zoeMabj+o9E+roPdTzs845rVEf6Kvd0PObbUjrJX73Y8fAvOu7wSsctn+2418iOmz2NdX/End8JcTPdphtJAAbeJTRSWDtr8fbbEaux0716hVY+/3x4rwDDML/7bmnYhOe0fzEtXz7HfuGF8JweeCAU4bnnwj6Bbg55XXxxCCNHCL/22shQ/uiCX4UNHn88PLCePcN7JSN/8oCNHzgc+Q4GeYQu5qMCTzwR1ZAPwDvef/+q9pc1Pd/2o4/i7RPKnLtjtlWLf/jhtJT6B/WF82/99PxbN/MvYX5aW4Nx22Tb8rgtNry+WNURpdb/uuH998P3efDBEttylJwaNy7AmvK4MrdcBRIoe/cO10M6EMsEpdb/uqHBKpA5bvubqatAlsdtc9XI/LpBjAbZskK4TZhtJ0yYUvvB5Ppg/L+xA9GsPoSc+bgt9xx7znTc9qCpq0Dm+7wcfwVI5B6Rx+tkuPPoT27Di1UgbYjlgKmrQK5UfKwPw4Iq+W9bfN1w+NRVILEnRwE+//Wv0rgteeQL6pcVq0AKBE86KS5H/aqRq0A6PuigJjlu22Q3bIu2+KQYDZ6SbfMpH99WeMVZACnogSeJHEm4gXeGHRWSF5c99liA+K23Qh4DgtGOOwaH2mXOUenfP2C3/fYRlIEgVkXBnAI+AnmeRb6+k0WgdZAVN+xafJv0j3+En9K3bxREEy66KEIwnE5Jzjgj6pNTK8zF1qjZtglu07GtgCk/JuSHIkdw4lGK8VlrLLzJJsGhglGAWWed0mNYzgRIS+Tkkke7ZHoXKw1SBJ4BhxR0X345OJG7AKgcVcgUV7VtG/kjcX2HrAVtgwZFWHbcccG28k/59G2ffjpcE/J8WyEvFeCgqEYuiMWV3muvqi0uaXq+bZPdWOPOnQOsXINcBm3EiHhWlg+1ROsOjjkmcOyvnYBdSsoz7/kirRSAk4+OdwzB/uLZzNOvfCQ6wLYEXHhMsXpVWR4Fi+YEXCSPPz6QLcpTE5nzu50ikK8qE863eknms44Gi11+z62ZbRvXNo1tx44NkIABNAIJn7RXr4AQhCDHK4oJm8AjEXvggYEZwCsPaouiyMOkSyDNAUCShzHotUtMfOY7oC6HdsLEnFKovylPJhXBMXfE3yxFcXmV49Qp6dBLzIWHHlq143VVt5aWX/vBWzPbNpIN23INWHgIyG83OY+AKBFK2GrQybffb7stGPDkk8PbBTVYZKKTYTkRfId8zeD000v+ggtvvjl80lNOCb9VJorw63LIE/4L4s49N4hVufkaA3bmrqoDpyD93wbyymUb5KPaX3wRKepAnj8+F1sz2zaubTq2BTwgBDl4g1KwAQ/OKaxyV5EdkIj0EWuCPD//gTp/ER88DxxYQhqUAphr8SkxYdmjj8YxjXBhOX/FSbGDOsAj4pSXoTBLJkI9Hm4WRz7tQTl/oZuNH+3aysqq3fs0s22T2d57L+L3F18Mhr3jjhhRGjYsgp2HH47hfBEWRPbrFweoEFCgEwULuPbfvzSwtdxyQak4d6GF4oEAcAudVl659BJionbAgMiEi3FTsSa8KC9fcrz77sgfj/+mWMkYjmkFekXT0m+9Nfay/AcfxDgyjO6wQ1R79OiICj/6KGDdPJLQlLbpRhKE53vvHZ/MYrrnnw+aA7OnngpQ4bjFFotRVIxMZvDgIF9YhTFof/nleOTAkOPiX/86htHuvTcA5tp99w2mbt06xsd22aX0mT6Czvx5Eg5eeCF853ffDdjvt18oDtjzjuV/441xIH8VANE33oiRB5C+774YOttyy7gHSkegpqZqq0ubRxKazJZPyd4v1tzNrxsAqFWr0hxgTz4Z1jjnAINmVvqss+KxVc58ZoPCli3j9SNh2oILxlOs8lM1vgDeTK/h8ccjE84y2s2nXvl1Ay+g/tcN9Z+SQWpOS1Z+Ska+PAdYsi2enTSpiX/d0AS36dj2z1PX3C1/3YBAhw8PagPORRcN/xec9twz2A0C77knCBHaa2tjODW/boDn+k/JWPqcA+zKK2Polna4EDg5E8idRjh4553S1xALLxzy/AYZ0hEH5TnAEHQ+JeOdoHJ0nxMw2eRGoIl+3YAgdBVHKSdhmDAh9FmK/a23Gh6UjxGTds9PGBucaiCffwmPmdt36+bx5jYBDg54qZQf08HcmmsGaNZdN19SiXdfcCX2FPiDkTB/iy3iIYM72rtYSRDDMviMPC/gggtKs6HDek6CB3O4VSa0ArJdzpOFQk4rgy//G26I93hoCxwjdO6DdAGaS+Sc8j17hjy/gLOQUwhhWzz+5pvhFCtxLrZGzbbjx095e8SkxOC/dX//7bmdPGVebdOxLYsu9AFgBMp4QxEkP/RQOKfg9ItfBMlC9T77xBMqERhPFsbI8GFFYHjqmmsCUaw4lPI9XYthXbLhhjFEtuOOkXm+vTtoUIxC8GQdcCCEbrwQsCcvEyA85JDIH9vK1l8sDP/PPRfvjUE+dxvVtm8f99C5c1Tg+eerNu/R9HxbtktzaPFvvom/H38co43Igt+n7bCM8ITCa+ju3UO96Ql5fUDh9Wu+g0LeX/2EeviAyIJF1T3ML04hlmM6P6mNmWF4oWH55eMF7PXWixeKeQT4FI3+6U/Bm5zcNm0ipNq8mHOdwBJLxAAC3KyySiCYy7nttqUpEzH1ZpvF8UorxbsNnNZf/rI04boiMn85r7NO+BHyJ7bJJtGSa68dx36xPM1xlUL5F3Jbdtl4gVxxypU/b5qOXX55ZCWRPO6uN1fR990aNduOqpt82inje5/4Yu+rv+qdK7n+YUjvP4+JvxIv/aT3Ka/0PvCR3l1H9r6gLg5S5oR/9N7upqnyQ0P++GG9r/i898kv975mXO8DHu591dje+w7sfd77va8c0/uUV4kdsMe4hmX/SNt0XzccWqwKDMCQ6Xe11QKT1BCBguUii4SFBk40miCHQNhzAMyAV1HM57vWWpGOXgGsbdvAG1ABLRUGtrwQODP/TJEtAb/zzReRXC4gu8IKkXhoMR0w70Se3OdttimVK7GYZTHcCxbCtVtsUbX0UU2PbfUNR1UwO66AlMCEE8cYolo9yugxhoJrgYBw+447ImQmnyaLu/r66xG/3F+sR8Lb2n33cMp4ygSQ72uvRahCrE+fsG8/qS2jJOBln0Vh3boFnriWMHHLLUGIV1wRvgDzfuqpEZHp1wEDggdhhVOs3V56Kd7C4XJCIb9AiIRzmX2Myc/FnmIokT5+ZHuku/yII2KIlibkJLYYU5NKYaJYNS4JjnYVM0BtVio+aYdmwhhWnkhc13QqJos6tJjKnSM8/QQa32tr1Gw78v0pF3abPOasi8a8NQrEYj/0uDGvjYy/Ep95dUz3P4/ZZb8xVX8bc9+gMbvuX5I5s/uYX6xQOj7s+DHDR44548Ix/xwx5oLLx7z9cci/UTvmV7uOueOBMa++N+aiq4itt+4Pt2fzdpvOtwU88BP4s/p0lmMEojSRqgIPGMOe8EgiFAE5lwiZMvMUlvw554QLBVqQmV/ToE44pPK4mApwldLbveeeGNESqFVXx+dLyPRnPwvtQJ3+5ko/SiTP2aIs3DIeg0CNw8sSQDi/++GHQ5h+oRqE3qVL1T63/2gzLn7fbV6ybc4mXmbb8mziQrXyl7vbF6tAIgK6TZ5vhV5T/oQTGn65O3ZsUInOwMsAYbutWNXxJ7XNOG6L18Ts7YoJocuz0vAcMZ24CZpz3HbXXcOQoMURIyJM01B8ihy3FfhrImiDKl7Dh8Uq6/XHbfPrBv7v3XeHc0EelNGxRoNXZ+G1rq70dQNnFqGDPjM246w0ZzX5tRvqzyaeG3OlfbKd88tdnlR5FcjchFn8rNyOKtbaKH+5y0vgLnxQTE17330xQCmqmxJk8hPZZjluCxL0C4TcafnrhtxyVpoct8V94n30yl2o/3XD4YdH1kiwb994tMXJvbXe2g30umPHaMz8uoE6/Md/hMo/+2y4F+QpOy14vFhGT7CrMrlMSefOweBvvz2TWWl+9LUbvtc2z9hW86EDeptsC2r5yl4+4eHx7bRTcIcG0rv66cknQ16ci6oYTw22997RAUyl7pQbFtDlmEtb9+8f5EWMoWNFf1JbM9s2crZ9/73JJ3SYOOSIPkMG/AvEYt/t4iH9R8VfiX1GDDn+jiHtTx/S45khPZ8fsvXpJZk/9B6yyE6l4917DLlr9JDD/jqk38ghx/Ybct+nQ9p3HlL10ZC2HYdcNHTInR9GDkOmrLvG/2r7zGZrZtsGW2NjW53k/rXUhGJKTYFTv36RqPkQhEZHIhrummuCLp2SSJ7fSsaxzs9j8hqvV68gGv16dbGGqLN6lxgBvfKT2sCU48Mbyvm8OTUtWgTscJzf1VcvzZEKlxgWCW65ZYxQI1anePrszVtvxS9m5Csdd1ywp1NYeOutI+d8ycblYjFFSHF5+/YR6C21VARlCsLvuJj7v/POAVxnpdABYSD5JZeM4YU11ojxB5VRQ+kqDPFii5RXaFNl288/m3xr3wkDen86oGoiiMX+l48G3DUh/kq889sBfT8bcO2oAf2+GnDbVwN6jSrJ9P5kQM93psp/HPJ//WRA//ED+nwWF/b6MFKuqRtw69gB/SdEDgOm/PXa4pHGT2CbbtxWTAkt8IDsQGKXXSJORZFAsuOOMWAKe3YcJ6LfeONA1C9+EW474eOPjzexJAIYxG5QTAO29NKBQynginalQLVjuG3TJnALz/g6p6rZd9+IetdeO8CprJyQjJizfDXYRvoKWmmlcBHUR4krrBADX6rRsqWCqja5oOrWuW3Yxsa2dBXP1p97O1Nmv38XmQb7XDzM+bds33wTbw4ytkOHlvbBg+Pv009HCFlOzwO/nJz89ZeMndFOmTyVB+W/udMPkplevtxeU1NKYYTKZT3zzLTMXZj1ySo1KCh3x1zdudgaNdsC1Iwo+3fsE2c+tfePsE1jW9UaPrwhqBwLgwg5KIOnPm4zEcxefDH2Brh1NjMh8NprJRDWL6KcW7m43F1IONHrL3+Z21tfYGiB6vK1RUFV3Wua3rhtk934tqw6c73YYuHb5lI3DG9G7v7uv38MknAtuQMMPkfSzoYLuJxN31MOHAdOsRiKVRd2rbVWeL7kebgdOsSYH9cj3WR/c9IQHseCC4bXIN3lHFVeCQ+C8ZdhFr3ssvE4QimHHhq+sJSU50cT3nXX2LkzTzfd+W2b4Dadb3viieFaCqegTuDF0xQkOeZOCsXE8rzO9dcPRHFR4VAY5C/MgBlIC5W22ipwJa6HJZdgLjmImaCXl8pXXXHFAC0AA6cIj38qtgNvvjB/Fj7lxrmWIb3gKYMuhZKhS37/+9AIf9u1i7Pzzx8pMkz5E0+sWqNT03snoclu5XFbCKi/wvmBB4bT2mDthhy3tffuHdQsEV8AennthvLXDfixZ88Yb+3aNQZqxV/o8oEHGn7dUF67YdiweOXu4ovD2xVhlb9uWGKJGNW5444QEJepDMQPHBhwryxWOG/yazc0wa3huO1vi+ni6uqmnDl1hfN3340hrMcfD94szwE2alSwas4Bdu21QXaXXhoDg/D/2GPhZMCSrHMOsNNPD7Qj1rvvLk31Wf66oTxue8jUtRtynBe3StxjjxiCI19eu+EvfwmNyDnAFl00nF9WgSuz+eZN9euG88+Pl0JoeH7a7Jh5PPXU8PZ1Rn58jQ6k5IGduZPul83MSVJyotWzinX3Ugw1aGUe39HFDC+8ufPOC8pwkF+n+FUWSSba5cTgQPdIcUrOrKI969OrV7xDlrNeyUHOkJSXg0hmAh+qlPLXXRcLQM1+a2bbRs62H4+edMrx34BSpz/8K37tf/yi0ykTO3UYG/sJX0fK0Z/H74nfdOr4badTJ0fiyRPi4KjPOx0/rtMpk+KSP03qdOzYTsd9FQfSU/iozyIrZ/2VfnQh71eiv/KXrV+5ESB20viohvxTpizv1y792DGdjv8qTslZijqkvByO+KRU6DFfhsCxY/vdMvMQu5ltG2yNjW3RQT741pq1tfH5B21HOvkegpTu3WPoXZtS8hxARG3SEeIZZ8RLfJoYKfTrF5lI79Yt3hUTR8tEK4u+5YC2sCROsYtTiOn4V18NySuuiA7Idx6g57bboi/hwDFuevbZ0ret4KKeGA0TCYiwzFVXRVZ/+1ugwTGUqBh5FwLTJ3Na4RzPulyh4nQ8Ljy/4YYwDxpkxIiABQTL2U2pz0UXxaiCQJ4BgJVcpne55UpTKIDvjTdGiHT11YFCt+Y2CcNuvkZ+2mlhBgDdhXAMowsvHBGZdI2Duw84IPAqOnNWyOZeWrSI1nNHWhgLqwz5zp0jXUTGotiV2IRHEnTg2V0mx/SWq2+Tk1zWHnBq7d8/qP1Tz9qTL629oipSdvxD/Paqrr3jqdq3vq798921j75eO+Lr2h0Pr+3Wt/aFUbX7nFQ7bHTtiZfUnnVt7T//p7bi0Nq+j9XeNiQOnh5Zu8fxce2rn9fudERttz61ux5d+2xtyMtftp2urH14eG3fR6Pcqr/XrrJl7ZVVtbscVTv47drfHFV7Qe/a3Y6pveS22P98T+3R59ee3zsu3+GQ2h79al8fU7vqViE/6M3a9SqilH5P1h5yBoFh+1181mEznyF6upEEzg2wbbRRwAzTwSHqgWcUec458UEBLaCGJxWrCMIzBtxmm7Dx4JrTce25Z2SSI2AXXBBaRpepG/RKpBQrrRSQ5nnkpGKcmPXWi2tx7hprBD5diDd4A4qm6ZSIJvqFed6GsvxlD/74xxg649itvXb4ImuthYirNjy36Y0k5Pu2GKT8/mz5fVt9acMRr70WL3AglHfeiQfxesiGEXCQni6/b9umTaTrHkZVc8vkvvuCI+SAxPEFZtHxeMqGtiCGJG+OHwoNuebzww9HF+YMQ/BBpTARDGFe9YQ1fNq6dZA+J5HkQw9FvzqWuYqRx8v6eI5sy+qq8CKLhG+LwjDafPPFY1l/99svfpdZJtIXWihIkKcJeUsuGYl2YqiWjIPFFgsBiax3yuf3ZosvHn9hKx1b+z33NDyov997b9yLFnPMojieUabBzobNat7p77Y1arYtvW87pVjJNLd83xYeRCHcBVu+ZisE4TEIKeDklVfiQDqg8gqZUjEH657v22IW/SWScADtAh3XAmp+0sqCAhh5+dsQHA9ALyhXLLjaaqE1rLIwRQAEhIAE1XblMrdg7HIg5/R98UUQlkpSLnhWSiL//fdjhfNDR5buaPptGtvqd7cAYLDHXcgd5PL1AB4DdoNPiey6RMfQ64DMsssG4UIslBLLHBZYIP6SEXU5AOZ8PbE+3mbzVyM8+GBgWKKDGTE/w+VVZw6rumNunz82NrbVMcIEriXehADRBB7UWHAjUQrbheDgEnpefjkeZTJ60nM6QWcZLuzAt0VeORmgoAaJy0S0wgxiHMzLT8TOPDgWjxgPmmKQZDBFOoIRRLz77tETTsEx1MI6LIJ4fjqsntBJbdhtbAvEUM63RbKOKYCKkQd0pcyRbZWe89u6ZOedQxNWXz3GK+TPgEtREz77jjuG602Bcb1ssaGbUkPtQ9NIqp4DLUBvybuQN6p6HH+uK+f9J7w1brZ9f0rlBV8HlvhWuTqRLhAwoU5gEAZJQW1+dSs6GD06cCKexaq77BJAFbGx5cgRhDh0YhrdJybDGg6gnY0XiKBIbmBO9C5/8lwN2SroqaeCKJX75JPBtmiUzLBh35ih2gAAWNlJREFUEWAxBa6iNdwUbMspueSSABse53/wjVVb5pRLrOYAGcHVa6/VHXTanNn2k2KlHAZAbdElt9GtUTGgLa+UQ2cdKI5S0x3mhKXhvQrIaDE1cdfU/5//DPBTXjVRZzGowJH2Ue3PP29Yg3m3VVXFPpdbY2NbniYQgAX2vPLKwJwuR6N8K1wmRSSC+zQ9Af4jKkGv0o8+Oq7Vc+SBm2V259IzuEaOZ58doIRXOcgQ+0AKQkwxv7qWJJt/WTF3nC4HGlc5BZoiF74tgha5ZMXI40dZ0RAQkSFfALKVKFuVyakOyctqjpPg5Lgt8HFyIRJYoR92cw6w+uO2lFMTTZgQCsyLR6zHFu+Kv/tu6AmTICvmJMdhYb08bptfiPyEt8bNtu9MqPzdkMDGbsWKG/bDDgsAAAZ2gz0pescvs6ev9QucCJkdgBDYABWi1LlcDZ2bEZXL0RABaIcl10oHe7wJk/JHWwAg2w4dArHkaco//hFQlz/7TVi58j+omCqeajDhLsnJ5AADPmEeYFhupSBopSgU8i+6qO6Q07v88eOGd1ts07HtDjtEPMqRxLYz/bqhPAcY2s1xXuEAHVEW8qWhaoKyc04v7k79cdtcKaeZbXObZ2yrhyBjxh0UANEB9PjNQQDgqy+Tp+xoURem/IxnDyqmkYdIBO1Yr4O7gwbCMsGYAKUFqY2uwO+Zib8Sy5KuxdGgxg53KOarlw7od98ddfCXE/odn5I1s23jZdsR31Zucut0YMtfKAU2MPDXjuNABfbKgGxwAEX69JhijYPyqfIxZwINHX98QEW/ozMHM+ZzTLEmwjHFygV4Vsjlb546uljmICtjlyEgoWwC6nlgoSBoDvUXAnUHntrluE8b3m2xNbNtg62xsa1QgpkdODDcSRyhobEGKhHsoBLdyWKLLHQeiOy0Uwgw9VDFqoOgkARi0NzSS0dcJtwGGuGVDMFI4O8YqsQpZ54ZRZCHLSWy54MHl76CxVYCn57FmkXgwh1QFgChVMJYHmjQ6Ouvx5sAvGAhD15TLtdghRWCiDnXdIanjC6hSvg/x5GEUaMinFx33Sga1DbdNNi2det4bdDNStloo/i7zjpR+VzkfIMNSonrrRe/7YqVnyVuskmk53u45DMlL+F6/4S3xs2270+u3POpMJPJHbCHrfiY2OSZZ2KEB9523jmGd/CdEHuNNQJd5FdZJSJ39ARd5AVMHF6Al0MiHD7lmd+5/va38axiySUDbHhWzcnjU5CDwCeeCD9auffdF3gGFcqi6+EQLHOyzZdfDiSz01wWFEkFZLvNNnEtsA0dGo8EaJlCFY1tD/jTrEYSXn01bhSs2rWZ1G7Nz9pt8HW71t+2W+1f7VqNa7fhN+3WG9uu9fh2a30e6Wt81q7t5DhY+4sQtq8/rt0mk9qtMybENpkY6a2+CmHHKe+AjLMbT4izjuXQriju37BTOEo8l1tjY9tf/zowAZ0wVFs73VMyDFKelUbbMIAcNwLoD7UBB1iXZ6VZailqFBnmU7J0UUHQMXOqdVGhIsiDJkKEmldeicGjGZ+SnXNOaALjnzEaNN97b2SST8lc/uabgdF8U4pK8IhzkIEB4FOwEP/1X3Nm2+atsbPtyCmVBwyLYdMcsZ1S7ylZTU3gBD5/W8xKkzO0MoQvvBDyq60W5JvpOSuNSwDelrPS5JjvL38ZiVgb2y6xRDgQQEgLziuWl+cNpINSfkoGz+2KSTawLaCKuhA0LeBa8jrROtj/5jcxCkzj2rePa6kV5Vp11SjFWUUfeeRsnpI1bw22ZrZtZttGszWzbTPbNuqtsbHtFlsEqkTByy8fgfCGG8b7H3AJkYssEsFXixaBHig59dQ4JgAcLVsGzflNebGziF743LZtnCUvQ8Jic8cECDtWhNDp8GK1WmdbtYoATUEuXHTRuJAYol9rrTjVs2d8WbjMMiHgWgeqpz5LLx3FqfYpp0RctuCCpfy33jqGa51affV4X2fuXo1qIlvjZtu3vqlc8sLobqACJ9gDEoAERaiDE3gDsO22C+gCzEILxRAQCMGMXzBLeTBDnU7JgTyY0Qh5EoM90MLdcsuJt/kBrpWzdMAD1AQnlXGhRL9rrx2v6Ky0UtQtcatKJFdZJVhbzvlNthSwV1UVw9eHHhr+zTLL1P1yqy6H1za82+ZtZtsPZNvJkyePGzfuq6++Gj9+vGMHX3/99aRJkzLd8eTpZ3WZl2wLQ3AALsl3aMvxr34Vb94BCsQwwgSAg0EGu223DTw5BsqKioCI3YWbb15iXhjCgyDFU3CAeeXsEvKu7dMnAArl22wTr/5BG1pn5F0oN6WDJgXIud9XXDEuV6j6yLZ1saIBsXbFZPLOypCAosvycvDLZZ7VRrUUyh2Wp6ygP5XE5Vl/+knrNEtm5ZREu5Q2bULeLn3hhaP1Uh6/c69k637dqfq4BWqpkhpTKfY0Tg7ciJBCbtkC2TLqn2qsVvlQZfvtp+UvdEh5t0/tFZTyUibO/Luj77I1crb9trJlZTSpxpx//ugUBJeWPvtLQ+k7Dqbu0M4Qq6mz0SCWjdfCwOavRDILLBCZwIDQ7Xe/i57S2vLXI/Inw9WFNy2vc2VOKcjjU12TKybkUzX9pSaKcC2cpLyedZAIp0o0QtjncokCu5QvalW35razeidh2LMT9trt2x0WfqbiV+Mr1h5ZsfXXFSsOr9jmm4pVairWeq9ixdcrthxbsdH/VLT5tGKpYRWbfVnxi+ciZdG/l+TbF/Jbf1Ox4NCKDUYHKpd8oWKjjyuWf61i088rFnu2ot3nFRt+VLH95Ir1R1Ws/EbFAkOiCGKt6ip+8WzFtt9W7DA55Nt+WrHEsIotxlYs/2pF648i//U/jEJdQn65VyLdVb98q2KrcRWti/w3+6Jiu4kVrT6s2PKrqNX6H2668ug7b/mR5gBDr/fee++dd945ZMiQN95448Ybb7z77rtri61Pnz79+/d3UF9+nrFtjiRgRo4hF0D4I9K54ooYFphxJOGdd4JNRPSXXhqPEURqtnvuid2FSDlnE8cv5ZEEXHDttQHfDPZ5tTffHLE/HRBh5UgC4D71VCmUe+ONgO/zz8cLMQTY/KuuCox+8klgOr9usOVIApZ8990IwYYNi2CNvEThWC4hMavtyivjqUW3bvFOgq26OirvTj/8MB7BqcZXX4UG5pe4tIKkRHv5nQQ7D90pwaP0ceNCIffdN94dZhTznQTtI+TML3dzY73OPz8O6Pznn4ewPkXTlcUqkPXnt73mmhhUGT8+8qm/CqRTWqay+HK3yc9vO91IAp8xI3oto/VgzH7RRUGID0ydTRywL7ssOkVH50iCG8+vGyT+85/ByBArjNPRWNWmtfPrBvkfe2wc6/ezi0XhciRBobvsUvpaAWzKXzeoSb6aTQvA7K234il/4q08kiARFCUOHx56ce+9+YVL3QV/TcEZt9I7Ca02nvaOgQuhAp7rrwKZX+5SCiYEp7Mr9d9JqC1Wgcz5pnMVyJzflmv/zDOhCIB3663xDDy/3LX17h2Gh+IDm0o++WRc+OqrQQ53311a6+HxYn7bkVO/3J1SLEHmHskfdli05Nix4Wnll7s/7vy2EP/2228PGzYMt3bo0GHQoEHI98knn7z++uvRbr9+/R588MFJ9eZFbGbbZrZtZttmtm1m24ZJ9bZZsq0Nmb700kt9+/bdeuutJ06c+NRTTz3wwAMnnnjiiBEjHn/88VtvvXW8Voj3RN87//zzd9hhh48++qhhFj9gQweUVnOceWYgQyik1Rz/vZjLXQ8hUGwr5LnwwuAaUU8+UoNgkAKsq6+O3YW6TR9ffHH0pYZ0FkHr/nOLpSRdctRR0T3o1S9KGjw4XtXq0SPiaH0PvnJ76KFgW5cDNwEUBtwCMSQrlIPO/v1DDAc5BQ2DBgVedTnGHDIknjaAApNwww0N77S8sSXq1r17gENWjIHK43SX77RTcBnFa9cumkVW1BWU1c2ODSlzr16xsxOCRMpGHicqcf/9QwC4zzgjEPboo4FXzZXqZ6exrJoD9/vCC/Fkxg0yJFqGkVD/E0+M4TzVkGd+tqRJaZH8t9gi5IcUi2UwchpZ+6i/Rmg8bDt58uSqqqojjjjiPOw211u8Abbr4+EloBWWjDXSGloG3oDKfsopMZKjDSm5DgJsTY0iNSPEolTGkrXzV0/psqWXjp6lEWCPTfQUboOW556LD14ZtjvuCNLUUzk7BxgrNDVIh4KNbPfbL570qglcATlA6ikpu+0WF8pTNchzX/AObDv74INxFSNhv/HGupMu6XL6zD9pDbbt9T/fttoo2NOdwifUQRHNdWtoTlb5xiSV4QO1aRMP91q1CvWhJk88ETLkKY66qQziu7GY4gOu1l8/btC9Ax7UHVKsKqYIYspCqRSfOrsv3AqQzBWUUhz50w7mzSWK4Osk5uk7lnBTlEX+L74Y9k/TIZMePaoOHlDVf5r7+MO2H862H3744bXXXvvCCy/stdden3322ZAhQwYOHHj66ae//PLLDz/88O233z5hwrQ+mGe+rebGbsAEr3aM4JeziW6Ab4EF4vmDflpyydIpKdIXXbQ0IQB2cPlii4XYz38e+Sy4YAjklJeoyoU6nre78MJxlYNDiglhXe6U3OQph/nnj4G21q2nPdyQLlu/MiSWsxDIxCk2XPpSSwVtOeuURBc6m3V2Nj3lmW60C9Xqdc571jbrv9xypXnp5eyvRAccWJkz7HYHcCNRhdXfr4Ik/uxnMRjnNrMa3KI114zB6JyEoWXLUrNIRL4cWw2VhdrxL7/VLdS/X2JYmFi2w3/+Z7QJPQdxbZ7CWbSuqRfxfN/tf5ltc5tnvu2bX1cuck70gs5KGGhVB5pIg+dfbMtxS1ToOIk6xV+thyPgzbVSpOv37HE9i0DtujXftlaEsyS5bDJMpCUgUx38xcXyxCP55Dbx40CifLJoifCA3x07QLXO5nwaflVb0YstVrfIWrN6ShZse+yj315+TWmehLwqNUiQKk96lLBUIlc9J0agL37zkYOynFKloqyoPCXdeedQ2Pnmi/AxZZzN2moc8kqhCDRahnmhRPpCKbhW5fZ3KnVH0erg2rwpiXohAe+4qGTVwgf/aLPSjB07tkOHDqeddtrQoUP79OlTWVnZq1ev559/Hs9edNFFl19+OVe3/oOyeca2qIGTj33sTK5Q65xzQv9xIv+RheeGMIZMMTPurHCDocYRlJ8vJpHze+qp4cphEw6FwIpHIPTgi7HYLCGTDljgeNxx4Slw3OysH39W/umywb3EXJ6eSRS829WEl40BVfL008Nua1yeHbOpGkJpET1iUjpurS5mTfTLv6Yqs/FtcxPQqU++IcRP4bZTLfn8plhbjIlWNNf1ttvCHdAyfAduCN/WTWkxBpxbzVbzI5A+h0Ll3Z1G4wvnu2t26ZwvO9dG3XKuLzDl4JBxX6rBL9Da2tmtaQ2JypI5z0LpioZdsSf/XdiohTU7YTuxoUMb3tf32Ro32743qbLjB9EXTz8dIz9ajx8HfjpIa/NhGdRtixkCuZwgof3hNieuBK10wSTCm/j6oIPCF/ZXjA9aAICAXKv34QFOgF93u8rluEMvyJ+VBRKkA0XwAx45fAFOQAWKssI+3EzyFAcXOwYV+HQAbNxGPqOrOIkc4TPPrDv2gi6nzbxHpvuWTFzIiwc5vzJRnJviPGkQKKKA1ApImO2XX46CFAFm6kmP3Jp4jk6pPO+YGPcfjDm8vHKKqQ3BUoaqRJ6ZyahOvKtBlOivqmgHbUsAX4O9xpHhQVNX3nUV3MK8/MUHrnUhJ3fDDdFFVdvuPybb3nXXXffff/+zzz7LsaUD6PXLL78cN27cY4895phAffl5xrZMok0EYc83W/UTaOoMnj8m1ayKLq+5m3OGCTQAN+cAy3kM/vWvaMQc5833bUUWAjQsDMrkc5oFHZOTALG3s3/f1q4mwj29pRexdl1dYFdZQjCOpI5EVTmhjFjScXkOsNmP2+ZW/8tdQP/009KXuwcW89t+8UVp3BZS8/N2mef8ttTV5kL+xZ1T57d9fOqXuxpNy+BEBia/3IU/O+S9805AWSOj0Ry3pQOqkeO22gFw/VJsrZr1yXFbLkZ5FUi/OW7bt6nPbxvjtjkH2DffRI9rNAgBP9Bi2nXB6NElazrjuK0K4GXwLr9viyBwDZkct23wvq3uRsQ5buvyfN825wxTD05DdTERKE4BIaEhNizPAcYflEPKcx1IHnFEdJxE9Uz6LM8Bdv31cx63TbYtf7mb38WdfHK0QC5xTfUwePnLXRguz2+rtjluSx575vvyfYv5bdctVoEkX57ftncxpyJ5vExAs8AkdUDN7rQ8bkvAjeD9w4v5bbUkJb2zWAWSo5bz21ITwWt53LaJzm/bzLbNbNvMts1s28y2s9nmGdsKagS8QGNHdtDpV8ShfYUM7doFKDXixhtH2OWsA2d1qtgNLrUpAUENNoE2oN9ii7gQrUgXZwFrt27ReWI6+FCcTOziNWKbbhrFyX+jjSIS1JHkYTfrIwdkqm9UA0x1La5XBJjmEmH6uE2bKB2sHTsrihHWqQNUzn6jM3gcy7sk6+8XGuQsq8MOi6KdAlM1F87TUvVhb5SrxXJ6ZpVU9HrrhdlgANydPOkVA+Mu7NLpuT2Xe8KqwAfWmb82rCumsiQpc/foN29W5rJVuvZZe+34i3DVQRFiQ2J2zfiPfzS8r++zNW62HTG+crM7oy/gp23baDT9pZ2Bk3MAbACjMTU+BnGgW/OUdK2HlOHNcfaUXgYAMrLKpWd1E0rCHRpfd8sH9rYsVgD7dTFhNkUgrFAQffLJKBrsQcKFLlEfUJSV7mPCU1733XNPlJUTirpQnijDr0rC3s471/3qgC4dZv4l5DS2Za3dNWzIMAdPODSpDhpECj1KAWW98UbcICdGsH9IMW2TKsFethglJa/mrVqFS2TPxN8WC0Ookr/cL9qKi0m6MOXlzwXp0CEaxI1o29SmbEnt4Cq3SV4lAVX7uJCCtG6teas2OKfqpuni9R+wNTa2RZegn74SS86scabYRumcL44YAvIrxQETyr7dVqxnzl6RZNmcuvHGuJy8X38dpLx01tXOL37ssZCXFWvvlGNiiibDxmah/Gseq5wzkbC/Dz0UBtwl0u1ZVXWQ4qzMy/WXYpf/3/425znAxowJ5GZl3JQKyG3gwKhn/lWuv2ql0GwBB24878iFZHgQ2Q6uJZPHDvyqlZ1ktqEUd6Ta/uZTYxV2mxpQNbjAyJR8tk82b2aYB/bbby/1havKMhyluVjMuHGz7bsTK/d/Ppox259vqMsSCbmX+06H+pvtmTGT9odwu5bMnrKnsKzK+PTrQoaZvAtvLGaqJanx80IyCX5ZZd/1LcY3s3P9ZlmZpxQXZvepW2aS8grK0m+4oe6K/l1OnjkNTWNb7nwqRVbDgdxeeCFOZ6HiM7/OZgUSbwme/M2ZebWMUyrjrEjOsZrY80IHeWHfQrkS2OX2cS1HXrn82Sz0hmLedNmqSR6nKjFFusZxFlQQRdXxj1fdHi9Zzc3W2NiWlWMGO+W6Tp2i4xklQbFGFBD5yxCJsMRsHTuGZ8cJ9StGWHXVaHTHZ5wRJpTZFHlJ59mtsUb8cuKkn39+WEhOmcsTcCye/PXEKquENd6lWAQUlfB/9dAmm4RjyI8Qf+k/voPq8RBFNMstF3ZStk6RP+64cFiEgXyEtMliIolkJH468znrpm1iRgEdj4OPwyNQH9Xo3j2cmkUWifzVnE0+++ww13wcVWXec7kakR7Xyf2KQMnAHNvOhRfyuwXZiqrUimcqEcLcEcypv0S3s8wy4Szz5bkA5AFahXkKmsJfPoK/Sy8dbZVeNnUSunKUtAYBJXJVhI3OckAmTmx4a995a/Rsu+19AVqdpXH0l7YFDx2qv+xUceWVo/VgGEo1nRSIlcjldFBVFR2qW/UL11LHkdHIUIRbiXEYBd3Dh5eenbqKgG7NF8X4dzCQD4ggJ98G062ABCR6U91WWilcv5YtI1KhDuDUokV08UUXhTwfmSQZ8Dv66KjqjjvW7XR4l8M+aHi3xTbdSALP1CV+3TWQqH++uHZWsfgNMlXzjP+oKg9d6RpHbamVWzutWL1pww1D/sILI8p89dVQB0EtPeWfUlKN4Gw6+FC37LJxm5qaomlz/KBNtAa/1YV8akrhFtwvuuApQ/4GGwQ751f466wTT/Pkv+aaalK1W++mt3aDnnD/5XHbq4rFvvLrBlwmBuH/P/dcvK/HKNk0qE1br7ZajNdMKb5uQJQ6JsdtuXvIa6bjtqLmkVNnpQGX9dcP0hGJ6LYvvggcv/deENDVV4cwyiYPN4KsHLdFUmhXtmsWc/XrRQAqr5STFlvid1wppzxuK2el1B+3XWih0rgtheE85tcNlFksJkKE9/y6gQpRHggeNy5Y75FHgm3ffbfh/LYffxx3JH+amdiFdR60so4sZlEZPTpQiNNZFH9ztp0llgh7oH3YLU6HQgGaqhCgJACta5q/bih/3ZDjtox03+m/bgBvKM03AXLcdkqBeSCfEt8URWQ9q3Fb/Pj884HS116L7rZBMoH8ugFOmECYlwj8gweHO6k7ctz2rqmz0uhrmec3vkj8gQcipcG47brrlr4+mNOsNA3HbeEc4dJfIMlxW4n5tQI1xG4QBb0oTw2VrnHKXzfkuG1+3SD0dCGjAlf5mU95fltngd+NwDk7ka/TUrQvvwyCtmHtYcMajttqebyBoH5VLA2ju9kwHgkbIJhz/KN/3fC9tnnJtixwspXOwAvwB5RaXKO88kqpjYYOjSbWz7rHrw7AtpjFMXjpVA6aTkXZYMc/BabexRA7NmQkK4sVOMiPGBGuBFiPGhXWj8+r+9Gce9H3QMzsX1ksAAEQ5P3NcV7AwrYMrGwRMXksSTHuvz9IMIMdzoJEHoccvgvb0hbsCeVKQXmyhSq4wbZZH2xLS2kLa08ncSU4kpGolbjYimbngVJAMHBgsG1+ZSSSytfXUDCboT7y19Q8LLfGl9EsvF3opHIAmmxLc/KBjPyxrbbSPoALrwqlD7lsD9yrpK5xls43Zbbd//loRp3F7kIItoVSrQ0kdkSjzd95JxxYKNW2PXoEYiXqPgdjxkSzM7o6Kz93JMOkQZE8keDTTwdKsS2ckGcdCehWKWDMLYV58MbszOfrr0ciXoNn/cXFpia4FXL8YlspnEG9r4sHDQp2Q2ok0SJ1uPzy6OjDD4+1G74j27pElToVa1DBVbItOlNbbCtspVxAwmCoIWXUOGpLrSiv+80X43AelGoi9fl98QUH1eCyXFYssiu3HKY48cRgW86TpqZo6Y2pCm7Ftm6EvFZ1C7mGNN6A/Bw1Vi4y4RQja9aRN/bXv1YdN6jpsS0wbb11MJR9v/0CGXx+LMZk4Q5UAjpQolMx3fnFqplkdAau4WTlsZbde+/w9fTWUcWinuSFNltuGbEbGyiHg4oFF/TxRhuFeWSQBSb6WB/oRX9RKh5RnO5Uh6WWim7jX6A8bOgS+eNx2TqFtsirjGwZAMGUalA2idwNF85xJIGyqrMMEaVC3S82Vw0ZYlv5Q6py/YVgPrWqgiad0Uo4F3qoEOddVbEwzeSwqKEblCFkqxX5fDImBRe4X1lJXHzxwKUGBzu3DK/Kwpvu0eUS5b/kknGsUAzrAJr1lPslr0RtqGhneeVNlm3fnlC5QZ9oRr2m/bWh1tirWH8eSOx4U48AA8sNHjoCsCFWItQ5wLM6VDivs7bZphRr80wTn2AMt36FGhBOHr/oi8yqPI2RxHwkhY8kgsfSS8evS+SG2aEItGBMyu+LjyB0Mb0oy1MleINzVW3Tpm6LveY8ksDFRmS7FM9UVQNIaDGb4b5AhcbhR/rrAEhUUg2d0iZSGBW+Ng11I2iXvMpoGX53Xp4jDFI0gr/UxK9S3BdJtoGiqbD8tYmmUyf3qCNok/bR7Non3TjxnEaQoVtmZlyuUE26555Vm/eo6vcjzUrzfbd5xrZUl0ul58AOUaJUINC+Cy5YelcBPqBBG+lRnaGliKEMxzoYLsm4Vt+485xZDpXAEHldKyUpQyJlcIxA7bqHNUYrEvPzMJGyvgeInB4MneEvXetYfegSJeFKkBcTqZJTehob4nR6suiicVYdKJ5b+C5sm4+DaZd85C8HoMR9LockheYTbU2kUKXQFkpFXhFpkDSde/n5z4NeVVtTqA9QHlQ8HE8jxABAIWuvKQhwNNwXUlBbmXOotae/eFkjyFa6XlATt+kvoPNz0zBI5z0BrpbXnvnJWdNl2/GVWw+IRgBUjYw09SZQZcvDT3ar9tSMWhjNbV9Mq6bB2S1ndYTuczkwIz4drX+BkFHM7xVdCyR8QGCTv65P+PlVEHnaoROJ6S+djtfIcC2zE3knSBb44RPjQwsOUlVncZNuzbcCXAUV8lfKH/5Q13b32X1Llmybk/MyuqlcCI6u5WCFCoMi5mXdnaJiDrTDr4tp5+hdjt5iQM3CYyDv3oWGqeBUW7PIClYJO6V90j79pvg6VBNJV2c5K11TX3hhNC93wV3oAuAUVQCqZvQXwrWSBu/QIYrwV/477VS18flVt37d8A6/59bY2FYfsJO4QFuIkgDiww8jzsWAIllh0eOPh3oPHhxDojnGSl7kol+BSecLPUBTuCGS0qbY4cUXI2S44YZIB6Arrii90ycHUR7i3q1Yof6rr2J8/YZiQTCBtjhId+a3Pd27ByZUQL9CqiCltjYchAwVRTF0BqH37BkxOw9F6fPPHwRUUxO4l/l3GUlgn4U5+F0+oh637FpZyb99+/BGKYMbZMCxw9lnl+TVQT179Yr3bXkQqsEyiQT3Kha1FsEBd36WzjXQkuIldzRgQNh5AR15lXzwwdKIgaYQ2V1wQeTvEq1BV12iXAbj5mKpIQClXYAlkyFDQs/J//OfoVFau8my7cgpld3HR7uJXsWnGlnYC1SaUbAPP1Ctv5gx3TRwYIQsUpzNhW+F4QCGCFyekwwwtOR1OnrVoeRh1bWwsXux3Byo6PeuXSOaURBg6GidpYiHH45BXpqCZF96KVL6949RAvR0TbHEJGYEOdaRfL7Ph3+BWRiOguUM5OBXU1N3cOc5jyTI+f77g0MpVL5xQcUoCIUiQREeL+aJBifKqGjqc+mlgRaqQXk5CjTatdws9SSf79tqw/POiwahuYq47rpI0UoskPzBGxdrvWuvjTulenLgjz/9dDSj9sG/b74Z+ARpFdCM/tJfjcZX+PjjqOruxRucN9xQdeyjVXf98NdpcmuEbKsV9DSHTpfgUySipUTrQAYZmpKLl0/JyODT+l83IMfyUzJNjFsPnv7rBik5BxiBvsXLLr8vnqSDoI3kTcUcYPmUbNfi6wZ+omAH9MtPyfKpWvkpGaBzK0BHd776aukp2XzzBX/l1w3f/SkZeRDPrxuw7d7TzwE241My8hmr4otvvgklZ9Kx7eefh7cCncCNbes/JaMV7ijZNqc1kcljjwWOc16PfEoG5XRYa+RTMrv8ydNG1ohy5ienVJpXkjzS/JSsckr0C//g3nuDIDS1TcufWyx3D9V6mWummwYNim7KrxuwiWYHHgA7pJgDDBK0JOtY/ykZttW5dxZzgOFQ8qByY7F2g/xtgDHj2g05IcuRU+cA4zzeULxNxVts8HUDIwrM9dduwPjvv/+dnpIBxiOPhMeTb5vl1w0tWpTmACs/JQOn+nOA0U3KeFGxLDGNLj8lO7yYAwzbakOOqgZp8HUDtqXvmoV8rhrZrlilQm4zfUoG0niDd5UT8uVTspwDTEvKsIk+JWPcdKBWZl2BAw/iBSgWRvkLVcyXbsNHnToFoTDI5BlVpANS+dq/FuRNoDwAEixoXF3LX4Ch/PgEBAnoGHYvxy51Xr5hKnGllaLLGUkeKx7RPfLnA6oAVlVDqNL3WCzfmIFvMYsohr1NZxBPLbxwxEf9irkd6NscRxLoFf5SZ/nDFpux+OJBhSqABMVi4CtKcoMwRA95K8BKHhGXv8qnpaqxwAKh7e4LDyJcQRacqZUgkVmik3Rb/tg25WVyySVB5U65wXw5wT1qSQXRdoXa+b/uC6BXXDEe76B4mkDn9QJ5mqBKWr7Jsu17kyqPfjPaGfXocbDRnkAlRU/BD1Rrau2s2RPSGhm94jjNnq8xgavLkUUG9eSFzLpeh5J31rX6TqxDHlRogVJwkIIAQ3frKV3Gs9Pp+dUATsnOhUb6AthA26pVKE67Ym458jzrfFEMc/F/oQIIQbdfv7pdj5rzuC16rSwmiqNQqkS5oIKCQAXzQxGobb7mpal5M9THvfOs6R3sHVlMZOpaNoA8aGFnqooEOAEaRM5psbSqv+yB/DULea0H5ziB6lFtzg2IgiJfRLkqRpKJUoH0sbbcMuS1DDOjqlpS4mmnVe1xc9N7Awy3phGGMwd+NYq+AR1/9S2o6UK/SEEj8vLK8n4J2B07my+LOJZnWd5fvQV2+Tdz0NyuQmokpQNEfhbJFLN3mbNLZHhb8QmvHAjo5swH26rSzcXXEHIgoMJ9i5e3KYn0J58Mx3D225gxkT95hWb+qpFfYcifwqhP32LqDZVRaNYk83egnsSk+8vmOyYmMQX8pryKSXTAn8rK51np2sEppWd74lZ+seKkZ6tm/lqSQpJMeX/lQz5vXFZN+euGPwyPRkig6n3tA1TZ8tlK2ke7aUYNrqmJZaNlp7tQdyTMsped6lt8qgAPNxVfNEjMY/KZQ/qzCspeyK6Rv+PexacEWaieEkcn5hPV8klEpbxulbmrUiPkX2Cjrtd9c/66Qc4uvLX4lMCe+ExoZf5K18iZf+/iK5jM31k+uFsm4EI4zzYZNaqkqqnIjqWn/rr2luIlsJRXhAN5yjlVOwe4Hi8WM3Rtam7m7+Cm4kOSrEPquMT+/av+9HTTY1vOEUu7XrEqDLPM62SWmaNMZNv98trs4hFnN900fkmSYajbtAnrx/NiFVNePkylUy7JjyD5EY5dyFXcpljrxbUOnGrdOuTZPQIO8skGAWddxeoynmpCfqONSmP2mf/qxbo4ipboKk5xPpSTJ0kp0Dz7TbjEPefRsPzuyN25cb4z/2W33cLXkL+7y+dj7YrFzPNDeOz2A/YpMb3rv2Wfi61xs+2Ibytb/TX6BRj0Xb5LAxKbFdNjwolTOhck+FPrF2vrwcnOxae05DfYIGQSNnkKbHh/4pv8sGWdYokjeHA2n6FBgl3mUJHyAAMkoAIwypIhQLow36txOTyrhiJS/r+KCeo2KRbpIZPyQnilyIpA+/bxlOyYmav2dF/u5us97iIfkalS3ogWkC2FSs3N+gs61by6+PJwpvuUfxs+Z7FX3TW56Y0k5LgtQPQuPhxkuHTkTL9uEDfdW7wtKLxl6C4r1twVrDHXOW4LdjkOO3ZsPN+/uRi3HTkyAg05CHa6d4+hIjkcN4u1G6TPuObuBx/EuITw5LXXpo3bLrZY1NY2fHhclW9K/oBZaciL06uL2Ro/+SSiJ/wuf96xGh59dAgI+oRCNLBly/B8/w9tjZtty+O2wPDpp/Ek4ONiOa8ct7253tcN+dUWcIIozw67kcdBMD9oUHQrVxHeRNbCC04r1Ily8uuGO+uN26YZzq8bLpm6dkP564bVVgsM55q7nxQzIsofntUnP0AAVKWXx23B7OZi3BaV3/n9Z6WhFDmVAT2iXKraokUMGU+ZOisNTncBSuO3LrVUlM7f/Mls6tXMts1s28y2/96tmW2b2XZKE2VbgQYqFJJAGwLNp+rw9MIL8aTokUeiz3SSwHyffeIsrsSh6FgHH3VUPLTN9xmuuiqCGoihASC1+OKRfvDBgSpoczkknXpqPCKQwyGHRA4YdqdiZZpf/jLQduml8Zgi32VBcJ07B0bhcujQ4F+oeuihkCcmf2zes2e8m/K3v0UpJPFghw4BegGaAOovf2l4pw02NCoWI0/ZUDM2HzYs9I1RoTD0Rw3/u1hWBPapH+Hllgvl/D+0NW62fX9y5en/isc+wKDvcspqkNCJAAkn8MaUglm3buFPACecDxwYIw/kRfQw369fvLsNPPB26KGloVVW9o474qkR5CO1f/wj4v3zipm5Benwn/JgfMMNMfIArrcV8yrAvPzXXruUP4aV20nF1A354cyDDwYy99sv8gdjlYTqdYslqfr0iYd13brVnVw5q5Vy+DyuZjVGvPDZiHNuGrHD0SO2PXzEU6NHHHvpiHteHrFkuxGPvKMBRlTeM+KuF0e02j1+r35oxN//Z8TibUYcfO6Iax+Osz+NnQ9215zm6Zvj1tjYdvNi8ef55w9+XHLJ2FcsFnfBLBJbTF0LJJcDWbJYL0eig1WLlTNcBVts/iqrlFbpIODyXL1jkUVKKdtsEwX9oliJJxcy8avcXG4nV46xS2lRLDHid4UVAtybbVZadSNXH0l5dfvP/ywV1KJY6YQArOcSHWTkf8usV8rJ7bPP4vUDPE55WI5TTgl1Qqwc5HwZi3pwaQlw4eltp05BuGDyf2hr3Gz75teVi50XaAEG/a7T/a6xRvhxEnMJGdDKU2AMNnDrd775wtXlgYI6BmxRLJKUK7s4RrKAB07E/JUPdk7QpoLIPFejwfLKyvVgZJXwzktWLRZYIpbyJP2mOkjJJZ1aFksokVcQF2SZYj2exRevW2ydLn+oa3i3xcadv/LK8F66njup62lju570adeOn3Q9Z2LXU7/s2mV81w6ju549Ic6ePq7rWeO7nvA/8XvG1yHs1J++iOOuxeU/gZ1iMUlzuTU2tuXb3ntvRBmMs44X7COaTTcNIw92XFEBkWD/umLqa2exmDidF8mw51uxgCI9ZzCAXZYcM7LwSGqrrcLa/+53Ee7xOMRcvAksJqCTv8iL67rnnuGfKjHfWZHnYYcF3znLI+7fP3LmAgCreuolXsatt0b+6G/LLcMpljOLj/FVjGPLYnJAbryx4Z02bzNsjZtt35tUeezbMQQEJxAIZnxYEIJJ5lNcAtXgwcxLEaLxbVlTPHv77RGoXXhhxECCbiAEZu7ngAGBeSjCBNDILV1nnXA5eRLbbRfAFvrSAi7wkcVnr3jTjVx0UYRZ9AjD0h1XkQfaJ5+MKt1ZvJDgEn+FTbxdiY7z4V6vXqEa990Xqrf33lGlysq6I8/p0mlOb9Q0b8XW2Ni2/HXD22+HO3nv1LUbGqxwXh63feONiLPqr90A3/dMXeH8jDNieIuJ/vLLCLIw4IxrN8gfyuvPAfbVV8Hpjz8eMBVY5bhtzgEm3idATyD7o4+CSUVbb745bQ4wAn8u1m4AaDU84juscN68Td0aN9vO+HUDHOa3Wzi0/HUDfsxx23Hjwk7j2fLXDSw9MoU3LMwPADCYB290/MQTAfu2bWNYCY3muG1tbWDs7LMDt+Ie3mi+DfbYYwHF8rgteXRsUyX4BGOXTCkGlHkJxx4bx2q14YahViBNAcGbbbh3DiucN28Ntma2/cmz7YgRUYr7pVHVxVdb5Z3muMGnnpr2t3xQ/7h+orrVPzurnf/+yCMxBk0bqZyrXn89RoEb5O+XmFYqp5fPvvBCDKmXU1Jg8OB4n+aHbs1s27jY9ovPJg1+YuKM4GqM+4PVE+s+mNjwDr/n1tjYVjAOf1AFf1gSKH9dzL6x336BAPGReAdp7rxz0B8WRmTOkseYoq1ttokRq4qK0juGG2wQI63zzRdx1hZbxCBDngJNYMrXaZW4/PKlWTmA21lB3wrFiugp37p1PPlF4ipGHwj4O//8EWqpD0z/9rdRvcw/pzXaZZe4Sm75PQyZvsVsvDPdaCNqvvzyGLGlbIS7FpOCMy3XFctJiPISEXvtFQLEcm4zmuZvzqdDIVkL8qqB97UG20MJXUVeEfL5Q7EGcL9+Ia8B1103rA7jsUKxmCs60Fxsj5SDDiqZCtUm5u5k5ZSbuuqqSNl66whCDzwwBPzqEcaMSjfleRJGfFvZpl90um76dTGlC3yCBEzmC7bwplvhU4e2bBkwk57UqcF1k+Zl1+EH8bVpEwBbaqlpr81q5KWXjl956gJioEULCG+8cXwbttBCpXe0gXa99UJsq61K8gRyKhaXqJ6C/FVV+JfoWK3oF1SANF3IGZfUf/vt6zbepcux/2p4t8X2jxcmXXbJpAHrn1l9+5jqEx6u3v6y6m0vqb750+q1T64+85nAbKtO1fv2LTHaRmdVXz6i+pznqzc4Pc5e/Gr1vjdVn/Rodeeh1Wc9E/uhd1WvfmJ112HVl7xe3f6i6k26VPd8K+QHTKw+4+nqHq9FJl3/UX3M/dX731K901XVR91bvfkF1cc/VL3bddWbnF3d5uyQP/f56o3Pqj7jqchk9+uqt7useuerq3f8c/VxA6t/95fqHsOr1zq5evfrq698p7rtudWXvVnId1GHrvu/du89kxre4ffcGhvb8gTffTcs8DvvxEHu/nIKuAB5wBkkxvKz7RI5FWT8qsCoUdP+EnM25f11itPhwC6fLKJcUGbuWCa0PcslqSButfTMNg8Y/8y/XJ9y9bI+6mbPglyi6NlwECOB/bt1a/gG2Nix4ZjcUkzdm1uDN8DOmTpPwtXF7OAU7JVXQqWHDy/NJv7AA+Fp5mziKnbeeZFJziZOgEbdUcyTsOCCpXkS0K57rCxWgeT+KMJ9MUV8IlVyimLnPAnsDWbfuXk28WlbvJNwxqfRIPpdayeoEja6KcGTOIGNumIme3+dgh9IBjxgc0BSOuEywKQIzhJ7mbnElEmA5YF0Ydlnn5VKlNvo0VFQymfpVEDpWUT+lqualZQik7yLItu6Zz+Y1TsJpTfAWm3c8A0wLhGzYWMVuDW5ld8A26CYTVyUycBce208huEr2OE/P5rP2cQh+ZlnQn78+AgC8vUbZy++ONwFHgCV2bZYMf6QQ+L1UObn6adj0FxBtEZsyolRpVyWUGDBe5BImw47LF5Q27x5NnHWnqkX7GtHysx1ZZ81k/bSEyedFAZ/2WXjcTwXzF/9mnNxcs0kksz5W888Mx5kcSj2KVaCQUzS9dDpp0cncXXBYtviM/OczA2PnHBC/HL3oA1KeAHyZPwvvLA0UWZ+uqZBpeMmNKTXVYzHob/VRG48BUXnnE/KksKDID+beRKa2Xbq1rjZ9p0Jla1vCt3WdPADunpE1yOOnFQwF/R0ioDA69xzAzxQlzMQnnJKEBMAi+qk84gXXTRStK3wSweBAZDYyYsnUIm4Tf75CSVJjjAA3HlnaRbHfD1cti1aRC9vUSyHKrfOnaMydEHFZJJQ52JDuFo5BT9Kz9fUOnWq23LvLofNdsbFZradujU2thUHCYS1Mkz06BFx64knlj55zq8bhK5AXB63zXFeFIlN8usGBHRlscI5NgEjfSmmK4/bjixmM7qtmERDESOnrpSj82Zc4fy9YqUcWqHD3nqrBF8dnw+X0ehrr8WQHPZ58MGoyfvFCud8h8w8x3nnOAcYHIgT3ey+xUTUv/hFoAR0kB1duuKK4Dhn7fBBgBrj/Z12ioDR35y6VCWXW6403QauZGaOLBZ/BU13Tds1TpZC1cnTTzQqf/hYZJGAL7VftVimiMrJX+bundlAwSJZAHVHLsk584W3/mouAhAvN1mpkttvsmw705VyQKhr13jJX8vrhQEDGo7b5tcN0o89NoZccYqz0K49sWqDrxtyc4AjBg4MjgYwZwGGZHnc9shiDrC3Z7ZSju5GahwIRee4re6mcSwBhFMcZ7Etu25TpTnOAdZx8LfrbRTGXoVbtQpdAwm8j9/dMmrjryR6uREU56CDAkXOsgcKpaEUhPGwaxky7uWPxfopKc88dO8e8qokEzhUYenEaJkc8Kn7hdiVV457JO9CCks+P76npznXM+MkUd1ot6prRjdIHsXvv39V+8uqbptb4DWz7U+ebYV+quHWUOQCC8TbmvPNN+VnP4thOH+Bg++ZL+3yUqHn5z+ftjtVPvj//r/SX1e5nGcEf5g0U+yyzTznnz8S/c1Xg7MsipHvEWeGxKBQfJB/XUJG3dQHmh3kC6ESs7aZvz5qZtsmxba9v/z2nQ8Cw3QHAAAjoQgwDqTkAXjk0HCiJdMTWnnsN3HrON+Uz/fu7QmtRGCmwF6+BZ+naDpvOgvN/FctFprkKMjf39QpxyBNjxzkqby8qE/Vz/at6vNlwzv8nltjY1shjJ4Wk0IqrKj9LruU3vlfY42IGjCdJrvk/2/v3qOlqO58gf85ycxdWTdmLWO880jijM5MHo5Xo+bmThzNBKOJUUfmiDOTqGiuwdeIICoqCjRCBFGeAkIEQeSAiCjP5iEEFXm0CIhR3shbQY4ICMKB+9m1D51jYYBDN0d7pX6rVq/qXb/927v2/v6+v9+u7qrqGqKc7E/UghUBCmPGiAflDg0aFH71EtxkbaYZR4PRz5O3N1o+Y3BVNEFZi4gbvaJIqy3EvWlTID5QdvSC5KWh7AAi7pbxmUXAlQOaUWnpzcmLG2KO+UTyUl798VWH+yfvJVNXOnmYKwlRVh+8c1e7fMCqRqZQvJIQ/03BkfgGUWgbMCC0GB3DKSxfHi6bKE+9BdLpcHUO3759WHOJAYZURUFo0aLgdUi5piZcomHwsoPPt6UPxE/We74tteKVB/rOjimD0Dp7vm2Q8Ayw/zsmUOrgwQEGxs0CJT6REm6NjEm07LDONRFWykYeDqGO58OYRAHObZI+4IFV3GTihG2LEuOPUKqTt6PCvzGHMeXs/yh5Z5IpEAilKQAApZqIz26WK8Bnq1YhD9A303f77SGRtJTRMQmEdQkIAQMPgiJph1SGsoZ89u27vsk1R37iYrySwKyYo2OgyI+kQQKJs45vafyLvwi9PZBcUig+35b+unV1z7fFnr///SeuJBDWKMQrCYaLBcMl7RA51Fq6NKRcui232LEjuDkcymENghP8m78J9jnsypXhZOXIOMQgAP+8eQH83OFP+koC1kNw4g/yOv30EMZRhjkDKYVIUCIprFm9iqWAIgbShxUlZkiIxr8oQF2ZlwwCYuifm7yqBF3iVqwR7dPEaCZPKxD5gx8E40jHYic+kyn++Cs5BXQGzzgjXKI9J3lXkgU7GrV4Z0RDUmBMDcdMsUkB4rXOQvz/jeB/eMnYttLZdtmebqcNCBCCMXmZ6QMPsJRaKuTn2POs5PVxZufyy0OCBjwOwQnYo06fpoxm1Mee9CEc9gRs1enHmyp/lLzhyQTxiIhkIGcB36FLdAalqnMNxjERfVBkhyOYUMbNvrljHyX95Cdh0x8lViedOgXA24H2889ff9r59zU/7HXbjG0PSqWxrSiNYtCodMDIWvLEvx9ZswPWPclbacHXyr1Hj5AF2KcvgJv5eEeZJNQ5SyohHoMgBdi1HLvqqjAZVljm+J/+KdiniYawNnhxgBdfDNNpSrCnWQQXHAqUSjAgg1oEWQmv5IWRmTMD9wHNN74R/nUAdyYeybKWzwfmtbiD9fgshcNcSYiSsW2ls+2a/d3u3x6y17lzA3K6dAlTz6uVGCLgBMj48AHjDC3Yp0PyJvPXXw/TAdhYCdrloapYxdM3/nhH/guQxr9P8iQauQJWAkv7VleOAjPvWLw4fLIDq2PGBBibUyst+KGPeU3TfyQvjpI7o2+YGT486EOvjT78DBwYUBefEAI/8+evv6rV4a4kZGxbTyqNbeN12OIzwHr3DpcF4jPAUCTyRUbxuu2QT74pp3jddsjBZ4DVv24LFwbCRJpOFsRt9occfFNO8botNtciWBTflKMtE8wBGFQF4p9KnkzGSPEZYNicxFsDuFm8uwGakVGT5O28h79uG2Xz5gAaBvE1roRRKZL+/yx56ihX4TmTJoWz0FWFNqRpox9/X74ieRGvcsCyQz/u25HgM8X5qfnKmopF+xYE1DTqa6xIn+M5hHzZdwo2anoVm3aUBZrxkaxF+6i59tj/tFjZbFu8u0GES123je8dwJ7PPRcWOiM/+aacDRvqrttyd/5PObIwfYi9+OCbcgR1JGLkv/zlcGGqz8E35cSgyztASJL764PPAPvzP//Dddv4DDBdgh+wvPvgM8CKdzeQqB/vbhDCgRzbHvG6bWRbZHr//QEqIKEhUAQJ+zAWwRbhBCSwKt/31XlFNMYqkn2LSwqGDhQj5K5MXt5KAfAU8iam4FBdJQ5FfYUR6j41zb5hlA34qq5a0TV0SS0NKfeprg4oof+zn40+68HRw3alz7CBcoxsu3jx4p49e06fPn3Xrl1jxowZNGjQwoULd+/ePXbs2IEDBy5ZsqT2k05VNrYt5rYYFjiEIwt5SdObb6ZzW5lpKre9OXnvyB/LbUV1zDtrVtABTfYPzW1vuy2kJIL8smVHm9ti0pjbjk5elFA/t5WzgBoO4lRHZNtt20LEVpcnYGpx5Y7kEWVArAOsOUfBnAJ/5qJ27r23riTq29SlL5PirjLlWFc5Bcp2ivp33RUyHbmqoOKUGRSrFPJhnaegOZ9Mxeo+mVJ4d/IXumJ/fJoUXdWW3srNS5DKZtuY2xrGEnNbykef28KMo5CZym3RdzG3PeWUT+S2mjv63LZQWP8frY/MtpZT0o6ILniI+AG5XPI6UZjRWycFRcojkJx71IwoBVfVFQKhEgo2Y8gImxFmUBof1WTfWHE3Bh1luUPyGEmJqnOPDbHMYOwJO/Efk2an2J8IWs0Z8M6dRzcb+Zm94XzNmjWjRo3q2rXrSy+99Pjjj8+fP3/o0KGTJk3q1KnT7NmzoXPHjk+8PKNsbHtu8nx7QDz55LBkOOmkuodsgREgWhPJWCUFFrAUbOjPihsuETTusxhReMIJoa5Y/cPknQ70EeJ3vhMSZ8zrqEPRfnzuUXzkEuOqx6u6p58eeNO6Iz4b6StfCcoWJickzyTTqJ34I77yP/uzQMq+ygh0Pl7DjU9vOvHEuivLPOHwgqdwvZSHi8ITIKsuWZaDWPpZKKFsIzBtWsjfJU3yXA4sPiN9AKIvnVGXvi5xFcq6Sv+ll+p+2aMQH0FCX6h36KqrggNzP/m4gRLkZf0sSFF5JjWm1AJKi4mFC4NvG2RhSX94uEAV37Wnq+BLTVslyGfCtnBbHrZdurvbib8J0w02X/pSAJVP+0pAKILHZ3xAV9yBFtA9NXmnvfU1XMWndkXYxGd3GWrlpg8sTUf8D4kqX0uebAfGdnwCuVmIcGXcmow+I9r6whfCEtB+tB/VtKs/EcnW+7boCI5SVshlfJ566vqvnnFUv5JhNOwPwHIgfBdvU5TloEupzPz5wU9Fdz0HPEcFGIEHZiANAvGgvEo0MhQI1GhIiQQJ1mbMCBFFAIBSZIocJUzoVQ40ZUrYgFZF0cI6WCRz7rNnB0aGz/uSV8CxM2JEINyzzgr0ymv0R0+AlpsbsdmzR5/zm8/yPwkLFizAtsOGDSsUCtu3b7fTvn37iRMnbtu2rX///ps2bYpqwCr5/eUvf/lufHZyiRL/2ok1xHajKYVU8p//GYjAAIGR4cNr/fqFBBYvIEcBnLKxw0fqxv+QSzkdEsMtT2JcNaxCN/D1Sl4mKqsFJjnaf/93OMQCZNM0u9JhoFQCCli+Z/IWdDkCDmrVKujbELdy2TSSApf4t1N0yVWiDvvWOAr1yhwf0Z+L123//eA7d4vXbXfuDH5VfJq4iG1AJDu6JzFZty4wJn0L0njd1ighRCdS/7ot6qQQr9v2Se5uiBeznnoqJCaSAoNggQbl8e4GWRU1puhL59nfnbxzV3K0dm3oDz8zBT6hX1c5j4G1dChBGp9t586d27t3bzhPH2i4hP8kNJsXPFksN3SgKwNAK0ps8AnVgGH1hoaQC9jgOFkYzAiNJghrxL9zmWib2XQIfsT+ceMCGxp8S+BHHw0XJcVLduQTIihTLAAkhmrZMhzCpyYdThylj4yAE6TPS24HUChJjG8FVUW8xLCM8DXhIRKlnFpvW7def8Ut991+pPeSpa7bGs+nnw5nZEmqXEPbt9fd3SAvgcnU3Q0ffRQa5e+IOF7nPeec9HXbW5LX7hbvbjAsxmrFigBjg6CiFM2Yx7sbtmypywO4kvydxHfuOkEZgxHWH+nX4OSdu0LOli2jz+sx+onPAdtKbCPbduzYcfz48ZFtN2/eHNX27t2rJJfLlSe3jf+3hRiOXf+6rYm0QIOJ+IZzg2VKDPTXk8fKYSIUHJ9KY8Tj/22/+926p9iIbFbHUG4iTXa8zrt8ecgUJkwIa66bk3c3nHFG3bse4rsYrkueKoCD6CMgXINQJk4M65d43bb+U2nkdPHXJBR/c713NxzNU2miZGz7WbCtJZpGBxxxdo5C/nDd9ksH391QvG4LCdAY/29bvG5bfHcDDF9W76k08f+2NkTskNFGfFgGiWMK62Lz+GzyOHx2midPpYnvbgDIn/607v+2+oCJECuQrE+eTvt88nLPeN1WlddeC/MFTj+r939bYJYYwgzLNx/5qTQZ26bkGNkWCuG+Xbt206ZN69mzJ+YdMmTI9OnT27ZtO2fOnEGDBu1EAfWkbFcSsC3ol5FtzbQpr8+2QBnZ1vorvme3yLY0n0zerBPZ1tTioHi3QpFt6dsQn/JRoz6dbaGtoWzLhXAfrwMyXbUIgvtLklehSN6t77gcCFL41+SV7OiSh5x9diiJ+nCjLn0ZU7xP9NxzgwMr/2HyFks7hs7nvyQv/tOcVZXe4ugmTUKOYwTitQWgZOf85NEqUZ99avGXYi3+XfKIa637lN3rarxCbUlYgjQ+2x4o43XbyLa4NbKtYTwatl206BNsC2PWbUOHhq3ItuYCFL/4xTq2RZdFtuUdRbYVViPbQp0+QHh9tkWd9dm2UCidbUUKMR0uqv5tb9V5m6rOXF512Z6q779T9c8bqn60peoHa6su2RXKz1pedcW+qn9cXHXxh1V/v7Dq0t3h6E8+qDpnddW/bK664L2qprVVP9xYdfaqUHjG21Xnv1v1rTfCVnWg6v+sDTbpU/7x+1VNtgXj564J+t9bWfXz3cHad94MFb85r+qsFUHnH9+o6w8Fn+qyo2lmaap41sqkP4tC02csrTp1Af2fnrVx/PPH/neaKMfItq+88kqrVq3atGkza9aswYMHd+jQ4bXXXqutra2urm7fvv3ixYv3799fX79sbMu3RTPBp/hLrqGyNve5cGFwb/yLOMaMCYt3zMj5wc6yiM9LCtAHbqKPBynncoEm+BJ6ZQEK6ccHfsd70sRJmMbILOAmmoDIJbAMU+CIX/A+fXDnM5LE+HsUVlIukYn5CH63lEP3AiZ8s8Y+JCoUMwA63jZ+UKwGNhyTbNq0qZHJqDGlstl21b5u1/8+oPGUU0LKZ32A+6SoLVqEDZ1BNeQ0axawjTefSB4lKmlAxKAiWRNcYQzwgNCG78AJfsTCKVPCtYjbbw+mYKlTp5B/sAPVAj8EWvUDpIAHhNYfsA2BHCH+RUGW0LlzgCVA+qoKTudcuqcKjzvvvGCE7/AjhSxjXlW6dl3fouN9rUv9sf5PRI6RbRsqZWNbEwwKIOITcGHIPtqFTuiBv3hJq7jdemu4xmQHsBAxOMbyeHNtUZ8ROIYhNlMWNMH4bcmrHO68M/CpWpYt8Z8AwE3BZ6pW7OGhG00MDr6gLDWYPLnuSbJS13pixdCvKK1a9bvoon4XXng025OXXbZREiS1GZ28tc5W3D9M4dHopPTr78yZE54Mcvylstl2xcfdzn3mD/AQiXErvIFl27YBqCnM+BoLozICjUi+oR7yi45gbUT5ULyBtIyaGuDF+27rV4fkQ13mMBsjWvEpO4nQnTx5/fAZ993xUfpsE6nZWjstv6+Inc/h9tzofRtKfmrt0Uulse3FF4e8AFbEZ59CtFAcUWu9c+GFgTQt2C11e/QI0V7kl0TEu3UBhf7w4WHJZqH9xhshubB2FsYtpa2qEKgUAw9KZinLNeJrIjmDjECKwSZ0/q/kKeZSDCs4sLNqljVb69kYlx0o/Ku/CrlJ/P+A/px8ckhGHGJN8oKs2Tn4Q+KhsnPnzpqi3HVXzXnn1fTqVdO0ac1jj9U0aVJzzz01d9xR89d/XdO9e82tt9YMG1Zz5pk17drV3H33B9/73l4Dwri83ppUf4xJ69Yh2f/2t0OipBvWktJ/kcPK0Upfmk8/5tqGRQYkMtH/1rfC331k4n/7tyGdj/rSqKhPWbii79MqtaYmfQ7HQSqbbVfVdvt/bwfOgklLIoN29dVhOsRvE2Sov/71ACTBGH7Mi7mz6JFvGnzDbh5hSeCHZztwyxEg0FdOEVNOHmHS//Iv68BpviJTDx0arlaB9/nnh2zXnELyaaeFq2ocx1wDqklnR6NWY5JoEDXRkmvIt/6zjLOe066h4FZ85KCsXx+KP1UK82sfPXvYxJOvy7d/Jf/wovyVA/PD38tf81R+6Mb8rc/n/31A/om1+c7z84+8kf9hx/wLu/P3zsxf2C0/Ymv+gZfy3Rflr/pt/ukt+auH5W99If/Ay/nrRuQHr893npd/dEn+n9uHuq0nBWv0280KRui3mhT2KduufSo/ZEP+wXn5y/vk+y7L/3Zt/o58sBb0fxeUc3O6PFj77DMZ2/4xiXc3dE/emwumSG3NmrrrtgsWBKBgjR07AtRmzAh/Ulm1KpCp5Es6CTdRX5C30l+4MOife274MwNwv/hiuBqg4mOPhaS1W/LgGFK8bout4BsR17+7AR2feGL41yRqtq2p9+6GQ99wDtmTJgVr0Xi8XnZEib8OL1tW9yvZrw+5l4z9S+rdS7Z5c/Dk+CvZmkOeuBjvJeNFsqG1awNdHnovWdRP3UuGCNatC1dy4r1k8V/3Olb8lez4S2Wz7afe3TAkeUCHQmj86lfDb1Pjx4e1jkPmDqoFM4Pv68UH33BuIp5IHtxuos27r/GOgJUr667bfvnLIaw6VHx3AzAwdc89n7i74ZvfDE5Ef+vWABIpNjsaxQhSmfirGhZ+4eAbzgXd+IZzUeGG5AbFRA7DtuFXstM77DnhqyGEOynIr38vmRz5zTeD03E94YTbShG4JIvFX8noC/OURyRPXKx/L5m6xrD4K1nxXjIIH5E8obH+vWRz5wZn4dqs0Y/3kj37LOeOP5I1jmRs+7lnWzBqkbzyT9rCFZs2DY7BGU45JWQfuVx4CqfcPN5eIakH69tuCwmL9JO+M9LWi8mDO7ku/VdeCZenLQKwNtqlbzDp80NnpIdR3+hxEgvYk04Kf8Llq/SNf9SnbCFpceDTRBzxOQ/lkIxtK4ttUVzva+bOOennhS75Qt9XC1VdCuM3Fq7tVRi7ttByaKHpg4VnVhYemVkYMK9w7o2Fl3cVOowrXNC6MOndwoOTC4/NKTR7qDBhU+GanoXbhxU6Ty5c37cwelWiP79wdotQ9+5RhWt7B/1OEwv95xWu6lq4a2QhNyEo25r3LoxZU+g+o/Dz9oUn3yiMWlFo+0ywNnFzITe+0Kxrodv0R7vXjhh+7Dc3NlQqjW2t9PFmXDrBk0WTlVTLlgEQIqF0DxmBF2QoQXnWaNK3XydPH0cl9C2C7Jx9dsB3/Mc4LqNvIP7rv8JO/DX/oosCheEmdVXBdEhHQygMdcKXLE/2p9bXvhYIlCkbTfR97bWBkUH528nTyuWYX/hCWME5hJG///262xY3bkyf3aeK4MHlAF2QEJN5VLwdU9+eTN4uzHnwnUIBwA5aF+Edsvqjz204tghOn29HfYEHOzsa9Xla1JdEKIz6KCDqS2N9jfqsRX3Kduj7lFN/9OlX7sorFc22q1bWXnrRnty/vphr9X7u74fmLnk19w/Dck0X5M6fGgpv3pT7H51zN27I/fLt3DXLc//wVK71ttzpI3PNFufu/Shonjok92+v5ZqvyF06N+yc1Ct314fBjq/fGZH7xsBg9uKXcxdMy30xl7t8fu60obk2H+T+97PBfsstwdR5U0Kts8fmqhbmrl+VO+Hh3PfH5S6Ynmu7K/et4aE/7LSuyZ35bO67I3M/mZW7e0fu7wbnfvFW7swxuV+tzn2le67p67lbN+f+Z7fc98ZCU9yEZi74qYKI5QMjOi6t7l9TPejD6u7rq4d9XN1jY/XQPdW936vuvq76yd3Vj38QDnVeWf10bXW/96sfWh10+m8LhY8k+o9uDMpKemw6qL8j6A/ZXd13S7BWp78j6Cvpty00Z3Mo6j+8tvqJndVDPvqDPh3KAz747eP7lizcm+73cZNKY1urV5slcHxchR1fWd6yJcxtLFSyJnmZ3erkzR9K7Me3jER9nzZO6zNqxk8KdqJa3CjjRMtknw6J9hIBO9FsrKXRDQdfsRN3YtP1++Ornfj1neT9Ood/O0592b8//Aa1e/fnd2uUn8gOVDjb7t17YN3a/RvW7t2wfv+GNR9vWLsvfK7bF0pi4erd4XNdbdgc2rB/wzsfh32Q9zUqr68NFe2s3hMUop13kqPqrk2ssbMuMR4sHDROJxzaE0rYZCfuK2Q/HE3sxCrxa7RPWUnUDx3YX1cxccS4/bHbziHXXB2Kl8/VVnt0XlgWqTS2zeRPWCqabTPJJGPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSiJWPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSiJWPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSiJWPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSiJWPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSiJWPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSiJWPbTCpGMrbNpKIlY9tMKkYyts2koiVj20wqRjK2zaSipcxsW1tb+84777z11ls7d+6sX56xbSalS8a2mVS0lJltt27d2rFjx65du06dOhXzFsszts2kdMnYNpOKljKz7cyZM/v167do0aK+fft++OGHsRDtduvWbfPmzbWZZFKCWDNNmjSpMdl2//79cDtw4MB0VzLJpOFSZrYdM2ZMdXX1qlWrevTosW3bNiXbt2+fN2/eTTfd9Pzzz888njJixIh0UVll+vTpY8eOTZeWT2bMmDFx4sQpU6akD5RVXnjhhWnTpqVLyyrDhw9PF5VJRo4cqf+NybYrVqyA2zvvvDPdlbLKhETSpWUVa81x48alS8sqvON4QwsA0kVlFT7+3HPPpUvLJzfeeGMaYfWkwWw7e/bsXr16odf+/fvv2LFDic8lS5a0bNnSfM8/nnLppZemi8oq+t+2bdt0aflkzpw5lgWjRo1KHyirdO7c2WI8XVpW+fGPf5wuKpPA1Zo1a2rrXaE63rJ27VoM0q5du3RXyiq9EkmXllWkQV26dEmXllWMEkJPl5ZVmjVrli4qq8h1WrdunS4tnxQKhTTC6kmD2bampqZNmzbNmzdH5PW9ohGu2/7iF79IF5VV3n333T59+qRLyycff/wxQn/jjTfSB8oqTz311IYNG9KlZZXLL788XVTJ0gjXbccnki4tq7z99ttPP/10urSsMmDAAIvadGlZpUWLFumissqmTZswVbq0saTBbPvHZNGiRR999FG6tKzy4osvpovKKvovSU+Xlk8Ep3Xr1r3//vvpA2WVZcuWpf4uUnbJ5/PpokoW8/7WW2+lS8sq7ySSLi2rSIOWL1+eLi2rvPnmm9u3b0+XllVeeumldFFZZdeuXZgqXdpYUh623bt378qVKxcuXGjK08eOVfbt2ydcL1682M6HH34oJfQVYZlvhUuXLtVouk4DhbXYbckga79PJFIV+zzQ3KTrNFD27Nmjt0hct52FmdZW8SxE2v3796frNFw++OADvXUK2NxA7d6926Bxb6cWr/aUIsxy49dff93gGBCzoAn27ccRk7On61SUmBdzsXr16jJewTAjBn/r1q3m16SY9xhlTYq27Jc+73zNRIAWxGpIExs3bmRWuSY2b96crtBAYUpvzTsfgSifEVqmO3pNuS6va2X9+vXRIyBNu0bP6fBBMEtrN1CMxoIFC1jbtm2b8XE6HFATPu1zwDJO+hGlPGy7YsWKxx9//Nlnn7WWKZfvyTgYvOuuuwzT9OnThw8f/sgjjxi4CRMmKLf0M9/pOg0UTUybNu2FF1548sknNWGJof+vvvqqKe/YseOQIUM0V6JX8ARNWN1PmjRp0KBB1mKaAKznn3++X79+Bq30ZMGAT548+eabb7YiHjhwYO/evadMmcLZhg4dqrlnnnkmXaGBYvzZGTt2rLzjtddecy4PPfTQq4k4Fy3Onj07XadyxPwCgJMyF7wxffhYxYq7S5cuv/vd79577z2WwXXMmDE4pVu3bmZk3LhxaCtdpyGCI6DUvEOpqYkToRUYGD9+vP3S18tGBknBUp8+fTTEZt++fWfMmMHZQQt6uWG6TsNFnOjcuTPjGuLjxkf8njhxolF67LHHZNPpCg0UoGXTslg68sQTTzDL2Z1C165d+eCoUaN0IF3nuEkZ2Fb8mTp1KsgClvkAqbTGMYnJBh1sKwrhKalHoVC49957DZaSefPmjRgxosTQpwmgl2iA6fXXXy+0Cnfmo0OHDubbPMFxicFDE1In3uUULrjggnfffRdzSXZatWolpLdp02bt2rXpOg0UHAH6N910Ew/Hg5B03XXXiRPGB5KaN29eYvQ2CxH3rAEoktXQjTfeCKm8Ef/i99LXGZ+VmB1RXLKGDVFJ+vCxCtggIyF8/vz5ZkcO1atXr+rq6sGDB5txFFm6kxtzrcyZM6djIqCFQXzFs5ylRYsWpf+OognowoYPPPCAzi9btuyOO+6YNWsW7xCDL7roohJzEciUhTz44IM9evRAHfwin8936tQJm9sHLc6ertNAYbB///4wjD2EPVlz5HGF9kWp430Rqb6UgW1NiXAqekjTcApvT2scq1gFt23b1rjDK3TaueGGG4YNG2bfokY+UvpaxnwjbtPwq1/9ilnLZCRlP/IIhJV+MRqDt27dmqkLL7yQe5jmNWvWIERnJ3iUONkMinPGpGXLlniwZ8+eMHTxxRfbhyrRCNuWeD1EPw01H7j77rtlB3jcoDVt2hRSuZ/+a7T0UfqshB8++uijOIVbGsn04WMVQyTEYlsxm1PwEYTCyWNSgkTKkkeba4mn2UFYiE8SJ2DkcjlUKE0pPTHkXwYHekVxgZZZcLJWc15OEJ5LnHdRIVqGLijigBast9xyCzzLgZYuXSrnLTGjWrJkiVDUvn17gyOpimQlnRo9ejS/4CNyrBJjxtFLGdjWuJuA5557DowMWVlgFCWyLYOGST4lqBo42YGSl19+eeTIkSXOxIGECvXZ4uU3v/kNhvLVNECAT34CByXmtqqbYAwltZELyGflttJ/WS2qwl8l/ngCMRIli9ZLLrmED4MUH5PXQC02FJ/slzhK5ndPIldffbVRmj17NoDefvvtclutSKaQSOXmtjAmt7UqdzoAlj58rFJkWzE7Zm1gJtuVN8htYbj03Bb3CX6AyqCsVhM+pdLdu3cHLYuPEnNbpwC9PnkfwpWFSHEg1ighLBi+4oorSlw26S1EtWvXTvIhfZaFWE06i7h+5ePDhw8vkQrhFjhNhNWkgBfzWfFPFJEdso/T03WOm5SBbQ8kMYrDCyBiRYn0VBQ8IuVs0qSJ6M0s4/fcc0+8piN6C4b203UaKFaRAqlM1ujPnDkTA1o04VwMiE0wO1cpcbLjFSKmJkyYIAe58847hWvtynHsO7USf8XSPQPF3+6//34OAEPOQoJjXWmBrwkBI12ngSLLkGsYGVPAsomWR+PZuXPnSqm0a5TSdSpHDKCpcRbxZsj04WMViyRxDrSmTp1qUkAXhk0K3EqspCalX7flCFdeeaVJ0X/0JJlF6yjYjOPEhx9+uETo7ty5U0Jjus2ydAotghMGlPcgdDCbPHlyuk4DBVfAP5sWfE4H4d53331cxhnFhLTEf1kYgUKhoOesCX7cjU3pDn+RUZkImUq8RatxpDxsa+755KZNm0rEUH1h0ypPxBaiTTxP8DVeaY37JcbVA0kTTMmUWRMD30skRostiZQeOZjVBDeL6aEhQrXFs4j76ToNFydSU1PjU+hmVjDfn/ywq7nST0FqLBEzSnrLsmFxOuyzHEesxNz5M5c4RwawLHMRxfwaMfEPm8RJiYvuuF/itZ0oLGvCFLPJQSK0DiS//WqixCh+IAEVJmLWp3n3GaFlumN5uebd+OttnIX4b43oHVBXuo8bal2Nvmx8ig7oM0K6jJN+RCkP22aSSSaZZHJ4ydg2k0wyyaQxJGPbTDLJJJPGkIxtM8kkk0waQzK2zSSTTDJpDMnYNpNMMsmkMSRj20wyySSTxpCMbTPJJJNMGkP+PyCHP4NkC0EtAAAAAElFTkSuQmCC)"
      ],
      "metadata": {
        "id": "2-tvMGAe-MZh"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "text_list[1][:1000]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "outputId": "f79fddf8-9605-4133-8ef8-c6951d134ef4",
        "id": "aBdQG37B14rL"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Query Article Response Topic(s)\\n“Are IL-6 inhibitors key to SARS-CoV-2 and COVID- “...monoclonal antibody against IL-6, is being tested in a Clinical Treatment\\nCOVID-19?” 19: is interleukin-6 (IL-6) clinical trial against COVID-19 (Sarilimumab COVID-\\nthe ’culprit lesion’ of ARDS 19). Another drug that showed potential inhibition of IL-\\nonset? 6 related JAK/STAT pathway is glatiramer acetate which\\nshowed potential to downregulate both IL-17 and IL-6”\\n“When is the salivary vi- Elective, Non-urgent Pro- “Patients with COVID-19 have demonstrated high viral Clinical Treatment\\nral load highest for COVID- cedures and Aesthetic loads in the upper respiratory tract soon after their infec-\\n19?” Surgery in the Wake of tion, with the highest load assumed to be the day before\\nSARS–COVID-19 [...] symptoms appear.”\\n“Is convalescent plasma ther- COVID-19 convalescent “...a passive immunotherapy, has been used as a possi- Clinical Treatment,\\napy a precursor to vaccine?” plasma transfusion ble therapeut'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 44
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Checking page 2 of the extracted text from *outside* table bounding boxes, we can see the text that is actually inside the table. Further experimentation is required with pdfplumber's various table extraction strategies and methods but it is unlikely that there will be a one-size-fits-all approach for all formats of tables in PDFs.\n",
        "\n",
        "Check the first 1000 characters on the first page to see if the text has been extracted correctly."
      ],
      "metadata": {
        "id": "O-ISKTUhEkif"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "text_list[0][:1000]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "outputId": "d34d4314-e324-410d-9b02-d3d036810dd7",
        "id": "icIvjMec14rL"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'AWS CORD-19 Search: A Neural Search Engine for COVID-19\\nLiterature\\nParminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\\nBen Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\\nRamesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\\nAmazon Web Services AI\\nparmib, liuall, arumae, nimpourd, suyogd, snivelyb,\\nmonamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\\nzhiheng, bxiang, tahak @amazon.com\\nAbstract the machine learning community with biomedical\\ndomain experts and policy makers in a race to iden-\\nCoronavirus disease (COVID-19) has been de-\\ntify effective treatments and management policies\\nclared as a pandemic by WHO with thousands\\nfor COVID-19. In accordance with this initiative\\nof cases being reported each day. Numerous\\nscientific articles are being published on the our goal is to provide a scalable solution aimed at\\ndisease raising the need for a service which aiding COVID-19 researchers and policy makers in\\ncan organize, and que'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 43
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "It appears that the text has not been extracted according to the two-column layout which we achieved earlier with pdfminer. Instead it has been extracted row-wise across the entire page instead of row-columnwise."
      ],
      "metadata": {
        "id": "2EBUQ4kXF1CZ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "print(text_list[0][:1000])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YXWl_zz8EI9g",
        "outputId": "73b18850-e438-4ac2-9acb-e3f2791a8f0c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "AWS CORD-19 Search: A Neural Search Engine for COVID-19\n",
            "Literature\n",
            "Parminder Bhatia, Lan Liu, Kristjan Arumae, Nima Pourdamghani, Suyog Deshpande,\n",
            "Ben Snively, Mona Mona, Colby Wise, George Price, Shyam Ramaswamy, Xiaofei Ma,\n",
            "Ramesh Nallapati, Zhiheng Huang, Bing Xiang, Taha Kass-Hout\n",
            "Amazon Web Services AI\n",
            "parmib, liuall, arumae, nimpourd, suyogd, snivelyb,\n",
            "monamo, colbywi, gwprice, shyar, xiaofeim, rnallapa,\n",
            "zhiheng, bxiang, tahak @amazon.com\n",
            "Abstract the machine learning community with biomedical\n",
            "domain experts and policy makers in a race to iden-\n",
            "Coronavirus disease (COVID-19) has been de-\n",
            "tify effective treatments and management policies\n",
            "clared as a pandemic by WHO with thousands\n",
            "for COVID-19. In accordance with this initiative\n",
            "of cases being reported each day. Numerous\n",
            "scientific articles are being published on the our goal is to provide a scalable solution aimed at\n",
            "disease raising the need for a service which aiding COVID-19 researchers and policy makers in\n",
            "can organize, and que\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We will try another example using a seven page PDF with three tables on page 6."
      ],
      "metadata": {
        "id": "XuqVvn7_G2op"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pdf_url = 'http://export.arxiv.org/pdf/2212.03911'\n",
        "tables_list, text_list = get_text(pdf_url)"
      ],
      "metadata": {
        "id": "l3jhUABn14rL"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(text_list)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1ad44f71-54c3-46a6-e2d8-fac14e50fb95",
        "id": "tsyD08SN14rL"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "7"
            ]
          },
          "metadata": {},
          "execution_count": 46
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "tables_list"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dBypDCZz4UKB",
        "outputId": "184ed2fb-9a86-478c-99d2-0407b656d0aa"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['',\n",
              " '',\n",
              " '',\n",
              " '',\n",
              " '',\n",
              " 'Scoring function Drugs Proposed\\nTransE L2 Ribavirin\\nMethylprednisolone\\nColchicine\\nTransE L1 Ribavirin\\nMethylprednisolone\\nColchicine\\nChloroquine\\nRotateE Colchicine\\nDeferoxamine\\nMavrilimumab\\nRESCAL Colchicine\\nDeferoxamine\\nMavrilimumab\\nDistMult Mavrilimumab\\nBevacizumab\\nThalidomide\\nComplEx Ribavirin\\nChloroquine\\nTetrandrine\\nModel Mean Rank Mean Hits @ N =3\\nReciprocal\\nRank\\nTransE 66.3 0.407 0.477 Model based on encoder Drugs Proposed\\nRotateE 100 0.45 0.5 CompGCN with ConvE Azithromycin\\nscoring function Ribavirin\\nRESCAL 57 0.66 0.7\\nTransformer Dexamethasone\\nDistMult 76.8 0.48 0.52 Chloroquine\\nHydroxychloroquine\\nBevacizumab\\nComplEx 83 0.65 0.7',\n",
              " '']"
            ]
          },
          "metadata": {},
          "execution_count": 65
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Although the code has identified a table this time, it is only the largest table on the page instead of all three.\n",
        "\n",
        "We will try one more example using a 33 page PDF with three tables, two of which extend over multiple pages."
      ],
      "metadata": {
        "id": "fFcFDBy7HII5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pdf_url = 'http://export.arxiv.org/pdf/2202.05145v1'\n",
        "tables_list, text_list = get_text(pdf_url)"
      ],
      "metadata": {
        "id": "yOhcLUFV56Uz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(text_list)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a9455766-ef96-418a-ef4f-0af1aa92f451",
        "id": "OGprefYA6Q6u"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "33"
            ]
          },
          "metadata": {},
          "execution_count": 70
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The first table, an authors table, appears on page 1 so, conversely, viewing the non-table text for that page should only reveal the article title and 'Authors:'"
      ],
      "metadata": {
        "id": "rZkJZDiaIQL8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "text_list[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "outputId": "235bd93e-11f1-479c-a9ac-41305ca396e0",
        "id": "I16guhuy6Q67"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Deep learning for drug repurposing: methods, databases, and\\napplications\\nAuthors:\\n1'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 72
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "This appears as expected so we will look at the table text that has been extracted."
      ],
      "metadata": {
        "id": "k55cxVIrIyz5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "print(tables_list[0])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "fb37a84c-f808-4656-dd10-0ffe4d06db2f",
        "id": "IepeA0x66Q68"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Xiaoqin Pan\n",
            "ORCID ID: 0000-0002-2083-338X\n",
            "Affiliation: School of Computer Science and Engineering, Hunan University\n",
            "Email: pxq123@hnu.edu.cn\n",
            "Xuan Lin\n",
            "ORCID ID: 0000-0001-7216-1821\n",
            "Affiliation: School of Computer Science, Xiangtan University\n",
            "Email: jack_lin@xtu.edu.cn\n",
            "Dongsheng Cao\n",
            "ORCID ID: 0000-0003-3604-3785\n",
            "Affiliation: Xiangya School of Pharmaceutical Sciences, Central South University\n",
            "Email: oriental-cds@163.com\n",
            "Xiangxiang Zeng\n",
            "ORCID ID: 0000-0003-1081-7658\n",
            "Affiliation: School of Computer Science and Engineering, Hunan University\n",
            "Email: xzeng@hnu.edu.cn\n",
            "Philip S. Yu\n",
            "ORCID ID: 0000-0002-3491-5968\n",
            "Affiliation: Department of Computer Science, University of Illinois at Chicago\n",
            "Email: psyu@uic.edu\n",
            "Lifang He\n",
            "ORCID ID: 0000-0001-7810-9071\n",
            "Affiliation: Department of Computer Science and Engineering, Lehigh University\n",
            "Email: lih319@lehigh.edu\n",
            "Ruth Nussinov\n",
            "ORCID ID: 0000-0002-8115-6415\n",
            "Affiliation: Computational Structural Biology Section, Basic Science Program, Frederick\n",
            "National Laboratory for Cancer Research, National Cancer Institute at Frederick,\n",
            "Frederick, MD 21702, USA\n",
            "Department of Human Molecular Genetics and Biochemistry, Sackler School of Medicine,\n",
            "Tel Aviv University, Tel Aviv 69978, Israel\n",
            "Email: nussinor@mail.nih.gov\n",
            "Feixiong Cheng\n",
            "ORCID ID: 0000-0002-1736-2847\n",
            "Affiliation: Genomic Medicine Institute, Lerner Research Institute, Cleveland Clinic,\n",
            "Cleveland, OH 44195, USA\n",
            "Department of Molecular Medicine, Cleveland Clinic Lerner College of Medicine, Case\n",
            "Western Reserve University, Cleveland, OH 44195, USA.\n",
            "Case Comprehensive Cancer Center, Case Western Reserve University School of\n",
            "Medicine, Cleveland, OH 44106, USA\n",
            "Email: chengf@ccf.org\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "This appears to have worked and printing out all of the tables below shows that the three have been extracted albeit requiring further formatting work."
      ],
      "metadata": {
        "id": "-PLWR7oOJOr_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "for table in tables_list:\n",
        "    print(table)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "H0o4L2Q_6tyL",
        "outputId": "91eafeda-f2a7-4af9-ea77-b0ee64fd0b5e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Xiaoqin Pan\n",
            "ORCID ID: 0000-0002-2083-338X\n",
            "Affiliation: School of Computer Science and Engineering, Hunan University\n",
            "Email: pxq123@hnu.edu.cn\n",
            "Xuan Lin\n",
            "ORCID ID: 0000-0001-7216-1821\n",
            "Affiliation: School of Computer Science, Xiangtan University\n",
            "Email: jack_lin@xtu.edu.cn\n",
            "Dongsheng Cao\n",
            "ORCID ID: 0000-0003-3604-3785\n",
            "Affiliation: Xiangya School of Pharmaceutical Sciences, Central South University\n",
            "Email: oriental-cds@163.com\n",
            "Xiangxiang Zeng\n",
            "ORCID ID: 0000-0003-1081-7658\n",
            "Affiliation: School of Computer Science and Engineering, Hunan University\n",
            "Email: xzeng@hnu.edu.cn\n",
            "Philip S. Yu\n",
            "ORCID ID: 0000-0002-3491-5968\n",
            "Affiliation: Department of Computer Science, University of Illinois at Chicago\n",
            "Email: psyu@uic.edu\n",
            "Lifang He\n",
            "ORCID ID: 0000-0001-7810-9071\n",
            "Affiliation: Department of Computer Science and Engineering, Lehigh University\n",
            "Email: lih319@lehigh.edu\n",
            "Ruth Nussinov\n",
            "ORCID ID: 0000-0002-8115-6415\n",
            "Affiliation: Computational Structural Biology Section, Basic Science Program, Frederick\n",
            "National Laboratory for Cancer Research, National Cancer Institute at Frederick,\n",
            "Frederick, MD 21702, USA\n",
            "Department of Human Molecular Genetics and Biochemistry, Sackler School of Medicine,\n",
            "Tel Aviv University, Tel Aviv 69978, Israel\n",
            "Email: nussinor@mail.nih.gov\n",
            "Feixiong Cheng\n",
            "ORCID ID: 0000-0002-1736-2847\n",
            "Affiliation: Genomic Medicine Institute, Lerner Research Institute, Cleveland Clinic,\n",
            "Cleveland, OH 44195, USA\n",
            "Department of Molecular Medicine, Cleveland Clinic Lerner College of Medicine, Case\n",
            "Western Reserve University, Cleveland, OH 44195, USA.\n",
            "Case Comprehensive Cancer Center, Case Western Reserve University School of\n",
            "Medicine, Cleveland, OH 44106, USA\n",
            "Email: chengf@ccf.org\n",
            "\n",
            "\n",
            "\n",
            "Database Describe URL Ref API\n",
            "BindingDB A public database of protein-ligand http://www.bindingd [30] √\n",
            "binding affinities. b.org/bind\n",
            "CCLE Cancer Cell Line Encyclopedia (CCLE) https://portals.broadi [31]\n",
            "is a large cancer cell line collection that nstitute.org/ccle\n",
            "broadly captures the genomic diversity\n",
            "of human cancers and provides\n",
            "valuable insight into anti-cancer drug\n",
            "responses.\n",
            "CellMinerCDB An interactive web application that https://discover.nci. [32]\n",
            "simplifies the access and exploration of nih.gov/cellminercd\n",
            "cancer cell line pharmacogenomic data b/\n",
            "across different sources.\n",
            "ChEMBL A manually curated database of https://www.ebi.ac.u [33] √\n",
            "bioactive molecules with drug-like k/chembl/\n",
            "properties. It brings together chemical,\n",
            "bioactivity and genomic data to aid the\n",
            "translation of genomic information into\n",
            "effective new drugs.\n",
            "ChemDB It provides chemical structures and http://cdb.ics.uci.ed [34]\n",
            "molecular properties. ChemDB also u/\n",
            "predicts 3D structures of molecules.\n",
            "ChemicalCheck It provides processed, harmonized and https://chemicalchec [35] √\n",
            "er integrated bioactivity data. ker.org/\n",
            "CGI Cancer Genome Interpreter (CGI) https://www.cancerg [36]\n",
            "supports the identification of tumor enomeinterpreter.or\n",
            "alterations that drive the disease and g/\n",
            "flag those that may be therapeutically\n",
            "actionable.\n",
            "Comparative Toxicogenomics [37]\n",
            "CTD Database (CTD) provides manually\n",
            "(Comparative curated information about chemical-\n",
            "http://ctdbase.org/\n",
            "Toxicogenomic gene or protein interactions, chemical-\n",
            "s Database) disease and gene-disease\n",
            "relationships.\n",
            "DGIdb http://www.dgidb.org [38] √\n",
            "Drug–target interactions mined from >\n",
            "/\n",
            "30 trusted sources, including\n",
            "DrugBank, PharmGKB, Chembl, Drug\n",
            "Target Commons, Therapeutic Target\n",
            "Database.\n",
            "It is a discovery platform containing [39] √\n",
            "publicly available collections of genes http://www.\n",
            "DisGeNET\n",
            "and variants associated with human disgenet.org/\n",
            "diseases.\n",
            "DrugBank It combines drug data (i.e., chemical, http://www.drugbank [28] √\n",
            "pharmacological and pharmaceutical) .ca\n",
            "information with drug target information\n",
            "(i.e., sequence, structure and pathway).\n",
            "DrugCentral It provides information on active http://drugcentral.or [40] √\n",
            "chemical entities and drug mode of g/\n",
            "action.\n",
            "DTC Drug Target Commons (DTC) manually http://drugtargetcom [41] √\n",
            "curates bioactivity data along with mons.fimm.fi/\n",
            "protein classification into superfamilies,\n",
            "clinical phase and adverse effects as\n",
            "well as disease indications.\n",
            "DTP Drug Target Profiler (DTP) contains http://drugtargetprofi [42]\n",
            "drug target bioactivity data and ler.fimm.fi/\n",
            "implements network visualizations.\n",
            "DTP also contains cell-based response\n",
            "profiles of the drugs and their clinical\n",
            "phase information.\n",
            "GeneCards Automatically integrates gene-centric https://www.genecar [43]\n",
            "data from 150 web sources, including ds.org/\n",
            "genomic, transcriptomic, proteomic,\n",
            "genetic, clinical and functional\n",
            "information.\n",
            "GLIDA It contains drug-target interactions for http://pharminfo.pha [44]\n",
            "G-protein-coupled receptors (GPCRs). rm.kyoto-\n",
            "u.ac.jp/services/glid\n",
            "a/\n",
            "GtopDB It contains quantitative bioactivity data http://www.guidetop [45] √\n",
            "for approved drugs and investigational harmacology.org/\n",
            "compounds.\n",
            "KEGG It is a knowledge base for systematic http://www.genome.j [27] √\n",
            "analysis of gene functions, linking p/kegg\n",
            "genomic information with higher order\n",
            "functional information.\n",
            "LINCS It contains details about the drug http://www.lincsproj [46] √\n",
            "assays, cell types, and perturbagens ect.org/LINCS/\n",
            "that are currently part of the library, as\n",
            "well as software that can be used for\n",
            "analyzing the data.\n",
            "It is a comprehensive, authoritative [47] √\n",
            "compendium of human genes and\n",
            "genetic phenotypes that is freely\n",
            "available and updated daily. The full-\n",
            "text, referenced overviews in OMIM https://www.omim.or\n",
            "OMIM\n",
            "contain information on all known g/\n",
            "mendelian disorders and over 16,000\n",
            "genes, and it focuses on the\n",
            "relationship between phenotype and\n",
            "genotype.\n",
            "PathBank PathBank is designed specifically to https://pathbank.org [48]\n",
            "support pathway elucidation and /\n",
            "discovery in transcriptomics,\n",
            "proteomics, metabolomics and systems\n",
            "biology.\n",
            "PathwayComm Pathways including biochemical http://www.pathway [49] √\n",
            "on reactions, complex assembly and commons.org/\n",
            "physical interactions involving proteins,\n",
            "DNA, RNA, small molecules and\n",
            "complexes.\n",
            "PDSP Ki It contains bioactivity data in terms of k https://pdspdb.unc.e [50] √\n",
            "i\n",
            "especially for GPCRs, ion channels, du/pdspWeb/\n",
            "transporters and enzymes.\n",
            "PharmGKB It contains comprehensive data on https://www.pharmg [51] √\n",
            "genetic variation on drug response for kb.org/\n",
            "clinicians and researchers.\n",
            "Probes & A public resource joining together https://www.probesd [52]\n",
            "Drugs Portal focused libraries of bioactive rugs.org/home/\n",
            "compounds (e.g., probes, drugs,\n",
            "specific inhibitor sets).\n",
            "It provides varieties of molecular [29] √\n",
            "https://pubchem.ncb\n",
            "Pubchem information including chemical structure\n",
            "i.nlm.nih.gov/\n",
            "and physical properties, biological\n",
            "activities, safety and toxicity\n",
            "information, patents, literature citations\n",
            "and so on.\n",
            "STITCH It stores known and predicted http://stitch.embl.de/ [53] √\n",
            "interactions of chemicals and proteins,\n",
            "and currently covers 9,643,763 proteins\n",
            "from 2,031 organisms.\n",
            "Supertarget A data resource is used for analyzing http://bioinf- [54]\n",
            "drug-target interactions and drug side apache.charite.de/s\n",
            "effects. upertarget/\n",
            "SwissTarget- It contains information on predicted http://www.swisstarg [55]\n",
            "Prediction targets of drugs based on the similarity etprediction.ch/\n",
            "principle through reverse screening.\n",
            "Therapeutic Target Database (TTD)\n",
            "provides information about the known\n",
            "and explored therapeutic protein and\n",
            "https://db.idrblab.or\n",
            "TTD nucleic acid targets, the targeted [56]\n",
            "g/ttd/\n",
            "disease, pathway information and the\n",
            "corresponding drugs directed at each\n",
            "of these targets.\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Target-centered models\n",
            "input\n",
            "Model network architecture type year\n",
            "protein compound\n",
            "Protein SPS\n",
            "RNN, CNN, Attention\n",
            "DeepAffintiy[16] (Structural property SMILES DTA 2019\n",
            "Mechanism\n",
            "sequence)\n",
            "Protein sequence,\n",
            "structural,\n",
            "Rifaioglu et\n",
            "evolutionary and SMILES CNN DTA 2020\n",
            "al.[73]\n",
            "physicochemical\n",
            "properties\n",
            "Molecular\n",
            "GraphDTA[17] Protein sequence GCN, CNN DTA 2019\n",
            "graph\n",
            "DeepConv-\n",
            "Protein sequence Fingerprint CNN, DNN DTI 2019\n",
            "DTI[92]\n",
            "ECFP &\n",
            "Amino acid\n",
            "MCPINN[93] Mol2Vec & DNN CPI 2019\n",
            "sequence & ProtVec\n",
            "SMILES\n",
            "GCN, LSTM, two-\n",
            "Amino acid Molecular\n",
            "Gao et al.[95] way Attention DTI 2018\n",
            "sequence graph\n",
            "Mechanism\n",
            "TransformerC Molecular\n",
            "Protein sequence Transformer CPI 2020\n",
            "PI[98] graph\n",
            "Tsubaki et Amino acid Molecular GCN, CNN, Attention\n",
            "CPI 2019\n",
            "al.[94] sequence graph Mechanism\n",
            "Eight individual drug or target-\n",
            "NeoDTI[100] GCN DTI 2019\n",
            "related networks\n",
            "15 types of chemical, genomic,\n",
            "DeepDTnet[77] Autoencoder DTI 2020\n",
            "phenotypic, and cellular networks\n",
            "15 networks covering chemical,\n",
            "genomic, phenotypic and network\n",
            "AOPEDF[58] Deep forest algorithm DTI 2020\n",
            "profiles among drugs,\n",
            "proteins/targets and diseases.\n",
            "Biomedical knowledge graphs about Knowledge Graph\n",
            "Trimodel[87] DTI 2019\n",
            "drug and target Embedding\n",
            "Disease-centered models\n",
            "input\n",
            "network architecture type year\n",
            "Model Drug & Disease\n",
            "Similarity Network Drug-\n",
            "Drug-related similarity information, Fusion (SNF), disease\n",
            "SNF-CVAE[102] 2020\n",
            "Drug-Disease associations Collective Variational association\n",
            "Autoencoder (cVAE) prediction\n",
            "Drug-\n",
            "Drug network, disease network, disease\n",
            "Xuan et al.[104] CNN, BiLSTM 2019\n",
            "Drug-Disease associations association\n",
            "prediction\n",
            "Multimodal deep Drug-\n",
            "Drug-Disease, Drug-Side-effect,\n",
            "autoencoder (MDA), disease\n",
            "DeepDR[105] Drug-Target and seven Drug-Drug 2019\n",
            "Collective Variational association\n",
            "networks\n",
            "Autoencoder (cVAE) prediction\n",
            "Drug-\n",
            "Drug-Protein, Disease-Protein and disease\n",
            "Wang et al.[106] Bipartite GCN 2020\n",
            "PPIs association\n",
            "prediction\n",
            "Drug-\n",
            "Knowledge graph disease\n",
            "Cov-KGE[108] Biomedical knowledge graphs 2020\n",
            "embedding association\n",
            "prediction\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 8. Conclusion\n",
        "\n",
        "Several Python libraries exist for text, table and image extraction although it appears that no single tool can extract all of the data in a structured way. The focus here is extracting text containing subject-predicate-object triples suitable for downstream NLP tasks such as Named Entity Recognition and Relation Extraction therefore text within tables and images has been excluded.\n",
        "\n",
        "Pdfminer is useful for page layout analysis and text extraction from documents with multi-column layouts but does not provide tools for table extraction. It provides the foundation for pdfplumber which contains table extraction methods but a scalable approach for extracting tables and text from structurally diverse multi-column layout PDFs from various sources would be challenging. [Suggested solutions](https://stackoverflow.com/questions/70170544/pdfplumber-extract-text-from-dynamic-column-layouts) include a document layout analysis algorithm to identify whether a page is multi-column or not, and the coordinates; or cropping the page keeping the middle 5% and running text extraction on it to see if there is any text or not, assuming the page is a two-column layout.\n",
        "\n",
        "Further work will include the investigation of Large Language Models for information extraction using orchestration frameworks such as LangChain, LlamaIndex and Haystack, to make an otherwise time-consuming and manual process more efficient, accurate, and scalable."
      ],
      "metadata": {
        "id": "3QsmHumYYkv_"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### References\n",
        "\n",
        "*  pdfminer.six https://github.com/pdfminer/pdfminer.six\n",
        "\n",
        "*  pdfminer.six documentation https://pdfminersix.readthedocs.io/\n",
        "\n",
        "*  Original pdfminer library of which pdfminer.six is a fork https://github.com/euske/pdfminer  \n",
        "\n",
        "* pdfminer documentation https://pdfminer-docs.readthedocs.io/\n",
        "\n",
        "*  pdfplumber https://github.com/jsvine/pdfplumber\n",
        "\n",
        "*   https://stackoverflow.com/questions/67380474/how-to-ignore-table-and-its-content-while-extracting-text-from-pdf\n",
        "\n",
        "*   https://stackoverflow.com/questions/70170544/pdfplumber-extract-text-from-dynamic-column-layouts\n",
        "\n",
        "\n",
        "*   https://github.com/janedoesrepo/pdfreader/blob/master/pdfreader.ipynb\n",
        "\n",
        "*   https://www.degeneratestate.org/posts/2016/Jun/15/extracting-tabular-data-from-pdfs/\n",
        "\n",
        "*   https://towardsdatascience.com/translate-long-pdf-reports-in-python-eab3be08ceb4\n",
        "\n",
        "*   https://towardsdatascience.com/pdf-text-extraction-in-python-5b6ab9e92dd\n",
        "\n",
        "*   https://towardsdatascience.com/extracting-text-from-pdf-files-with-python-a-comprehensive-guide-9fc4003d517\n",
        "\n"
      ],
      "metadata": {
        "id": "abVMvzkuYEXO"
      }
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}