{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# GROBID (GeneRation Of BIbliographic Data)"
      ],
      "metadata": {
        "id": "lck84urZuemw"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 1. Introduction\n",
        "\n",
        "[GROBID](https://github.com/kermitt2/grobid) is a machine learning library for extracting, parsing and restructuring raw documents such as PDF into structured XML/TEI encoded documents with a particular focus on technical and scientific publications.  \n",
        "\n",
        "Functionalities include full text extraction and structuring from PDF articles including a model for the overall document segmentation and models for the structuring of the text body (paragraph, section titles, reference and footnote callouts, figures, tables, data availability statements, etc.).\n",
        "\n",
        "We will use this to extract the full text from the arXiv PDFs for which we downloaded the metadata."
      ],
      "metadata": {
        "id": "dXDtbQacsaz6"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 2. Install/import libraries"
      ],
      "metadata": {
        "id": "Buttu_RvuDRG"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "sfDU-adrx6HM"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import pickle\n",
        "import re\n",
        "import requests\n",
        "import urllib\n",
        "import time\n",
        "import concurrent\n",
        "\n",
        "from multiprocessing.pool import ThreadPool\n",
        "from functools import lru_cache\n",
        "from bs4 import BeautifulSoup, NavigableString, Tag\n",
        "from itertools import chain\n",
        "from collections import Counter"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dwe8GC3F4udi"
      },
      "source": [
        "## 3. Import arXiv PDF metadata\n",
        "\n",
        "Read in metadata collected via arXiv API with PDF URLs from export.arxiv.org subdomain."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_304 = pd.read_pickle('2023-01-06_arxiv_results_for_dl.pickle')\n",
        "article_results_arxiv_304"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "FSZWbfdSEqva",
        "outputId": "2cc1336c-75eb-4d0c-bd7f-111ceae42b73"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         arxiv-id   published     revised  \\\n",
              "0    2109.06377v4  2021-09-14  2022-12-22   \n",
              "1    2212.09867v1  2022-12-19  2022-12-19   \n",
              "2    2212.09610v1  2022-12-19  2022-12-19   \n",
              "3    2212.00023v2  2022-11-30  2022-12-08   \n",
              "4    2212.03911v1  2022-12-07  2022-12-07   \n",
              "..            ...         ...         ...   \n",
              "299  2003.13665v1  2020-03-30  2020-03-30   \n",
              "300  2003.14258v1  2020-03-30  2020-03-30   \n",
              "301  2003.12454v1  2020-03-26  2020-03-26   \n",
              "302  2003.04524v1  2020-03-10  2020-03-10   \n",
              "303  2003.00751v1  2020-03-02  2020-03-02   \n",
              "\n",
              "                                                 title journal  \\\n",
              "0    ASGARD: A Single-cell Guided pipeline to Aid R...    None   \n",
              "1    Detecting Contradictory COVID-19 Drug Efficacy...    None   \n",
              "2    Drying of Bio-colloidal Sessile Droplets: Adva...    None   \n",
              "3    Random Copolymer inverse design system orienti...    None   \n",
              "4    Analysis of Drug repurposing Knowledge graphs ...    None   \n",
              "..                                                 ...     ...   \n",
              "299  Genomics-guided molecular maps of coronavirus ...    None   \n",
              "300  Nanomechanical sonification of the 2019-nCoV c...    None   \n",
              "301  A Machine Learning alternative to placebo-cont...    None   \n",
              "302  Old Drugs for Newly Emerging Viral Disease, CO...    None   \n",
              "303  Regulation of interferon production as a poten...    None   \n",
              "\n",
              "                                               authors   doi  \\\n",
              "0    Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...  None   \n",
              "1    Daniel N. Sosa, Malavika Suresh, Christopher P...  None   \n",
              "2           Anusuya Pal, Amalesh Gope, Anupam Sengupta  None   \n",
              "3                                 Tianyu Wu, Yang Tang  None   \n",
              "4                                  Ajay Kumar Gogineni  None   \n",
              "..                                                 ...   ...   \n",
              "299                                    Gennadi Glinsky  None   \n",
              "300                                  Markus J. Buehler  None   \n",
              "301   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc  None   \n",
              "302                                Mohammad Reza Dayer  None   \n",
              "303             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei  None   \n",
              "\n",
              "                                      pdf_url  \n",
              "0    http://export.arxiv.org/pdf/2109.06377v4  \n",
              "1    http://export.arxiv.org/pdf/2212.09867v1  \n",
              "2    http://export.arxiv.org/pdf/2212.09610v1  \n",
              "3    http://export.arxiv.org/pdf/2212.00023v2  \n",
              "4    http://export.arxiv.org/pdf/2212.03911v1  \n",
              "..                                        ...  \n",
              "299  http://export.arxiv.org/pdf/2003.13665v1  \n",
              "300  http://export.arxiv.org/pdf/2003.14258v1  \n",
              "301  http://export.arxiv.org/pdf/2003.12454v1  \n",
              "302  http://export.arxiv.org/pdf/2003.04524v1  \n",
              "303  http://export.arxiv.org/pdf/2003.00751v1  \n",
              "\n",
              "[304 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-bce49e86-2422-4608-abb9-237bb6b603fb\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>arxiv-id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>2109.06377v4</td>\n",
              "      <td>2021-09-14</td>\n",
              "      <td>2022-12-22</td>\n",
              "      <td>ASGARD: A Single-cell Guided pipeline to Aid R...</td>\n",
              "      <td>None</td>\n",
              "      <td>Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2109.06377v4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>2212.09867v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Detecting Contradictory COVID-19 Drug Efficacy...</td>\n",
              "      <td>None</td>\n",
              "      <td>Daniel N. Sosa, Malavika Suresh, Christopher P...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09867v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2212.09610v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Drying of Bio-colloidal Sessile Droplets: Adva...</td>\n",
              "      <td>None</td>\n",
              "      <td>Anusuya Pal, Amalesh Gope, Anupam Sengupta</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09610v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>2212.00023v2</td>\n",
              "      <td>2022-11-30</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>Random Copolymer inverse design system orienti...</td>\n",
              "      <td>None</td>\n",
              "      <td>Tianyu Wu, Yang Tang</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.00023v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>2212.03911v1</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>Analysis of Drug repurposing Knowledge graphs ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ajay Kumar Gogineni</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.03911v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>299</th>\n",
              "      <td>2003.13665v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Genomics-guided molecular maps of coronavirus ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Gennadi Glinsky</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.13665v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>300</th>\n",
              "      <td>2003.14258v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Nanomechanical sonification of the 2019-nCoV c...</td>\n",
              "      <td>None</td>\n",
              "      <td>Markus J. Buehler</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.14258v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>301</th>\n",
              "      <td>2003.12454v1</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>A Machine Learning alternative to placebo-cont...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ezequiel Alvarez, Federico Lamagna, Manuel Szewc</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.12454v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>302</th>\n",
              "      <td>2003.04524v1</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>Old Drugs for Newly Emerging Viral Disease, CO...</td>\n",
              "      <td>None</td>\n",
              "      <td>Mohammad Reza Dayer</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.04524v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>303</th>\n",
              "      <td>2003.00751v1</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>Regulation of interferon production as a poten...</td>\n",
              "      <td>None</td>\n",
              "      <td>Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.00751v1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>304 rows Ã— 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-bce49e86-2422-4608-abb9-237bb6b603fb')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-bce49e86-2422-4608-abb9-237bb6b603fb button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-bce49e86-2422-4608-abb9-237bb6b603fb');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XXOn0JwFAZ4n"
      },
      "source": [
        "## 4. Download full text as XML\n",
        "\n",
        "Create a function to download parsed XML for each PDF using a Python wrapper for the public GROBID demo server available for testing purposes. Please note according to the [GROBID repo](https://github.com/kermitt2/grobid/) quota and query limitations apply to the demo server and for any serious works you will need to deploy and use your own GROBID server. If a document exceeds the max token limit you will get an exception e.g. Exception('[TOO_MANY_TOKENS] The document has 1374455 tokens, but the limit is 1000000').\n",
        "\n",
        "The code below used the previous demo server hosted at [https://cloud.science-miner.com/grobid](https://cloud.science-miner.com/grobid) which has since been updated and redirects to a new demo server with a combination of Deep Learning models and CRF models hosted on Hugging Face Spaces at [https://kermitt2-grobid.hf.space/](https://kermitt2-grobid.hf.space/) or [ https://huggingface.co/spaces/kermitt2/grobid](https://huggingface.co/spaces/kermitt2/grobid).\n",
        "\n",
        "A faster demo with CRF only is available at https://kermitt2-grobid-crf.hf.space/ or https://huggingface.co/spaces/kermitt2/grobid-crf.\n",
        "\n",
        "Updating the GROBID URL in the function to [https://kermitt2-grobid.hf.space/api/processFulltextDocument](https://kermitt2-grobid.hf.space/api/processFulltextDocument) will be necessary.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "EF0OEp_S5L26"
      },
      "outputs": [],
      "source": [
        "# Adapted from https://github.com/titipata/scipdf_parser/blob/master/scipdf/pdf/parse_pdf.py\n",
        "\n",
        "def parse_pdf(pdf_url: str):\n",
        "    \"\"\"\n",
        "    Parse PDF to XML using GROBID tool\n",
        "\n",
        "    :param pdf_url: str, URL to article PDF\n",
        "\n",
        "    :return: XML of parsed article\n",
        "    \"\"\"\n",
        "    # GROBID URL for the cloud service to parse full text of the article\n",
        "    url = \"https://cloud.science-miner.com/grobid/api/processFulltextDocument\"\n",
        "\n",
        "    if isinstance(pdf_url, str):\n",
        "            page = urllib.request.urlopen(pdf_url).read()\n",
        "            resp = requests.post(url, files={\"input\": page})\n",
        "            if resp.status_code != 200:\n",
        "              raise Exception(resp.text)\n",
        "            parsed_article = resp.text\n",
        "            time.sleep(3)\n",
        "    else:\n",
        "        raise TypeError(\"Need to supply a url\")\n",
        "\n",
        "\n",
        "    return parsed_article"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Multithreading is used here since the code is I/O-bound rather than CPU-bound. Executing multiple threads concurrently speeds up the process as opposed to just iterating through the pdf_url list using a for loop calling the parse_pdf() function sequentially for each URL."
      ],
      "metadata": {
        "id": "w8EqDCZH1Aqm"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5p1eEYTP-FZT"
      },
      "outputs": [],
      "source": [
        "with concurrent.futures.ThreadPoolExecutor(4) as executor:\n",
        "     futures = [executor.submit(parse_pdf, pdf_url) for pdf_url in article_results_arxiv_304.pdf_url]\n",
        "     concurrent.futures.wait(futures)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create a dictionary of futures which are proxies for results that do not yet exist but will in the future."
      ],
      "metadata": {
        "id": "bqe_UdEwWO2T"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "futures_map = dict(zip(article_results_arxiv_304.pdf_url, futures))\n",
        "futures_map"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HCEcIWie5YcS",
        "outputId": "2c2174a3-2793-48e6-d425-b4bbc8c6525e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'http://export.arxiv.org/pdf/2109.06377v4': <Future at 0x7f9fc2e3d490 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2212.09867v1': <Future at 0x7f9fc6be80d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2212.09610v1': <Future at 0x7f9fc2e3db50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2212.00023v2': <Future at 0x7f9fc2e3de20 state=finished raised Exception>,\n",
              " 'http://export.arxiv.org/pdf/2212.03911v1': <Future at 0x7f9fc6174670 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2212.01575v1': <Future at 0x7f9fc616dac0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.02009v2': <Future at 0x7f9fc616d910 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2204.08697v2': <Future at 0x7f9fc616d8e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2207.09551v2': <Future at 0x7f9fc616da00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2107.02905v2': <Future at 0x7f9fc616d730 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2211.00201v2': <Future at 0x7f9fc2e380d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.11576v3': <Future at 0x7f9fc2e381c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.00127v2': <Future at 0x7f9fc2e382b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2205.11966v2': <Future at 0x7f9fc2e383a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2210.04774v1': <Future at 0x7f9fc2e38490 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2209.04431v2': <Future at 0x7f9fc2e38580 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2206.07375v2': <Future at 0x7f9fc2e38670 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2210.01882v1': <Future at 0x7f9fc2e38760 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2209.13773v1': <Future at 0x7f9fc2e38850 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.15803v3': <Future at 0x7f9fc2e38940 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2209.11508v1': <Future at 0x7f9fc2e38a30 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2209.09810v1': <Future at 0x7f9fc2e38b20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2209.05069v1': <Future at 0x7f9fc2e38c10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.00728v3': <Future at 0x7f9fc2e38d00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2209.00114v1': <Future at 0x7f9fc6174a00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2208.06362v3': <Future at 0x7f9fc617a1f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2208.06355v1': <Future at 0x7f9fc617aee0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2208.05019v1': <Future at 0x7f9fc2e40760 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2206.01047v2': <Future at 0x7f9fc2e40820 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2206.14358v1': <Future at 0x7f9fc2e40970 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2206.13634v1': <Future at 0x7f9fc2e40430 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2206.06111v1': <Future at 0x7f9fc2e404f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.11584v2': <Future at 0x7f9fc2e40670 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2206.02970v1': <Future at 0x7f9fc2e40160 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2204.09781v3': <Future at 0x7f9fc2e40220 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2205.14781v1': <Future at 0x7f9fc2e40340 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2205.05618v1': <Future at 0x7f9fc2e40280 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.12263v3': <Future at 0x7f9fc2e40250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.07531v2': <Future at 0x7f9fc2e69ee0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2204.09471v1': <Future at 0x7f9fc2e69190 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2204.04017v1': <Future at 0x7f9fc2e69c10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2201.00287v3': <Future at 0x7f9fc2e69be0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.05470v5': <Future at 0x7f9fc2e693d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2203.10338v1': <Future at 0x7f9fc2e69f70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.10177v7': <Future at 0x7f9fc2e69e20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2202.05145v1': <Future at 0x7f9fc2e695e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2202.01962v1': <Future at 0x7f9fc2e69790 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.10931v3': <Future at 0x7f9fc2e69a90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2202.00143v1': <Future at 0x7f9fc2e69b20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2201.12237v1': <Future at 0x7f9fc2e69a00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2201.08176v1': <Future at 0x7f9fc2e694f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.13751v3': <Future at 0x7f9fc2e696a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2201.00237v1': <Future at 0x7f9fc2e69700 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2112.02560v2': <Future at 0x7f9fc3aaed90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2112.01318v1': <Future at 0x7f9fc3aae250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2109.06711v2': <Future at 0x7f9fc3aaea00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2111.14480v1': <Future at 0x7f9fc3aaeaf0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2111.14283v1': <Future at 0x7f9fc3aae520 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2111.11955v1': <Future at 0x7f9fc3aa9d90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.13522v2': <Future at 0x7f9fc3aa9400 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.13971v1': <Future at 0x7f9fc3aa9e50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.12450v1': <Future at 0x7f9fc3a20880 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.09606v1': <Future at 0x7f9fc3a20250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2110.06397v1': <Future at 0x7f9fc3a204c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.07567v2': <Future at 0x7f9fc3a20d30 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2109.13739v2': <Future at 0x7f9fc3a20400 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2109.03565v2': <Future at 0x7f9fc3a208e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2106.13375v2': <Future at 0x7f9fc3a20430 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.02723v3': <Future at 0x7f9fc3a20640 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2109.00100v4': <Future at 0x7f9fc3a20b50 state=finished raised Exception>,\n",
              " 'http://export.arxiv.org/pdf/2109.00435v3': <Future at 0x7f9fc3a206a0 state=finished raised Exception>,\n",
              " 'http://export.arxiv.org/pdf/2010.16413v3': <Future at 0x7f9fc3a20f40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.02843v2': <Future at 0x7f9fc3a20eb0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.13764v1': <Future at 0x7f9fc3a20fa0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.12150v1': <Future at 0x7f9fc615fe20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2107.05347v2': <Future at 0x7f9fc615f880 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.09739v1': <Future at 0x7f9fc61b9c10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.07046v2': <Future at 0x7f9fc2e38e20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.06473v1': <Future at 0x7f9fc2e38f10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.09394v2': <Future at 0x7f9fc2e50040 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.12556v3': <Future at 0x7f9fc2e50130 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2108.00664v1': <Future at 0x7f9fc2e50220 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2107.12799v1': <Future at 0x7f9fc2e50310 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2107.12463v1': <Future at 0x7f9fc2e50400 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2107.09217v1': <Future at 0x7f9fc2e504f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2107.09602v1': <Future at 0x7f9fc2e505e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.01981v3': <Future at 0x7f9fc2e506d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2106.14631v1': <Future at 0x7f9fc2e507c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2106.12297v1': <Future at 0x7f9fc2e508b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.09507v4': <Future at 0x7f9fc2e509a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2106.06911v1': <Future at 0x7f9fc2e50a90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2106.05440v1': <Future at 0x7f9fc2e50b80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.01533v2': <Future at 0x7f9fc2e50c70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.14005v3': <Future at 0x7f9fc2e50d60 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2106.00356v1': <Future at 0x7f9fc2e50e50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2105.14433v1': <Future at 0x7f9fc2e50f40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2105.14333v1': <Future at 0x7f9fc2e57070 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2105.12267v1': <Future at 0x7f9fc2e57160 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2105.11410v1': <Future at 0x7f9fc2e57250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.04700v3': <Future at 0x7f9fc2e57340 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.11308v2': <Future at 0x7f9fc2e57430 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.00576v6': <Future at 0x7f9fc2e57520 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2105.05220v1': <Future at 0x7f9fc2e57610 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2105.10489v1': <Future at 0x7f9fc2e57700 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.03146v3': <Future at 0x7f9fc2e577f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.13224v1': <Future at 0x7f9fc2e578e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.12003v1': <Future at 0x7f9fc2e579d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.11674v1': <Future at 0x7f9fc2e57ac0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.15561v2': <Future at 0x7f9fc2e57bb0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.08867v1': <Future at 0x7f9fc2e57ca0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.07367v1': <Future at 0x7f9fc2e57d90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.05321v1': <Future at 0x7f9fc2e57e80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.05188v1': <Future at 0x7f9fc2e57f70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2104.01113v2': <Future at 0x7f9fc2e5e0a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.16192v1': <Future at 0x7f9fc2e5e190 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.04172v4': <Future at 0x7f9fc2e5e280 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.14102v1': <Future at 0x7f9fc2e5e370 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.13844v1': <Future at 0x7f9fc2e5e460 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.13297v1': <Future at 0x7f9fc2e5e550 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.03354v2': <Future at 0x7f9fc2e5e640 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.06578v1': <Future at 0x7f9fc2e5e730 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.02202v2': <Future at 0x7f9fc2e5e820 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.06357v2': <Future at 0x7f9fc2e5e910 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.10445v2': <Future at 0x7f9fc2e5ea00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2103.01896v1': <Future at 0x7f9fc2e5eaf0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.13103v1': <Future at 0x7f9fc2e5ebe0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.08628v1': <Future at 0x7f9fc2e5ecd0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.07610v1': <Future at 0x7f9fc2e5edc0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.07022v1': <Future at 0x7f9fc2e5eeb0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.04977v1': <Future at 0x7f9fc2e5efa0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.09600v2': <Future at 0x7f9fc0fb20d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.02250v1': <Future at 0x7f9fc0fb21c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.00971v1': <Future at 0x7f9fc0fb22b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2102.00197v1': <Future at 0x7f9fc0fb23a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.12547v1': <Future at 0x7f9fc0fb2490 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.11990v1': <Future at 0x7f9fc0fb2580 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.07824v1': <Future at 0x7f9fc0fb2670 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.04617v1': <Future at 0x7f9fc0fb2760 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.11336v3': <Future at 0x7f9fc0fb2850 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.11710v1': <Future at 0x7f9fc0fb2940 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.03133v1': <Future at 0x7f9fc0fb2a30 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2101.01884v1': <Future at 0x7f9fc0fb2b20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.13493v4': <Future at 0x7f9fc0fb2c10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.13605v1': <Future at 0x7f9fc0fb2d00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.06139v1': <Future at 0x7f9fc0fb2df0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.04682v2': <Future at 0x7f9fc0fb2ee0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.07856v2': <Future at 0x7f9fc0fb2fd0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.02151v1': <Future at 0x7f9fc0fb8100 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.01953v1': <Future at 0x7f9fc0fb81f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.00627v1': <Future at 0x7f9fc0fb82e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.14241v1': <Future at 0x7f9fc0fb83d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.04651v2': <Future at 0x7f9fc0fb84c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.12350v1': <Future at 0x7f9fc0fb85b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.01453v4': <Future at 0x7f9fc0fb86a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.05836v1': <Future at 0x7f9fc0fb8790 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.08902v1': <Future at 0x7f9fc0fb8880 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.00336v2': <Future at 0x7f9fc0fb8970 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.15681v2': <Future at 0x7f9fc0fb8a60 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.03983v1': <Future at 0x7f9fc0fb8b50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.00653v1': <Future at 0x7f9fc0fb8c40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2012.00652v1': <Future at 0x7f9fc0fb8d30 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.15209v3': <Future at 0x7f9fc0fb8e20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2011.01251v1': <Future at 0x7f9fc0fb8f10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.04438v2': <Future at 0x7f9fc0fbf040 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.16325v1': <Future at 0x7f9fc0fbf130 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.07883v2': <Future at 0x7f9fc0fbf220 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.11367v2': <Future at 0x7f9fc0fbf310 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.12332v1': <Future at 0x7f9fc0fbf400 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.10517v1': <Future at 0x7f9fc0fbf4f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.08855v1': <Future at 0x7f9fc0fbf5e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.07586v1': <Future at 0x7f9fc0fbf6d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.06574v1': <Future at 0x7f9fc0fbf7c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.02417v2': <Future at 0x7f9fc0fbf8b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.10276v2': <Future at 0x7f9fc0fbf9a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.04420v1': <Future at 0x7f9fc0fbfa90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.04115v1': <Future at 0x7f9fc0fbfb80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.09186v3': <Future at 0x7f9fc0fbfc70 state=finished raised Exception>,\n",
              " 'http://export.arxiv.org/pdf/2005.02261v2': <Future at 0x7f9fc0fbfd60 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2010.01916v1': <Future at 0x7f9fc0fbfe50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.12817v1': <Future at 0x7f9fc0fbff40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.11568v1': <Future at 0x7f9fc0fc7070 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.10333v1': <Future at 0x7f9fc0fc7160 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.10049v1': <Future at 0x7f9fc0fc7250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.12668v3': <Future at 0x7f9fc0fc7340 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.07736v4': <Future at 0x7f9fc0fc7430 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.03397v2': <Future at 0x7f9fc0fc7520 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.13602v1': <Future at 0x7f9fc0fc7610 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.03068v1': <Future at 0x7f9fc0fc7700 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.01094v1': <Future at 0x7f9fc0fc77f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.00744v1': <Future at 0x7f9fc0fc78e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2009.08861v1': <Future at 0x7f9fc0fc79d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.12353v1': <Future at 0x7f9fc0fc7ac0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.09649v1': <Future at 0x7f9fc0fc7bb0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.07488v2': <Future at 0x7f9fc0fc7ca0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.07880v1': <Future at 0x7f9fc0fc7d90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.00979v3': <Future at 0x7f9fc0fc7e80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.06974v1': <Future at 0x7f9fc0fc7f70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.06755v1': <Future at 0x7f9fc0fcd0a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.14492v4': <Future at 0x7f9fc0fcd190 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.05377v1': <Future at 0x7f9fc0fcd280 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.10233v5': <Future at 0x7f9fc0fcd370 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.07229v2': <Future at 0x7f9fc0fcd460 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.03283v1': <Future at 0x7f9fc0fcd550 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.00163v2': <Future at 0x7f9fc2e69940 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.00621v2': <Future at 0x7f9fc0fcd5e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.07350v1': <Future at 0x7f9fc0fcd6d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2008.04045v1': <Future at 0x7f9fc0fcd7c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.02772v2': <Future at 0x7f9fc0fcd8b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.16177v1': <Future at 0x7f9fc0fcd9a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.01902v2': <Future at 0x7f9fc0fcda90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.15835v1': <Future at 0x7f9fc0fcdb80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.14965v1': <Future at 0x7f9fc0fcdc70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.08095v3': <Future at 0x7f9fc0fcdd60 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.14375v1': <Future at 0x7f9fc0fcde50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.14805v1': <Future at 0x7f9fc0fcdf40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.12731v1': <Future at 0x7f9fc0fd4070 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.12242v1': <Future at 0x7f9fc0fd4160 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.07955v2': <Future at 0x7f9fc0fd4250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.10261v1': <Future at 0x7f9fc0fd4340 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.06435v3': <Future at 0x7f9fc0fd4430 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.10706v4': <Future at 0x7f9fc0fd4520 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.00662v5': <Future at 0x7f9fc0fd4610 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.04809v1': <Future at 0x7f9fc0fd4700 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/1808.04773v2': <Future at 0x7f9fc0fd47f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.03678v1': <Future at 0x7f9fc0fd48e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.13285v3': <Future at 0x7f9fc0fd49d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.02557v1': <Future at 0x7f9fc0fd4ac0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.02338v1': <Future at 0x7f9fc0fd4bb0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.10287v1': <Future at 0x7f9fc0fd4ca0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2007.01800v1': <Future at 0x7f9fc0fd4d90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.13997v4': <Future at 0x7f9fc0fd4e80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.01284v3': <Future at 0x7f9fc0fd4f70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.01226v2': <Future at 0x7f9fc0fdc0a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.02936v2': <Future at 0x7f9fc0fdc190 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.03110v3': <Future at 0x7f9fc0fdc280 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.14790v1': <Future at 0x7f9fc0fdc370 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.14707v1': <Future at 0x7f9fc0fdc460 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.01215v2': <Future at 0x7f9fc0fdc550 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.02631v4': <Future at 0x7f9fc0fdc640 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.16926v1': <Future at 0x7f9fc0fdc730 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.10584v1': <Future at 0x7f9fc0fdc820 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.09125v2': <Future at 0x7f9fc0fdc910 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.06452v1': <Future at 0x7f9fc0fdca00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.12124v2': <Future at 0x7f9fc0fdcaf0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.06523v1': <Future at 0x7f9fc0fdcbe0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.05502v1': <Future at 0x7f9fc0fdccd0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.11088v2': <Future at 0x7f9fc0fdcdc0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.00652v1': <Future at 0x7f9fc0fdceb0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.00343v1': <Future at 0x7f9fc0fdcfa0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.02343v1': <Future at 0x7f9fc0fe20d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2006.02431v1': <Future at 0x7f9fc0fe21c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.13653v1': <Future at 0x7f9fc0fe22b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.06891v2': <Future at 0x7f9fc0fe23a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.11260v2': <Future at 0x7f9fc0fe2490 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.11767v1': <Future at 0x7f9fc0fe2580 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.11458v1': <Future at 0x7f9fc0fe2670 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.11254v1': <Future at 0x7f9fc0fe2760 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.05625v2': <Future at 0x7f9fc0fe2850 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.10797v1': <Future at 0x7f9fc0fe2940 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.10831v1': <Future at 0x7f9fc0fe2a30 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.10028v1': <Future at 0x7f9fc0fe2b20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.09805v1': <Future at 0x7f9fc0fe2c10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.12137v1': <Future at 0x7f9fc0fe2d00 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.02666v2': <Future at 0x7f9fc0fe2df0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.08224v1': <Future at 0x7f9fc0fe2ee0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.06380v2': <Future at 0x7f9fc0fe2fd0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.05954v1': <Future at 0x7f9fc0fe8100 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.07121v1': <Future at 0x7f9fc0fe81f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.03004v1': <Future at 0x7f9fc0fe82e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.02289v1': <Future at 0x7f9fc0fe83d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.02188v1': <Future at 0x7f9fc0fe84c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.04108v1': <Future at 0x7f9fc0fe85b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.07697v2': <Future at 0x7f9fc0fe86a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.01042v1': <Future at 0x7f9fc0fe8790 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.14333v2': <Future at 0x7f9fc0fe8880 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2005.00485v1': <Future at 0x7f9fc0fe8970 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.14114v1': <Future at 0x7f9fc0fe8a60 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.13457v1': <Future at 0x7f9fc0fe8b50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.12420v1': <Future at 0x7f9fc0fe8c40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.12055v1': <Future at 0x7f9fc0fe8d30 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.11695v1': <Future at 0x7f9fc0fe8e20 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.10878v1': <Future at 0x7f9fc0fe8f10 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.10641v1': <Future at 0x7f9fc0f6f040 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.08920v1': <Future at 0x7f9fc0f6f130 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.08557v1': <Future at 0x7f9fc0f6f220 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.08491v1': <Future at 0x7f9fc0f6f310 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.00217v2': <Future at 0x7f9fc0f6f400 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.07750v1': <Future at 0x7f9fc0f6f4f0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.10642v2': <Future at 0x7f9fc0f6f5e0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.07086v1': <Future at 0x7f9fc0f6f6d0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.06447v1': <Future at 0x7f9fc0f6f7c0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.05684v1': <Future at 0x7f9fc0f6f8b0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.02136v3': <Future at 0x7f9fc0f6f9a0 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.04874v1': <Future at 0x7f9fc0f6fa90 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.03766v1': <Future at 0x7f9fc0f6fb80 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.02085v1': <Future at 0x7f9fc0f6fc70 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.13670v4': <Future at 0x7f9fc0f6fd60 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.01634v1': <Future at 0x7f9fc0f6fe50 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2004.00746v1': <Future at 0x7f9fc0f6ff40 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.13665v1': <Future at 0x7f9fc0f76070 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.14258v1': <Future at 0x7f9fc0f76160 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.12454v1': <Future at 0x7f9fc0f76250 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.04524v1': <Future at 0x7f9fc0f76340 state=finished returned str>,\n",
              " 'http://export.arxiv.org/pdf/2003.00751v1': <Future at 0x7f9fc0f76430 state=finished returned str>}"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5177bbf0-e13f-4f57-b96e-b0af9a2674e5",
        "id": "gy5k3cQw6DFK"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "304"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ],
      "source": [
        "len(futures_map)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create dictionary of exceptions with PDF URL as key and exception error message as value."
      ],
      "metadata": {
        "id": "9pNffeHPW6pi"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "exceptions = {url: f.exception() for url, f in futures_map.items() if f.exception() is not None}\n",
        "exceptions"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-3xfWu-45g31",
        "outputId": "ea0a7392-5be3-46ab-db59-e3be6e4e8d5f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'http://export.arxiv.org/pdf/2212.00023v2': Exception('[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1'),\n",
              " 'http://export.arxiv.org/pdf/2109.00100v4': Exception('[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1'),\n",
              " 'http://export.arxiv.org/pdf/2109.00435v3': Exception('[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1'),\n",
              " 'http://export.arxiv.org/pdf/2007.09186v3': Exception('[GENERAL] An exception occurred while running Grobid.')}"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Four exceptions were found. The three with Exception(' [BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1') are caused by the PDFs being unavailable so no XML would have been extracted. The one with Exception('[GENERAL] An exception occurred while running Grobid.') is a legitimate PDF but something went wrong during processing."
      ],
      "metadata": {
        "id": "3lrLWeKM2BDV"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1e77cd86-98a4-45bc-b277-e79c8ff468a6",
        "id": "SJPmOlDS6DFL"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'http://export.arxiv.org/pdf/2109.06377v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2212.09867v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2212.09610v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2212.03911v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2212.01575v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.02009v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2204.08697v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2207.09551v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2107.02905v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2211.00201v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.11576v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.00127v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2205.11966v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2210.04774v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2209.04431v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2206.07375v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2210.01882v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2209.13773v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.15803v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2209.11508v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2209.09810v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2209.05069v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.00728v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2209.00114v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2208.06362v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2208.06355v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2208.05019v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2206.01047v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2206.14358v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2206.13634v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2206.06111v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.11584v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2206.02970v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2204.09781v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2205.14781v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2205.05618v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.12263v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.07531v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2204.09471v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2204.04017v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2201.00287v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.05470v5': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2203.10338v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.10177v7': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2202.05145v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2202.01962v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.10931v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2202.00143v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2201.12237v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2201.08176v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.13751v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2201.00237v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2112.02560v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2112.01318v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2109.06711v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2111.14480v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2111.14283v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2111.11955v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.13522v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.13971v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.12450v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.09606v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2110.06397v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.07567v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2109.13739v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2109.03565v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2106.13375v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.02723v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.16413v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.02843v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.13764v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.12150v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2107.05347v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.09739v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.07046v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.06473v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.09394v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.12556v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2108.00664v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2107.12799v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2107.12463v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2107.09217v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2107.09602v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.01981v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2106.14631v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2106.12297v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.09507v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2106.06911v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2106.05440v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.01533v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.14005v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2106.00356v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2105.14433v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2105.14333v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2105.12267v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2105.11410v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.04700v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.11308v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.00576v6': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2105.05220v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2105.10489v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.03146v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.13224v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.12003v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.11674v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.15561v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.08867v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.07367v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.05321v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.05188v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2104.01113v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.16192v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.04172v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.14102v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.13844v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.13297v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.03354v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.06578v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.02202v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.06357v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.10445v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2103.01896v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.13103v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.08628v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.07610v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.07022v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.04977v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.09600v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.02250v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.00971v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2102.00197v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.12547v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.11990v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.07824v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.04617v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.11336v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.11710v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.03133v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2101.01884v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.13493v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.13605v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.06139v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.04682v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.07856v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.02151v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.01953v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.00627v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.14241v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.04651v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.12350v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.01453v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.05836v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.08902v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.00336v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.15681v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.03983v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.00653v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2012.00652v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.15209v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2011.01251v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.04438v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.16325v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.07883v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.11367v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.12332v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.10517v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.08855v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.07586v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.06574v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.02417v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.10276v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.04420v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.04115v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.02261v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2010.01916v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.12817v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.11568v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.10333v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.10049v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.12668v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.07736v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.03397v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.13602v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.03068v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.01094v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.00744v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2009.08861v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.12353v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.09649v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.07488v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.07880v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.00979v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.06974v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.06755v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.14492v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.05377v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.10233v5': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.07229v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.03283v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.00163v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.00621v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.07350v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2008.04045v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.02772v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.16177v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.01902v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.15835v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.14965v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.08095v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.14375v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.14805v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.12731v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.12242v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.07955v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.10261v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.06435v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.10706v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.00662v5': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.04809v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/1808.04773v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.03678v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.13285v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.02557v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.02338v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.10287v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2007.01800v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.13997v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.01284v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.01226v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.02936v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.03110v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.14790v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.14707v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.01215v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.02631v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.16926v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.10584v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.09125v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.06452v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.12124v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.06523v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.05502v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.11088v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.00652v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.00343v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.02343v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2006.02431v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.13653v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.06891v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.11260v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.11767v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.11458v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.11254v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.05625v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.10797v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.10831v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.10028v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.09805v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.12137v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.02666v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.08224v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.06380v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.05954v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.07121v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.03004v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.02289v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.02188v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.04108v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.07697v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.01042v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.14333v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2005.00485v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.14114v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.13457v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.12420v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.12055v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.11695v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.10878v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.10641v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.08920v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.08557v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.08491v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.00217v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.07750v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.10642v2': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.07086v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.06447v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.05684v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.02136v3': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.04874v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.03766v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.02085v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.13670v4': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.01634v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2004.00746v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.13665v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.14258v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.12454v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.04524v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"',\n",
              " 'http://export.arxiv.org/pdf/2003.00751v1': '<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\"'}"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ],
      "source": [
        "results = {url: f.result()[:100] for url, f in futures_map.items() if f.exception() is None}\n",
        "results"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Amend the parse_pdf function with error handling for specific exceptions and run again to download the full text.\n",
        "\n",
        "We will use the @lru_cache decorator from Python's functools module which helps in reducing the execution time of the function for the same inputs by using the memoization technique."
      ],
      "metadata": {
        "id": "aHQgCEss62kW"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "_uoMS_pc62kW"
      },
      "outputs": [],
      "source": [
        "@lru_cache(maxsize=None)\n",
        "def parse_pdf(pdf_url: str):\n",
        "    \"\"\"\n",
        "    Parse PDF to XML using GROBID tool\n",
        "\n",
        "    :param pdf_url: str, URL to article PDF\n",
        "\n",
        "    :return: XML of parsed article\n",
        "    \"\"\"\n",
        "    # GROBID URL for the cloud service to parse full text of the article\n",
        "    url = \"https://cloud.science-miner.com/grobid/api/processFulltextDocument\"\n",
        "\n",
        "    if isinstance(pdf_url, str):\n",
        "            page = urllib.request.urlopen(pdf_url).read()\n",
        "            resp = requests.post(url, files={\"input\": page})\n",
        "            if resp.status_code != 200:\n",
        "                if resp.status_code >= 500:\n",
        "                  retry = 1\n",
        "                else:\n",
        "                  return \"500\"\n",
        "                if resp.text in ['[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1']:\n",
        "                    return \"1\"\n",
        "                if resp.text in ['{\\n  \"message\":\"An invalid response was received from the upstream server\"\\n}',\n",
        "                                '[GENERAL] An exception occurred while running Grobid.']:\n",
        "                    return \"0\"\n",
        "                else:\n",
        "                    raise Exception(resp.text)\n",
        "            parsed_article = resp.text\n",
        "            time.sleep(3)\n",
        "    else:\n",
        "        raise TypeError(\"Need to supply a url\")\n",
        "\n",
        "\n",
        "    return parsed_article"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "4odecqma62kX"
      },
      "outputs": [],
      "source": [
        "with ThreadPool(4) as pool:\n",
        "  dl_pdf_xml = pool.map(parse_pdf, article_results_arxiv_304.pdf_url)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_grobid_xml_str_v2.pickle', \"wb\") as f:\n",
        "    pickle.dump(dl_pdf_xml, f)"
      ],
      "metadata": {
        "id": "OsyzV3mX3-l8"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(dl_pdf_xml)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "bde86dad-eda2-4792-e468-fb6f85b8891c",
        "id": "Lnottc4Y62kX"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "304"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check XML output for first article which was extracted successfully."
      ],
      "metadata": {
        "id": "83KBtVb8U5vW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "dl_pdf_xml[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 237
        },
        "outputId": "d68e8e5d-9b2e-438c-bef1-c54ccd0678c7",
        "id": "JEAKwzim62kX"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" \\nxmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" \\nxsi:schemaLocation=\"http://www.tei-c.org/ns/1.0 https://raw.githubusercontent.com/kermitt2/grobid/master/grobid-home/schemas/xsd/Grobid.xsd\"\\n xmlns:xlink=\"http://www.w3.org/1999/xlink\">\\n\\t<teiHeader xml:lang=\"en\">\\n\\t\\t<fileDesc>\\n\\t\\t\\t<titleStmt>\\n\\t\\t\\t\\t<title level=\"a\" type=\"main\">ASGARD: A Single-cell Guided Pipeline to Aid Repurposing of Drugs</title>\\n\\t\\t\\t</titleStmt>\\n\\t\\t\\t<publicationStmt>\\n\\t\\t\\t\\t<publisher/>\\n\\t\\t\\t\\t<availability status=\"unknown\"><licence/></availability>\\n\\t\\t\\t</publicationStmt>\\n\\t\\t\\t<sourceDesc>\\n\\t\\t\\t\\t<biblStruct>\\n\\t\\t\\t\\t\\t<analytic>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Bing</forename><surname>He</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Medical School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Yao</forename><surname>Xiao</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Medical School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Haodong</forename><surname>Liang</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff1\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Statistics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">College of Literature, Science, and the Arts</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Qianhui</forename><surname>Huang</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Medical School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Yuheng</forename><surname>Du</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Medical School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Yijun</forename><surname>Li</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Medical School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author role=\"corresp\">\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">David</forename><surname>Garmire</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<email>lgarmire@med.umich.edu</email>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff2\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Electrical Engineering and Computer Science</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">College of Engineering</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Duxin</forename><surname>Sun</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff3\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Pharmaceutical Sciences</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">College of Pharmacy</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Lana</forename><forename type=\"middle\">X</forename><surname>Garmire</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Medical School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff4\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Department of Computational Medicine and Bioinformatics</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\" key=\"instit1\">North Campus Research Complex</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\" key=\"instit2\">University of Michigan</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<addrLine>1600 Huron Parkway</addrLine>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>48105</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Ann Arbor</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MI</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<title level=\"a\" type=\"main\">ASGARD: A Single-cell Guided Pipeline to Aid Repurposing of Drugs</title>\\n\\t\\t\\t\\t\\t</analytic>\\n\\t\\t\\t\\t\\t<monogr>\\n\\t\\t\\t\\t\\t\\t<imprint>\\n\\t\\t\\t\\t\\t\\t\\t<date/>\\n\\t\\t\\t\\t\\t\\t</imprint>\\n\\t\\t\\t\\t\\t</monogr>\\n\\t\\t\\t\\t\\t<idno type=\"MD5\">D07A27EC15F206FD458ECE081E1236AE</idno>\\n\\t\\t\\t\\t</biblStruct>\\n\\t\\t\\t</sourceDesc>\\n\\t\\t</fileDesc>\\n\\t\\t<encodingDesc>\\n\\t\\t\\t<appInfo>\\n\\t\\t\\t\\t<application version=\"0.7.3-SNAPSHOT\" ident=\"GROBID\" when=\"2023-01-11T15:39+0000\">\\n\\t\\t\\t\\t\\t<desc>GROBID - A machine learning software for extracting information from scholarly documents</desc>\\n\\t\\t\\t\\t\\t<ref target=\"https://github.com/kermitt2/grobid\"/>\\n\\t\\t\\t\\t</application>\\n\\t\\t\\t</appInfo>\\n\\t\\t</encodingDesc>\\n\\t\\t<profileDesc>\\n\\t\\t\\t<abstract>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><p>Intercellular heterogeneity is a major obstacle to successful precision medicine. Single-cell RNA sequencing (scRNA-seq) technology has enabled in-depth analysis of intercellular heterogeneity in various diseases. However, its full potential for precision medicine has yet to be reached. Towards this, we propose a new drug recommendation system called: A Single-cell Guided Pipeline to Aid Repurposing of Drugs (ASGARD). ASGARD defines a novel drug score predicting drugs by considering all cell clusters to address the intercellular heterogeneity within each patient. We tested ASGARD on multiple diseases, including breast cancer, acute lymphoblastic leukemia, and coronavirus disease 2019 (COVID-19). On single-drug therapy, ASGARD shows significantly better average accuracy (AUC of 0.92) compared to two other bulk-cell-based drug repurposing methods (AUC of 0.80 and 0.76). It is also considerably better (AUC of 0.82) than other cell cluster level predicting methods (AUC of 0.67 and 0.55). In addition, ASGARD is also validated by the drug response prediction method TRANSACT with Triple-Negative-Breast-Cancer patient samples. Many top-ranked drugs are either approved by FDA or in clinical trials treating corresponding diseases. In silico cell-type specific drop-out experiments using triple-negative breast cancers show the importance of T cells in the tumor microenvironment in affecting drug predictions. In conclusion, ASGARD is a promising drug repurposing recommendation tool guided by single-cell RNA-seq for personalized medicine.</p></div>\\n\\t\\t\\t</abstract>\\n\\t\\t</profileDesc>\\n\\t</teiHeader>\\n\\t<text xml:lang=\"en\">\\n\\t\\t<body>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Introduction</head><p>Heterogeneity, or more specifically, the diverse cell populations within the diseased tissue, is the leading cause of treatment failure for many complex diseases, such as cancers 1 , Alzheimer\\'s disease <ref type=\"bibr\" target=\"#b1\">2</ref> , stroke <ref type=\"bibr\" target=\"#b2\">3</ref> , and coronavirus disease 2019 (COVID-19) <ref type=\"bibr\" target=\"#b3\">4</ref> , etc., as well as a major obstacle to successful precision medicine <ref type=\"bibr\" target=\"#b4\">[5]</ref><ref type=\"bibr\" target=\"#b5\">[6]</ref><ref type=\"bibr\" target=\"#b6\">[7]</ref> . Recent significant advances in single-cell technologies, especially the single-cell RNA sequencing (scRNA-seq) technology, have enabled the analysis of intercellular heterogeneity at a very fine resolution <ref type=\"bibr\" target=\"#b7\">8,</ref><ref type=\"bibr\" target=\"#b8\">9</ref> and helped us to have many breakthroughs in understanding disease mechanisms <ref type=\"bibr\" target=\"#b9\">10</ref> , such as breast cancer <ref type=\"bibr\" target=\"#b10\">11</ref> , liver cancer <ref type=\"bibr\" target=\"#b11\">12</ref> and COVID-19 <ref type=\"bibr\" target=\"#b12\">13</ref> . However, its full potential for precision medicine has not been fulfilled <ref type=\"bibr\" target=\"#b13\">14,</ref><ref type=\"bibr\" target=\"#b14\">15</ref> .</p><p>Drug repurposing (also known as drug reposition, reprofiling, or re-tasking) is a strategy to identify new drug uses outside the scope of its original medical approval or investigation <ref type=\"bibr\" target=\"#b15\">16</ref> . So far, few drug repurposing methods have been developed to utilize the highly valuable information contained in scRNA-seq data. The pipeline by Alakwaa identifies significantly differentiated genes (DEGs) for a specific group of cells, then predicts candidate drugs for DEGs using the Connectivity Map Linked User Environment (CLUE) platform, followed by prioritizing these drugs using a comprehensive ranking score system <ref type=\"bibr\" target=\"#b16\">17</ref> . This pipeline identified didanosine as a potential treatment for COVID-19 using scRNA-seq data <ref type=\"bibr\" target=\"#b16\">17</ref> . Another pipeline by Guo et al.   uses a simple combination of Seurat <ref type=\"bibr\" target=\"#b17\">18</ref> , a tool for scRNA-seq analysis, and CLUE to identify 281 FDA-approved drugs that can potentially be effective for treating COVID-19 <ref type=\"bibr\" target=\"#b18\">19</ref> . In general, the above pipelines predict drugs for each cell cluster within the patient. However, in heterogeneous diseases caused by multiple types of cells, efficient drugs should be able to address multiple cell clusters <ref type=\"bibr\" target=\"#b19\">20</ref> . Neither of these pipelines mentioned above can predict drugs for multiple cell clusters, limiting their utility in the era of precision medicine.</p><p>Here we propose A Single-cell Guided Pipeline to Aid Repurposing of Drugs (ASGARD) to overcome the issue above. ASGARD defines a novel drug score to predict drugs for multiple diseased cell clusters within each patient. The benchmarking results show that the performance of ASGARD on single drugs is more accurate and robust than other pipelines handling bulk and single-cell RNA-Seq data. We tested ASGARD on multiple cancer scRNA-Seq datasets, including patient-Derived Xenografts (PDXs) models for advanced metastatic breast cancers, from GEO with accession numbers GSE145926 <ref type=\"bibr\" target=\"#b12\">13</ref> and GSE158055 <ref type=\"bibr\" target=\"#b12\">13,</ref><ref type=\"bibr\" target=\"#b22\">23</ref> .</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Processing of scRNA-seq data</head><p>ASGARD accepts processed scRNA-seq data from the Seurat package <ref type=\"bibr\" target=\"#b17\">18</ref> . In this study, genes identified in fewer than three cells are removed from the dataset. We used the same criteria as in their original studies to filter cells <ref type=\"bibr\" target=\"#b10\">11,</ref><ref type=\"bibr\" target=\"#b12\">13,</ref><ref type=\"bibr\" target=\"#b20\">21</ref> . Preprocessing steps remove the following cells from the dataset: (1) epithelial cells from breast cancer PDXs and healthy breast tissues with fewer than 200 unique genes, (2) PBMC cells from leukemia patients and healthy controls with fewer than 200 unique genes, and (3) BALF cells from COVID-19 patients with fewer than 200 unique genes, more than 6000 unique genes or have a proportion of mitochondrial genes larger than 10% <ref type=\"bibr\" target=\"#b12\">13</ref> . For consistency, cells from TNBC patients with fewer than 200 unique genes are also removed from the dataset <ref type=\"bibr\" target=\"#b23\">24</ref> . We used cell cycle marker genes and linear transformation to scale the expression of each gene and remove the effects of the cell cycle on gene expression.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Cell pairwise correspondences</head><p>ASGARD suggests using functions from Seurat for cell pairwise correspondences. In this study, gene counts for each cell were divided by the total counts for that cell and multiplied by a scaling factor (default is set to 10000). The count matrix was then transformed by log 2(count+1) in R. To identify gene variance across cells, we first fitted a line to the relationship of log(variance) and log(mean) using local polynomial regression (loess). Then we standardized the feature values using the observed mean and expected variance (given by the fitted line).</p><p>Gene variance was then calculated on the standardized values. We used the 2,000 genes with the highest standardized variance for downstream analysis. Then we identified the K-nearest neighbors (KNNs) between disease and normal cells based on the L2-normalized canonical correlation vectors (CCV). Finally, we built up the pairwise cell correspondences by identifying mutual nearest neighbors <ref type=\"bibr\" target=\"#b17\">18</ref> .</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Cell clustering and annotation</head><p>We applied principal component analysis (PCA) from Seurat on the scaled data to perform the linear dimensional reduction. Then we used a graph-based clustering approach <ref type=\"bibr\" target=\"#b17\">18</ref> . In this approach, we first constructed a KNN graph based on the euclidean distance in PCA space and refined the edge weights between any two cell pairs using Jaccard similarity. Then we applied the Louvain algorithm of modularity optimization to iteratively group cell pairs together. We further ran non-linear dimensional reduction (UMAP) to place similar cells within the graph-based clusters determined above together in low-dimensional space. To annotate clusters of cells, we ran an automatic annotation of single cells based on similarity to the referenced single-cell panel using the SingleR package <ref type=\"bibr\" target=\"#b24\">25</ref> . We used the dominant cell type (&gt;50% cells) as the cell type of the cluster.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Drug repurposing</head><p>ASGARD supports importing differentially expressed genes calculated from multiple external methods, including Limma <ref type=\"bibr\" target=\"#b25\">26</ref> , Seurat (Wilcoxon Rank Sum test) <ref type=\"bibr\" target=\"#b17\">18</ref> , DESeq2 <ref type=\"bibr\" target=\"#b26\">27</ref> , and edgeR <ref type=\"bibr\" target=\"#b27\">28</ref> .</p><p>The differentially expressed gene list in disease is transformed into a gene rank list. ASGARD uses 21,304 drugs/compounds with response gene expression profiles in 98 cell lines from the LINCS L1000 project <ref type=\"bibr\" target=\"#b28\">29</ref> . A differential gene expression list in response to drug treatment is also transformed into a gene rank list. ASGARD further identifies potential candidate drugs that yield reversed gene expression patterns from those of diseased vs. normal cells, using the DrInsight package <ref type=\"bibr\" target=\"#b29\">30</ref> (version: 0.1.1). Specifically, it identifies consistently differentially expressed genes, which are up-regulated in cells from diseased tissue but down-regulated in cells with drug treatment, or down-regulated in cells from diseased tissue but up-regulated in cells with drug treatment. It then calculates the outlier-sum (OS) statistic <ref type=\"bibr\" target=\"#b30\">31</ref> , representing the effect of reversed differential gene pattern by the drug treatment. The Kolmogorov-Smirnov test (K-S test) is then applied to the OS statistic, to show the significance level of one drug treatment relative to the background of all other drugs in the dataset. The reference drug dataset contains gene rank lists of 591,697 drug/compound treatments from the LINCS L1000 data, as mentioned above.</p><p>The Benjamini-Hochberg procedure is used to adjust P-values from the K-S test to control False Discovery Rate (FDR) of multiple hypothesis testing <ref type=\"bibr\" target=\"#b31\">32</ref> .</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Drug evaluation</head><p>ASGARD defines a novel drug score at the individual patient level (Formula 1), which calculates the drug efficacy across all single-cell clusters in a given patient\\'s scRNA-Seq data. The drug score estimates drug efficacy using the cell type proportion, the significance of reversed differential gene expression pattern (FDR), and the ratio of reversed significantly deregulated genes over disease-related (or selected) single-cell clusters. The drug score is estimated by the following formula: is the ð‘˜ ð‘ð‘¢ð‘š (ð·ð‘–ð‘ ð‘’ð‘Žð‘ ð‘’ð‘‘ðºð‘’ð‘›ð‘’ð‘ ) ð‘˜ number of significantly deregulated genes in a cluster , while is the ð‘˜ ð‘ð‘¢ð‘š (ð‘…ð‘’ð‘£ð‘’ð‘Ÿð‘ ð‘’ð‘‘ðºð‘’ð‘›ð‘’ð‘ ) ð‘˜ number of significantly deregulated genes in a cluster that can be reversed by the drug. To ð‘˜ allow a comparison of drug efficacy across patients, ASGARD also provides a standardized drug score, which has a scale of 0 to 1 (Formula 2).</p><p>(</p><formula xml:id=\"formula_0\">) ð‘†ð‘¡ð‘Žð‘›ð‘‘ð‘Žð‘Ÿð‘‘ð‘–ð‘§ð‘’ð‘‘ ð·ð‘Ÿð‘¢ð‘” ð‘†ð‘ð‘œð‘Ÿð‘’ = 1 âˆ’ ð‘…ð‘Žð‘›ð‘˜(ð·ð‘Ÿð‘¢ð‘”) ð‘‡ð‘œð‘¡ð‘Žð‘™ ð‘ð‘¢ð‘š(ð·ð‘Ÿð‘¢ð‘”)<label>2</label></formula><p>Besides the drug score, ASGARD further provides Fisher\\'s combined P-value <ref type=\"bibr\" target=\"#b32\">33</ref> over the original P value of every cluster. The combined p-value is calculated as the right-tail probability , where</p><p>. The BH FDR is used to adjust Fisher\\'s combined ð‘ƒ </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Benchmarking ASGARD</head><p>We use the receiver operating characteristic curves (ROCs) and the areas under the ROC curves (AUCs) to compare the performance of ASGARD with those of the other two pipelines, as well as bulk methods. Since these pipelines/ methods report both drugs and compounds, we let ASGARD report both drugs and compounds in the comparisons with other pipelines/methods. ROCs and AUCs are calculated for each pipeline using the pROC package 34   .</p><p>In ROC and AUC estimation, we regarded FDA-approved drugs and compounds used in advanced clinical trials or have been proven effective in animal models as positive cases (Supplementary Table <ref type=\"table\">1</ref>), and all other drugs as negative cases. To identify drugs and compounds used in advanced clinical trials or have been proven effective in animal models, we used three databases that are ClinicalTrials.gov, PubMed, and PubChem, and all the drugs and compounds we found are listed in Supplementary Table <ref type=\"table\">1</ref>.</p><p>We determined the need to assess the robustness of the three pipelines on different (1) sizes,</p><p>(2) similarities, and (3) unbalances of single-cell populations. For (1), the Bootstrapping method in R <ref type=\"bibr\" target=\"#b34\">35</ref> generated simulation data of different sizes by randomly drawing the same number of disease and normal cells from GSE123926 and GSE113197. For (2), additional simulation data are generated by adjusting the differential gene expression levels from 20% to 90% of the original differential levels of the single-cell cluster, based on GSE123926 and GSE113197. For</p><p>(3), simulation data is generated by randomly drawing 5000 cells with diseased cell proportions ranging from 20% to 90%, thereby yielding unbalanced populations.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Drug score analysis</head><p>To examine the impact of each cell type on a drug score, we conducted in silico drop-one-out experiments, excluding one cell type from the scRNA-Seq data at a time. The difference between the new drug score and the original drug score is then calculated to reflect the contribution of each cell type to the drug prediction.</p><p>To validate ASGARD, we compared it with another drug response prediction method TRANSACT <ref type=\"bibr\" target=\"#b35\">36</ref> , on the TNBC dataset (GSE161529). Since TRANSACT is a method working on bulk gene expression data, we took the mean expression value of each gene across all cells as the pseudo-bulk expression value of that gene. To fit TRANSACT with the dataset we used, we changed two parameters, number_pc[\\'target\\'] and n_pv, to 3 and maintained all other parameters at the same value as the authors\\' original report <ref type=\"bibr\" target=\"#b35\">36</ref> .</p><p>the performance comparison and validation are available at:</p><p>https://github.com/lanagarmire/Single-cell-drug-repositioning/tree/master/ROC</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Results</head></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Summary of A Single-cell Guided pipeline to Aid Repurposing of Drugs (ASGARD)</head><p>Using scRNA-seq data, ASGARD repurposes drugs for disease by fully accounting for the cellular heterogeneity of patients (Figure <ref type=\"figure\" target=\"#fig_3\">1</ref>, Formula 1 in Methods). In ASGARD, every cell cluster in the diseased sample is paired to that in the normal (or control) sample, according to \"anchor\" genes that are consistently expressed between diseased and normal cells. It then imports differentially expressed (DE) genes (P-value &lt; 0.05) between the paired diseased and normal clusters in the scRNA-seq data, as determined by an DE detection method at the user\\'s choice. These individual clusters can be optionally annotated to specific cell types. To identify drugs for each single cluster (cell type), then ASGARD uses these consistently differentially expressed genes as inputs to identify drugs that can significantly (single-cluster FDR &lt; 0.05) reverse their expression levels in the L1000 drug response dataset <ref type=\"bibr\" target=\"#b28\">29</ref> . To identify drugs for multiple clusters, ASGARD defines a novel drug score (Formula 1 in Methods) to evaluate the drug efficacy across multiple cell clusters selected by the user. The drug score estimates drug efficacy by taking into account the cell type proportion, the significance of reversing the differential gene expression pattern (single-cluster FDR) by the drug treatment in each selected cell cluster, and the ratio of significantly deregulated genes (adjust P-value &lt; 0.05) that the drug treatment can reverse in each selected cell cluster. Finally, ASGARD uses the drug score to rank and choose drugs for the disease.</p><p>We evaluated the power of the drug score by comparing ASGARD with traditional bulk-cell-based repurposing methods and single-cell-based repurposing methods using multiple independent scRNA-seq datasets, including PDX models from advanced metastatic TNBC <ref type=\"bibr\" target=\"#b10\">11</ref> , an acute lymphoblastic leukemia dataset <ref type=\"bibr\" target=\"#b21\">22</ref> , and coronavirus disease 2019 (COVID-19) datasets <ref type=\"bibr\" target=\"#b12\">13,</ref><ref type=\"bibr\" target=\"#b22\">23</ref> (see Methods).</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Comparing ASGARD to bulk-cell based repurposing methods</head><p>Before comparing ASGARD to bulk-cell-based repurposing methods, we first evaludated several external differential expression (DE) methods, on three datasets from three diseases: advanced metastatic breast cancer <ref type=\"bibr\" target=\"#b10\">11,</ref><ref type=\"bibr\" target=\"#b20\">21</ref> , acute lymphoblastic leukemia <ref type=\"bibr\" target=\"#b21\">22</ref> , and coronavirus disease 2019 <ref type=\"bibr\" target=\"#b12\">13,</ref><ref type=\"bibr\" target=\"#b22\">23</ref> (see Methods). We selected Limma 26 , Seurat 18 , DESeq2 <ref type=\"bibr\" target=\"#b26\">27</ref> , and edgeR <ref type=\"bibr\" target=\"#b27\">28</ref> for DE methods, given that they were top-ranked methods in a benchmark study of confronting false discoveries in single-cell differential expression <ref type=\"bibr\" target=\"#b36\">37</ref> . We conducted systematic comparison of these methods under different modes (Supplementary Figure <ref type=\"figure\" target=\"#fig_4\">2</ref>). For limma, we compared three modes: empirical Bayes without trend (Bayes), empirical Bayes approach prior trend (trend), and precision weights (voom) <ref type=\"bibr\" target=\"#b25\">26</ref> . For Seurat, we compared three different DE tests:</p><p>Wilcoxon rank-sum test (Wilcox), t-test, and logistic regression (LR). For DESeq2, we compared the Wald test (Wald) and the likelihood ratio test (LRT). For edgeR, we compared the likelihood ratio test (LRT) and the quasi-likelihood F-test (QLF). We identified DE genes using the above methods for each cell cluster as the inputs of ASGARD for drug repurposing. The subsequent drug prediction accuracies by ASGARD are determined by the receiver operating characteristic curves (ROCs) and the areas under the ROC curves (AUCs), using FDA-approved drugs and candidate drugs validated in advanced clinical trials as the positive data (see Methods).</p><p>The systematic comparison is shown in Supplementary Figure <ref type=\"figure\" target=\"#fig_3\">1</ref>, and the results from each method under the best-performing mode are shown in Figure <ref type=\"figure\" target=\"#fig_4\">2A</ref>. Limma (Bayes) method yields the best AUC in all three datasets ranging from (0.90-0.92), significantly (P-value &lt; 0.05, student\\'s t-test) better than other DE methods (Figure <ref type=\"figure\" target=\"#fig_3\">1A</ref>). Seurat (Wilcox test) and edgeR perform similarly overall, where edgeR has slightly higher AUC (0.83-0.86) than Seurat (0.80-8.86). DEseq2 on the other hand, tends to generate some of the lowest AUCs in comparison. Therefore, we used DE results from the limma-Bayes package for the following analysis, while keeping other DE methods as options for the inputs to ASGARD.</p><p>To compare ASGARD with those drug repurposing methods using bulk RNA-Seq samples, we summarized scRNA-seq data into pseudo-bulk RNA-Seq data. We then applied bulk methods CLUE <ref type=\"bibr\" target=\"#b37\">38</ref> and DrInsight <ref type=\"bibr\" target=\"#b29\">30</ref> to the pseudo-bulk RNA-Seq query data and compared their results with ASGARD\\'s on predicting both drugs and compounds (Figure <ref type=\"figure\" target=\"#fig_4\">2B</ref>). We used the same scRNA-seq data from the same three datasets above. Since CLUE and DrInsight predict both drugs and compounds, we added compounds validated in animal models to the true positive dataset for the AUC evaluation of drug/compound predictions. As a result, the AUCs obtained from ASGARD on drugs and compounds (Figure <ref type=\"figure\" target=\"#fig_4\">2B</ref>) are slightly different from those on drugs only (Figure <ref type=\"figure\" target=\"#fig_4\">2A</ref>). On the breast cancer dataset, ASGARD yields an overall AUC of 0.92, much better than CLUE and DrInsight, with values of 0.74 and 0.81, respectively. On precursor T cell acute lymphoblastic leukemia data, ASGARD yields an AUC of 0.95 in drug/compound repurposing for leukemia patients, while CLUE and DrInsight achieve worse average AUCs of 0.82 and 0.73, respectively. For the COVID-19 datasets, ASGARD shows an AUC of 0.88 in drug/compound repurposing, while CLUE and DrInsight have lower AUCs of 0.85 and 0.73, respectively, for the same patients (Figure <ref type=\"figure\" target=\"#fig_4\">2B</ref>). In summary, by paying attention to heterogeneity at single-cell levels, ASGARD shows much better drug repurposing predictability than methods that rely on bulk samples.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Comparing ASGARD to other single-cell-based repurposing methods</head><p>We also compared single drug prediction using ASGARD with two other pipelines developed by Alakwaa et al. <ref type=\"bibr\" target=\"#b16\">17,</ref><ref type=\"bibr\" target=\"#b18\">19</ref> and Guo et al. <ref type=\"bibr\" target=\"#b16\">17,</ref><ref type=\"bibr\" target=\"#b18\">19</ref> , which were reported to handle scRNA-Seq data. Note that ASGARD offers more functionalities than those two methods. Alakwaa\\' and Guo\\' pipelines can only repurpose drug/compounds for every cluster, but not on a multi-cluster level. On the other hand, ASGARD can compute both the single-cluster-level drug significance and the multi-cluster novel drug score (Formula 1 in Methods). The above section shows that the ASGARD multi-cluster drug score shows AUCs of 0.92, 0.95, and 0.88 for breast cancer, leukemia, and and normal conditions (Supplementary Figure <ref type=\"figure\" target=\"#fig_5\">3B</ref>), and different proportions of disease cells among the scRNA-Seq data (Supplementary Figure <ref type=\"figure\" target=\"#fig_5\">3C</ref>).</p><p>We demonstrate that ASGARD is a promising drug recommendation pipeline through computational and clinical validation. In the following sections, we further illustrate the results of ASGARD applied to breast cancer, leukemia, and COVID-19, respectively.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Drug repurposing for breast cancers</head><p>To evaluate the reliability of ASGARD on breast cancer patient data, we downloaded four Triple Negative Breast Cancer (TNBC) samples along with four controls from the GSE161529 dataset 40</p><p>, in order to compare with the drug prediction results from the PDX models of TNBC described earlier. After the preprocessing procedure by Seurat, the TNBC samples contain an average of 5580 cells. We aligned all 8 samples, paired the cases vs. controls, and clustered them into 6 groups: B-cell, endothelial cell, epithelial cell, macrophage, T cell, and tissue stem cell (Figure <ref type=\"figure\" target=\"#fig_6\">4A</ref>). Epithelial cells are the largest group covering 45.98% of cells on average, while endothelial cells are the smallest group as expected, accounting for only 1.152% of total cells (Figure <ref type=\"figure\" target=\"#fig_6\">4A</ref>).</p><p>ASGARD predicted 13 drugs with significant FDR p-values in at least one of the four TNBC patients (Figure <ref type=\"figure\" target=\"#fig_6\">4B</ref>). For comparison, we also performed drug predictions on the 2 PDX models of TNBC patients, using the same procedures (Supplementary Figure <ref type=\"figure\" target=\"#fig_6\">4C</ref>). Of great interest, four of the most significant drugs from TNBC patients overlap with those predicted by the two PDX samples. These drugs are mebendazole, crizotinib, neratinib, and vinblastine (Figure <ref type=\"figure\" target=\"#fig_6\">4B</ref>). Both neratinib and vinblastine have been proven by the FDA for the treatment of breast cancer <ref type=\"bibr\" target=\"#b40\">41,</ref><ref type=\"bibr\" target=\"#b41\">42</ref> .</p><p>Mebendazole is a well-known anti-helminthic drug with wide clinical use. It has been reported to have anti-cancer properties in preclinical studies and has been in many clinical trial studies for treating various cancers, including liver, lung cancers, and glioma <ref type=\"bibr\" target=\"#b42\">43</ref> . Crizotinib is a receptor tyrosine kinase inhibitor showing tumor-reducing effects in vitro and in vivo <ref type=\"bibr\" target=\"#b43\">44,</ref><ref type=\"bibr\" target=\"#b44\">45</ref> . It is now in phase 2 clinical trial for treating patients with TNBC (ClinicalTrials.gov Identifier: NCT03620643).</p><p>To show quantitatively that ASGARD prediction on the TNBC samples is valuable, we next conducted two additional sets of analyses. First, we compared its results with those using TRANSACT <ref type=\"bibr\" target=\"#b35\">36</ref> , another computational method to calculate drug sensitivity. Since the TRANSACT can only predict drugs existing in the GDSC dataset <ref type=\"bibr\" target=\"#b45\">46</ref> , thus we can only compare the subset of drugs predicted by ASGARD in the GDSC dataset. As shown in Figure <ref type=\"figure\" target=\"#fig_6\">4C</ref>, ASGARD and TRANSACT results are well correlated. As ASGARD\\'s drug score increases, the drug sensitivity in TRANSACT also increases. Second, we investigate the effect of the tumor microenvironment on the drug scores. Thus we did in silico drop-one-out experiment, which excluded one cell type at a time. Among all cell types in the tumor microenvironment, T cell leads to the most drastic drug score changes, as well as the most variable drug score changes among different drugs (Figure <ref type=\"figure\" target=\"#fig_6\">4D</ref>). Moreover, the drug score changes also differ among the four TNBC patients, showing the sensitivity of ASGARD in personalized drug prediction.</p><p>To explore the potential molecular mechanisms of the top drug candidate, we next investigated the target genes and pathways of Mebendazole across the six cell clusters (Figure <ref type=\"figure\" target=\"#fig_6\">4E</ref> &amp;<ref type=\"figure\" target=\"#fig_6\">4F</ref>).</p><p>Mebendazole targets many important genes and pathways in TNBC, such as signal transducer and activator of transcription 1 (STAT1) in Toll-like receptor signaling pathway, Vascular Cell Adhesion Molecule 1 (VCAM1) in NF-kappa B signaling pathway, Matrix Metallopeptidase 14 (MMP14) in TNF signaling pathway, signal transducer and activator of transcription 2 (STAT2) in NOD-like receptor signaling pathway, cyclin-dependent kinase inhibitor 1A (CDKN1A) in PI3K-Akt signaling pathway, etc. These targeted genes and pathways are essential for the proliferation, migration, and invasion of TNBC cells <ref type=\"bibr\" target=\"#b46\">47,</ref><ref type=\"bibr\" target=\"#b47\">48</ref> , and were suggested as therapeutic drug targets for TNBC in previous studies <ref type=\"bibr\" target=\"#b48\">49,</ref><ref type=\"bibr\" target=\"#b49\">50</ref> .</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Drug repurposing for precursor T cell acute lymphoblastic leukemia (Pre-T ALL)</head><p>We further applied ASGARD to the collected scRNA-seq data from 2 Pre-T ALL patients and three normal healthy controls <ref type=\"bibr\" target=\"#b21\">22</ref> . ASGARD identifies eight types of cells (Figure <ref type=\"figure\" target=\"#fig_7\">5A</ref>), in which T cells are further clustered into four sub-populations (Figure <ref type=\"figure\" target=\"#fig_7\">5B</ref>). Cluster 1 (C1) is the largest one, covering 47.29% of cells, while cluster 4 (C4) is the smallest, accounting for only 2.11% of cells (Figure <ref type=\"figure\" target=\"#fig_7\">5B</ref>). The differentially expressed genes (adjusted P-value &lt; 0.05, Pre-T ALL vs.</p><p>normal) in the T cell clusters are significantly enriched in 6 pathways, including apoptosis, cell cycle, cGMPâˆ’PKG signaling, NFâˆ’kappa B signaling, p53 signaling, and T cell receptor signaling pathways (Figure <ref type=\"figure\" target=\"#fig_7\">5C</ref>).</p><p>Among the predicted drugs by ASGARD, the first candidate, tretinoin, has been approved for the treatment of leukemia <ref type=\"bibr\" target=\"#b50\">51</ref> (Figure <ref type=\"figure\" target=\"#fig_7\">5D</ref>, Supplementary Table <ref type=\"table\">2</ref>). Tretinoin is a vitamin A derivative. We further explored the potential molecular mechanisms of the FDA-approved top1 candidate tretinoin. Tretinoin targets many leukemia-related genes and all the significant pathways in the 4 T cell clusters, including: the regulator MDM4 in the p53 signaling pathway, cyclin D3 (CCND3) in cell cycle and p53 signaling pathways, G protein subunit alpha q (GNAQ)</p><p>and phospholipase C beta 1 (PLCB1) in the cGMPâˆ’PKG signaling pathway, Fos protooncogene (FOS) and p21 (RAC1) activated kinase 2 (PAK2) in the T cell receptor signaling pathway, spectrin alpha non-erythrocytic 1 (SPTAN1) in the apoptosis pathway, and zeta chain of T cell receptor-associated protein kinase 70 (ZAP70) in apoptosis and NFâˆ’kappa B signaling pathways (Figure <ref type=\"figure\" target=\"#fig_7\">5E</ref>). All these genes and pathways were previously shown to have significance in the pathogenesis of Pre-T ALL <ref type=\"bibr\" target=\"#b51\">[52]</ref><ref type=\"bibr\" target=\"#b52\">[53]</ref><ref type=\"bibr\" target=\"#b53\">[54]</ref> . The drug target genes and pathways in the T cell clusters explain why ASGARD predicts tretinoin for leukemia and how tretinoin treats leukemia.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Drug repurposing for severe patients with coronavirus disease 2019 (COVID-19)</head><p>cure rates exceeding 80% <ref type=\"bibr\" target=\"#b70\">71</ref> . The successful prediction of FDA-approved drugs supports the reliability of ASGARD.</p><p>Beyond the above described FDA-approved cases, ASGARD also predicts novel candidate drugs for breast cancer and leukemia. Crizotinib is a candidate from both TNBC patient and PDX model data (Figure <ref type=\"figure\" target=\"#fig_6\">4B</ref>, Supplementary Figure <ref type=\"figure\" target=\"#fig_6\">4C</ref>). Crizotinib is a receptor tyrosine kinase inhibitor that inhibits the growth, migration, and invasion of breast cancer cells in preclinical studies <ref type=\"bibr\" target=\"#b43\">44,</ref><ref type=\"bibr\" target=\"#b44\">45</ref> . A case report showed the TNBC patient harboring the ALK fusion mutation had a dramatic response to crizotinib treatment <ref type=\"bibr\" target=\"#b68\">69</ref> . It is also in a phase 2 clinical trial for treating patients with TNBC (ClinicalTrials.gov Identifier: NCT03620643). For leukemia, ASGARD predicts Vorinostat, a histone deacetylase (HDAC) inhibitor, as one of the novel candidate drugs. Vorinostat was approved by FDR for treating patients with progressive, persistent, or recurrent cutaneous T-cell lymphoma <ref type=\"bibr\" target=\"#b71\">72</ref> . It induces cell apoptosis in one T cell leukemia cell line in vitro <ref type=\"bibr\" target=\"#b72\">73</ref> , and improves the outcome of acute T cell lymphoblastic leukemia in animal models <ref type=\"bibr\" target=\"#b73\">74</ref> .</p><p>The results of a recent clinical trial (ClinicalTrials.gov Identifier: NCT04467931) show vorinostat is a promising candidate drug for T cell acute lymphoblastic leukemia <ref type=\"bibr\" target=\"#b74\">75</ref> .</p><p>Since ASGARD repurposes candidate drugs to reverse \"diseased\" cells to \"normal\" cells, it\\'s important to set proper controls according to the aim of the study. The best controls are arguably from the normal tissues of the same patient, and the next best ones are from the patients without such a disease but matched on other major confounders. Although consortiums such as Human Cell Atlas aim to obtain normal tissues from clinically healthy samples, it may not be easy to obtain normal samples for some diseases. Under such scenarios, samples from the very early stage of the disease or controls from tissues with the most relevant origin could be substitutes, until the data from the normal tissues are available. Additionally, a recent report has proposed using deep-learning based approach to identify the best reference control tissue, an attractive strategy that relies much less on prior-assumptions <ref type=\"bibr\" target=\"#b75\">76</ref> . Additionally, ASGARD built the drug reference using drug responses data from LINCS L1000 project <ref type=\"bibr\" target=\"#b28\">29</ref> , which were collected from 98 cell lines. We divided the drug reference into several tissue specific drug references according to the tissue origin of the cell lines. It\\'s highly recommended to select tissue specific drug references when using ASGARD. For example, for drug repurposing in breast cancer, it is best to use drug responses collected from breast cell lines. If there isn\\'t a proper cell-type specific reference for the target disease, it might be worthy identifying the cell line whose base-line gene expression profiles are most similar to \"control\" samples, after adjusting for systematic differences between cell line vs. primary tissues (eg. using transfer learning).</p><p>Altogether, this study shows clear evidence that ASGARD defines a single-cell-based reliable drug score for repurposing confident drugs, which were approved or in clinical trials for breast cancer, leukemia, and COVID-19, respectively. It also provides new applications for drugs that warrant further clinical studies. In all, ASGARD is a single-cell guided pipeline with significant potential to recommend repurposeful drugs.      </p><note type=\"other\">Figure Legends</note></div><figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_0\"><head></head><label></label><figDesc>-Breast-Cancer (TNBC) patients. Additionally, with the ongoing worldwide COVID-19 pandemic, we applied ASGARD to scRNA-seq data from severe COVID-19 patients and predicted potential therapies to reduce deaths of severe COVID-19 patients.MethodsSingle-cell RNA sequencing (scRNA-seq) dataWe obtained multiple scRNA-seq datasets from the Gene Expression Omnibus (GEO) database. ScRNA-seq data of cells from 4 Triple-Negative-Breast-Cancer (TNBC) patients and 4 healthy controls are from GSE161529. Epithelial cells from Patient-Derived Xenografts (PDXs) models of 2 patients with advanced metastatic TNBCs and adult human breast epithelial cells from 3 healthy women are from GEO with accession numbers GSE123926<ref type=\"bibr\" target=\"#b10\">11</ref> and GSE113197 21 , respectively. Another scRNA-seq pediatric bone marrow mononuclear cells (PBMMC) dataset from 2 Pre-T acute lymphoblastic leukemia patients and three healthy controls is from GEO with accession number GSE132509<ref type=\"bibr\" target=\"#b21\">22</ref> . The last set of scRNA-seq datasets are single cells from the bronchoalveolar lavage fluid (BALF) of 15 severe COVID-19 patients (4 deceased and 11 cured)</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_1\"><head>)</head><label></label><figDesc>In this formula, is a particular single-cell cluster, represents all disease-related (or selected)ð‘˜ ð‘› single-cell clusters, represents cellular proportion of the cluster in all diseased . Specifically, is the total number of cells in the sample and ð‘ð‘¢ð‘š(ð‘‡ð‘œð‘¡ð‘Žð‘™. ð¶ð‘’ð‘™ð‘™) is the number of cells in the cluster .is the drug\\'s FDR-adjusted p-value ð‘ð‘¢ð‘š(ð¶ð‘’ð‘™ð‘™) ð‘˜ ð‘˜ ð¹ð·ð‘… ð‘˜ (significance of reversed differential gene pattern) for cluster .</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_2\"><head></head><label></label><figDesc>. The adjusted Fisher\\'s combined P-value (FDR) is independent of the drug score. The FDR and drug score can be used together or independently for drug selection. By default, ASGARD uses the drug score for drug selection. Drugs with a higher value of drug score are supposed to have better therapeutic effects than those with a lower value.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_3\"><head>Figure 1 .</head><label>1</label><figDesc>Figure 1. The workflow of the ASGARD drug repurposing pipeline. The workflow of the</figDesc><graphic coords=\"26,73.50,109.10,468.00,501.75\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_4\"><head>Figure 2 .</head><label>2</label><figDesc>Figure 2. Comparing ASGARD to bulk-cell-based repurposing methods. The receiver</figDesc><graphic coords=\"27,73.50,207.99,468.00,327.75\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_5\"><head>Figure 3 .</head><label>3</label><figDesc>Figure 3. Comparing ASGARD to other single-cell-based repurposing methods.</figDesc><graphic coords=\"28,73.50,132.10,468.00,468.00\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_6\"><head>Figure 4 .</head><label>4</label><figDesc>Figure 4. Drug repurposing in TNBC patient samples. (A) UMAP plots of single-cell data</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_7\"><head>Figure 5 .</head><label>5</label><figDesc>Figure 5. Drug repurposing for precursor T cell acute lymphoblastic leukemia (Pre-T</figDesc><graphic coords=\"32,73.50,73.50,468.00,389.25\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_8\"><head>Figure 6 .</head><label>6</label><figDesc>Figure 6. Drug repurposing for reducing mortality of severe COVID-19 patients. (A)</figDesc><graphic coords=\"33,73.50,73.50,468.00,303.75\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\"><head></head><label></label><figDesc></figDesc><graphic coords=\"34,73.50,109.10,468.00,263.25\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\"><head></head><label></label><figDesc></figDesc><graphic coords=\"35,73.50,73.50,468.00,597.00\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\"><head></head><label></label><figDesc></figDesc><graphic coords=\"36,73.50,132.10,468.00,312.00\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\"><head></head><label></label><figDesc></figDesc><graphic coords=\"37,73.50,73.50,468.00,453.75\" type=\"bitmap\" /></figure>\\n\\t\\t</body>\\n\\t\\t<back>\\n\\n\\t\\t\\t<div type=\"availability\">\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Code availability</head><p>ASGARD is available as an R package in Github (https://github.com/lanagarmire/ASGARD) under the PolyForm Noncommercial License. The true positive datasets, results, and scripts for</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Data and materials availability</head><p>ScRNA-Seq data are available in Gene Expression Omnibus (Accession number: GSE161529,</p></div>\\n\\t\\t\\t</div>\\n\\n\\n\\t\\t\\t<div type=\"funding\">\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Funding</head><p>This research was supported by the National Institute of Environmental Health Sciences through funds provided by the trans-NIH Big Data to Knowledge (BD2K) initiative [K01ES025434]; the US National Library of Medicine [R01 LM012373, R01 LM12907]; and the National Institute of Child Health and Human Development [R01 HD084633; to L.X. Garmire].</p></div>\\n\\t\\t\\t</div>\\n\\n\\t\\t\\t<div type=\"annex\">\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><p>COVID-19, respectively (Figure <ref type=\"figure\">2B</ref>). For a fair comparison, we further tested the single-cluster-level drug prediction accuracies of these three methods (Figure <ref type=\"figure\">3</ref>, Supplementary Figure <ref type=\"figure\">2</ref>). Even at the single-cluster-level, ASGARD still shows the best AUCs on every individual cluster from breast cancer, leukemia, and COVID-19 datasets (Figure <ref type=\"figure\">3</ref>). On the 8 clusters of the breast cancer dataset, ASGARD yields an averaged AUC of 0.83 (0.80-0.86), significantly better (P-value&lt;0.001, student\\'s t-test) than Alakwaa\\' and Guo\\' pipelines, with averaged AUC values of 0.72 (0.62-0.79) and 0.56 (0.54-0.59) respectively (Figure <ref type=\"figure\">3A</ref>, Supplementary Figure <ref type=\"figure\">2</ref>). On the 4 clusters of precursor T cell acute lymphoblastic leukemia data, ASGARD has an averaged AUC of 0.81 (0.76-0.85), again significantly better (P-value &lt;0.001, student\\'s t-test) than Alakwaa\\' and Guo\\' pipelines, with averaged AUC values of 0.51 (0.49-0.56) and 0.52 (0.49-0.55) respectively (Figure <ref type=\"figure\">3B</ref>, Supplementary Figure <ref type=\"figure\">2</ref>). Similar trends exist in the neutrophile, NK, T cell and monocytes that have increased cell proportions in the decreased severe vs. cured severe COVID-19 patients. While ASGARD achieves average AUCs of 0.82 (0.77-0.88), Alakwaa\\' and Guo\\' methods have reduced average AUCs of 0.72 (0.63-0.80), and 0.58 (0.55-0.62) (Figure <ref type=\"figure\">3C</ref>, Supplementary Figure <ref type=\"figure\">2</ref>). These results support the conclusion that ASGARD predicts drugs more accurately than Alakwaa\\' and Guo\\' pipelines.</p><p>Additionally, given that sample size, cell population similarity, and proportion of disease cells impact significantly on differential gene analysis <ref type=\"bibr\" target=\"#b38\">39</ref> , we further performed robustness assessments of the three pipelines across different sizes of single-cell populations, different differential levels of single-cell populations, and different proportions of diseased cells using simulation data based on GSE123926 and GSE113197 dataset (see Methods). The AUCs of the three single-cell drug repurposing pipelines on the simulation data show that ASGARD, as well as the other two pipelines, have very robust performance across different sizes of single-cell populations (Supplementary Figure <ref type=\"figure\">3A</ref>), different degrees of DE between disease The immune response activated by the SARS-CoV-2 virus infection is a double-edged sword. It protects the human body from viral infection. But the deregulated immune response in severe COVID-19 patients damages the alveolar to cause respiratory failure that kills the patients <ref type=\"bibr\" target=\"#b54\">55,</ref><ref type=\"bibr\" target=\"#b55\">56</ref> .</p><p>To find drugs that may help to reduce the mortality of severe COVID-19, we collected scRNA-seq data from the bronchoalveolar lavage fluid (BALF) of 15 severe COVID-19 patients 13,23   . Among them, 11 patients were cured (cured severe), while four died (deceased severe) afterward. To identify immune cells that correlate with the death of severe patients, we compared the scRNA-seq data between deceased severe and cured severe patients. In total, there are seven types of cells, including six types of immune cells and epithelial cell types (Figure <ref type=\"figure\">6A</ref>), in the BALF samples collected from severe COVID-19 patients. Monocyte is the largest T cell population in both deceased and cured severe COVID-19 patients(Figure <ref type=\"figure\">6B</ref>).</p><p>The population of neutrophil, NK cell, T cell, and monocyte increased in deceased severe COVID patients compared to the cured ones, suggesting the important role of these four types of cells in COVID-19-related death <ref type=\"bibr\" target=\"#b56\">[57]</ref><ref type=\"bibr\" target=\"#b57\">[58]</ref><ref type=\"bibr\" target=\"#b58\">[59]</ref><ref type=\"bibr\" target=\"#b59\">[60]</ref> (Figure <ref type=\"figure\">6B</ref>). The differentially expressed genes We identified the differential gene expression profiles of the four cell types, including neutrophil, NK cell, T cell, and monocyte, by comparing decreased severe patients to cured severe ones.</p><p>Then we put the differential gene expression profiles to ASGARD to identify drug candidates using the multi-cluster drug score. Among the predicted drugs, rescinnamine (2nd) and enalapril (4th) caught our attention (Figure <ref type=\"figure\">6D</ref>, Supplementary Table <ref type=\"table\">3</ref>). Both rescinnamine and enalapril are angiotensin-converting enzyme (ACE) inhibitors. Angiotensin-converting enzyme 2 (ACE2) mediates the SARS-CoV-2 cell entry. It\\'s interesting to see rescinnamine and enalapril are predicted by ASGARD for treating severe COVID-19. So we further explored their target genes and pathways in the four cell types. Rescinnamine and enalapril share most of the key genes on all the significant pathways in monocyte, NK cell, neutrophil, and T cell, respectively (Figure <ref type=\"figure\">6E</ref>). In monocyte, rescinnamine and enalapril share 47 key target genes, including Janus In T cell, rescinnamine and enalapril share 30 key target genes, such as CCL2, CCL8, C-X-C Motif Chemokine Ligand 9 (CXCL9), JAK3, TNF, and Lymphocyte Cytosolic Protein 2 (LCP2), and all the 6 significant pathways of T cell. The shared target genes and pathways in corresponding cells were previously shown related to death from COVID-19 <ref type=\"bibr\" target=\"#b56\">[57]</ref><ref type=\"bibr\" target=\"#b57\">[58]</ref><ref type=\"bibr\" target=\"#b58\">[59]</ref><ref type=\"bibr\" target=\"#b59\">[60]</ref> .</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Discussion</head><p>This study presents a Single-cell Guided pipeline to Aid Repurposing of Drugs (ASGARD) as a new generation of personalized drug recommendation system. To evaluate the accuracy of ASGARD in single drug repurposing, we compared ASGARD to other repurposing methods that utilize bulk cell RNA-Seq (CLUE and DrInsight) or single-cell RNA-Seq data (Alakwaa\\'s and Guo\\'s) on a variety of diseases, including breast cancer, leukemia, and COVID-19. ASGARD performs much better than all these methods in predicting drugs/compounds (Figure <ref type=\"figure\">2</ref>, 3, Supplementary Figure <ref type=\"figure\">2</ref>). The performance of ASGARD is also robust across different sizes and proportions of cell populations, as well as differential expression levels (Supplementary Figure <ref type=\"figure\">3</ref>). Moreover, we highlight that ASGARD defines a novel drug score to summarize drug efficiency across multiple selected cell clusters. These important functions are missing in other simpler single-cell RNA-Seq drug reposition pipelines by Alakwaa and Guo. Both Alakwaa\\' and Luo\\'s pipelines use the CLUE platform, a cloud-based platform developed by the LINCS Center for signature-gene based drug ranking <ref type=\"bibr\" target=\"#b60\">61</ref> . Additionally, Luo\\'s method uses log fold change as the additional threshold to filter the gene query. ASGARD on each cluster is related to DrInsight, a concordantly expressed genes (CEG) based, enhanced drug repurposing method compared to other signature-based searching methods <ref type=\"bibr\" target=\"#b29\">30</ref> . It uses order statistics to directly measure the concordance (e.g. inverse association) between the disease data and drug-perturbed data and identifies concordantly expressed genes (CEGs). CEGS are used as features to further formulate an outlier sum statistic for drug selection, rather than the connectivity score (usually -90) based cut-off for drug selection. The CEG and outlier sum statistics contribute to higher performance in ASGARD.</p><p>ASGARD achieves drug ranking for the disease/patient by a novel drug score that evaluates the treatment efficacy across the user-selected cell clusters (Formula 1 in Methods). The prediction using the multi-cluster drug score shows a significantly (P-value &lt; 0.05, student\\'s t-test) better AUC than the prediction based on individual clusters (Figure <ref type=\"figure\">2</ref>, 3). It suggests that targeting an individual cell cluster isn\\'t sufficient for successful drug prediction. Instead, targeting multiple essential diseased cell clusters is a more appropriate strategy for drug prediction. On the other hand, it is not ideal to propose drug repurposing using bulk RNA-seq, a mixture of all cells, as done by traditional methods (e.g. CLUE and DrInsight). Significant heterogeneity exists in different T cell populations; not all these cells play equal roles in the diseases <ref type=\"bibr\" target=\"#b61\">62,</ref><ref type=\"bibr\" target=\"#b62\">63</ref> , reflected by different gene expression responses to drug treatment <ref type=\"bibr\" target=\"#b63\">64</ref> . ASGARD can distinguish more important T cell types from others and repurpose drugs accordingly, explaining why ASGARD has significantly (P-value &lt;0.05, student\\'s t-test) better AUC performance than traditional bulk methods (Figure <ref type=\"figure\">2B</ref>). Moreover, ASGARD also demonstrates variations in drug scores across different patients (Figure <ref type=\"figure\">4B</ref>, 5D, and 6D). This result stresses that personalized therapy is necessary for the best therapeutic effect, and utilizing single-cell sequencing information may help to achieve that. Sparsity and heterogeneity are two major challenges in analyzing single-cell data, and usually cause false discoveries of differentially expressed genes <ref type=\"bibr\" target=\"#b64\">65</ref> . Previous benchmark study showed that Seurat 18 , DESeq2 27 , edgeR <ref type=\"bibr\" target=\"#b27\">28</ref> , and Limma 26 are among the top methods in discovering differentially expressed genes using single-cell data <ref type=\"bibr\" target=\"#b36\">37</ref> . We here compared the effect of these methods and different parameterization on the downstream drug repurposing, using AUC metric. AUC performance still varies with methods for single-cell differential expression (Figure <ref type=\"figure\">2A</ref>, Limma (Bayes) method showed the best average AUC performance compared to all other three methods. Within limma method, approaches that model mean-variance with the empirical Bayes approach (limma Bayes and limma trend) showed better AUCs than that with the precision weights approach (limma voom). Similar observations were observed in some of the comparisons in a benchmark study of single-cell differential expression <ref type=\"bibr\" target=\"#b36\">37</ref> . The empirical Bayes approach is usually more powerful than the precision weights approach when the library sizes are not quite variable between samples <ref type=\"bibr\" target=\"#b65\">66</ref> . Seurat, a method widely used in single-cell studies, has the 2nd best AUCs in general. In particular, the default mode of Wilcoxon rank-sum test in Seurat has slightly better average AUC than the t-test and logistic regression (LR) modes. Our comparison revealed that DE methods should be carefully selected according to the status of the dataset to achieve the best performance. Accordingly, ASGARD was designed as a flexible framework supporting various methods for single-cell differential gene expression analysis.</p><p>We chose breast cancer or leukemia datasets to illustrate the utilities of ASGARD, given the relative abundance of prior drug knowledge. FDA has approved many drugs predicted by ASGARD, such as neratinib and vinblastine for treating breast cancer <ref type=\"bibr\" target=\"#b66\">67,</ref><ref type=\"bibr\" target=\"#b67\">68</ref> (Figure <ref type=\"figure\">4B</ref>, Supplementary Figure <ref type=\"figure\">4C</ref>), and tretinoin for treating leukemia <ref type=\"bibr\" target=\"#b50\">51</ref> (Figure <ref type=\"figure\">5D</ref>). Vinblastine and neratinib were predicted for breast cancer in both TNBC patient and PDX datasets. Vinblastine is a vinca alkaloid that has been used in the treatment of metastatic breast cancer since the early 1980s. The regimen of vinblastine/mitomycin is an effective salvage regimen and an excellent first-line chemotherapeutic treatment for women with metastatic breast cancer <ref type=\"bibr\" target=\"#b41\">42</ref> .</p><p>Neratinib is a protein kinase inhibitor that was approved in July 2017 as an extended adjuvant therapy in breast cancer <ref type=\"bibr\" target=\"#b67\">68</ref> . Recently, a randomized phase III clinical trial of 621 patients from 28 countries showed neratinib significantly improved the progression-free survival of patients with advanced breast cancer <ref type=\"bibr\" target=\"#b68\">69</ref> . Tretinoin, also known as all-trans-retinoic acid (ATRA), is the first candidate predicted by ASGARD for leukemia. Tretinoin targets all significant pathways, such as p53 signaling, cell cycle, and apoptosis pathways, for each diseased cell cluster in leukemia patients (Figure <ref type=\"figure\">5E</ref>). These pathways play important roles in the survival of leukemia patients <ref type=\"bibr\" target=\"#b53\">54</ref> .</p><p>Consistent with our prediction, tretinoin was approved by the FDA to induce remission in patients with acute leukemia <ref type=\"bibr\" target=\"#b50\">51</ref> . Tretinoin significantly improves the survival of acute leukemia <ref type=\"bibr\" target=\"#b69\">70</ref> .</p><p>Tretinoin with chemotherapy has become the standard treatment for acute leukemia, resulting in</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Competing interests</head><p>The authors declare that they have no competing interests. GSE123926, GSE113197, GSE132509, GSE158055, and GSE145926). Phase I LINCS L1000 data are available in Gene Expression Omnibus (Accession number: GSE92742). Phase II LINCS L1000 data are available in Gene Expression Omnibus (Accession number GSE70138).</p><p>ASGARD is available as an R package on GitHub (https://github.com/lanagarmire/ASGARD).</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Scripts</head><p>used in this study are available on GitHub (https://github.com/lanagarmire/Single-cell-drug-repositioning). We collected scRNA-seq data from 24,741 epithelial cells of advanced metastatic breast cancer Patient-Derived Xenografts (PDXs) models <ref type=\"bibr\" target=\"#b10\">11</ref>   <ref type=\"figure\">3B</ref>). Cluster 7 (C7) has the largest number of 7 significant pathways, while C1 and C6 each have only 1 significant pathway.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Supplementary Materials</head><note type=\"other\">Supplementary Figure 1</note><p>We first applied ASGARD for multi-cluster drug repurposing prediction and predicted 11 drugs (FDR&lt;0.05 and overall drug score &gt;0.99 quantiles) for advanced metastatic breast cancer (Supplementary Figure <ref type=\"figure\">3C</ref>, Supplementary Table <ref type=\"table\">4</ref>). Fostamatinib is the top 1 drug candidate (Supplementary Figure <ref type=\"figure\">3C</ref>). It is a tyrosine kinase inhibitor medication approved for the treatment of chronic immune thrombocytopenia 77 . Colchicine, the second best candidate, is an alkaloid approved for treating the inflammatory symptoms of familial Mediterranean fever 78 .</p><p>Both fostamatinib and colchicine have shown antitumor and anti-metastasis effects in animal models of breast cancer 79,80 . Moreover, the 4th candidate fulvestrant and 7th candidate neratinib have been approved by the Food and Drug Administration (FDA) for breast cancer treatment <ref type=\"bibr\" target=\"#b57\">58,</ref><ref type=\"bibr\" target=\"#b58\">59</ref> .</p><p>To explore the potential molecular mechanisms of the top 2 candidates, we next investigated the target genes and pathways of fostamatinib and colchicine across the eight cell clusters (Supplementary Figure <ref type=\"figure\">3D</ref>). Fostamatinib and colchicine both target all the significant pathways in each cluster. Fostamatinib and colchicine are complementary in targeting genes of these pathways. Among the 143 target genes from these significant pathways, only 29 target genes are shared by fostamatinib and colchicine (Supplementary Figure <ref type=\"figure\">3D</ref>). The fostamatinib and colchicine also show biologically synergistic targeting of multiple genes on the same significant pathways. For example, fostamatinib inhibits Cyclin D1 (CCND1) to produce G1 arrest in the p53 signaling pathway, while colchicine inhibits Cyclin-dependent kinase 1 (CDK1) to produce G2 arrest in the p53 signaling pathway and cell cycle pathway 81 (Supplementary Figure <ref type=\"figure\">3D</ref>).</p><p>Additionally, the drug scores of top drug candidates vary from one PDX model to another (Supplementary Figure <ref type=\"figure\">3D</ref>), demonstrating that ASGARD is a forward-looking precision medicine strategy in silico. </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Supplementary</head></div>\\t\\t\\t</div>\\n\\t\\t\\t<div type=\"references\">\\n\\n\\t\\t\\t\\t<listBibl>\\n\\n<biblStruct xml:id=\"b0\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Tumour heterogeneity and resistance to cancer therapies</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Dagogo-Jack</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">T</forename><surname>Shaw</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Rev. Clin. Oncol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">15</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"81\" to=\"94\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b1\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Heterogeneity of Alzheimer&apos;s disease: consequence for drug trials?</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Devi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Scheltens</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Alzheimers. Res. Ther</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">122</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b2\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Cerebrovascular Disease: Primary and Secondary Stroke Prevention</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">Z</forename><surname>Caprio</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">A</forename><surname>Sorond</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Med. Clin. North Am</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">103</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"295\" to=\"308\" />\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b3\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">One size does not fit all -Patterns of vulnerability and resilience in the COVID-19 pandemic and why heterogeneity of disease matters</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Sominsky</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">W</forename><surname>Walker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">J</forename><surname>Spencer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Brain Behav. Immun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">87</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1\" to=\"3\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b4\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Genomic Heterogeneity as a Barrier to Precision Medicine in Gastroesophageal Adenocarcinoma</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Pectasides</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cancer Discov</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">8</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"37\" to=\"48\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b5\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Heterogeneity in Colorectal Cancer: A Challenge for Personalized Medicine?</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Molinari</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Int. J. Mol. Sci</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">19</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b6\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Harnessing big &apos;omics&apos; data and AI for drug discovery in hepatocellular carcinoma</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Rev. Gastroenterol. Hepatol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">17</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"238\" to=\"251\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b7\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Single-cell analysis tools for drug discovery and development</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">R</forename><surname>Heath</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Ribas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">S</forename><surname>Mischel</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Rev. Drug Discov</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">15</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"204\" to=\"216\" />\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b8\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Evaluation of Cell Type Annotation R Packages on Single-cell RNA-seq Data</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Du</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">X</forename><surname>Garmire</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno type=\"DOI\">10.1016/j.gpb.2020.07.004</idno>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Genomics Proteomics Bioinformatics</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b9\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Using single-cell multiple omics approaches to resolve tumor heterogeneity</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Ortega</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Clin. Transl. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">6</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">46</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b10\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Barcoding reveals complex clonal behavior in patient-derived xenografts of metastatic triple negative breast cancer</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Merino</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">766</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b11\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Intratumoral heterogeneity and clonal evolution in liver cancer</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Losic</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">11</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">291</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b12\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Single-cell landscape of bronchoalveolar immune cells in patients with COVID-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Liao</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">26</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"842\" to=\"844\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b13\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Eleven grand challenges in single-cell data science</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>LÃ¤hnemann</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Genome Biol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">21</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">31</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b14\">\\n\\t<monogr>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">X</forename><surname>Garmire</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G.-C</forename><surname>Yuan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Fan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">W</forename><surname>Yeo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Quackenbush</surname></persName>\\n\\t\\t</author>\\n\\t\\t<title level=\"m\">SINGLE CELL ANALYSIS, WHAT IS IN THE FUTURE? in Biocomputing</title>\\n\\t\\t\\t\\t<imprint>\\n\\t\\t\\t<publisher>WORLD SCIENTIFIC</publisher>\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2019. 2018</date>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"332\" to=\"337\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b15\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Drug repurposing: progress, challenges and recommendations</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Pushpakom</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Rev. Drug Discov</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">18</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"41\" to=\"58\" />\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b16\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Repurposing Didanosine as a Potential Treatment for COVID-19 Using Single-Cell RNA Sequencing Data. mSystems 5</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">M</forename><surname>Alakwaa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b17\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Comprehensive Integration of Single-Cell Data</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Stuart</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">177</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e21</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b18\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Identification of Repurposal Drugs and Adverse Drug Reactions for Various Courses of Coronavirus Disease 2019 (COVID-19) Based on Single-cell RNA Sequencing Data</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2005\">2005. 2020</date>\\n\\t\\t\\t<biblScope unit=\"page\">7856</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b19\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Combination therapeutics in complex diseases</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>He</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Cell. Mol. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">20</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2231\" to=\"2240\" />\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b20\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Profiling human breast epithelial cells using single cell RNA sequencing identifies cell diversity</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><forename type=\"middle\">H</forename><surname>Nguyen</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">9</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">2028</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b21\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Single-cell analysis of childhood leukemia reveals a link between developmental states and ribosomal protein expression as a source of intra-individual heterogeneity</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Caron</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Sci. Rep</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">8079</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b22\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">COVID-19 immune features revealed by a large-scale single-cell transcriptome atlas</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Ren</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">184</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e19</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b23\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">A single-cell RNA expression atlas of normal, preneoplastic and tumorigenic states in the human breast</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Pal</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">EMBO J</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">40</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e107333</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b24\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Aran</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Immunol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">20</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"163\" to=\"172\" />\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b25\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">limma powers differential expression analyses for RNA-sequencing and microarray studies</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Ritchie</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nucleic Acids Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">43</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e47</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2015\">2015</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b26\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">I</forename><surname>Love</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Huber</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Anders</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Genome Biol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">15</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">550</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2014\">2014</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b27\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">J</forename><surname>Mccarthy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">K</forename><surname>Smyth</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nucleic Acids Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">40</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"4288\" to=\"4297\" />\\n\\t\\t\\t<date type=\"published\" when=\"2012\">2012</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b28\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Subramanian</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">171</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e17</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b29\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Breaking the paradigm: Dr Insight empowers signature-free, enhanced drug repurposing</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Chan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Turner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">E</forename><surname>Baldwin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Gu</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Bioinformatics</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">35</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2818\" to=\"2826\" />\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b30\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Outlier sums for differential gene expression analysis</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Tibshirani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Hastie</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Biostatistics</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">8</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2\" to=\"8\" />\\n\\t\\t\\t<date type=\"published\" when=\"2007\">2007</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b31\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Prediction of repurposed drugs for treating lung injury in COVID-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>He</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Garmire</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">F1000Res. 2020</date>\\n\\t\\t\\t<biblScope unit=\"volume\">9</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">609</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b32\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Powerful p-value combination methods to detect incomplete association</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Yoon</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Baik</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Park</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Nam</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Sci. Rep</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">11</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">6980</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b33\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">pROC: an open-source package for R and S+ to analyze and compare ROC curves</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Robin</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">BMC Bioinformatics</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">77</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2011\">2011</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b34\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\">R Foundation for Statistical Computing, R. C. R: A language and environment for statistical computing. R Foundation for Statistical Computing</title>\\n\\t\\t\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b35\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Predicting clinical drug response from model systems by non-linear subspace-based transfer learning</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Mourragui</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno type=\"DOI\">10.1101/2020.06.29.177139</idno>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n\\t<note type=\"report_type\">Preprint at</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b36\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Confronting false discoveries in single-cell differential expression</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">W</forename><surname>Squair</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">5692</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b37\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">The Drug Repurposing Hub: a next-generation drug library and information resource</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">M</forename><surname>Corsello</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">23</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"405\" to=\"408\" />\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b38\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Power analysis and sample size estimation for RNA-Seq differential expression</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Ching</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">X</forename><surname>Garmire</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">RNA</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">20</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1684\" to=\"1696\" />\\n\\t\\t\\t<date type=\"published\" when=\"2014\">2014</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b39\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">A single-cell RNA expression atlas of normal, preneoplastic and tumorigenic states in the human breast</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Pal</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">EMBO J</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">40</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e107333</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b40\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Neratinib approved by FDA for breast cancer</title>\\n\\t\\t<ptr target=\"https://www.cancer.gov/news-events/cancer-currents-blog/2017/neratinib-breast-cancer-fda\" />\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">National Cancer Institute</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b41\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">First-line and salvage therapy of metastatic breast cancer with mitomycin/vinblastine</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">M</forename><surname>Sedlacek</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Oncology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">50</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"16\" to=\"21\" />\\n\\t\\t\\t<date type=\"published\" when=\"1993\">1993</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>Suppl</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b42\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Albendazole and Mebendazole as Anti-Parasitic and Anti-Cancer Agents: an Update</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J.-Y</forename><surname>Chai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B.-K</forename><surname>Jung</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-J</forename><surname>Hong</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Korean J. Parasitol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">59</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"189\" to=\"225\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b43\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Crizotinib, a MET inhibitor, inhibits growth, migration, and invasion of breast cancer cells in vitro and synergizes with chemotherapeutic agents</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">M</forename><surname>Ayoub</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>Al-Shami</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Alqudah</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">M</forename><surname>Mhaidat</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Onco. Targets. Ther</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"4869\" to=\"4883\" />\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b44\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Single oral dose acute and subacute toxicity of a c-MET tyrosine kinase inhibitor and CDK 4/6 inhibitor combination drug therapy</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Smith</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Am. J. Cancer Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">8</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"183\" to=\"191\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b45\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Genomics of Drug Sensitivity in Cancer (GDSC): a resource for therapeutic biomarker discovery in cancer cells</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nucleic Acids Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">41</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"D955\" to=\"D961\" />\\n\\t\\t\\t<date type=\"published\" when=\"2013\">2013</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b46\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Targeting the PI3K/AKT/mTOR pathway in triple-negative breast cancer: a review</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L B</forename><surname>Costa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">S</forename><surname>Han</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">J</forename><surname>Gradishar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Breast Cancer Res. Treat</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">169</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"397\" to=\"406\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b47\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Dual inhibition of STAT1 and STAT3 activation downregulates expression of PD-L1 in human breast cancer cells</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Sasidharan</forename><surname>Nair</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Toor</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">M</forename><surname>Ali</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">R</forename><surname>Elkord</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Expert Opin. Ther. Targets</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">22</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"547\" to=\"557\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b48\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">The potential and controversy of targeting STAT family members in cancer</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Verhoeven</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Semin. Cancer Biol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">60</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"41\" to=\"56\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b49\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">A novel immunotherapy targeting MMP-14 limits hypoxia, immune suppression and metastasis in triple-negative breast cancer models</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Ling</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Oncotarget</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">8</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"58372\" to=\"58385\" />\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b50\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Tretinoin in the treatment of acute promyelocytic leukemia</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">M</forename><surname>Wong</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cancer Pract</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">4</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"220\" to=\"223\" />\\n\\t\\t\\t<date type=\"published\" when=\"1996\">1996</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b51\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Therapeutic targeting of the cyclin D3:CDK4/6 complex in T cell leukemia</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M</forename><surname>Sawai</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cancer Cell</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">22</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"452\" to=\"465\" />\\n\\t\\t\\t<date type=\"published\" when=\"2012\">2012</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b52\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Activation of endogenous c-fos proto-oncogene expression by human T-cell leukemia virus type I-encoded p40tax protein in the human T-cell line</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Nagata</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ohtani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Nakamura</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sugamura</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Jurkat. J. Virol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">63</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"3220\" to=\"3226\" />\\n\\t\\t\\t<date type=\"published\" when=\"1989\">1989</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b53\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">T-cell acute lymphoblastic leukemia</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">A</forename><surname>Raetz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">T</forename><surname>Teachey</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Hematology Am. Soc. Hematol. Educ. Program</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"580\" to=\"588\" />\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016. 2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b54\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Causes of death and comorbidities in hospitalized patients with COVID-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Elezkurtaj</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Sci. Rep</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">11</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">4263</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b55\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Uncontrolled Innate and Impaired Adaptive Immune Responses in Patients with COVID-19 Acute Respiratory Distress Syndrome</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Hue</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Am. J. Respir. Crit. Care Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">202</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1509\" to=\"1519\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b56\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Natural killer cells associated with SARS-CoV-2 viral RNA shedding, antibody response and mortality in COVID-19 patients</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Bao</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Exp. Hematol. Oncol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">5</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b57\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Monocyte-driven atypical cytokine storm and aberrant neutrophil activation as key mediators of COVID-19 disease severity</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Vanderbeke</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">4117</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b58\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">T cells in COVID-19 -united in diversity</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Swadling</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">K</forename><surname>Maini</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature immunology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">21</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1307\" to=\"1308\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b59\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Neutrophil Extracellular Traps as Prognostic Markers in COVID-19: A Welcome Piece to the Puzzle</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">S</forename><surname>Ondracek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><forename type=\"middle\">M</forename><surname>Lang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Arteriosclerosis, thrombosis, and vascular biology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">41</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"995\" to=\"998\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b60\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">The Connectivity Map: using gene-expression signatures to connect small molecules, genes, and disease</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lamb</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Science</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">313</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1929\" to=\"1935\" />\\n\\t\\t\\t<date type=\"published\" when=\"2006\">2006</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b61\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Detecting heterogeneity in single-cell RNA-Seq data by non-negative matrix factorization</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Ching</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Pan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">M</forename><surname>Weissman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Garmire</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">PeerJ</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">5</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e2888</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b62\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Intercellular signaling dynamics from a single cell atlas of the biomaterials response</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Cherry</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno type=\"DOI\">10.1101/2020.07.24.218537</idno>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b63\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Drug discovery in traditional Chinese medicine: from herbal fufang to combinatory drugs</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>He</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Science</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2015\">2015</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b64\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Bayesian approach to single-cell differential expression analysis</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">V</forename><surname>Kharchenko</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Silberstein</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">T</forename><surname>Scadden</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Methods</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">11</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"740\" to=\"742\" />\\n\\t\\t\\t<date type=\"published\" when=\"2014\">2014</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b65\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Precision weights unlock linear model analysis tools for RNA-seq read counts</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">W</forename><surname>Law</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Shi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">K</forename><surname>Smyth</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Voom</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Genome Biol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">15</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">R29</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2014\">2014</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b66\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Farooq</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">P</forename><surname>Patel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Fulvestrant</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t\\t<pubPlace>StatPearls</pubPlace>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b67\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Neratinib for breast cancer. Aust Prescr</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">42</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"209\" to=\"210\" />\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b68\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Neratinib Plus Capecitabine Versus Lapatinib Plus Capecitabine in HER2-Positive Metastatic Breast Cancer Previously Treated With â‰¥ 2 HER2-Directed Regimens: Phase III NALA Trial</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Saura</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Clin. Oncol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">38</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"3138\" to=\"3149\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b69\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Improvement of prognosis in refractory and relapsed acute promyelocytic leukemia over recent years: the role of all-trans retinoic acid therapy</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Thomas</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Ann. Hematol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">75</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"195\" to=\"200\" />\\n\\t\\t\\t<date type=\"published\" when=\"1997\">1997</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b70\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Retinoic acid and arsenic trioxide for acute promyelocytic leukemia</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Lo-Coco</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">369</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"111\" to=\"121\" />\\n\\t\\t\\t<date type=\"published\" when=\"2013\">2013</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b71\">\\n\\t<monogr>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Vorinostat</surname></persName>\\n\\t\\t</author>\\n\\t\\t<title level=\"m\">LiverTox: Clinical and Research Information on Drug-Induced Liver Injury (National Institute of Diabetes and Digestive and Kidney Diseases</title>\\n\\t\\t\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b72\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Therapeutic potential and functional interaction of carfilzomib and vorinostat in T-cell leukemia/lymphoma</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Oncotarget</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">7</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"29102\" to=\"29115\" />\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b73\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Vorinostat and quinacrine have synergistic effects in T-cell acute lymphoblastic leukemia through reactive oxygen species increase and mitophagy inhibition</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Jing</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell Death Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">9</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">589</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b74\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Phase 1 study of the Aurora kinase A inhibitor alisertib (MLN8237) combined with the histone deacetylase inhibitor vorinostat in lymphoid malignancies</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Siddiqi</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Leuk. Lymphoma</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">61</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"309\" to=\"317\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b75\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">OCTAD: an open workspace for virtually screening therapeutics targeting precise cancer patient groups using gene expression features</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Zeng</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Protoc</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">16</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"728\" to=\"753\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n\\t\\t\\t\\t</listBibl>\\n\\t\\t\\t</div>\\n\\t\\t</back>\\n\\t</text>\\n</TEI>\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check output for the first exception in futures_map dictionary above with Exception('[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1')."
      ],
      "metadata": {
        "id": "86Z1rat1VDic"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "dl_pdf_xml[3]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        },
        "outputId": "be74f9ae-2b98-43c2-df03-7b1f12b6de21",
        "id": "07QviQ1F62kX"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'1'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "EtYIxblRxfPK"
      },
      "source": [
        "## 5. Clean and return XML as Beautiful Soup object\n",
        "\n",
        "Function to remove XML namespaces (xmlns) attribute and return the cleaned, parsed text as a [Beautiful Soup](https://www.crummy.com/software/BeautifulSoup/bs4/doc/) object. This object represents the document as a nested data structure so that we can navigate and extract text using XML tags.\n",
        "\n",
        "The simplest way to navigate the parse tree is to find a tag by name so we will use the find() method to return the XML for the `<body>` tag content.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "cQeG0S_lxfPL"
      },
      "outputs": [],
      "source": [
        "# remove xmlns references from XML and convert to soup object\n",
        "\n",
        "def soupify(text):\n",
        "    cleaned_text = re.sub('\\s*xmlns(:\\w+)?=\\\"[^\\\"]*\\\"', '', text)\n",
        "    return  BeautifulSoup(cleaned_text, 'lxml-xml').find(\"body\")"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View output for a 'soupified' article."
      ],
      "metadata": {
        "id": "mzW-TBp6g2yJ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soupify_article_6 = soupify(dl_pdf_xml[6])\n",
        "soupify_article_6"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RIJC1Nrq24Zv",
        "outputId": "cba16e17-1da4-4486-af3a-5001ec87c6ef"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<body>\n",
              "<div><head>INTRODUCTION</head><p>The global impact of the COVID-19 pandemic on both human health and socioeconomic activity has brought to light the importance of a nuanced understanding of the way that epidemics spread in cities. Urban spread of disease is inherently tied to human mobility: as we move through and between cities, we serve as vectors that allow disease to spread to new individuals and communities. The nature of the relationship between mobility and spread of disease has been extensively studied; it is widely understood that human travel is a driving force behind disease spread <ref target=\"#b0\" type=\"bibr\">[1]</ref><ref target=\"#b1\" type=\"bibr\">[2]</ref><ref target=\"#b2\" type=\"bibr\">[3]</ref><ref target=\"#b3\" type=\"bibr\">[4]</ref><ref target=\"#b4\" type=\"bibr\">[5]</ref>. For this reason, many public policy interventions implemented worldwide to contain the spread of COVID-19 focused on limiting mobility, restricting the radius that individuals could travel from their homes and limiting travel between cities and countries.</p><p>As we continue to learn more about the ways that humans move, it is important to apply new discoveries about human mobility to the study of disease spread, deepening our understanding of urban epidemics for use in future public health crises. Recent research uncovered a clear, universal pattern in urban mobility: the total distance that the average visitor to a given location travels to reach it is constant across a city, unrelated to the lo-cation's overall attractiveness <ref target=\"#b5\" type=\"bibr\">[6]</ref>. This value, equivalent to the distance of a location to an individual's home multiplied by the number of times that they visit it over the course of some time period, can be thought of as an \"exploration velocity\": it is the effective distance individuals travel towards any given location per unit of time. It is a simple but critical parameter to understanding human movement.</p><p>Given the simplicity and apparent universality of this parameter of human mobility, its relationship to disease spread is a natural next question. Many travel restrictions and recommendations during the 2020 and 2021 COVID-19 pandemic focused on confining movements to a certain geographical area-for example, restricting the radius that people are able to travel to one's neighborhood, city, state, or country. But what if speed of disease spread depends not just on radius of travel r, but also on the product v := r â€¢ f ? If v has additional impact on disease spread, beyond average travel radius r, COVID-19 policies and restrictions that focused solely on restricting radius may have left significant unmet potential for further disease mitigation. This paper investigates the relationship between exploration velocity v := r â€¢ f and spread of disease, opening the door to future research on the ways in which that relationship can be utilized to contain epidemics.</p></div>\n",
              "<div><head>RESULTS</head><p>Simulating reduction of exploration velocity on real data. In order to understand how v = r â€¢ f interacts with speed of disease spread, we start with largescale datasets of individual human movements in New York City and Dakar, Senegal, which we call M real . The datasets each consist of a set of trips for N individuals over different time periods T , where each trip indicates a given individual moving between two locations. We then use agent based susceptible-infected (SI), susceptible-infected-recovered (SIR), and susceptibleexposed-infected-recovered (SEIR) models calibrated with estimates for COVID-19 <ref target=\"#b6\" type=\"bibr\">[7]</ref> to simulate disease spread as agents follow the trajectories in our datasets. Aside from their unique trajectories, each individual is assumed identical. In each simulation, we vary two parameters: the maximum travel distance r, measured relative to each agent's home location (see Methods for details), and the travel frequency f , the maximum number of times each location was visited. By varying these two parameters together, we are able to manually adjust exploration velocity v in the system. In practice, this means discarding any trip of length greater than r and all but f randomly selected trips to a given location from our datasets. The details of the spatial partitionings used as well as other simulation details are given in the Methods.</p><p>Figures <ref target=\"#fig_7\" type=\"figure\">1(a</ref>) and (b) plot the time it takes for the disease to reach some set proportion Ï„ of the population against exploration velocity for each dataset. The value of Ï„ is set to 29% for our Dakar data and 57% for our New York City data. Thresholds are chosen to be the total epidemic size at the end of the study period under the strictest restrictions for each study area, respectively. Simulations were run over the entire length of the datasets (14 days for the Dakar dataset; 28 days for New York City) with 10,000 agents and a 5% initial infected population. The trends are intuitive and not surprising. For a given f , Ï„ decreases monotonically with r: the further people are allowed travel, the faster the epidemic spreads. Similarly, for a given r, Ï„ decreases monotonically with f : more frequent return trips leads to faster-spreading epidemics. What is surprising, however, is that the Ï„ (r) curves for each value of f have similar shape. This echoes a previous finding <ref target=\"#b5\" type=\"bibr\">[6]</ref> and hints that Ï„, r and f might have a simple relationship. Figures 1(c) and (d) show that they do. Under the rescaling r â†’ r â€¢f all the data appear to merge, collapsing to a single curve that depends on the unifying factor v := r â€¢ f , or exploration velocity. This curve can be empirically characterized as</p><formula xml:id=\"formula_0\">Ï„ (r â€¢ f ) = (r â€¢ f ) a â€¢ b</formula><p>where a and b vary across cities and model types. Figure <ref type=\"figure\">5</ref> in the Supplementary Material shows the same scaling collapse is achieved when the SI and SIR models are used, and Figure <ref type=\"figure\">6</ref> in the SM shows it persists for different values of the disease parameter R 0 . Taken together, these findings suggest the r â†’ râ€¢f collapse persists across multiple disease processes.</p><p>After observing the relationship between r â€¢ f and epidemic size, we check whether there is a relationship between r â€¢ f and spatial dispersion of the disease. We analyze the spatial concentration of infected persons in our New York City simulations using the M function developed by Marcon and Puech, which is calculated as a function of some radius k around each agent <ref target=\"#b7\" type=\"bibr\">[8]</ref>-see Methods for a detailed description of this statistic. High M indicates tighter clusters of cases; low M indicates a more homogeneous distribution of cases across space. We find that for a given radius k, as r and f increase, M (k, r, f ) decreases: infections become consistently more dispersed across the city (see Figure <ref target=\"#fig_7\" type=\"figure\">2a</ref> for this relationship when k = 700m). As with epidemic size, under the rescaling r â†’ r â€¢ f , this relationship collapses to a single curve, as shown in Figure <ref target=\"#fig_2\" type=\"figure\">2b</ref>. This indicates that restricting r â€¢ f is more effective than restricting r alone at containing geographic spread of disease, just as with slowing epidemic spread. This relationship is robust to the choice of k-as k increases, M (k, r, f ) decreases (as would be expected-the larger radius you look at around an agent, the more representative sample of the whole population you will capture) but stays significant and maintains its relationship to r â€¢ f , as illustrated in Figure <ref type=\"figure\">7</ref> in the Supplementary Material. This relationship is demonstrated visually in Figures <ref target=\"#fig_7\" type=\"figure\">2c</ref> and<ref target=\"#fig_2\" type=\"figure\">2d</ref>. When r is fixed at 8, increasing f from 1 to 6 has a clear effect on the extent to which infections spread across the city (Figure <ref target=\"#fig_2\" type=\"figure\">2c</ref>); while significant portions of the city remain untouched by infection when f = 1, many more grid cells contain at least one infection when f = 6. In Figure <ref target=\"#fig_2\" type=\"figure\">2d</ref>, on the other hand, we see that when r á¸Ÿ is held fixed at 12, spatial distribution of infection rates is almost identical for different r and f values, demonstrating the relationship between spatial distribution and r â€¢f .</p></div>\n",
              "<div><head>Mechanisms.</head><p>In order to illustrate the potential mechanism by which r â€¢ f determines disease spread time, we assess the relationship between râ€¢f and both number and variance of contacts between agents in our simulations. Think of our simulations as a network, where agents are nodes and edges are formed whenever two agents come within some distance of one another, enabling disease transmission. It is a known result of the SIR model that speed at which a disease spreads through a network of individuals depends not only on the average degree, or number of contacts per individual k , but also on the variance of contacts (via k 2 , or mean squared degree) according to the following relation:</p><formula xml:id=\"formula_1\">Ï„ = k k 2 âˆ’ (Î³ + Î²) k ,</formula><p>where Ï„ is the characteristic time, or time it takes for the disease to reach 1/e = 36% of the population, Î³ is the daily recovery parameter and Î² is the infection parameter. This relation isn't perfectly applicable to our context-it assumes exposure to all contacts at all timesteps, whereas our simulations incorporate movement in space and thus non-constant degree k over time. However, it is valuable in illustrating the effects of restricting r and f in our dataset. Restricting radius of travel r and frequency of return f affects both k and k 2 . The closer you stay to home and the fewer trips you take, the fewer unique individuals you have the opportunity to encounter (reducing average number of contacts k ). Further, as stricter restrictions are enforced, high-r, high-f trips are removed and the mobility patterns of high-frequency, long-distance travellers start to look more like the mobility patterns of low-frequency travellers who stay close to home (reducing variance of contacts k 2 ). Figure <ref target=\"#fig_3\" type=\"figure\">3</ref> plots these relationships in our New York City data, showing exploration velocity r â€¢ f on the x-axis against average number of contacts k and average squared number of contacts k 2 on the y-axis in panels a and b, respectively. Again, we see a logarithmic relationship. Plugging both k and k 2 into the relation above, we see that characteristic time as predicted by the degree distribution (Ï„ ) is proportional to the true characteristic time (Ï„ ) that we see in our simulations -Figure <ref target=\"#fig_3\" type=\"figure\">3</ref>, panel c. Taken together, these findings show that the shape of the relationship between r â€¢ f and Ï„ can be predicted by decreases in number and variance of contacts between agents in combination with fundamental SIR modeling results.</p><p>These results can be replicated by a simple modification of the preferential exploration and preferential return (PEPR) mobility model presented in previous literature <ref target=\"#b5\" type=\"bibr\">[6]</ref>. Under the PEPR model as proposed in <ref target=\"#b5\" type=\"bibr\">[6]</ref>, at each timestep, agents either explore a new location with a fixed probability P new or return to a previously-visited location with complementary probability 1 âˆ’ P new . In order to choose a new location, agents draw a radius of travel âˆ†r from a heavy-tailed distribution with exponent P (âˆ†r) âˆ¼ |âˆ†r| âˆ’1âˆ’Î± , with Î± = 0.55 taken from <ref target=\"#b5\" type=\"bibr\">[6]</ref>, and they draw an angle of travel Î¸ with preference towards Î¸ that are heavily visited by other agents (see Methods for details). As demonstrated in <ref target=\"#b5\" type=\"bibr\">[6]</ref>, the PEPR model replicates the universality of r â€¢ f . However, when we run our SEIR model on mobility trajectories simulated with the PEPR model, we do not see the same relationship between r â€¢ f and Ï„ that we see in our real data. This is because, in the original PEPR model, agents are always moving from place to place without necessarily returning home, whereas in our real mobility trajectories, time spent at home means less exposure to new, unique contacts (and more exposure to the same neighbors). This leads to the PEPR model being unable to replicate the changes in k that are associated with loosening or tightening radius restrictions, as seen in Figures <ref target=\"#fig_7\" type=\"figure\">4a</ref> and<ref type=\"figure\">4b</ref>. In order to correct for this, we add a probability of travel P travel -at each timestep, with probability P travel , the agent travels to a new location according to the protocol above, whereas with complementary probability 1 âˆ’ P travel the agent stays or returns home. Figures <ref target=\"#fig_7\" type=\"figure\">4c</ref> and<ref type=\"figure\">4d</ref> show the results of rerunning the same SEIR simulations that were run on our real mobility traces M real on a set of simulated trajectories M sim derived from the modified PEPR model. In order to implement   radius restrictions within the PEPR model we restrict the power law distribution from which âˆ†r is drawn and in order to implement frequency restrictions we delete all trips beyond frequency f to the same location. We set P travel = .25, the average proportion of time spent away from home in M real , and run the PEPR simulations on a unit square (see Methods for more details). We see the same logarithmic relationship between r â€¢ f and both Ï„ and Ï„ (expected Ï„ derived from k and k 2 ) that we see in our simulations over M real , although the simulated results are less precise than the results we see in M real . We also replicate the simulations with P travel = .40 and see similar results (see Figure <ref target=\"#fig_8\" type=\"figure\">8</ref> in the SM). This indicates that the modified PEPR model is able to replicate the relationship between exploration velocity and speed of epidemic spread.</p></div>\n",
              "<div><head>DISCUSSION</head><p>The significance of r â€¢ f was revealed in recent work <ref target=\"#b5\" type=\"bibr\">[6]</ref> on human mobility patterns which shows the number of people who visit a given location f times from a distance r during a certain reference period follow a universal, inverse square law N (r, f ) âˆ 1/(rf ) 2 . We suspect the Ï„ (rf ) curve found in this paper derives from this inverse square law; while we have discussed the relationship between exploration velocity and network contacts, deriving the exact mechanism linking the universal inverse square law to Ï„ (rf ) is an open question for future work.</p><p>An important implication of the Ï„ (r â€¢ f ) curve is that restricting short, high frequency trips in a simulated environment does as much to slow disease spread as restricting long, low frequency trips. Agents in our simulations that take multiple trips around their neighbourhood spread disease as quickly as a those taking single, long trips to the city center. This is interesting academically, but more importantly, it has potential implications for policy: it means that restricting travel distance r but ignoring travel frequency f could be ineffective in containing the spreading of a disease.</p><p>That said, an important limitation of this study is that it restricts exploration velocity in a very specific way, removing trips that extend beyond a certain r or f . In reality, it is unclear how people would respond to policies that aim to restrict exploration velocity. True behavioral responses to different travel policies and what policies or mechanisms would be required to change individuals' behavior in a way that reduces exploration velocity are areas for future research. Further limitations of this study include the inherent limitations of simple epidemiological modeling; it is known that the SI, SIR, SEIR and other classic disease models make assumptions which bound their accuracy <ref target=\"#b8\" type=\"bibr\">[9]</ref><ref target=\"#b9\" type=\"bibr\">[10]</ref><ref target=\"#b10\" type=\"bibr\">[11]</ref>. The estimates for the model parameters likely carry error <ref target=\"#b11\" type=\"bibr\">[12]</ref>, and we also assumed that each agent was identical and that each exposure within a set radius carried equal likelihood of infection. Moreover, the scale of our analysis is restricted to the city level (because our datasets are collected at this level) and so our findings do not necessarily generalize to the perhaps more important case of country-level and international travel policies. If the r â€¢ f invariance, or some form of inverse relation between r and f , holds for international mobility patterns, then, as at the city scale, lax distance limits could be compensated by strict frequency restrictions. This is a bold hypothesis, which should be tested in future work. Metapopulation disease models <ref target=\"#b12\" type=\"bibr\">[13]</ref><ref target=\"#b13\" type=\"bibr\">[14]</ref><ref target=\"#b14\" type=\"bibr\">[15]</ref>, with their convenient trade off between realism and parsimony, seem like a good theoretical starting point for this effort.</p><p>Despite these limitations, we believe our results reveal novel insights and bring up an important component of human mobility that could prove useful for future policies on COVID-19 and other epidemic diseases. In the current COVID-19 pandemic and in future pandemics, evidence-based policies at the city scale (not to mention the national or international scale) are needed to mitigate speed and intensity of disease spread. Our results have the potential to help this effort by opening up investigation into the relationship between exploration velocity and epidemic spread. They indicate that the exploration velocity of a city's inhabitants r â€¢ f must be boundedto bound distance r but not visitation frequency f leaves potential for disease containment unmet. Furthermore, and more optimistically, if a bound on exploration velocity is possible and effective, it would mean that strict distance restrictions at the city or town level -as adopted by, e.g., Ireland and Italy at the beginning of the infection -are perhaps unnecessary. Given a desired bound on epidemic speed Ï„ bound , a large r could be offset by a small f ; allowing citizens to travel infrequently to distant services (doctors, hospitals etc) may be safe. This inverse relation between travel distance and frequency (r âˆ Ï„ bound /f ) could also vitally inform remote working policies supporting the hypothesis that working from home multiple days per week -and thereby limiting the visitation frequency to workplaces -helps prevent the spread of disease.</p></div>\n",
              "<div><head>METHODS</head><p>New York City data. Individuals' movements in New York City are inferred from GPS traces collected from mobile phones by the company X-Mode over a span of one month (February 2020). The raw data contains about 479,163 anonymized users; our analysis uses 10,000 users randomly selected from those that appear in the dataset every day in the month of February. Dakar data. The Dakar dataset is based on anonymized Call Detailed Records (CDR) provided by the Data for Development (D4D) Challenge. The detailed information of this dataset is provided in <ref target=\"#b15\" type=\"bibr\">[16]</ref>. Here, we use the SET2, which includes individual trajectories for 300,000 FIG. <ref type=\"figure\">4</ref>: PEPR simulation results. When we run SEIR models across a set of trajectories M sim which have been created using the modified PEPR model with P travel = .25, we see a similar relationship between r â€¢ f and Ï„ to that in our real trajectories M real . When we run SEIR models across trajectories which have been created using the original PEPR model (with P travel = 1), we do not see this relationship. sampled users in Senegal, and after the preprocess, we have 173,000 users and 173 cells in Dakar region during two weeks of January, 2013. We subselect for users who appear at least 200 times in the dataset to ensure that we have adequate information about their trajectories over the two weeks. Data preprocessing. The X-Mode data from NYC is generated on an very fine spatial and temporal scale, with exact latitude and longitude coordinate updates as frequently as every second. The CDR data from Dakar, on the other hand, are generated only for voice calls, text messages or data exchanges and therefore have limited resolution in time. The geographic location of the cell towers and their density determines the accuracy of location measurements through triangularization techniques. Therefore, the trajectories extracted from CDRs constitute a discrete approximation of the moving population M (x; y; t). There are several steps in preprocessing of the data before it can be suitable for use in our analysis, which vary between the X-Mode data and the CDR data.</p><p>The main steps for the NYC data are: i) We use density-based spatial clustering of applications with noise (DBSCAN) to group tightly-clustered latitude/longitude pairs in each individual's trajectory into locations <ref target=\"#b16\" type=\"bibr\">[17]</ref>. If a cluster of at least five latitude/longitude points exists such that no point is more than .0004 degrees (about 56 meters) from two other points in the cluster, those points are grouped together as a single location. ii) Each agent is assigned the DBSCAN cluster it visits most as its home location. iii) We drop all locations in the trajectory that have been visited for less than a minimum time Ï„ min = 15min. iv) In order to restrict travel distance r, we calculate distance between locations by the haversine formula, which derives the great-circle distance between two points on a sphere. All locations that are more than r km from an agents home location are removed from their trajectory. v) In order to restrict travel frequency f , for each DBSCAN cluster that an agent visits more than f distinct times (where distinct visits are determined by an agent leaving a location and then coming back to it), we randomly select f visits to include in their trajectory and drop the rest.</p><p>The main steps for the Dakar data are: i) We view each cell tower as a different location in the city. ii) For each person, we determine the home location as the cell tower location which has been visited for the most cumulative time. By summing over all days in a given time window, one can find the home cell with high level of confidence for the majority of subjects. iii) We drop all locations in the trajectory that have been visited for less than a minimum time Ï„ min = 10min. iv) In order to restrict travel distance r, we calculate distance between cell towers by the haversine formula, which derives the great-circle distance between two points on a sphere. All cell towers that are more than r km from an agent's home location are removed from their trajectory. v) In order to restrict travel frequency f , for each location that an agent visits more than f distinct times (where distinct visits are determined by an agent leaving a location and then coming back to it), we randomly select f visits to include in their trajectory and drop the rest (excepting visits to an agent's home location, which are not restricted).</p><p>The duration of stay, frequency, and distance criteria on defining cell visits yields a list of cells visited by that subject over the study period for a given frequency restriction f and distance restriction r. Simulation details. We run an agent-based SEIR, SIR, and SI models with N = 10, 000 agents, 5% of which are initialized to be infected (1% in the SI and SIR Dakar models). Each agent is assigned the trajectory of a real person from our dataset, with location updated every 900 seconds (15 minutes) for the NYC simulations or every 600 seconds (10 minutes) for the Dakar simulations. At each time step, each user's location is updated according to their assigned trajectory and infection status is updated according to the following parameters, drawn from Chen 2020 <ref target=\"#b6\" type=\"bibr\">[7]</ref>'s estimates of R0 = 3.58, incubation period = 5.2 days, and infection period = 5. In addition, we reproduce our results with disease parameters estimated for the more highly-contagious Delta variant of COVID-19 (Î² = 1.42, all other parameters remain the same, <ref target=\"#b17\" type=\"bibr\">[18]</ref>) and the 2009 H1N1 influenza strain (Î² = .913, Î³ = 1.6, Ïƒ = 1, <ref target=\"#b18\" type=\"bibr\">[19,</ref><ref target=\"#b19\" type=\"bibr\">20]</ref>). Finally, let I local and N local be the number of infected agents and total agents within a 190 meter radius of the agent's current location for the NYC data or within the same cell tower location for the Dakar data. Then, transition probabilities are:</p><formula xml:id=\"formula_2\">P[S â†’ E] = Î² * * I local N local P[E â†’ I] = Ïƒ * P[I â†’ R] = Î³ *</formula><p>In the SIR and SI models, if an agent becomes infected on a given day, they will become contagious at the start of the next day. Quantifying dispersion. We use the M function developed in <ref target=\"#b7\" type=\"bibr\">[8]</ref> to quantify spatial dispersion of disease in our New York City simulations for a given r, f . The M function is calculated as follows: for each infected agent I and some radius k, we calculate the ratio between the proportion of agents within k of I which are infected to the proportion of agents in the total population which are infected. Summing this value over all infected agents I and dividing by N âˆ’ 1, where N is the number of infected agents, gives M (k), the M function evaluated at k. While M is generally analyzed as a function over all reasonable k, we evaluate the M function at a specific k in order to compare spatial dispersion across r â€¢ f values at that k, and then show that the relationship is robust to choice of k.</p><p>Confidence intervals for M (k) are obtained by Monte Carlo simulation-for a given epidemic size Ïˆ, we randomly assign Ïˆ infections across the population 1,000 times and calculate M Ïˆ each time. By taking the .025 and .975 quantiles of these simulated M , we form an upper and lower bound on M Ïˆ . It is notable that our empirical M never reaches this confidence band, implying that spatial dispersion is significantly non-homogenous for every k and Ïˆ. Robustness to model parameters. Here we demonstrate that the universal Ï„ (râ€¢f ) curve is robust to changes in the parameters R 0 by running our simulations with estimated transmission parameters for the 2009 H1N1 influenza strain (Î² = .913, Î³ = 1.6, Ïƒ = 1 <ref target=\"#b18\" type=\"bibr\">[19,</ref><ref target=\"#b19\" type=\"bibr\">20]</ref>) and the Delta variant of COVID-19 (Î² = 1.42, <ref target=\"#b17\" type=\"bibr\">[18]</ref>). Figure <ref type=\"figure\">6</ref> shows that, with these parameters, the Ï„ (r â€¢ f ) relationship still holds. Preferential return model. The preferential return model proposed in <ref target=\"#b5\" type=\"bibr\">[6]</ref> is based off of that proposed in Song et al. <ref target=\"#b20\" type=\"bibr\">[21]</ref>. With some probability P new , agents return to a location they have already visited; with probability 1 âˆ’ P new they visit a new location with distance drawn from the empirical distance distribution and direction drawn uniformly at random. Waiting times between trips are also drawn from the empirical distribution. The probability p is a function of the number of locations already visited:</p><formula xml:id=\"formula_3\">P new = ÏS âˆ’Î³</formula><p>where S is the number of locations visited. The parameters Ï and Î³ are fit to the real data using least-squares regression (using the NYC dataset, we find Ï = .500 and Î³ = .267.)</p><p>We add the following modification: while in the original PEPR model agents travel at each timestep, we set some probability P travel with which agents travel at any given timestep. Conversely, with probability 1 âˆ’ P travel , agents stay home. The simulation results reported here are from running our modified PEPR on a unit square for 500 timesteps.   </p></div><figure xml:id=\"fig_0\"><head>FIG. 1 :</head><label>1</label><figDesc>FIG. 1: Epidemics sizes for SEIR model. Top row: plots of epidemic spread speed in units of 10-minute increments against maximum allowed travel distance r for different max travel frequency f . Bottom row: top row plotted against r â€¢ f . Each data point is the average of 5 simulations. For other simulation details see Methods. R 2 values for best-fit lines are .981 (NYC) and .902 (Dakar). Best-fit line parameters are a = âˆ’0.08, b = 2292.58 (NYC) and a = âˆ’0.01, b = 1853.57 (Dakar).</figDesc><graphic coords=\"3,70.94,229.91,228.64,152.43\" type=\"bitmap\"/></figure>\n",
              "<figure xml:id=\"fig_1\"><head/><label/><figDesc>(a) Relationship between frequency restriction and dispersion for radius = 700m. (b) Scaling collapse for radius = 700m. R 2 value of best-fit line is .97.(c) Holding r constant but restricting f , we can see that loosening restrictions on f increases spatial dispersion of infections-when f is at 1 (left), infections reach 59.0% of grid cells; when f is relaxed to 6 (right), infections reach 73.8% of grid cells.(d) In the simulations above, r and f are individually different but r â€¢ f is the same. The similar spatial distribution of infections in the two simulations reflects the scaling collapse in dispersion of infection: dispersion depends on r â€¢ f as opposed to just r or f . Here, color represents proportion of the grid cell population that is infected at the end of the simulation.</figDesc></figure>\n",
              "<figure xml:id=\"fig_2\"><head>FIG. 2 :</head><label>2</label><figDesc>FIG. 2:Scaling collapse of spatial dispersion of infections. Like Ï„ , spatial dispersion M shows a universal scaling collapse with the product r â€¢ f , as demonstrated in panels (a) and (b). Here, M is calculated using a radius of k = 700 meters. This is visually represented in panel (c), where we see the final spatial distribution of infections for two simulations with the same r but different f , and in panel (d), where we see the final spatial distribution of infections for two simulations with the same r â€¢ f but different r and f .</figDesc><graphic coords=\"4,66.70,528.79,152.43,101.62\" type=\"bitmap\"/></figure>\n",
              "<figure xml:id=\"fig_3\"><head>FIG. 3 :</head><label>3</label><figDesc>FIG.3: Relationship between spreading speed and number and variance of contacts. Both mean number of contacts k and mean squared number of contacts k 2 of the agents in our dataset show a clear logarithmic relationship with exploration velocity r â€¢ f . The relationship between r â€¢ f and k , k 2 can predict spreading speed Ï„ using a simple relation that is a known result of the SIR model.</figDesc><graphic coords=\"4,231.83,528.79,152.43,101.62\" type=\"bitmap\"/></figure>\n",
              "<figure xml:id=\"fig_4\"><head/><label/><figDesc>(a) Relationship between distance restriction and Ï„ in PEPR simulations with P travel = .25. R 2 value of best fit line is .875. (b) Relationship between distance restriction and Ï„ as predicted by k , k 2 in PEPR simulations with P travel = .25. R 2 value of best fit line is .924. (c) Relationship between distance restriction and Ï„ in PEPR simulations with P travel = 1. R 2 value of best fit line is .640. (d) Relationship between distance restriction and Ï„ as predicted by k , k 2 in PEPR simulations with P travel = 1. R 2 value of best fit line is .480.</figDesc></figure>\n",
              "<figure xml:id=\"fig_5\"><head/><label/><figDesc>8 days: Î² = daily transmission parameter = 3.58 5.8 = .617 Ïƒ = daily rate at which an exposed person becomes infective = 1/5.2 Î³ = daily recovery parameter = 1/5.8 Let s be the number of time steps in a day. We then transform the above daily parameters into timestep parameters as follows: Î² * = time step transmission probability = Î²/s Î³ * = time step recovery probability = 1 âˆ’ s âˆš 1 âˆ’ Î³ Ïƒ * = time step probability that an exposed person becomes infective = 1 âˆ’ s âˆš 1 âˆ’ Ïƒ</figDesc></figure>\n",
              "<figure xml:id=\"fig_6\"><head>FIG. 5 :FIG. 6 :FIG. 7 :</head><label>567</label><figDesc>FIG. 5: Scaling collapse in SI and SIR model. Top row: scaling collapse for SI model. R 2 values for best-fit lines are, from left to right, .958 and .896. Best-fit line parameters are a = âˆ’0.18, b = 1821.23 (NYC) and a = âˆ’0.04, b = 1797.87 (Dakar). Bottom row, scaling collapse for SIR model. R 2 values for best-fit lines are, from left to right, .974 and .937. Best-fit line parameters are a = âˆ’0.17, b = 1802.79 (NYC) and a = âˆ’0.04, b = 1760.42 (Dakar).</figDesc><graphic coords=\"10,70.94,229.91,228.64,152.43\" type=\"bitmap\"/></figure>\n",
              "<figure xml:id=\"fig_7\"><head>( a )</head><label>a</label><figDesc>Relationship between distance restriction and Ï„ in PEPR simulations. (b) Relationship between distance restriction and Ï„ as predicted by k , k 2 in PEPR simulations.</figDesc></figure>\n",
              "<figure xml:id=\"fig_8\"><head>FIG. 8 :</head><label>8</label><figDesc>FIG.8: PEPR simulation results, where P travel = .40. When we run SEIR models across a set of trajectories M sim which have been created using the PEPR model with P travel = .40, we see a similar relationship between r â€¢ f and Ï„ to that in our real trajectories M real .</figDesc><graphic coords=\"11,70.94,440.95,228.64,152.43\" type=\"bitmap\"/></figure>\n",
              "</body>"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_arxiv = list(map(soupify, dl_pdf_xml))"
      ],
      "metadata": {
        "id": "PCeYnZPMxfPM"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "type(soup_results_arxiv)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b489db0c-84d5-4a63-bab4-e0fe085079a6",
        "id": "w54yxqhpxfPN"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "list"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_arxiv)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "301e43e4-8392-425b-e828-5dfec402ef3a",
        "id": "jaR_Y_d3xfPM"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "304"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 6. Navigating the tree\n",
        "\n",
        "A `<Tag>` object corresponds to an XML tag in the original document, in this case the `<body>` tag."
      ],
      "metadata": {
        "id": "qEabBaAjhNiR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "type(soup_results_arxiv[6])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "cba07734-f277-4ff1-ca65-3f61b98adefe",
        "id": "MpTTZSuYxfPN"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "bs4.element.Tag"
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "To find the `<head`> tag we can just use the format soup.head"
      ],
      "metadata": {
        "id": "P7wj52FTjRXV"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_arxiv[6].head"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ddf34ed4-e93e-4b62-f8f5-c504ef0aee74",
        "id": "AgKUXK8bxfPN"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<head>INTRODUCTION</head>"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View the result when returning the type of object for the first exception in futures_map dictionary above with Exception('[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1')."
      ],
      "metadata": {
        "id": "MP5kB-SLjqSB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "type(soup_results_arxiv[3])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DUqHExoH3ppj",
        "outputId": "54f13b4b-b1ca-4dd9-f5f2-18b5d46aa924"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "NoneType"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 7. Find all tags including descendants\n",
        "\n",
        "Previously we used the find() method for each article which gives only the *first* tag by that name as each article only has one `<body>` and `<head>` tag.\n",
        "\n",
        "The function below uses the find_all() method to return *all* of the direct child and descendant tags of the `<body>` tag for each article."
      ],
      "metadata": {
        "id": "mtR3hhkOxfPO"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_all_tags(article):\n",
        "\n",
        "    tags_list = []\n",
        "\n",
        "    try:\n",
        "        for tag in article.find_all(True):\n",
        "            if tag.name != None:\n",
        "              tags_list.append(tag.name)\n",
        "    except Exception as e:\n",
        "      print(e)\n",
        "\n",
        "\n",
        "    return sorted(list(set(tags_list)))\n"
      ],
      "metadata": {
        "id": "N23LqasXxfPO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "all_tags = list(map(find_all_tags, soup_results_arxiv))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ys84qqg33zcZ",
        "outputId": "a9cb68d7-1617-4b3a-ba80-16be3c28bf80"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "As expected, these are the four exceptions we found earlier which show up as empty lists."
      ],
      "metadata": {
        "id": "Y_k81qAMqAXT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "all_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3aec54d2-00c1-427a-87a2-e3ee88aecae9",
        "id": "fOpCuL3hxfPP"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " [],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref', 'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " [],\n",
              " [],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'formula', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'head', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " [],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'p', 'ref', 'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'head', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'formula', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref']]"
            ]
          },
          "metadata": {},
          "execution_count": 20
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in all_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "fFkDkYcHsBpQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_all_tags = sum(all_tags, [])\n",
        "sum_all_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "191bb3b4-2d74-4661-b696-da72f54f52df",
        "id": "PsQ6bNlQxfPP"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Counter class with additional most_common() method to return a list of the *n* most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "20-lOqLXsPjy"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_all_unique_tags = Counter(sum_all_tags).most_common()\n",
        "num_all_unique_tags = num_all_unique_tags[::]\n",
        "\n",
        "for tag, count in num_all_unique_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2c34b9a7-d25d-4d45-e77e-df6542a4f710",
        "id": "xfwP5qtyxfPQ"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "div 300\n",
            "head 300\n",
            "p 300\n",
            "ref 299\n",
            "label 298\n",
            "figDesc 297\n",
            "figure 297\n",
            "graphic 249\n",
            "table 240\n",
            "cell 235\n",
            "row 235\n",
            "formula 179\n",
            "note 177\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 8. Find direct child tags of `<body>` tag\n",
        "\n",
        "Every tag has a name which can be accessed using the `.name` attribute.\n",
        "\n",
        "We will use this to find the direct child tags of the `<body>` tag.\n"
      ],
      "metadata": {
        "id": "LotA3KLkxfPQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_body_child_tags(article):\n",
        "\n",
        "    body_child_list = []\n",
        "\n",
        "    try:\n",
        "        for tag in article:\n",
        "            if tag.name != None:\n",
        "                  body_child_list.append(tag.name)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "    return sorted(list(set(body_child_list)))"
      ],
      "metadata": {
        "id": "O0BlIqINxfPQ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "body_child_tags = list(map(find_body_child_tags, soup_results_arxiv))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "741da578-c48a-4f5d-9492-2fdd14480542",
        "id": "OZ-Zk3F8xfPQ"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We can see below that the `<body>` tag has direct child tags for `<div>`, `<figure>` and `<note>`."
      ],
      "metadata": {
        "id": "R-doePj_vxyl"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "body_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f64468ed-c3c5-4427-d11b-a745d51dad25",
        "id": "LF1jZsgMwQK0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " [],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " [],\n",
              " [],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " [],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure']]"
            ]
          },
          "metadata": {},
          "execution_count": 25
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in body_child_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "qydRGwCIx7cx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_body_child_tags = sum(body_child_tags, [])\n",
        "sum_body_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f6f1d160-5018-4496-8c6f-ca33072de265",
        "id": "DtesdnaRxfPR"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure']"
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again use the Counter class with most_common() method to return a list of the *n* most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "3zY5hHveyM0R"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_body_child_tags = Counter(sum_body_child_tags).most_common()\n",
        "num_unique_body_child_tags = num_unique_body_child_tags[::]\n",
        "\n",
        "for tag, count in num_unique_body_child_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Bt8oDoG94KxO",
        "outputId": "a0cad949-e5a7-4d8a-f8ed-f54bdadcfb1e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "div 300\n",
            "figure 297\n",
            "note 105\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##  9. Find direct child tags of `<div>` tags\n",
        "\n",
        "Function to find all `<div>` tags and append all direct child tags to a list.\n",
        "\n"
      ],
      "metadata": {
        "id": "VGgoq5DFxfPS"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_div_tags(article):\n",
        "\n",
        "    div_tags = []\n",
        "\n",
        "    try:\n",
        "        divs = article.find_all(\"div\")\n",
        "        for div in divs:\n",
        "          for tag in div:\n",
        "              if tag.name != None:\n",
        "                  div_tags.append(tag.name)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "    return sorted(list(set(div_tags)))"
      ],
      "metadata": {
        "id": "h6kasfyBxfPS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "div_tags = list(map(find_div_tags, soup_results_arxiv))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "89b78192-fdb2-40c1-ded6-dcfa0757fedc",
        "id": "Ki8ibP69xfPS"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "div_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0dd77c5d-bb2b-4d16-b440-be7a98078199",
        "id": "j9feMi_xxfPS"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " [],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " [],\n",
              " [],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " [],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p']]"
            ]
          },
          "metadata": {},
          "execution_count": 30
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in div_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "CnW9joiBzQ4K"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_div_tags = sum(div_tags, [])\n",
        "sum_div_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b91d8956-ece1-4726-c09f-61357409c358",
        "id": "69qBhWrGxfPT"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p']"
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again use the Counter class with most_common() method to return a list of the *n* most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "PIHYv5TVzfbC"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_div_tags = Counter(sum_div_tags).most_common()\n",
        "num_unique_div_tags = num_unique_div_tags[::]\n",
        "\n",
        "for tag, count in num_unique_div_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oFJxWmzy43uc",
        "outputId": "1204f40f-0ee8-46dc-e3b5-1d8cd61ae866"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "p 300\n",
            "head 297\n",
            "formula 179\n",
            "note 15\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "traCdqLcxfPT"
      },
      "source": [
        "## 10. Remove unwanted tags and content\n",
        "\n",
        "Define functions to remove unwanted tags and their contents using decompose() method which removes a tag from the tree, then completely destroys it and its contents."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "me7LDtCTxfPT"
      },
      "outputs": [],
      "source": [
        "def remove_headings(article):\n",
        "\n",
        "    for head in article(\"head\"):\n",
        "        head.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_figures(article):\n",
        "\n",
        "    for figure in article(\"figure\"):\n",
        "        figure.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_tables(article):\n",
        "\n",
        "    for table in article(\"table\"):\n",
        "        table.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_formulas(article):\n",
        "\n",
        "    for formula in article(\"formula\"):\n",
        "        formula.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_labels(article):\n",
        "\n",
        "    for label in article(\"label\"):\n",
        "        label.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_refs(article):\n",
        "\n",
        "    for ref in article(\"ref\"):\n",
        "        ref.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_graphics(article):\n",
        "\n",
        "    for graphic in article(\"graphic\"):\n",
        "        graphic.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_notes(article):\n",
        "\n",
        "    for note in article(\"note\"):\n",
        "        note.decompose()\n",
        "\n",
        "    return article\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "g6vN_WZlxfPU"
      },
      "source": [
        "Function to call the functions above on each article and return new article list."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def remove_tags(article):\n",
        "\n",
        "    new_article_list = []\n",
        "\n",
        "    if article != None:\n",
        "        try:\n",
        "              new_article = remove_headings(article)\n",
        "              new_article = remove_figures(new_article)\n",
        "              new_article = remove_tables(new_article)\n",
        "              new_article = remove_formulas(new_article)\n",
        "              new_article = remove_labels(new_article)\n",
        "              new_article = remove_refs(new_article)\n",
        "              new_article = remove_graphics(article)\n",
        "              new_article = remove_notes(article)\n",
        "              new_article_list.append(new_article)\n",
        "        except Exception as e:\n",
        "            print(e)\n",
        "\n",
        "    return new_article_list"
      ],
      "metadata": {
        "id": "dCk2zMArxfPU"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5rClP9_MxfPU"
      },
      "outputs": [],
      "source": [
        "soup_results_arxiv_removed_tags = list(map(remove_tags, soup_results_arxiv))"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_arxiv_removed_tags)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a3567da1-2946-4feb-b45b-fa25c113616c",
        "id": "FzNvAk2pxfPU"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "304"
            ]
          },
          "metadata": {},
          "execution_count": 36
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View article with unwanted tags and contents removed."
      ],
      "metadata": {
        "id": "wyMpq4Hq3ILu"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_arxiv_removed_tags[6]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "521ad2dc-51b0-4a5a-f82b-6d74539010d5",
        "id": "zKcRKpUDxfPV"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<body>\n",
              " <div><p>The global impact of the COVID-19 pandemic on both human health and socioeconomic activity has brought to light the importance of a nuanced understanding of the way that epidemics spread in cities. Urban spread of disease is inherently tied to human mobility: as we move through and between cities, we serve as vectors that allow disease to spread to new individuals and communities. The nature of the relationship between mobility and spread of disease has been extensively studied; it is widely understood that human travel is a driving force behind disease spread . For this reason, many public policy interventions implemented worldwide to contain the spread of COVID-19 focused on limiting mobility, restricting the radius that individuals could travel from their homes and limiting travel between cities and countries.</p><p>As we continue to learn more about the ways that humans move, it is important to apply new discoveries about human mobility to the study of disease spread, deepening our understanding of urban epidemics for use in future public health crises. Recent research uncovered a clear, universal pattern in urban mobility: the total distance that the average visitor to a given location travels to reach it is constant across a city, unrelated to the lo-cation's overall attractiveness . This value, equivalent to the distance of a location to an individual's home multiplied by the number of times that they visit it over the course of some time period, can be thought of as an \"exploration velocity\": it is the effective distance individuals travel towards any given location per unit of time. It is a simple but critical parameter to understanding human movement.</p><p>Given the simplicity and apparent universality of this parameter of human mobility, its relationship to disease spread is a natural next question. Many travel restrictions and recommendations during the 2020 and 2021 COVID-19 pandemic focused on confining movements to a certain geographical area-for example, restricting the radius that people are able to travel to one's neighborhood, city, state, or country. But what if speed of disease spread depends not just on radius of travel r, but also on the product v := r â€¢ f ? If v has additional impact on disease spread, beyond average travel radius r, COVID-19 policies and restrictions that focused solely on restricting radius may have left significant unmet potential for further disease mitigation. This paper investigates the relationship between exploration velocity v := r â€¢ f and spread of disease, opening the door to future research on the ways in which that relationship can be utilized to contain epidemics.</p></div>\n",
              " <div><p>Simulating reduction of exploration velocity on real data. In order to understand how v = r â€¢ f interacts with speed of disease spread, we start with largescale datasets of individual human movements in New York City and Dakar, Senegal, which we call M real . The datasets each consist of a set of trips for N individuals over different time periods T , where each trip indicates a given individual moving between two locations. We then use agent based susceptible-infected (SI), susceptible-infected-recovered (SIR), and susceptibleexposed-infected-recovered (SEIR) models calibrated with estimates for COVID-19  to simulate disease spread as agents follow the trajectories in our datasets. Aside from their unique trajectories, each individual is assumed identical. In each simulation, we vary two parameters: the maximum travel distance r, measured relative to each agent's home location (see Methods for details), and the travel frequency f , the maximum number of times each location was visited. By varying these two parameters together, we are able to manually adjust exploration velocity v in the system. In practice, this means discarding any trip of length greater than r and all but f randomly selected trips to a given location from our datasets. The details of the spatial partitionings used as well as other simulation details are given in the Methods.</p><p>Figures ) and (b) plot the time it takes for the disease to reach some set proportion Ï„ of the population against exploration velocity for each dataset. The value of Ï„ is set to 29% for our Dakar data and 57% for our New York City data. Thresholds are chosen to be the total epidemic size at the end of the study period under the strictest restrictions for each study area, respectively. Simulations were run over the entire length of the datasets (14 days for the Dakar dataset; 28 days for New York City) with 10,000 agents and a 5% initial infected population. The trends are intuitive and not surprising. For a given f , Ï„ decreases monotonically with r: the further people are allowed travel, the faster the epidemic spreads. Similarly, for a given r, Ï„ decreases monotonically with f : more frequent return trips leads to faster-spreading epidemics. What is surprising, however, is that the Ï„ (r) curves for each value of f have similar shape. This echoes a previous finding  and hints that Ï„, r and f might have a simple relationship. Figures 1(c) and (d) show that they do. Under the rescaling r â†’ r â€¢f all the data appear to merge, collapsing to a single curve that depends on the unifying factor v := r â€¢ f , or exploration velocity. This curve can be empirically characterized as</p><p>where a and b vary across cities and model types. Figure  in the Supplementary Material shows the same scaling collapse is achieved when the SI and SIR models are used, and Figure  in the SM shows it persists for different values of the disease parameter R 0 . Taken together, these findings suggest the r â†’ râ€¢f collapse persists across multiple disease processes.</p><p>After observing the relationship between r â€¢ f and epidemic size, we check whether there is a relationship between r â€¢ f and spatial dispersion of the disease. We analyze the spatial concentration of infected persons in our New York City simulations using the M function developed by Marcon and Puech, which is calculated as a function of some radius k around each agent -see Methods for a detailed description of this statistic. High M indicates tighter clusters of cases; low M indicates a more homogeneous distribution of cases across space. We find that for a given radius k, as r and f increase, M (k, r, f ) decreases: infections become consistently more dispersed across the city (see Figure  for this relationship when k = 700m). As with epidemic size, under the rescaling r â†’ r â€¢ f , this relationship collapses to a single curve, as shown in Figure . This indicates that restricting r â€¢ f is more effective than restricting r alone at containing geographic spread of disease, just as with slowing epidemic spread. This relationship is robust to the choice of k-as k increases, M (k, r, f ) decreases (as would be expected-the larger radius you look at around an agent, the more representative sample of the whole population you will capture) but stays significant and maintains its relationship to r â€¢ f , as illustrated in Figure  in the Supplementary Material. This relationship is demonstrated visually in Figures  and. When r is fixed at 8, increasing f from 1 to 6 has a clear effect on the extent to which infections spread across the city (Figure ); while significant portions of the city remain untouched by infection when f = 1, many more grid cells contain at least one infection when f = 6. In Figure , on the other hand, we see that when r á¸Ÿ is held fixed at 12, spatial distribution of infection rates is almost identical for different r and f values, demonstrating the relationship between spatial distribution and r â€¢f .</p></div>\n",
              " <div><p>In order to illustrate the potential mechanism by which r â€¢ f determines disease spread time, we assess the relationship between râ€¢f and both number and variance of contacts between agents in our simulations. Think of our simulations as a network, where agents are nodes and edges are formed whenever two agents come within some distance of one another, enabling disease transmission. It is a known result of the SIR model that speed at which a disease spreads through a network of individuals depends not only on the average degree, or number of contacts per individual k , but also on the variance of contacts (via k 2 , or mean squared degree) according to the following relation:</p><p>where Ï„ is the characteristic time, or time it takes for the disease to reach 1/e = 36% of the population, Î³ is the daily recovery parameter and Î² is the infection parameter. This relation isn't perfectly applicable to our context-it assumes exposure to all contacts at all timesteps, whereas our simulations incorporate movement in space and thus non-constant degree k over time. However, it is valuable in illustrating the effects of restricting r and f in our dataset. Restricting radius of travel r and frequency of return f affects both k and k 2 . The closer you stay to home and the fewer trips you take, the fewer unique individuals you have the opportunity to encounter (reducing average number of contacts k ). Further, as stricter restrictions are enforced, high-r, high-f trips are removed and the mobility patterns of high-frequency, long-distance travellers start to look more like the mobility patterns of low-frequency travellers who stay close to home (reducing variance of contacts k 2 ). Figure  plots these relationships in our New York City data, showing exploration velocity r â€¢ f on the x-axis against average number of contacts k and average squared number of contacts k 2 on the y-axis in panels a and b, respectively. Again, we see a logarithmic relationship. Plugging both k and k 2 into the relation above, we see that characteristic time as predicted by the degree distribution (Ï„ ) is proportional to the true characteristic time (Ï„ ) that we see in our simulations -Figure , panel c. Taken together, these findings show that the shape of the relationship between r â€¢ f and Ï„ can be predicted by decreases in number and variance of contacts between agents in combination with fundamental SIR modeling results.</p><p>These results can be replicated by a simple modification of the preferential exploration and preferential return (PEPR) mobility model presented in previous literature . Under the PEPR model as proposed in , at each timestep, agents either explore a new location with a fixed probability P new or return to a previously-visited location with complementary probability 1 âˆ’ P new . In order to choose a new location, agents draw a radius of travel âˆ†r from a heavy-tailed distribution with exponent P (âˆ†r) âˆ¼ |âˆ†r| âˆ’1âˆ’Î± , with Î± = 0.55 taken from , and they draw an angle of travel Î¸ with preference towards Î¸ that are heavily visited by other agents (see Methods for details). As demonstrated in , the PEPR model replicates the universality of r â€¢ f . However, when we run our SEIR model on mobility trajectories simulated with the PEPR model, we do not see the same relationship between r â€¢ f and Ï„ that we see in our real data. This is because, in the original PEPR model, agents are always moving from place to place without necessarily returning home, whereas in our real mobility trajectories, time spent at home means less exposure to new, unique contacts (and more exposure to the same neighbors). This leads to the PEPR model being unable to replicate the changes in k that are associated with loosening or tightening radius restrictions, as seen in Figures  and. In order to correct for this, we add a probability of travel P travel -at each timestep, with probability P travel , the agent travels to a new location according to the protocol above, whereas with complementary probability 1 âˆ’ P travel the agent stays or returns home. Figures  and show the results of rerunning the same SEIR simulations that were run on our real mobility traces M real on a set of simulated trajectories M sim derived from the modified PEPR model. In order to implement   radius restrictions within the PEPR model we restrict the power law distribution from which âˆ†r is drawn and in order to implement frequency restrictions we delete all trips beyond frequency f to the same location. We set P travel = .25, the average proportion of time spent away from home in M real , and run the PEPR simulations on a unit square (see Methods for more details). We see the same logarithmic relationship between r â€¢ f and both Ï„ and Ï„ (expected Ï„ derived from k and k 2 ) that we see in our simulations over M real , although the simulated results are less precise than the results we see in M real . We also replicate the simulations with P travel = .40 and see similar results (see Figure  in the SM). This indicates that the modified PEPR model is able to replicate the relationship between exploration velocity and speed of epidemic spread.</p></div>\n",
              " <div><p>The significance of r â€¢ f was revealed in recent work  on human mobility patterns which shows the number of people who visit a given location f times from a distance r during a certain reference period follow a universal, inverse square law N (r, f ) âˆ 1/(rf ) 2 . We suspect the Ï„ (rf ) curve found in this paper derives from this inverse square law; while we have discussed the relationship between exploration velocity and network contacts, deriving the exact mechanism linking the universal inverse square law to Ï„ (rf ) is an open question for future work.</p><p>An important implication of the Ï„ (r â€¢ f ) curve is that restricting short, high frequency trips in a simulated environment does as much to slow disease spread as restricting long, low frequency trips. Agents in our simulations that take multiple trips around their neighbourhood spread disease as quickly as a those taking single, long trips to the city center. This is interesting academically, but more importantly, it has potential implications for policy: it means that restricting travel distance r but ignoring travel frequency f could be ineffective in containing the spreading of a disease.</p><p>That said, an important limitation of this study is that it restricts exploration velocity in a very specific way, removing trips that extend beyond a certain r or f . In reality, it is unclear how people would respond to policies that aim to restrict exploration velocity. True behavioral responses to different travel policies and what policies or mechanisms would be required to change individuals' behavior in a way that reduces exploration velocity are areas for future research. Further limitations of this study include the inherent limitations of simple epidemiological modeling; it is known that the SI, SIR, SEIR and other classic disease models make assumptions which bound their accuracy . The estimates for the model parameters likely carry error , and we also assumed that each agent was identical and that each exposure within a set radius carried equal likelihood of infection. Moreover, the scale of our analysis is restricted to the city level (because our datasets are collected at this level) and so our findings do not necessarily generalize to the perhaps more important case of country-level and international travel policies. If the r â€¢ f invariance, or some form of inverse relation between r and f , holds for international mobility patterns, then, as at the city scale, lax distance limits could be compensated by strict frequency restrictions. This is a bold hypothesis, which should be tested in future work. Metapopulation disease models , with their convenient trade off between realism and parsimony, seem like a good theoretical starting point for this effort.</p><p>Despite these limitations, we believe our results reveal novel insights and bring up an important component of human mobility that could prove useful for future policies on COVID-19 and other epidemic diseases. In the current COVID-19 pandemic and in future pandemics, evidence-based policies at the city scale (not to mention the national or international scale) are needed to mitigate speed and intensity of disease spread. Our results have the potential to help this effort by opening up investigation into the relationship between exploration velocity and epidemic spread. They indicate that the exploration velocity of a city's inhabitants r â€¢ f must be boundedto bound distance r but not visitation frequency f leaves potential for disease containment unmet. Furthermore, and more optimistically, if a bound on exploration velocity is possible and effective, it would mean that strict distance restrictions at the city or town level -as adopted by, e.g., Ireland and Italy at the beginning of the infection -are perhaps unnecessary. Given a desired bound on epidemic speed Ï„ bound , a large r could be offset by a small f ; allowing citizens to travel infrequently to distant services (doctors, hospitals etc) may be safe. This inverse relation between travel distance and frequency (r âˆ Ï„ bound /f ) could also vitally inform remote working policies supporting the hypothesis that working from home multiple days per week -and thereby limiting the visitation frequency to workplaces -helps prevent the spread of disease.</p></div>\n",
              " <div><p>New York City data. Individuals' movements in New York City are inferred from GPS traces collected from mobile phones by the company X-Mode over a span of one month (February 2020). The raw data contains about 479,163 anonymized users; our analysis uses 10,000 users randomly selected from those that appear in the dataset every day in the month of February. Dakar data. The Dakar dataset is based on anonymized Call Detailed Records (CDR) provided by the Data for Development (D4D) Challenge. The detailed information of this dataset is provided in . Here, we use the SET2, which includes individual trajectories for 300,000 FIG. : PEPR simulation results. When we run SEIR models across a set of trajectories M sim which have been created using the modified PEPR model with P travel = .25, we see a similar relationship between r â€¢ f and Ï„ to that in our real trajectories M real . When we run SEIR models across trajectories which have been created using the original PEPR model (with P travel = 1), we do not see this relationship. sampled users in Senegal, and after the preprocess, we have 173,000 users and 173 cells in Dakar region during two weeks of January, 2013. We subselect for users who appear at least 200 times in the dataset to ensure that we have adequate information about their trajectories over the two weeks. Data preprocessing. The X-Mode data from NYC is generated on an very fine spatial and temporal scale, with exact latitude and longitude coordinate updates as frequently as every second. The CDR data from Dakar, on the other hand, are generated only for voice calls, text messages or data exchanges and therefore have limited resolution in time. The geographic location of the cell towers and their density determines the accuracy of location measurements through triangularization techniques. Therefore, the trajectories extracted from CDRs constitute a discrete approximation of the moving population M (x; y; t). There are several steps in preprocessing of the data before it can be suitable for use in our analysis, which vary between the X-Mode data and the CDR data.</p><p>The main steps for the NYC data are: i) We use density-based spatial clustering of applications with noise (DBSCAN) to group tightly-clustered latitude/longitude pairs in each individual's trajectory into locations . If a cluster of at least five latitude/longitude points exists such that no point is more than .0004 degrees (about 56 meters) from two other points in the cluster, those points are grouped together as a single location. ii) Each agent is assigned the DBSCAN cluster it visits most as its home location. iii) We drop all locations in the trajectory that have been visited for less than a minimum time Ï„ min = 15min. iv) In order to restrict travel distance r, we calculate distance between locations by the haversine formula, which derives the great-circle distance between two points on a sphere. All locations that are more than r km from an agents home location are removed from their trajectory. v) In order to restrict travel frequency f , for each DBSCAN cluster that an agent visits more than f distinct times (where distinct visits are determined by an agent leaving a location and then coming back to it), we randomly select f visits to include in their trajectory and drop the rest.</p><p>The main steps for the Dakar data are: i) We view each cell tower as a different location in the city. ii) For each person, we determine the home location as the cell tower location which has been visited for the most cumulative time. By summing over all days in a given time window, one can find the home cell with high level of confidence for the majority of subjects. iii) We drop all locations in the trajectory that have been visited for less than a minimum time Ï„ min = 10min. iv) In order to restrict travel distance r, we calculate distance between cell towers by the haversine formula, which derives the great-circle distance between two points on a sphere. All cell towers that are more than r km from an agent's home location are removed from their trajectory. v) In order to restrict travel frequency f , for each location that an agent visits more than f distinct times (where distinct visits are determined by an agent leaving a location and then coming back to it), we randomly select f visits to include in their trajectory and drop the rest (excepting visits to an agent's home location, which are not restricted).</p><p>The duration of stay, frequency, and distance criteria on defining cell visits yields a list of cells visited by that subject over the study period for a given frequency restriction f and distance restriction r. Simulation details. We run an agent-based SEIR, SIR, and SI models with N = 10, 000 agents, 5% of which are initialized to be infected (1% in the SI and SIR Dakar models). Each agent is assigned the trajectory of a real person from our dataset, with location updated every 900 seconds (15 minutes) for the NYC simulations or every 600 seconds (10 minutes) for the Dakar simulations. At each time step, each user's location is updated according to their assigned trajectory and infection status is updated according to the following parameters, drawn from Chen 2020 's estimates of R0 = 3.58, incubation period = 5.2 days, and infection period = 5. In addition, we reproduce our results with disease parameters estimated for the more highly-contagious Delta variant of COVID-19 (Î² = 1.42, all other parameters remain the same, ) and the 2009 H1N1 influenza strain (Î² = .913, Î³ = 1.6, Ïƒ = 1, ). Finally, let I local and N local be the number of infected agents and total agents within a 190 meter radius of the agent's current location for the NYC data or within the same cell tower location for the Dakar data. Then, transition probabilities are:</p><p>In the SIR and SI models, if an agent becomes infected on a given day, they will become contagious at the start of the next day. Quantifying dispersion. We use the M function developed in  to quantify spatial dispersion of disease in our New York City simulations for a given r, f . The M function is calculated as follows: for each infected agent I and some radius k, we calculate the ratio between the proportion of agents within k of I which are infected to the proportion of agents in the total population which are infected. Summing this value over all infected agents I and dividing by N âˆ’ 1, where N is the number of infected agents, gives M (k), the M function evaluated at k. While M is generally analyzed as a function over all reasonable k, we evaluate the M function at a specific k in order to compare spatial dispersion across r â€¢ f values at that k, and then show that the relationship is robust to choice of k.</p><p>Confidence intervals for M (k) are obtained by Monte Carlo simulation-for a given epidemic size Ïˆ, we randomly assign Ïˆ infections across the population 1,000 times and calculate M Ïˆ each time. By taking the .025 and .975 quantiles of these simulated M , we form an upper and lower bound on M Ïˆ . It is notable that our empirical M never reaches this confidence band, implying that spatial dispersion is significantly non-homogenous for every k and Ïˆ. Robustness to model parameters. Here we demonstrate that the universal Ï„ (râ€¢f ) curve is robust to changes in the parameters R 0 by running our simulations with estimated transmission parameters for the 2009 H1N1 influenza strain (Î² = .913, Î³ = 1.6, Ïƒ = 1 ) and the Delta variant of COVID-19 (Î² = 1.42, ). Figure  shows that, with these parameters, the Ï„ (r â€¢ f ) relationship still holds. Preferential return model. The preferential return model proposed in  is based off of that proposed in Song et al. . With some probability P new , agents return to a location they have already visited; with probability 1 âˆ’ P new they visit a new location with distance drawn from the empirical distance distribution and direction drawn uniformly at random. Waiting times between trips are also drawn from the empirical distribution. The probability p is a function of the number of locations already visited:</p><p>where S is the number of locations visited. The parameters Ï and Î³ are fit to the real data using least-squares regression (using the NYC dataset, we find Ï = .500 and Î³ = .267.)</p><p>We add the following modification: while in the original PEPR model agents travel at each timestep, we set some probability P travel with which agents travel at any given timestep. Conversely, with probability 1 âˆ’ P travel , agents stay home. The simulation results reported here are from running our modified PEPR on a unit square for 500 timesteps.   </p></div>\n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " </body>]"
            ]
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "And one with Exception('[BAD_INPUT_DATA] PDF to XML conversion failed with error code: 1')."
      ],
      "metadata": {
        "id": "9tyWLkukqrYq"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_arxiv_removed_tags[3]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7RFHgYOf5FHG",
        "outputId": "d68fa99a-9219-4c35-a81b-51d87ca79a12"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[]"
            ]
          },
          "metadata": {},
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bzKanZNKxfPW"
      },
      "source": [
        "## 11.  Strip markup and keep text\n",
        "\n",
        "We only want to keep the human-readable text so we will use the get_text() method to return all the text in the articles as a single Unicode string without the `<body>`, `<div>` and `<p>` tags."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "kmWaXegyxfPW"
      },
      "outputs": [],
      "source": [
        "def strip_markup(articles):\n",
        "\n",
        "    for article in articles:\n",
        "\n",
        "        return article.get_text()\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "VnQTLrw7xfPW"
      },
      "outputs": [],
      "source": [
        "stripped_markup_articles = list(map(strip_markup, soup_results_arxiv_removed_tags))"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "66c4ed56-8fc4-4859-ec5d-c4a703abcdfd",
        "id": "9GmEMnV_xfPW"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "304"
            ]
          },
          "metadata": {},
          "execution_count": 41
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View article with all tags removed."
      ],
      "metadata": {
        "id": "jZypoVXR5Ckw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles[6]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 237
        },
        "outputId": "32d65dd8-2ecb-48fb-cc5e-7b38d3b793f5",
        "id": "2udgg55BxfPW"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'\\nThe global impact of the COVID-19 pandemic on both human health and socioeconomic activity has brought to light the importance of a nuanced understanding of the way that epidemics spread in cities. Urban spread of disease is inherently tied to human mobility: as we move through and between cities, we serve as vectors that allow disease to spread to new individuals and communities. The nature of the relationship between mobility and spread of disease has been extensively studied; it is widely understood that human travel is a driving force behind disease spread . For this reason, many public policy interventions implemented worldwide to contain the spread of COVID-19 focused on limiting mobility, restricting the radius that individuals could travel from their homes and limiting travel between cities and countries.As we continue to learn more about the ways that humans move, it is important to apply new discoveries about human mobility to the study of disease spread, deepening our understanding of urban epidemics for use in future public health crises. Recent research uncovered a clear, universal pattern in urban mobility: the total distance that the average visitor to a given location travels to reach it is constant across a city, unrelated to the lo-cation\\'s overall attractiveness . This value, equivalent to the distance of a location to an individual\\'s home multiplied by the number of times that they visit it over the course of some time period, can be thought of as an \"exploration velocity\": it is the effective distance individuals travel towards any given location per unit of time. It is a simple but critical parameter to understanding human movement.Given the simplicity and apparent universality of this parameter of human mobility, its relationship to disease spread is a natural next question. Many travel restrictions and recommendations during the 2020 and 2021 COVID-19 pandemic focused on confining movements to a certain geographical area-for example, restricting the radius that people are able to travel to one\\'s neighborhood, city, state, or country. But what if speed of disease spread depends not just on radius of travel r, but also on the product v := r â€¢ f ? If v has additional impact on disease spread, beyond average travel radius r, COVID-19 policies and restrictions that focused solely on restricting radius may have left significant unmet potential for further disease mitigation. This paper investigates the relationship between exploration velocity v := r â€¢ f and spread of disease, opening the door to future research on the ways in which that relationship can be utilized to contain epidemics.\\nSimulating reduction of exploration velocity on real data. In order to understand how v = r â€¢ f interacts with speed of disease spread, we start with largescale datasets of individual human movements in New York City and Dakar, Senegal, which we call M real . The datasets each consist of a set of trips for N individuals over different time periods T , where each trip indicates a given individual moving between two locations. We then use agent based susceptible-infected (SI), susceptible-infected-recovered (SIR), and susceptibleexposed-infected-recovered (SEIR) models calibrated with estimates for COVID-19  to simulate disease spread as agents follow the trajectories in our datasets. Aside from their unique trajectories, each individual is assumed identical. In each simulation, we vary two parameters: the maximum travel distance r, measured relative to each agent\\'s home location (see Methods for details), and the travel frequency f , the maximum number of times each location was visited. By varying these two parameters together, we are able to manually adjust exploration velocity v in the system. In practice, this means discarding any trip of length greater than r and all but f randomly selected trips to a given location from our datasets. The details of the spatial partitionings used as well as other simulation details are given in the Methods.Figures ) and (b) plot the time it takes for the disease to reach some set proportion Ï„ of the population against exploration velocity for each dataset. The value of Ï„ is set to 29% for our Dakar data and 57% for our New York City data. Thresholds are chosen to be the total epidemic size at the end of the study period under the strictest restrictions for each study area, respectively. Simulations were run over the entire length of the datasets (14 days for the Dakar dataset; 28 days for New York City) with 10,000 agents and a 5% initial infected population. The trends are intuitive and not surprising. For a given f , Ï„ decreases monotonically with r: the further people are allowed travel, the faster the epidemic spreads. Similarly, for a given r, Ï„ decreases monotonically with f : more frequent return trips leads to faster-spreading epidemics. What is surprising, however, is that the Ï„ (r) curves for each value of f have similar shape. This echoes a previous finding  and hints that Ï„, r and f might have a simple relationship. Figures 1(c) and (d) show that they do. Under the rescaling r â†’ r â€¢f all the data appear to merge, collapsing to a single curve that depends on the unifying factor v := r â€¢ f , or exploration velocity. This curve can be empirically characterized aswhere a and b vary across cities and model types. Figure  in the Supplementary Material shows the same scaling collapse is achieved when the SI and SIR models are used, and Figure  in the SM shows it persists for different values of the disease parameter R 0 . Taken together, these findings suggest the r â†’ râ€¢f collapse persists across multiple disease processes.After observing the relationship between r â€¢ f and epidemic size, we check whether there is a relationship between r â€¢ f and spatial dispersion of the disease. We analyze the spatial concentration of infected persons in our New York City simulations using the M function developed by Marcon and Puech, which is calculated as a function of some radius k around each agent -see Methods for a detailed description of this statistic. High M indicates tighter clusters of cases; low M indicates a more homogeneous distribution of cases across space. We find that for a given radius k, as r and f increase, M (k, r, f ) decreases: infections become consistently more dispersed across the city (see Figure  for this relationship when k = 700m). As with epidemic size, under the rescaling r â†’ r â€¢ f , this relationship collapses to a single curve, as shown in Figure . This indicates that restricting r â€¢ f is more effective than restricting r alone at containing geographic spread of disease, just as with slowing epidemic spread. This relationship is robust to the choice of k-as k increases, M (k, r, f ) decreases (as would be expected-the larger radius you look at around an agent, the more representative sample of the whole population you will capture) but stays significant and maintains its relationship to r â€¢ f , as illustrated in Figure  in the Supplementary Material. This relationship is demonstrated visually in Figures  and. When r is fixed at 8, increasing f from 1 to 6 has a clear effect on the extent to which infections spread across the city (Figure ); while significant portions of the city remain untouched by infection when f = 1, many more grid cells contain at least one infection when f = 6. In Figure , on the other hand, we see that when r á¸Ÿ is held fixed at 12, spatial distribution of infection rates is almost identical for different r and f values, demonstrating the relationship between spatial distribution and r â€¢f .\\nIn order to illustrate the potential mechanism by which r â€¢ f determines disease spread time, we assess the relationship between râ€¢f and both number and variance of contacts between agents in our simulations. Think of our simulations as a network, where agents are nodes and edges are formed whenever two agents come within some distance of one another, enabling disease transmission. It is a known result of the SIR model that speed at which a disease spreads through a network of individuals depends not only on the average degree, or number of contacts per individual k , but also on the variance of contacts (via k 2 , or mean squared degree) according to the following relation:where Ï„ is the characteristic time, or time it takes for the disease to reach 1/e = 36% of the population, Î³ is the daily recovery parameter and Î² is the infection parameter. This relation isn\\'t perfectly applicable to our context-it assumes exposure to all contacts at all timesteps, whereas our simulations incorporate movement in space and thus non-constant degree k over time. However, it is valuable in illustrating the effects of restricting r and f in our dataset. Restricting radius of travel r and frequency of return f affects both k and k 2 . The closer you stay to home and the fewer trips you take, the fewer unique individuals you have the opportunity to encounter (reducing average number of contacts k ). Further, as stricter restrictions are enforced, high-r, high-f trips are removed and the mobility patterns of high-frequency, long-distance travellers start to look more like the mobility patterns of low-frequency travellers who stay close to home (reducing variance of contacts k 2 ). Figure  plots these relationships in our New York City data, showing exploration velocity r â€¢ f on the x-axis against average number of contacts k and average squared number of contacts k 2 on the y-axis in panels a and b, respectively. Again, we see a logarithmic relationship. Plugging both k and k 2 into the relation above, we see that characteristic time as predicted by the degree distribution (Ï„ ) is proportional to the true characteristic time (Ï„ ) that we see in our simulations -Figure , panel c. Taken together, these findings show that the shape of the relationship between r â€¢ f and Ï„ can be predicted by decreases in number and variance of contacts between agents in combination with fundamental SIR modeling results.These results can be replicated by a simple modification of the preferential exploration and preferential return (PEPR) mobility model presented in previous literature . Under the PEPR model as proposed in , at each timestep, agents either explore a new location with a fixed probability P new or return to a previously-visited location with complementary probability 1 âˆ’ P new . In order to choose a new location, agents draw a radius of travel âˆ†r from a heavy-tailed distribution with exponent P (âˆ†r) âˆ¼ |âˆ†r| âˆ’1âˆ’Î± , with Î± = 0.55 taken from , and they draw an angle of travel Î¸ with preference towards Î¸ that are heavily visited by other agents (see Methods for details). As demonstrated in , the PEPR model replicates the universality of r â€¢ f . However, when we run our SEIR model on mobility trajectories simulated with the PEPR model, we do not see the same relationship between r â€¢ f and Ï„ that we see in our real data. This is because, in the original PEPR model, agents are always moving from place to place without necessarily returning home, whereas in our real mobility trajectories, time spent at home means less exposure to new, unique contacts (and more exposure to the same neighbors). This leads to the PEPR model being unable to replicate the changes in k that are associated with loosening or tightening radius restrictions, as seen in Figures  and. In order to correct for this, we add a probability of travel P travel -at each timestep, with probability P travel , the agent travels to a new location according to the protocol above, whereas with complementary probability 1 âˆ’ P travel the agent stays or returns home. Figures  and show the results of rerunning the same SEIR simulations that were run on our real mobility traces M real on a set of simulated trajectories M sim derived from the modified PEPR model. In order to implement   radius restrictions within the PEPR model we restrict the power law distribution from which âˆ†r is drawn and in order to implement frequency restrictions we delete all trips beyond frequency f to the same location. We set P travel = .25, the average proportion of time spent away from home in M real , and run the PEPR simulations on a unit square (see Methods for more details). We see the same logarithmic relationship between r â€¢ f and both Ï„ and Ï„ (expected Ï„ derived from k and k 2 ) that we see in our simulations over M real , although the simulated results are less precise than the results we see in M real . We also replicate the simulations with P travel = .40 and see similar results (see Figure  in the SM). This indicates that the modified PEPR model is able to replicate the relationship between exploration velocity and speed of epidemic spread.\\nThe significance of r â€¢ f was revealed in recent work  on human mobility patterns which shows the number of people who visit a given location f times from a distance r during a certain reference period follow a universal, inverse square law N (r, f ) âˆ 1/(rf ) 2 . We suspect the Ï„ (rf ) curve found in this paper derives from this inverse square law; while we have discussed the relationship between exploration velocity and network contacts, deriving the exact mechanism linking the universal inverse square law to Ï„ (rf ) is an open question for future work.An important implication of the Ï„ (r â€¢ f ) curve is that restricting short, high frequency trips in a simulated environment does as much to slow disease spread as restricting long, low frequency trips. Agents in our simulations that take multiple trips around their neighbourhood spread disease as quickly as a those taking single, long trips to the city center. This is interesting academically, but more importantly, it has potential implications for policy: it means that restricting travel distance r but ignoring travel frequency f could be ineffective in containing the spreading of a disease.That said, an important limitation of this study is that it restricts exploration velocity in a very specific way, removing trips that extend beyond a certain r or f . In reality, it is unclear how people would respond to policies that aim to restrict exploration velocity. True behavioral responses to different travel policies and what policies or mechanisms would be required to change individuals\\' behavior in a way that reduces exploration velocity are areas for future research. Further limitations of this study include the inherent limitations of simple epidemiological modeling; it is known that the SI, SIR, SEIR and other classic disease models make assumptions which bound their accuracy . The estimates for the model parameters likely carry error , and we also assumed that each agent was identical and that each exposure within a set radius carried equal likelihood of infection. Moreover, the scale of our analysis is restricted to the city level (because our datasets are collected at this level) and so our findings do not necessarily generalize to the perhaps more important case of country-level and international travel policies. If the r â€¢ f invariance, or some form of inverse relation between r and f , holds for international mobility patterns, then, as at the city scale, lax distance limits could be compensated by strict frequency restrictions. This is a bold hypothesis, which should be tested in future work. Metapopulation disease models , with their convenient trade off between realism and parsimony, seem like a good theoretical starting point for this effort.Despite these limitations, we believe our results reveal novel insights and bring up an important component of human mobility that could prove useful for future policies on COVID-19 and other epidemic diseases. In the current COVID-19 pandemic and in future pandemics, evidence-based policies at the city scale (not to mention the national or international scale) are needed to mitigate speed and intensity of disease spread. Our results have the potential to help this effort by opening up investigation into the relationship between exploration velocity and epidemic spread. They indicate that the exploration velocity of a city\\'s inhabitants r â€¢ f must be boundedto bound distance r but not visitation frequency f leaves potential for disease containment unmet. Furthermore, and more optimistically, if a bound on exploration velocity is possible and effective, it would mean that strict distance restrictions at the city or town level -as adopted by, e.g., Ireland and Italy at the beginning of the infection -are perhaps unnecessary. Given a desired bound on epidemic speed Ï„ bound , a large r could be offset by a small f ; allowing citizens to travel infrequently to distant services (doctors, hospitals etc) may be safe. This inverse relation between travel distance and frequency (r âˆ Ï„ bound /f ) could also vitally inform remote working policies supporting the hypothesis that working from home multiple days per week -and thereby limiting the visitation frequency to workplaces -helps prevent the spread of disease.\\nNew York City data. Individuals\\' movements in New York City are inferred from GPS traces collected from mobile phones by the company X-Mode over a span of one month (February 2020). The raw data contains about 479,163 anonymized users; our analysis uses 10,000 users randomly selected from those that appear in the dataset every day in the month of February. Dakar data. The Dakar dataset is based on anonymized Call Detailed Records (CDR) provided by the Data for Development (D4D) Challenge. The detailed information of this dataset is provided in . Here, we use the SET2, which includes individual trajectories for 300,000 FIG. : PEPR simulation results. When we run SEIR models across a set of trajectories M sim which have been created using the modified PEPR model with P travel = .25, we see a similar relationship between r â€¢ f and Ï„ to that in our real trajectories M real . When we run SEIR models across trajectories which have been created using the original PEPR model (with P travel = 1), we do not see this relationship. sampled users in Senegal, and after the preprocess, we have 173,000 users and 173 cells in Dakar region during two weeks of January, 2013. We subselect for users who appear at least 200 times in the dataset to ensure that we have adequate information about their trajectories over the two weeks. Data preprocessing. The X-Mode data from NYC is generated on an very fine spatial and temporal scale, with exact latitude and longitude coordinate updates as frequently as every second. The CDR data from Dakar, on the other hand, are generated only for voice calls, text messages or data exchanges and therefore have limited resolution in time. The geographic location of the cell towers and their density determines the accuracy of location measurements through triangularization techniques. Therefore, the trajectories extracted from CDRs constitute a discrete approximation of the moving population M (x; y; t). There are several steps in preprocessing of the data before it can be suitable for use in our analysis, which vary between the X-Mode data and the CDR data.The main steps for the NYC data are: i) We use density-based spatial clustering of applications with noise (DBSCAN) to group tightly-clustered latitude/longitude pairs in each individual\\'s trajectory into locations . If a cluster of at least five latitude/longitude points exists such that no point is more than .0004 degrees (about 56 meters) from two other points in the cluster, those points are grouped together as a single location. ii) Each agent is assigned the DBSCAN cluster it visits most as its home location. iii) We drop all locations in the trajectory that have been visited for less than a minimum time Ï„ min = 15min. iv) In order to restrict travel distance r, we calculate distance between locations by the haversine formula, which derives the great-circle distance between two points on a sphere. All locations that are more than r km from an agents home location are removed from their trajectory. v) In order to restrict travel frequency f , for each DBSCAN cluster that an agent visits more than f distinct times (where distinct visits are determined by an agent leaving a location and then coming back to it), we randomly select f visits to include in their trajectory and drop the rest.The main steps for the Dakar data are: i) We view each cell tower as a different location in the city. ii) For each person, we determine the home location as the cell tower location which has been visited for the most cumulative time. By summing over all days in a given time window, one can find the home cell with high level of confidence for the majority of subjects. iii) We drop all locations in the trajectory that have been visited for less than a minimum time Ï„ min = 10min. iv) In order to restrict travel distance r, we calculate distance between cell towers by the haversine formula, which derives the great-circle distance between two points on a sphere. All cell towers that are more than r km from an agent\\'s home location are removed from their trajectory. v) In order to restrict travel frequency f , for each location that an agent visits more than f distinct times (where distinct visits are determined by an agent leaving a location and then coming back to it), we randomly select f visits to include in their trajectory and drop the rest (excepting visits to an agent\\'s home location, which are not restricted).The duration of stay, frequency, and distance criteria on defining cell visits yields a list of cells visited by that subject over the study period for a given frequency restriction f and distance restriction r. Simulation details. We run an agent-based SEIR, SIR, and SI models with N = 10, 000 agents, 5% of which are initialized to be infected (1% in the SI and SIR Dakar models). Each agent is assigned the trajectory of a real person from our dataset, with location updated every 900 seconds (15 minutes) for the NYC simulations or every 600 seconds (10 minutes) for the Dakar simulations. At each time step, each user\\'s location is updated according to their assigned trajectory and infection status is updated according to the following parameters, drawn from Chen 2020 \\'s estimates of R0 = 3.58, incubation period = 5.2 days, and infection period = 5. In addition, we reproduce our results with disease parameters estimated for the more highly-contagious Delta variant of COVID-19 (Î² = 1.42, all other parameters remain the same, ) and the 2009 H1N1 influenza strain (Î² = .913, Î³ = 1.6, Ïƒ = 1, ). Finally, let I local and N local be the number of infected agents and total agents within a 190 meter radius of the agent\\'s current location for the NYC data or within the same cell tower location for the Dakar data. Then, transition probabilities are:In the SIR and SI models, if an agent becomes infected on a given day, they will become contagious at the start of the next day. Quantifying dispersion. We use the M function developed in  to quantify spatial dispersion of disease in our New York City simulations for a given r, f . The M function is calculated as follows: for each infected agent I and some radius k, we calculate the ratio between the proportion of agents within k of I which are infected to the proportion of agents in the total population which are infected. Summing this value over all infected agents I and dividing by N âˆ’ 1, where N is the number of infected agents, gives M (k), the M function evaluated at k. While M is generally analyzed as a function over all reasonable k, we evaluate the M function at a specific k in order to compare spatial dispersion across r â€¢ f values at that k, and then show that the relationship is robust to choice of k.Confidence intervals for M (k) are obtained by Monte Carlo simulation-for a given epidemic size Ïˆ, we randomly assign Ïˆ infections across the population 1,000 times and calculate M Ïˆ each time. By taking the .025 and .975 quantiles of these simulated M , we form an upper and lower bound on M Ïˆ . It is notable that our empirical M never reaches this confidence band, implying that spatial dispersion is significantly non-homogenous for every k and Ïˆ. Robustness to model parameters. Here we demonstrate that the universal Ï„ (râ€¢f ) curve is robust to changes in the parameters R 0 by running our simulations with estimated transmission parameters for the 2009 H1N1 influenza strain (Î² = .913, Î³ = 1.6, Ïƒ = 1 ) and the Delta variant of COVID-19 (Î² = 1.42, ). Figure  shows that, with these parameters, the Ï„ (r â€¢ f ) relationship still holds. Preferential return model. The preferential return model proposed in  is based off of that proposed in Song et al. . With some probability P new , agents return to a location they have already visited; with probability 1 âˆ’ P new they visit a new location with distance drawn from the empirical distance distribution and direction drawn uniformly at random. Waiting times between trips are also drawn from the empirical distribution. The probability p is a function of the number of locations already visited:where S is the number of locations visited. The parameters Ï and Î³ are fit to the real data using least-squares regression (using the NYC dataset, we find Ï = .500 and Î³ = .267.)We add the following modification: while in the original PEPR model agents travel at each timestep, we set some probability P travel with which agents travel at any given timestep. Conversely, with probability 1 âˆ’ P travel , agents stay home. The simulation results reported here are from running our modified PEPR on a unit square for 500 timesteps.   \\n\\n\\n\\n\\n\\n\\n\\n\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 42
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "print(stripped_markup_articles[3])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b868f134-9b55-47c0-c431-4529a5cde10d",
        "id": "FLNN6zHrxfPX"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "None\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "JlVmEySwxfPX"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-06_stripped_markup_articles_304.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles, f)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 12. Add extracted full text to DataFrame\n",
        "\n",
        "Read in  arXiv metadata."
      ],
      "metadata": {
        "id": "idc1SjNzxfPY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_304 = pd.read_pickle('2023-01-06_arxiv_results_for_dl.pickle')\n",
        "article_results_arxiv_304"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "a061e78d-7082-4215-f46f-c572168f462f",
        "id": "T2V6yChTxfPX"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         arxiv-id   published     revised  \\\n",
              "0    2109.06377v4  2021-09-14  2022-12-22   \n",
              "1    2212.09867v1  2022-12-19  2022-12-19   \n",
              "2    2212.09610v1  2022-12-19  2022-12-19   \n",
              "3    2212.00023v2  2022-11-30  2022-12-08   \n",
              "4    2212.03911v1  2022-12-07  2022-12-07   \n",
              "..            ...         ...         ...   \n",
              "299  2003.13665v1  2020-03-30  2020-03-30   \n",
              "300  2003.14258v1  2020-03-30  2020-03-30   \n",
              "301  2003.12454v1  2020-03-26  2020-03-26   \n",
              "302  2003.04524v1  2020-03-10  2020-03-10   \n",
              "303  2003.00751v1  2020-03-02  2020-03-02   \n",
              "\n",
              "                                                 title journal  \\\n",
              "0    ASGARD: A Single-cell Guided pipeline to Aid R...    None   \n",
              "1    Detecting Contradictory COVID-19 Drug Efficacy...    None   \n",
              "2    Drying of Bio-colloidal Sessile Droplets: Adva...    None   \n",
              "3    Random Copolymer inverse design system orienti...    None   \n",
              "4    Analysis of Drug repurposing Knowledge graphs ...    None   \n",
              "..                                                 ...     ...   \n",
              "299  Genomics-guided molecular maps of coronavirus ...    None   \n",
              "300  Nanomechanical sonification of the 2019-nCoV c...    None   \n",
              "301  A Machine Learning alternative to placebo-cont...    None   \n",
              "302  Old Drugs for Newly Emerging Viral Disease, CO...    None   \n",
              "303  Regulation of interferon production as a poten...    None   \n",
              "\n",
              "                                               authors   doi  \\\n",
              "0    Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...  None   \n",
              "1    Daniel N. Sosa, Malavika Suresh, Christopher P...  None   \n",
              "2           Anusuya Pal, Amalesh Gope, Anupam Sengupta  None   \n",
              "3                                 Tianyu Wu, Yang Tang  None   \n",
              "4                                  Ajay Kumar Gogineni  None   \n",
              "..                                                 ...   ...   \n",
              "299                                    Gennadi Glinsky  None   \n",
              "300                                  Markus J. Buehler  None   \n",
              "301   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc  None   \n",
              "302                                Mohammad Reza Dayer  None   \n",
              "303             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei  None   \n",
              "\n",
              "                                      pdf_url  \n",
              "0    http://export.arxiv.org/pdf/2109.06377v4  \n",
              "1    http://export.arxiv.org/pdf/2212.09867v1  \n",
              "2    http://export.arxiv.org/pdf/2212.09610v1  \n",
              "3    http://export.arxiv.org/pdf/2212.00023v2  \n",
              "4    http://export.arxiv.org/pdf/2212.03911v1  \n",
              "..                                        ...  \n",
              "299  http://export.arxiv.org/pdf/2003.13665v1  \n",
              "300  http://export.arxiv.org/pdf/2003.14258v1  \n",
              "301  http://export.arxiv.org/pdf/2003.12454v1  \n",
              "302  http://export.arxiv.org/pdf/2003.04524v1  \n",
              "303  http://export.arxiv.org/pdf/2003.00751v1  \n",
              "\n",
              "[304 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-675f079f-bc07-4109-9867-f6553aed30b1\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>arxiv-id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>2109.06377v4</td>\n",
              "      <td>2021-09-14</td>\n",
              "      <td>2022-12-22</td>\n",
              "      <td>ASGARD: A Single-cell Guided pipeline to Aid R...</td>\n",
              "      <td>None</td>\n",
              "      <td>Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2109.06377v4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>2212.09867v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Detecting Contradictory COVID-19 Drug Efficacy...</td>\n",
              "      <td>None</td>\n",
              "      <td>Daniel N. Sosa, Malavika Suresh, Christopher P...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09867v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2212.09610v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Drying of Bio-colloidal Sessile Droplets: Adva...</td>\n",
              "      <td>None</td>\n",
              "      <td>Anusuya Pal, Amalesh Gope, Anupam Sengupta</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09610v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>2212.00023v2</td>\n",
              "      <td>2022-11-30</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>Random Copolymer inverse design system orienti...</td>\n",
              "      <td>None</td>\n",
              "      <td>Tianyu Wu, Yang Tang</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.00023v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>2212.03911v1</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>Analysis of Drug repurposing Knowledge graphs ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ajay Kumar Gogineni</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.03911v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>299</th>\n",
              "      <td>2003.13665v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Genomics-guided molecular maps of coronavirus ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Gennadi Glinsky</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.13665v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>300</th>\n",
              "      <td>2003.14258v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Nanomechanical sonification of the 2019-nCoV c...</td>\n",
              "      <td>None</td>\n",
              "      <td>Markus J. Buehler</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.14258v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>301</th>\n",
              "      <td>2003.12454v1</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>A Machine Learning alternative to placebo-cont...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ezequiel Alvarez, Federico Lamagna, Manuel Szewc</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.12454v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>302</th>\n",
              "      <td>2003.04524v1</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>Old Drugs for Newly Emerging Viral Disease, CO...</td>\n",
              "      <td>None</td>\n",
              "      <td>Mohammad Reza Dayer</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.04524v1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>303</th>\n",
              "      <td>2003.00751v1</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>Regulation of interferon production as a poten...</td>\n",
              "      <td>None</td>\n",
              "      <td>Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.00751v1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>304 rows Ã— 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-675f079f-bc07-4109-9867-f6553aed30b1')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-675f079f-bc07-4109-9867-f6553aed30b1 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-675f079f-bc07-4109-9867-f6553aed30b1');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 46
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Add full text with stripped markup as 'text' column to DataFrame."
      ],
      "metadata": {
        "id": "2DcFoKlL5lEG"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_304['text'] = stripped_markup_articles\n",
        "article_results_arxiv_304"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "a48330b6-da75-4e04-97dd-f4527dcb955e",
        "id": "3HuLBNWPxfPY"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         arxiv-id   published     revised  \\\n",
              "0    2109.06377v4  2021-09-14  2022-12-22   \n",
              "1    2212.09867v1  2022-12-19  2022-12-19   \n",
              "2    2212.09610v1  2022-12-19  2022-12-19   \n",
              "3    2212.00023v2  2022-11-30  2022-12-08   \n",
              "4    2212.03911v1  2022-12-07  2022-12-07   \n",
              "..            ...         ...         ...   \n",
              "299  2003.13665v1  2020-03-30  2020-03-30   \n",
              "300  2003.14258v1  2020-03-30  2020-03-30   \n",
              "301  2003.12454v1  2020-03-26  2020-03-26   \n",
              "302  2003.04524v1  2020-03-10  2020-03-10   \n",
              "303  2003.00751v1  2020-03-02  2020-03-02   \n",
              "\n",
              "                                                 title journal  \\\n",
              "0    ASGARD: A Single-cell Guided pipeline to Aid R...    None   \n",
              "1    Detecting Contradictory COVID-19 Drug Efficacy...    None   \n",
              "2    Drying of Bio-colloidal Sessile Droplets: Adva...    None   \n",
              "3    Random Copolymer inverse design system orienti...    None   \n",
              "4    Analysis of Drug repurposing Knowledge graphs ...    None   \n",
              "..                                                 ...     ...   \n",
              "299  Genomics-guided molecular maps of coronavirus ...    None   \n",
              "300  Nanomechanical sonification of the 2019-nCoV c...    None   \n",
              "301  A Machine Learning alternative to placebo-cont...    None   \n",
              "302  Old Drugs for Newly Emerging Viral Disease, CO...    None   \n",
              "303  Regulation of interferon production as a poten...    None   \n",
              "\n",
              "                                               authors   doi  \\\n",
              "0    Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...  None   \n",
              "1    Daniel N. Sosa, Malavika Suresh, Christopher P...  None   \n",
              "2           Anusuya Pal, Amalesh Gope, Anupam Sengupta  None   \n",
              "3                                 Tianyu Wu, Yang Tang  None   \n",
              "4                                  Ajay Kumar Gogineni  None   \n",
              "..                                                 ...   ...   \n",
              "299                                    Gennadi Glinsky  None   \n",
              "300                                  Markus J. Buehler  None   \n",
              "301   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc  None   \n",
              "302                                Mohammad Reza Dayer  None   \n",
              "303             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei  None   \n",
              "\n",
              "                                      pdf_url  \\\n",
              "0    http://export.arxiv.org/pdf/2109.06377v4   \n",
              "1    http://export.arxiv.org/pdf/2212.09867v1   \n",
              "2    http://export.arxiv.org/pdf/2212.09610v1   \n",
              "3    http://export.arxiv.org/pdf/2212.00023v2   \n",
              "4    http://export.arxiv.org/pdf/2212.03911v1   \n",
              "..                                        ...   \n",
              "299  http://export.arxiv.org/pdf/2003.13665v1   \n",
              "300  http://export.arxiv.org/pdf/2003.14258v1   \n",
              "301  http://export.arxiv.org/pdf/2003.12454v1   \n",
              "302  http://export.arxiv.org/pdf/2003.04524v1   \n",
              "303  http://export.arxiv.org/pdf/2003.00751v1   \n",
              "\n",
              "                                                  text  \n",
              "0    \\nHeterogeneity, or more specifically, the div...  \n",
              "1    \\nThe COVID-19 pandemic caused by the novel SA...  \n",
              "2    \\nVirus Emulating Particles (VEP) White Blood ...  \n",
              "3                                                 None  \n",
              "4    \\nIn the initial stages of a viral outbreak su...  \n",
              "..                                                 ...  \n",
              "299  \\nCoronavirus pandemic 2020 caused by the newl...  \n",
              "300  \\nProteins are the building blocks of virtuall...  \n",
              "301  \\nCurrent and last decades research in drug di...  \n",
              "302  \\nThe outbreak of coronavirus  in 2019-2020 is...  \n",
              "303  \\nNovel Coronavirus SARS-CoV-2 is a single str...  \n",
              "\n",
              "[304 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-4ed1a1a8-4280-43d8-b6c2-3881f7974af3\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>arxiv-id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>2109.06377v4</td>\n",
              "      <td>2021-09-14</td>\n",
              "      <td>2022-12-22</td>\n",
              "      <td>ASGARD: A Single-cell Guided pipeline to Aid R...</td>\n",
              "      <td>None</td>\n",
              "      <td>Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2109.06377v4</td>\n",
              "      <td>\\nHeterogeneity, or more specifically, the div...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>2212.09867v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Detecting Contradictory COVID-19 Drug Efficacy...</td>\n",
              "      <td>None</td>\n",
              "      <td>Daniel N. Sosa, Malavika Suresh, Christopher P...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09867v1</td>\n",
              "      <td>\\nThe COVID-19 pandemic caused by the novel SA...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2212.09610v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Drying of Bio-colloidal Sessile Droplets: Adva...</td>\n",
              "      <td>None</td>\n",
              "      <td>Anusuya Pal, Amalesh Gope, Anupam Sengupta</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09610v1</td>\n",
              "      <td>\\nVirus Emulating Particles (VEP) White Blood ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>2212.00023v2</td>\n",
              "      <td>2022-11-30</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>Random Copolymer inverse design system orienti...</td>\n",
              "      <td>None</td>\n",
              "      <td>Tianyu Wu, Yang Tang</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.00023v2</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>2212.03911v1</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>Analysis of Drug repurposing Knowledge graphs ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ajay Kumar Gogineni</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.03911v1</td>\n",
              "      <td>\\nIn the initial stages of a viral outbreak su...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>299</th>\n",
              "      <td>2003.13665v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Genomics-guided molecular maps of coronavirus ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Gennadi Glinsky</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.13665v1</td>\n",
              "      <td>\\nCoronavirus pandemic 2020 caused by the newl...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>300</th>\n",
              "      <td>2003.14258v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Nanomechanical sonification of the 2019-nCoV c...</td>\n",
              "      <td>None</td>\n",
              "      <td>Markus J. Buehler</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.14258v1</td>\n",
              "      <td>\\nProteins are the building blocks of virtuall...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>301</th>\n",
              "      <td>2003.12454v1</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>A Machine Learning alternative to placebo-cont...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ezequiel Alvarez, Federico Lamagna, Manuel Szewc</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.12454v1</td>\n",
              "      <td>\\nCurrent and last decades research in drug di...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>302</th>\n",
              "      <td>2003.04524v1</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>Old Drugs for Newly Emerging Viral Disease, CO...</td>\n",
              "      <td>None</td>\n",
              "      <td>Mohammad Reza Dayer</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.04524v1</td>\n",
              "      <td>\\nThe outbreak of coronavirus  in 2019-2020 is...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>303</th>\n",
              "      <td>2003.00751v1</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>Regulation of interferon production as a poten...</td>\n",
              "      <td>None</td>\n",
              "      <td>Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.00751v1</td>\n",
              "      <td>\\nNovel Coronavirus SARS-CoV-2 is a single str...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>304 rows Ã— 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4ed1a1a8-4280-43d8-b6c2-3881f7974af3')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-4ed1a1a8-4280-43d8-b6c2-3881f7974af3 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-4ed1a1a8-4280-43d8-b6c2-3881f7974af3');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 47
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_article_results_arxiv_304_full_text.pickle', 'wb') as f:\n",
        "  pickle.dump(article_results_arxiv_304, f)"
      ],
      "metadata": {
        "id": "WshnadmqxfPY"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 13. Check for missing text\n",
        "\n",
        "Concise summary of DataFrame to see if there are any articles with missing text."
      ],
      "metadata": {
        "id": "5Ailt6ik7v0I"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_304.info()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NNH3m30L3aI0",
        "outputId": "24cf0ac1-2e16-4526-8940-5a86c377fdb4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 304 entries, 0 to 303\n",
            "Data columns (total 9 columns):\n",
            " #   Column     Non-Null Count  Dtype \n",
            "---  ------     --------------  ----- \n",
            " 0   arxiv-id   304 non-null    object\n",
            " 1   published  304 non-null    object\n",
            " 2   revised    304 non-null    object\n",
            " 3   title      304 non-null    object\n",
            " 4   journal    39 non-null     object\n",
            " 5   authors    304 non-null    object\n",
            " 6   doi        63 non-null     object\n",
            " 7   pdf_url    304 non-null    object\n",
            " 8   text       300 non-null    object\n",
            "dtypes: object(9)\n",
            "memory usage: 21.5+ KB\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "As expected, the four with exceptions."
      ],
      "metadata": {
        "id": "xyqf_DMF729I"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_304[article_results_arxiv_304['text'].isna()]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 586
        },
        "id": "BJaGZMXQ4jay",
        "outputId": "d3ee2bf2-4144-4bf6-fc1b-d9c31452bc90"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         arxiv-id   published     revised  \\\n",
              "3    2212.00023v2  2022-11-30  2022-12-08   \n",
              "69   2109.00100v4  2021-08-31  2021-09-07   \n",
              "70   2109.00435v3  2021-09-01  2021-09-07   \n",
              "176  2007.09186v3  2020-07-17  2020-10-07   \n",
              "\n",
              "                                                 title journal  \\\n",
              "3    Random Copolymer inverse design system orienti...    None   \n",
              "69   Proceedings of KDD 2021 Workshop on Data-drive...    None   \n",
              "70   Proceedings of KDD 2020 Workshop on Data-drive...    None   \n",
              "176  AWS CORD-19 Search: A Neural Search Engine for...    None   \n",
              "\n",
              "                                               authors   doi  \\\n",
              "3                                 Tianyu Wu, Yang Tang  None   \n",
              "69   Snehalkumar, S. Gaikwad, Shankar Iyer, Dalton ...  None   \n",
              "70   Snehalkumar, S. Gaikwad, Shankar Iyer, Dalton ...  None   \n",
              "176  Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...  None   \n",
              "\n",
              "                                      pdf_url  text  \n",
              "3    http://export.arxiv.org/pdf/2212.00023v2  None  \n",
              "69   http://export.arxiv.org/pdf/2109.00100v4  None  \n",
              "70   http://export.arxiv.org/pdf/2109.00435v3  None  \n",
              "176  http://export.arxiv.org/pdf/2007.09186v3  None  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-f50ee9c8-913e-426f-ad13-2b0c4b0ba7f0\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>arxiv-id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>2212.00023v2</td>\n",
              "      <td>2022-11-30</td>\n",
              "      <td>2022-12-08</td>\n",
              "      <td>Random Copolymer inverse design system orienti...</td>\n",
              "      <td>None</td>\n",
              "      <td>Tianyu Wu, Yang Tang</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.00023v2</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>69</th>\n",
              "      <td>2109.00100v4</td>\n",
              "      <td>2021-08-31</td>\n",
              "      <td>2021-09-07</td>\n",
              "      <td>Proceedings of KDD 2021 Workshop on Data-drive...</td>\n",
              "      <td>None</td>\n",
              "      <td>Snehalkumar, S. Gaikwad, Shankar Iyer, Dalton ...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2109.00100v4</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>70</th>\n",
              "      <td>2109.00435v3</td>\n",
              "      <td>2021-09-01</td>\n",
              "      <td>2021-09-07</td>\n",
              "      <td>Proceedings of KDD 2020 Workshop on Data-drive...</td>\n",
              "      <td>None</td>\n",
              "      <td>Snehalkumar, S. Gaikwad, Shankar Iyer, Dalton ...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2109.00435v3</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>176</th>\n",
              "      <td>2007.09186v3</td>\n",
              "      <td>2020-07-17</td>\n",
              "      <td>2020-10-07</td>\n",
              "      <td>AWS CORD-19 Search: A Neural Search Engine for...</td>\n",
              "      <td>None</td>\n",
              "      <td>Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2007.09186v3</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-f50ee9c8-913e-426f-ad13-2b0c4b0ba7f0')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-f50ee9c8-913e-426f-ad13-2b0c4b0ba7f0 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-f50ee9c8-913e-426f-ad13-2b0c4b0ba7f0');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We will drop the three that have no PDF and handle separately the valid PDF which GROBID failed to process."
      ],
      "metadata": {
        "id": "q99DZOPt8YcI"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_301 = article_results_arxiv_304.copy()"
      ],
      "metadata": {
        "id": "9352Qnkk5COc"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_301.drop([3,69,70], axis=0, inplace=True)\n",
        "article_results_arxiv_301"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "BpKu24Be7yFp",
        "outputId": "28b3ad98-5685-4def-f418-4876028a8388"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         arxiv-id   published     revised  \\\n",
              "0    2109.06377v4  2021-09-14  2022-12-22   \n",
              "1    2212.09867v1  2022-12-19  2022-12-19   \n",
              "2    2212.09610v1  2022-12-19  2022-12-19   \n",
              "4    2212.03911v1  2022-12-07  2022-12-07   \n",
              "5    2212.01575v1  2022-12-03  2022-12-03   \n",
              "..            ...         ...         ...   \n",
              "299  2003.13665v1  2020-03-30  2020-03-30   \n",
              "300  2003.14258v1  2020-03-30  2020-03-30   \n",
              "301  2003.12454v1  2020-03-26  2020-03-26   \n",
              "302  2003.04524v1  2020-03-10  2020-03-10   \n",
              "303  2003.00751v1  2020-03-02  2020-03-02   \n",
              "\n",
              "                                                 title journal  \\\n",
              "0    ASGARD: A Single-cell Guided pipeline to Aid R...    None   \n",
              "1    Detecting Contradictory COVID-19 Drug Efficacy...    None   \n",
              "2    Drying of Bio-colloidal Sessile Droplets: Adva...    None   \n",
              "4    Analysis of Drug repurposing Knowledge graphs ...    None   \n",
              "5    Multi-view deep learning based molecule design...    None   \n",
              "..                                                 ...     ...   \n",
              "299  Genomics-guided molecular maps of coronavirus ...    None   \n",
              "300  Nanomechanical sonification of the 2019-nCoV c...    None   \n",
              "301  A Machine Learning alternative to placebo-cont...    None   \n",
              "302  Old Drugs for Newly Emerging Viral Disease, CO...    None   \n",
              "303  Regulation of interferon production as a poten...    None   \n",
              "\n",
              "                                               authors   doi  \\\n",
              "0    Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...  None   \n",
              "1    Daniel N. Sosa, Malavika Suresh, Christopher P...  None   \n",
              "2           Anusuya Pal, Amalesh Gope, Anupam Sengupta  None   \n",
              "4                                  Ajay Kumar Gogineni  None   \n",
              "5    Chao Pang, Yu Wang, Yi Jiang, Ruheng Wang, Ran...  None   \n",
              "..                                                 ...   ...   \n",
              "299                                    Gennadi Glinsky  None   \n",
              "300                                  Markus J. Buehler  None   \n",
              "301   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc  None   \n",
              "302                                Mohammad Reza Dayer  None   \n",
              "303             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei  None   \n",
              "\n",
              "                                      pdf_url  \\\n",
              "0    http://export.arxiv.org/pdf/2109.06377v4   \n",
              "1    http://export.arxiv.org/pdf/2212.09867v1   \n",
              "2    http://export.arxiv.org/pdf/2212.09610v1   \n",
              "4    http://export.arxiv.org/pdf/2212.03911v1   \n",
              "5    http://export.arxiv.org/pdf/2212.01575v1   \n",
              "..                                        ...   \n",
              "299  http://export.arxiv.org/pdf/2003.13665v1   \n",
              "300  http://export.arxiv.org/pdf/2003.14258v1   \n",
              "301  http://export.arxiv.org/pdf/2003.12454v1   \n",
              "302  http://export.arxiv.org/pdf/2003.04524v1   \n",
              "303  http://export.arxiv.org/pdf/2003.00751v1   \n",
              "\n",
              "                                                  text  \n",
              "0    \\nHeterogeneity, or more specifically, the div...  \n",
              "1    \\nThe COVID-19 pandemic caused by the novel SA...  \n",
              "2    \\nVirus Emulating Particles (VEP) White Blood ...  \n",
              "4    \\nIn the initial stages of a viral outbreak su...  \n",
              "5    \\nDe novo drug design is a time-consuming and ...  \n",
              "..                                                 ...  \n",
              "299  \\nCoronavirus pandemic 2020 caused by the newl...  \n",
              "300  \\nProteins are the building blocks of virtuall...  \n",
              "301  \\nCurrent and last decades research in drug di...  \n",
              "302  \\nThe outbreak of coronavirus  in 2019-2020 is...  \n",
              "303  \\nNovel Coronavirus SARS-CoV-2 is a single str...  \n",
              "\n",
              "[301 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-2efe703a-7d61-4eed-b67e-253962072b4c\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>arxiv-id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>2109.06377v4</td>\n",
              "      <td>2021-09-14</td>\n",
              "      <td>2022-12-22</td>\n",
              "      <td>ASGARD: A Single-cell Guided pipeline to Aid R...</td>\n",
              "      <td>None</td>\n",
              "      <td>Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2109.06377v4</td>\n",
              "      <td>\\nHeterogeneity, or more specifically, the div...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>2212.09867v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Detecting Contradictory COVID-19 Drug Efficacy...</td>\n",
              "      <td>None</td>\n",
              "      <td>Daniel N. Sosa, Malavika Suresh, Christopher P...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09867v1</td>\n",
              "      <td>\\nThe COVID-19 pandemic caused by the novel SA...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2212.09610v1</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Drying of Bio-colloidal Sessile Droplets: Adva...</td>\n",
              "      <td>None</td>\n",
              "      <td>Anusuya Pal, Amalesh Gope, Anupam Sengupta</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.09610v1</td>\n",
              "      <td>\\nVirus Emulating Particles (VEP) White Blood ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>2212.03911v1</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>2022-12-07</td>\n",
              "      <td>Analysis of Drug repurposing Knowledge graphs ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ajay Kumar Gogineni</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.03911v1</td>\n",
              "      <td>\\nIn the initial stages of a viral outbreak su...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>2212.01575v1</td>\n",
              "      <td>2022-12-03</td>\n",
              "      <td>2022-12-03</td>\n",
              "      <td>Multi-view deep learning based molecule design...</td>\n",
              "      <td>None</td>\n",
              "      <td>Chao Pang, Yu Wang, Yi Jiang, Ruheng Wang, Ran...</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2212.01575v1</td>\n",
              "      <td>\\nDe novo drug design is a time-consuming and ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>299</th>\n",
              "      <td>2003.13665v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Genomics-guided molecular maps of coronavirus ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Gennadi Glinsky</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.13665v1</td>\n",
              "      <td>\\nCoronavirus pandemic 2020 caused by the newl...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>300</th>\n",
              "      <td>2003.14258v1</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>2020-03-30</td>\n",
              "      <td>Nanomechanical sonification of the 2019-nCoV c...</td>\n",
              "      <td>None</td>\n",
              "      <td>Markus J. Buehler</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.14258v1</td>\n",
              "      <td>\\nProteins are the building blocks of virtuall...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>301</th>\n",
              "      <td>2003.12454v1</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>2020-03-26</td>\n",
              "      <td>A Machine Learning alternative to placebo-cont...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ezequiel Alvarez, Federico Lamagna, Manuel Szewc</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.12454v1</td>\n",
              "      <td>\\nCurrent and last decades research in drug di...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>302</th>\n",
              "      <td>2003.04524v1</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>2020-03-10</td>\n",
              "      <td>Old Drugs for Newly Emerging Viral Disease, CO...</td>\n",
              "      <td>None</td>\n",
              "      <td>Mohammad Reza Dayer</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.04524v1</td>\n",
              "      <td>\\nThe outbreak of coronavirus  in 2019-2020 is...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>303</th>\n",
              "      <td>2003.00751v1</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>2020-03-02</td>\n",
              "      <td>Regulation of interferon production as a poten...</td>\n",
              "      <td>None</td>\n",
              "      <td>Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2003.00751v1</td>\n",
              "      <td>\\nNovel Coronavirus SARS-CoV-2 is a single str...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>301 rows Ã— 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-2efe703a-7d61-4eed-b67e-253962072b4c')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-2efe703a-7d61-4eed-b67e-253962072b4c button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-2efe703a-7d61-4eed-b67e-253962072b4c');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(article_results_arxiv_301)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TQef-s0X94Sv",
        "outputId": "c9baece4-1166-4517-af7c-c52df764b0d4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "301"
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_301.reset_index(drop=True, inplace=True)"
      ],
      "metadata": {
        "id": "Apzpcjcu99AS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check that the articles have been dropped and the reindexing has worked by seeing if the articles that were previously in rows 69 and 70 have gone."
      ],
      "metadata": {
        "id": "_3vr7XkK9Zjb"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "article_results_arxiv_301.loc[68:71]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 551
        },
        "id": "RyFyuDu2-eDK",
        "outputId": "c6d9e2bb-49d7-42cc-be79-c3ce600dd6a4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "        arxiv-id   published     revised  \\\n",
              "68  2010.16413v3  2020-10-09  2021-09-05   \n",
              "69  2103.02843v2  2021-03-04  2021-09-04   \n",
              "70  2108.13764v1  2021-08-31  2021-08-31   \n",
              "71  2108.12150v1  2021-08-27  2021-08-27   \n",
              "\n",
              "                                                title  \\\n",
              "68  Artificial Intelligence (AI) in Action: Addres...   \n",
              "69  Pandemic Drugs at Pandemic Speed: Infrastructu...   \n",
              "70  Virtual screening of Microalgal compounds as p...   \n",
              "71  A Nested Multi-Scale Model for COVID-19 Viral ...   \n",
              "\n",
              "                                              journal  \\\n",
              "68  Annual Review of Biomedical Data Science 4 (2021)   \n",
              "69            Interface Focus. 2021. 11 (6): 20210018   \n",
              "70                                               None   \n",
              "71                                               None   \n",
              "\n",
              "                                              authors  \\\n",
              "68  Qingyu Chen, Robert Leaman, Alexis Allot, Ling...   \n",
              "69  Agastya P. Bhati, Shunzhou Wan, Dario AlfÃ¨, Au...   \n",
              "70                                   Ibrahim Mohammed   \n",
              "71     Bishal Chhetri, D. K. K Vamsi, Carani Sanjeevi   \n",
              "\n",
              "                                         doi  \\\n",
              "68  10.1146/annurev-biodatasci-021821-061045   \n",
              "69                    10.1098/rsfs.2021.0018   \n",
              "70                                      None   \n",
              "71                                      None   \n",
              "\n",
              "                                     pdf_url  \\\n",
              "68  http://export.arxiv.org/pdf/2010.16413v3   \n",
              "69  http://export.arxiv.org/pdf/2103.02843v2   \n",
              "70  http://export.arxiv.org/pdf/2108.13764v1   \n",
              "71  http://export.arxiv.org/pdf/2108.12150v1   \n",
              "\n",
              "                                                 text  \n",
              "68  \\nSince the initial reports of an outbreak of ...  \n",
              "69  \\ndiscovery lie at the interface between machi...  \n",
              "70  \\nCorona virus disease-19  is caused by Severe...  \n",
              "71  \\nCOVID -19 is a contagious respiratory and va...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-7682770b-e3dd-4eea-93fe-c6d9c1d0bb23\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>arxiv-id</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>68</th>\n",
              "      <td>2010.16413v3</td>\n",
              "      <td>2020-10-09</td>\n",
              "      <td>2021-09-05</td>\n",
              "      <td>Artificial Intelligence (AI) in Action: Addres...</td>\n",
              "      <td>Annual Review of Biomedical Data Science 4 (2021)</td>\n",
              "      <td>Qingyu Chen, Robert Leaman, Alexis Allot, Ling...</td>\n",
              "      <td>10.1146/annurev-biodatasci-021821-061045</td>\n",
              "      <td>http://export.arxiv.org/pdf/2010.16413v3</td>\n",
              "      <td>\\nSince the initial reports of an outbreak of ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>69</th>\n",
              "      <td>2103.02843v2</td>\n",
              "      <td>2021-03-04</td>\n",
              "      <td>2021-09-04</td>\n",
              "      <td>Pandemic Drugs at Pandemic Speed: Infrastructu...</td>\n",
              "      <td>Interface Focus. 2021. 11 (6): 20210018</td>\n",
              "      <td>Agastya P. Bhati, Shunzhou Wan, Dario AlfÃ¨, Au...</td>\n",
              "      <td>10.1098/rsfs.2021.0018</td>\n",
              "      <td>http://export.arxiv.org/pdf/2103.02843v2</td>\n",
              "      <td>\\ndiscovery lie at the interface between machi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>70</th>\n",
              "      <td>2108.13764v1</td>\n",
              "      <td>2021-08-31</td>\n",
              "      <td>2021-08-31</td>\n",
              "      <td>Virtual screening of Microalgal compounds as p...</td>\n",
              "      <td>None</td>\n",
              "      <td>Ibrahim Mohammed</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2108.13764v1</td>\n",
              "      <td>\\nCorona virus disease-19  is caused by Severe...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>71</th>\n",
              "      <td>2108.12150v1</td>\n",
              "      <td>2021-08-27</td>\n",
              "      <td>2021-08-27</td>\n",
              "      <td>A Nested Multi-Scale Model for COVID-19 Viral ...</td>\n",
              "      <td>None</td>\n",
              "      <td>Bishal Chhetri, D. K. K Vamsi, Carani Sanjeevi</td>\n",
              "      <td>None</td>\n",
              "      <td>http://export.arxiv.org/pdf/2108.12150v1</td>\n",
              "      <td>\\nCOVID -19 is a contagious respiratory and va...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7682770b-e3dd-4eea-93fe-c6d9c1d0bb23')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-7682770b-e3dd-4eea-93fe-c6d9c1d0bb23 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-7682770b-e3dd-4eea-93fe-c6d9c1d0bb23');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 34
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_article_results_arxiv_301_full_text.pickle', 'wb') as f:\n",
        "  pickle.dump(article_results_arxiv_301, f)"
      ],
      "metadata": {
        "id": "1u-pf5CqDRvj"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### References\n",
        "\n",
        "* GROBID https://github.com/kermitt2/grobid\n",
        "\n",
        "* GROBID documentation https://grobid.readthedocs.io/\n",
        "\n",
        "\n",
        "* Python PDF parser for scientific publications: content and figures https://github.com/titipata/scipdf_parser\n"
      ],
      "metadata": {
        "id": "aXYPOc1ntoKG"
      }
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}